<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05829</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05829 septum site-determining protein MinD [Phormidium_A]</h3>
    <div>MPQVYISLYCSPVWIVPKPFLIMSRIIVVTSGKGGVGKTTTTANLGMALAQRKRKVILVDADFGLRNLDLLLGLENRVVYTAVEAIAGECRLDQAIVRDKRQNTLALLPAAQNRMKDAVTPAQMRQLVMALAKKFDYVLVDCPAGIEQGFQNAIAAAREALIVTTPEIAAVRDADRVVGLLEANGVRKVGLIINRIRPEMVEANDMMSVEDVQELLAVPLVGVIPDDERVIVSTNRGEPLVLSEEPSRAGQAYGNIARRLEGETVEFLDLSVPPKGLFSRLRTLLNTRII</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>TPA: septum site-determining protein MinD</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>septum site-determining protein MinD</td>
            <td id=cr>FACHB-46</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>65</td>
            <td id=cg>88</td>
            <td id=cg>98</td>
            <td id=al>TIGR01968 septum site-determining protein MinD. This model describes the bacterial and chloroplast form of MinD, a multifunctional cell division protein that guides correct placement of the septum. The homologous archaeal MinD proteins, with many archaeal genomes having two or more forms, are described by a separate model. [Cellular processes, Cell division]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=al>COG2894 Septum formation inhibitor-activating ATPase MinD [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>90</td>
            <td id=cg>93</td>
            <td id=al>CHL00175 septum-site determining protein; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=al>cd02036 septum site-determining protein MinD. Septum site-determining protein MinD is part of the operon MinCDE that determines the site of the formation of a septum at mid-cell, an important part of bacterial cell division. MinC is a nonspecific inhibitor of the septum protein FtsZ. MinE is the supressor of MinC. MinD plays a pivotal role, selecting the mid-cell over other sites through the activation and regulation of MinC and MinE. MinD is a membrane-associated ATPase, related to nitrogenase iron protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=al>PRK10818 septum site-determining protein MinD. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>84</td>
            <td id=cg>97</td>
            <td id=al>TIGR01969 cell division ATPase MinD, archaeal. This model represents the archaeal branch of the MinD family. MinD, a weak ATPase, works in bacteria with MinC as a generalized cell division inhibitor and, through interaction with MinE, prevents septum placement inappropriate sites. Often several members of this family are found in archaeal genomes, and the function is uncharacterized. More distantly related proteins ParA chromosome partitioning proteins. The exact roles of the various archaeal MinD homologs are unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>85</td>
            <td id=cg>94</td>
            <td id=al>COG0455 MinD-like ATPase involved in chromosome partitioning or flagellar assembly [Cell cycle control, cell division, chromosome partitioning, Cell motility]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cb>75</td>
            <td id=cg>98</td>
            <td id=al>pfam01656 CobQ/CobB/MinD/ParA nucleotide binding domain. This family consists of various cobyrinic acid a,c-diamide synthases. These include CbiA and CbiP from S.typhimurium, and CobQ from R. capsulatus. These amidases catalyze amidations to various side chains of hydrogenobyrinic acid or cobyrinic acid a,c-diamide in the biosynthesis of cobalamin (vitamin B12) from uroporphyrinogen III. Vitamin B12 is an important cofactor and an essential nutrient for many plants and animals and is primarily produced by bacteria. The family also contains dethiobiotin synthetases as well as the plasmid partitioning proteins of the MinD/ParA family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=al>cd03111 pilus assembly ATPase CpaE. This protein family consists of proteins similar to the cpaE protein of the Caulobacter pilus assembly and the orf4 protein of Actinobacillus pilus formation gene cluster. The function of these proteins are unkown. The Caulobacter pilus assembly contains 7 genes: pilA, cpaA, cpaB, cpaC, cpaD, cpaE and cpaF. These genes are clustered together on chromosome.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=al>cd02038 MinD-like ATPase FlhG. FlhG is a member of the SIMIBI superfamily. FlhG (also known as YlxH) is a major determinant for a variety of flagellation patterns. It effects location and number of bacterial flagella during C-ring assembly.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>COG1192 Cellulose biosynthesis protein BcsQ [Cell motility]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cb>64</td>
            <td id=cb>70</td>
            <td id=al>COG0489 Chromosome partitioning ATPase, Mrp family, contains Fe-S cluster [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pfam10609 NUBPL iron-transfer P-loop NTPase. This family contains ATPases involved in plasmid partitioning. It also contains the cytosolic Fe-S cluster assembling factor NBP35 which is required for biogenesis and export of both ribosomal subunits.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cb>62</td>
            <td id=cg>96</td>
            <td id=al>cd05387 bacterial tyrosine-kinase. Bacterial tyrosine (BY)-kinases catalyze the autophosphorylation on a C-terminal tyrosine cluster and also phosphorylate endogenous protein substrates by using ATP as phosphoryl donor. Besides their capacity to function as tyrosine kinase, most of these proteins are also involved in the production and transport of exopolysaccharides. BY-kinases are involved in a number of physiological processes ranging from stress resistance to pathogenicity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=al>pfam13614 AAA domain. This family includes a wide variety of AAA domains including some that have lost essential nucleotide binding residues in the P-loop.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=al>cd02042 partition proteins ParAB family. ParA and ParB of Caulobacter crescentus belong to a conserved family of bacterial proteins implicated in chromosome segregation. ParB binds to DNA sequences adjacent to the origin of replication and localizes to opposite cell poles shortly following the initiation of DNA replication. ParB regulates the ParA ATPase activity by promoting nucleotide exchange in a fashion reminiscent of the exchange factors of eukaryotic G proteins. ADP-bound ParA binds single-stranded DNA, whereas the ATP-bound form dissociates ParB from its DNA binding sites. Increasing the fraction of ParA-ADP in the cell inhibits cell division, suggesting that this simple nucleotide switch may regulate cytokinesis. ParA shares sequence similarity to a conserved and widespread family of ATPases which includes the repA protein of the repABC operon in Rhizobium etli symbiotic plasmid. This operon is involved in the plasmid replication and partition.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG1149 MinD superfamily P-loop ATPase, contains an inserted ferredoxin domain  [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=al>cd02037 Mrp/NBP35 ATP-binding protein family. Mrp/NBP35 ATP-binding family protein are typically iron-sulfur (FeS) cluster scaffolds that function to assemble nascent FeS clusters for transfer to FeS-requiring enzymes. Members include the eukaryotic nucleotide-binding protein 1 (NUBP1) which is a component of the cytosolic iron-sulfur (Fe/S) protein assembly (CIA) machinery and the archael [NiFe] hydrogenase maturation protein HypB which is required for nickel insertion into [NiFe] hydrogenase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>94</td>
            <td id=cb>74</td>
            <td id=al>COG4963 Flp pilus assembly protein, ATPase CpaE  [Intracellular trafficking, secretion, and vesicular transport, Extracellular structures]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=al>cd02034 accessory protein CooC1. The accessory protein CooC1, a nickel-binding ATPase, participates in the incorporation of nickel into the complex active site ([Ni-4Fe-4S]) cluster of Ni,Fe-dependent carbon monoxide dehydrogenase (CODH). CODH from Rhodospirillum rubrum catalyzes the reversible oxidation of CO to CO2. CODH contains a nickel-iron-sulfur cluster (C-center) and an iron-sulfur cluster (B-center). CO oxidation occurs at the C-center. Three accessory proteins encoded by cooCTJ genes are involved in nickel incorporation into a nickel site. CooC functions as a nickel insertase that mobilizes nickel to apoCODH using energy released from ATP hydrolysis. CooC is a homodimer and has NTPase activities. Mutation at the P-loop abolishs its function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>COG3640 CO dehydrogenase nickel-insertion accessory protein CooC1 [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>cd03110 bacterial and archaeal subfamily of SIMIBI. Uncharacterized bacterial and archaeal subfamily of SIMIBI superfamily. Proteins in this superfamily contain an ATP-binding domain and use energy from hydrolysis of ATP to transfer electron or ion. The specific function of this family is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>81</td>
            <td id=cg>96</td>
            <td id=al>TIGR03371 cellulose synthase operon protein YhjQ. Members of this family are the YhjQ protein, found immediately upsteam of bacterial cellulose synthase (bcs) genes in a broad range of bacteria, including both copies of the bcs locus in Klebsiella pneumoniae. In several species it is seen clearly as part of the bcs operon. It is identified as a probable component of the bacterial cellulose metabolic process not only by gene location, but also by partial phylogenetic profiling, or Haft-Selengut algorithm (), based on a bacterial cellulose biosynthesis genome property profile. Cellulose plays an important role in biofilm formation and structural integrity in some bacteria. Mutants in yhjQ in Escherichia coli, show altered morphology an growth, but the function of YhjQ has not yet been determined. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>80</td>
            <td id=cg>93</td>
            <td id=al>cd02035 Arsenical pump-driving ATPase ArsA. ArsA ATPase functions as an efflux pump located on the inner membrane of the cell. This ATP-driven oxyanion pump catalyzes the extrusion of arsenite, antimonite and arsenate. Maintenance of a low intracellular concentration of oxyanion produces resistance to the toxic agents. The pump is composed of two subunits, the catalytic ArsA subunit and the membrane subunit ArsB, which are encoded by arsA and arsB genes, respectively. Arsenic efflux in bacteria is catalyzed by either ArsB alone or by ArsAB complex. The ATP-coupled pump, however, is more efficient. ArsA is composed of two homologous halves, A1 and A2, connected by a short linker sequence.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>62</td>
            <td id=cg>88</td>
            <td id=al>TIGR01007 capsular exopolysaccharide family. This model describes the capsular exopolysaccharide proteins in bacteria. The exopolysaccharide gene cluster consists of several genes which encode a number of proteins which regulate the exoploysaccharide biosynthesis(EPS). Atleast 13 genes espA to espM in streptococcus species seem to direct the EPS proteins and all of which share high homology. Functional roles were characterized by gene disruption experiments which resulted in exopolysaccharide-deficient phenotypes. [Transport and binding proteins, Carbohydrates, organic alcohols, and acids]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=al>cd01983 SIMIBI (signal recognition particle, MinD and BioD)-class NTPases. SIMIBI (after signal recognition particle, MinD, and BioD), consists of signal recognition particle (SRP) GTPases, the assemblage of MinD-like ATPases, which are involved in protein localization, chromosome partitioning, and membrane transport, and a group of metabolic enzymes with kinase or related phosphate transferase activity. Functionally, proteins in this superfamily use the energy from hydrolysis of NTP to transfer electron or ion.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>90</td>
            <td id=cb>71</td>
            <td id=al>PRK11670 iron-sulfur cluster carrier protein ApbC. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>62</td>
            <td id=cr>23</td>
            <td id=al>TIGR01005 exopolysaccharide transport protein family. The model describes the exopolysaccharide transport protein family in bacteria. The transport protein is part of a large genetic locus which is associated with exopolysaccharide (EPS) biosynthesis. Detailed molecular characterization and gene fusion analysis revealed atleast seven gene products are involved in the overall regulation, which among other things, include exopolysaccharide biosynthesis, property of conferring virulence and exopolysaccharide export. [Transport and binding proteins, Carbohydrates, organic alcohols, and acids]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>64</td>
            <td id=cg>80</td>
            <td id=al>pfam06564 Cellulose biosynthesis protein BcsQ. This is a family of bacterial proteins involved in cellulose biosynthesis. (Roemling U. and Galperin M.Y. "Bacterial cellulose biosynthesis. Diversity of operons and subunits" (manuscript in preparation)). A second component of the extracellular matrix of the multicellular morphotype (rdar) of Salmonella typhimurium and Escherichia coli is cellulose. The family does contain a P-loop sequence motif suggesting a nucleotide binding function, but this has not been confirmed.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4771620</td></tr>
<tr><td id=al>end</td><td id=al>4772493</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>septum site-determining protein MinD</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>291</td></tr>
</table>

    </div>
</body>
</html>
