<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02896</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02896 ribonuclease HI [Phormidium_A]</h3>
    <div>MTMKSVTIYTDGACSGNPGPGGYGTVLIYRDFRKELSGGFRRTTNNRMEMMAAIVGLKLLKQACQVTLYSDSKYIVDAMRQGWAKRWKAKGWWRNKKERAKNPDLWEELLQLSDFHQVEFVWVKGHAGNKENECCDRLAVAASKQSDLPADEVFERGDA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease HI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>95</td>
            <td id=cg>91</td>
            <td id=al>ribonuclease HI</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>95</td>
            <td id=cg>93</td>
            <td id=al>ribonuclease HI</td>
            <td id=co>Rippkaea</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>98</td>
            <td id=cg>96</td>
            <td id=al>ribonuclease HI</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>96</td>
            <td id=cg>96</td>
            <td id=al>ribonuclease HI</td>
            <td id=co>Rippkaea</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>52</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>cd09278 RNase HI family found mainly in prokaryotes. Ribonuclease H (RNase H) is classified into two evolutionarily unrelated families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is involved in DNA replication, repair and transcription. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD), residues and have the same catalytic mechanism and functions in cells. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. Prokaryotic RNase H varies greatly in domain structures and substrate specificities. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>87</td>
            <td id=cg>93</td>
            <td id=al>PRK00203 ribonuclease H; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=al>COG0328 Ribonuclease HI [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>pfam00075 RNase H. RNase H digests the RNA strand of an RNA/DNA hybrid. Important enzyme in retroviral replication cycle, and often found as a domain associated with reverse transcriptases. Structure is a mixed alpha+beta fold with three a/b/a layers.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>cd09280 Eukaryotic RNase H is essential and is longer and more complex than their prokaryotic counterparts. Ribonuclease H (RNase H) is classified into two families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is involved in DNA replication, repair and transcription. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. RNase H is widely present in various organisms, including bacteria, archaea and eukaryote and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD) residues and have the same catalytic mechanism and functions in cells. Eukaryotic RNase H is longer and more complex than in prokaryotes. Almost all eukaryotic RNase HI have highly conserved regions at their N-termini called hybrid binding domain (HBD). It is speculated that the HBD contributes to binding the RNA/DNA hybrid. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability, but RNase H is essential in higher eukaryotes. RNase H knockout mice lack mitochondrial DNA replication and die as embryos.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=al>cd06222 Ribonuclease H-like superfamily, including RNase H, HI, HII, HIII, and RNase-like domain IV of spliceosomal protein Prp8. Ribonuclease H (RNase H) enzymes are divided into two major families, Type 1 and Type 2, based on amino acid sequence similarities and biochemical properties. RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner in the presence of divalent cations. It is widely present in various organisms, including bacteria, archaea, and eukaryotes. Most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site residues and have the same catalytic mechanism and functions in cells. RNase H is involved in DNA replication, repair and transcription. An important RNase H function is to remove Okazaki fragments during DNA replication. RNase H inhibitors have been explored as anti-HIV drug targets since RNase H inactivation inhibits reverse transcription. This model also includes the Prp8 domain IV, which adopts the RNase fold but shows low sequence homology; domain IV is implicated in key spliceosomal interactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>cd09279 RNAse HI family that includes archaeal, some bacterial as well as plant RNase HI. Ribonuclease H (RNase H) is classified into two evolutionarily unrelated families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is involved in DNA replication, repair and transcription. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD) residues and have the same catalytic mechanism and functions in cells. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. Most archaeal genomes contain only type 2 RNase H (RNase HII); however, a few contain RNase HI as well. Although archaeal RNase HI sequences conserve the DEDD active-site motif, they lack other common features important for catalytic function, such as the basic protrusion region. Archaeal RNase HI homologs are more closely related to retroviral RNase HI than bacterial and eukaryotic type I RNase H in enzymatic properties.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>cd13934 Fungal (dikarya) Ribonuclease H, uncharacterized. This family contains dikarya RNase H, many of which are uncharacterized. Ribonuclease H (RNase H) enzymes are divided into two major families, Type 1 and Type 2, based on amino acid sequence similarities and biochemical properties. RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner in the presence of divalent cations. It is widely present in various organisms, including bacteria, archaea and eukaryotes. Most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site residues and have the same catalytic mechanism and functions in cells. RNase H is involved in DNA replication, repair and transcription. An important RNase H function is to remove Okazaki fragments during DNA replication.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=al>PRK08719 ribonuclease H; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>90</td>
            <td id=cg>89</td>
            <td id=al>PRK06548 ribonuclease H; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>cd09277 Bacterial RNase HI containing a hybrid binding domain (HBD) at the N-terminus. Ribonuclease H (RNase H) enzymes are divided into two major families, Type 1 and Type 2, based on amino acid sequence similarities and biochemical properties. RNase H is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner in the presence of divalent cations. RNase H is involved in DNA replication, repair and transcription. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes and most prokaryotic and eukaryotic genomes contain multiple RNase H genes. Despite the lack of amino acid sequence homology, Type 1 and type 2 RNase H share a main-chain fold and steric configurations of the four acidic active-site (DEDD) residues and have the same catalytic mechanism and functions in cells. One of the important functions of RNase H is to remove Okazaki fragments during DNA replication. Prokaryotic RNase H varies greatly in domain structures and substrate specificities. Prokaryotes and some single-cell eukaryotes do not require RNase H for viability. Some bacteria distinguished from other bacterial RNase HI in the presence of a hybrid binding domain (HBD) at the N-terminus which is commonly present at the N-termini of eukaryotic RNase HI. It has been reported that this domain is required for dimerization and processivity of RNase HI upon binding to RNA-DNA hybrids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>cd09276 non-LTR RNase HI domain of reverse transcriptases. Ribonuclease H (RNase H) is classified into two families, type 1 (prokaryotic RNase HI, eukaryotic RNase H1 and viral RNase H) and type 2 (prokaryotic RNase HII and HIII, and eukaryotic RNase H2). Ribonuclease HI (RNase HI) is an endonuclease that cleaves the RNA strand of an RNA/DNA hybrid in a sequence non-specific manner. RNase H is widely present in various organisms, including bacteria, archaea and eukaryotes. RNase HI has also been observed as an adjunct domain to the reverse transcriptase gene in retroviruses, long-term repeat (LTR)-bearing retrotransposons and non-LTR retrotransposons. RNase HI in LTR retrotransposons perform degradation of the original RNA template, generation of a polypurine tract (the primer for plus-strand DNA synthesis), and final removal of RNA primers from newly synthesized minus and plus strands. The catalytic residues for RNase H enzymatic activity, three aspartatic acids and one glutamic acid residue (DEDD), are unvaried across all RNase H domains. The position of the RNase domain of non-LTR and LTR transposons is at the carboxyl terminal of the reverse transcriptase (RT) domain and their RNase domains group together, indicating a common evolutionary origin. Many non-LTR transposons have lost the RNase domain because their activity is at the nucleus and cellular RNase may suffice; however LTR retrotransposons always encode their own RNase domain because it requires RNase activity in RNA-protein particles in the cytoplasm. RNase H inhibitors have been explored as an anti-HIV drug target because RNase H inactivation inhibits reverse transcription.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>43</td>
            <td id=al>PRK07238 bifunctional RNase H/acid phosphatase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2422179</td></tr>
<tr><td id=al>end</td><td id=al>2422659</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>ribonuclease HI</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>160</td></tr>
</table>

    </div>
</body>
</html>
