<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04542</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04542 adenylosuccinate lyase, partial [Phormidium_A]</h3>
    <div>MSIPRYTLPEMGDLWTDRYKYQTWLQVEIAVCEAQAELGYIPAEAVEEIKAKANFEPERILEIEAEVRHDVIAFLTNVNEYVGDAGRYIHLGMTSSDMLDTALALQLVASTNVLLQRLEDTIQAIRYQAQQHRHTVMVGRSHGIHAEPITFGFKLAGWLAEMLRHRDRLVAVRQAVAVGKISGAVGTYANLDPRIEALACQKLGLEPDMASTQVISRDRHAEFVQALALVAASLERFSVEIRNLQRTDVLEVEEYFSKGQKGSSAMPHKRNPIRSERLSGLARVIRGNAVAALENVALWHERDISHSSVERMILPDSCTVLHFMLHETTELVKHLLVYPENMLRNMNLYGGVIFSQRVLLTLVGKGVSREDAYAIVQSCAHSAWNQPNGDFRKLISQDPRVAQHLTPDEVDECFDPQYHLRNLDQVYQRLNI</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase PurB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TPA: adenylosuccinate lyase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=co>Phormidium_B</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=co>Phormidium_B</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TPA: adenylosuccinate lyase</td>
            <td id=co>UBA11371</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>adenylosuccinate lyase</td>
            <td id=co>Crinalium</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>78</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>PRK07380 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>99</td>
            <td id=cg>98</td>
            <td id=al>PRK07492 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>86</td>
            <td id=cg>96</td>
            <td id=al>cd01360 Adenylsuccinate lyase (ASL)_subgroup 1. This subgroup contains bacterial and archeal proteins similar to ASL, a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and, the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TIGR00928 adenylosuccinate lyase. This family consists of adenylosuccinate lyase, the enzyme that catalyzes step 8 in the purine biosynthesis pathway for de novo synthesis of IMP and also the final reaction in the two-step sequence from IMP to AMP. [Purines, pyrimidines, nucleosides, and nucleotides, Purine ribonucleotide biosynthesis]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>COG0015 Adenylosuccinate lyase [Nucleotide transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>86</td>
            <td id=cg>98</td>
            <td id=al>cd01595 Adenylsuccinate lyase (ASL)_like. This group contains ASL, prokaryotic-type 3-carboxy-cis,cis-muconate cycloisomerase (pCMLE), and related proteins. These proteins are members of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and; the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP). pCMLE catalyzes the cyclization of 3-carboxy-cis,cis-muconate (3CM) to 4-carboxy-muconolactone, in the beta-ketoadipate pathway. ASL deficiency has been linked to several pathologies including psychomotor retardation with autistic features, epilepsy and muscle wasting.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>PRK08470 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>96</td>
            <td id=cg>93</td>
            <td id=al>PRK08540 adenylosuccinate lyase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>97</td>
            <td id=cg>96</td>
            <td id=al>cd01597 prokaryotic 3-carboxy-cis,cis-muconate cycloisomerase (CMLE)_like. This subgroup contains pCLME and related proteins, and belongs to the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. CMLE catalyzes the cyclization of 3-carboxy-cis,cis-muconate (3CM) to 4-carboxy-muconolactone in the beta-ketoadipate pathway. This pathway is responsible for the catabolism of a variety of aromatic compounds into intermediates of the citric cycle in prokaryotic and eukaryotic micro-organisms.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=al>cd01334 Lyase class I family; a group of proteins which catalyze similar beta-elimination reactions. The Lyase class I family contains class II fumarase, aspartase, adenylosuccinate lyase (ASL), argininosuccinate lyase (ASAL), prokaryotic-type 3-carboxy-cis,cis-muconate cycloisomerase (pCMLE), and related proteins. It belongs to the Lyase_I superfamily. Proteins of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>96</td>
            <td id=cg>92</td>
            <td id=al>PRK06390 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=co>51</td>
            <td id=cg>100</td>
            <td id=al>PRK08937 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>99</td>
            <td id=cg>95</td>
            <td id=al>PRK09053 3-carboxy-cis,cis-muconate cycloisomerase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cb>77</td>
            <td id=cg>98</td>
            <td id=al>TIGR02426 3-carboxy-cis,cis-muconate cycloisomerase. Members of this family are 3-carboxy-cis,cis-muconate cycloisomerase, the enzyme the catalyzes the second step in the protocatechuate degradation to beta-ketoadipate and then to succinyl-CoA and acetyl-CoA. 4-hydroxybenzoate, 3-hydroxybenzoate, and vanillate all can be converted in one step to protocatechuate. All members of the seed alignment for this model were chosen from within protocatechuate degradation operons of at least three genes of the pathway, from genomes with the complete pathway through beta-ketoadipate. [Energy metabolism, Other]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>cd03302 Adenylsuccinate lyase (ASL)_subgroup 2. This subgroup contains mainly eukaryotic proteins similar to ASL, a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two steps in the de novo purine biosynthesis: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and, the conversion of adenylsuccinate (SAMP) into adenosine monophosphate (AMP). ASL deficiency has been linked to several pathologies including psychomotor retardation with autistic features, epilepsy and muscle wasting.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cb>71</td>
            <td id=cg>99</td>
            <td id=al>pfam00206 Lyase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=al>cd01594 Lyase class I_like superfamily: contains the lyase class I family, histidine ammonia-lyase and phenylalanine ammonia-lyase, which catalyze similar beta-elimination reactions. Lyase class I_like superfamily of enzymes that catalyze beta-elimination reactions and are active as homotetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. This superfamily contains the lyase class I family, histidine ammonia-lyase and phenylalanine ammonia-lyase. The lyase class I family comprises proteins similar to class II fumarase, aspartase, adenylosuccinate lyase, argininosuccinate lyase, and 3-carboxy-cis, cis-muconate lactonizing enzyme which, for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. Histidine or phenylalanine ammonia-lyase catalyze a beta-elimination of ammonia from histidine and phenylalanine, respectively.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cb>71</td>
            <td id=cg>87</td>
            <td id=al>PRK05975 3-carboxy-cis,cis-muconate cycloisomerase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=al>smart00998 Adenylosuccinate lyase C-terminus. Adenylosuccinate lyase catalyses two steps in the synthesis of purine nucleotides: the conversion of succinylaminoimidazole-carboxamide ribotide into aminoimidazole-carboxamide ribotide (the fifth step of de novo IMP biosynthesis); the formation of adenosine monophosphate (AMP) from adenylosuccinate (the final step in the synthesis of AMP from IMP). This entry represents the C-terminal, seven alpha-helical, domain of adenylosuccinate lyase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cb>76</td>
            <td id=cb>77</td>
            <td id=al>cd01598 PurB_like adenylosuccinases (adenylosuccinate lyase, ASL). This subgroup contains EcASL, the product of the purB gene in Escherichia coli, and related proteins. It is a member of the Lyase class I family of the Lyase_I superfamily. Members of the Lyase class I family function as homotetramers to catalyze similar beta-elimination reactions in which a Calpha-N or Calpha-O bond is cleaved with the subsequent release of fumarate as one of the products. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASL catalyzes two non-sequential steps in the de novo purine biosynthesis pathway: the conversion of 5-aminoimidazole-(N-succinylocarboxamide) ribotide (SAICAR) into 5-aminoimidazole-4-carboxamide ribotide (AICAR) and; the conversion of adenylosuccinate (SAMP) into adenosine monophosphate (AMP).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>95</td>
            <td id=cg>95</td>
            <td id=al>cd01359 Argininosuccinate lyase (argininosuccinase, ASAL). This group contains ASAL and related proteins. It is a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. ASAL is a cytosolic enzyme which catalyzes the reversible breakdown of argininosuccinate to arginine and fumarate during arginine biosynthesis. In ureotelic species ASAL also catalyzes a reaction involved in the production of urea. Included in this group are the major soluble avian eye lens proteins from duck, delta 1 and delta 2 crystallin. Of these two isoforms only delta 2 has retained ASAL activity. These crystallins may have evolved by, gene recruitment of ASAL followed by gene duplication. In humans, mutations in ASAL result in the autosomal recessive disorder argininosuccinic aciduria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>76</td>
            <td id=cb>73</td>
            <td id=al>PRK09285 adenylosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=al>pfam10397 Adenylosuccinate lyase C-terminus. This is the C-terminal seven alpha helices of the structure whose full length represents the enzyme adenylosuccinate lyase. This sequence lies C-terminal to the conserved motif necessary for beta-elimination reactions, Adenylosuccinate lyase catalyzes two steps in the synthesis of purine nucleotides: the conversion of succinylaminoimidazole-carboxamide ribotide into aminoimidazole-carboxamide ribotide, the eighth step of the de novo pathway, and the formation of adenosine monophosphate (AMP) from adenylosuccinate, the second step in the conversion of inosine monophosphate into AMP.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>93</td>
            <td id=cg>88</td>
            <td id=al>COG0165 Argininosuccinate lyase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>90</td>
            <td id=cg>85</td>
            <td id=al>PRK00855 argininosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cb>67</td>
            <td id=cb>64</td>
            <td id=al>PLN02848 adenylosuccinate lyase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>67</td>
            <td id=cb>65</td>
            <td id=al>cd01357 Aspartase. This subgroup contains Escherichia coli aspartase (L-aspartate ammonia-lyase), Bacillus aspartase and related proteins. It is a member of the Lyase class I family, which includes both aspartase (L-aspartate ammonia-lyase) and fumarase class II enzymes. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. Aspartase catalyzes the reversible deamination of aspartic acid.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>71</td>
            <td id=cb>68</td>
            <td id=al>cd01596 aspartase (L-aspartate ammonia-lyase) and fumarase class II enzymes. This group contains aspartase (L-aspartate ammonia-lyase), fumarase class II enzymes, and related proteins. It is a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. Aspartase catalyzes the reversible deamination of aspartic acid. Fumarase catalyzes the reversible hydration/dehydration of fumarate to L-malate during the Krebs cycle.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>57</td>
            <td id=co>54</td>
            <td id=al>cd01362 Class II fumarases. This subgroup contains Escherichia coli fumarase C, human mitochondrial fumarase, and related proteins.  It is a member of the Lyase class I family. Members of this family for the most part catalyze similar beta-elimination reactions in which a C-N or C-O bond is cleaved with the release of fumarate as one of the products. These proteins are active as tetramers. The four active sites of the homotetrameric enzyme are each formed by residues from three different subunits. Fumarase catalyzes the reversible hydration/dehydration of fumarate to L-malate during the Krebs cycle.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>55</td>
            <td id=co>52</td>
            <td id=al>PRK04833 argininosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>70</td>
            <td id=cb>65</td>
            <td id=al>COG0114 Fumarate hydratase class II [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>50</td>
            <td id=cr>24</td>
            <td id=al>PRK02186 argininosuccinate lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>73</td>
            <td id=cb>66</td>
            <td id=al>PLN02646 argininosuccinate lyase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>59</td>
            <td id=co>42</td>
            <td id=al>PRK12308 argininosuccinate lyase. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3707045</td></tr>
<tr><td id=al>end</td><td id=al>3708344</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>adenylosuccinate lyase, partial</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>433</td></tr>
</table>

    </div>
</body>
</html>
