<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05654</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05654 tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB, partial [Phormidium_A]</h3>
    <div>MTLSPRRYHITTFGCQMNKADSERMAGILETLNFQATEDPNLADLVLYNTCTIRDNAEQKVYSYLGRQAKRKQNDPNLILAVAGCVAQQEGEALMRRVPELDLVMGPQHANRLGDLLEQVFSGNQVVATEPVHIMEDITKPRRDSAVSAWVNVIYGCNERCTYCVVPNVRGVEQSRTPEAIRAEMEELGRQGYKEVTLLGQNIDAYGRDLPGSTPEGRHLHTLTDLLYFVHDVPGIERIRFATSHPRYFTERLIRACQELPKVCEHFHIPFQSGDNEILKAMARGYTHEKYRRIIDQIRHYMPDAAISADAIVGFPGETEEQFQNTLDLTADIGFDILNTAAYSPRPGTPAAHWENQLSDEVKQDRLQRLNHLVSVKAAERSQRYANRVEEVLVEEQNPKDPTQVMGRTRGNRLTFFEGNLEELKGKTLSVEISEVRPFSLTGRVLALV</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>bifunctional tRNA-i(6)A37 thiotransferase / methyltransferase MiaB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TPA: tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=co>UBA11371</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cr>O-77</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TPA: tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=co>UBA11371</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB</td>
            <td id=cr>O-77</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA A37 methylthiotransferase MiaB [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>radical SAM methylthiotransferase, MiaB/RimO family. This subfamily contains the tRNA-i(6)A37 modification enzyme, MiaB (TIGR01574). The phylogenetic tree indicates 4 distinct clades, one of which corresponds to MiaB. The other three clades are modelled by hypothetical equivalogs (TIGR01125, TIGR01579 and TIGR01578). Together, the four models hit every sequence hit by the subfamily model without any overlap between them. This subfamily is aparrently a part of a larger superfamily of enzymes utilizing both a 4Fe4S cluster and S-adenosyl methionine (SAM) to initiate radical reactions. MiaB acts on a particular isoprenylated Adenine base of certain tRNAs causing thiolation at an aromatic carbon, and probably also transferring a methyl grouyp from SAM to the thiol. The particular substrate of the three other clades is unknown but may be very closely related.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA-N(6)-(isopentenyl)adenosine-37 thiotransferase enzyme MiaB. This model represents homologs of the MiaB enzyme responsible for the modification of the isopentenylated adenine-37 base of most bacterial and eukaryotic tRNAs that read codons beginning with uracil (all except tRNA(I,V) Ser). Adenine-37 is next to the anticodon on the 3' side in these tRNA's, and lack of modification at this site leads to an increased spontaneous mutation frequency. Isopentenylated A-37 is modified by methylthiolation at position 2, either by MiaB alone or in concert with a separate methylase yet to be discovered (MiaC?). MiaB contains a 4Fe-4S cluster which is labile under oxidizing conditions. Additionally, the sequence is homologous (via PSI-BLAST searches) to the biotin synthetase, BioB, which utilizes both an iron-sulfur cluster and S-adenosym methionine (SAM) to generate a radical which is responsible for initiating the insertion of sulfur into the substrate. It is reasonable to surmise that the methyl group of SAM becomes the methyl group of the product, but this has not been shown, and the possibility of a separate methylase exists. This equivalog is a member of a subfamily (TIGR00089) which contains several other hypothetical equivalogs which are all probably enzymes with similar function acting on different substrates. These enzymes contain a TRAM domain (pfam01938) which is believed to be responsible for binding to tRNAs. Hits to this model span all major groups of bacteria and eukaryotes, but not archaea, which are known to lack this particular tRNA modification. The enzyme from Thermotoga maritima has been cloned, expressed, spectroscopically characterized and shown to complement the E. coli MiaB enzyme. [Protein synthesis, tRNA and rRNA base modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(dimethylallyl)adenosine tRNA methylthiotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MiaB-like tRNA modifying enzyme. This clade of sequences is closely related to MiaB, a modifier of isopentenylated adenosine-37 of certain eukaryotic and bacterial tRNAs (see TIGR01574). Sequence alignments suggest that this equivalog performs the same chemical transformation as MiaB, perhaps on a different (or differently modified) tRNA base substrate. This clade is a member of a subfamily (TIGR00089) and spans low GC Gram positive bacteria, alpha and epsilon proteobacteria, Campylobacter, Porphyromonas, Aquifex, Thermotoga, Chlamydia, Treponema and Fusobacterium. [Protein synthesis, tRNA and rRNA base modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ribosomal protein S12 methylthiotransferase RimO. Members of this protein are the methylthiotransferase RimO, which modifies a conserved Asp residue in ribosomal protein S12. This clade of radical SAM family proteins is closely related to the tRNA modification bifunctional enzyme MiaB (see TIGR01574), and it catalyzes the same two types of reactions: a radical-mechanism sulfur insertion, and a methylation of the inserted sulfur. This clade spans alpha and gamma proteobacteria, cyano bacteria, Deinococcus, Porphyromonas, Aquifex, Helicobacter, Campylobacter, Thermotoga, Chlamydia, Streptococcus coelicolor and Clostridium, but does not include most other gram positive bacteria, archaea or eukaryotes. [Protein synthesis, Ribosomal proteins: synthesis and modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MiaB-like tRNA modifying enzyme, archaeal-type. This clade of sequences is closely related to MiaB, a modifier of isopentenylated adenosine-37 of certain eukaryotic and bacterial tRNAs (see TIGR01574). Sequence alignments suggest that this equivalog performs the same chemical transformation as MiaB, perhaps on a different (or differently modified) tRNA base substrate. This clade is a member of a subfamily (TIGR00089) and spans the archaea and eukaryotes. The only archaeal miaB-like genes are in this clade, while eukaryotes have sequences described by this model as well as ones falling within the scope of the MiaB equivalog model. [Protein synthesis, tRNA and rRNA base modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongator protein 3, MiaB family, Radical SAM. This superfamily contains MoaA, NifB, PqqE, coproporphyrinogen III oxidase, biotin synthase and MiaB families, and includes a representative in the eukaryotic elongator subunit, Elp-3. Some members of the family are methyltransferases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized protein family UPF0004. This family is the N terminal half of the Prosite family. The C-terminal half has been shown to be related to MiaB proteins. This domain is a nearly always found in conjunction with pfam04055 and pfam01938 although its function is uncertain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>30S ribosomal protein S12 methylthiotransferase RimO. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Radical SAM superfamily. Radical SAM proteins catalyze diverse reactions, including unusual methylations, isomerisation, sulphur insertion, ring formation, anaerobic oxidation and protein radical formation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>Radical SAM superfamily enzyme YgiQ, UPF0313 family [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Radical SAM superfamily. Enzymes of this family generate radicals by combining a 4Fe-4S cluster and S-adenosylmethionine (SAM) in close proximity. They are characterized by a conserved CxxxCxxC motif, which coordinates the conserved iron-sulfur cluster. Mechanistically, they share the transfer of a single electron from the iron-sulfur cluster to SAM, which leads to its reductive cleavage to methionine and a 5'-deoxyadenosyl radical, which, in turn, abstracts a hydrogen from the appropriately positioned carbon atom. Depending on the enzyme, SAM is consumed during this process or it is restored and reused. Radical SAM enzymes catalyze steps in metabolism, DNA repair, the biosynthesis of vitamins and coenzymes, and the biosynthesis of many antibiotics. Examples are biotin synthase (BioB), lipoyl synthase (LipA), pyruvate formate-lyase (PFL), coproporphyrinogen oxidase (HemN), lysine 2,3-aminomutase (LAM), anaerobic ribonucleotide reductase (ARR), and  MoaA, an enzyme of the biosynthesis of molybdopterin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TRAM domain. This small domain has no known function. However it may perform a nucleic acid binding role (Bateman A. unpublished observation).</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4627831</td></tr>
<tr><td id=al>end</td><td id=al>4629181</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>tRNA (N6-isopentenyl adenosine(37)-C2)-methylthiotransferase MiaB, partial</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
