<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01569</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01569 S41 family peptidase [Phormidium_A]</h3>
    <div>MNRPQPPLFHSLTLVSTVVFTTAVSFLNPLSSRSSQAASFENSPKAVLDEAWQIVNRDYVDDTFRQNDWQQVREELLGQDYSSREQAYAALRRALARLDDPYTRFLDPEAFEALNEQTSGELSGVGLRLERDNQREILSIVEPIPNSPASREGIREGDRILAIDGDSTRTMDVDDASQRIRGTVGTPVTLTIQRGEREFDVTLVRSRIELPAVHHSLSLEGDTRVGYIRLSEFSSHAHEQMYRAIEDLKEQGVDAFVLDLRGNPGGLLSVSIQIARMWLNEGAIVSTVDRHGNVDAVEATRTALTDLPLAVLVDRNSASASEIVAGALQDNGRATIIGTQTFGKALVQSVNSLSDGSGLAVTVAHYYTPNGTDISQNGILPDTRVEMTGDQQLLLSRESHRRGTLSDPVYSRAVDLFQRQAPVATTNGADTKVLQLKQR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S41 family peptidase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S41 family peptidase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>PDZ domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>PDZ domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>carboxyl-terminal processing protease</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>S41 family peptidase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain-containing protein, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>S41 family peptidase</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>S41 family peptidase</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>S41 family peptidase</td>
            <td id=cb>LEGE-11467</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>C-terminal processing protease CtpA/Prc, contains a PDZ domain [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>C-terminal peptidase (prc). A C-terminal peptidase with different substrates in different species including processing of D1 protein of the photosystem II reaction center in higher plants and cleavage of a peptide of 11 residues from the precursor form of penicillin-binding protein in E.coli E.coli and H influenza have the most distal branch of the tree and their proteins have an N-terminal 200 amino acids that show no homology to other proteins in the database. [Protein fate, Degradation of proteins, peptides, and glycopeptides, Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>carboxyl-terminal processing protease; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>C-terminal processing peptidase; serine protease family S41. The C-terminal processing peptidase (CPP, EC 3.4.21.102) also known as tail-specific protease (tsp), the photosystem II D1 C-terminal processing protease (D1P), and other related S41 protease family members are present in this CD. CPP is synthesized as a precursor form with a carboxyl-terminal extension. It specifically recognizes a C-terminal tripeptide, Xaa-Yaa-Zaa, in which Xaa is preferably Ala or Leu, Yaa is preferably Ala or Tyr and Zaa is preferably Ala, but then cleaves at a variable distance from the C-terminus. The C-terminal carboxylate group is essential, and proteins where this group is amidated are not substrates. This family of proteases contains the PDZ domain that promotes protein-protein interactions and is important for substrate recognition. The active site consists of a serine/lysine catalytic dyad. The bacterial CCP-1 is believed to be important for the degradation of incorrectly synthesized proteins as well as protection from thermal and osmotic stresses. In E. coli, it is involved in the cleavage of a C-terminal peptide of 11 residues from the precursor form of penicillin-binding protein 3 (PBP3). In the plant chloroplast, the enzyme removes the C-terminal extension of the D1 polypeptide of photosystem II, allowing the light-driven assembly of the tetranuclear manganese cluster, which is responsible for photosynthetic water oxidation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Peptidase family S41. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>tail specific protease. tail specific protease</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>C-terminal processing peptidase family S41. Peptidase family S41 (C-terminal processing peptidase or CTPase family) contains very different subfamilies; it includes photosystem II D1 C-terminal processing protease (CTPase), interphotoreceptor retinoid-binding protein IRBP and tricorn protease (TRI). CTPase and TRI both contain the PDZ domain while IRBP, although being very similar to the tail-specific protease domain, lacks the PDZ insertion domain and hydrolytic activity. These serine proteases have distinctly different active sites: in CTPase, the active site consists of a serine/lysine catalytic dyad while in tricorn core protease, it is a tetrad (serine, histidine, serine, glutamate). CPases with different substrate specificities in different species include processing of D1 protein of the photosystem II reaction center in higher plants and cleavage of a peptide of 11 residues from the precursor form of penicillin-binding protein; and others such as tricorn protease (TRI) act as a carboxypeptidase, involved in the degradation of proteasomal products. CTPase homolog IRBP, secreted by photoreceptors into the interphotoreceptor matrix, having arisen in the early evolution of the vertebrate eye, promotes the release of all-trans retinol from photoreceptors and facilitates its delivery to the retinal pigment epithelium.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>carboxy terminal-processing peptidase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>Tricorn protease; serine protease family S41. The tricorn protease (TRI), a member of the S41 peptidase family and named for its tricorn-like shape, exists only in some archaea and eubacteria. It has been shown to act as a carboxypeptidase, involved in the degradation of proteasomal products to preferentially yield di- and tripeptides, with subsequent and final degradations to free amino acid residues by tricorn interacting factors, F1, F2 and F3. Tricorn is a hexameric D3-symmetric protease of 720kD, and can self-associate further into a giant icosahedral capsid structure containing twenty copies of the complex. Each tricorn peptidase monomer consists of five structural domains: a six-bladed beta-propeller and a seven-bladed beta-propeller that limit access to the active site, the two domains (C1 and C2) that carry the active site residues, and a PDZ-like domain (proposed to be important for substrate recognition) between the C1 and C2 domains. The active site tetrad residues are distributed between the C1 and C2 domains, with serine and histidine on C1 and serine and glutamate on C2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>C-terminal processing peptidase-like; serine protease family S41. Bacterial protease homologs of the S41 family related to C-terminal processing peptidase (CPP).  CPP-1 is believed to be important for the degradation of incorrectly synthesized proteins as well as protection from thermal and osmotic stresses. CPP is synthesized with an extension on its carboxyl-terminus and specifically recognizes a C-terminal tripeptide, but cleaves at variable distance from the C-terminus. The CPP active site consists of a serine/lysine catalytic dyad. Conservation of these residues is seen in the CPP-like proteins of this group. CPP proteins contain a PDZ domain that promotes protein-protein interactions and is important for substrate recognition however, most of CPP-like proteins only have an internal fragment or lack the PDZ domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain of C-terminal processing-, tail-specific-, and tricorn proteases, which function in posttranslational protein processing, maturation, and disassembly or degradation, in Bacteria, Archaea, and plant chloroplasts. May be responsible for substrate recognition and/or binding, as most PDZ domains bind C-terminal polypeptides, and binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. In this subfamily of protease-associated PDZ domains a C-terminal beta-strand forms the peptide-binding groove base, a circular permutation with respect to PDZ domains found in Eumetazoan signaling proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>Interphotoreceptor retinoid-binding protein; serine protease family S41. Interphotoreceptor retinoid-binding protein (IRBP) is a homolog of the S41 protease, C-terminal processing peptidase (CTPase) family. It is thought to facilitate the compartmentalization of the visual cycle that requires poorly soluble and potentially toxic retinoids to cross the aqueous subretinal space between the photoreceptors and the retinal pigment epithelium (RPE). IRBP is secreted by photoreceptors into the interphotoreceptor matrix (IPM) where it is rapidly turned over by a combination of RPE and photoreceptor endocytosis. It is the most abundant soluble protein component of the IPM, consisting of homologous modules, each repeat structure arising through the duplication (as in teleost IRBP) or quadruplication (in tetrapods) of an ancient gene, arisen in the early evolution of the vertebrate eye. IRBP has been shown to promote the release of all-trans retinol from photoreceptors and facilitates its delivery to the RPE. Conversely, IRBP can promote the release of 11-cis-retinal from the RPE, prevent its isomerization in the subretinal space, and transfer it to photoreceptors. In vivo evidence implicates IRBP as a retinoid transporter in the visual cycle, suggesting a critical role for IRBP in cone function essential for human vision. IRBP is a dominant autoimmune antigen in the eye; IRBP proteolysis analysis has proven a useful biomarker for autoimmune uveitis (AU) disorders, a major cause of blindness. This family also includes a chlamydia-secreted protein, designated chlamydia protease-like activity factor (CPAF), known to degrade host proteins, enabling Chlamydia to evade host defenses and replicate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>Domain present in PSD-95, Dlg, and ZO-1/2. Also called DHR (Dlg homologous region) or GLGF (relatively well conserved tetrapeptide in these domains). Some PDZs have been shown to bind C-terminal polypeptides; others appear to bind internal (non-C-terminal) polypeptides. Different PDZs possess different binding specificities.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>PDZ domain (Also known as DHR or GLGF). PDZ domains are found in diverse signaling proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain, also called DHR (Dlg homologous region) or GLGF (after a conserved sequence motif). Many PDZ domains bind C-terminal polypeptides, though binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. Heterodimerization through PDZ-PDZ domain interactions adds to the domain's versatility, and PDZ domain-mediated interactions may be modulated dynamically through target phosphorylation. Some PDZ domains play a role in scaffolding supramolecular complexes. PDZ domains are found in diverse signaling proteins in bacteria, archebacteria, and eurkayotes. This CD contains two distinct structural subgroups with either a N- or C-terminal beta-strand forming the peptide-binding groove base. The circular permutation placing the strand on the N-terminus appears to be found in Eumetazoa only, while the C-terminal variant is found in all three kingdoms of life, and seems to co-occur with protease domains. PDZ domains have been named after PSD95(post synaptic density protein), DlgA (Drosophila disc large tumor suppressor), and ZO1, a mammalian tight junction protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain. This entry represents the PDZ domain from a wide variety of proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain of bacterial and plant zinc metalloprotases, presumably membrane-associated or integral membrane proteases, which may be involved in signalling and regulatory mechanisms. May be responsible for substrate recognition and/or binding, as most PDZ domains bind C-terminal polypeptides, and binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. In this subfamily of protease-associated PDZ domains a C-terminal beta-strand forms the peptide-binding groove base, a circular permutation with respect to PDZ domains found in Eumetazoan signaling proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>PDZ domain found in a variety of Eumetazoan signaling molecules, often in tandem arrangements. May be responsible for specific protein-protein interactions, as most PDZ domains bind C-terminal polypeptides, and binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. In this subfamily of PDZ domains an N-terminal beta-strand forms the peptide-binding groove base, a circular permutation with respect to PDZ domains found in proteases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>PDZ domain of trypsin-like serine proteases, such as DegP/HtrA, which are oligomeric proteins involved in heat-shock response, chaperone function, and apoptosis. May be responsible for substrate recognition and/or binding, as most PDZ domains bind C-terminal polypeptides, though binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. In this subfamily of protease-associated PDZ domains a C-terminal beta-strand forms the peptide-binding groove base, a circular permutation with respect to PDZ domains found in Eumetazoan signaling proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PDZ domain associated with archaeal and bacterial M61 glycyl-aminopeptidases. May be responsible for substrate recognition and/or binding, as most PDZ domains bind C-terminal polypeptides, and binding to internal (non-C-terminal) polypeptides and even to lipids has been demonstrated. In this subfamily of protease-associated PDZ domains a C-terminal beta-strand is presumed to form the peptide-binding groove base, a circular permutation with respect to PDZ domains found in Eumetazoan signaling proteins.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1326704</td></tr>
<tr><td id=al>end</td><td id=al>1328024</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>S41 family peptidase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al>SP</td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
