<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02107</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02107 pantetheine-phosphate adenylyltransferase [Phormidium_A]</h3>
    <div>MPSDSHVVAIYPGSFDPITLGHLDIIERGVRLFDKVVVAVVTNPNKSPLFEVDERIRLIEQSTEHLDKIEIDRFDGLTVNYAKLKGASVLLRGLRVLSDFEMELQMAHTNKTLSDSIETVFLATSNEYSFLSSSLVKEVARFGGSVDRFVPPPVATEIYRCYVRTQPGSTXXXXXXXXXXXXXRL</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>91</td>
            <td id=cg>91</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>91</td>
            <td id=cg>91</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>90</td>
            <td id=cg>91</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>91</td>
            <td id=cg>91</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>91</td>
            <td id=cg>91</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>88</td>
            <td id=cg>88</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>88</td>
            <td id=cg>90</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>87</td>
            <td id=cg>89</td>
            <td id=al>TPA: pantetheine-phosphate adenylyltransferase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>87</td>
            <td id=cg>89</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=co>SIO1A7</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>87</td>
            <td id=cg>88</td>
            <td id=al>pantetheine-phosphate adenylyltransferase</td>
            <td id=cb>LEGE-11467</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>60</td>
            <td id=cg>83</td>
            <td id=cg>97</td>
            <td id=al>PRK00168 phosphopantetheine adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=al>cd02163 Phosphopantetheine adenylyltransferase. Phosphopantetheine adenylyltransferase (PPAT). PPAT is an essential enzyme in bacteria, responsible for catalyzing the rate-limiting step in coenzyme A (CoA) biosynthesis.  The dinucleotide-binding fold of PPAT is homologous to class I aminoacyl-tRNA synthetases. CoA has been shown to inhibit PPAT and competes with ATP, PhP, and dPCoA. PPAT is a homohexamer in E. coli.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>83</td>
            <td id=cg>97</td>
            <td id=al>COG0669 Phosphopantetheine adenylyltransferase [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cb>79</td>
            <td id=cg>95</td>
            <td id=al>TIGR01510 pantetheine-phosphate adenylyltransferase, bacterial. This model describes pantetheine-phosphate adenylyltransferase, the penultimate enzyme of coenzyme A (CoA) biosynthesis in bacteria. It does not show any strong homology to eukaryotic enzymes of coenzyme A biosynthesis. This protein was previously designated KdtB and postulated (because of cytidyltransferase homology and proximity to kdtA) to be an enzyme of LPS biosynthesis, a cytidyltransferase for 3-deoxy-D-manno-2-octulosonic acid. However, no activity toward that compound was found with either CTP or ATP. The phylogenetic distribution of this enzyme is more consistent with coenzyme A biosynthesis than with LPS biosynthesis. [Biosynthesis of cofactors, prosthetic groups, and carriers, Pantothenate and coenzyme A]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cb>73</td>
            <td id=cg>97</td>
            <td id=al>PRK13964 pantetheine-phosphate adenylyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=al>pfam01467 Cytidylyltransferase-like. This family includes: Cholinephosphate cytidylyltransferase; glycerol-3-phosphate cytidylyltransferase. It also includes putative adenylyltransferases, and FAD synthases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=al>cd02156 nucleotidyl transferase superfamily. nt_trans (nucleotidyl transferase) This superfamily includes the class I amino-acyl tRNA synthetases, pantothenate synthetase (PanC), ATP sulfurylase, and the cytidylyltransferases, all of which have a conserved dinucleotide-binding domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=al>TIGR00125 cytidyltransferase-like domain. Protein families that contain at least one copy of this domain include citrate lyase ligase, pantoate-beta-alanine ligase, glycerol-3-phosphate cytidyltransferase, ADP-heptose synthase, phosphocholine cytidylyltransferase, lipopolysaccharide core biosynthesis protein KdtB, the bifunctional protein NadR, and a number whose function is unknown. Many of these proteins are known to use CTP or ATP and release pyrophosphate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>99</td>
            <td id=cg>93</td>
            <td id=al>COG1057 Nicotinic acid mononucleotide adenylyltransferase [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>TIGR00482 nicotinate (nicotinamide) nucleotide adenylyltransferase. This model represents the predominant bacterial/eukaryotic adenylyltransferase for nicotinamide-nucleotide, its deamido form nicotinate nucleotide, or both. The first activity, nicotinamide-nucleotide adenylyltransferase (EC 2.7.7.1), synthesizes NAD by the salvage pathway, while the second, nicotinate-nucleotide adenylyltransferase (EC 2.7.7.18) synthesizes the immediate precursor of NAD by the de novo pathway. In E. coli, NadD activity is biased toward the de novo pathway while salvage activity is channeled through the multifunctional NadR protein, but this division of labor may be exceptional. The given name of this model, nicotinate (nicotinamide) nucleotide adenylyltransferase, reflects the lack of absolute specificity with respect to substrate amidation state in most species. [Biosynthesis of cofactors, prosthetic groups, and carriers, Pyridine nucleotides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=al>cd02039 Cytidylyltransferase-like domain. Cytidylyltransferase-like domain. Many of these proteins are known to use CTP or ATP and release pyrophosphate. Protein families that contain at least one copy of this domain include citrate lyase ligase, pantoate-beta-alanine ligase, glycerol-3-phosphate cytidyltransferase, ADP-heptose synthase, phosphocholine cytidylyltransferase, lipopolysaccharide core biosynthesis protein KdtB, the bifunctional protein NadR, and a number whose function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>cd02165 Nicotinamide/nicotinate mononucleotide adenylyltransferase. Nicotinamide/nicotinate mononucleotide (NMN/ NaMN)adenylyltransferase (NMNAT).  NMNAT represents the primary bacterial and eukaryotic adenylyltransferases for nicotinamide-nucleotide and for the deamido form, nicotinate nucleotide.  It is an indispensable enzyme in the biosynthesis of NAD(+) and NADP(+). Nicotinamide-nucleotide adenylyltransferase synthesizes NAD via the salvage pathway, while nicotinate-nucleotide adenylyltransferase synthesizes the immediate precursor of NAD via the de novo pathway. Human NMNAT displays unique dual substrate specificity toward both NMN and NaMN, and can participate in both de novo and salvage pathways of NAD synthesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>PRK00071 nicotinate-nucleotide adenylyltransferase. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1778843</td></tr>
<tr><td id=al>end</td><td id=al>1779401</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>pantetheine-phosphate adenylyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>186</td></tr>
</table>

    </div>
</body>
</html>
