<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04222</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04222 NAD-dependent epimerase/dehydratase family protein [Phormidium_A]</h3>
    <div>MKVLVIGGDGYCGWATALYLSNRGHDVAILDSLIRRHWDSQLQIETLTPIAPIQRRLQRWYELTGKRIELFLGDINDYPFLSQAMHQFEPEAVVHFGEQRSAPFSMIDREHAVLTQANNVIGNLNLLYILKEDFPDCHLVKLGTMGEYGTPNIDIEEGYITIEHNGRKDTLPYPKQPGSFYHLSKVHDSHNIQFACKIWGLRATDLNQGIVYGVLTEETGMDELLVNRLDYDGVYGTALNRFCIQAAIGHPLTVYGKGGQTRALLDIRDTVRCIEIAVENPANPGEFRVFNQFTEMFSVADLALKVQQAGSALGLKVSIDNLENPRVELEDHYFNAKNTKLIDLGLQPHNLSDSLLDSLLNFAVKYQDRVDKDQILPKVRWRR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>UDP-sulfoquinovose synthase SqdB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=cr>Nodosilinea</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>NAD-dependent epimerase/dehydratase family protein</td>
            <td id=co>Phormidium</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>73</td>
            <td id=cg>99</td>
            <td id=cg>86</td>
            <td id=al>PLN02572 UDP-sulfoquinovose synthase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>cd05255 UDP_sulfoquinovose_synthase (Arabidopsis thaliana SQD1 and related proteins), extended (e) SDRs. Arabidopsis thaliana UDP-sulfoquinovose-synthase ( SQD1), an extended SDR,  catalyzes the transfer of SO(3)(-) to UDP-glucose in the biosynthesis of plant sulfolipids. Members of this subgroup share the conserved SDR catalytic residues, and a partial match to the characteristic extended-SDR NAD-binding motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>COG0451 Nucleoside-diphosphate-sugar epimerase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=al>cd08946 extended (e) SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>75</td>
            <td id=cg>95</td>
            <td id=al>cd05256 UDP-N-acetylglucosamine 4-epimerase, extended (e) SDRs. This subgroup contains UDP-N-acetylglucosamine 4-epimerase of Pseudomonas aeruginosa, WbpP,  an extended SDR, that catalyzes the NAD+ dependent conversion of UDP-GlcNAc and UDPGalNA to UDP-Glc and UDP-Gal.  This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=al>pfam01370 NAD dependent epimerase/dehydratase family. This family of proteins utilize NAD as a cofactor. The proteins in this family use nucleotide-sugar substrates for a variety of chemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>54</td>
            <td id=cb>64</td>
            <td id=al>cd05247 UDP-glucose 4 epimerase, subgroup 1, extended (e) SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>33</td>
            <td id=cr>38</td>
            <td id=al>COG1087 UDP-glucose 4-epimerase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>35</td>
            <td id=co>40</td>
            <td id=al>PRK10675 UDP-galactose-4-epimerase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>79</td>
            <td id=cg>90</td>
            <td id=al>cd05258 CDP-tyvelose 2-epimerase, extended (e) SDRs. CDP-tyvelose 2-epimerase is a tetrameric SDR that catalyzes the conversion of CDP-D-paratose to CDP-D-tyvelose, the last step in tyvelose biosynthesis. This subgroup is a member of the extended SDR subfamily, with a characteristic active site tetrad and NAD-binding motif. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cb>77</td>
            <td id=cg>99</td>
            <td id=al>cd05264 UDP-glucose 4-epimerase (G4E), subgroup 5, extended (e) SDRs. This subgroup partially conserves the characteristic active site tetrad and NAD-binding motif of the extended SDRs, and has been identified as possible UDP-glucose 4-epimerase (aka UDP-galactose 4-epimerase), a homodimeric member of the extended SDR family. UDP-glucose 4-epimerase catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3449605</td></tr>
<tr><td id=al>end</td><td id=al>3450757</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>NAD-dependent epimerase/dehydratase family protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>384</td></tr>
</table>

    </div>
</body>
</html>
