<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00982</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00982 thioredoxin [Phormidium_A]</h3>
    <div>MSEPVTINDSQFQAEVLEASQPVLAYFWASWCGPCRLVSPSIDWAAGEYGDRLKIVKLEVDANPESVKRCQVEGVPALRFYRQGEMLEAKEGAITRSQLQGLLDSLLGA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin 1</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>thioredoxin</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>99</td>
            <td id=cg>98</td>
            <td id=al>thioredoxin</td>
            <td id=cr>JAAUOQ01</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>97</td>
            <td id=cg>96</td>
            <td id=al>thioredoxin</td>
            <td id=cr>JAAUOQ01</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>thioredoxin</td>
            <td id=co>Gloeothece</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>33</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>pfam00085 Thioredoxin. Thioredoxins are small enzymes that participate in redox reactions, via the reversible oxidation of an active centre disulfide bond. Some members with only the active site are not separated from the noise.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=al>cd02947 TRX family; composed of two groups: Group I, which includes proteins that exclusively encode a TRX domain; and Group II, which are composed of fusion proteins of TRX and additional domains. Group I TRX is a small ancient protein that alter the redox state of target proteins via the reversible oxidation of an active site dithiol, present in a CXXC motif, partially exposed at the protein's surface. TRX reduces protein disulfide bonds, resulting in a disulfide bond at its active site. Oxidized TRX is converted to the active form by TRX reductase, using reducing equivalents derived from either NADPH or ferredoxins. By altering their redox state, TRX regulates the functions of at least 30 target proteins, some of which are enzymes and transcription factors. It also plays an important role in the defense against oxidative stress by directly reducing hydrogen peroxide and certain radicals, and by serving as a reductant for peroxiredoxins. At least two major types of functional TRXs have been reported in most organisms; in eukaryotes, they are located in the cytoplasm and the mitochondria. Higher plants contain more types (at least 20 TRX genes have been detected in the genome of Arabidopsis thaliana), two of which (types f amd m) are located in the same compartment, the chloroplast. Also included in the alignment are TRX-like domains which show sequence homology to TRX but do not contain the redox active CXXC motif. Group II proteins, in addition to either a redox active TRX or a TRX-like domain, also contain additional domains, which may or may not possess homology to known proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>TIGR01068 thioredoxin. Several proteins, such as protein disulfide isomerase, have two or more copies of a domain closely related to thioredoxin. This model is designed to recognize authentic thioredoxin, a small protein that should be hit exactly once by this model. Any protein that hits once with a score greater than the second (per domain) trusted cutoff may be taken as thioredoxin. [Energy metabolism, Electron transport]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>94</td>
            <td id=cg>95</td>
            <td id=al>PRK09381 thioredoxin TrxA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>95</td>
            <td id=cr>34</td>
            <td id=al>COG3118 Negative regulator of GroEL, contains thioredoxin-like and TPR-like domains  [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>85</td>
            <td id=cb>67</td>
            <td id=al>PRK10996 thioredoxin 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>90</td>
            <td id=cg>81</td>
            <td id=al>cd03011 TlpA-like family, suppressor for copper sensitivity D protein (ScsD) and actinobacterial DsbE homolog subfamily; composed of ScsD, the DsbE homolog of Mycobacterium tuberculosis (MtbDsbE) and similar proteins, all containing a redox-active CXXC motif. The Salmonella typhimurium ScsD is a thioredoxin-like protein which confers copper tolerance to copper-sensitive mutants of E. coli. MtbDsbE has been characterized as an oxidase in vitro, catalyzing the disulfide bond formation of substrates like hirudin. The reduced form of MtbDsbE is more stable than its oxidized form, consistent with an oxidase function. This is in contrast to the function of DsbE from gram-negative bacteria which is a specific reductase of apocytochrome c.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cb>70</td>
            <td id=cb>60</td>
            <td id=al>COG0526 Thiol-disulfide isomerase or thioredoxin [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>85</td>
            <td id=cg>97</td>
            <td id=al>cd02956 ybbN protein family; ybbN is a hypothetical protein containing a redox-inactive TRX-like domain. Its gene has been sequenced from several gammaproteobacteria and actinobacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>82</td>
            <td id=cg>92</td>
            <td id=al>PTZ00051 thioredoxin; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>73</td>
            <td id=cg>83</td>
            <td id=al>cd02949 TRX domain, novel NADPH thioredoxin reductase (NTR) family; composed of fusion proteins found only in oxygenic photosynthetic organisms containing both TRX and NTR domains. The TRX domain functions as a protein disulfide reductase via the reversible oxidation of an active center dithiol present in a CXXC motif, while the NTR domain functions as a reductant to oxidized TRX. The fusion protein is  bifunctional, showing both TRX and NTR activities, but it is not an independent NTR/TRX system. In plants, the protein is found exclusively in shoots and mature leaves and is localized in the chloroplast. It is involved in plant protection against oxidative stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>75</td>
            <td id=cg>82</td>
            <td id=al>cd02961 Protein Disulfide Isomerase (PDIa) family, redox active TRX domains; composed of eukaryotic proteins involved in oxidative protein folding in the endoplasmic reticulum (ER) by acting as catalysts and folding assistants. Members of this family include PDI and PDI-related proteins like ERp72, ERp57 (or ERp60), ERp44, P5, PDIR, ERp46 and the transmembrane PDIs. PDI, ERp57, ERp72, P5, PDIR and ERp46 are all oxidases, catalyzing the formation of disulfide bonds of newly synthesized polypeptides in the ER. They also exhibit reductase activity in acting as isomerases to correct any non-native disulfide bonds, as well as chaperone activity to prevent protein aggregation and facilitate the folding of newly synthesized proteins. These proteins usually contain multiple copies of a redox active TRX (a) domain containing a CXXC motif, and may also contain one or more redox inactive TRX-like (b) domains. Only one a domain is required for the oxidase function but multiple copies are necessary for the isomerase function. The different types of PDIs may show different substrate specificities and tissue-specific expression, or may be induced by stress. PDIs are in their reduced form at steady state and are oxidized to the active form by Ero1, which is localized in the ER through ERp44. Some members of this family also contain a DnaJ domain in addition to the redox active a domains; examples are ERdj5 and Pfj2. Also included in the family is the redox inactive N-terminal TRX-like domain of ERp29.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>80</td>
            <td id=cg>87</td>
            <td id=al>TIGR01126 protein disulfide-isomerase domain. This model describes a domain of eukaryotic protein disulfide isomerases, generally found in two copies. The high cutoff for total score reflects the expectation of finding both copies. The domain is similar to thioredoxin but the redox-active disulfide region motif is APWCGHCK. [Protein fate, Protein folding and stabilization]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>78</td>
            <td id=cg>88</td>
            <td id=al>cd02984 TRX domain, PICOT (for PKC-interacting cousin of TRX) subfamily; PICOT is a protein that interacts with protein kinase C (PKC) theta, a calcium independent PKC isoform selectively expressed in skeletal muscle and T lymphocytes. PICOT contains an N-terminal TRX-like domain, which does not contain the catalytic CXXC motif, followed by one to three glutaredoxin domains. The TRX-like domain is required for interaction with PKC theta. PICOT inhibits the activation of c-Jun N-terminal kinase and the transcription factors, AP-1 and NF-kB, induced by PKC theta or T-cell activating stimuli.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>90</td>
            <td id=co>44</td>
            <td id=al>PTZ00443 Thioredoxin domain-containing protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>81</td>
            <td id=cb>63</td>
            <td id=al>cd02950 TRX-like protein A (TxlA) family; TxlA was originally isolated from the cyanobacterium Synechococcus. It is found only in oxygenic photosynthetic organisms. TRX is a small enzyme that participate in redox reactions, via the reversible oxidation of an active site dithiol present in a CXXC motif. Disruption of the txlA gene suggests that the protein is involved in the redox regulation  of the structure and function of photosynthetic apparatus. The plant homolog (designated as HCF164) is localized in the chloroplast and is involved in the assembly of the cytochrome b6f complex, which takes a central position in photosynthetic electron transport.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>33</td>
            <td id=cr>31</td>
            <td id=al>cd02966 TlpA-like family; composed of  TlpA, ResA, DsbE and similar proteins. TlpA, ResA and DsbE are bacterial protein disulfide reductases with important roles in cytochrome maturation. They are membrane-anchored proteins with a soluble TRX domain containing a CXXC motif located in the periplasm. The TRX domains of this family contain an insert, approximately 25 residues in length, which correspond to an extra alpha helix and a beta strand when compared with TRX. TlpA catalyzes an essential reaction in the biogenesis of cytochrome aa3, while ResA and DsbE are essential proteins in cytochrome c maturation. Also included in this family are proteins containing a TlpA-like TRX domain with domain architectures similar to E. coli DipZ protein, and the N-terminal TRX domain of PilB protein from Neisseria which acts as a disulfide reductase that can recylce methionine sulfoxide reductases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>83</td>
            <td id=cg>88</td>
            <td id=al>cd02997 PDIa family, PDIR subfamily; composed of proteins similar to human PDIR (for Protein Disulfide Isomerase Related). PDIR is composed of three redox active TRX (a) domains and an N-terminal redox inactive TRX-like (b) domain. Similar to PDI, it is involved in oxidative protein folding in the endoplasmic reticulum (ER) through its isomerase and chaperone activities. These activities are lower compared to PDI, probably due to PDIR acting only on a subset of proteins. PDIR is preferentially expressed in cells actively secreting proteins and its expression is induced by stress. Similar to PDI, the isomerase and chaperone activities of PDIR are independent; CXXC mutants lacking isomerase activity retain chaperone activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>72</td>
            <td id=cb>77</td>
            <td id=al>cd03001 PDIa family, P5 subfamily; composed of eukaryotic proteins similar to human P5, a PDI-related protein with a domain structure of aa'b (where a and a' are redox active TRX domains and b is a redox inactive TRX-like domain). Like PDI, P5 is located in the endoplasmic reticulum (ER) and displays both isomerase and chaperone activities, which are independent of each other. Compared to PDI, the isomerase and chaperone activities of P5 are lower. The first cysteine in the CXXC motif of both redox active domains in P5 is necessary for isomerase activity. The P5 gene was first isolated as an amplified gene from a hydroxyurea-resistant hamster cell line. The zebrafish P5 homolog has been implicated to play a critical role in establishing left/right asymmetries in the embryonic midline. Some members of this subfamily are P5-like proteins containing only one redox active TRX domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>80</td>
            <td id=cr>19</td>
            <td id=al>TIGR01130 protein disulfide isomerase, eukaryotic. This model represents eukaryotic protein disulfide isomerases retained in the endoplasmic reticulum (ER) and closely related forms. Some members have been assigned alternative or additional functions such as prolyl 4-hydroxylase and dolichyl-diphosphooligosaccharide-protein glycotransferase. Members of this family have at least two protein-disulfide domains, each similar to thioredoxin but with the redox-active disulfide in the motif PWCGHCK, and an ER retention signal at the extreme C-terminus (KDEL, HDEL, and similar motifs).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>75</td>
            <td id=cb>79</td>
            <td id=al>cd02998 PDIa family, endoplasmic reticulum protein 38 (ERp38) subfamily; composed of proteins similar to the P5-like protein first isolated from alfalfa, which contains two redox active TRX (a) domains at the N-terminus, like human P5, and a C-terminal domain with homology to the C-terminal domain of ERp29, unlike human P5. The cDNA clone of this protein (named G1) was isolated from an alfalfa cDNA library by screening with human protein disulfide isomerase (PDI) cDNA. The G1 protein is constitutively expressed in all major organs of the plant and its expression is induced by treatment with tunicamycin, indicating that it may be a glucose-regulated protein. The G1 homolog in the eukaryotic social amoeba Dictyostelium discoideum is also described as a P5-like protein, which is located in the endoplasmic reticulum (ER) despite the absence of an ER-retrieval signal. G1 homologs from Aspergillus niger and Neurospora crassa have also been characterized, and are named TIGA and ERp38, respectively. Also included in the alignment is an atypical PDI from Leishmania donovani containing a single a domain, and the C-terminal a domain of a P5-like protein from Entamoeba histolytica.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>802921</td></tr>
<tr><td id=al>end</td><td id=al>803251</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>thioredoxin</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>110</td></tr>
</table>

    </div>
</body>
</html>
