<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05705</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05705 methyltransferase domain-containing protein [Phormidium_A]</h3>
    <div>MISQSKQHLEAVRQQFERSPYPRISLEETPKDNSSYLYQHHFASAYYHRNQQVIDPKDKLILDAGCGSGYKALALAYANPGAKIVGIDFSEESVKLARTRLEYHGIENVEFHAMAIESIASLGMEFDYINCDEVLYLLPDPEAGLRALKSVLKPQGIIRGNLHSALQRVNFFRAQKLFKMMGLMEDDVTDEMAMELSQETMEALRDSVDLKKKVWNSRIKGNPEGLLANYLLREDRGYTIPDLFELLGKTDLEFISMVNWRQWNPFDLFVEEDKLPAFLGMSLPATSIEERLHLYELLNPIHRLLDFWVGHPGGGCEWQPPEEWDSQTWNLAQVHLHPLLQTDSTRRELETCLQHYRAFTISQQLAIPGVGKVMIDGTTAGCLLPLFQSPQSFSNLTGFFRELRPRNPVTAEATSDEEAFAAVRGLLQSLESLSFVFLSLDPSLAKA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>methyltransferase domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>methyltransferase domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>79</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>Methyltransferase domain</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>79</td>
            <td id=cg>96</td>
            <td id=cg>98</td>
            <td id=al>methyltransferase domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>96</td>
            <td id=cg>98</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=co>Nostoc</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=co>FACHB-T130</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>class I SAM-dependent methyltransferase</td>
            <td id=co>SIO3C6</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>40</td>
            <td id=cr>23</td>
            <td id=cb>70</td>
            <td id=al>pfam13847 Methyltransferase domain. This family appears to have methyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>pfam13649 Methyltransferase domain. This family appears to be a methyltransferase domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>27</td>
            <td id=co>51</td>
            <td id=al>COG2227 2-polyprenyl-3-methyl-5-hydroxy-6-metoxy-1,4-benzoquinol methylase [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>pfam08242 Methyltransferase domain. Members of this family are SAM dependent methyltransferases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>22</td>
            <td id=cg>93</td>
            <td id=al>cd02440 S-adenosylmethionine-dependent methyltransferases (SAM or AdoMet-MTase), class I;  AdoMet-MTases are enzymes that use S-adenosyl-L-methionine (SAM or AdoMet) as a substrate for methyltransfer, creating the product S-adenosyl-L-homocysteine (AdoHcy). There are at least five structurally distinct families of AdoMet-MTases, class I being the largest and most diverse. Within this class enzymes can be classified by different substrate specificities (small molecules, lipids, nucleic acids, etc.) and different target atoms for methylation (nitrogen, oxygen, carbon, sulfur, etc.).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>24</td>
            <td id=co>46</td>
            <td id=al>PRK05134 bifunctional 2-polyprenyl-6-hydroxyphenol methylase/3-demethylubiquinol 3-O-methyltransferase UbiG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>pfam08241 Methyltransferase domain. Members of this family are SAM dependent methyltransferases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>22</td>
            <td id=co>42</td>
            <td id=al>COG2226 Ubiquinone/menaquinone biosynthesis C-methylase UbiE [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>21</td>
            <td id=co>40</td>
            <td id=al>PRK08317 hypothetical protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>48</td>
            <td id=cg>84</td>
            <td id=al>COG0500 SAM-dependent methyltransferase [Secondary metabolites biosynthesis, transport and catabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>22</td>
            <td id=co>44</td>
            <td id=al>PRK07580 Mg-protoporphyrin IX methyl transferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>27</td>
            <td id=co>48</td>
            <td id=al>COG2519 tRNA A58 N-methylase Trm61 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>12</td>
            <td id=cr>20</td>
            <td id=al>PRK09328 N5-glutamine S-adenosyl-L-methionine-dependent methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>13</td>
            <td id=cr>21</td>
            <td id=al>COG2890 Methylase of polypeptide chain release factors [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>23</td>
            <td id=cb>64</td>
            <td id=al>pfam13489 Methyltransferase domain. This family appears to be a methyltransferase domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>26</td>
            <td id=co>43</td>
            <td id=al>PRK11873 arsenite methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>12</td>
            <td id=cr>34</td>
            <td id=al>pfam05175 Methyltransferase small domain. This domain is found in ribosomal RNA small subunit methyltransferase C as well as other methyltransferases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>24</td>
            <td id=co>45</td>
            <td id=al>TIGR02072 malonyl-acyl carrier protein O-methyltransferase BioC. This enzyme, which is found in biotin biosynthetic gene clusters in proteobacteria, firmicutes, green-sulfur bacteria, fusobacterium and bacteroides, carries out an enzymatic step prior to the formation of pimeloyl-CoA, namely O-methylation of the malonyl group preferentially while on acyl carrier protein. The enzyme is recognizable as a methyltransferase by homology. [Biosynthesis of cofactors, prosthetic groups, and carriers, Biotin]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>22</td>
            <td id=co>44</td>
            <td id=al>pfam01209 ubiE/COQ5 methyltransferase family. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>15</td>
            <td id=cr>23</td>
            <td id=al>TIGR00536 HemK family putative methylases. The gene hemK from E. coli was found to contribute to heme biosynthesis and originally suggested to be protoporphyrinogen oxidase. Functional analysis of the nearest homolog in Saccharomyces cerevisiae, YNL063w, finds it is not protoporphyrinogen oxidase and sequence analysis suggests that HemK homologs have S-adenosyl-methionine-dependent methyltransferase activity (Medline 99237242). Homologs are found, usually in a single copy, in nearly all completed genomes, but varying somewhat in apparent domain architecture. Both E. coli and H. influenzae have two members rather than one. The members from the Mycoplasmas have an additional C-terminal domain. [Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>26</td>
            <td id=co>48</td>
            <td id=al>PRK00216 bifunctional demethylmenaquinone methyltransferase/2-methoxy-6-polyprenyl-1,4-benzoquinol methylase UbiE. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4665947</td></tr>
<tr><td id=al>end</td><td id=al>4667291</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>methyltransferase domain-containing protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>448</td></tr>
</table>

    </div>
</body>
</html>
