<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04312</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04312 aminomethyl-transferring glycine dehydrogenase [Phormidium_A]</h3>
    <div>MLDLNTTPNPQQPSSGTSRTAPPHLSPHGEFDFARRHNGSTPDDVAQMLSALDYESLAQLIADIIPEQIRLKQPLNLPQPLTEAEALRKLRAIAQQNQLYRSFLGLGYHDCITPPVIQRNILENPGWYTAYTPYQAEIAQGRLEALLNFQTLVIDLTGLEIANASLLDEGTAAAEAMSLSWGASKAKCNTYFVAADCHPQTIEVVQTRAEPLGIEVIVGDWQAFDFQTPIFGALLQYPTTKGAVCDYGEFVERAHQHQALVTVAADLLSLTLLRPPGEFGADIAVGSSQRFGVPLGYGGPHAAYFATKETYQRKLPGRLVGVSKDSSGQAALRLALQTREQHIRRDKATSNICTAQVLLAVMASMYAVYHGPAGLRGIAQGIHERAVMLAEGLNRLKFRIGESVLFDTLTVQVGSAQRMIQRAAAQRINLRHVSETEVGISLDETTSDADLLDLWQIFADALELPFTLDQMRDQVTESKYGIGRSPLARESEYLTQPVFNHYHSETEFLRYVHRLERKDLSLTTSMIPLGSCTMKLNATAEMLPVSWPEFAKIHPFAPKSQTQGYQLLFGQLQAWLAEITGFAGVSLQPNAGSQGEYAGLQVIRAYHRDRGEAQRRVCLIPESAHGTNPASAVMCGMKVVAIACDESGNIDLEDLRRKAEKHAAELAAIMVTYPSTHGVFEESIVEVCEIVHGFGGQVYLDGANLNAQVGLCRPGDFGADVCHLNLHKTFCIPHGGGGPGMGPIGVAAHLVPFLPGHSLVSPQSQAHPQGIGAISAAPWGSASILPISWMYIAMMGAQGLTEATILAILHANYLAARLDGVYPVLYKGASGRVAHECILDLRPLRKSAGVTVEDVAKRLMDYGFHAPTMSWPVAGTMMVEPTESESLAEIDRFCEAMIAIRGEIEAIEWGEVDEGNNLLKNAPHTAESLLSGEWERPYSREQAAYPLPWCREAKFWVPVGRIDAAFGDRNLVCSCVGMEAYGEMPVETVESVEVVEETEVLNPRPMELFLGDGGDG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glycine dehydrogenase (aminomethyl-transferring)</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>decarboxylating glycine dehydrogenase GcvP, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TPA: aminomethyl-transferring glycine dehydrogenase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TPA: aminomethyl-transferring glycine dehydrogenase</td>
            <td id=co>DVEG01</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=co>PCC-10802</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=co>FACHB-40</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase</td>
            <td id=co>PCC7113</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>glycine dehydrogenase (decarboxylating)</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glycine dehydrogenase (decarboxylating). This apparently ubiquitous enzyme is found in bacterial, mammalian and plant sources. The enzyme catalyzes the reaction: GLYCINE + LIPOYLPROTEIN = S-AMINOMETHYL-DIHYDROLIPOYLPROTEIN + CO2. It is part of the glycine decarboxylase multienzyme complex (GDC) consisting of four proteins P, H, L and T. Active site in E.coli is located as the (K) residues at position 713 of the SEED alignment. [Energy metabolism, Amino acids and amines]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glycine dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Glycine cleavage system protein P (pyridoxal-binding), C-terminal domain [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Glycine cleavage system P-protein. This family consists of Glycine cleavage system P-proteins EC:1.4.4.2 from bacterial, mammalian and plant sources. The P protein is part of the glycine decarboxylase multienzyme complex EC:2.1.2.10 (GDC) also annotated as glycine cleavage system or glycine synthase. GDC consists of four proteins P, H, L and T. The reaction catalyzed by this protein is:- Glycine + lipoylprotein <=> S-aminomethyldihydrolipoylprotein + CO2</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Glycine cleavage system protein P (pyridoxal-binding), N-terminal domain [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Glycine cleavage system P-protein, alpha- and beta-subunits. This family consists of Glycine cleavage system P-proteins EC:1.4.4.2 from bacterial, mammalian and plant sources. The P protein is part of the glycine decarboxylase multienzyme complex EC:2.1.2.10 (GDC) also annotated as glycine cleavage system or glycine synthase. GDC consists of four proteins P, H, L and T. The reaction catalysed by this protein is: Glycine + lipoylprotein <=> S-aminomethyldihydrolipoylprotein + CO2. Alpha-beta-type dimers associate to form an alpha(2)beta(2) tetramer, where the alpha- and beta-subunits are structurally similar and appear to have arisen by gene duplication and subsequent divergence with a loss of one active site. The members of this CD are widely dispersed among all three forms of cellular life.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Glycine cleavage system P-protein, alpha- and beta-subunits. This family consists of Glycine cleavage system P-proteins EC:1.4.4.2 from bacterial, mammalian and plant sources. The P protein is part of the glycine decarboxylase multienzyme complex EC:2.1.2.10 (GDC) also annotated as glycine cleavage system or glycine synthase. GDC consists of four proteins P, H, L and T. The reaction catalysed by this protein is: Glycine + lipoylprotein <=> S-aminomethyldihydrolipoylprotein + CO2. Alpha-beta-type dimers associate to form an alpha(2)beta(2) tetramer, where the alpha- and beta-subunits are structurally similar and appear to have arisen by gene duplication and subsequent divergence with a loss of one active site. The members of this CD are widely dispersed among all three forms of cellular life.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase subunit GcvPB. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>aminomethyl-transferring glycine dehydrogenase subunit GcvPA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Aspartate aminotransferase (AAT) superfamily (fold type I) of pyridoxal phosphate (PLP)-dependent enzymes. PLP combines with an alpha-amino acid to form a compound called a Schiff base or aldimine intermediate, which depending on the reaction, is the substrate in four kinds of reactions (1) transamination (movement of amino groups), (2) racemization (redistribution of enantiomers), (3) decarboxylation (removing COOH groups), and (4) various side-chain reactions depending on the enzyme involved. Pyridoxal phosphate (PLP) dependent enzymes were previously classified into alpha, beta and gamma classes, based on the chemical characteristics (carbon atom involved) of the reaction they catalyzed. The availability of several structures allowed a comprehensive analysis of  the evolutionary classification of PLP dependent enzymes, and it was found that the functional classification did not always agree with the evolutionary history of these enzymes. Structure and sequence analysis has revealed that the PLP dependent enzymes can be classified into four major groups of different evolutionary origin: aspartate aminotransferase superfamily (fold type I), tryptophan synthase beta superfamily (fold type II), alanine racemase superfamily (fold type III), and D-amino acid superfamily (fold type IV) and Glycogen phophorylase family (fold type V).</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3480209</td></tr>
<tr><td id=al>end</td><td id=al>3483260</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>aminomethyl-transferring glycine dehydrogenase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
