<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03582</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03582 SDR family oxidoreductase [Phormidium_A]</h3>
    <div>MLGAIAMTSHSINRSAKTIVLTGVSRGLGQAMLEGFIAEGHRLFGCATDEGAIAQLQKTYGDPHQFAVVDLADDGQVQRWSQQVLQEIQAPDILINNAGVINEPARVWEMSAPEFDRVIDVNIKGVANVLRHFVPAMIQQEGGIIVNFSSTWGRSTSPEVAPYCATKWAIEGLTRSLSQELPRHLAAVALNPGVINTDMLQTCFGSSANNYTTPKAWARSAVPFILNLTPQDNGKAATVPG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Short-chain alcohol dehydrogenase of unknown specificity</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>TPA: SDR family oxidoreductase</td>
            <td id=cr>HRBIN36</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>FACHB-152</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>Anabaenopsis</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>classical (c) SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>3-oxoacyl-ACP reductase FabG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>short chain dehydrogenase. This family contains a wide variety of dehydrogenases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>NAD(P)-dependent dehydrogenase, short-chain alcohol dehydrogenase family [Lipid transport and metabolism, Secondary metabolites biosynthesis, transport and catabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>17beta hydroxysteroid dehydrogenase-like, classical (c) SDRs. 17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>NADP-dependent 3-hydroxy acid dehydrogenase YdfG [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>beta-Keto acyl carrier protein reductase (BKR), involved in Type II FAS, classical (c) SDRs. This subgroup includes the Escherichai coli K12 BKR, FabG. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet) NAD(P)(H) binding region and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues.   Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD binding motif and characteristic NAD-binding and catalytic sequence patterns.  These enzymes have a 3-glycine N-terminal NAD(P)(H) binding pattern: TGxxxGxG in classical SDRs.  Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif.  Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P) binding motif and  an altered active site motif (YXXXN).  Fungal type type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.  Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P) binding motif and missing or unusual active site residues.  Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site.  Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr-151 and Lys-155, and well as Asn-111 (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>67</td>
            <td id=al>classical (c) SDR, subgroup 5. These proteins are members of the classical SDR family, with a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>67</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Short-chain dehydrogenase [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>3-ketodihydrosphingosine reductase (KDSR) and related proteins, classical (c) SDR. These proteins include members identified as KDSR, ribitol type dehydrogenase, and others. The group shows strong conservation of the active site tetrad and glycine rich NAD-binding motif of the classical SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>gluconate 5-dehydrogenase (Ga5DH)-like, classical (c) SDRs. Ga5DH catalyzes the NADP-dependent conversion of carbon source D-gluconate and 5-keto-D-gluconate. This SDR subgroup has a classical Gly-rich NAD(P)-binding motif and a conserved active site tetrad pattern. However, it has been proposed that Arg104 (Streptococcus suis Ga5DH numbering), as well as an active site Ca2+, play a critical role in catalysis. In addition to Ga5DHs this subgroup contains Erwinia chrysanthemi KduD which is involved in pectin degradation, and is a putative 2,5-diketo-3-deoxygluconate dehydrogenase. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107,15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Enoyl-(Acyl carrier protein) reductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR), subgroup 3, classical (c) SDR. This subgroup includes the putative Brucella melitensis biovar Abortus 2308 BKR, FabG, Mesorhizobium loti MAFF303099 FabG, and other classical SDRs. BKR, a member of the SDR family, catalyzes the NADPH-dependent reduction of acyl carrier protein in the first reductive step of de novo fatty acid synthesis (FAS).  FAS consists of 4 elongation steps, which are repeated to extend the fatty acid chain thru the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I Fas utilizes one or 2 multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>human 17-beta-hydroxysteroid dehydrogenase XI-like, classical (c) SDRs. 17-beta-hydroxysteroid dehydrogenases (17betaHSD) are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. 17betaHSD type XI, a classical SDR, preferentially converts 3alpha-Adiol to androsterone but not numerous other tested steroids. This subgroup of classical SDRs also includes members identified as retinol dehydrogenases, which convert retinol to retinal, a property that overlaps with 17betaHSD activity. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>short chain dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>short chain dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>L-iditol 2-dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>sorbitol-6-phosphate 2-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>classical (c) SDR, subgroup 7. These proteins are members of the classical SDR family, with a canonical active site triad (and also an active site Asn) and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>classical (c) SDR, subgroup 12. These are classical SDRs, with the canonical active site tetrad and glycine-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>aldose dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>insect type alcohol dehydrogenase (ADH)-like, classical (c) SDRs. This subgroup contains insect type ADH, and 15-hydroxyprostaglandin dehydrogenase (15-PGDH) type I; these proteins are classical SDRs. ADH catalyzes the NAD+-dependent oxidation of alcohols to aldehydes/ketones. This subgroup is distinct from the zinc-dependent alcohol dehydrogenases of the medium chain dehydrogenase/reductase family, and evolved in fruit flies to allow the digestion of fermenting fruit. 15-PGDH catalyzes the NAD-dependent interconversion of (5Z,13E)-(15S)-11alpha,15-dihydroxy-9-oxoprost-13-enoate and (5Z,13E)-11alpha-hydroxy-9,15-dioxoprost-13-enoate, and has a typical SDR glycine-rich NAD-binding motif, which is not fully present in ADH.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>classical (c) SDR, subgroup 3. These proteins are members of the classical SDR family, with a canonical active site triad (and also active site Asn) and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>carbonyl reductase sniffer-like, classical (c) SDRs. Sniffer is an NADPH-dependent carbonyl reductase of the classical SDR family. Studies in Drosophila melanogaster implicate Sniffer in the prevention of neurodegeneration due to aging and oxidative-stress. This subgroup also includes Rhodococcus sp. AD45 IsoH, which is an NAD-dependent 1-hydroxy-2-glutathionyl-2-methyl-3-butene dehydrogenase involved in isoprene metabolism, Aspergillus nidulans StcE encoded by a gene which is part of a proposed sterigmatocystin biosynthesis gene cluster, Bacillus circulans SANK 72073 BtrF encoded by a gene found in the butirosin biosynthesis gene cluster, and Aspergillus parasiticus nor-1 involved in the biosynthesis of aflatoxins. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>classical (c) SDR, subgroup 4. This subgroup has a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>tetrahydroxynaphthalene/trihydroxynaphthalene reductase-like, classical (c) SDRs. 1,3,6,8-tetrahydroxynaphthalene reductase (4HNR) of Magnaporthe grisea and the related 1,3,8-trihydroxynaphthalene reductase (3HNR) are typical members of the SDR family containing the canonical glycine rich NAD(P)-binding site and active site tetrad, and function in fungal melanin biosynthesis. This subgroup also includes an SDR from Norway spruce that may function to protect against both biotic and abitoic stress. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>meso-2,3-butanediol dehydrogenase-like, classical (c) SDRs. 2,3-butanediol dehydrogenases (BDHs) catalyze the NAD+ dependent conversion of 2,3-butanediol to acetonin; BDHs are classified into types according to their stereospecificity as to substrates and products. Included in this subgroup are Klebsiella pneumonia meso-BDH which catalyzes meso-2,3-butanediol to D(-)-acetonin, and Corynebacterium glutamicum L-BDH which catalyzes lX+)-2,3-butanediol to L(+)-acetonin. This subgroup is comprised of classical SDRs with the characteristic catalytic triad and NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Clostridium thermocellum A3DFK9-like, a putative carbohydrate or polyalcohol metabolizing SDR, classical (c) SDRs. This subgroup includes a putative carbohydrate or polyalcohol metabolizing SDR (A3DFK9) from Clostridium thermocellum. Its members have a TGXXXGXG classical-SDR glycine-rich NAD-binding motif, and some have a canonical SDR active site tetrad (A3DFK9 lacks the upstream Asn). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>Porcine testicular carbonyl reductase (PTCR)-like, classical (c) SDRs. PTCR is a classical SDR which catalyzes the NADPH-dependent reduction of ketones on steroids and prostaglandins. Unlike most SDRs, PTCR functions as a monomer. This subgroup also includes human carbonyl reductase 1 (CBR1) and CBR3. CBR1 is an NADPH-dependent SDR with broad substrate specificity and may be responsible for the in vivo reduction of quinones, prostaglandins, and other carbonyl-containing compounds. In addition it includes poppy NADPH-dependent salutaridine reductase which catalyzes the stereospecific reduction of salutaridine to 7(S)-salutaridinol in the biosynthesis of morphine, and Arabidopsis SDR1,a menthone reductase, which catalyzes the reduction of menthone to neomenthol, a compound with antimicrobial activity; SDR1  can also carry out neomenthol oxidation. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>3beta17beta hydroxysteroid dehydrogenase-like, classical (c) SDRs. This subgroup includes members identified as 3beta17beta hydroxysteroid dehydrogenase, 20beta hydroxysteroid dehydrogenase, and R-alcohol dehydrogenase. These proteins exhibit the canonical active site tetrad and glycine rich NAD(P)-binding motif of the classical SDRs. 17beta-dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens, and include members of the SDR family. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>2,3 dihydro-2,3 dihydrozybenzoate dehydrogenases, classical (c) SDRs. 2,3 dihydro-2,3 dihydrozybenzoate dehydrogenase shares the characteristics of the classical SDRs. This subgroup includes Escherichai coli EntA which catalyzes the NAD+-dependent oxidation of 2,3-dihydro-2,3-dihydroxybenzoate to 2,3-dihydroxybenzoate during biosynthesis of the siderophore Enterobactin. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>secoisolariciresinol dehydrogenase (secoisolariciresinol-DH)-like, classical (c) SDRs. Podophyllum secoisolariciresinol-DH is a homo tetrameric, classical SDR that catalyzes the NAD-dependent conversion of (-)-secoisolariciresinol to (-)-matairesinol via a (-)-lactol intermediate. (-)-Matairesinol is an intermediate to various 8'-lignans, including the cancer-preventive mammalian lignan, and those involved in vascular plant defense. This subgroup also includes rice momilactone A synthase which catalyzes the conversion of 3beta-hydroxy-9betaH-pimara-7,15-dien-19,6beta-olide into momilactone A, Arabidopsis ABA2 which during abscisic acid (ABA) biosynthesis, catalyzes the conversion of xanthoxin to abscisic aldehyde and, maize Tasselseed2 which participate in the maize sex determination pathway. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>human DHRS6-like, classical (c) SDRs. Human DHRS6, and similar proteins. These proteins are classical SDRs, with a canonical active site tetrad and a close match to the typical Gly-rich NAD-binding motif. Human DHRS6 is a cytosolic type 2 (R)-hydroxybutyrate dehydrogenase, which catalyses the conversion of (R)-hydroxybutyrate to acetoacetate. Also included in this subgroup is Escherichia coli UcpA (upstream cys P). Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.    Note: removed :  needed to make this chiodl smaller when drew final trees: rmeoved text form description: Other proteins in this subgroup include Thermoplasma acidophilum aldohexose dehydrogenase, which has high dehydrogenase activity against D-mannose, Bacillus subtilis BacC involved in the biosynthesis of the dipeptide bacilysin and its antibiotic moiety anticapsin, Sphingomonas paucimobilis strain B90 LinC, involved in the degradation of hexachlorocyclohexane isomers...... P).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>HetN oxidoreductase-like, classical (c) SDR. This subgroup includes Anabaena sp. strain PCC 7120 HetN, a putative oxidoreductase involved in heterocyst differentiation, and related proteins.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>sepiapterin reductase (SPR)-like, classical (c) SDRs. Human SPR, a member of the SDR family, catalyzes the NADP-dependent reduction of sepiaptern to 7,8-dihydrobiopterin (BH2). In addition to SPRs, this subgroup also contains Bacillus cereus yueD, a benzil reductase, which catalyzes the stereospecific reduction of benzil to (S)-benzoin. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>2,3-dihydroxybenzoate-2,3-dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>classical (c) SDR, subgroup 9. This subgroup has the canonical active site tetrad and NAD-binding motif of the classical SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>xylulose reductase-like, classical (c) SDRs. Members of this subgroup include proteins identified as L-xylulose reductase (XR) and carbonyl reductase; they are members of the SDR family. XR, catalyzes the NADP-dependent reduction of L-xyulose and other sugars. Tetrameric mouse carbonyl reductase is involved in the metabolism of biogenic and xenobiotic carbonyl compounds. This subgroup also includes tetrameric chicken liver D-erythrulose reductase, which catalyzes the reduction of D-erythrulose to D-threitol. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>11beta-hydroxysteroid dehydrogenase type 1 (11beta-HSD1)-like, classical (c) SDRs. Human 11beta_HSD1 catalyzes the NADP(H)-dependent interconversion of cortisone and cortisol. This subgroup also includes human dehydrogenase/reductase SDR family member 7C (DHRS7C) and DHRS7B. These proteins have the GxxxGxG nucleotide binding motif and S-Y-K catalytic triad characteristic of the SDRs, but have an atypical C-terminal domain that contributes to homodimerization contacts. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-oxoacyl-ACP reductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>glucose 1 dehydrogenase (GlcDH), classical (c) SDRs. GlcDH, is a tetrameric member of the SDR family, it catalyzes the NAD(P)-dependent oxidation of beta-D-glucose to D-glucono-delta-lactone. GlcDH has a typical NAD-binding site glycine-rich pattern as well as the canonical active site tetrad (YXXXK motif plus upstream Ser and Asn). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>SDR family oxidoreductase UcpA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>human 17beta-hydroxysteroid dehydrogenase type 2 (type 2 17beta-HSD)-like, classical (c) SDRs. 17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. This classical-SDR subgroup includes the human proteins: type 2 17beta-HSD, type 6 17beta-HSD,  type 2 11beta-HSD, dehydrogenase/reductase SDR family member 9,  short-chain dehydrogenase/reductase family 9C member 7, 3-hydroxybutyrate dehydrogenase type 1, and retinol dehydrogenase 5. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>Sorbitol dehydrogenase (SDH), classical (c) SDR. This bacterial subgroup includes Rhodobacter sphaeroides SDH, and other SDHs. SDH  preferentially interconverts D-sorbitol (D-glucitol) and D-fructose, but also interconverts L-iditol/L-sorbose and galactitol/D-tagatose. SDH is NAD-dependent and is a dimeric member of the SDR family. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>hypothetical protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>tropinone reductase-I and II (TR-1, and TR-II)-like, classical (c) SDRs. This subgroup includes TR-I and TR-II; these proteins are members of the SDR family. TRs catalyze the NADPH-dependent reductions of the 3-carbonyl group of tropinone, to a beta-hydroxyl group. TR-I and TR-II produce different stereoisomers from tropinone, TR-I produces tropine (3alpha-hydroxytropane), and TR-II, produces pseudotropine (sigma-tropine, 3beta-hydroxytropane).  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>70</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>short chain dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>gluconate 5-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>mannitol dehydrogenase (MDH)-like, classical (c) SDRs. NADP-mannitol dehydrogenase catalyzes the conversion of fructose to mannitol, an acyclic 6-carbon sugar. MDH is a tetrameric member of the SDR family. This subgroup also includes various other tetrameric SDRs, including Pichia stipitis D-arabinitol dehydrogenase (aka polyol dehydrogenase), Candida albicans Sou1p, a sorbose reductase, and Candida parapsilosis (S)-specific carbonyl reductase (SCR, aka S-specific alcohol dehydrogenase) which catalyzes the enantioselective reduction of 2-hydroxyacetophenone into (S)-1-phenyl-1,2-ethanediol. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>classical (c) SDR, subgroup 11. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>clavulanic acid dehydrogenase (CAD), classical (c) SDR. CAD catalyzes the NADP-dependent reduction of clavulanate-9-aldehyde to clavulanic acid, a beta-lactamase inhibitor. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>human Mgc4172-like, classical (c) SDRs. Human Mgc4172-like proteins, putative SDRs. These proteins are members of the SDR family, with a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>cyclohexanol reductases, including levodione reductase, classical (c) SDRs. Cyloclohexanol reductases,including (6R)-2,2,6-trimethyl-1,4-cyclohexanedione (levodione) reductase of Corynebacterium aquaticum, catalyze the reversible oxidoreduction of hydroxycyclohexanone derivatives. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>gluconate 5-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>dihydroanticapsin 7-dehydrogenase. Members of this family are dihydroanticapsin 7-dehydrogenase (EC 1.1.1.385), one of seven key molecular markers for biosynthesis of the non-cognate amino acid anticapsin, a building block for the dipeptide antibiotic natural product bacilysin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>classical (c) SDR, subgroup 2. Short-chain dehydrogenases/reductases (SDRs, aka Tyrosine-dependent oxidoreductases) are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase NodG; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>classical (c) SDR, subgroup 6. These proteins are members of the classical SDR family, with a canonical active site tetrad  and a fairly well conserved typical Gly-rich  NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>(S)-benzoin forming benzil reductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR)-like, SDR. This subgroup resembles the SDR family, but does not have a perfect match to the NAD-binding motif or the catalytic tetrad characteristic of the SDRs. It includes the SDRs, Q9HYA2 from Pseudomonas aeruginosa PAO1 and APE0912 from Aeropyrum pernix K1. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>17hydroxysteroid dehydrogenase type 10 (HSD10)-like, classical (c) SDRs. HSD10, also known as amyloid-peptide-binding alcohol dehydrogenase (ABAD), was previously identified as a L-3-hydroxyacyl-CoA dehydrogenase, HADH2. In fatty acid metabolism, HADH2 catalyzes the third step of beta-oxidation, the conversion of a hydroxyl to a keto group in the NAD-dependent oxidation of L-3-hydroxyacyl CoA. In addition to alcohol dehydrogenase and HADH2 activites, HSD10 has steroid dehydrogenase activity. Although the mechanism is unclear, HSD10 is implicated in the formation of amyloid beta-petide in the brain (which is linked to the development of Alzheimer's disease). Although HSD10 is normally concentrated in the mitochondria, in the presence of amyloid beta-peptide it translocates into the plasma membrane, where it's action may generate cytotoxic aldehydes and may lower estrogen levels through its use of 17-beta-estradiol as a substrate. HSD10 is a member of the SRD family, but differs from other SDRs by the presence of two insertions of unknown function. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>human dehydrogenase/reductase (SDR family) member 1 (DHRS1) and human hydroxysteroid dehydrogenase-like protein 2 (HSDL2), classical (c) SDRs. This subgroup includes human DHRS1 and human HSDL2 and related proteins. These are members of the classical SDR family, with a canonical Gly-rich  NAD-binding motif and the typical YXXXK active site motif. However, the rest of the catalytic tetrad is not strongly conserved. DHRS1 mRNA has been detected in many tissues, liver, heart, skeletal muscle, kidney and pancreas; a longer transcript is predominantly expressed in the liver , a shorter one in the heart. HSDL2 may play a part in fatty acid metabolism, as it is found in peroxisomes. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>RhlG and related beta-ketoacyl reductases, classical (c) SDRs. Pseudomonas aeruginosa RhlG is an SDR-family beta-ketoacyl reductase involved in Rhamnolipid biosynthesis. RhlG is similar to but distinct from the FabG family of beta-ketoacyl-acyl carrier protein (ACP) of type II fatty acid synthesis. RhlG and related proteins are classical SDRs, with a canonical active site tetrad and glycine-rich NAD(P)-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>d-3-hydroxybutyrate dehydrogenase (HBDH), classical (c) SDRs. DHBDH, an NAD+ -dependent enzyme, catalyzes the interconversion of D-3-hydroxybutyrate and acetoacetate. It is a classical SDR, with the canonical NAD-binding motif and active site tetrad. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>2-dehydro-3-deoxy-D-gluconate 5-dehydrogenase KduD. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>1-cyclohexenylcarbonyl_coenzyme A_reductase (ChcA)_like, classical (c) SDRs. This subgroup contains classical SDR proteins, including members identified as 1-cyclohexenylcarbonyl coenzyme A reductase. ChcA of Streptomyces collinus is implicated in the final reduction step of shikimic acid to ansatrienin. ChcA shows sequence similarity to the SDR family of NAD-binding proteins, but it lacks the conserved Tyr of the characteristic catalytic site. This subgroup also contains the NADH-dependent enoyl-[acyl-carrier-protein(ACP)] reductase FabL from Bacillus subtilis. This enzyme participates in bacterial fatty acid synthesis, in type II fatty-acid synthases and catalyzes the last step in each elongation cycle. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>gluconate 5-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>17-beta-hydroxysteroid dehydrogenases (17beta-HSDs) types -1, -3, and -12, -like, classical (c) SDRs. This subgroup includes various 17-beta-hydroxysteroid dehydrogenases and 3-ketoacyl-CoA reductase, these are members of the SDR family, and contain the canonical active site tetrad and glycine-rich NAD-binding motif of the classical SDRs. 3-ketoacyl-CoA reductase (KAR, aka 17beta-HSD type 12, encoded by HSD17B12) acts in fatty acid elongation; 17beta- hydroxysteroid dehydrogenases are isozymes that catalyze activation and inactivation of estrogen and androgens, and include members of the SDR family. 17beta-estradiol dehydrogenase (aka 17beta-HSD type 1, encoded by HSD17B1) converts estrone to estradiol. Estradiol is the predominant female sex hormone. 17beta-HSD type 3 (aka testosterone 17-beta-dehydrogenase 3, encoded by HSD17B3) catalyses the reduction of androstenedione to testosterone, it also accepts estrogens as substrates. This subgroup also contains a putative steroid dehydrogenase let-767 from Caenorhabditis elegans, mutation in which results in  hypersensitivity to cholesterol limitation.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>Trans-2-enoyl-CoA reductase (TER) and 2,4-dienoyl-CoA reductase (DECR), atypical (a) SDR. TTER is a peroxisomal protein with a proposed role in fatty acid elongation. Fatty acid synthesis is known to occur in the both endoplasmic reticulum and mitochondria; peroxisomal TER has been proposed as an additional fatty acid elongation system, it reduces the double bond at C-2 as the last step of elongation.  This system resembles the mitochondrial system in that acetyl-CoA is used as a carbon donor. TER may also function in phytol metabolism, reducting phytenoyl-CoA to phytanoyl-CoA in peroxisomes. DECR processes double bonds in fatty acids to increase their utility in fatty acid metabolism; it reduces 2,4-dienoyl-CoA to an enoyl-CoA. DECR is active in mitochondria and peroxisomes. This subgroup has the Gly-rich NAD-binding motif of the classical SDR family, but does not display strong identity to the canonical active site tetrad, and lacks the characteristic Tyr at the usual position. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>retinal dehydrogenase-like, classical (c) SDR. These classical SDRs includes members identified as retinol dehydrogenases, which convert retinol to retinal, a property that overlaps with 17betaHSD activity. 17beta-dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens, and include members of the short-chain dehydrogenases/reductase family. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>3-hydroxybutyrate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>human dehydrogenase/reductase (SDR family) member 1 (DHRS1) -like, classical (c) SDRs. This subgroup includes human DHRS1 and related proteins. These are members of the classical SDR family, with a canonical Gly-rich  NAD-binding motif and the typical YXXXK active site motif. However, the rest of the catalytic tetrad is not strongly conserved. DHRS1 mRNA has been detected in many tissues, liver, heart, skeletal muscle, kidney and pancreas; a longer transcript is predominantly expressed in the liver , a shorter one in the heart. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>2-dehydro-3-deoxy-D-gluconate 5-dehydrogenase KduD. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cr>25</td>
            <td id=al>bifunctional aldolase/short-chain dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Polyketide ketoreductase, classical (c) SDR. Polyketide ketoreductase (KR) is a classical SDR with a characteristic NAD-binding pattern and active site tetrad.  Aromatic polyketides include various aromatic compounds of pharmaceutical interest. Polyketide KR, part of the type II polyketide synthase (PKS) complex, is comprised of stand-alone domains that resemble the domains found in fatty acid synthase and multidomain type I PKS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>(S)-acetoin forming diacetyl reductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>cis-2,3-dihydrobiphenyl-2,3-diol dehydrogenase. Members of this family occur as the BphD protein of biphenyl catabolism and as the TodD protein of toluene catabolism. Members catalyze the second step in each pathway and proved interchangeable when tested; the first and fourth enzymes in each pathway confer metabolic specificity. In the context of biphenyl degradation, the enzyme acts as cis-2,3-dihydrobiphenyl-2,3-diol dehydrogenase (EC 1.3.1.56), while in toluene degradation it acts as cis-toluene dihydrodiol dehydrogenase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>acetoacetyl-CoA reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>Short-chain dehydrogenase involved in D-alanine esterification of teichoic acids [Cell wall/membrane/envelope biogenesis, Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>rhamnulose-1-phosphate aldolase/alcohol dehydrogenase, classical (c) SDRs. This family has bifunctional proteins with an N-terminal aldolase and a C-terminal classical SDR domain. One member is identified as a rhamnulose-1-phosphate aldolase/alcohol dehydrogenase. The SDR domain has a canonical SDR glycine-rich NAD(P) binding motif and a match to the characteristic active site triad. However, it lacks an upstream active site Asn typical of SDRs. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>gluconate 5-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>glucose-1-dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>classical (c) SDR, subgroup 8. This subgroup has a fairly well conserved active site tetrad and domain size of the classical SDRs, but has an atypical NAD-binding motif ([ST]G[GA]XGXXG). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>bifunctional NADP-dependent 3-hydroxy acid dehydrogenase/3-hydroxypropionate dehydrogenase YdfG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>xanthoxin dehydrogenase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Tthb094 and related proteins, classical (c) SDRs. Tthb094 from Thermus Thermophilus is a classical SDR which binds NADP. Members of this subgroup contain the YXXXK active site characteristic of SDRs. Also, an upstream Asn residue of the canonical catalytic tetrad is partially conserved in this subgroup of proteins of undetermined function. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>sorbitol-6-phosphate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>classical (c) SDR, subgroup 1. These proteins are members of the classical SDR family, with a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>2,3-dihydroxy-2,3-dihydrophenylpropionate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>3-ketoacyl-ACP reductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>pteridine reductase (PR), classical (c) SDRs. Pteridine reductases (PRs), members of the SDR family, catalyzes the NAD-dependent reduction of folic acid, dihydrofolate and related compounds. In Leishmania, pteridine reductase (PTR1) acts to circumvent the anti-protozoan drugs that attack dihydrofolate reductase activity. Proteins in this subgroup have an N-terminal NAD-binding motif and a YxxxK active site motif, but have an Asp instead of the usual upstream catalytic Ser. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>putative beta-ketoacyl acyl carrier protein [ACP] reductase (BKR), subgroup 1, classical (c) SDR. This subgroup includes Escherichia coli CFT073 FabG. The Escherichai coli K12 BKR, FabG, belongs to a different subgroup. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS. SDRs are a functionally diverse family of oxidoreductases that have a single domain with structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet) NAD(P)(H) binding region and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues.   Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD binding motif and characteristic NAD-binding and catalytic sequence patterns.  These enzymes have a 3-glycine N-terminal NAD(P)(H) binding pattern: TGxxxGxG in classical SDRs. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif.  Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P) binding motif and  an altered active site motif (YXXXN).  Fungal type type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.  Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P) binding motif and missing or unusual active site residues.  Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site.  Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr-151 and Lys-155, and well as Asn-111 (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>cis-biphenyl-2,3-dihydrodiol-2,3-dehydrogenase (BphB)-like, classical (c) SDRs. cis-biphenyl-2,3-dihydrodiol-2,3-dehydrogenase (BphB) is a classical SDR, it is of particular importance for its role in the degradation of biphenyl/polychlorinated biphenyls(PCBs); PCBs are a significant source of environmental contamination. This subgroup also includes Pseudomonas putida F1 cis-biphenyl-1,2-dihydrodiol-1,2-dehydrogenase (aka cis-benzene glycol dehydrogenase, encoded by the bnzE gene), which participates in benzene metabolism. In addition it includes Pseudomonas sp. C18 putative 1,2-dihydroxy-1,2-dihydronaphthalene dehydrogenase (aka dibenzothiophene dihydrodiol dehydrogenase, encoded by the doxE gene) which participates in an upper naphthalene catabolic pathway. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>human estrogenic 17beta-hydroxysteroid dehydrogenase type 1 (type 1 17beta-HSD)-like, classical (c) SDRs. 17beta-hydroxysteroid dehydrogenases are a group of isozymes that catalyze activation and inactivation of estrogen and androgens. This classical SDR subgroup includes human type 1 17beta-HSD, human retinol dehydrogenase 8, zebrafish photoreceptor associated retinol dehydrogenase type 2, and a chicken ovary-specific 17beta-hydroxysteroid dehydrogenase. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>2-dehydro-3-deoxy-D-gluconate 5-dehydrogenase KduD. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>enoyl-[acyl-carrier-protein] reductase FabL. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>putative beta-ketoacyl acyl carrier protein [ACP]reductase (BKR), subgroup 2, classical (c) SDR. This subgroup includes Rhizobium sp. NGR234 FabG1. The Escherichai coli K12 BKR, FabG, belongs to a different subgroup. BKR catalyzes the NADPH-dependent reduction of ACP in the first reductive step of de novo fatty acid synthesis (FAS). FAS consists of four elongation steps, which are repeated to extend the fatty acid chain through the addition of two-carbo units from malonyl acyl-carrier protein (ACP): condensation, reduction, dehydration, and a final reduction. Type II FAS, typical of plants and many bacteria, maintains these activities on discrete polypeptides, while type I FAS utilizes one or two multifunctional polypeptides. BKR resembles enoyl reductase, which catalyzes the second reduction step in FAS.  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>putative D-mannonate oxidoreductase, classical (c) SDR. D-mannonate oxidoreductase catalyzes the NAD-dependent interconversion of D-mannonate and D-fructuronate. This subgroup includes Bacillus subtitils UxuB/YjmF, a putative D-mannonate oxidoreductase; the B. subtilis UxuB gene is part of a putative ten-gene operon (the Yjm operon) involved in hexuronate catabolism. Escherichia coli UxuB does not belong to this subgroup. This subgroup has a canonical active site tetrad and a typical Gly-rich NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>D-mannonate oxidoreductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>retinol dehydrogenase (retinol-DH), Light dependent Protochlorophyllide (Pchlide) OxidoReductase (LPOR) and related proteins, classical (c) SDRs. Classical SDR subgroup containing retinol-DHs, LPORs, and related proteins. Retinol is processed by a medium chain alcohol dehydrogenase followed by retinol-DHs. Pchlide reductases act in chlorophyll biosynthesis. There are distinct enzymes that catalyze Pchlide reduction in light or dark conditions. Light-dependent reduction is via an NADP-dependent SDR, LPOR. Proteins in this subfamily share the glycine-rich NAD-binding motif of the classical SDRs, have a partial match to the canonical active site tetrad, but lack the typical active site Ser. This subgroup includes the human proteins: retinol dehydrogenase -12, -13 ,and -14, dehydrogenase/reductase SDR family member (DHRS)-12 , -13 and -X (a DHRS on chromosome X), and WWOX (WW domain-containing oxidoreductase), as well as a Neurospora crassa SDR encoded by the blue light inducible bli-4 gene. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>64</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>(3R)-hydroxyacyl-CoA dehydrogenase-like, classical(c)-like SDRs. Beta oxidation of fatty acids in eukaryotes occurs by a four-reaction cycle, that may take place in mitochondria or in peroxisomes. (3R)-hydroxyacyl-CoA dehydrogenase is part of rat peroxisomal multifunctional MFE-2, it is a member of the NAD-dependent SDRs, but contains an additional small C-terminal domain that completes the active site pocket and participates in dimerization. The atypical, additional C-terminal extension allows for more extensive dimerization contact than other SDRs. MFE-2 catalyzes the second and third reactions of the peroxisomal beta oxidation cycle. Proteins in this subgroup have a typical catalytic triad, but have a His in place of the usual upstream Asn. This subgroup also contains members identified as 17-beta-hydroxysteroid dehydrogenases, including human peroxisomal 17-beta-hydroxysteroid dehydrogenase type 4 (17beta-HSD type 4, aka MFE-2, encoded by HSD17B4 gene) which is involved in fatty acid beta-oxidation and steroid metabolism. This subgroup also includes two SDR domains of the Neurospora crassa and Saccharomyces cerevisiae multifunctional beta-oxidation protein (MFP, aka Fox2).  SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>short chain dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=co>41</td>
            <td id=al>Porcine peroxisomal carbonyl reductase like, classical (c) SDR. This subgroup contains porcine peroxisomal carbonyl reductase and similar proteins. The porcine enzyme efficiently reduces retinals. This subgroup also includes human dehydrogenase/reductase (SDR family) member 4 (DHRS4), and human DHRS4L1. DHRS4 is a peroxisomal enzyme with 3beta-hydroxysteroid dehydrogenase activity; it catalyzes the reduction of 3-keto-C19/C21-steroids into 3beta-hydroxysteroids more efficiently than it does the retinal reduction. The human DHRS4 gene cluster contains DHRS4, DHRS4L2 and DHRS4L1. DHRS4L2 and DHRS4L1 are paralogs of DHRS4, DHRS4L2 being the most recent member. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>3-hydroxybutyrate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>Sorbitol 6-phosphate dehydrogenase (SDH), classical (c) SDRs. Sorbitol 6-phosphate dehydrogenase (SDH, aka glucitol 6-phosphate dehydrogenase) catalyzes the NAD-dependent interconversion of D-fructose 6-phosphate to D-sorbitol 6-phosphate. SDH is a member of the classical SDRs, with the characteristic catalytic tetrad, but without a complete match to the typical NAD-binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>Escherichia coli K-12 YCIK-like, classical (c) SDRs. Escherichia coli K-12 YCIK and related proteins have a canonical classical SDR nucleotide-binding motif and active site tetrad. They are predicted oxoacyl-(acyl carrier protein/ACP) reductases. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase (DHB DH)-like, classical (c) SDR. DHB DH (aka 1,2-dihydroxycyclohexa-3,5-diene-1-carboxylate dehydrogenase) catalyzes the NAD-dependent conversion of 1,2-dihydroxycyclohexa-3,4-diene carboxylate to a catechol. This subgroup also contains Pseudomonas putida F1 CmtB, 2,3-dihydroxy-2,3-dihydro-p-cumate dehydrogenase, the second enzyme in  the pathway for catabolism of p-cumate catabolism. This subgroup shares the glycine-rich NAD-binding motif of the classical SDRs and shares the same catalytic triad; however, the upstream Asn implicated in cofactor binding or catalysis in other SDRs is generally substituted by a Ser. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>classical (c) SDR, subgroup  10. This subgroup resembles the classical SDRs, but has an incomplete match to the canonical glycine rich NAD-binding motif and lacks the typical active site tetrad (instead of the critical active site Tyr, it has Phe, but contains the nearby Lys). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>7-alpha-hydroxysteroid dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>pteridine reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>41</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>retinol dehydrogenases (retinol-DHs), classical (c) SDRs. Classical SDR-like subgroup containing retinol-DHs and related proteins. Retinol is processed by a medium chain alcohol dehydrogenase followed by retinol-DHs. Proteins in this subfamily share the glycine-rich NAD-binding motif of the classical SDRs, have a partial match to the canonical active site tetrad, but lack the typical active site Ser. This subgroup includes the human proteins: retinol dehydrogenase -12, -13 ,and -14. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>Short-chain dehydrogenases/reductases (SDR). SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRs are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human prostaglandin dehydrogenase (PGDH) numbering). In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, PGDH numbering) and/or an Asn (Asn-107, PGDH numbering) contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Extended SDRs have additional elements in the C-terminal region, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase (KR) domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type KRs have a TGXXXGX(1-2)G NAD(P)-binding motif. Some atypical SDRs have lost catalytic activity and/or have an unusual NAD(P)-binding motif and missing or unusual active site residues. Reactions catalyzed within the SDR family include isomerization, decarboxylation, epimerization, C=N bond reduction, dehydratase activity, dehalogenation, Enoyl-CoA reduction, and carbonyl-alcohol oxidoreduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>putative oxoacyl-(acyl carrier protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>67</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>Nucleoside-diphosphate-sugar epimerase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>short chain dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>SDR family oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>7 alpha-hydroxysteroid dehydrogenase (7 alpha-HSDH), classical (c) SDRs. This bacterial subgroup contains 7 alpha-HSDHs,  including Escherichia coli 7 alpha-HSDH. 7 alpha-HSDH, a member of the SDR family, catalyzes the NAD+ -dependent dehydrogenation of a hydroxyl group at position 7 of  the steroid skeleton of bile acids. In humans the two primary bile acids are cholic and chenodeoxycholic acids, these are formed from cholesterol in the liver. Escherichia coli 7 alpha-HSDH dehydroxylates these bile acids in the human intestine. Mammalian 7 alpha-HSDH activity has been found in livers. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold (alpha/beta folding pattern with a central beta-sheet), an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Classical SDRs are typically about 250 residues long, while extended SDRS are approximately 350 residues. Sequence identity between different SDR enzymes are typically in the 15-30% range, but the enzymes share the Rossmann fold NAD-binding motif and characteristic NAD-binding and catalytic sequence patterns. These enzymes have a 3-glycine N-terminal NAD(P)(H)-binding pattern (typically, TGxxxGxG in classical SDRs and TGxxGxxG in extended SDRs), while substrate binding is in the C-terminal region. A critical catalytic Tyr residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase (15-PGDH) numbering), is often found in a conserved YXXXK pattern. In addition to the Tyr and Lys, there is often an upstream Ser (Ser-138, 15-PGDH numbering) and/or an Asn (Asn-107, 15-PGDH numbering) or additional Ser, contributing to the active site. Substrates for these enzymes include sugars, steroids, alcohols, and aromatic compounds. The standard reaction mechanism is a proton relay involving the conserved Tyr and Lys, as well as Asn (or Ser). Some SDR family members, including 17 beta-hydroxysteroid dehydrogenase contain an additional helix-turn-helix motif that is not generally found among SDRs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>3-ketoacyl-(acyl-carrier-protein) reductase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3034526</td></tr>
<tr><td id=al>end</td><td id=al>3035252</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>SDR family oxidoreductase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
