<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04582</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04582 translational GTPase TypA [Phormidium_A]</h3>
    <div>MSLPIRNVAIIAHVDHGKTTLVDALLKQSGIFREGEDVPDCVMDSNDLERERGITILSKNTAVHYKDTLINIVDTPGHADFGGEVERVLGMVDGCILIVDANEGPMPQTRFVLKKALEKGLRPIVVVNKIDRGQADPHVAVDKVLDLFIDLGADDDQCDFPYLFASGLGGFAKDNLDDEDKDMQPLFEAILGHVPPPIGDPKKPLQLQVTTLDYSDYLGRIVIGRIHNGTIKAGQQAMLLQEGEKVVKGKITKLMGFEGLARVELEEATAGMIVAVSGFADANIGETIACPDEPQALPLIKVDEPTLQMTFLVNDSPFAGQEGQFVTSRQLRDRLMRELETNVALRVEPTDSPDKFAVSGRGELHLGILIETMRREGYEFQVSQPQVIYREVNGQPYEPFELLVLDVPEDAVGGCMERIGERKGEMQDMRMTAGRSQLEFVIPARGLIGFRGDFVRITRGDGIMNHSFLEYRRLSGDVETRRNGVLTSFEEGVATFYALKNAEDRGVFFIEPGTKVYKGMIIGENNRPQDLELNVCKTKQLTNHRAASGDELVQLQSPVEMSLERALEYIGADELVEVTPESIRLRKMSAKKLARR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein TypA/BipA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=co>Rivularia</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translational GTPase TypA</td>
            <td id=co>Rivularia</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>COG1217 Predicted membrane GTPase involved in stress response [Signal transduction mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TIGR01394 GTP-binding protein TypA/BipA. This bacterial (and Arabidopsis) protein, termed TypA or BipA, a GTP-binding protein, is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways, but the precise function is unknown. [Regulatory functions, Other, Cellular processes, Adaptations to atypical conditions, Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>99</td>
            <td id=cg>97</td>
            <td id=al>PRK10218 translational GTPase TypA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=al>cd01891 Tyrosine phosphorylated protein A (TypA)/BipA family belongs to ribosome-binding GTPases. BipA is a protein belonging to the ribosome-binding family of GTPases and is widely distributed in bacteria and plants. BipA was originally described as a protein that is induced in Salmonella typhimurium after exposure to bactericidal/permeability-inducing protein (a cationic antimicrobial protein produced by neutrophils), and has since been identified in E. coli as well. The properties thus far described for BipA are related to its role in the process of pathogenesis by enteropathogenic E. coli. It appears to be involved in the regulation of several processes important for infection, including rearrangements of the cytoskeleton of the host, bacterial resistance to host defense peptides, flagellum-mediated cell motility, and expression of K5 capsular genes. It has been proposed that BipA may utilize a novel mechanism to regulate the expression of target genes. In addition, BipA from enteropathogenic E. coli has been shown to be phosphorylated on a tyrosine residue, while BipA from Salmonella and from E. coli K12 strains is not phosphorylated under the conditions assayed. The phosphorylation apparently modifies the rate of nucleotide hydrolysis, with the phosphorylated form showing greatly increased GTPase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=al>pfam00009 Elongation factor Tu GTP binding domain. This domain contains a P-loop motif, also found in several other families such as pfam00071, pfam00025 and pfam00063. Elongation factor Tu consists of three structural domains, this plus two C-terminal beta barrel domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>82</td>
            <td id=cb>70</td>
            <td id=al>COG0480 Translation elongation factor EF-G, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>86</td>
            <td id=cg>85</td>
            <td id=al>COG0481 Translation elongation factor EF-4, membrane-bound GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>80</td>
            <td id=cb>65</td>
            <td id=al>PRK07560 elongation factor EF-2; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>82</td>
            <td id=cb>71</td>
            <td id=al>PRK13351 elongation factor G-like protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cb>79</td>
            <td id=cb>65</td>
            <td id=al>TIGR00490 translation elongation factor aEF-2. This model represents archaeal elongation factor 2, a protein more similar to eukaryotic EF-2 than to bacterial EF-G, both in sequence similarity and in sharing with eukaryotes the property of having a diphthamide (modified His) residue at a conserved position. The diphthamide can be ADP-ribosylated by diphtheria toxin in the presence of NAD. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=al>cd00881 GTP translation factor family primarily contains translation initiation, elongation and release factors. The GTP translation factor family consists primarily of translation initiation, elongation, and release factors, which play specific roles in protein translation. In addition, the family includes Snu114p, a component of the U5 small nuclear riboprotein particle which is a component of the spliceosome and is involved in excision of introns, TetM, a tetracycline resistance gene that protects the ribosome from tetracycline binding, and the unusual subfamily CysN/ATPS, which has an unrelated function (ATP sulfurylase) acquired through lateral transfer of the EF1-alpha gene and development of a new function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cr>13</td>
            <td id=cg>100</td>
            <td id=al>cd16263 Domain III of GTP-binding protein BipA (TypA). BipA (also called TypA) is a highly conserved protein with global regulatory properties in Escherichia coli. BipA is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways. BipA functions as a translation factor that is required specifically for the expression of the transcriptional modulator Fis. BipA binds to ribosomes at a site that coincides with that of EF-G and has a GTPase activity that is sensitive to high GDP:GTP ratios. It is stimulated by 70S ribosomes programmed with mRNA and aminoacylated tRNAs. The growth rate-dependent induction of BipA allows the efficient expression of Fis, thereby modulating a range of downstream processes, including DNA metabolism and type III secretion.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>89</td>
            <td id=cg>89</td>
            <td id=al>PRK05433 GTP-binding protein LepA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=al>cd01885 Elongation Factor 2 (EF2) in archaea and eukarya. Translocation requires hydrolysis of a molecule of GTP and is mediated by EF-G in bacteria and by eEF2 in eukaryotes. The eukaryotic elongation factor eEF2 is a GTPase involved in the translocation of the peptidyl-tRNA from the A site to the P site on the ribosome. The 95-kDa protein is highly conserved, with 60% amino acid sequence identity between the human and yeast proteins. Two major mechanisms are known to regulate protein elongation and both involve eEF2. First, eEF2 can be modulated by reversible phosphorylation. Increased levels of phosphorylated eEF2 reduce elongation rates presumably because phosphorylated eEF2 fails to bind the ribosomes. Treatment of mammalian cells with agents that raise the cytoplasmic Ca2+ and cAMP levels reduce elongation rates by activating the kinase responsible for phosphorylating eEF2. In contrast, treatment of cells with insulin increases elongation rates by promoting eEF2 dephosphorylation. Second, the protein can be post-translationally modified by ADP-ribosylation. Various bacterial toxins perform this reaction after modification of a specific histidine residue to diphthamide, but there is evidence for endogenous ADP ribosylase activity. Similar to the bacterial toxins, it is presumed that modification by the endogenous enzyme also inhibits eEF2 activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>82</td>
            <td id=cb>73</td>
            <td id=al>PRK12740 elongation factor G-like protein EF-G2. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=al>cd01890 LepA also known as Elongation Factor 4 (EF4). LepA (also known as elongation factor 4, EF4) belongs to the GTPase family and exhibits significant homology to the translation factors EF-G and EF-Tu, indicating its possible involvement in translation and association with the ribosome. LepA is ubiquitous in bacteria and eukaryota (e.g. yeast GUF1p), but is missing from archaea. This pattern of phyletic distribution suggests that LepA evolved through a duplication of the EF-G gene in bacteria, followed by early transfer into the eukaryotic lineage, most likely from the promitochondrial endosymbiont. Yeast GUF1p is not essential and mutant cells did not reveal any marked phenotype.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cr>15</td>
            <td id=cg>100</td>
            <td id=al>cd03691 Domain II of BipA. BipA (also called TypA) is a highly conserved protein with global regulatory properties in Escherichia coli.  BipA is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways. BipA functions as a translation factor that is required specifically for the expression of the transcriptional modulator Fis. BipA binds to ribosomes at a site that coincides with that of EF-G and has a GTPase activity that is sensitive to high GDP:GTP ratios and is stimulated  by 70S ribosomes programmed with mRNA and aminoacylated tRNAs. The growth rate-dependent induction of BipA allows the efficient expression of Fis, thereby modulating a range of downstream processes, including DNA metabolism and type III secretion. The domain II of BipA shows similarity to the domain II of the elongation factors (EFs) EF-G and EF-Tu.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cr>23</td>
            <td id=cb>60</td>
            <td id=al>cd04168 Tet(M)-like family includes Tet(M), Tet(O), Tet(W), and OtrA, containing tetracycline resistant proteins. Tet(M), Tet(O), Tet(W), and OtrA are tetracycline resistance genes found in Gram-positive and Gram-negative bacteria. Tetracyclines inhibit protein synthesis by preventing aminoacyl-tRNA from binding to the ribosomal acceptor site. This subfamily contains tetracycline resistance proteins that function through ribosomal protection and are typically found on mobile genetic elements, such as transposons or plasmids, and are often conjugative. Ribosomal protection proteins are homologous to the elongation factors EF-Tu and EF-G. EF-G and Tet(M) compete for binding on the ribosomes. Tet(M) has a higher affinity than EF-G, suggesting these two proteins may have overlapping binding sites and that Tet(M) must be released before EF-G can bind. Tet(M) and Tet(O) have been shown to have ribosome-dependent GTPase activity. These proteins are part of the GTP translation factor family, which includes EF-G, EF-Tu, EF2, LepA, and SelB.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>82</td>
            <td id=cb>71</td>
            <td id=al>PRK12739 elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>23</td>
            <td id=cr>16</td>
            <td id=al>PTZ00416 elongation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cr>23</td>
            <td id=co>51</td>
            <td id=al>cd01886 Elongation factor G (EF-G) family involved in both the elongation and ribosome recycling phases of protein synthesis. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains both eukaryotic and bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cr>12</td>
            <td id=cg>96</td>
            <td id=al>cd03710 BipA_TypA_C: a C-terminal portion of BipA or TypA having homology to the C terminal domains of the elongation factors EF-G and EF-2. A member of the ribosome binding GTPase superfamily, BipA is widely distributed in bacteria and plants.  BipA is a highly conserved protein with global regulatory properties in Escherichia coli. BipA is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways. BipA functions as a translation factor that is required specifically for the expression of the transcriptional modulator Fis.  BipA binds to ribosomes at a site that coincides with that of EF-G and has a GTPase activity that is sensitive to high GDP:GTP ratios and, is stimulated  by 70S ribosomes programmed with mRNA and aminoacylated tRNAs. The growth rate-dependent induction of BipA allows the efficient expression of Fis, thereby modulating a range of downstream processes, including DNA metabolism and type III secretion.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>84</td>
            <td id=cb>72</td>
            <td id=al>PRK00007 elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>22</td>
            <td id=cb>64</td>
            <td id=al>cd04167 Snu114p, a spliceosome protein, is a GTPase. Snu114p subfamily. Snu114p is one of several proteins that make up the U5 small nuclear ribonucleoprotein (snRNP) particle. U5 is a component of the spliceosome, which catalyzes the splicing of pre-mRNA to remove introns. Snu114p is homologous to EF-2, but typically contains an additional N-terminal domain not found in Ef-2. This protein is part of the GTP translation factor family and the Ras superfamily, characterized by five G-box motifs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>24</td>
            <td id=cr>17</td>
            <td id=al>PLN00116 translation elongation factor EF-2 subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>54</td>
            <td id=cb>76</td>
            <td id=al>COG5256 Translation elongation factor EF-1alpha (GTPase)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>24</td>
            <td id=co>54</td>
            <td id=al>cd04170 Elongation factor G (EF-G) family. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains only bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>47</td>
            <td id=cb>71</td>
            <td id=al>COG0050 Translation elongation factor EF-Tu, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>23</td>
            <td id=co>51</td>
            <td id=al>cd04169 Release Factor 3 (RF3) protein involved in the terminal step of translocation in bacteria. Peptide chain release factor 3 (RF3) is a protein involved in the termination step of translation in bacteria. Termination occurs when class I release factors (RF1 or RF2) recognize the stop codon at the A-site of the ribosome and activate the release of the nascent polypeptide. The class II release factor RF3 then initiates the release of the class I RF from the ribosome. RF3 binds to the RF/ribosome complex in the inactive (GDP-bound) state. GDP/GTP exchange occurs, followed by the release of the class I RF. Subsequent hydrolysis of GTP to GDP triggers the release of RF3 from the ribosome. RF3 also enhances the efficiency of class I RFs at less preferred stop codons and at stop codons in weak contexts.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>23</td>
            <td id=cr>26</td>
            <td id=al>COG4108 Peptide chain release factor RF-3 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>48</td>
            <td id=cb>73</td>
            <td id=al>PRK12736 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>45</td>
            <td id=cb>63</td>
            <td id=al>PRK12317 elongation factor 1-alpha; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=al>cd01884 Elongation Factor Tu (EF-Tu) GTP-binding proteins. EF-Tu subfamily. This subfamily includes orthologs of translation elongation factor EF-Tu in bacteria, mitochondria, and chloroplasts. It is one of several GTP-binding translation factors found in the larger family of GTP-binding elongation factors. The eukaryotic counterpart, eukaryotic translation elongation factor 1 (eEF-1 alpha), is excluded from this family. EF-Tu is one of the most abundant proteins in bacteria, as well as, one of the most highly conserved, and in a number of species the gene is duplicated with identical function. When bound to GTP, EF-Tu can form a complex with any (correctly) aminoacylated tRNA except those for initiation and for selenocysteine, in which case EF-Tu is replaced by other factors. Transfer RNA is carried to the ribosome in these complexes for protein translation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>48</td>
            <td id=cb>73</td>
            <td id=al>TIGR00485 translation elongation factor TU. This model models orthologs of translation elongation factor EF-Tu in bacteria, mitochondria, and chloroplasts, one of several GTP-binding translation factors found by the more general pfam model GTP_EFTU. The eukaryotic conterpart, eukaryotic translation elongation factor 1 (eEF-1 alpha), is excluded from this model. EF-Tu is one of the most abundant proteins in bacteria, as well as one of the most highly conserved, and in a number of species the gene is duplicated with identical function. When bound to GTP, EF-Tu can form a complex with any (correctly) aminoacylated tRNA except those for initiation and for selenocysteine, in which case EF-Tu is replaced by other factors. Transfer RNA is carried to the ribosome in these complexes for protein translation. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>43</td>
            <td id=co>58</td>
            <td id=al>PLN03127 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>49</td>
            <td id=cb>72</td>
            <td id=al>CHL00071 elongation factor Tu</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>43</td>
            <td id=cb>64</td>
            <td id=al>PRK00049 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>24</td>
            <td id=cg>90</td>
            <td id=al>TIGR00231 small GTP-binding protein domain. Proteins with a small GTP-binding domain recognized by this model include Ras, RhoA, Rab11, translation elongation factor G, translation initiation factor IF-2, tetratcycline resistance protein TetM, CDC42, Era, ADP-ribosylation factors, tdhF, and many others. In some proteins the domain occurs more than once.This model recognizes a large number of small GTP-binding proteins and related domains in larger proteins. Note that the alpha chains of heterotrimeric G proteins are larger proteins in which the NKXD motif is separated from the GxxxxGK[ST] motif (P-loop) by a long insert and are not easily detected by this model. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>41</td>
            <td id=cb>62</td>
            <td id=al>PRK12735 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>14</td>
            <td id=cg>97</td>
            <td id=al>pfam00679 Elongation factor G C-terminus. This domain includes the carboxyl terminal regions of Elongation factor G, elongation factor 2 and some tetracycline resistance proteins and adopt a ferredoxin-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>22</td>
            <td id=cb>78</td>
            <td id=al>cd01887 Initiation Factor 2 (IF2)/ eukaryotic Initiation Factor 5B (eIF5B) family. IF2/eIF5B contribute to ribosomal subunit joining and function as GTPases that are maximally activated by the presence of both ribosomal subunits. As seen in other GTPases, IF2/IF5B undergoes conformational changes between its GTP- and GDP-bound states. Eukaryotic IF2/eIF5Bs possess three characteristic segments, including a divergent N-terminal region followed by conserved central and C-terminal segments. This core region is conserved among all known eukaryotic and archaeal IF2/eIF5Bs and eubacterial IF2s.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=al>cd01883 Elongation Factor 1-alpha (EF1-alpha) protein family. EF1 is responsible for the GTP-dependent binding of aminoacyl-tRNAs to the ribosomes. EF1 is composed of four subunits: the alpha chain which binds GTP and aminoacyl-tRNAs, the gamma chain that probably plays a role in anchoring the complex to other cellular components and the beta and delta (or beta') chains. This subfamily is the alpha subunit, and represents the counterpart of bacterial EF-Tu for the archaea (aEF1-alpha) and eukaryotes (eEF1-alpha). eEF1-alpha interacts with the actin of the eukaryotic cytoskeleton and may thereby play a role in cellular transformation and apoptosis. EF-Tu can have no such role in bacteria. In humans, the isoform eEF1A2 is overexpressed in 2/3 of breast cancers and has been identified as a putative oncogene. This subfamily also includes Hbs1, a G protein known to be important for efficient growth and protein synthesis under conditions of limiting translation initiation in yeast, and to associate with Dom34. It has been speculated that yeast Hbs1 and Dom34 proteins may function as part of a complex with a role in gene expression.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>13</td>
            <td id=cg>98</td>
            <td id=al>cd01514 Elongation factor G C-terminus. This domain includes the carboxyl terminal regions of elongation factors (EFs) bacterial EF-G, eukaryotic and archeal EF-2 and eukaryotic mitochondrial mtEFG1s and mtEFG2s. This group also includes proteins similar to the ribosomal protection proteins Tet(M) and Tet(O), BipA, LepA and, spliceosomal proteins: human 116kD U5 small nuclear ribonucleoprotein (snRNP) protein (U5-116 kD) and yeast counterpart Snu114p.  This domain adopts a ferredoxin-like fold consisting of an alpha-beta sandwich with anti-parallel beta-sheets, resembling the topology of domain III found in the elongation factors EF-G and eukaryotic EF-2, with which it forms the C-terminal block. The two domains however are not superimposable and domain III lacks some of the characteristics of this domain.  EF-2/EF-G in complex with GTP, promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site, the uncharged tRNA from the P site to the E-site and, the mRNA is shifted one codon relative to the ribosome. Tet(M) and Tet(O) mediate Tc resistance. Typical Tcs bind to the ribosome and inhibit the elongation phase of protein synthesis, by inhibiting the occupation of site A by aminoacyl-tRNA. Tet(M) and Tet(O) catalyze the release of tetracycline (Tc) from the ribosome in a GTP-dependent manner.  BipA is a highly conserved protein with global regulatory properties in Escherichia coli. Yeast Snu114p is essential for cell viability and for splicing in vivo. Experiments suggest that GTP binding and probably GTP hydrolysis is important for the function of the U5-116 kD/Snu114p. The function of LepA proteins is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>45</td>
            <td id=co>46</td>
            <td id=al>TIGR00475 selenocysteine-specific elongation factor SelB. In prokaryotes, the incorporation of selenocysteine as the 21st amino acid, encoded by TGA, requires several elements: SelC is the tRNA itself, SelD acts as a donor of reduced selenium, SelA modifies a serine residue on SelC into selenocysteine, and SelB is a selenocysteine-specific translation elongation factor. 3-prime or 5-prime non-coding elements of mRNA have been found as probable structures for directing selenocysteine incorporation. This model describes the elongation factor SelB, a close homolog rf EF-Tu. It may function by replacing EF-Tu. A C-terminal domain not found in EF-Tu is in all SelB sequences in the seed alignment except that from Methanococcus jannaschii. This model does not find an equivalent protein for eukaryotes. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>48</td>
            <td id=cb>60</td>
            <td id=al>PLN03126 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>23</td>
            <td id=cr>27</td>
            <td id=al>PRK00741 peptide chain release factor 3; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>54</td>
            <td id=cb>74</td>
            <td id=al>COG2895 Sulfate adenylyltransferase subunit 1, EFTu-like GTPase family  [Inorganic ion transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>23</td>
            <td id=cr>18</td>
            <td id=al>CHL00189 translation initiation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>21</td>
            <td id=cb>75</td>
            <td id=al>cd04171 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec, and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains bacterial SelBs, as well as, one from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>24</td>
            <td id=cb>75</td>
            <td id=al>cd01889 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains eukaryotic SelBs and some from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>24</td>
            <td id=cr>19</td>
            <td id=al>PRK05306 translation initiation factor IF-2; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=al>cd00882 Rat sarcoma (Ras)-like superfamily of small guanosine triphosphatases (GTPases). Ras-like GTPase superfamily. The Ras-like superfamily of small GTPases consists of several families with an extremely high degree of structural and functional similarity. The Ras superfamily is divided into at least four families in eukaryotes: the Ras, Rho, Rab, and Sar1/Arf families. This superfamily also includes proteins like the GTP translation factors, Era-like GTPases, and G-alpha chain of the heterotrimeric G proteins. Members of the Ras superfamily regulate a wide variety of cellular functions: the Ras family regulates gene expression, the Rho family regulates cytoskeletal reorganization and gene expression, the Rab and Sar1/Arf families regulate vesicle trafficking, and the Ran family regulates nucleocytoplasmic transport and microtubule organization. The GTP translation factor family regulates initiation, elongation, termination, and release in translation, and the Era-like GTPase family regulates cell division, sporulation, and DNA replication. Members of the Ras superfamily are identified by the GTP binding site, which is made up of five characteristic sequence motifs, and the switch I and switch II regions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>22</td>
            <td id=cr>29</td>
            <td id=al>COG3276 Selenocysteine-specific translation elongation factor [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>12</td>
            <td id=cg>100</td>
            <td id=al>cd16257 Domain III of Elongation factor G (EF-G) and related proteins. Bacterial Elongation factor G (EF-G) and related proteins play a role in translation and share a similar domain architecture. Elongation factor EFG participates in the elongation phase during protein biosynthesis on the ribosome by stimulating translocation. Its functional cycles depend on GTP binding and its hydrolysis. Domain III is involved in the activation of GTP hydrolysis. This domain III, which is different from domain III in EF-TU and related elongation factors, is found in several translation factors, like bacterial release factors RF3, elongation factor 4, elongation factor 2, GTP-binding protein BipA and tetracycline resistance protein Tet.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>46</td>
            <td id=cb>62</td>
            <td id=al>PTZ00141 elongation factor 1- alpha; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>25</td>
            <td id=cr>29</td>
            <td id=al>COG0532 Translation initiation factor IF-2, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>25</td>
            <td id=cb>74</td>
            <td id=al>cd04166 CysN, together with protein CysD, forms the ATP sulfurylase (ATPS) complex. CysN_ATPS subfamily. CysN, together with protein CysD, form the ATP sulfurylase (ATPS) complex in some bacteria and lower eukaryotes. ATPS catalyzes the production of ATP sulfurylase (APS) and pyrophosphate (PPi) from ATP and sulfate. CysD, which catalyzes ATP hydrolysis, is a member of the ATP pyrophosphatase (ATP PPase) family. CysN hydrolysis of GTP is required for CysD hydrolysis of ATP; however, CysN hydrolysis of GTP is not dependent on CysD hydrolysis of ATP. CysN is an example of lateral gene transfer followed by acquisition of new function. In many organisms, an ATPS exists which is not GTP-dependent and shares no sequence or structural similarity to CysN.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>12</td>
            <td id=cg>100</td>
            <td id=al>pfam03144 Elongation factor Tu domain 2. Elongation factor Tu consists of three structural domains, this is the second domain. This domain adopts a beta barrel structure. This the second domain is involved in binding to charged tRNA. This domain is also found in other proteins such as elongation factor G and translation initiation factor IF-2. This domain is structurally related to pfam03143, and in fact has weak sequence matches to this domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>pfam01926 50S ribosome-binding GTPase. The full-length GTPase protein is required for the complete activity of the protein of interacting with the 50S ribosome and binding of both adenine and guanine nucleotides, with a preference for guanine nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>44</td>
            <td id=co>58</td>
            <td id=al>PLN00043 elongation factor 1-alpha; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>12</td>
            <td id=cg>89</td>
            <td id=al>cd04088 Domain II of bacterial elongation factor G and C-terminal domain of mitochondrial Elongation factors G1 and G2. This family represents the domain II of bacterial Elongation factor G (EF-G)and mitochondrial Elongation factors G1 (mtEFG1) and G2 (mtEFG2). During the process of peptide synthesis and tRNA site changes, the ribosome is moved along the mRNA a distance equal to one codon with the addition of each amino acid. In bacteria this translocation step is catalyzed by EF-G_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. mtEFG1 and mtEFG2 show significant homology to bacterial EF-Gs. Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. No clear phenotype has been found for mutants in the yeast homolog of mtEFG2, MEF2.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3734725</td></tr>
<tr><td id=al>end</td><td id=al>3736516</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>translational GTPase TypA</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>597</td></tr>
</table>

    </div>
</body>
</html>
