<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02091</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02091 dTDP-4-dehydrorhamnose reductase [Phormidium_A]</h3>
    <div>MRILLTGVNGQLGGELQKTLVPLGEVVGVGRDRLDLSQSDSIPPVVEAIQPDLIINCAAYTAVDKAETEVELAHQVNAIAPGTLAKAASQVGATLLHLSTDYVFDGTQSHPYVETDTTNPIGEYGKSKFAGEMAILQTGAPHIILRTAWVYGVGGTGNFVKTMLRLGKDREEIRVVADQVGSPSWTGDIAGAIAKIAGRHQHLEGTYHFTNSGVASWYDFAVAIFEEAAALGFPLQVKRVVPITTADYPTPAQRPAYSVLSGKKLEAILKEPAPHWRQGLRQMLAGLQAALTDEL</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase RmlD</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>TPA: dTDP-4-dehydrorhamnose reductase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>96</td>
            <td id=cg>98</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>dTDP-4-dehydrorhamnose reductase</td>
            <td id=co>Limnospira</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>51</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>pfam04321 RmlD substrate binding domain. L-rhamnose is a saccharide required for the virulence of some bacteria. Its precursor, dTDP-L-rhamnose, is synthesized by four different enzymes the final one of which is RmlD. The RmlD substrate binding domain is responsible for binding a sugar nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=al>TIGR01214 dTDP-4-dehydrorhamnose reductase. This enzyme catalyzes the last of 4 steps in making dTDP-rhamnose, a precursor of LPS core antigen, O-antigen, etc. [Cell envelope, Biosynthesis and degradation of surface polysaccharides and lipopolysaccharides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=al>COG1091 dTDP-4-dehydrorhamnose reductase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=al>cd05254 dTDP-6-deoxy-L-lyxo-4-hexulose reductase and related proteins, extended (e) SDRs. dTDP-6-deoxy-L-lyxo-4-hexulose reductase, an extended SDR, synthesizes dTDP-L-rhamnose from alpha-D-glucose-1-phosphate,  providing the precursor of L-rhamnose, an essential cell wall component of many pathogenic bacteria. This subgroup has the characteristic active site tetrad and NADP-binding motif. This subgroup also contains human MAT2B, the regulatory subunit of methionine adenosyltransferase (MAT); MAT catalyzes S-adenosylmethionine synthesis. The human gene encoding MAT2B encodes two major splicing variants which are induced in human cell liver cancer and regulate HuR, an mRNA-binding protein which stabilizes the mRNA of several cyclins, to affect cell proliferation. Both MAT2B variants include this extended SDR domain. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>98</td>
            <td id=cg>97</td>
            <td id=al>PRK09987 dTDP-4-dehydrorhamnose reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=al>cd08946 extended (e) SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG0451 Nucleoside-diphosphate-sugar epimerase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cb>63</td>
            <td id=cb>61</td>
            <td id=al>cd05232 UDP-glucose 4 epimerase, subgroup 4, extended (e) SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup is comprised of bacterial proteins, and includes the Staphylococcus aureus capsular polysaccharide Cap5N, which may have a role in the synthesis of UDP-N-acetyl-d-fucosamine. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>62</td>
            <td id=cb>78</td>
            <td id=al>pfam01370 NAD dependent epimerase/dehydratase family. This family of proteins utilize NAD as a cofactor. The proteins in this family use nucleotide-sugar substrates for a variety of chemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>86</td>
            <td id=cg>87</td>
            <td id=al>cd05263 Pseudomonas fluorescens MupV-like, extended (e) SDRs. This subgroup of extended SDR family domains have the characteristic active site tetrad and a well-conserved NAD(P)-binding motif. This subgroup is not well characterized, its members are annotated as having a variety of putative functions. One characterized member is Pseudomonas fluorescens MupV a protein  involved in the biosynthesis of Mupirocin, a polyketide-derived antibiotic. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=al>cd05226 Extended (e) and atypical (a) SDRs. Extended or atypical short-chain dehydrogenases/reductases (SDRs, aka tyrosine-dependent oxidoreductases) are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>54</td>
            <td id=co>50</td>
            <td id=al>cd05260 GDP-mannose 4,6 dehydratase, extended (e) SDRs. GDP-mannose 4,6 dehydratase, a homodimeric SDR, catalyzes the NADP(H)-dependent conversion of GDP-(D)-mannose to GDP-4-keto, 6-deoxy-(D)-mannose in the fucose biosynthesis pathway. These proteins have the canonical active site triad and NAD-binding pattern, however the active site Asn is often missing and may be substituted with Asp. A Glu residue has been identified as an important active site base. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>cd05271 NADH dehydrogenase (ubiquinone) 1 alpha subcomplex, subunit 9, 39 kDa, (NDUFA9) -like, atypical (a) SDRs. This subgroup of extended SDR-like proteins are atypical SDRs. They have a glycine-rich NAD(P)-binding motif similar to the typical SDRs, GXXGXXG, and have the YXXXK active site motif (though not the other residues of the SDR tetrad). Members identified include NDUFA9 (mitochondrial) and putative nucleoside-diphosphate-sugar epimerase. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=co>59</td>
            <td id=co>57</td>
            <td id=al>cd08957 Bordetella bronchiseptica enzymes WbmH and WbmG-like, extended (e) SDRs. Bordetella bronchiseptica enzymes WbmH and WbmG, and related proteins. This subgroup exhibits the active site tetrad and NAD-binding motif of the extended SDR family. It has been proposed that the active site in Bordetella WbmG and WbmH cannot function as an epimerase, and that it plays a role in O-antigen synthesis pathway from UDP-2,3-diacetamido-2,3-dideoxy-l-galacturonic acid. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>53</td>
            <td id=co>48</td>
            <td id=al>pfam16363 GDP-mannose 4,6 dehydratase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>67</td>
            <td id=cb>65</td>
            <td id=al>cd05256 UDP-N-acetylglucosamine 4-epimerase, extended (e) SDRs. This subgroup contains UDP-N-acetylglucosamine 4-epimerase of Pseudomonas aeruginosa, WbpP,  an extended SDR, that catalyzes the NAD+ dependent conversion of UDP-GlcNAc and UDPGalNA to UDP-Glc and UDP-Gal.  This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1767134</td></tr>
<tr><td id=al>end</td><td id=al>1768022</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>dTDP-4-dehydrorhamnose reductase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>296</td></tr>
</table>

    </div>
</body>
</html>
