<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03840</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03840 ATP-dependent Clp protease proteolytic subunit [Phormidium_A]</h3>
    <div>MPIGVPKVPYQMPGDTYTQWIDIYNRLYRERIIFIGRDIDDELANQVIAVMLYLASEDPDKDIMVYINSPGGVVSSGMAIYDTMQHIKADVVTICVGLAASMGSFLLSAGTPGKRLALPHSRIMIHQPSGGTRGQASDIQIEAKEILRIRHQLNGIYAANTGQTIEKIEKDMDRDFFLSAEQAKEYGLIDRVLEGKPV</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease, protease subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=co>Hassallia</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=co>Hassallia</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>ATP-dependent Clp protease proteolytic subunit</td>
            <td id=co>Hassallia</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>PRK00277 ATP-dependent Clp protease proteolytic subunit; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>96</td>
            <td id=cg>96</td>
            <td id=al>CHL00028 ATP-dependent Clp protease proteolytic subunit</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=al>pfam00574 Clp protease. The Clp protease has an active site catalytic triad. In E. coli Clp protease, ser-111, his-136 and asp-185 form the catalytic triad. Some members have lost active site residues and are therefore inactive, some contain one or two large insertions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=al>cd07017 Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease. Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>95</td>
            <td id=cg>95</td>
            <td id=al>COG0740 ATP-dependent protease ClpP, protease subunit [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>98</td>
            <td id=cg>94</td>
            <td id=al>PRK12553 ATP-dependent Clp protease proteolytic subunit; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>86</td>
            <td id=cg>89</td>
            <td id=al>TIGR00493 ATP-dependent Clp endopeptidase, proteolytic subunit ClpP. This model for the proteolytic subunit ClpP has been rebuilt to a higher stringency. In every bacterial genome with the ClpXP machine, a ClpP protein will be found that scores well with this model. In general, this ClpP member will be encoded adjacent to the clpX gene, as were all examples used in the seed alignment. A large fraction of genomes have one or more additional ClpP paralogs, sometimes encoded nearby and sometimes elsewhere. The stringency of the trusted cutoff used here excludes the more divergent ClpP paralogs from being called authentic ClpP by this model. [Protein fate, Degradation of proteins, peptides, and glycopeptides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>88</td>
            <td id=cg>90</td>
            <td id=al>PRK12551 ATP-dependent Clp protease proteolytic subunit; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>88</td>
            <td id=cg>80</td>
            <td id=al>PRK14514 ATP-dependent Clp endopeptidase proteolytic subunit ClpP. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>86</td>
            <td id=cg>86</td>
            <td id=al>PRK14513 ATP-dependent Clp protease proteolytic subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>PRK12552 ATP-dependent Clp protease proteolytic subunit. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=al>cd07013 Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease. Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. Additionally, they are implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>83</td>
            <td id=cg>84</td>
            <td id=al>PRK14512 ATP-dependent Clp protease proteolytic subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=al>cd00394 Caseinolytic protease (ClpP) is an ATP-dependent protease. Clp protease (caseinolytic protease; ClpP; endopeptidase Clp; Peptidase S14; ATP-dependent protease, ClpAP)-like enzymes are highly conserved serine proteases and belong to the ClpP/Crotonase superfamily. Included in this family are Clp proteases that are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. The functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP. Active site consists of the triad Ser, His and Asp, preferring hydrophobic or non-polar residues at P1 or P1' positions. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function. Another family included in this class of enzymes is the signal peptide peptidase A (SppA; S49) which is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Mutagenesis studies suggest that the catalytic center of SppA comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. In addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members, the E. coli SppA contains an amino-terminal domain. Others, including sohB peptidase, protein C, protein 1510-N and archaeal signal peptide peptidase, do not contain the amino-terminal domain. The third family included in this hierarchy is nodulation formation efficiency D (NfeD) which is a membrane-bound Clp-class protease and only found in bacteria and archaea. Majority of the NfeD genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named stomatin operon partner protein (STOPP). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 from Pyrococcus horikoshii has been shown to possess serine protease activity having a Ser-Lys catalytic dyad.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cb>79</td>
            <td id=cg>98</td>
            <td id=al>cd07016 Caseinolytic protease (ClpP) is an ATP-dependent, highly conserved serine protease. Clp protease (caseinolytic protease; ClpP; Peptidase S14) is a highly conserved serine protease present throughout in bacteria and eukaryota, but seems to be absent in archaea, mollicutes and some fungi. This subfamily only contains bacterial sequences. Clp proteases are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. ClpP has also been linked to the tight regulation of virulence genes in the pathogens Listeria monocytogenes and Salmonella typhimurium. This enzyme belong to the family of ATP-dependent proteases; the functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP, although the proteolytic subunit alone does possess some catalytic activity. Active site consists of the triad Ser, His and Asp; some members have lost all of these active site residues and are therefore inactive, while others may have one or two large insertions. ClpP seems to prefer hydrophobic or non-polar residues at P1 or P1' positions in its substrate. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3239156</td></tr>
<tr><td id=al>end</td><td id=al>3239753</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>ATP-dependent Clp protease proteolytic subunit</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>199</td></tr>
</table>

    </div>
</body>
</html>
