<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03525</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03525 excinuclease ABC subunit UvrB [Phormidium_A]</h3>
    <div>MTSFQFTSPFEATGDQPQAIAKLSHNLQKNSRFQTLLGATGTGKTYTMARVIENLGKPTLVLAHNKTLAAQLCNELRGFFPNNAVEYFVSYYDYYQPEAYLPVTDTYIEKTASINDEIDMLRHSATRSLFEQRDVIVVASISCIYGLGIPSEYLNASILFQVGQEIDQRQLLRDLVTIQYQRNDLEMGRGKFRVKGDVVEIGPAYEDRLIRIEFFGDEIDAIRYIDPVTGQVLYSVNTLRIYPARHFVTPGDRLDMACDDIQRELETQLTLLENAGKLLEAQRLEQRTRYDLEMLREVGYTNGVENYSRHLAGRQAGEPPECLVDYFPEDWLLVVDESHVTVPQIRGMYNGDQARKRILIDHGFRLPSAADNRPLKADEFWRKVRQCVFVSATPGDWEIEQSDGQVVEQIIRPTGVVDPEIFVRPTEGQVDDLLGEIRDRISRKERTLVTTLTKRMAEDLTEYFQERGVLVRYLHSDINSIERIEILQSLRQGEFDVLIGVNLLREGLDLPEVSLVAILDADKEGFLRAERSLIQTIGRAARHVRGQAILYGDNLTKSMEKAISETQRRREIQLAYNDKHGITPTSIVKSIDNSILSFLEISRRLNRQQLDTAYEQVDEIPLSQIPELIQQLEAEMQEAAKNLEFETAASLRDRIKHLRDKLVGRRPEP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=co>LEGE-07092</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=cr>Cyanothece</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=co>Chroogloeocystis</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=co>Chroogloeocystis</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>excinuclease ABC subunit UvrB</td>
            <td id=co>Chroogloeocystis</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>excinuclease ABC subunit UvrB. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Excinuclease UvrABC helicase subunit UvrB [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>excinuclease ABC, B subunit. All proteins in this family for wich functions are known are DNA helicases that function in the nucleotide excision repair and are endonucleases that make the 3' incision next to DNA damage. They are part of a pathway requiring UvrA, UvrB, UvrC, and UvrD homologs. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University) [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DEXH-box helicase domain of excinuclease ABC subunit B. Excinuclease ABC subunit B (or UvrB) plays a central role in nucleotide excision repair (NER). Together with other components of the NER system, like UvrA, UvrC, UvrD (helicase II) and DNA polymerase I, it recognizes and cleaves damaged DNA in a multistep ATP-dependent reaction. UvrB is critical for the second phase of damage recognition by verifying the nature of the damage and forming the pre-incision complex. Its ATPase site becomes activated in the presence of UvrA and damaged DNA, but its activity is strand destabilization via distortion of the DNA at lesion site, with very limited DNA unwinding. UvrB is a member of the DEAD-like helicases, a diverse family of proteins involved in ATP-dependent RNA unwinding, needed in a variety of cellular processes including splicing, ribosome biogenesis and RNA degradation. The name derives from the sequence of the Walker B motif (motif II). This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>C-terminal helicase domain of the UvrB family helicases. Excinuclease ABC subunit B (or UvrB) plays a central role in nucleotide excision repair (NER). Together with other components of the NER system, like UvrA, UvrC, UvrD (helicase II), and DNA polymerase I, it recognizes and cleaves damaged DNA in a multistep ATP-dependent reaction. UvrB is critical for the second phase of damage recognition by verifying the nature of the damage and forming the pre-incision complex. Its ATPase site becomes activated in the presence of UvrA and damaged DNA. Its activity is strand destabilization via distortion of the DNA at lesion site, with very limited DNA unwinding. UvrB is a DEAD-like helicase belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>UvrB interaction domain. This domain is found in the UvrB protein where it interacts with the UvrA protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>Transcription-repair coupling factor (superfamily II helicase) [Replication, recombination and repair, Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Ultra-violet resistance protein B. This domain family is found in bacteria, archaea and eukaryotes, and is approximately 40 amino acids in length. The family is found in association with pfam00271, pfam02151, pfam04851. There are two conserved sequence motifs: YAD and RRR. This family is the C terminal region of the UvrB protein which conveys mutational resistance against UV light to various different species.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helicase conserved C-terminal domain. The Prosite family is restricted to DEAD/H helicases, whereas this domain family is found in a wide variety of helicases and helicase related proteins. It may be that this is not an autonomously folding unit, but an integral part of the helicase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>helicase superfamily c-terminal domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>39</td>
            <td id=al>Superfamily II DNA and RNA helicase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>C-terminal helicase domain of the DEAD box helicases. DEAD-box helicases comprise a diverse family of proteins involved in ATP-dependent RNA unwinding, needed in a variety of cellular processes including splicing, ribosome biogenesis, and RNA degradation. They are superfamily (SF)2 helicases that, similar to SF1, do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>Superfamily II DNA or RNA helicase [Transcription, Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>C-terminal helicase domain of ATP-dependent helicase PriA. PriA, also known as replication factor Y or primosomal protein N', is a 3'-->5' DNA helicase that acts to remodel stalled replication forks and as a specificity factor for origin-independent assembly of a new replisome at the stalled fork. PriA is a DEAD-like helicase belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>primosome assembly protein PriA; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>Superfamily II DNA/RNA helicase required for DNA uptake (late competence protein)  [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>C-terminal helicase domain of EcoAI HsdR-like restriction enzyme family helicases. This family is composed of helicase restriction enzymes, including the HsdR subunit of restriction-modification enzymes such as Escherichia coli type I restriction enzyme EcoAI R protein (R.EcoAI). The EcoAI enzyme recognizes 5'-GAGN(7)GTCA-3'. The HsdR or R subunit is required for both nuclease and ATPase activities, but not for modification. These proteins are DEAD-like helicases belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>Type III restriction enzyme, res subunit. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>N-terminal DEAD/H-box helicase domain of superfamily 2 helicases. The DEAD/H-like superfamily 2 helicases comprise a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This N-terminal domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>DEAD-like helicases superfamily. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cr>5</td>
            <td id=al>excinuclease ABC subunit UvrC. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>UvrB/uvrC motif. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>Primosomal protein N' (replication factor Y) - superfamily II helicase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2984031</td></tr>
<tr><td id=al>end</td><td id=al>2986041</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>excinuclease ABC subunit UvrB</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
