<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04939</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04939 S-methyl-5'-thioadenosine phosphorylase [Phormidium_A]</h3>
    <div>MTDLAIGIIGGSGLYQMDDLENRQDHQLDTPFGPPSDALISGTLQGVPVVFLARHGRNHHLLPSELPFRANIHAFKQLGVQYLISASAVGSLAAAIKPLDMVVPDQFIDRTQHRTASFFGEGIVAHIGFADPVCPQLASLIAETASSLPLDGVDVHSGGTYICMEGPAFSTRAESHLYRSWGASVIGMTNLTEAKLAREAEIAYATLALVTDYDCWHPDHDSVTVEMVVNNLRLNAHNAQQVIRETVRRLHDAPFVSEAHSALKFAVLTPWDKVPETTKGKLELLLRPYL</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>5'-methylthioadenosine phosphorylase MtaP</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=co>Pleurocapsa</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=co>SIO2C1</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=co>Phormidium_B</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase</td>
            <td id=co>Coleofasciculus</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>5'-deoxy-5'-methylthioadenosine phosphorylases (MTAP) similar to Sulfolobus solfataricus MTAPII and Pseudomonas aeruginosa PAO1 5'-methylthioinosine phosphorylase (MTIP). MTAP catalyzes the reversible phosphorolysis of 5'-deoxy-5'-methylthioadenosine (MTA) to adenine and 5-methylthio-D-ribose-1-phosphate. This subfamily includes human MTAP which is highly specific for MTA, and Sulfolobus solfataricus MTAPII which accepts adenosine in addition to MTA. Two MTAPs have been isolated from S. solfataricus: SsMTAP1 and SsMTAPII, SsMTAP1 belongs to a different subfamily of the nucleoside phosphorylase-I (NP-I) family. This group also includes Pseudomonas aeruginosa PAO1 MTI phosphorylase (MTIP) which uses 5'-methylthioinosine (MTI) as a preferred substrate, and does not use MTA. NP-I family members accept a range of purine nucleosides as well as the pyrimidine nucleoside uridine. The NP-1 family  includes phosphorolytic nucleosidases, such as purine nucleoside phosphorylase (PNPs, EC. 2.4.2.1), uridine phosphorylase (UP, EC 2.4.2.3), and 5'-deoxy-5'-methylthioadenosine phosphorylase (MTAP, EC 2.4.2.28), and hydrolytic nucleosidases, such as AMP nucleosidase (AMN, EC 3.2.2.4), and 5'-methylthioadenosine/S-adenosylhomocysteine (MTA/SAH) nucleosidase (MTAN, EC 3.2.2.16). The NP-I family is distinct from nucleoside phosphorylase-II, which belongs to a different structural family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>5'-deoxy-5'-methylthioadenosine phosphorylase. This model represents the methylthioadenosine phosphorylase found in metazoa, cyanobacteria and a limited number of archaea such as Sulfolobus, Aeropyrum, Pyrobaculum, Pyrococcus, and Thermoplasma. This enzyme is responsible for the first step in the methionine salvage pathway after the transfer of the amino acid moiety from S-adenosylmethionine. The enzyme from human is well-characterized including a crystal structure. A misleading characterization is found for a Sulfolobus solfataricus enzyme, which is called a MTAP. In fact, as uncovered by the genome sequence of S. solfataricus, there are at least two nucleotide phosphorylases and the one found in the MTAP clade is not the one annotated as such. The sequence in this clade has not been isolated but is likely to be the authentic SsMTAP as it displays all of the conserved active site residues found in the human enzyme. This explains the finding that the characterized enzyme has greater efficiency towards the purines inosine, guanosine and adenosine over MTA. In fact, this mis-naming of this enzyme has been carried forward to several publications including a crystal stucture. In between the trusted and noise cutoffs are: 1) several archaeal sequences which appear to contain several residues characteristic of phosphorylases which act on guanosine or inosine (according to the crystal structure of MTAP and alignments). In any case, these residues are not conserved. 2) sequences from Mycobacterium tuberculosis and Streptomyces coelicolor which have better, although not perfect retention of the active site residues, but considering the general observation that bacteria utilize the MTA/SAH nucleotidase enzyme and a kinase to do this reaction, these have been excluded pending stronger evidence of their function, and 3) a sequence from Drosophila which appears to be a recent divergence (long branch in neighbor-joining trees) and lacks some of the conserved active site residues. [Central intermediary metabolism, Other, Purines, pyrimidines, nucleosides, and nucleotides, Salvage of nucleosides and nucleotides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Purine nucleoside phosphorylase [Nucleotide transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>5'-methylthioadenosine phosphorylase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>S-methyl-5'-thioadenosine phosphorylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>S-methyl-5'-thioinosine phosphorylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphorylase superfamily. Members of this family include: purine nucleoside phosphorylase (PNP) Uridine phosphorylase (UdRPase) 5'-methylthioadenosine phosphorylase (MTA phosphorylase)</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>nucleoside phosphorylase-I family. The nucleoside phosphorylase-I family members accept a range of purine nucleosides as well as the pyrimidine nucleoside uridine. The NP-1 family  includes phosphorolytic nucleosidases such as purine nucleoside phosphorylase (PNP, EC. 2.4.2.1), uridine phosphorylase (UP, EC 2.4.2.3), and 5'-deoxy-5'-methylthioadenosine phosphorylase (MTAP, EC 2.4.2.28), and hydrolytic nucleosidases such as AMP nucleosidase (AMN, EC 3.2.2.4) and 5'-methylthioadenosine/S-adenosylhomocysteine (MTA/SAH) nucleosidase (MTAN, EC 3.2.2.16). Members of this family display different physiologically relevant quaternary structures: hexameric (trimer-of-dimers arrangement of Shewanella oneidensis MR-1 UP); homotrimeric (human PNP and Escherichia coli PNPII or XapA); hexameric (with some evidence for co-existence of a trimeric form) such as E. coli PNPI (DeoD); or homodimeric such as human and Trypanosoma brucei UP. The NP-I family is distinct from nucleoside phosphorylase-II, which belongs to a different structural family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>purine nucleoside phosphorylases similar to human PNP and Escherichia coli PNP-II (XapA). Human PNP catalyzes the reversible phosphorolysis of the purine nucleosides and deoxynucleosides inosine, guanosine, deoxyinosine, and deoxyguanosine. Patients with PNP deficiency typically present with severe immunodeficiency, neurological dysfunction, and autoimmunity. Escherichia coli PNPII, product of the xapA/pndA gene, catalyzes the phosphorolysis of xanthosine, inosine and guanosine with equal efficiency and has been referred to as xanthosine phosphorylase and inosine-guanosine phosphorylase. E. coli PNPII is also capable of converting nicotinamide to nicotinamide riboside, and may be involved in the NAD+ salvage pathway. It is one of two purine nucleoside phosphorylases found in E. coli, which also contains PNPI, which displays a different substrate specificity and belongs to a different subgroup of the nucleoside phosphorylase-I (NP-I) family than PNPII. NP-I family members accept a range of purine nucleosides as well as the pyrimidine nucleoside uridine. The NP-1 family  includes phosphorolytic nucleosidases, such as purine nucleoside phosphorylase (PNPs, EC. 2.4.2.1), uridine phosphorylase (UP, EC 2.4.2.3), and 5'-deoxy-5'-methylthioadenosine phosphorylase (MTAP, EC 2.4.2.28), and hydrolytic nucleosidases, such as AMP nucleosidase (AMN, EC 3.2.2.4), and 5'-methylthioadenosine/S-adenosylhomocysteine (MTA/SAH) nucleosidase (MTAN, EC 3.2.2.16). The NP-I family is distinct from nucleoside phosphorylase-II, which belongs to a different structural family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>purine nucleoside phosphorylase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4033385</td></tr>
<tr><td id=al>end</td><td id=al>4034258</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>S-methyl-5'-thioadenosine phosphorylase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
