<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05303</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05303 L-lactate dehydrogenase [Phormidium_A]</h3>
    <div>MFEHLFTTLPDPQFRALQPRKAVIIGAGQVGMACAYSLLIQNCFDELVLQDLAADKVEGEVMDLAHGLPFLAPTAIAAGTVADAGRNADIVIITAGAAQKPGETRLDLVQKNVAIFRRLIPEIAEFCPEAILLIVTNPVDIMTYISLELSGFSASRVLGTGTVLDTARFRFLLARELCLDARSLHAYIIGEHGDSEVPVWSKVNIAGMPLCKGGWQGSPDADDPRLQRIFQETRDAAYEIIKRKGYTSYAIGLGVTDIVQSILRDQNRVMTVSSLVHGVANIENVCLSLPTVVNRQGADRMLQLSLTPTESRQLEQSAKVMLNTIKEVSL</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase Ldh</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-lactate dehydrogenase</td>
            <td id=co>JAAURB01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>54</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>cd05292 A subgroup of L-lactate dehydrogenases. L-lactate dehydrogenases (LDH) are tetrameric enzymes catalyzing the last step of glycolysis in which pyruvate is converted to L-lactate. This subgroup is composed predominantly of bacterial LDHs and a few fungal LDHs. Bacterial LDHs may be non-allosteric or may be activated by an allosteric effector such as fructose-1,6-bisphosphate. LDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>94</td>
            <td id=cg>99</td>
            <td id=al>PRK00066 L-lactate dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=al>TIGR01771 L-lactate dehydrogenase. This model represents the NAD-dependent L-lactate dehydrogenases from bacteria and eukaryotes. This enzyme function as as the final step in anaerobic glycolysis. Although lactate dehydrogenases have in some cases been mistaken for malate dehydrogenases due to the similarity of these two substrates and the apparent ease with which evolution can toggle these activities, critical residues have been identified which can discriminate between the two activities. At the time of the creation of this model no hits above the trusted cutoff contained critical residues typical of malate dehydrogenases. [Energy metabolism, Anaerobic, Energy metabolism, Glycolysis/gluconeogenesis]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>cd05291 L-2-hydroxyisocapronate dehydrogenases and some bacterial L-lactate dehydrogenases. L-2-hydroxyisocapronate dehydrogenase (HicDH) catalyzes the conversion of a variety of 2-oxo carboxylic acids with medium-sized aliphatic or aromatic side chains. This subfamily is composed of HicDHs and some bacterial L-lactate dehydrogenases (LDH). LDHs catalyze the last step of glycolysis in which pyruvate is converted to L-lactate. Bacterial LDHs can be non-allosteric or may be activated by an allosteric effector such as fructose-1,6-bisphosphate. Members of this subfamily with known structures such as the HicDH of Lactobacillus confusus, the non-allosteric LDH of Lactobacillus pentosus, and the allosteric LDH of Bacillus stearothermophilus, show that they exist as homotetramers. The HicDH-like subfamily is part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>93</td>
            <td id=cg>98</td>
            <td id=al>cd05293 A subgroup of L-lactate dehydrogenases. L-lactate dehydrogenases (LDH) are tetrameric enzymes catalyzing the last step of glycolysis in which pyruvate is converted to L-lactate. This subgroup is composed of eukaryotic LDHs. Vertebrate LDHs are non-allosteric. This is in contrast to some bacterial LDHs that are activated by an allosteric effector such as fructose-1,6-bisphosphate. LDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>cd00300 L-lactate dehydrogenase-like enzymes. Members of this subfamily are tetrameric NAD-dependent 2-hydroxycarboxylate dehydrogenases including LDHs, L-2-hydroxyisocaproate dehydrogenases (L-HicDH), and LDH-like malate dehydrogenases (MDH). Dehydrogenases catalyze the conversion of carbonyl compounds to alcohols or amino acids. LDHs catalyze the last step of glycolysis in which pyruvate is converted to L-lactate. Vertebrate LDHs are non-allosteric, but some bacterial LDHs are activated by an allosteric effector such as fructose-1,6-bisphosphate. L-HicDH catalyzes the conversion of a variety of 2-oxo carboxylic acids with medium-sized aliphatic or aromatic side chains. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. The LDH-like subfamily is part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>PLN02602 lactate dehydrogenase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=al>COG0039 Malate/lactate dehydrogenase [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>PRK06223 malate dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>cd05290 A subgroup of L-lactate dehydrogenases. L-lactate dehydrogenases (LDH) are tetrameric enzymes catalyzing the last step of glycolysis in which pyruvate is converted to L-lactate. This subgroup is composed of some bacterial LDHs from firmicutes, gammaproteobacteria, and actinobacteria. Vertebrate LDHs are non-allosteric, but some bacterial LDHs are activated by an allosteric effector such as fructose-1,6-bisphosphate. LDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenase, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=al>cd01339 L-lactate dehydrogenase-like malate dehydrogenase proteins. Members of this subfamily have an LDH-like structure and an MDH enzymatic activity. Some members, like MJ0490 from Methanococcus jannaschii, exhibit both MDH and LDH activities. Tetrameric MDHs, including those from phototrophic bacteria, are more similar to LDHs than to other MDHs. LDH catalyzes the last step of glycolysis in which pyruvate is converted to L-lactate. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. The LDH-like MDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=al>cd05294 A lactate dehydrogenases-like structure with malate dehydrogenase enzymatic activity. The LDH-like MDH proteins have a lactate dehyhydrogenase-like (LDH-like) structure and malate dehydrogenase (MDH) enzymatic activity. This subgroup is composed of some archaeal LDH-like MDHs that prefer NADP(H) rather than NAD(H) as a cofactor. One member, MJ0490 from Methanococcus jannaschii, has been observed to form dimers and tetramers during crystalization, although it is believed to exist primarilly as a tetramer in solution. In addition to its MDH activity, MJ0490 also possesses fructose-1,6-bisphosphate-activated LDH activity. Members of this subgroup have a higher sequence similarity to LDHs than to other MDHs. LDH catalyzes the last step of glycolysis in which pyruvate is converted to L-lactate. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. The LDH-like MDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)- binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenase, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>cd00650 NAD-dependent, lactate dehydrogenase-like, 2-hydroxycarboxylate dehydrogenase family. Members of this family include ubiquitous enzymes like L-lactate dehydrogenases (LDH), L-2-hydroxyisocaproate dehydrogenases, and some malate dehydrogenases (MDH). LDH catalyzes the last step of glycolysis in which pyruvate is converted to L-lactate. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. The LDH/MDH-like proteins are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>TIGR01763 malate dehydrogenase, NAD-dependent. This enzyme converts malate into oxaloacetate in the citric acid cycle. The critical residues which discriminate malate dehydrogenase from lactate dehydrogenase have been characterized, and have been used to set the cutoffs for this model. Sequences showing [aflimv][ap]R[rk]pgM[st] and [ltv][ilm]gGhgd were kept above trusted, while those in which the capitalized residues in the patterns were found to be Q, E and E were kept below the noise cutoff. Some sequences in the grey zone have been annotated as malate dehydrogenases, but none have been characterized. Phylogenetically, a clade of sequences from eukaryotes such as Toxoplasma and Plasmodium which include a characterized lactate dehydrogenase and show abiguous critical residue patterns appears to be more closely related to these bacterial sequences than other eukaryotic sequences. These are relatively long branch and have been excluded from the model. All other sequences falling below trusted appear to be phylogenetically outside of the clade including the trusted hits. The annotation of Botryococcus braunii as lactate dehydrogenase appears top be in error. This was initially annotated as MDH by Swiss-Prot and then changed. The rationale for either of these annotations is not traceable. [Energy metabolism, TCA cycle]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=al>PTZ00117 malate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>PTZ00082 L-lactate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=co>42</td>
            <td id=cg>99</td>
            <td id=al>pfam00056 lactate/malate dehydrogenase, NAD binding domain. L-lactate dehydrogenases are metabolic enzymes which catalyze the conversion of L-lactate to pyruvate, the last step in anaerobic glycolysis. L-2-hydroxyisocaproate dehydrogenases are also members of the family. Malate dehydrogenases catalyze the interconversion of malate to oxaloacetate. The enzyme participates in the citric acid cycle. L-lactate dehydrogenase is also found as a lens crystallin in bird and crocodile eyes. N-terminus (this family) is a Rossmann NAD-binding fold. C-terminus is an unusual alpha+beta fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>51</td>
            <td id=cg>98</td>
            <td id=al>pfam02866 lactate/malate dehydrogenase, alpha/beta C-terminal domain. L-lactate dehydrogenases are metabolic enzymes which catalyze the conversion of L-lactate to pyruvate, the last step in anaerobic glycolysis. L-2-hydroxyisocaproate dehydrogenases are also members of the family. Malate dehydrogenases catalyze the interconversion of malate to oxaloacetate. The enzyme participates in the citric acid cycle. L-lactate dehydrogenase is also found as a lens crystallin in bird and crocodile eyes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TIGR01772 malate dehydrogenase, NAD-dependent. This model represents the NAD-dependent malate dehydrogenase found in eukaryotes and certain gamma proteobacteria. The enzyme is involved in the citric acid cycle as well as the glyoxalate cycle. Several isoforms exidt in eukaryotes. In S. cereviseae, for example, there are cytoplasmic, mitochondrial and peroxisomal forms. Although malate dehydrogenases have in some cases been mistaken for lactate dehydrogenases due to the similarity of these two substrates and the apparent ease with which evolution can toggle these activities, critical residues have been identified which can discriminate between the two activities. At the time of the creation of this model no hits above the trusted cutoff contained critical residues typical of lactate dehydrogenases. [Energy metabolism, TCA cycle]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cb>60</td>
            <td id=cb>64</td>
            <td id=al>cd01337 Glyoxysomal and mitochondrial malate dehydrogenases. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. Members of this subfamily are localized to the glycosome and mitochondria. MDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=al>PTZ00325 malate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>58</td>
            <td id=cb>62</td>
            <td id=al>PRK05086 malate dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>59</td>
            <td id=cb>61</td>
            <td id=al>PLN00106 malate dehydrogenase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>84</td>
            <td id=cg>86</td>
            <td id=al>cd00704 Malate dehydrogenase. Malate dehydrogenase (MDH) is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. MDHs belong to the NAD-dependent, lactate dehydrogenase (LDH)-like, 2-hydroxycarboxylate dehydrogenase family, which also includes the GH4 family of glycoside hydrolases. They are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>73</td>
            <td id=cb>75</td>
            <td id=al>TIGR01758 malate dehydrogenase, NAD-dependent. This model represents the NAD-dependent cytosolic malate dehydrogenase from eukaryotes. The enzyme from pig has been studied by X-ray crystallography</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cb>61</td>
            <td id=cb>65</td>
            <td id=al>TIGR01756 lactate dehydrogenase. This model represents a family of protist lactate dehydrogenases which have aparrently evolved from a recent protist malate dehydrogenase ancestor. Lactate dehydrogenase converts the hydroxyl at C-2 of lactate to a carbonyl in the product, pyruvate. The preference of this enzyme for NAD or NADP has not been determined. A critical residue in malate dehydrogenase, arginine-91 (T. vaginalis numbering) has been mutated to a leucine, eliminating the positive charge which complemeted the carboxylate in malate which is absent in lactate. Several other more subtle changes are proposed to make the active site smaller to accomadate the less bulky lactate molecule.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>77</td>
            <td id=cb>78</td>
            <td id=al>cd01336 Cytoplasmic and cytosolic Malate dehydrogenases. MDH is one of the key enzymes in the citric acid cycle, facilitating both the conversion of malate to oxaloacetate and replenishing levels of oxalacetate by reductive carboxylation of pyruvate. Members of this subfamily are eukaryotic MDHs localized to the cytoplasm and cytosol. MDHs are part of the NAD(P)-binding Rossmann fold superfamily, which includes a wide variety of protein families including the NAD(P)-binding domains of alcohol dehydrogenases, tyrosine-dependent oxidoreductases, glyceraldehyde-3-phosphate dehydrogenases, formate/glycerate dehydrogenases, siroheme synthases, 6-phosphogluconate dehydrogenases, aminoacid dehydrogenases, repressor rex, and NAD-binding potassium channel domains, among others.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4314046</td></tr>
<tr><td id=al>end</td><td id=al>4315039</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>L-lactate dehydrogenase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>331</td></tr>
</table>

    </div>
</body>
</html>
