<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04814</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04814 small ribosomal subunit biogenesis GTPase RsgA [Phormidium_A]</h3>
    <div>MSDRPAMESELRYPQGTVLAVQANFYQVQLDPDPPPPTPSLLCTRRTRLKKVGQQVMVGDRVVVEEPDWEGNRGAIGDVLPRRTELDRPPVANADRILLLFTLEEPSLDPWQLSRFLVKAESTQLTVELCLNKCDLVSEEEQEAWRSRLGQWGYDPLLMSVKTGWGLEALEQRLEQGITILAGPSGVGKSSTINRLIPQANLRVAKVSGKLGRGRHTTRHVELFELEQGGLLADTPGFNQPQLTTTAKELADCFPEIRQRRQGDDCHFSDCRHRDEPGCQIRGDWERYEHYLTFLEEAIAQDTAQHHSGEEESSFKSKAQAGGERRYEPKLASKKYRRQSRRRQHQTLQDLCEELDEK</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>ribosome biogenesis GTPase EngC</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cr>Pantanalinema</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=co>Planktothricoides</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA</td>
            <td id=cr>JACYLU01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Putative ribosome biogenesis GTPase RsgA [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTPase RsgA; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Ribosomal interacting GTPase YjeQ/EngC, a circularly permuted subfamily of the Ras GTPases. YjeQ (YloQ in Bacillus subtilis) is a ribosomal small subunit-dependent GTPase; hence also known as RsgA. YjeQ is a late-stage ribosomal biogenesis factor involved in the 30S subunit maturation, and it represents a protein family whose members are broadly conserved in bacteria and have been shown to be essential to the growth of E. coli and B. subtilis. Proteins of the YjeQ family contain all sequence motifs typical of the vast class of P-loop-containing GTPases, but show a circular permutation, with a G4-G1-G3 pattern of motifs as opposed to the regular G1-G3-G4 pattern seen in most GTPases. All YjeQ family proteins display a unique domain architecture, which includes an N-terminal OB-fold RNA-binding domain, the central permuted GTPase domain, and a zinc knuckle-like C-terminal cysteine domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ribosome small subunit-dependent GTPase A. Members of this protein were designated YjeQ and are now designated RsgA (ribosome small subunit-dependent GTPase A). The strongest motif in the alignment of these proteins is GXSGVGKS[ST], a classic P-loop for nucleotide binding. This protein has been shown to cleave GTP and remain bound to GDP. A role as a regulator of translation has been suggested. The Aquifex aeolicus ortholog is split into consecutive open reading frames. Consequently, this model was build in fragment mode (-f option). [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RsgA GTPase. RsgA (also known as EngC and YjeQ) represents a protein family whose members are broadly conserved in bacteria and are indispensable for growth. The GTPase domain of RsgA is very similar to several P-loop GTPases, but differs in having a circular permutation of the GTPase structure described by a G4-G1-G3 pattern.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>small ribosomal subunit biogenesis GTPase RsgA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>GTPase RsgA; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>39</td>
            <td id=al>Ribosome biogenesis GTPase A [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>Circularly permuted YlqF-related GTPases. These proteins are found in bacteria, eukaryotes, and archaea.  They all exhibit a circular permutation of the GTPase signature motifs so that the order of the conserved G box motifs is G4-G5-G1-G2-G3, with G4 and G5 being permuted from the C-terminal region of proteins in the Ras superfamily to the N-terminus of YlqF-related GTPases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>Circularly permuted YlqF GTPase. Proteins of the YlqF family contain all sequence motifs typical of the vast class of P-loop-containing GTPases, but show a circular permutation, with a G4-G1-G3 pattern of motifs as opposed to the regular G1-G3-G4 pattern seen in most GTPases. The YlqF subfamily is represented in all eukaryotes as well as a phylogenetically diverse array of bacteria (including gram-positive bacteria, proteobacteria, Synechocystis, Borrelia, and Thermotoga).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>An uncharacterized, circularly permuted subfamily of the Ras GTPases. This family represents archaeal GTPase typified by the protein MJ1464 from Methanococcus jannaschii. The members of this family show a circular permutation of the GTPase signature motifs so that C-terminal strands 5, 6, and 7 (strands 6 contain the NKxD motif) are relocated to the N terminus.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>50S ribosome-binding GTPase. The full-length GTPase protein is required for the complete activity of the protein of interacting with the 50S ribosome and binding of both adenine and guanine nucleotides, with a preference for guanine nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>GTP-binding protein YsxC; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>GTP-binding protein EngB required for normal cell division [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>A circularly permuted subfamily of the Ras GTPases. Human HSR1 is localized to the human MHC class I region and is highly homologous to a putative GTP-binding protein, MMR1 from mouse. These proteins represent a new subfamily of GTP-binding proteins that has only eukaryote members. This subfamily shows a circular permutation of the GTPase signature motifs so that the C-terminal strands 5, 6, and 7 (strand 6 contains the G4 box with sequence NKXD) are relocated to the N-terminus.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>65</td>
            <td id=al>Circularly permuted YqeH GTPase. YqeH is an essential GTP-binding protein. Depletion of YqeH induces an excess initiation of DNA replication, suggesting that it negatively controls initiation of chromosome replication. The YqeH subfamily is common in eukaryotes and sporadically present in bacteria with probable acquisition by plants from chloroplasts. Proteins of the YqeH family contain all sequence motifs typical of the vast class of P-loop-containing GTPases, but show a circular permutation, with a G4-G1-G3 pattern of motifs as opposed to the regular G1-G3-G4 pattern seen in most GTPases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>34</td>
            <td id=al>YihA (EngB) GTPase family. The YihA (EngB) subfamily of GTPases is typified by the E. coli YihA, an essential protein involved in cell division control. YihA and its orthologs are small proteins that typically contain less than 200 amino acid residues and consists of the GTPase domain only (some of the eukaryotic homologs contain an N-terminal extension of about 120 residues that might be involved in organellar targeting). Homologs of yihA are found in most Gram-positive and Gram-negative pathogenic bacteria, with the exception of Mycobacterium tuberculosis. The broad-spectrum nature of YihA and its essentiality for cell viability in bacteria make it an attractive antibacterial target.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3923672</td></tr>
<tr><td id=al>end</td><td id=al>3924749</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[secondary-metabolites]</td></tr>
<tr><td id=al>descr</td><td id=al>small ribosomal subunit biogenesis GTPase RsgA</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al>(region 6) NRPS-like cds((PP-binding or NAD_binding_4) and (AMP-binding or A-OX))  </td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
