<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05348</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05348 HAD-IIB family hydrolase [Phormidium_A]</h3>
    <div>MPMHSEGCGRLKPLESVFLPALKRRGFQPSQKLLRDDMTLVVFTDLDGTLLNSHDYRFDEAIPTLTLLKQRNIPIIPVTSKTRVEVEPLLDNLQLQEPFIVENGSGLFIHPQDQRFDLSETESWQGYRLKQFGCLYHDARSALQQLSQRLGVNLQGFGDLTVPDLTTLTGLSPQAAQQAQTREFSEPFITPRERSSQEIEQAVTELGFRVVVGDRFSHLIGPQASKGTAVRWLLERFQQSGPRQIVGLGNSPNDLDMLEAVESPIIIPNNKGLIHPGLQHLNAQIAPHAGARGWAAAIAPLLGQ</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>93</td>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>79</td>
            <td id=cg>86</td>
            <td id=cg>98</td>
            <td id=al>mannosyl-3-phosphoglycerate phosphatase MpgP</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>86</td>
            <td id=cg>98</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>86</td>
            <td id=cg>98</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cr>JACYMB01</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>85</td>
            <td id=cg>95</td>
            <td id=al>haloacid dehalogenase</td>
            <td id=co>Lyngbya</td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>86</td>
            <td id=cg>95</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cb>RDYZ01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>42</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>PRK00192 mannosyl-3-phosphoglycerate phosphatase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=al>TIGR01486 mannosyl-3-phosphoglycerate phosphatase family. This small group of proteins is a member of the IIB subfamily (TIGR01484) of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. Several members of this family from thermophiles (and from Dehalococcoides ethenogenes) are now known to act as mannosyl-3-phosphoglycerate (MPG) phosphatase. In these cases, the enzyme acts after MPG synthase to make the compatible solute mannosylglycerate. We propose that other mesophilic members of this family do not act as mannosyl-3-phosphoglycerate phosphatase. A member of this family is found in Escherichia coli, which appears to lack MPG synthase. Mannosylglycerate is imported in E. coli by phosphoenolpyruvate-dependent transporter (), but it appears the phosphorylation is not on the glycerate moiety, that the phosphorylated import is degraded by an alpha-mannosidase from an adjacent gene, and that E. coli would have no pathway to obtain MPG.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=al>cd07507 haloacid dehalogenase-like superfamily phosphatase similar to Pyrococcus horikoshii mannosyl-3-phosphoglycerate phosphatase and Persephonella marina glucosyl-3-phosphoglycerate phosphatase. This family includes Pyrococcus horikoshii and Thermus thermophilus HB27 mannosyl-3-phosphoglycerate phosphatases (MpgPs) which catalyze the dephosphorylation of alpha-mannosyl-3-phosphoglycerate (MPG) to produce alpha-mannosylglycerate (MG), and Persephonella marina glucosyl-3-phosphoglycerate phosphatase (GpgP) which catalyzes the dephosphorylation of glucosyl-3-phosphoglycerate (GPG) to produce glucosylglycerate (GG). It also includes Methanococcoides burtonii MpgP protein which is able to dephosphorylate GPG to GG, and MPG to MG. Similar flexibilities in substrate specificity have been confirmed in vitro for the MpgPs from Thermus thermophiles and Pyrococcus horikoshii. Screens with natural substrates have not yet detected activity for another member Escherichia Coli YedP. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=al>TIGR02461 mannosyl-3-phosphoglycerate phosphatase. Members of this family are mannosyl-3-phosphoglycerate phosphatase (EC 3.1.3.70). It acts sequentially after mannosyl-3-phosphoglycerate synthase (EC 2.4.1.217) in a two-step pathway of biosynthesis of the compatible solute mannosylglycerate, a typical osmolyte of thermophiles.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>PRK03669 mannosyl-3-phosphoglycerate phosphatase-related protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>86</td>
            <td id=cg>96</td>
            <td id=al>COG3769 Predicted mannosyl-3-phosphoglycerate phosphatase, HAD superfamily [Carbohydrate transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>91</td>
            <td id=co>40</td>
            <td id=al>PRK14502 bifunctional mannosyl-3-phosphoglycerate synthase/mannosyl-3 phosphoglycerate phosphatase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>82</td>
            <td id=cg>83</td>
            <td id=al>PRK12702 mannosyl-3-phosphoglycerate phosphatase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=al>TIGR01484 HAD-superfamily hydrolase, subfamily IIB. This subfamily falls within the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The Class II subfamilies are characterized by a domain that is located between the second and third conserved catalytic motifs of the superfamily domain. The IIB subfamily is distinguished from the IIA subfamily (TIGR01460) by homology and the predicted secondary structure of this domain by PSI-PRED. The IIB subfamily's Class II domain has the following predicted structure: Helix-Sheet-Sheet-(Helix or Sheet)-Helix-Sheet-(variable)-Helix-Sheet-Sheet. The IIB subfamily consists of Trehalose-6-phosphatase (TIGR00685), plant and cyanobacterial Sucrose-phosphatase and a closely related group of bacterial and archaeal sequences, eukaryotic phosphomannomutase (pfam03332), a large subfamily ("Cof-like hydrolases", TIGR00099) containing many closely related bacterial sequences, a hypothetical equivalog containing the E. coli YedP protein, as well as two small clusters containing OMNI|TC0379 and OMNI|SA2196 whose relationship to the other groups is unclear. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>COG0561 Hydroxymethylpyrimidine pyrophosphatase and other HAD family phosphatases [Coenzyme transport and metabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>78</td>
            <td id=cg>93</td>
            <td id=al>TIGR00099 Cof subfamily of IIB subfamily of haloacid dehalogenase superfamily. This subfamily of sequences falls within the Class-IIB subfamily (TIGR01484) of the Haloacid Dehalogenase superfamily of aspartate-nucleophile hydrolases. The use of the name "Cof" as an identifier here is arbitrary and refers to the E. coli Cof protein. This subfamily is notable for the large number of recent paralogs in many species. Listeria, for instance, has 12, Clostridium, Lactococcus and Streptococcus pneumoniae have 8 each, Enterococcus and Salmonella have 7 each, and Bacillus subtilus, Mycoplasma, Staphylococcus and E. coli have 6 each. This high degree of gene duplication is limited to the gamma proteobacteria and low-GC gram positive lineages. The profusion of genes in this subfamily is not coupled with a high degree of divergence, so it is impossible to determine an accurate phylogeny at the equivalog level. Considering the relationship of this subfamily to the other known members of the HAD-IIB subfamily (TIGR01484), sucrose and trehalose phosphatases and phosphomannomutase, it seems a reasonable hypothesis that these enzymes act on phosphorylated sugars. Possibly the diversification of genes in this subfamily represents the diverse sugars and polysaccharides that various bacteria find in their biological niches. The members of this subfamily are restricted almost exclusively to bacteria (one sequences from S. pombe scores above trusted, while another is between trusted and noise). It is notable that no archaea are found in this group, the closest relations to the archaea found here being two Deinococcus sequences. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>81</td>
            <td id=cg>98</td>
            <td id=al>pfam08282 haloacid dehalogenase-like hydrolase. This family contains haloacid dehalogenase-like hydrolase enzymes.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4352180</td></tr>
<tr><td id=al>end</td><td id=al>4353095</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>HAD-IIB family hydrolase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>305</td></tr>
</table>

    </div>
</body>
</html>
