<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04442</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04442 rod shape-determining protein [Phormidium_A]</h3>
    <div>MGIDLGTANTLVYVSGKGIVLQEPSVVAIDQDDKLPLAVGEEAKQMLGRTPGNVVALRPLRDGVIADFDTAELMLKHFIRQVHGGRTGLVGPRIVIGIPSGVTGVERRAVMDAAIQAGARDVRLIDEPIAAAIGAGLPVAEPTGNMIVDIGGGTTEVAVLSLQGIVLSESVRVAGDELSDSITQYMKKVHNLVIGERTAEEIKIQVASAYPTQEDEIIMDVRGLHLLSGLPRTVTIKGPEIRESMSEPLSVIIEAVKRTLERTPPELAADIIDRGIMLAGGGALLRGLDTLISHETGIVTHVAADPLSCVVLGTGRVLENFKQLERVFRGQSRSL</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=cg>96</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein MreB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=co>SIO1D9</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>rod shape-determining protein</td>
            <td id=cr>GCF-001939115</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>65</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>PRK13927 rod shape-determining protein MreB; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>97</td>
            <td id=cg>97</td>
            <td id=al>PRK13930 rod shape-determining protein MreB; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>pfam06723 MreB/Mbl protein. This family consists of bacterial MreB and Mbl proteins as well as two related archaeal sequences. MreB is known to be a rod shape-determining protein in bacteria and goes to make up the bacterial cytoskeleton. Genes coding for MreB/Mbl are only found in elongated bacteria, not in coccoid forms. It has been speculated that constituents of the eukaryotic cytoskeleton (tubulin, actin) may have evolved from prokaryotic precursor proteins closely related to today's bacterial proteins FtsZ and MreB/Mbl.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=al>cd10225 MreB and similar proteins. MreB is a bacterial protein which assembles into filaments resembling those of eukaryotic F-actin. It is involved in determining the shape of rod-like bacterial cells, by assembling into large fibrous spirals beneath the cell membrane. MreB has also been implicated in chromosome segregation; specifically MreB is thought to bind to and segregate the replication origin of bacterial chromosomes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>PRK13928 rod shape-determining protein Mbl; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>TIGR00904 cell shape determining protein, MreB/Mrl family. MreB (mecillinam resistance) in E. coli (also called envB) and the paralogous pair MreB and Mrl of Bacillus subtilis have all been shown to help determine cell shape. This protein is present in a wide variety of bacteria, including spirochetes, but is missing from the Mycoplasmas and from Gram-positive cocci. Most completed bacterial genomes have a single member of this family. In some species it is an essential gene. A close homolog is found in the Archaeon Methanobacterium thermoautotrophicum, and a more distant homolog in Archaeoglobus fulgidus. The family is related to cell division protein FtsA and heat shock protein DnaK. [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>98</td>
            <td id=cg>97</td>
            <td id=al>COG1077 Actin-like ATPase involved in cell morphogenesis [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>97</td>
            <td id=cg>97</td>
            <td id=al>PRK13929 rod-share determining protein MreBH; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd10170 Nucleotide-binding domain of the HSP70 family. HSP70 (70-kDa heat shock protein) family chaperones assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. Typically, HSP70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. HSP70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). Some HSP70 family members are not chaperones but instead, function as NEFs to remove ADP from their HSP70 chaperone partners during the ATP hydrolysis cycle, some may function as both chaperones and NEFs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>COG0443 Molecular chaperone DnaK (HSP70) [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>98</td>
            <td id=co>55</td>
            <td id=al>pfam00012 Hsp70 protein. Hsp70 chaperones help to fold many proteins. Hsp70 assisted folding involves repeated cycles of substrate binding and release. Hsp70 activity is ATP dependent. Hsp70 proteins are made up of two regions: the amino terminus is the ATPase domain and the carboxyl terminus is the substrate binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd10235 Nucleotide-binding domain of Escherichia coli HscC and similar proteins. This subfamily  includes Escherichia coli HscC (also called heat shock cognate protein C, Hsc62, or YbeW) and the the putative DnaK-like protein Escherichia coli ECs0689. It belongs to the heat shock protein 70 (Hsp70) family of chaperones that assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. Typically, Hsp70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. Hsp70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). Two genes in the vicinity of the HscC gene code for potential cochaperones: J-domain containing proteins, DjlB/YbeS and DjlC/YbeV. HscC and its co-chaperone partners may play a role in the SOS DNA damage response. HscC does not appear to require a NEF.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>48</td>
            <td id=cg>97</td>
            <td id=al>pfam14450 Cell division protein FtsA. FtsA is essential for bacterial cell division, and co-localizes to the septal ring with FtsZ. It has been suggested that the interaction of FtsA-FtsZ has arisen through coevolution in different bacterial strains. The FtsA protein contains two structurally related actin-like ATPase domains which are also structurally related to the ATPase domains of HSP70 (see PF00012). FtsA has a SHS2 domain PF02491 inserted in to the RnaseH fold PF02491.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>60</td>
            <td id=co>48</td>
            <td id=al>COG0849 Cell division ATPase FtsA [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>63</td>
            <td id=co>55</td>
            <td id=al>cd10230 Nucleotide-binding domain of human HYOU1 and similar proteins. This subgroup includes human HYOU1 (also known as human hypoxia up-regulated 1, GRP170; HSP12A; ORP150; GRP-170; ORP-150; the human HYOU1 gene maps to11q23.1-q23.3) and Saccharomyces cerevisiae Lhs1p (also known as Cer1p, SsI1). Mammalian HYOU1 functions as a nucleotide exchange factor (NEF) for HSPA5 (alos known as BiP, Grp78 or HspA5) and may also function as a HSPA5-independent chaperone. S. cerevisiae Lhs1p, does not have a detectable endogenous ATPase activity like canonical HSP70s, but functions as a NEF for Kar2p; it's interaction with Kar2p is stimulated by nucleotide-binding. In addition, Lhs1p has a nucleotide-independent holdase activity that prevents heat-induced aggregation of proteins in vitro. This subgroup belongs to the 105/110 kDa heat shock protein (HSP105/110) subfamily of the HSP70-like family. HSP105/110s are believed to function generally as co-chaperones of HSP70 chaperones, acting as NEFs, to remove ADP from their HSP70 chaperone partners during the ATP hydrolysis cycle. HSP70 chaperones assist in protein folding and assembly, and can direct incompetent "client" proteins towards degradation. Like HSP70 chaperones, HSP105/110s have an N-terminal nucleotide-binding domain (NBD) and a C-terminal substrate-binding domain (SBD). For HSP70 chaperones, the nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. Hsp70 chaperone activity is also regulated by J-domain proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>66</td>
            <td id=co>58</td>
            <td id=al>cd10228 Nucleotide-binding domain of 105/110 kDa heat shock proteins including HSPA4 and similar proteins. This subgroup includes the human proteins, HSPA4 (also known as 70-kDa heat shock protein 4, APG-2, HS24/P52, hsp70 RY, and HSPH2; the human HSPA4 gene maps to 5q31.1), HSPA4L (also known as 70-kDa heat shock protein 4-like, APG-1, HSPH3, and OSP94; the human HSPA4L gene maps to 4q28), and HSPH1 (also known as heat shock 105kDa/110kDa protein 1, HSP105; HSP105A; HSP105B; NY-CO-25; the human HSPH1 gene maps to 13q12.3), Saccharomyces cerevisiae Sse1p and Sse2p, and a sea urchin sperm receptor. It belongs to the 105/110 kDa heat shock protein (HSP105/110) subfamily of the HSP70-like family, and includes proteins believed to function generally as co-chaperones of HSP70 chaperones, acting as nucleotide exchange factors (NEFs), to remove ADP from their HSP70 chaperone partners during the ATP hydrolysis cycle. HSP70 chaperones assist in protein folding and assembly, and can direct incompetent "client" proteins towards degradation. Like HSP70 chaperones, HSP105/110s have an N-terminal nucleotide-binding domain (NBD) and a C-terminal substrate-binding domain (SBD). For HSP70 chaperones, the nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. Hsp70 chaperone activity is also regulated by J-domain proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>cd10231 Escherichia coli YegD, a putative chaperone protein, and related proteins. This bacterial subfamily includes the uncharacterized Escherichia coli YegD. It belongs to the heat shock protein 70 (HSP70) family of chaperones that assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. Typically, HSP70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. YegD lacks the SBD. HSP70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). Some family members are not chaperones but instead, function as NEFs for their Hsp70 partners, other family members function as both chaperones and NEFs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cb>69</td>
            <td id=cb>62</td>
            <td id=al>cd10241 Nucleotide-binding domain of human HSPA5 and similar proteins. This subfamily includes human HSPA5 (also known as 70-kDa heat shock protein 5, glucose-regulated protein 78/GRP78, and immunoglobulin heavy chain-binding protein/BIP, MIF2; the gene encoding HSPA5 maps to 9q33.3.), Sacchaormyces cerevisiae Kar2p (also known as Grp78p), and related proteins. This subfamily belongs to the heat shock protein 70 (HSP70) family of chaperones that assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. HSPA5 and Kar2p are chaperones of the endoplasmic reticulum (ER). Typically, HSP70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. HSP70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). Multiple ER DNAJ domain proteins have been identified and may exist in distinct complexes with HSPA5 in various locations in the ER, for example DNAJC3-p58IPK in the lumen. HSPA5-NEFs include SIL1 and an atypical HSP70 family protein HYOU1/ORP150. The ATPase activity of Kar2p is stimulated by the NEFs: Sil1p and Lhs1p.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=co>59</td>
            <td id=co>53</td>
            <td id=al>TIGR01174 cell division protein FtsA. This bacterial cell division protein interacts with FtsZ, the bacterial homolog of tubulin. It is an ATP-binding protein and shows structural similarities to actin and heat shock cognate protein 70. [Cellular processes, Cell division]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>72</td>
            <td id=co>58</td>
            <td id=al>cd10237 Nucleotide-binding domain of human HSPA13 and similar proteins. Human HSPA13 (also called 70-kDa heat shock protein 13,  STCH, "stress 70 protein chaperone, microsome-associated, 60kD", "stress 70 protein chaperone, microsome-associated, 60kDa"; the gene encoding HSPA13 maps to 21q11.1) belongs to the heat shock protein 70 (HSP70) family of chaperones that assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. Typically, HSP70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. HSP70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). STCH contains an NBD but lacks an SBD. STCH may function to regulate cell proliferation and survival, and modulate the TRAIL-mediated cell death pathway. The HSPA13 gene is a candidate stomach cancer susceptibility gene; a mutation in the NBD coding region of HSPA13 has been identified in stomach cancer cells. The NBD of HSPA13 interacts with the ubiquitin-like proteins Chap1 and Chap2, implicating HSPA13 in regulating cell cycle and cell death events. HSPA13 is induced by the Ca2+ ionophore A23187.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>cd10236 Nucleotide-binding domain of HscA and similar proteins. Escherichia coli HscA (heat shock cognate protein A, also called Hsc66), belongs to the heat shock protein 70 (HSP70) family of chaperones that assist in protein folding and assembly and can direct incompetent "client" proteins towards degradation. Typically, HSP70s have a nucleotide-binding domain (NBD) and a substrate-binding domain (SBD). The nucleotide sits in a deep cleft formed between the two lobes of the NBD. The two subdomains of each lobe change conformation between ATP-bound, ADP-bound, and nucleotide-free states. ATP binding opens up the substrate-binding site; substrate-binding increases the rate of ATP hydrolysis. HSP70 chaperone activity is regulated by various co-chaperones: J-domain proteins and nucleotide exchange factors (NEFs). HscA's partner J-domain protein is HscB; it does not appear to require a NEF, and has been shown to be induced by cold-shock. The HscA-HscB chaperone/co-chaperone pair is involved in [Fe-S] cluster assembly.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>47</td>
            <td id=co>59</td>
            <td id=al>PRK15080 ethanolamine utilization protein EutJ; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>PRK13411 molecular chaperone DnaK; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>28</td>
            <td id=cr>16</td>
            <td id=al>PRK01433 chaperone protein HscA; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3608304</td></tr>
<tr><td id=al>end</td><td id=al>3609312</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>rod shape-determining protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>336</td></tr>
</table>

    </div>
</body>
</html>
