<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00599</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00599 DNA-formamidopyrimidine glycosylase [Phormidium_A]</h3>
    <div>MPELPEVETVRRGLNAHTCQQRICGGAVLLDRSLARPAHADEFIEGLQGQSIRTWHRRGKYLIAQLGEEDICHRLSPTAAGWLGVHLRMTGQLLWCHPDDPLQTHTRVRLFFEGDRELRFVDIRTFGRLWWVPPDTSPETVITGLLSLGPEPFRPEFSTEYLIATLANRRKPIKTALLDQTIVAGVGNIYADESLFLSGIPPTRLCSQLTPEEIDRLRQQVIAVLGQAIEAGGTTLRDFRDVKGINGNYGGMAWVYNRTGQPCRQCDTPIERIKLGGRSSHFCPRCQPDPKTP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>94</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>97</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>83</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>99</td>
            <td id=cg>97</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA-formamidopyrimidine glycosylase</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>formamidopyrimidine-DNA glycosylase</td>
            <td id=co>Planktothrix</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>61</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>PRK13945 formamidopyrimidine-DNA glycosylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>PRK01103 bifunctional DNA-formamidopyrimidine glycosylase/DNA-(apurinic or apyrimidinic site) lyase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>TIGR00577 DNA-formamidopyrimidine glycosylase. All proteins in the FPG family with known functions are FAPY-DNA glycosylases that function in base excision repair. Homologous to endonuclease VIII (nei). This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>COG0266 Formamidopyrimidine-DNA glycosylase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>PRK14810 formamidopyrimidine-DNA glycosylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK14811 formamidopyrimidine-DNA glycosylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=al>pfam01149 Formamidopyrimidine-DNA glycosylase N-terminal domain. Formamidopyrimidine-DNA glycosylase (Fpg) is a DNA repair enzyme that excises oxidized purines from damaged DNA. This family is the N-terminal domain contains eight beta-strands, forming a beta-sandwich with two alpha-helices parallel to its edges.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=al>cd08966 N-terminal domain of Escherichia coli Fpg1/MutM and related bacterial DNA glycosylases. This family contains the N-terminal domain of Escherichia coli Fpg1/MutM and related bacterial DNA glycosylases. It belongs to the FpgNei_N, [N-terminal domain of Fpg (formamidopyrimidine-DNA glycosylase, MutM)_Nei (endonuclease VIII)] domain superfamily. DNA glycosylases maintain genome integrity by recognizing base lesions created by ionizing radiation, alkylating or oxidizing agents, and endogenous reactive oxygen species. They initiate the base-excision repair process, which is completed with the help of enzymes such as phosphodiesterases, AP endonucleases, DNA polymerases and DNA ligases. DNA glycosylases cleave the N-glycosyl bond between the sugar and the damaged base, creating an AP (apurinic/apyrimidinic) site. Most FpgNei DNA glycosylases use their N-terminal proline residue as the key catalytic nucleophile, and the reaction proceeds via a Schiff base intermediate.  Escherichia coli Fpg mainly recognizes and excises damaged purines such as 8-oxo-7,8-dihydroguanine (8-oxoG) and 2,6-diamino-4-hydroxy-5-formamidopyrimidine (FapyG). It is bifunctional, having both a DNA glycosylase (recognition activity) and a AP lyase activity. In addition to this EcFpg-like_N domain, enzymes belonging to this family contain a helix-two turn-helix (H2TH) domain and a zinc-finger motif, which also contribute residues to the active site.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=al>smart00898 Formamidopyrimidine-DNA glycosylase N-terminal domain. This entry represents the catalytic domain of DNA glycosylase/AP lyase enzymes, which are involved in base excision repair of DNA damaged by oxidation or by mutagenic agents. Most damage to bases in DNA is repaired by the base excision repair pathway. These enzymes are primarily from bacteria, and have both DNA glycosylase activity and AP lyase activity. Examples include formamidopyrimidine-DNA glycosylases (Fpg; MutM) and endonuclease VIII (Nei). Formamidopyrimidine-DNA glycosylases (Fpg, MutM) is a trifunctional DNA base excision repair enzyme that removes a wide range of oxidation-damaged bases (N-glycosylase activity; ) and cleaves both the 3'- and 5'-phosphodiester bonds of the resulting apurinic/apyrimidinic site (AP lyase activity; ). Fpg has a preference for oxidised purines, excising oxidized purine bases such as 7,8-dihydro-8-oxoguanine (8-oxoG). ITs AP (apurinic/apyrimidinic) lyase activity introduces nicks in the DNA strand, cleaving the DNA backbone by beta-delta elimination to generate a single-strand break at the site of the removed base with both 3'- and 5'-phosphates. Fpg is a monomer composed of 2 domains connected by a flexible hinge. The two DNA-binding motifs (a zinc finger and the helix-two-turns-helix motifs) suggest that the oxidized base is flipped out from double-stranded DNA in the binding mode and excised by a catalytic mechanism similar to that of bifunctional base excision repair enzymes. Fpg binds one ion of zinc at the C-terminus, which contains four conserved and essential cysteines.. Endonuclease VIII (Nei) has the same enzyme activities as Fpg above, but with a preference for oxidized pyrimidines, such as thymine glycol, 5,6-dihydrouracil and 5,6-dihydrothymine. These protein contains three structural domains: an N-terminal catalytic core domain, a central helix-two turn-helix (H2TH) module and a C-terminal zinc finger (see PDB:1K82). The N-terminal catalytic domain and the C-terminal zinc finger straddle the DNA with the long axis of the protein oriented roughly orthogonal to the helical axis of the DNA. Residues that contact DNA are located in the catalytic domain and in a beta-hairpin loop formed by the zinc finger.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=al>pfam06831 Formamidopyrimidine-DNA glycosylase H2TH domain. Formamidopyrimidine-DNA glycosylase (Fpg) is a DNA repair enzyme that excises oxidized purines from damaged DNA. This family is the central domain containing the DNA-binding helix-two turn-helix domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=al>cd08773 N-terminal domain of Fpg (formamidopyrimidine-DNA glycosylase, MutM)_Nei (endonuclease VIII) base-excision repair DNA glycosylases. DNA glycosylases maintain genome integrity by recognizing base lesions created by ionizing radiation, alkylating or oxidizing agents, and endogenous reactive oxygen species. These enzymes initiate the base-excision repair process, which is completed with the help of enzymes such as phosphodiesterases, AP endonucleases, DNA polymerases and DNA ligases. DNA glycolsylases cleave the N-glycosyl bond between the sugar and the damaged base, creating an AP (apurinic/apyrimidinic) site. The FpgNei DNA glycosylases represent one of the two structural superfamilies of DNA glycosylases that recognize oxidized bases (the other is the HTH-GPD superfamily exemplified by Escherichia coli Nth). Most FpgNei DNA glycosylases use their N-terminal proline residue as the key catalytic nucleophile, and the reaction proceeds via a Schiff base intermediate. One exception is mouse Nei-like glycosylase 3 (Neil3) which forms a Schiff base intermediate via its N-terminal valine. In addition to this FpgNei_N domain, FpgNei proteins have a helix-two-turn-helix (H2TH) domain and a zinc (or zincless)-finger motif which also contribute residues to the active site. FpgNei DNA glycosylases have a broad substrate specificity. They are bifunctional, in addition to the glycosylase (recognition) activity, they have a lyase (cleaving) activity on the phosphodiester backbone of the DNA at the AP site. This superfamily includes eukaryotic, bacterial, and viral proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>50</td>
            <td id=co>56</td>
            <td id=al>PRK10445 endonuclease VIII; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=al>cd08976 Uncharacterized bacterial subgroup of the N-terminal domain of Fpg (formamidopyrimidine-DNA glycosylase, MutM)_Nei  (endonuclease VIII) base-excision repair DNA glycosylases. This family is an uncharacterized bacterial subgroup of the FpgNei_N domain superfamily. DNA glycosylases maintain genome integrity by recognizing base lesions created by ionizing radiation, alkylating or oxidizing agents, and endogenous reactive oxygen species. They initiate the base-excision repair process, which is completed with the help of enzymes such as phosphodiesterases, AP endonucleases, DNA polymerases and DNA ligases. DNA glycosylases cleave the N-glycosyl bond between the sugar and the damaged base, creating an AP (apurinic/apyrimidinic) site. Most FpgNei DNA glycosylases use their N-terminal proline residue as the key catalytic nucleophile, and the reaction proceeds via a Schiff base intermediate. This N-terminal proline is conserved in this family. Escherichia coli Fpg prefers 8-oxo-7,8-dihydroguanine (8-oxoG) and oxidized purines and Escherichia coli Nei recognizes oxidized pyrimidines. However, neither Escherichia coli Fpg or Nei belong to this family. In addition to this BaFpgNei_N_4 domain, most enzymes belonging to this family contain a helix-two turn-helix (H2TH) domain and a zinc-finger motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>40</td>
            <td id=cg>98</td>
            <td id=al>cd08973 Uncharacterized bacterial subgroup of the N-terminal domain of Fpg (formamidopyrimidine-DNA glycosylase, MutM)_Nei  (endonuclease VIII) base-excision repair DNA glycosylases. This family is an uncharacterized bacterial subgroup of the FpgNei_N domain superfamily. DNA glycosylases maintain genome integrity by recognizing base lesions created by ionizing radiation, alkylating or oxidizing agents, and endogenous reactive oxygen species. They initiate the base-excision repair process, which is completed with the help of enzymes such as phosphodiesterases, AP endonucleases, DNA polymerases and DNA ligases. DNA glycosylases cleave the N-glycosyl bond between the sugar and the damaged base, creating an AP (apurinic/apyrimidinic) site. Most FpgNei DNA glycosylases use their N-terminal proline residue as the key catalytic nucleophile, and the reaction proceeds via a Schiff base intermediate. This N-terminal proline is conserved in this family. Escherichia coli Fpg prefers 8-oxo-7,8-dihydroguanine (8-oxoG) and oxidized purines and Escherichia coli Nei recognizes oxidized pyrimidines. However, neither Escherichia coli Fpg or Nei belong to this family. In addition to this BaFpgNei_N_1 domain, enzymes belonging to this family contain a helix-two turn-helix (H2TH) domain and a zinc-finger motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>51</td>
            <td id=cg>100</td>
            <td id=al>cd08972 N-terminal domain of the plant and fungal Nei and related proteins. This family contains the N-terminal domain of plant and Fungi Nei and related proteins. It belongs to the FpgNei_N, [N-terminal domain of Fpg (formamidopyrimidine-DNA glycosylase, MutM)_Nei (endonuclease VIII)] domain superfamily. DNA glycosylases maintain genome integrity by recognizing base lesions created by ionizing radiation, alkylating or oxidizing agents, and endogenous reactive oxygen species. They initiate the base-excision repair process, which is completed with the help of enzymes such as phosphodiesterases, AP endonucleases, DNA polymerases and DNA ligases. DNA glycosylases cleave the N-glycosyl bond between the sugar and the damaged base, creating an AP (apurinic/apyrimidinic) site. Most FpgNei DNA glycosylases use their N-terminal proline residue as the key catalytic nucleophile, and the reaction proceeds via a Schiff base intermediate. The plant and fungal FpgNei glycosylases prefer the oxidized pyrimidines spiroiminodihydantoin (Sp), guanidinohydantoin (Gh) over 8-oxoguanine in double stranded oligonucleotides and also show weak activity on single stranded DNA. In addition to this domain, enzymes belonging to this family contain a helix-two turn-helix (H2TH) domain and a characteristic zincless finger motif. They share a common ancestor not shared with other eukaryotic members of the FpgNei family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cr>9</td>
            <td id=cg>100</td>
            <td id=al>pfam06827 Zinc finger found in FPG and IleRS. This zinc binding domain is found at the C-terminus of isoleucyl tRNA synthetase and the enzyme Formamidopyrimidine-DNA glycosylase EC:3.2.2.23.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>506622</td></tr>
<tr><td id=al>end</td><td id=al>507504</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA-formamidopyrimidine glycosylase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>294</td></tr>
</table>

    </div>
</body>
</html>
