<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03452</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03452 DNA polymerase I [Phormidium_A]</h3>
    <div>MEENPSSRRVPVSSVLLVDGHSLAFRAYYAFAKSREGGLRTSTGLPTSVCFGFLRSLLDSIATHQPSAVAIAFDTSTPTFRHETDANYKAQRAETPEDFIPDVENLKKLLEALNFQVFTAPGYEADDILGTLAQQAHQQQQRVQLLSGDQDLFQLIAEQPPITVLHLAGGWGRKNTRPQEFQAQEVQEKLGILPQQVVDYKALCGDSSDNLPGVKGIGPKTAVKLLAEYGSLDEIYAHLDEIKGATGKKLEAGREAAYHSQYMATINCEVPLKVNWEQLQLRGFEPERVLPLLEQLELHSLRNSLERLQEQLGGTLETTPTQPSEDQPEDTGFWSWQETQASQPPEESRLQPQIIDSEAKLEQLLEQLQTQTDPNSPVAWDTETTSTKATEADLVGLGCCWGNAPQDLAYIPLGHHQGQQLAQDKVLQRLRPILEGDRYPKVLQNAKYDRLVLRTQGIQLKGVVFDTLLASYLINPETTHNLTDLALRYLDLTVSSFKEIVPKGKTMADVSIPVAAHYCAEDVHVTYRLVPLLRAQLQPTPQLEALLQDVELPLEPVLAEMEWLGIRLDIDYLHQLSEQLGQQLQQLEQQAYELAEQSFNLNSPKQLGTILFEQLGLDKRKTSQTKTGNYSTNVKVLEKLQGDHPIVDIILEYRSLAKLKSTYVDALPELVQLDEQQAPRIHTDFNQTITATGRLSSSNPNLQNIPIRTEFSRHIRKAFLPRQGWTLVTADYSQIELRILAHLSQEPILLEAYNTGQDVHRLTAQLLFEKPDISPEERRLGKTINFGVIYGMGAQRFAREAGVSTLEGREFIDRYNSRYSRVFEYLQQMKREAIARGYVETLCGRRRYFNFESPSLRKLKGESPEQIDLDSLSKISLADSQALRAAANAPIQGSSADIIKIAMVKLHQALGDRAARILLQVHDELVLEMPQEEWQPLELTIRDTMESALKLSIPLQVEIHAGENWMEAK</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>DNA polymerase I</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I PolA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>DNA polymerase I</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>TPA: DNA polymerase I</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA polymerase I</td>
            <td id=cr>Nodosilinea</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>45</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>PRK05755 DNA polymerase I; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TIGR00593 DNA polymerase I. All proteins in this family for which functions are known are DNA polymerases Many also have an exonuclease motif. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=al>COG0749 DNA polymerase I - 3'-5' exonuclease and polymerase domains [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=al>pfam00476 DNA polymerase family A. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=al>cd08637 Polymerase I functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. Family A polymerase (polymerase I) functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I (pol I) ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5'-3' polymerase and 5'-3' exonuclease. Pol I uses its 5' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used to search for protein signatures. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=al>cd08638 DNA polymerase theta is a low-fidelity family A enzyme implicated in translesion synthesis and in somatic hypermutation. DNA polymerase theta is a low-fidelity family A enzyme implicated in translesion synthesis (TLS) and in somatic hypermutation (SHM). DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. Pol theta is an exception among family A polymerases and generates processive single base substitutions. Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I (pol I) ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5'-3' polymerase and 5'-3' exonuclease. Pol I uses its 5' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. Polymerase theta mostly has amino-terminal helicase domain, a carboxy-terminal polymerase domain and an intervening space region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=al>smart00475 5'-3' exonuclease. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>55</td>
            <td id=cg>97</td>
            <td id=al>PRK14975 bifunctional 3'-5' exonuclease/DNA polymerase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>34</td>
            <td id=cg>96</td>
            <td id=al>cd06444 Family A polymerase primarily fills DNA gaps that arise during DNA repair, recombination and replication. DNA polymerase family A, 5'-3' polymerase domain. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified into six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaeota polymerase II (class D), human polymerase  beta (class X), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerases are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I, mitochondrial polymerase gamma, and several bacteriophage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic polymerase I (pol I) has two functional domains located on the same polypeptide; a 5'-3' polymerase and a 5'-3' exonuclease. Pol I uses its 5' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and the DNA polymerase activity to fill in the resulting gap. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=al>cd08639 Phylum Aquificae Pol A is different from Escherichia coli  Pol A by three signature sequences. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5'-3' polymerase and 5'-3' exonuclease. Pol I uses its 5' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used for phylogenetic anaylsis of bacteria. Species of the phylum Aquificae grow in extreme thermophilic environments. The Aquificae are non-spore-forming, Gram-negative rods and strictly thermophilic. Phylum Aquificae Pol A is different from E. coli Pol I by three signature sequences consisting of a 2 amino acids (aa) insert, a 5-6 aa insert and a 6 aa deletion. These signature sequences may provide a molecular marker for the family Aquificaceae and related species.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>31</td>
            <td id=cg>97</td>
            <td id=al>COG0258 5'-3' exonuclease [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=al>smart00482 DNA polymerase A domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>36</td>
            <td id=cg>94</td>
            <td id=al>cd08640 DNA polymerase A type from plastids of higher plants possibly involve in DNA replication or in the repair of errors occurring during replication. DNA polymerase A type from plastids of higher plants possibly involve in DNA replication or in the repair of errors occurring during replication. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7).   The three-dimensional structure of plastid DNA polymerase has substantial similarity to Pol I. The structure of Pol I resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=al>PRK14976 5'-3' exonuclease; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=al>cd06139 DEDDy 3'-5' exonuclease domain of Escherichia coli DNA polymerase I and similar bacterial family-A DNA polymerases. Escherichia coli-like Polymerase I (Pol I), a subgroup of family-A DNA polymerases, contains a DEDDy-type DnaQ-like 3'-5' exonuclease domain in the same polypeptide chain as the polymerase domain. The exonuclease domain contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. The 3'-5' exonuclease domain of DNA polymerases has a fundamental role in reducing polymerase errors and is involved in proofreading activity. E. coli DNA Pol I is involved in genome replication but is not the main replicating enzyme. It is also implicated in DNA repair.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=al>pfam02739 5'-3' exonuclease, N-terminal resolvase-like domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>cd09859 FEN-like PIN domains of PIN domain of the 5'-3' exonuclease of Thermus aquaticus DNA polymerase I (Taq) and homologs. The 5'-3' exonuclease (53EXO) PIN (PilT N terminus) domain of multi-domain DNA polymerase I and single domain protein homologs are included in this family. Taq contains a polymerase domain for synthesizing a new DNA strand and a 53EXO PIN domain for cleaving RNA primers or damaged DNA strands. Taq's 53EXO PIN domain recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp. The 53EXO PIN domain cleaves the unpaired 5'-arm of the overlap flap DNA substrate. 5'-3' exonucleases are members of the structure-specific, 5' nuclease family (FEN-like) that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. Canonical members of the FEN-like family possess a PIN domain with a two-helical structure insert (also known as the helical arch, helical clamp or I domain) of variable length (approximately 16 to 800 residues), and at the C-terminus of the PIN domain a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region. Both the H3TH domain (not included in this model) and the helical arch/clamp region are involved in DNA binding. The PIN domain belongs to a large nuclease superfamily. The structural properties of the PIN domain indicate its putative active center, consisting of invariant acidic amino acid residues (putative metal-binding residues), is geometrically similar in the active center of structure-specific 5' nucleases, PIN-domain ribonucleases of eukaryotic rRNA editing proteins, and bacterial toxins of toxin-antitoxin (TA) operons.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cr>9</td>
            <td id=cg>100</td>
            <td id=al>pfam01367 5'-3' exonuclease, C-terminal SAM fold. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>23</td>
            <td id=cg>87</td>
            <td id=al>PRK09482 flap endonuclease-like protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cr>7</td>
            <td id=cg>100</td>
            <td id=al>cd09898 H3TH domain of the 5'-3' exonuclease of Taq DNA polymerase I and homologs. H3TH (helix-3-turn-helix) domains of the 5'-3' exonuclease (53EXO) of mutli-domain DNA polymerase I and single domain protein homologs are included in this family. Taq DNA polymerase I contains a polymerase domain for synthesizing a new DNA strand and a 53EXO domain for cleaving RNA primers or damaged DNA strands. Taq's 53EXO recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp. The 53EXO cleaves the unpaired 5'-arm of the overlap flap DNA substrate. 5'-3' exonucleases are members of the structure-specific, 5' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+ or Mn2+ or Zn2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and two Asp residues from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5' nucleases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>17</td>
            <td id=cg>95</td>
            <td id=al>cd06140 inactive DEDDy 3'-5' exonuclease domain of Bacillus stearothermophilus DNA polymerase I and similar family-A DNA polymerases. Bacillus stearothermophilus-like Polymerase I (Pol I), a subgroup of the family-A DNA polymerases, contains an inactive DnaQ-like 3'-5' exonuclease domain in the same polypeptide chain as the polymerase region. The exonuclease-like domain of these proteins possess the same fold as the Klenow fragment (KF) of Escherichia coli Pol I, but does not contain the four critical metal-binding residues necessary for activity. The function of this domain is unknown. It might act as a spacer between the polymerase and the 5'-3' exonuclease domains. Some members of this subgroup, such as those from Bacillus sphaericus and Thermus aquaticus, are thermostable DNA polymerases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=al>pfam01612 3'-5' exonuclease. This domain is responsible for the 3'-5' exonuclease proofreading activity of E. coli DNA polymerase I (polI) and other enzymes, it catalyzes the hydrolysis of unpaired or mismatched nucleotides. This domain consists of the amino-terminal half of the Klenow fragment in E. coli polI it is also found in the Werner syndrome helicase (WRN), focus forming activity 1 protein (FFA-1) and ribonuclease D (RNase D). Werner syndrome is a human genetic disorder causing premature aging; the WRN protein has helicase activity in the 3'-5' direction. The FFA-1 protein is required for formation of a replication foci and also has helicase activity; it is a homolog of the WRN protein. RNase D is a 3'-5' exonuclease involved in tRNA processing. Also found in this family is the autoantigen PM/Scl thought to be involved in polymyositis-scleroderma overlap syndrome.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cr>7</td>
            <td id=cg>100</td>
            <td id=al>cd00080 H3TH domains of structure-specific 5' nucleases (or flap endonuclease-1-like) involved in DNA replication, repair, and recombination. The 5' nucleases of this superfamily are capable of both 5'-3' exonucleolytic activity and cleaving bifurcated or branched DNA, in an endonucleolytic, structure-specific manner, and are involved in DNA replication, repair, and recombination. The superfamily includes the H3TH (helix-3-turn-helix) domains of Flap Endonuclease-1 (FEN1), Exonuclease-1 (EXO1), Mkt1, Gap Endonuclease 1 (GEN1) and Xeroderma pigmentosum complementation group G (XPG) nuclease. Also included are the H3TH domains of the 5'-3' exonucleases of DNA polymerase I and single domain protein homologs, as well as, the bacteriophage T4 RNase H, T5-5'nuclease, and other homologs. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the C-terminal region of the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. Typically, the nucleases within this superfamily have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+, Mn2+, Zn2+, or Co2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one or two Asp residues from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5' nucleases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=al>smart00474 3'-5' exonuclease. 3\' -5' exonuclease proofreading domain present in DNA polymerase I, Werner syndrome helicase, RNase D and other enzymes</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>16</td>
            <td id=cg>99</td>
            <td id=al>cd00008 FEN-like PIN domains of the 5'-3' exonucleases of DNA polymerase I, bacteriophage T4 RNase H and T5-5' nucleases, and homologs. PIN (PilT N terminus) domains of the 5'-3' exonucleases (53EXO) of multi-domain DNA polymerase I and single domain protein homologs, as well as, the PIN domains of bacteriophage T5-5'nuclease (T5FEN or 5'-3'exonuclease), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar nucleases are included in this family. The 53EXO of DNA polymerase I recognizes and endonucleolytically cleaves a structure-specific DNA substrate that has a bifurcated downstream duplex and an upstream template-primer duplex that overlaps the downstream duplex by 1 bp. The T5-5'nuclease is a 5'-3'exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5'- 3'exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. These nucleases are members of the structure-specific, 5' nuclease family (FEN-like) that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. Canonical members of the FEN-like family possess a PIN domain with a two-helical structure insert (also known as the helical arch, helical clamp or I domain) of variable length (approximately 16 to 800 residues), and at the C-terminus of the PIN domain a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region. Both the H3TH domain (not included in this model) and the helical arch/clamp region are involved in DNA binding. The PIN domain belongs to a large nuclease superfamily. The structural properties of the PIN domain indicate its putative active center, consisting of invariant acidic amino acid residues (putative metal-binding residues), is geometrically similar in the active center of structure-specific 5' nucleases, PIN-domain ribonucleases of eukaryotic rRNA editing proteins, and bacterial toxins of toxin-antitoxin (TA) operons.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>15</td>
            <td id=cg>95</td>
            <td id=al>cd09860 FEN-like PIN domains of bacteriophage T3, T4 RNase H, T5-5'nuclease, and homologs. PIN (PilT N terminus) domain of bacteriophage T5-5'nuclease (5'-3' exonuclease or T5FEN), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar 5' nucleases are included in this family. T5-5'nuclease is a 5'-3'exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5'- 3'exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. Bacteriophage T3 is believed to function in the removal of DNA-linked RNA primers and is essential for phage DNA replication and also necessary for host DNA degradation and phage genetic recombination. These nucleases are members of the structure-specific, 5' nuclease family (FEN-like) that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. Canonical members of the FEN-like family possess a PIN domain with a two-helical structure insert (also known as the helical arch, helical clamp or I domain) of variable length (approximately 16 to 800 residues), and at the C-terminus of the PIN domain a H3TH (helix-3-turn-helix) domain, an atypical helix-hairpin-helix-2-like region. Both the H3TH domain (not included in this model) and the helical arch/clamp region are involved in DNA binding. The PIN domain belongs to a large nuclease superfamily. The structural properties of the PIN domain indicate its putative active center, consisting of invariant acidic amino acid residues (putative metal-binding residues), is geometrically similar in the active center of structure-specific 5' nucleases, PIN-domain ribonucleases of eukaryotic rRNA editing proteins, and bacterial toxins of toxin-antitoxin (TA) operons. In the T5-5'nuclease, structure-specific endonuclease activity requires binding of a single metal ion in the high-affinity, metal binding site 1, whereas exonuclease activity requires both, the high-affinity, metal binding site 1 and the low-affinity, metal binding site 2 to be occupied by a divalent cofactor. The T5-5'nuclease is reported to be able to bind several metal ions including, Mg2+, Mn2+, Zn2+ and Co2+, as co-factors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=co>42</td>
            <td id=cg>96</td>
            <td id=al>cd08643 Polymerase I functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. Family A polymerase functions primarily to fill DNA gaps that arise during DNA repair, recombination and replication. DNA-dependent DNA polymerases can be classified in six main groups based upon phylogenetic relationships with E. coli polymerase I (classA), E. coli polymerase II (class B), E.coli polymerase III (class C), euryarchaaeota polymerase II (class D), human polymerase  beta (class x), E. coli UmuC/DinB and eukaryotic RAP 30/Xeroderma pigmentosum variant (class Y). Family A polymerase are found primarily in organisms related to prokaryotes and include prokaryotic DNA polymerase I ,mitochondrial polymerase delta, and several bacteriphage polymerases including those from odd-numbered phage (T3, T5, and T7). Prokaryotic Pol Is have two functional domains located on the same polypeptide; a 5'-3' polymerase and 5'-3' exonuclease. Pol I uses its 5' nuclease activity to remove the ribonucleotide portion of newly synthesized Okazaki fragments and DNA polymerase activity to fill in the resulting gap. A combination of phylogenomic and signature sequence-based (or phonetic) approaches is used to understand the evolutionary relationships among bacteria. DNA polymerase I is one of the conserved proteins that is used to search for protein signatures. The structure of these polymerases resembles in overall morphology a cupped human right hand, with fingers (which bind an incoming nucleotide and interact with the single-stranded template), palm (which harbors the catalytic amino acid residues and also binds an incoming dNTP) and thumb (which binds double-stranded DNA) subdomains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>15</td>
            <td id=cg>100</td>
            <td id=al>cd06128 DEDDy 3'-5' exonuclease domain of family-A DNA polymerases. The 3'-5' exonuclease domain of family-A DNA polymerases has a fundamental role in reducing polymerase errors and is involved in proofreading activity. Family-A DNA polymerases contain a DnaQ-like exonuclease domain in the same polypeptide chain as the polymerase domain, similar to family-B DNA polymerases. The exonuclease domain contains three conserved sequence motifs termed ExoI, ExoII and ExoIII, which are clustered around the active site and contain four invariant acidic residues that serve as ligands for the two metal ions required for catalysis. The Klenow fragment (KF) of Escherichia coli Pol I, the Thermus aquaticus (Taq) Pol I, and Bacillus stearothermophilus (BF) Pol I are examples of family-A DNA polymerases. They are involved in nucleotide excision repair and in the processing of Okazaki fragments that are generated during lagging strand synthesis. The N-terminal domains of BF Pol I and Taq Pol I resemble the fold of the 3'-5' exonuclease domain of KF without the proofreading activity of KF. The four critical metal-binding residues are not conserved in BF Pol I and Taq Pol I, and they are unable to bind metals necessary for exonuclease activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cr>3</td>
            <td id=cg>100</td>
            <td id=al>smart00279 Helix-hairpin-helix class 2 (Pol1 family) motifs. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>15</td>
            <td id=cg>100</td>
            <td id=al>cd09018 DEDDy 3'-5' exonuclease domain of family-A DNA polymerases, RNase D, WRN, and similar proteins. DEDDy exonucleases, part of the DnaQ-like (or DEDD) exonuclease superfamily, catalyze the excision of nucleoside monophosphates at the DNA or RNA termini in the 3'-5' direction. They contain four invariant acidic residues in three conserved sequence motifs termed ExoI, ExoII and ExoIII. DEDDy exonucleases are classified as such because of the presence of a specific YX(3)D pattern at ExoIII. The four conserved acidic residues serve as ligands for the two metal ions required for catalysis. This family of DEDDy exonucleases includes the proofreading domains of family A DNA polymerases, as well as RNases such as RNase D and yeast Rrp6p. The Egalitarian (Egl) and Bacillus-like DNA Polymerase I subfamilies do not possess a completely conserved YX(3)D pattern at the ExoIII motif. In addition, the Bacillus-like DNA polymerase I subfamily has inactive 3'-5' exonuclease domains which do not possess the metal-binding residues necessary for activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>19</td>
            <td id=co>48</td>
            <td id=al>PTZ00217 flap endonuclease-1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=al>cd06142 DEDDy 3'-5' exonuclease domain of Ribonuclease D and similar proteins. Ribonuclease (RNase) D is a bacterial enzyme involved in the maturation of small stable RNAs and the 3' maturation of tRNA. It contains a DEDDy-type DnaQ-like 3'-5' exonuclease domain possessing three conserved sequence motifs termed ExoI, ExoII and ExoIII, with a specific YX(3)D pattern at ExoIII. These motifs are clustered around the active site and contain four conserved acidic residues that serve as ligands for the two metal ions required for catalysis. In vivo, RNase D only becomes essential upon removal of other ribonucleases. Eukaryotic RNase D homologs include yeast Rrp6p, human PM/Scl-100, and the Drosophila melanogaster egalitarian protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>5</td>
            <td id=cb>77</td>
            <td id=al>cd09899 H3TH domain of bacteriophage T3, T4 RNase H, T5-5' nucleases, and homologs. H3TH (helix-3-turn-helix) domains of bacteriophage T5-5'nuclease (5'-3' exonuclease or T5FEN), bacteriophage T4 RNase H (T4FEN), bacteriophage T3 (T3 phage exodeoxyribonuclease) and other similar 5' nucleases are included in this family. The T5-5'nuclease is a 5'-3' exodeoxyribonuclease that also exhibits endonucleolytic activity on flap structures (branched duplex DNA containing a free single-stranded 5'end). T4 RNase H, which removes the RNA primers that initiate lagging strand fragments, has 5'- 3' exonuclease activity on DNA/DNA and RNA/DNA duplexes and has endonuclease activity on flap or forked DNA structures. Bacteriophage T3 is believed to function in the removal of DNA-linked RNA primers and is essential for phage DNA replication and also necessary for host DNA degradation and phage genetic recombination. These nucleases are members of the structure-specific, 5' nuclease family that catalyzes hydrolysis of DNA duplex-containing nucleic acid structures during DNA replication, repair, and recombination. They contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors required for nuclease activity. The first metal binding site (MBS-1) is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site (MBS-2) is composed generally of two Asp residues from the PIN domain and two Asp residues from the H3TH domain. In the T5-5'nuclease, structure-specific endonuclease activity requires binding of a single metal ion in the high-affinity, MBS-1, whereas exonuclease activity requires both, the high-affinity, MBS-1 and the low-affinity, MBS-2 to be occupied by a divalent cofactor. The T5-5'nuclease is reported to be able to bind several metal ions including, Mg2+, Mn2+, Zn2+ and Co2+, as co-factors. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5' nucleases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>5</td>
            <td id=cb>70</td>
            <td id=al>cd09907 H3TH domain of Flap Endonuclease-1, a structure-specific, divalent-metal-ion dependent, 5' nuclease: Eukaryotic homologs. Members of this subgroup include the H3TH (helix-3-turn-helix) domains of eukaryotic Flap endonuclease-1 (FEN1), 5' nucleases. FEN1 is involved in multiple DNA metabolic pathways, including DNA replication processes (5' flap DNA endonuclease activity and double stranded DNA 5'-exonuclease activity) and DNA repair processes (long-patch base excision repair) in eukaryotes and archaea. Interaction between FEN1 and PCNA (Proliferating cell nuclear antigen) is an essential prerequisite to FEN1's DNA replication functionality and stimulates FEN1 nuclease activity by 10-50 fold. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. The nucleases within this subfamily have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (Mg2+ or Mn2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one Asp residue from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5' nucleases. Also, FEN1 has a C-terminal extension containing residues forming the consensus PIP-box - Qxx(M/L/I)xxF(Y/F) which serves to anchor FEN1 to PCNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>6</td>
            <td id=cr>22</td>
            <td id=al>PRK03980 flap endonuclease-1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cr>4</td>
            <td id=cb>60</td>
            <td id=al>cd09897 H3TH domains of Flap endonuclease-1 (FEN1)-like structure specific 5' nucleases. The 5' nucleases within this family are capable of both 5'-3' exonucleolytic activity and cleaving bifurcated or branched DNA, in an endonucleolytic, structure-specific manner, and are involved in DNA replication, repair, and recombination. This family includes the H3TH (helix-3-turn-helix) domains of Flap Endonuclease-1 (FEN1), Exonuclease-1 (EXO1), Mkt1, Gap Endonuclease 1 (GEN1), Xeroderma pigmentosum complementation group G (XPG) nuclease, and other eukaryotic and archaeal homologs. These nucleases contain a PIN (PilT N terminus) domain with a helical arch/clamp region/I domain (not included here) and inserted within the PIN domain is an atypical helix-hairpin-helix-2 (HhH2)-like region. This atypical HhH2 region, the H3TH domain, has an extended loop with at least three turns between the first two helices, and only three of the four helices appear to be conserved. Both the H3TH domain and the helical arch/clamp region are involved in DNA binding. Studies suggest that a glycine-rich loop in the H3TH domain contacts the phosphate backbone of the template strand in the downstream DNA duplex. With the except of the Mkt1-like proteins, the nucleases within this family have a carboxylate rich active site that is involved in binding essential divalent metal ion cofactors (i. e., Mg2+, Mn2+, Zn2+, or Co2+) required for nuclease activity. The first metal binding site is composed entirely of Asp/Glu residues from the PIN domain, whereas, the second metal binding site is composed generally of two Asp residues from the PIN domain and one Asp residue from the H3TH domain. Together with the helical arch and network of amino acids interacting with metal binding ions, the H3TH region defines a positively charged active-site DNA-binding groove in structure-specific 5' nucleases.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2921236</td></tr>
<tr><td id=al>end</td><td id=al>2924146</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA polymerase I</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>970</td></tr>
</table>

    </div>
</body>
</html>
