<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05886</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05886 glutathione-disulfide reductase [Phormidium_A]</h3>
    <div>MTYDFDLFVIGAGSGGIATARRAAEYGAKVGIAEDDRLGGTCVNRGCVPKKLMVYASHFPEHFHDAVGYGWSAVQSQLDWPKMIKAVNDEVTRLNGIYQRMLDKSEVQVFRESARFVDSHTVQVGDRQVTADKILIAVGGKPVKPNIPGLEHAITSDEMFHLPQQPNHLVVLGGGYIGVEFACILRGLGSEVTVVIRREKILNGFDEDLRDCIQEGMANHGIRILNNTNITSVEKQEDGTLKLNLDSDEGSIIADTITLSATGRKPRLEGLGLENTQVEVKKGAIAVDQYSQTAEPHIFAVGDCTDRVNLTPVAINEGRVFADTHFGGKSRQMSYENIPTAVFSSPEASTVGLTEAEAREQYGDNIKVFRSKFRPMYHTLGGREERTLMKLVVDSQSDRVLGAHMVGESAAEIIQGVAIAVKMGATKADFDATVGIHPSSAEEFVTMR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=co>SIO2C1</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=cr>Acaryochloris</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glutathione-disulfide reductase</td>
            <td id=co>Pleurocapsa</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK06116 glutathione reductase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TIGR01424 glutathione-disulfide reductase, plant. The tripeptide glutathione is an important reductant, e.g., for maintaining the cellular thiol/disulfide status and for protecting against reactive oxygen species such as hydrogen peroxide. Glutathione-disulfide reductase regenerates reduced glutathione from oxidized glutathione (glutathione disulfide) + NADPH. This model represents one of two closely related subfamilies of glutathione-disulfide reductase. Both are closely related to trypanothione reductase, and separate models are built so each of the three can describe proteins with conserved function. This model describes glutathione-disulfide reductases of plants and some bacteria, including cyanobacteria. [Energy metabolism, Electron transport]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>PLN02507 glutathione reductase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>PLN02546 glutathione reductase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG1249 Pyruvate/2-oxoglutarate dehydrogenase complex, dihydrolipoamide dehydrogenase (E3) component or related enzyme [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>TIGR01421 glutathione-disulfide reductase, animal/bacterial. The tripeptide glutathione is an important reductant, e.g., for maintaining the cellular thiol/disulfide status and for protecting against reactive oxygen species such as hydrogen peroxide. Glutathione-disulfide reductase regenerates reduced glutathione from oxidized glutathione (glutathione disulfide) + NADPH. This model represents one of two closely related subfamilies of glutathione-disulfide reductase. Both are closely related to trypanothione reductase, and separate models are built so each of the three can describe proteins with conserved function. This model describes glutathione-disulfide reductases of animals, yeast, and a number of animal-resident bacteria. [Energy metabolism, Electron transport]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TIGR01438 thioredoxin and glutathione reductase selenoprotein. This homodimeric, FAD-containing member of the pyridine nucleotide disulfide oxidoreductase family contains a C-terminal motif Cys-SeCys-Gly, where SeCys is selenocysteine encoded by TGA (in some sequence reports interpreted as a stop codon). In some members of this subfamily, Cys-SeCys-Gly is replaced by Cys-Cys-Gly. The reach of the selenium atom at the C-term arm of the protein is proposed to allow broad substrate specificity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>PTZ00058 glutathione reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TIGR01423 trypanothione-disulfide reductase. Trypanothione, a glutathione-modified derivative of spermidine, is (in its reduced form) an important antioxidant found in trypanosomatids (Crithidia, Leishmania, Trypanosoma). This model describes trypanothione reductase, a possible antitrypanosomal drug target closely related to some forms of glutathione reductase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>PRK06292 dihydrolipoamide dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>PRK06370 FAD-containing oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>PRK06416 dihydrolipoamide dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>99</td>
            <td id=cg>96</td>
            <td id=al>TIGR01350 dihydrolipoamide dehydrogenase. This model describes dihydrolipoamide dehydrogenase, a flavoprotein that acts in a number of ways. It is the E3 component of dehydrogenase complexes for pyruvate, 2-oxoglutarate, 2-oxoisovalerate, and acetoin. It can also serve as the L protein of the glycine cleavage system. This family includes a few members known to have distinct functions (ferric leghemoglobin reductase and NADH:ferredoxin oxidoreductase) but that may be predicted by homology to act as dihydrolipoamide dehydrogenase as well. The motif GGXCXXXGCXP near the N-terminus contains a redox-active disulfide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>99</td>
            <td id=cg>96</td>
            <td id=al>TIGR02053 mercury(II) reductase. This model represents the mercuric reductase found in the mer operon for the detoxification of mercury compounds. MerA is a FAD-containing flavoprotein which reduces Hg(II) to Hg(0) utilizing NADPH. [Cellular processes, Detoxification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>PTZ00052 thioredoxin reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>PRK05976 dihydrolipoamide dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>PRK07846 mycothione reductase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>96</td>
            <td id=cg>93</td>
            <td id=al>PRK05249 Si-specific NAD(P)(+) transhydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TIGR03452 mycothione reductase. Mycothiol, a glutathione analog in Mycobacterium tuberculosis and related species, can form a disulfide-linked dimer called mycothione. This enzyme can reduce mycothione to regenerate two mycothiol molecules. The enzyme shows some sequence similarity to glutathione-disulfide reductase, trypanothione-disulfide reductase, and dihydrolipoamide dehydrogenase. The characterized protein from M. tuberculosis, a homodimer, has FAD as a cofactor, one per monomer, and uses NADPH as a substrate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=al>pfam07992 Pyridine nucleotide-disulphide oxidoreductase. This family includes both class I and class II oxidoreductases and also NADH oxidases and peroxidases. This domain is actually a small NADH binding domain within a larger FAD binding domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>94</td>
            <td id=cg>92</td>
            <td id=al>PRK06912 dihydrolipoamide dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>96</td>
            <td id=cb>77</td>
            <td id=al>PRK13748 putative mercuric reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK07251 FAD-containing oxidoreductase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>PRK06327 dihydrolipoamide dehydrogenase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>94</td>
            <td id=cg>88</td>
            <td id=al>PRK14727 putative mercuric reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>95</td>
            <td id=cg>92</td>
            <td id=al>PRK14694 putative mercuric reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>PRK06467 dihydrolipoamide dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK07818 dihydrolipoamide dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>PRK06115 dihydrolipoamide dehydrogenase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK08010 pyridine nucleotide-disulfide oxidoreductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>PRK07845 flavoprotein disulfide reductase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>PTZ00153 lipoamide dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=al>pfam02852 Pyridine nucleotide-disulphide oxidoreductase, dimerization domain. This family includes both class I and class II oxidoreductases and also NADH oxidases and peroxidases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cb>79</td>
            <td id=cb>79</td>
            <td id=al>PRK09564 coenzyme A disulfide reductase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>75</td>
            <td id=cg>83</td>
            <td id=al>COG1252 NADH dehydrogenase, FAD-containing subunit [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>44</td>
            <td id=co>53</td>
            <td id=al>PRK04965 NADH:flavorubredoxin reductase NorW. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>COG0446 NADPH-dependent 2,4-dienoyl-CoA reductase, sulfur reductase, or a related oxidoreductase [Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>16</td>
            <td id=cg>90</td>
            <td id=al>pfam00070 Pyridine nucleotide-disulphide oxidoreductase. This family includes both class I and class II oxidoreductases and also NADH oxidases and peroxidases. This domain is actually a small NADH binding domain within a larger FAD binding domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cb>67</td>
            <td id=cr>37</td>
            <td id=al>COG1251 NAD(P)H-nitrite reductase, large subunit [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=al>COG0492 Thioredoxin reductase [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>50</td>
            <td id=co>53</td>
            <td id=al>PTZ00318 NADH dehydrogenase-like protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=co>41</td>
            <td id=co>47</td>
            <td id=al>PRK09754 phenylpropionate dioxygenase ferredoxin reductase subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>51</td>
            <td id=cr>35</td>
            <td id=al>PRK12814 putative NADPH-dependent glutamate synthase small subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>47</td>
            <td id=co>48</td>
            <td id=al>PRK13512 coenzyme A disulfide reductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=co>41</td>
            <td id=cb>62</td>
            <td id=al>pfam13738 Pyridine nucleotide-disulphide oxidoreductase. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4825338</td></tr>
<tr><td id=al>end</td><td id=al>4826685</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[Metabolism, Citric Acid/Krebs/TCA cycle|pyruvate/2-oxoglutarate dehydrogenase, dihydrolipoamide dehydrogenase (E3) component|1.0] [Metabolism, Pyruvate|pyruvate/2-oxoglutarate dehydrogenase, dihydrolipoamide dehydrogenase (E3) component|1.0]</td></tr>
<tr><td id=al>descr</td><td id=al>glutathione-disulfide reductase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>449</td></tr>
</table>

    </div>
</body>
</html>
