<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00165</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00165 exodeoxyribonuclease III [Phormidium_A]</h3>
    <div>MKIATWNVNSIRTRQGLVCDWLRANPVEVLCVQETKVIDADFPRDPFEALGYELAVSGQKSYNGVAILSRLPLEQVDIGFSALLPPEQVGDLDEQKRAIAATVAGVRLINLYVPNGGDYNSDKYQYKLRWLALLYDYIQKTLDKNQNLCICGDFNIAPDDRDIHTPPGKTPVVGTTAAEREALQRCLSLGLADAFRKFNQEAQQYSWWDYRAGAFPRNRGWRIDHHYLTPELYQRATACTIDREPRQWTKPSDHTPVIVEWPD</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=co>Lyngbya</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>exodeoxyribonuclease III</td>
            <td id=cb>RDYZ01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>56</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>cd09086 Escherichia coli exonuclease III (ExoIII) and Neisseria meningitides NExo-like subfamily of the ExoIII family purinic/apyrimidinic (AP) endonucleases. This subfamily includes Escherichia coli ExoIII, Neisseria meningitides NExo,and related proteins. These are ExoIII family AP endonucleases and they belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER, the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, and this is followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, and have both mutagenic and cytotoxic effects. AP endonucleases can carry out a variety of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiencies. Many organisms have two AP endonucleases, usually one is the dominant AP endonuclease, the other has weak AP endonuclease activity. For example, Neisseria meningitides Nape and NExo, and exonuclease III (ExoIII) and endonuclease IV (EndoIV) in Escherichia coli. NExo and ExoIII  are found in this subfamily. NExo is the non-dominant AP endonuclease. It exhibits strong 3'-5' exonuclease and 3'-deoxyribose phosphodiesterase activities. Escherichia coli ExoIII is an active AP endonuclease, and in addition, it exhibits double strand (ds)-specific 3'-5' exonuclease, exonucleolytic RNase H, 3'-phosphomonoesterase and  3'-phosphodiesterase activities, all catalyzed by a single active site. Class II AP endonucleases have been classified into two families, designated ExoIII and EndoIV, based on their homology to the Escherichia coli enzymes ExoIII and endonuclease IV (EndoIV). This subfamily belongs to the ExoIII family; the EndoIV family belongs to a different superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG0708 Exonuclease III [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TIGR00195 exodeoxyribonuclease III. The model brings in reverse transcriptases at scores below 50, model also contains eukaryotic apurinic/apyrimidinic endonucleases which group in the same family [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>TIGR00633 exodeoxyribonuclease III (xth). All proteins in this family for which functions are known are 5' AP endonucleases that funciton in base excision repair and the repair of abasic sites in DNA.This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>cd09073 Escherichia coli exonuclease III (ExoIII)-like apurinic/apyrimidinic (AP) endonucleases. The ExoIII family AP endonucleases belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER, the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, which is then followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, which have both mutagenic and cytotoxic effects. AP endonucleases can carry out a wide range of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiences. Many organisms have two functional AP endonucleases, for example, APE1/Ref-1 and Ape2 in humans, Apn1 and Apn2 in bakers yeast, Nape and NExo in Neisseria meningitides, and exonuclease III (ExoIII) and endonuclease IV (EndoIV) in Escherichia coli. Usually, one of the two is the dominant AP endonuclease, the other has weak AP endonuclease activity, but exhibits strong 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, and 3'-phosphatase activities. Class II AP endonucleases have been classified into two families, designated ExoIII and EndoIV, based on their homology to the Escherichia coli enzymes. This family contains the ExoIII family; the EndoIV family belongs to a different superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd10281 Neisseria meningitides Nape-like subfamily of the ExoIII family purinic/apyrimidinic (AP) endonucleases. This subfamily includes Neisseria meningitides Nape and related proteins. These are Escherichia coli exonuclease III (ExoIII)-like AP endonucleases and belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, and this is followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, and have both mutagenic and cytotoxic effects. AP endonucleases can carry out a variety of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiences. Many organisms have two AP endonucleases, usually one is the dominant AP endonuclease, the other has weak AP endonuclease activity; for example, Neisseria meningitides Nape and NExo. Nape, found in this subfamily, is the dominant AP endonuclease. It exhibits strong AP endonuclease activity, and also exhibits 3'-5'exonuclease and 3'-deoxyribose phosphodiesterase activities.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK11756 exonuclease III; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd09085 Methanothermobacter thermautotrophicus Mth212-like subfamily of the ExoIII family purinic/apyrimidinic (AP) endonucleases. This subfamily includes the thermophilic archaeon Methanothermobacter thermautotrophicus Mth212and related proteins. These are Escherichia coli exonuclease III (ExoIII)-like AP endonucleases and they belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER, the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, and this is followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, and have both mutagenic and cytotoxic effects. AP endonucleases can carry out a variety of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiences. Mth212 is an AP endonuclease, and a DNA uridine endonuclease (U-endo) that nicks double-stranded DNA at the 5'-side of a 2'-d-uridine residue. After incision at the 5'-side of a 2'-d-uridine residue by Mth212, DNA polymerase B takes over the 3'-OH terminus and carries out repair synthesis, generating a 5'-flap structure that is resolved by a 5'-flap endonuclease. Finally, DNA ligase seals the resulting nick. This U-endo activity shares the same catalytic center as its AP-endo activity, and is absent from other AP endonuclease homologues.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd09087 Human Ape1-like subfamily of the ExoIII family apurinic/apyrimidinic (AP) endonucleases. This subfamily includes human Ape1 (also known as Apex, Hap1, or Ref-1) and related proteins. These are Escherichia coli exonuclease III (ExoIII)-like AP endonucleases and they belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER, the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, and this is followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, and have both mutagenic and cytotoxic effects. AP endonucleases can carry out a variety of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiences. Many organisms have two AP endonucleases, usually one is the dominant AP endonuclease, the other has weak AP endonuclease activity; for example, Ape1 and Ape2 in humans. Ape1 is found in this subfamily, it exhibits strong AP-endonuclease activity but shows weak 3'-5' exonuclease and 3'-phosphodiesterase activities. Class II AP endonucleases have been classified into two families, designated ExoIII and EndoIV, based on their homology to the Escherichia coli enzymes exonuclease III (ExoIII) and endonuclease IV (EndoIV). This subfamily belongs to the ExoIII family; the EndoIV family belongs to a different superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>pfam03372 Endonuclease/Exonuclease/phosphatase family. This large family of proteins includes magnesium dependent endonucleases and a large number of phosphatases involved in intracellular signalling. This family includes: AP endonuclease proteins EC:4.2.99.18, DNase I proteins EC:3.1.21.1, Synaptojanin an inositol-1,4,5-trisphosphate phosphatase EC:3.1.3.56, Sphingomyelinase EC:3.1.4.12, and Nocturnin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK13911 exodeoxyribonuclease III; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd09088 Human Ape2-like subfamily of the ExoIII family purinic/apyrimidinic (AP) endonucleases. This subfamily includes human APE2, Saccharomyces cerevisiae Apn2/Eth1, and related proteins. These are Escherichia coli exonuclease III (ExoIII)-like AP endonucleases and they belong to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds. AP endonucleases participate in the DNA base excision repair (BER) pathway. AP sites are one of the most common lesions in cellular DNA. During BER, the damaged DNA is first recognized by DNA glycosylase. AP endonucleases then catalyze the hydrolytic cleavage of the phosphodiester bond 5' to the AP site, and this is followed by the coordinated actions of DNA polymerase, deoxyribose phosphatase, and DNA ligase. If left unrepaired, AP sites block DNA replication, and have both mutagenic and cytotoxic effects. AP endonucleases can carry out a variety of excision and incision reactions on DNA, including 3'-5' exonuclease, 3'-deoxyribose phosphodiesterase, 3'-phosphatase, and occasionally, nonspecific DNase activities. Different AP endonuclease enzymes catalyze the different reactions with different efficiences. Many organisms have two AP endonucleases, usually one is the dominant AP endonuclease, the other has weak AP endonuclease activity. For examples, Ape1 and Ape2 in humans, and Apn1 and Apn2 in bakers yeast. Ape2 and Apn2/Eth1 are both found in this subfamily, and have the weaker AP endonuclease activity. Ape2 shows strong 3'-5' exonuclease and 3'-phosphodiesterase activities; it can reduce the mutagenic consequences of attack by reactive oxygen species by removing 3'-end adenine opposite from 8-oxoG, in addition to repairing 3'-damaged termini. Apn2/Eth1 exhibits AP endonuclease activity, but has 30-40 fold more active 3'-phosphodiesterase and 3'-5' exonuclease activities. Class II AP endonucleases have been classified into two families, designated ExoIII and EndoIV, based on their homology to the Escherichia coli enzymes exonuclease III (ExoIII) and endonuclease IV (EndoIV). This subfamily belongs to the ExoIII family; the EndoIV family belongs to a different superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd09076 Endonuclease domain (L1-EN) of the non-LTR retrotransposon LINE-1 (L1), and related domains. This family contains the endonuclease domain (L1-EN) of the non-LTR retrotransposon LINE-1 (L1), and related domains, including the endonuclease of Xenopus laevis Tx1. These retrotranspons belong to the subtype 2, L1-clade. LINES can be classified into two subtypes. Subtype 2 has two ORFs: the second (ORF2) encodes a modular protein consisting of an N-terminal apurine/apyrimidine endonuclease domain (EN), a central reverse transcriptase, and a zinc-finger-like domain at the C-terminus. LINE-1/L1 elements (full length and truncated) comprise about 17% of the human genome. This endonuclease nicks the genomic DNA at the consensus target sequence 5'TTTT-AA3' producing a ribose 3'-hydroxyl end as a primer for reverse transcription of associated template RNA. This subgroup also includes the endonuclease of Xenopus laevis Tx1, another member of the L1-clade. This family belongs to the large EEP (exonuclease/endonuclease/phosphatase) superfamily that contains functionally diverse enzymes that share a common catalytic mechanism of cleaving phosphodiester bonds.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd08372 Exonuclease-Endonuclease-Phosphatase (EEP) domain superfamily. This large superfamily includes the catalytic domain (exonuclease/endonuclease/phosphatase or EEP domain) of a diverse set of proteins including the ExoIII family of apurinic/apyrimidinic (AP) endonucleases, inositol polyphosphate 5-phosphatases (INPP5), neutral sphingomyelinases (nSMases), deadenylases (such as the vertebrate circadian-clock regulated nocturnin), bacterial cytolethal distending toxin B (CdtB), deoxyribonuclease 1 (DNase1), the endonuclease domain of the non-LTR retrotransposon LINE-1, and related domains. These diverse enzymes share a common catalytic mechanism of cleaving phosphodiester bonds; their substrates range from nucleic acids to phospholipids and perhaps proteins.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>134696</td></tr>
<tr><td id=al>end</td><td id=al>135488</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>exodeoxyribonuclease III</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>264</td></tr>
</table>

    </div>
</body>
</html>
