<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01577</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01577 transcription-repair coupling factor [Phormidium_A]</h3>
    <div>MAFSSVVRTLGRTALTQELTSSLERDGRLQLDGVSRFPKGLMVSALAHQRQQHLCLIAATLEEAGRWAAQLQEMDWPLVHFYPTSEASPYEASDSDEITWGQLQVLADLVHPTARTSSRFAIVATQRALHPHLPSREQFAQSCLCLKLGLELSLGQLKHQLSELGYESVSLVEAEGQWSRRGDIIDIFPVAAELPLRLEFFGDELEKMREFDPVSQRSLDKVQQVVLTPCQLNQGLGDSLGQEESDLSPSAQVGLAFDAPSSLLDYLPPETLFVLDEPEACQGHSDRWVEHVEQDFQSHRDRHPHLSPLHRPIQESLGTCTLDRVDLRELVLSDRDADNQPGKRLNLASRRLASTPHQFAKIAQILRDERDRGFNIFLISAQPSRTVSLLQEHDCNAQFLPNPRDYPAVDRLQTHHVPIALKYAGQAELEGFILPTLRLVVVSDREFFGQHSLASPSYVRKRRRAASKQVDPNQLQPGDYVVHHNHGIGQFLQLESLTLNGETREYLVLRYADGTLRVAADQLNALSRYRHTGSGRPQLNKMSGKAWERTKNKVRKAIKKIAIDLLKLYAERAKQQGFAYPADIPWQQELEDSFPYQPTPDQLKATQDIKHDMESDRPMDRLVCGDVGFGKTEVAIRAVFKAITAGKQVAFLAPTTILTQQHYHTLKERFAPYPIQVGLLNRFRSPEERKEIQQRLMTGELDVVVGTHQLLGKSVKFKDLGLLVVDEEQRFGVNQKEKIKALKTQVDVLTLTATPIPRTLYMSLSGIREISLITTPPPSRRPIKTHLARYDWETIRGAIAQELDRGGQVFYVVPRVEGIEERSAKIREVVPSARIAIAHGQMDVAELESVMLAFSSGEADVLVCTTIVESGLDIPRVNTILIEDAHRFGLSQLYQLRGRVGRSGIQAHAWLFYSQRQQLSEAAQKRLRALQEFSQLGSGYQLAMRDMEIRGVGNLLGAEQSGQMNAIGFDLYRTMLEEEIREIQGQAIPKVNDTQVDLNLTAFIPSNYIPDLDHKMTAYRQVATADSGEALQAIALDWQDCYGPLPKAAHQLLRVMELKQLAKTLGFSRIKPEGKQHVILETPMEEPAWGRLKSHLPKHLQSRFVYGSGRVTVRGLGVLPAEKQLDSLIQWLECLRPALEDIQDTDNLAASAS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>transcription-repair coupling factor Mfd</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TPA: transcription-repair coupling factor</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=cr>PCC-7407</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcription-repair coupling factor</td>
            <td id=co>FACHB-831</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG1197 Transcription-repair coupling factor (superfamily II helicase) [Replication, recombination and repair, Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=al>TIGR00580 transcription-repair coupling factor (mfd). All proteins in this family for which functions are known are DNA-dependent ATPases that function in the process of transcription-coupled DNA repair in which the repair of the transcribed strand of actively transcribed genes is repaired at a higher rate than the repair of non-transcribed regions of the genome and than the non-transcribed strand of the same gene. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). This family is closely related to the RecG and UvrB families. [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>93</td>
            <td id=cg>93</td>
            <td id=al>PRK10689 transcription-repair coupling factor; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>35</td>
            <td id=cb>60</td>
            <td id=al>PRK10917 ATP-dependent DNA helicase RecG; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>35</td>
            <td id=cb>61</td>
            <td id=al>COG1200 RecG-like helicase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd17991 DEXH/Q-box helicase domain of the transcription-repair coupling factor. Transcription-repair coupling factor (TrcF) dissociates transcription elongation complexes blocked at nonpairing lesions and mediates recruitment of DNA repair proteins. TrcF is a member of the DEAD-like helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>37</td>
            <td id=cb>67</td>
            <td id=al>TIGR00643 ATP-dependent DNA helicase RecG. [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cr>13</td>
            <td id=cg>100</td>
            <td id=al>cd18810 C-terminal helicase domain of the transcription-repair coupling factor. Transcription-repair coupling factor (TrcF) dissociates transcription elongation complexes blocked at nonpairing lesions and mediates recruitment of DNA repair proteins. TrcF is a DEAD-like helicase belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>18</td>
            <td id=cg>94</td>
            <td id=al>cd17992 DEXH/Q-box helicase domain of RecG. ATP-dependent DNA helicase RecG plays a critical role in recombination and DNA repair. It is a member of the DEAD-like helicases superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cr>13</td>
            <td id=cg>100</td>
            <td id=al>cd18792 C-terminal helicase domain of the RecG family helicases. The DEAD-like helicase RecG family contains recombination factor RecG and transcription-repair coupling factor TrcF. They are DEAD-like helicases belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd17918 DEXH/Q-box helicase domain of DEAD-like helicase RecG family proteins. The DEAD-like helicase RecG family is part of the DEAD-like helicases superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>14</td>
            <td id=cg>100</td>
            <td id=al>cd18811 C-terminal helicase domain of DNA helicase RecG. ATP-dependent DNA helicase RecG plays a critical role in recombination and DNA repair. RecG helps process Holliday junction intermediates to mature products by catalyzing branch migration. It is a DEAD-like helicase belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cr>8</td>
            <td id=cg>100</td>
            <td id=al>smart01058 CarD-like/TRCF domain. CarD is a Myxococcus xanthus protein required for the activation of light- and starvation-inducible genes. This family includes the presumed N-terminal domain. CarD interacts with the zinc-binding protein CarG, to form a complex that regulates multiple processes in Myxococcus xanthus. This family also includes a domain to the N-terminal side of the DEAD helicase of TRCF proteins. TRCF displaces RNA polymerase stalled at a lesion, binds to the damage recognition protein UvrA, and increases the template strand repair rate during transcription. This domain is involved in binding to the stalled RNA polymerase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cr>7</td>
            <td id=cg>95</td>
            <td id=al>pfam17757 UvrB interaction domain. This domain is found in the UvrB protein where it interacts with the UvrA protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>8</td>
            <td id=cg>100</td>
            <td id=al>pfam02559 CarD-like/TRCF domain. CarD is a Myxococcus xanthus protein required for the activation of light- and starvation-inducible genes. This family includes the presumed N-terminal domain, CdnL. CarD interacts with the zinc-binding protein CarG to form a complex that regulates multiple processes in Myxococcus xanthus. This family also includes a domain to the N-terminal side of the DEAD helicase of TRCF (transcription-repair-coupling factor) proteins. TRCF displaces RNA polymerase stalled at a lesion, binds to the damage recognition protein UvrA, and increases the template strand repair rate during transcription. This domain is involved in binding to the stalled RNA polymerase. The family includes members otherwise referred to as CdnL, for CarD N-terminal like, whichdiffer functionally from CarD. The TRCF domain mentioned above is the RNA polymerase-interacting domain or RID.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>8</td>
            <td id=cg>100</td>
            <td id=al>smart00982 This domain is found in proteins necessary for strand-specific repair in DNA such as TRCF in Escherichia coli. A lesion in the template strand blocks the RNA polymerase complex (RNAP). The RNAP-DNA-RNA complex is specifically recognised by the transcription-repair-coupling factor (TRCF) which releases RNAP and the truncated transcript.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>15</td>
            <td id=cg>100</td>
            <td id=al>pfam00270 DEAD/DEAH box helicase. Members of this family include the DEAD and DEAH box helicases. Helicases are involved in unwinding nucleic acids. The DEAD box helicases are involved in various aspects of RNA metabolism, including nuclear transcription, pre mRNA splicing, ribosome biogenesis, nucleocytoplasmic transport, translation, RNA decay and organellar gene expression.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>smart00487 DEAD-like helicases superfamily. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>12</td>
            <td id=cg>100</td>
            <td id=al>cd00046 N-terminal DEAD/H-box helicase domain of superfamily 2 helicases. The DEAD/H-like superfamily 2 helicases comprise a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This N-terminal domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>8</td>
            <td id=cg>94</td>
            <td id=al>pfam00271 Helicase conserved C-terminal domain. The Prosite family is restricted to DEAD/H helicases, whereas this domain family is found in a wide variety of helicases and helicase related proteins. It may be that this is not an autonomously folding unit, but an integral part of the helicase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>7</td>
            <td id=cg>90</td>
            <td id=al>pfam03461 TRCF domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>31</td>
            <td id=cg>81</td>
            <td id=al>COG4098 Superfamily II DNA/RNA helicase required for DNA uptake (late competence protein)  [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=co>42</td>
            <td id=cb>67</td>
            <td id=al>COG1198 Primosomal protein N' (replication factor Y) - superfamily II helicase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>9</td>
            <td id=cr>16</td>
            <td id=al>COG0556 Excinuclease UvrABC helicase subunit UvrB [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>11</td>
            <td id=cb>71</td>
            <td id=al>cd17929 DEXH-box helicase domain of PriA. PriA, also known as replication factor Y or primosomal protein N', is a 3'-->5' superfamily 2 DNA helicase that acts to remodel stalled replication forks and as a specificity factor for origin-independent assembly of a new replisome at the stalled fork. PriA is a member of the DEAD-like helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>6</td>
            <td id=cg>86</td>
            <td id=al>smart00490 helicase superfamily c-terminal domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>8</td>
            <td id=cr>14</td>
            <td id=al>PRK05298 excinuclease ABC subunit UvrB. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>34</td>
            <td id=co>58</td>
            <td id=al>PRK05580 primosome assembly protein PriA; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>11</td>
            <td id=cb>75</td>
            <td id=al>cd17921 DEXH-box helicase domain of DEAD-like helicase Ski2 family proteins. Ski2-like RNA helicases play an important role in RNA degradation, processing, and splicing pathways. They belong to the type II DEAD box helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cr>27</td>
            <td id=cg>90</td>
            <td id=al>cd09639 CRISPR/Cas system-associated protein Cas3. CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats) and associated Cas proteins comprise a system for heritable host defense by prokaryotic cells against phage and other foreign DNA; DEAD/DEAH box helicase DNA helicase cas3'; Often but not always is fused to HD nuclease domain; signature gene for Type I</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1333288</td></tr>
<tr><td id=al>end</td><td id=al>1336750</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>transcription-repair coupling factor</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>1154</td></tr>
</table>

    </div>
</body>
</html>
