<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04760</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04760 TIGR01548 family HAD-type hydrolase [Phormidium_A]</h3>
    <div>MAHPSSSASEPLALVIFDIDGVIRDVGSSYRRALADTVEQFTQGAYRPSPEDIDQLKAEGIWNNDWEGSMELIYRYFESQGGDRDRLNLDFDEIVAFFQSRYRGESAENPEQWTGYICDEPILAQRPYFESLTAANIGWGFFSGATRGSAHYILQRRIGLESPVLVAMEDAPGKPDPTGLFATLEQLQASPTCPVLYVGDTVADMYVVENAAKQDGDRPWIAVGILPPHVQTHPDSQAKHRQRLEDAGGRLVLNQVTDLTPELILKKVKGS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>98</td>
            <td id=cg>96</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=al>HAD superfamily phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=al>TIGR01548 family HAD-type hydrolase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>93</td>
            <td id=cg>92</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cr>ULC077BIN1</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>95</td>
            <td id=cg>93</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cr>JAAUTN01</td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>95</td>
            <td id=cg>98</td>
            <td id=al>TIGR01548 family HAD-type hydrolase</td>
            <td id=cr>Myxacorys</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>33</td>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=al>TIGR01548 haloacid dehalogenase superfamily, subfamily IA hydrolase, TIGR01548. This model represents a small and phylogenetically curious clade of sequences. Sequences are found from Halobacterium (an archaeon), Nostoc and Synechococcus (cyanobacteria) and Phytophthora (a stramenophile eukaryote). These appear to be members of the haloacid dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases by general homology and the conservation of all of the recognized catalytic motifs. The variable domain is found in between motifs 1 and 2, indicating membership in subfamily I and phylogeny and prediction of the alpha helical nature of the variable domain (by PSI-PRED) indicate membership in subfamily IA. All but the Halobacterium sequence currently found are annotated as "Imidazoleglycerol-phosphate dehydratase", however, the source of the annotation could not be traced and significant homology could not be found between any of these sequences and known IGPD's.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=al>TIGR01549 haloacid dehalogenase superfamily, subfamily IA, variant 1 with third motif having Dx(3-4)D or Dx(3-4)E. This model represents part of one structural subfamily of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called "capping domain", or the absence of such a domain. Subfamily I consists of sequences in which the capping domain is found in between the first and second catalytic motifs. Subfamily II consists of sequences in which the capping domain is found between the second and third motifs. Subfamily III sequences have no capping domain in either of these positions.The Subfamily IA and IB capping domains are predicted by PSI-PRED to consist of an alpha helical bundle. Subfamily I encompasses such a wide region of sequence space (the sequences are highly divergent) that representing it with a single model is impossible, resulting in an overly broad description which allows in many unrelated sequences. Subfamily IA and IB are separated based on an aparrent phylogenetic bifurcation. Subfamily IA is still too broad to model, but cannot be further subdivided into large chunks based on phylogenetic trees. Of the three motifs defining the HAD superfamily, the third has three variant forms: (1) hhhhsDxxx(x)(D/E), (2) hhhhssxxx(x)D and (3) hhhhDDxxx(x)s where _s_ refers to a small amino acid and _h_ to a hydrophobic one. All three of these variants are found in subfamily IA. Individual models were made based on seeds exhibiting only one of the variants each. Variant 1 (this model) is found in the enzymes phosphoglycolate phosphatase (TIGR01449) and enolase-phosphatase. These three variant models (see also TIGR01493 and TIGR01509) were created withthe knowledge that there will be overlap among them - this is by design and serves the purpose of eliminating the overlap with models of more distantly relatedHAD subfamilies caused by an overly broad single model. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>COG0546 Phosphoglycolate phosphatase, HAD superfamily [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=al>pfam13419 Haloacid dehalogenase-like hydrolase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=al>pfam00702 haloacid dehalogenase-like hydrolase. This family is structurally different from the alpha/beta hydrolase family (pfam00561). This family includes L-2-haloacid dehalogenase, epoxide hydrolases and phosphatases. The structure of the family consists of two domains. One is an inserted four helix bundle, which is the least well conserved region of the alignment, between residues 16 and 96 of Pseudomonas sp. (S)-2-haloacid dehalogenase 1. The rest of the fold is composed of the core alpha/beta domain. Those members with the characteristic DxD triad at the N-terminus are probably phosphatidylglycerolphosphate (PGP) phosphatases involved in cardiolipin biosynthesis in the mitochondria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=al>COG0637 Beta-phosphoglucomutase or related phosphatase, HAD superfamily  [Carbohydrate transport and metabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3875286</td></tr>
<tr><td id=al>end</td><td id=al>3876102</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>TIGR01548 family HAD-type hydrolase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>272</td></tr>
</table>

    </div>
</body>
</html>
