<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01213</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01213 histone deacetylase [Phormidium_A]</h3>
    <div>MFPIFYAEAFLDHRTGAYHPERPDRLRAIVQALKSQPWAEHLHWQLPGDRPELLNIITQVHPRDYVESVKRIASRGGGRIDMDTSISPSSYDVALLAVSAWLDAVDLVVHQGQPSFVLARPPGHHAERDRGMGFCIFSNAAIAARYALKQFTIDRVAVLDWDVHHGNGTQAILETHPQLAYCSLHESPNYPGTGSANETGYYQNVLNIPLPPNTTSDTYRQVFQDKVLPFLSHFRPDLLIISAGYDGTKADTISQMLLQPEDYRQFSEWCLEITPHLMFGLEGGYELESLAESVVKTIEPFMTAQG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>histone deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>histone deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>histone deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>histone deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>histone deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>histone deacetylase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>histone deacetylase</td>
            <td id=co>Crinalium</td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>TPA: histone deacetylase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>histone deacetylase</td>
            <td id=cr>FACHB-46</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histone deacetylases and histone-like deacetylases, classII. Class II histone deacetylases are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. This group includes animal HDAC4,5,6,7,8,9,10, fungal HOS3 and HDA1, plant HDA5 and HDA15 as well as other eukaryotes, archaeal and bacterial histone-like deacetylases. Eukaryotic deacetylases mostly use histones (H2, H3, H4) as substrates for deacetylation; however, non-histone substrates are known (for example, tubulin). Substrates for prokaryotic histone-like deacetylases are not known. Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Interaction partners of class II deacetylases include 14-3-3 proteins, MEF2 family of transcriptional factors, CtBP, calmodulin (CaM), SMRT, N-CoR, BCL6, HP1alpha and SUMO. Histone deacetylases play a role in the regulation of cell cycle, cell differentiation and survival. Class II mammalian HDACs are differentially inhibited by structurally diverse compounds with known antitumor activities, thus presenting them as potential drug targets for human diseases resulting from aberrant acetylation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histone deacetylases and histone-like deacetylases, classII. This subfamily includes eukaryotic as well as bacterial Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. In D. discoideum, where four homologs (HdaA, HdaB, HdaC, HdaD) have been identified, HDAC activity is important for regulating the timing of gene expression during development. Also, inhibition of HDAC activity by trichostatin A is shown to cause hyperacetylation of the histone and a delay in cell aggregation and differentiation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>Acetoin utilization deacetylase AcuC or a related deacetylase [Chromatin structure and dynamics, Secondary metabolites biosynthesis, transport and catabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histone deacetylase domain. Histones can be reversibly acetylated on several lysine residues. Regulation of transcription is caused in part by this mechanism. Histone deacetylases catalyze the removal of the acetyl group. Histone deacetylases are related to other proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histone deacetylase class IIa. This subfamily includes bacterial acetylpolyamine amidohydrolase (APAH) as well as other Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. Mycoplana ramosa APAH exhibits broad substrate specificity and catalyzes the deacetylation of polyamines such as putrescine, spermidine, and spermine by cleavage of a non-peptide amide bond.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Histone deacetylases and histone-like deacetylases, classII. This subfamily includes bacterial as well as eukaryotic Class II histone deacetylase (HDAC) and related proteins. Deacetylases of class II are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) and possibly other proteins to yield deacetylated histones/other proteins. Included in this family is a bacterial HDAC-like amidohydrolase (Bordetella/Alcaligenes species FB18817, denoted as FB188 HDAH) shown to be most similar in sequence and function to class II HDAC6 domain 3 or b (HDAC6b). FB188 HDAH is able to remove the acetyl moiety from acetylated histones, and can be inhibited by common HDAC inhibitors such as SAHA (suberoylanilide hydroxamic acid) as well as class II-specific but not class I specific inhibitors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>Histone deacetylase 6, domain 1 and histone deacetylase 10. Histone deacetylases 6 and 10 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. HDAC10 has an N-terminal deacetylase domain and a C-terminal pseudo-repeat that shares significant similarity with its catalytic domain. It is located in the nucleus and cytoplasm, and is involved in regulation of melanogenesis. It transcriptionally down-regulates thioredoxin-interacting protein (TXNIP), leading to altered reactive oxygen species (ROS) signaling in human gastric cancer cells. Known interaction partners of HDAC6 are alpha tubulin (substrate) and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD) while interaction partners of HDAC10 are Pax3, KAP1, hsc70 and HDAC3 proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Histone deacetylase 6, domain 2. Histone deacetylase 6 is a class IIb Zn-dependent enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. Known interaction partners of HDAC6 are alpha tubulin and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Class II Histone deacetylase  Clr3 and similar proteins. Clr3 is a class II Histone deacetylase Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Clr3 is the homolog of the class-II HDAC HdaI in S. cerevisiae, and is essential for silencing in heterochromatin regions, such as centromeric regions, ribosomal DNA, the mating-type region and telomeric loci. Clr3 has also been implicated in the regulation of stress-related genes; the histone acetyltransferase, Gcn5, in S. cerevisiae, preferentially acetylates global histone H3K14 while Clr3 preferentially deacetylates H3K14ac, and therefore, interplay between Gcn5 and Clr3 is crucial for the regulation of many stress-response genes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Class I histone deacetylase AcuC (Acetoin utilization protein)-like enzymes. AcuC (Acetoin utilization protein) is a class I deacetylase found only in bacteria and is involved in post-translational control of the acetyl-coenzyme A synthetase (AcsA). Deacetylase AcuC works in coordination with deacetylase SrtN (class III), possibly to maintain AcsA in active (deacetylated) form and let the cell grow under low concentration of acetate. B. subtilis AcuC is a member of operon acuABC; this operon is repressed by the presence of glucose and does not show induction by acetoin; acetoin is a bacterial fermentation product that can be converted to acetate via the butanediol cycle in absence of other carbon sources. Inactivation of AcuC leads to slower growth and lower cell yield under low-acetate conditions in Bacillus subtilis. In general, Class I histone deacetylases (HDACs) are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>Histone deacetylase 10. Histone deacetylases 10 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC10 has an N-terminal deacetylase domain and a C-terminal pseudo-repeat that shares significant similarity with its catalytic domain. It is located in the nucleus and cytoplasm, and is involved in regulation of melanogenesis. It transcriptionally down-regulates thioredoxin-interacting protein (TXNIP), leading to altered reactive oxygen species (ROS) signaling in human gastric cancer cells. Known interaction partners of HDAC10 are Pax3, KAP1, hsc70 and HDAC3 proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Histone deacetylases, class IIa. Class IIa histone deacetylases are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues of histones (EC 3.5.1.98) to yield deacetylated histones. This subclass includes animal HDAC4, HDAC5, HDAC7, and HDCA9. Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, they have N-terminal regulatory domain with two or three conserved serine residues, phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC9 is involved in regulation of gene expression and dendritic growth in developing cortical neurons. It also plays a role in hematopoiesis. HDAC7 is involved in regulation of myocyte migration and differentiation. HDAC5 is involved in integration of chronic drug (cocaine) addiction and depression with changes in chromatin structure and gene expression. HDAC4 participates in regulation of chondrocyte hypertrophy and skeletogenesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Histone deacetylase 6, domain 1. Histone deacetylases 6 are class IIb Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDACs usually act via association with DNA binding proteins to target specific chromatin regions. HDAC6 is the only histone deacetylase with internal duplication of two catalytic domains which appear to function independently of each other, and also has a C-terminal ubiquitin-binding domain. It is located in the cytoplasm and associates with microtubule motor complex, functioning as the tubulin deacetylase and regulating microtubule-dependent cell motility. Known interaction partners of HDAC6 are alpha tubulin (substrate) and ubiquitin-like modifier FAT10 (also known as Ubiquitin D or UBD).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Histone deacetylase (HDAC) classes I, II, IV and related proteins. The HDAC/HDAC-like family includes Zn-dependent histone deacetylase classes I, II and IV (class III HDACs, also called sirtuins, are NAD-dependent and structurally unrelated, and therefore not part of this family). Histone deacetylases catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98), as opposed to the acetylation reaction by some histone acetyltransferases (EC 2.3.1.48). Deacetylases of this family are involved in signal transduction through histone and other protein modification, and can repress/activate transcription of a number of different genes. They usually act via the formation of large multiprotein complexes. They are involved in various cellular processes, including cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and post-translational control of the acetyl coenzyme A synthetase. In mammals, they are known to be involved in progression of different tumors. Specific inhibitors of mammalian histone deacetylases are an emerging class of promising novel anticancer drugs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=co>58</td>
            <td id=al>Histone deacetylase 4. Histone deacetylase 4 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC4 participates in regulation of chondrocyte hypertrophy and skeletogenesis. However, biological substrates for HDAC4 have not been identified; only low lysine deacetylation activity has been demonstrated and active site mutant has enhanced activity toward acetylated lysines. HDAC4 does not bind DNA directly, but through transcription factors MEF2C (myocyte enhancer factor-2C) and MEF2D. Other known interaction partners of the protein are 14-3-3 proteins, SMRT and N-CoR co-repressors, BCL6, HP1, SUMO-1 ubiquitin-like protein, and ANKRA2. It appears to interact in a multiprotein complex with RbAp48 and HDAC3. Furthermore, HDAC4 is required for TGFbeta1-induced myofibroblastic differentiation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Histone deacetylase 7. Histone deacetylase 7 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC7 is involved in regulation of myocyte migration and differentiation. Known interaction partners of class IIa HDAC7 are myocyte enhancer factors - MEF2A, -2C, and -2D, 14-3-3 proteins, SMRT and N-CoR co-repressors, HDAC3, ETA (endothelin receptor). This enzyme is also involved in the development of the immune system as well as brain and heart development. Multiple alternatively spliced transcript variants encoding several isoforms have been found for this gene.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Histone deacetylase class IV also known as histone deacetylase 11. Class IV histone deacetylases (HDAC11; EC 3.5.1.98) are predicted Zn-dependent enzymes. This class includes animal HDAC11, plant HDA2 and related bacterial deacetylases. Enzymes in this subfamily participate in regulation of a number of different processes through protein modification (deacetylation). They catalyze hydrolysis of N(6)-acetyl-lysine of histones (or other proteins) to yield a deacetylated proteins. Histone deacetylases often act as members of large multi-protein complexes such as mSin3A or SMRT/N-CoR. Human HDAC11 does not associate with them but can interact with HDAC6 in vivo. It has been suggested that HDAC11 and HDAC6 may use non-histone proteins as their substrates and play a role other than to directly modulate chromatin structure. In normal tissues, expression of HDAC11 is limited to kidney, heart, brain, skeletal muscle and testis, suggesting that its function might be tissue-specific. In mammals, HDAC11 proteins are known to be involved in progression of various tumors. HDAC11 plays an essential role in regulating OX40 ligand (OX40L) expression in Hodgkin lymphoma (HL); selective inhibition of HDAC11 expression significantly up-regulates OX40L and induces apoptosis in HL cell lines. Thus, inhibition of HDAC11 could be a therapeutic drug option for antitumor immune response in HL patients.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>Histone deacetylase 5. Histone deacetylase 5 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, having N-terminal regulatory domain with two or three conserved serine residues; phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC5 is involved in integration of chronic drug (cocaine) addiction and depression with changes in chromatin structure and gene expression; cocaine regulates HDAC5 function to antagonize the rewarding impact of cocaine, possibly by blocking drug-stimulated gene expression that supports drug-induced behavioral change. It is also involved in regulation of angiogenesis and cell cycle as well as immune system development. HDAC5 and HDAC9 have been found to be significantly up-regulated in high-risk medulloblastoma compared with low-risk and may potentially be novel drug targets.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Histone deacetylase 9. Histone deacetylase 9 is a class IIa Zn-dependent enzyme that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Class IIa histone deacetylases are signal-dependent co-repressors, they have N-terminal regulatory domain with two or three conserved serine residues, phosphorylation of these residues is important for ability to shuttle between the nucleus and cytoplasm and act as transcriptional co-repressors. HDAC9 is involved in regulation of gene expression and dendritic growth in developing cortical neurons. It also plays a role in hematopoiesis. Its deregulated expression may be associated with some human cancers. HDAC5 and HDAC9 have been found to be significantly up-regulated in high-risk medulloblastoma compared with low-risk and may potentially be novel drug targets.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>Class II histone deacetylases Hos3 and related proteins. Fungal histone deacetylase Hos3 from Saccharomyces cerevisiae is a Zn-dependent enzyme belonging to HDAC class II. It catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. Hos3 deacetylase is homodimer, in vitro it shows specificity to H4, H3 and H2A.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>Histone deacetylase 8 (HDAC8). HDAC8 is a Zn-dependent class I histone deacetylase that catalyzes hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. Histone deacetylases usually act via association with DNA binding proteins to target specific chromatin regions. HDAC8 is found in human cytoskeleton-bound protein fraction and insoluble cell pellets. It plays a crucial role in intramembraneous bone formation; germline deletion of HDAC8 is detrimental to skull bone formation. HDAC8 is possibly associated with the smooth muscle actin cytockeleton and may regulate the contractive capacity of smooth muscle cells. HDAC8 is also involved in the metabolic control of the estrogen receptor related receptor (ERR)-alpha/peroxisome proliferator activated receptor (PPAR) gamma coactivator 1 alpha (PGC1-alpha) transcriptional complex as well as in the development of neuroblastoma and T-cell lymphoma. HDAC8-selective small-molecule inhibitors could be a therapeutic drug option for these diseases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>67</td>
            <td id=al>Class I histone deacetylases Hos1 and related proteins. Saccharomyces cerevisiae Hos1 is responsible for Smc3 deacetylation. Smc3 is an important player during the establishment of sister chromatid cohesion. Hos1 belongs to the class I histone deacetylases (HDACs). HDACs are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. Other class I HDACs are animal HDAC1, HDAC2, HDAC3, HDAC8, fungal RPD3 and HOS2, plant HDA9, protist, archaeal and bacterial (AcuC) deacetylases. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Class I histone deacetylases. Class I histone deacetylases (HDACs) are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98). Enzymes belonging to this group participate in regulation of a number of processes through protein (mostly different histones) modification (deacetylation). Class I histone deacetylases in general act via the formation of large multiprotein complexes. This group includes animal HDAC1, HDAC2, HDAC3, HDAC8, fungal RPD3, HOS1 and HOS2, plant HDA9, protist, archaeal and bacterial (AcuC) deacetylases. Members of this class are involved in cell cycle regulation, DNA damage response, embryonic development, cytokine signaling important for immune response and in posttranslational control of the acetyl coenzyme A synthetase. In mammals, they are known to be involved in progression of various tumors. Specific inhibitors of mammalian histone deacetylases are an emerging class of promising novel anticancer drugs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Class I histone deacetylases including ScHos2 and SpPhd1. This subfamily includes Class I histone deacetylase (HDAC) Hos2 from Saccharomyces cerevisiae as well as a histone deacetylase Phd1 from Schizosaccharomyces pombe. Hos2 binds to the coding regions of genes during gene activation, specifically it deacetylates the lysines in H3 and H4 histone tails. It is preferentially associated with genes of high activity genome-wide and is shown to be necessary for efficient transcription. Thus, Hos2 is directly required for gene activation in contrast to other class I histone deacetylases. Protein encoded by phd1 is inhibited by trichostatin A (TSA), a specific inhibitor of histone deacetylase, and is involved in the meiotic cell cycle in S. pombe. Class 1 HDACs are Zn-dependent enzymes that catalyze hydrolysis of N(6)-acetyl-lysine residues in histone amino termini to yield a deacetylated histone (EC 3.5.1.98).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>Histone deacetylase 1 (HDAC1). Histone deacetylase 1 (HDAC1) is a Zn-dependent class I enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDAC1 is involved in regulation through association with DNA binding proteins to target specific chromatin regions. In particular, HDAC1 appears to play a major role in pre-implantation embryogenesis in establishing a repressive chromatin state. Its interaction with retinoblastoma tumor-suppressor protein is essential in the control of cell proliferation and differentiation. Together with metastasis-associated protein-2 (MTA2), it deacetylates p53, thereby modulating its effect on cell growth and apoptosis. It participates in DNA-damage response, along with HDAC2; together, they promote DNA non-homologous end-joining. HDAC1 is also involved in tumorogenesis; its overexpression modulates cancer progression. Specific inhibitors of HDAC1 are currently used in cancer therapy.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>Histone deacetylase 3 (HDAC3). HDAC3 is a Zn-dependent class I histone deacetylase that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. In order to target specific chromatin regions, HDAC3 can interact with DNA-binding proteins (transcriptional factors) either directly or after forming complexes with a number of other proteins, as observed for the SMPT/N-CoR complex which recruits human HDAC3 to specific chromatin loci and activates deacetylation. Human HDAC3 is also involved in deacetylation of non-histone substrates such as RelA, SPY and p53 factors. This protein can also down-regulate p53 function and subsequently modulate cell growth and apoptosis. This gene is therefore regarded as a potential tumor suppressor gene. HDAC3 plays a role in various physiological processes, including subcellular protein localization, cell cycle progression, cell differentiation, apoptosis and survival. HDAC3 has been found to be overexpressed in some tumors including leukemia, lung carcinoma, colon cancer and maxillary carcinoma. Thus, inhibitors precisely targeting HDAC3 (in some cases together with retinoic acid or hyperthermia) could be a therapeutic drug option.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>Histone deacetylase 2 (HDAC2). Histone deacetylase 2 (HDAC2) is a Zn-dependent class I enzyme that catalyzes hydrolysis of N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). Histone acetylation/deacetylation process is important for mediation of transcriptional regulation of many genes. HDAC2 is involved in regulation through association with DNA binding proteins to target specific chromatin regions. It forms transcriptional repressor complexes by associating with several proteins, including the mammalian zinc-finger transcription factor YY1, thus playing an important role in transcriptional regulation, cell cycle progression and developmental events. Additionally, a few non-histone HDAC2 substrates have been found. HDAC2 plays a role in embryonic development and cytokine signaling important for immune response, and is over-expressed in several solid tumors including oral, prostate, ovarian, endometrial and gastric cancer. It participates in DNA-damage response, along with HDAC1; together, they can promote DNA non-homologous end-joining. HDAC2 is considered an important cancer prognostic marker. Inhibitors specifically targeting HDAC2 could be a therapeutic drug option.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>reduced potassium dependency-3 (RPD3)-like. Proteins of the Rpd3-like family are class I Zn-dependent Histone deacetylases that catalyze hydrolysis of an N(6)-acetyl-lysine residue of a histone to yield a deacetylated histone (EC 3.5.1.98). RPD3 is the yeast homolog of class I HDACs. The main function of RPD3-like group members is regulation of a number of different processes through protein (mostly different histones) modification (deacetylation). This group includes fungal RPD3 and acts via the formation of large multiprotein complexes. Members of this group are involved in cell cycle regulation, DNA damage response, embryonic development and cytokine signaling important for immune response. Histone deacetylation by yeast RPD3 represses genes regulated by the Ash1 and Ume6 DNA-binding proteins. In mammals, they are known to be involved in progression of various tumors. Specific inhibitors of mammalian histone deacetylases could be a therapeutic drug option.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>40</td>
            <td id=al>histone deacetylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>histone deacetylase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1020133</td></tr>
<tr><td id=al>end</td><td id=al>1021054</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>histone deacetylase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
