<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04197</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04197 tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE [Phormidium_A]</h3>
    <div>MSEAVLRRDTIVAIATAVVPQQGSVGIVRLSGDEAIAIAKRLFHAPGKQPWESHRILYGTIRHPGTSALVDEALLLLMVAPRSYTREDVVEFHCHGGMMVVQEVLQLCVEAGARLANPGEFTLRAFLNGRIDLTQAESIADLVGAQSQAASQAALAGVQGRLAQPIRDLRSQCLDILAEIEARVDFEEDLPPLDEGKVVQDLAAVLREIEAIEKTASQGELLRTGLKVVIVGRPNVGKSSLLNAWSRSDRAIVTDLPGTTRDVVESQLVVGGIPVQVLDTAGIRQTSDRIEQMGVERSRQALEAADLVLLVLDALSGWTAEDEAIYQRVCDRPLITILNKCDLVEVPPQLPHVQQPVLTAAAQQQGLEALEAAILEKANAQEIETANVDFCINQRQAAALTRAKLALEQVQETITAQLPLDFWTIDLREGIQALGEITGEEMTESVLDRIFSRFCIGK</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA modification GTPase trmE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>TPA: tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td>
            <td id=cb>DVED01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA U34 5-carboxymethylaminomethyl modifying GTPase MnmE/TrmE [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA modification GTPase TrmE. TrmE, also called MnmE and previously designated ThdF (thiophene and furan oxidation protein), is a GTPase involved in tRNA modification to create 5-methylaminomethyl-2-thiouridine in the wobble position of some tRNAs. This protein and GidA form an alpha2/beta2 heterotetramer. [Protein synthesis, tRNA and rRNA base modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MnmE helical domain. The tRNA modification GTPase MnmE consists of three domains. An N-terminal domain, a helical domain and a GTPase domain which is nested within the helical domain. This family represents the helical domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>trmE is a tRNA modification GTPase. TrmE (MnmE, ThdF, MSS1) is a 3-domain protein found in bacteria and eukaryotes. It controls modification of the uridine at the wobble position (U34) of tRNAs that read codons ending with A or G in the mixed codon family boxes. TrmE contains a GTPase domain that forms a canonical Ras-like fold. It functions a molecular switch GTPase, and apparently uses a conformational change associated with GTP hydrolysis to promote the tRNA modification reaction, in which the conserved cysteine in the C-terminal domain is thought to function as a catalytic residue. In bacteria that are able to survive in extremely low pH conditions, TrmE regulates glutamate-dependent acid resistance.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein TrmE N-terminus. This family represents the shorter, B, chain of the homo-dimeric structure which is a guanine nucleotide-binding protein that binds and hydrolyzes GTP. TrmE is homologous to the tetrahydrofolate-binding domain of N,N-dimethylglycine oxidase and indeed binds formyl-tetrahydrofolate. TrmE actively participates in the formylation reaction of uridine and regulates the ensuing hydrogenation reaction of a Schiff's base intermediate. This B chain is the N-terminal portion of the protein consisting of five beta-strands and three alpha helices and is necessary for mediating dimer formation within the protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-terminal domain of TrmE, a tRNA modification GTPase. This family contains the N-terminal domain of TrmE (also known as MnmE, ThdF, MSS1), a guanine nucleotide-binding protein conserved in all three kingdoms of life. It is involved in the modification of uridine bases (U34) at the first anticodon (wobble) position of tRNAs decoding two-family box triplets. TrmE is a three-domain protein comprising an N-terminal alpha/beta domain, a helical domain, and the GTPase domain which is nested within the helical domain. The N-terminal domain induces dimerization for self-assembly and is topologically homologous to the tetrahydrofolate (THF)-binding domain of N,N-dimethylglycine oxidase (DMGO). However, the THF-binding site in DMGO is encoded on a single polypeptide, while homodimerization would be required to create a similar THF-binding site in TrmE. Dimerization also creates a second, symmetry-related THF-binding site. Biochemical and structural studies show that TrmE indeed binds formyl-THF. A cysteine residue, necessary for modification of U34, is located close to the C1-group donor 5-formyl-tetrahydrofolate, suggesting a direct role of TrmE in the modification analogous to DNA modification enzymes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>EngA2 GTPase contains the second domain of EngA. This EngA2 subfamily CD represents the second GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=co>49</td>
            <td id=al>ribosome-associated GTPase EngA. EngA (YfgK, Der) is a ribosome-associated essential GTPase with a duplication of its GTP-binding domain. It is broadly to universally distributed among bacteria. It appears to function in ribosome biogenesis or stability. [Protein synthesis, Other]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=co>40</td>
            <td id=al>ribosome-associated GTPase EngA. EngA (YfgK, Der) is a ribosome-associated essential GTPase with a duplication of its GTP-binding domain. It is broadly to universally distributed among bacteria. It appears to function in ribosome biogenesis or stability. [Protein synthesis, Other]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>50S ribosome-binding GTPase. The full-length GTPase protein is required for the complete activity of the protein of interacting with the 50S ribosome and binding of both adenine and guanine nucleotides, with a preference for guanine nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=co>43</td>
            <td id=al>Predicted GTPases [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>41</td>
            <td id=al>Predicted GTPases [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>E. coli Ras-like protein (Era)-like GTPase. The Era (E. coli Ras-like protein)-like family includes several distinct subfamilies (TrmE/ThdF, FeoB, YihA (EngB), Era, and EngA/YfgK) that generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. TrmE is ubiquitous in bacteria and is a widespread mitochondrial protein in eukaryotes, but is absent from archaea. The yeast member of TrmE family, MSS1, is involved in mitochondrial translation; bacterial members are often present in translation-related operons. FeoB represents an unusual adaptation of GTPases for high-affinity iron (II) transport. YihA (EngB) family of GTPases is typified by the E. coli YihA, which is an essential protein involved in cell division control. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain. EngA and its orthologs are composed of two GTPase domains and, since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>EngA1 GTPase contains the first domain of EngA. This EngA1 subfamily CD represents the first GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>24</td>
            <td id=al>GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>22</td>
            <td id=al>bifunctional cytidylate kinase/GTPase Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>25</td>
            <td id=al>bifunctional cytidylate kinase/GTPase Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>small GTP-binding protein domain. Proteins with a small GTP-binding domain recognized by this model include Ras, RhoA, Rab11, translation elongation factor G, translation initiation factor IF-2, tetratcycline resistance protein TetM, CDC42, Era, ADP-ribosylation factors, tdhF, and many others. In some proteins the domain occurs more than once.This model recognizes a large number of small GTP-binding proteins and related domains in larger proteins. Note that the alpha chains of heterotrimeric G proteins are larger proteins in which the NKXD motif is separated from the GxxxxGK[ST] motif (P-loop) by a long insert and are not easily detected by this model. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>GTPase Era, involved in 16S rRNA processing [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Rat sarcoma (Ras)-like superfamily of small guanosine triphosphatases (GTPases). Ras-like GTPase superfamily. The Ras-like superfamily of small GTPases consists of several families with an extremely high degree of structural and functional similarity. The Ras superfamily is divided into at least four families in eukaryotes: the Ras, Rho, Rab, and Sar1/Arf families. This superfamily also includes proteins like the GTP translation factors, Era-like GTPases, and G-alpha chain of the heterotrimeric G proteins. Members of the Ras superfamily regulate a wide variety of cellular functions: the Ras family regulates gene expression, the Rho family regulates cytoskeletal reorganization and gene expression, the Rab and Sar1/Arf families regulate vesicle trafficking, and the Ran family regulates nucleocytoplasmic transport and microtubule organization. The GTP translation factor family regulates initiation, elongation, termination, and release in translation, and the Era-like GTPase family regulates cell division, sporulation, and DNA replication. Members of the Ras superfamily are identified by the GTP binding site, which is made up of five characteristic sequence motifs, and the switch I and switch II regions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>58</td>
            <td id=al>GTPase Era; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>E. coli Ras-like protein (Era) is a multifunctional GTPase. Era (E. coli Ras-like protein) is a multifunctional GTPase found in all bacteria except some eubacteria. It binds to the 16S ribosomal RNA (rRNA) of the 30S subunit and appears to play a role in the assembly of the 30S subunit, possibly by chaperoning the 16S rRNA. It also contacts several assembly elements of the 30S subunit. Era couples cell growth with cytokinesis and plays a role in cell division and energy metabolism. Homologs have also been found in eukaryotes. Era contains two domains: the N-terminal GTPase domain and a C-terminal domain KH domain that is critical for RNA binding. Both domains are important for Era function. Era is functionally able to compensate for deletion of RbfA, a cold-shock adaptation protein that is required for efficient processing of the 16S rRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>Circularly permuted YlqF GTPase. Proteins of the YlqF family contain all sequence motifs typical of the vast class of P-loop-containing GTPases, but show a circular permutation, with a G4-G1-G3 pattern of motifs as opposed to the regular G1-G3-G4 pattern seen in most GTPases. The YlqF subfamily is represented in all eukaryotes as well as a phylogenetically diverse array of bacteria (including gram-positive bacteria, proteobacteria, Synechocystis, Borrelia, and Thermotoga).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>Circularly permuted YqeH GTPase. YqeH is an essential GTP-binding protein. Depletion of YqeH induces an excess initiation of DNA replication, suggesting that it negatively controls initiation of chromosome replication. The YqeH subfamily is common in eukaryotes and sporadically present in bacteria with probable acquisition by plants from chloroplasts. Proteins of the YqeH family contain all sequence motifs typical of the vast class of P-loop-containing GTPases, but show a circular permutation, with a G4-G1-G3 pattern of motifs as opposed to the regular G1-G3-G4 pattern seen in most GTPases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>GTP-binding protein, GTP1/Obg family  [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>Ribosome biogenesis GTPase A [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>Nucleolar GTP-binding protein (NOG). NOG1 is a nucleolar GTP-binding protein present in eukaryotes ranging from trypanosomes to humans. NOG1 is functionally linked to ribosome biogenesis and found in association with the nuclear pore complexes and identified in many preribosomal complexes. Thus, defects in NOG1 can lead to defects in 60S biogenesis. The S. cerevisiae NOG1 gene is essential for cell viability, and mutations in the predicted G motifs abrogate function. It is a member of the ODN family of GTP-binding proteins that also includes the bacterial Obg and DRG proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>GTP-binding protein Era. Era is an essential GTPase in Escherichia coli and many other bacteria. It plays a role in ribosome biogenesis. Few bacteria lack this protein. [Protein synthesis, Other]</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3427679</td></tr>
<tr><td id=al>end</td><td id=al>3429056</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
