<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05384</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05384 bifunctional acetaldehyde-CoA/alcohol dehydrogenase [Phormidium_A]</h3>
    <div>MTEFVTNLEQLEAHIARVKAAQQAYATFSQEQVDVIFKKAALAANAARIPLAKLAVEETGMGVVEDKVIKNHFASEIIYNKYKSDKTCGIIEEDKSFGIQKIAEPVGILAGIVPTTNPTSTAVFKALITLKTRNAIIFSPHPRAKRCTIEAAKIVRDAAVAAGAPEHIIGWIDEPTVPLSQALMQHPDIKLILATGGPGMVKAAYSSGNPSLGVGAGNTPALIDSSAHIKMAVSSILLSKTFDNGMICASEQSVVVLDDVYEEVRQEFIDRGAYLLAPDEGDRLGKLLINNGRLNANIVGQSVESIAALADISIPDNTRVLISEVSDISTDEPFAYEKLSPVLAMYRAQDFNEAVDKAEALVEFGGRGHTAVLYINPSEFDHIKQFEDKMQTARVLINTPSSQGAIGDLYNFRLDPSLTLGCGTWGGNSISENVEPHHLLNVKTVAERRENMLWFRIPPKVYFKYGSLPVALRELAGKQRAMIITDKPLYDLGMTASVETVLEDIGLKYTVFSEVEPDPSLETVNRGLAVMNSFQPDVIIALGGGSPMDAAKIMWLMYEHPEIEFEGLAMRFMDIRKRVYDLPELGEKAVLVAIPTTSGTGSEVTPFAVVTDRRNDIKYPLADYALTPTMAIVDPELVLNMPKSLTAFGGVDALTHALEAYVSVLASEYTNGLALEAIRLIFKYLPDSYHQGAANPKAREKMHYAATMAGMAFANGFLGICHSIAHQLGATFHIPHGLANAVMISHIIRYNATNAPFKQATFSQYKYPNAKWRYARIADHLQLGGETENEKIVRLIQAVEHLKQEVGIPSAMKDLIQMDEAQFKAQVDQVADRAFDDQCTVSNPRYPLITDLKQLILDAYYGNLDLSGLDGGQTGGLDGVPDVPAEPEVVMQ</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>bifunctional acetaldehyde dehydrogenase / alcohol dehydrogenase AdhE</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acetaldehyde dehydrogenase</td>
            <td id=co>Lyngbya</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aldehyde-alcohol dehydrogenase</td>
            <td id=co>Lyngbya</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=co>CCY15215</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td>
            <td id=co>Coleofasciculus</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Coenzyme A acylating aldehyde dehydrogenase (ACDH), ALDH family 20-like. Coenzyme A acylating aldehyde dehydrogenase (ACDH, EC=1.2.1.10), an NAD+ and CoA-dependent acetaldehyde dehydrogenase, functions as a single enzyme (such as the Ethanolamine utilization protein, EutE, in Salmonella typhimurium) or as part of a multifunctional enzyme to convert acetaldehyde into acetyl-CoA . The E. coli aldehyde-alcohol dehydrogenase includes the functional domains, alcohol dehydrogenase (ADH), ACDH, and pyruvate-formate-lyase deactivase; and the Entamoeba histolytica aldehyde-alcohol dehydrogenase 2 (ALDH20A1) includes the functional domains ADH and ACDH and may be critical enzymes in the fermentative pathway.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>C-terminal alcohol dehydrogenase domain of the acetaldehyde dehydrogenase-alcohol dehydrogenase bifunctional two-domain protein (AAD). This alcohol dehydrogenase domain is located on the C-terminal of a bifunctional two-domain protein. The N-terminal of the protein contains an acetaldehyde-CoA dehydrogenase domain. This protein is involved in pyruvate metabolism whereby pyruvate is converted to acetyl-CoA and formate by pyruvate formate-lysase (PFL). Under anaerobic condition, acetyl-CoA is reduced to acetaldehyde and ethanol by this two-domain protein. Acetyl-CoA is first converted into an enzyme-bound thiohemiacetal by the N-terminal acetaldehyde dehydrogenase domain. The enzyme-bound thiohemiacetal is subsequently reduced by the C-terminal NAD+-dependent alcohol dehydrogenase domain. In E. coli, this protein is called AdhE and has been shown to have pyruvate formate-lyase (PFL) deactivase activity, which leads to the inactivation of PFL, a key enzyme in anaerobic metabolism. In Escherichia coli and Entamoeba histolytica, this enzyme forms homopolymeric peptides composed of more than 20 protomers associated in a helical rod-like structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Coenzyme A acylating aldehyde dehydrogenase (ACDH), Ethanolamine utilization protein EutE, and related proteins. Coenzyme A acylating aldehyde dehydrogenase (ACDH), an NAD+ and CoA-dependent acetaldehyde dehydrogenase, acetylating (EC=1.2.1.10), functions as a single enzyme (such as the Ethanolamine utilization protein, EutE, in Salmonella typhimurium) or as part of a multifunctional enzyme to convert acetaldehyde into acetyl-CoA. The E. coli aldehyde-alcohol dehydrogenase includes the functional domains, alcohol dehydrogenase (ADH), ACDH, and pyruvate-formate-lyase deactivase; and the Entamoeba histolytica aldehyde-alcohol dehydrogenase 2 (ALDH20A1) includes the functional domains ADH and ACDH, and may be critical enzymes in the fermentative pathway.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NADPH-dependent butanol dehydrogenase involved in the butanol and ethanol formation pathway in bacteria. NADPH-dependent butanol dehydrogenase (BDH) is involved in the butanol and ethanol formation pathway of some bacteria. The fermentation process is characterized by an acid producing growth phase, followed by a solvent producing phase. The latter phase is associated with the induction of solventogenic enzymes such as butanol dehydrogenase. The activity of the enzyme requires NADPH as cofactor, as well as divalent ions zinc or iron. This family is a member of the iron-containing alcohol dehydrogenase superfamily. Protein structure has a dehydroquinate synthase-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>acetaldehyde dehydrogenase (acetylating). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Alcohol dehydrogenase, class IV [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>1,3-propanediol dehydrogenase (PPD) catalyzes the reduction of 3-hydroxypropionaldehyde (3-HPA) to 1,3-propanediol in glycerol metabolism. 1,3-propanediol dehydrogenase (PPD) plays a role in glycerol metabolism of some bacteria in anaerobic conditions. In this degradation pathway, glycerol is converted in a two-step process to 1,3-propanediol (1,3-PD) which is then excreted into the extracellular medium. The first reaction involves the transformation of glycerol into 3-hydroxypropionaldehyde (3-HPA) by a coenzyme B-12-dependent dehydratase. The second reaction involves the dismutation of the 3-hydroxypropionaldehyde (3-HPA) to 1,3-propanediol by the NADH-linked 1,3-propanediol dehydrogenase (PPD). The enzyme requires iron ion for its function. Because many genes in this pathway are present in the propanediol utilization (pdu) operon, they are also named pdu genes. PPD is a member of the iron-containing alcohol dehydrogenase superfamily. The PPD structure has a dehydroquinate synthase-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains large metal-containing alcohol dehydrogenases (ADH), known as iron-containing alcohol dehydrogenases. They contain a dehydroquinate synthase-like protein structural fold and mostly contain iron. They are distinct from other alcohol dehydrogenases which contains different protein domains. There are several distinct families of alcohol dehydrogenases: Zinc-containing long-chain alcohol dehydrogenases, insect-type, or short-chain alcohol dehydrogenases, iron-containing alcohol dehydrogenases, among others. The iron-containing family has a Rossmann fold-like topology that resembles the fold of the zinc-dependent alcohol dehydrogenases, but lacks sequence homology, and differs in strand arrangement. ADH catalyzes the reversible oxidation of alcohol to acetaldehyde with the simultaneous reduction of NAD(P)+ to NAD(P)H.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>1,3-Propanediol (1,3-PD) dehydrogenase. This family includes 1,3-propanediol (1,3-PD) dehydrogenase, a key enzyme in the microbial production of 1,3-PD that has been previously characterized as the product of dhaT gene in Klebsiella pneumoniae. 1,3-PD dehydrogenase is a member of the family III metal-dependent polyol dehydrogenases, which are shown to require a divalent metal ion for catalysis. However, some members of this family showed a dependence on Fe(2+) or Zn(2+) for activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Lactadehyde:propanediol oxidoreductase (LPO) catalyzes the interconversion between L-lactaldehyde and L-1,2-propanediol in Escherichia coli and other enterobacteria. Lactadehyde:propanediol oxidoreductase (LPO) is a member of the group III iron-activated dehydrogenases which catalyze the interconversion between L-lactaldehyde and L-1,2-propanediol in Escherichia coli and other enterobacteria. L-fucose and L-rhamnose are used by Escherichia coli through an inducible pathway mediated by the fucose regulon comprising four linked operons fucO, fucA, fucPIK, and fucR. The fucA-encoded aldolase catalyzes the formation of dihydroxyacetone phosphate and L-lactaldehyde. Under anaerobic conditions, with NADH as a cofactor, lactaldehyde is converted by a fucO-encoded lactadehyde:propanediol oxidoreductase (LPO) to L-1,2-propanediol, which is excreted as a fermentation product. In mutant strains, E. coli adapted to grow on L-1,2-propanediol, FucO catalyzes the oxidation of the polyol to L-lactaldehyde. FucO is induced regardless of the respiratory conditions of the culture, remains fully active in the absence of oxygen. In the presence of oxygen, this enzyme becomes oxidatively inactivated by a metal-catalyzed oxidation mechanism. FucO is an iron-dependent metalloenzyme that is inactivated by other metals, such as zinc, copper, or cadmium. This enzyme can also reduce glycol aldehyde with similar efficiency.  Beside L-1,2-propanediol, the enzyme is also able to oxidize methanol as an alternative substrate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and belongs to the alcohol dehydrogenase-like superfamily. It is distinct from other alcohol dehydrogenases which contain different protein domain. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) most of which have not been characterized. Their specific function is unknown. The protein structure represents a dehydroquinate synthase-like fold and belongs to the alcohol dehydrogenase-like superfamily. It is distinct from other alcohol dehydrogenases which contain different protein domains. Alcohol dehydrogenase catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NAD(P)+-dependent aldehyde dehydrogenase-like (ALDH-like) family. The aldehyde dehydrogenase-like (ALDH-like) group of the ALDH superfamily of NAD(P)+-dependent enzymes which, in general, oxidize a wide range of  endogenous and exogenous aliphatic and aromatic aldehydes to their corresponding carboxylic acids and play an  important role in detoxification. This group includes families ALDH18, ALDH19, and ALDH20 and represents such proteins as gamma-glutamyl phosphate reductase, LuxC-like acyl-CoA reductase, and coenzyme A acylating aldehyde dehydrogenase.  All of these proteins have a conserved cysteine that aligns with the catalytic cysteine of the ALDH group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains proteins similar to iron-containing alcohol dehydrogenase (Fe-ADH), most of which have not been characterized. Their specific function is unknown. The protein structure represents a dehydroquinate synthase-like fold and belongs to the alcohol dehydrogenase-like superfamily. It is distinct from other alcohol dehydrogenases which contain different protein domains.  Alcohol dehydrogenase catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains iron-containing alcohol dehydrogenase-like (ADH) proteins. Alcohol dehydrogenase catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase fold and is a member of the iron-containing alcohol dehydrogenase-like family. They are distinct from other alcohol dehydrogenases which contain different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contain different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-1,2-propanediol oxidoreductase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>aldehyde dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>5-hydroxyvalerate dehydrogenase (HVD) catalyzes the oxidation of 5-hydroxyvalerate to 5-oxovalerate with NAD+ as cofactor. 5-hydroxyvalerate dehydrogenase (HVD) is an iron-containing (type III) NAD-dependent alcohol dehydrogenase. It plays a role in the cyclopentanol metabolism biochemical pathway. It catalyzes the oxidation of 5-hydroxyvalerate to 5-oxovalerate with NAD+ as cofactor. This cyclopentanol (cpn) degradation pathway is present in some bacteria which can use cyclopentanol as sole carbon source. In Comamonas sp. strain NCIMB 9872, this enzyme is encoded by the CpnD gene.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Ethanolamine utilization protein EutE-like. Coenzyme A acylating aldehyde dehydrogenase (ACDH), an NAD+ and CoA-dependent acetaldehyde dehydrogenase, acetylating (EC=1.2.1.10), converts acetaldehyde into acetyl-CoA.  This CD is limited to such monofunctional enzymes as the Ethanolamine utilization protein, EutE, in Salmonella typhimurium.  Mutations in eutE abolish the ability to utilize ethanolamine as a carbon source.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Iron-containing alcohol dehydrogenases-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contain different protein domain. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Hydroxyacid-oxoacid transhydrogenase (HOT) involved in gamma-hydroxybutyrate metabolism. This family contains hydroxyacid-oxoacid transhydrogenase (HOT), also known as D-2-hydroxyglutarate transhydrogenase. It catalyzes the conversion of gamma-hydroxybutyrate (GHB) to succinic semialdehyde (SSA), coupled to the stoichiometric conversion of alpha-ketoglutarate to D-2-hydroxyglutarate in gamma-Hydroxybutyrate catabolism. Unlike many other alcohols, which are oxidized by NAD-linked dehydrogenases, gamma-hydroxybutyrate is metabolized to succinate semialdehyde by hydroxyacid-oxoacid transhydrogenase which does not require free NAD or NADP; instead, it uses alpha-ketoglutarate as an acceptor, converting it to d-2-hydroxyglutarate. Alpha-ketoglutarate serves as an intermediate acceptor to regenerate NAD(P) required for the oxidation of GHB. HOT also catalyzes the reversible oxidation of a hydroxyacid obligatorily coupled to the reduction of an oxoacid, and requires no cofactor. In mammals, the HOT enzyme is located in mitochondria, and is expressed with an N-terminal mitochondrial targeting sequence. HOT enzyme is member of the metal-containing alcohol dehydrogenase family. It typically contains an iron although other metal ions may be used.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Hydroxyethylphosphoate dehydrogenase (HEPD) catalyzes the reduction of phosphonoacetaldehyde (PnAA) to hydroxyethylphosphoate (HEP). Hydroxyethylphosphoate dehydrogenase (HEPD) catalyzes the reduction of phosphonoacetaldehyde (PnAA) to hydroxyethylphosphoate (HEP) with either NADH or NADPH as a cofactor, although NADH is the preferred cofactor. PnAA is a biosynthetic intermediate for several phosphonates such as the antibiotic fosfomycin, phosphinothricin tripeptide (PTT), and 2-aminoethylphosphonate (AEP). This enzyme is named PhpC in PTT biosynthesis pathway in Streptomyces hygroscopicus and S. viridochromogenes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>putative alcohol dehydrogenase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>1,3-propanediol dehydrogenase-like (PPD). This family contains proteins similar to 1,3-propanediol dehydrogenase (PPD) which is a member of the iron-containing alcohol dehydrogenase superfamily, and exhibits a dehydroquinate synthase-like fold.  Protein sequence similarity search and other biochemical evidences suggest that they are close to the iron-containing 1,3-propanediol dehydrogenase (EC 1.1.1.202). 1,3-propanediol dehydrogenase catalyzes the oxidation of propane-1,3-diol to 3-hydroxypropanal with the simultaneous reduction of NADP+ to NADPH. The protein structure of Thermotoga maritima TM0920 gene contains one NADP+ and one iron ion.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Acyl-CoA reductase or other NAD-dependent aldehyde dehydrogenase [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ethanolamine utilization ethanol dehydrogenase EutG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Maleylacetate reductase is involved in many aromatic compounds degradation pathways of aerobic microbes. Maleylacetate reductase (MAR) plays an important role in the degradation of aromatic compounds  in aerobic microbes. In fungi and yeasts, the enzyme is involved in the catabolism of compounds such as phenol, tyrosine, benzoate, 4-hydroxybenzoate and resorcinol. In bacteria, the enzyme contributes to the degradation of resorcinol, 2,4-dihydroxybenzoate ([beta]-resorcylate) and 2,6-dihydroxybenzoate ([gamma]-resorcylate) via hydroxyquinol and maleylacetate. Maleylacetate reductase (MAR) catalyzes NADH- or NADPH-dependent reduction, at the carbon-carbon double bond, of maleylacetate or 2-chloromaleylacetate to 3-oxoadipate. In the case of 2-chloromaleylacetate, MAR initially catalyzes the NAD(P)H-dependent dechlorination to maleylacetate, which is then reduced to 3-oxoadipate. This enzyme is a homodimer. It is inhibited by thiol-blocking reagents such as p-chloromercuribenzoate and Hg++, indicating that the cysteine residue is probably necessary for the catalytic activity of maleylacetate reductase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Butanol dehydrogenase catalyzes the conversion of butyraldehyde to butanol with the cofactor NAD(P)H being oxidized in the process. The butanol dehydrogenase (BDH) is involved in the final step of the butanol formation pathway in anaerobic micro-organism. Butanol dehydrogenase catalyzes the conversion of butyraldehyde to butanol with the cofactor NAD(P)H being oxidized in the process. Activity in the reverse direction is 50-fold lower than that in the forward direction. The NADH-BDH has higher activity with longer chained aldehydes and is inhibited by metabolites containing an adenine moiety. This protein family belongs to the so-called iron-containing alcohol dehydrogenase superfamily. Since members of this superfamily use different divalent ions, preferentially iron or zinc, it has been suggested to be renamed to family III metal-dependent polyol dehydrogenases. This family also includes E. coli YqhD enzyme, an NADP-dependent dehydrogenase whose activity measurements with several alcohols demonstrate preference for alcohols longer than C3. The active site of YqhD contains a Zn metal, and a modified NADPH cofactor bearing OH groups on the saturated C5 and C6 atoms, possibly due to oxygen stress on the enzyme, which would functionally work under anaerobic conditions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>Iron-containing alcohol dehydrogenase. This family contains iron-containing alcohol dehydrogenase (ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. The ADH of hyperthermophilic archaeon Thermococcus hydrothermalis oxidizes a series of primary aliphatic and aromatic alcohols, preferentially from C2 to C8, but is also active towards methanol and glycerol, and is stereospecific for monoterpenes. It has been suggested that the type III ADHs in microorganisms are involved in acetaldehyde detoxication rather than in alcohol turnover.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>iron-containing alcohol dehydrogenases (Fe-ADH)-like. This family contains iron-containing alcohol dehydrogenase (Fe-ADH) which catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron ions. The protein structure represents a dehydroquinate synthase-like fold and is a member of the iron-activated alcohol dehydrogenase-like family. It is distinct from other alcohol dehydrogenases which contains different protein domains. Proteins of this family have not been characterized.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NAD(P)+-dependent aldehyde dehydrogenase superfamily. The aldehyde dehydrogenase superfamily (ALDH-SF) of  NAD(P)+-dependent enzymes, in general, oxidize a wide range of  endogenous and exogenous aliphatic and aromatic aldehydes to their corresponding carboxylic acids and play an  important role in detoxification. Besides aldehyde detoxification, many ALDH isozymes possess multiple additional catalytic and non-catalytic functions such as participating in metabolic pathways, or as binding proteins, or osmoregulants, to mention a few. The enzyme has three domains, a NAD(P)+ cofactor-binding domain, a catalytic domain, and a bridging domain; and the active enzyme is generally either homodimeric or homotetrameric. The catalytic mechanism is proposed to involve cofactor binding, resulting in a conformational change and activation of an invariant catalytic cysteine nucleophile. The cysteine and aldehyde substrate form an oxyanion thiohemiacetal intermediate resulting in hydride transfer to the cofactor and formation of a thioacylenzyme intermediate. Hydrolysis of the thioacylenzyme and release of the carboxylic acid product occurs, and in most cases, the reduced cofactor dissociates from the enzyme. The evolutionary phylogenetic tree of ALDHs appears to have an initial bifurcation between what has been characterized as the classical aldehyde dehydrogenases, the ALDH family (ALDH) and extended family members or aldehyde dehydrogenase-like (ALDH-L) proteins. The ALDH proteins are represented by enzymes which share a number of highly conserved residues necessary for catalysis and cofactor binding and they include such proteins as retinal dehydrogenase, 10-formyltetrahydrofolate dehydrogenase, non-phosphorylating glyceraldehyde 3-phosphate dehydrogenase, delta(1)-pyrroline-5-carboxylate dehydrogenases, alpha-ketoglutaric semialdehyde dehydrogenase, alpha-aminoadipic semialdehyde dehydrogenase, coniferyl aldehyde dehydrogenase and succinate-semialdehyde dehydrogenase.  Included in this larger group are all human, Arabidopsis, Tortula, fungal, protozoan, and Drosophila ALDHs identified in families ALDH1 through ALDH22 with the exception of families ALDH18, ALDH19, and ALDH20 which are present in the ALDH-like group. The ALDH-like group is represented by such proteins as gamma-glutamyl phosphate reductase, LuxC-like acyl-CoA reductase, and coenzyme A acylating aldehyde dehydrogenase. All of these proteins have a conserved cysteine that aligns with the catalytic cysteine of the ALDH group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>4-hydroxybutyrate dehydrogenase, also called gamma-hydroxybutyrate dehydrogenase, catalyzes the reduction of succinic simialdehyde to 4-hydroxybutyrate in the succinic degradation pathway. 4-hydroxybutyrate dehydrogenase (4HBD) is an iron-containing (type III) NAD-dependent alcohol dehydrogenase. It plays a role in the succinate metabolism biochemical pathway. It catalyzes the reduction of succinic simialdehide to 4-hydroxybutyrate in the succinate degradation pathway This succinate degradation pathway is present in some bacteria which can use succinate as sole carbon source.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Maleylacetate reductase is involved in many aromatic compounds degradation pathways of aerobic microbes. Maleylacetate reductase (MAR) plays an important role in the degradation of aromatic compounds  in aerobic microbes. In fungi and yeasts, the enzyme is involved in the catabolism of compounds such as phenol, tyrosine, benzoate, 4-hydroxybenzoate and resorcinol. In bacteria, the enzyme contributes to the degradation of resorcinol, 2,4-dihydroxybenzoate ([beta]-resorcylate) and 2,6-dihydroxybenzoate ([gamma]-resorcylate) via hydroxyquinol and maleylacetate. Maleylacetate reductase catalyzes NADH- or NADPH-dependent reduction, at the carbon-carbon double bond, of maleylacetate or 2-chloromaleylacetate to 3-oxoadipate. In the case of 2-chloromaleylacetate, MAR initially catalyzes the NAD(P)H-dependent dechlorination to maleylacetate, which is then reduced to 3-oxoadipate. This enzyme is a homodimer and is inhibited by thiol-blocking reagents such as p-chloromercuribenzoate and Hg++, indicating that the cysteine residue is probably necessary for the catalytic activity of maleylacetate reductase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Alcohol dehydrogenase YqhD, Fe-dependent ADH family [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Dehydroquinate synthase-like (DHQ-like) and iron-containing alcohol dehydrogenases (Fe-ADH). This superfamily consists of two subgroups: the dehydroquinate synthase (DHQS)-like, and a large metal-containing alcohol dehydrogenases (ADH), known as iron-containing alcohol dehydrogenases. Dehydroquinate synthase (DHQS) catalyzes the conversion of 3-deoxy-D-arabino-heptulosonate-7-phosphate (DAHP) to dehydroquinate (DHQ) in the second step of the shikimate pathway. This pathway involves seven sequential enzymatic steps in the conversion of erythrose 4-phosphate and phosphoenolpyruvate into chorismate for subsequent synthesis of aromatic compounds. Dehydroquinate synthase-like group includes dehydroquinate synthase, 2-deoxy-scyllo-inosose synthase, and 2-epi-5-epi-valiolone synthase. The alcohol dehydrogenases (ADHs) in this superfamily contain a dehydroquinate synthase-like protein structural fold and mostly contain iron. They are distinct from other alcohol dehydrogenases which contains different protein domains. There are several distinct families of alcohol dehydrogenases: Zinc-containing long-chain alcohol dehydrogenases; insect-type, or short-chain alcohol dehydrogenases; iron-containing alcohol dehydrogenases, and others. The iron-containing family has a Rossmann fold-like topology that resembles the fold of the zinc-dependent alcohol dehydrogenases, but lacks sequence homology, and differs in strand arrangement. ADH catalyzes the reversible oxidation of alcohol to acetaldehyde with the simultaneous reduction of NAD(P)+ to NAD(P)H.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>65</td>
            <td id=al>Iron-containing alcohol dehydrogenase similar to Shewanella oneidensis KdnB required for Kdo8N biosynthesis. This family contains iron-containing alcohol dehydrogenase-like proteins, many of which have not been characterized. Their specific function is unknown. The protein structure represents a dehydroquinate synthase-like fold and belongs to the iron-containing alcohol dehydrogenase-like superfamily. It is distinct from other alcohol dehydrogenases which contain different protein domains. Alcohol dehydrogenase catalyzes the reduction of acetaldehyde to alcohol with NADP as cofactor. Its activity requires iron or zinc ions. This family also includes Shewanella oneidensis KdnB which is required for biosynthesis of 8-Amino-3,8-dideoxy-D-manno-octulosonic acid (Kdo8N), a unique amino sugar that has thus far only been observed on the lipopolysaccharides of marine bacteria belonging to the genus Shewanella, and thought to be important for the integrity of the bacterial cell outer membrane. KdnB requires NAD(P) and zinc ion for activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Aldehyde dehydrogenase family. This family of dehydrogenases act on aldehyde substrates. Members use NADP as a cofactor. The family includes the following members: The prototypical members are the aldehyde dehydrogenases EC:1.2.1.3. Succinate-semialdehyde dehydrogenase EC:1.2.1.16. Lactaldehyde dehydrogenase EC:1.2.1.22. Benzaldehyde dehydrogenase EC:1.2.1.28. Methylmalonate-semialdehyde dehydrogenase EC:1.2.1.27. Glyceraldehyde-3-phosphate dehydrogenase EC:1.2.1.9. Delta-1-pyrroline-5-carboxylate dehydrogenase EC: 1.5.1.12. Acetaldehyde dehydrogenase EC:1.2.1.10. Glutamate-5-semialdehyde dehydrogenase EC:1.2.1.41. This family also includes omega crystallin, an eye lens protein from squid and octopus that has little aldehyde dehydrogenase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>NAD(P)+ dependent aldehyde dehydrogenase family. The aldehyde dehydrogenase family (ALDH) of NAD(P)+ dependent enzymes, in general, oxidize a wide range of  endogenous and exogenous aliphatic and aromatic aldehydes to their corresponding carboxylic acids and play an  important role in detoxification. Besides aldehyde detoxification, many ALDH isozymes possess multiple additional catalytic and non-catalytic functions such as participating in  metabolic pathways, or as  binding proteins, or as osmoregulants, to mention a few. The enzyme has three domains, a NAD(P)+ cofactor-binding domain, a catalytic domain, and a bridging domain; and the active enzyme  is generally either homodimeric or homotetrameric. The catalytic mechanism is proposed to involve cofactor binding, resulting in a conformational change and activation of an invariant catalytic cysteine nucleophile. The cysteine and aldehyde substrate form an oxyanion thiohemiacetal intermediate resulting in hydride transfer to the cofactor and formation of a thioacylenzyme intermediate. Hydrolysis of the thioacylenzyme and release of the carboxylic acid product occurs, and in most cases, the reduced cofactor dissociates from the enzyme. The evolutionary phylogenetic tree of ALDHs appears to have an initial bifurcation between what has been characterized as the classical aldehyde dehydrogenases, the ALDH family (ALDH) and extended family members or aldehyde dehydrogenase-like (ALDH-like) proteins. The ALDH proteins are represented by enzymes which share a number of highly conserved residues necessary for catalysis and cofactor binding and they include such proteins as retinal dehydrogenase, 10-formyltetrahydrofolate dehydrogenase, non-phosphorylating glyceraldehyde 3-phosphate dehydrogenase, delta(1)-pyrroline-5-carboxylate dehydrogenases, alpha-ketoglutaric semialdehyde dehydrogenase, alpha-aminoadipic semialdehyde dehydrogenase, coniferyl aldehyde dehydrogenase and succinate-semialdehyde dehydrogenase.  Included in this larger group are all human, Arabidopsis, Tortula, fungal, protozoan, and Drosophila ALDHs identified in families ALDH1 through ALDH22 with the exception of families ALDH18, ALDH19, and ALDH20 which are present in the ALDH-like group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>Glycerol dehydrogenase or related enzyme, iron-containing ADH family [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>Escherichia coli lactaldehyde dehydrogenase AldA-like. Lactaldehyde dehydrogenase from Escherichia coli (AldA, LactADH, EC=1.2.1.22), an NAD(+)-dependent enzyme involved in the metabolism of L-fucose and L-rhamnose, and other similar sequences are present in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>Methanocaldococcus jannaschii NAD+-dependent lactaldehyde dehydrogenase-like. NAD+-dependent, lactaldehyde dehydrogenase (EC=1.2.1.22) involved the biosynthesis of coenzyme F(420) in Methanocaldococcus jannaschii through the oxidation of lactaldehyde to lactate and generation of NAPH, and similar sequences are included in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>ALDH subfamily: NAD+-dependent, lactaldehyde dehydrogenase, ALDH family 21 A1, and related proteins. ALDH subfamily which includes Tortula ruralis aldehyde dehydrogenase ALDH21A1 (RNP123), and NAD+-dependent, lactaldehyde dehydrogenase (EC=1.2.1.22) and like sequences.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>alcohol dehydrogenase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>43</td>
            <td id=al>NAD+-dependent 4-trimethylaminobutyraldehyde dehydrogenase, ALDH family 9A1. NAD+-dependent, 4-trimethylaminobutyraldehyde dehydrogenase (TMABADH, EC=1.2.1.47), also known as aldehyde dehydrogenase family 9 member A1 (ALDH9A1) in humans, is a cytosolic tetramer which catalyzes the oxidation of gamma-aminobutyraldehyde involved in 4-aminobutyric acid (GABA) biosynthesis  and also oxidizes betaine aldehyde (gamma-trimethylaminobutyraldehyde) which is involved in carnitine biosynthesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Uncharacterized ALDH (y4uC) with similarity to Tortula ruralis aldehyde dehydrogenase ALDH21A1. Uncharacterized aldehyde dehydrogenase (ORF name y4uC) with sequence similarity to the moss Tortula ruralis aldehyde dehydrogenase ALDH21A1 (RNP123) believed to play an important role in the detoxification of aldehydes generated in response to desiccation- and salinity-stress, and similar sequences are included in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cb>64</td>
            <td id=al>Pseudomonas fluorescens 4-hydroxymuconic semialdehyde dehydrogenase-like. 4-hydroxymuconic semialdehyde dehydrogenase (HapE, EC=1.2.1.61) of Pseudomonas fluorescens ACB involved in 4-hydroxyacetophenone degradation, and putative hydroxycaproate semialdehyde dehydrogenase (ChnE) of Brachymonas petroleovorans involved in cyclohexane metabolism, and other similar sequences, are present in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>aldehyde dehydrogenase A; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Pseudomonas putida aldehyde dehydrogenase PsfA (ACA09737)-like. Included in this CD is the aldehyde dehydrogenase (PsfA, locus ACA09737) of Pseudomonas putida involved in furoic acid metabolism. Transcription of psfA was induced in response to 2-furoic acid, furfuryl alcohol, and furfural.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>Arabidopsis betaine aldehyde dehydrogenase 1 and 2, ALDH family 10A8 and 10A9-like. Present in this CD are the Arabidopsis betaine aldehyde dehydrogenase (BADH) 1 (chloroplast) and 2 (mitochondria), also known as, aldehyde dehydrogenase family 10 member A8 and aldehyde dehydrogenase family 10 member A9, respectively, and are putative dehydration- and salt-inducible BADHs (EC 1.2.1.8) that catalyze the oxidation of betaine aldehyde to the compatible solute glycine betaine.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>Uncharacterized aldehyde dehydrogenase of Synechococcus sp. PCC 7335 (EDX86601). Uncharacterized aldehyde dehydrogenase of Synechococcus sp. PCC 7335 (locus EDX86601) and other similar sequences, are present in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>Escherichia coli NAD+-dependent phenylacetaldehyde dehydrogenase PadA-like. NAD+-dependent, homodimeric, phenylacetaldehyde dehydrogenase (PADH, EC=1.2.1.39) PadA of Escherichia coli involved in the catabolism of 2-phenylethylamine, and other related sequences, are present in this CD. Also included is the Pseudomonas fluorescens ST StyD PADH involved in styrene catabolism, the Sphingomonas sp. LB126 FldD protein involved in fluorene degradation, and the Novosphingobium aromaticivorans NahF salicylaldehyde dehydrogenase involved in the NAD+-dependent conversion of salicylaldehyde to salicylate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>Uncharacterized Candidatus kuenenia aldehyde dehydrogenase AldH (CAJ73105)-like. Uncharacterized aldehyde dehydrogenase of Candidatus kuenenia AldH (locus CAJ73105) and similar sequences with similarity to alpha-aminoadipic semialdehyde dehydrogenase (AASADH, human ALDH7A1, EC=1.2.1.31), Arabidopsis ALDH7B4, and Streptomyces clavuligerus delta-1-piperideine-6-carboxylate dehydrogenase (P6CDH) are included in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>Streptomyces putative phosphonoformaldehyde dehydrogenase PhpJ-like. Putative phosphonoformaldehyde dehydrogenase (PhpJ), an aldehyde dehydrogenase homolog reportedly involved in the biosynthesis of phosphinothricin tripeptides in Streptomyces viridochromogenes DSM 40736, and similar sequences are included in this CD.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>Pseudomonas putida vanillin dehydrogenase-like. Vanillin dehydrogenase (Vdh, VaniDH) involved in the metabolism of ferulic acid and other related  sequences are included in this CD.  The E. coli vanillin dehydrogenase (LigV) preferred NAD+ to NADP+  and exhibited a broad substrate preference, including vanillin,  benzaldehyde, protocatechualdehyde, m-anisaldehyde, and p-hydroxybenzaldehyde.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>Aldehyde dehydrogenase family 21A1-like. Aldehyde dehydrogenase ALDH21A1 (gene name RNP123) was first described in the moss Tortula ruralis and is believed to play an important role in the detoxification of aldehydes generated in response to desiccation- and salinity-stress, and ALDH21A1 expression represents a unique stress tolerance mechanism. So far, of plants, only the bryophyte sequence has been observed, but similar protein sequences from bacteria and archaea are also present in this CD.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4387842</td></tr>
<tr><td id=al>end</td><td id=al>4390521</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>bifunctional acetaldehyde-CoA/alcohol dehydrogenase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
