<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03078</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03078 DNA mismatch repair endonuclease MutL [Phormidium_A]</h3>
    <div>MPVFSPTMPKIATLPAEVVHAIAAGEVIDSLAAVVRELAENALDAGASRITISVDPQQWFVSVADNGCGMSRQDLQQAASPHSTSKIQNFNDLHQITSLGFRGEALHSLAQLAQLNIYSRQGDDPGWQARYSRQGQIKTLTPTAIAPGTIVEVHQVFQHWPQRRDALPSPPQQLRAIQLMLQNLALCHPQVCWQVRKGNKPWFSLTPGPSPREILPQLLRGVTREDVYYSHTSSLELLLGRPDRCHRGRADWLKFAVNGRVVNLPDLEHTLLSALSRSLPRNRYPLCFLHLRVPPDQLDWNRHPAKSEIYLQHSEQWQRQLREAVDRALALGSLNSSGHSRVRQLLRVQDDSADYRIDSSXXXXXXXXXPRSLMPLRAIGQANRTYIVAEHPDGLWLVEQHIAHERVIYEQLCQDWALQELDPPIILSGLSPQQLENLEHIGITVDPFGDGLWALRQAPIPLAGRDDCADALQELSQGNLQEAQVATACRSAIRNGTLLTLKEMQQLLDRWQQTRNPHTCPHGRPIYLPLDESSLGRYFRRSWVIGKSHGI</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>79</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>DNA mismatch repair protein MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=co>Phormidium</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA mismatch repair endonuclease MutL</td>
            <td id=co>PCC-10802</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair endonuclease MutL. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair ATPase MutL [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA mismatch repair protein MutL. All proteins in this family for which the functions are known are involved in the process of generalized mismatch repair. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histidine kinase-like ATPase domain of DNA mismatch repair proteins Escherichia coli MutL, human MutL homologs (MLH/ PMS), and related domains. This family includes the histidine kinase-like ATPase (HATPase) domains of Escherichia coli MutL, human MLH1 (mutL homolog 1), human PMS1 (PMS1 homolog 1, mismatch repair system component), human MLH3 (mutL homolog 3), and human PMS2 (PMS1 homolog 2, mismatch repair system component). MutL homologs (MLH/PMS) participate in MMR (DNA mismatch repair), and in addition have role(s) in DNA damage signaling and suppression of homologous recombination (recombination between partially homologous parental DNAs). The primary role of MutL in MMR is to mediate protein-protein interactions during mismatch recognition and strand removal; a ternary complex is formed between MutS, MutL, and the mismatched DNA, which activates the MutH endonuclease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>MutL_Trans: transducer domain, having a ribosomal S5 domain 2-like fold, conserved in the C-terminal domain of DNA mismatch repair (MutL/MLH1/PMS2) family. This transducer domain is homologous to the second domain of the DNA gyrase B subunit, which is known to be important in nucleotide hydrolysis and the transduction of structural signals from ATP-binding site to the DNA breakage/reunion regions of the enzymes. Included in this group are proteins similar to human MLH1, hPMS2, hPMS1, hMLH3 and E. coli MutL,  MLH1 forms heterodimers with PMS2, PMS1 and MLH3. These three complexes have distinct functions in meiosis. hMLH1-hPMS2 also participates in the repair of all DNA mismatch repair (MMR) substrates. Roles for hMLH1-hPMS1 or hMLH1-hMLH3 in MMR have not been established. Cells lacking either hMLH1 or hPMS2 have a strong mutator phenotype and display microsatellite instability (MSI). Mutation in hMLH1 causes predisposition to HNPCC, Muir-Torre syndrome and Turcot syndrome (HNPCC variant). Mutation in hPMS2 causes predisposition to HPNCC and Turcot syndrome. Mutation in hMLH1 accounts for a large fraction of HNPCC families. There is no convincing evidence to support hPMS1 having a role in HNPCC predisposition. It has been suggested that hMLH3 may be a low risk gene for colorectal cancer; however there is little evidence to support it having a role in classical HNPCC.  It has been suggested that during initiation of DNA mismatch repair in E. coli, the mismatch recognition protein MutS recruits MutL in the presence of ATP.  The MutS(ATP)-MutL ternary complex formed, then recruits the latent endonuclease MutH.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>DNA mismatch repair protein, C-terminal domain. This family represents the C-terminal domain of the mutL/hexB/PMS1 family. This domain has a ribosomal S5 domain 2-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MutL C terminal dimerisation domain. MutL and MutS are key components of the DNA repair machinery that corrects replication errors. MutS recognises mispaired or unpaired bases in a DNA duplex and in the presence of ATP, recruits MutL to form a DNA signaling complex for repair. The N terminal region of MutL contains the ATPase domain and the C terminal is involved in dimerisation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>MutL C terminal dimerization domain. MutL and MutS are key components of the DNA repair machinery that corrects replication errors. MutS recognizes mispaired or unpaired bases in a DNA duplex and in the presence of ATP, recruits MutL to form a DNA signaling complex for repair. The N terminal region of MutL contains the ATPase domain and the C terminal is involved in dimerization.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>MutL_Trans: transducer domain, having a ribosomal S5 domain 2-like fold, conserved in the C-terminal domain of type II DNA topoisomerases (Topo II) and DNA mismatch repair (MutL/MLH1/PMS2) proteins. This transducer domain is homologous to the second domain of the DNA gyrase B subunit, which is known to be important in nucleotide hydrolysis and the transduction of structural signals from ATP-binding site to the DNA breakage/reunion regions of the enzymes. The GyrB dimerizes in response to ATP binding, and is homologous to the N-terminal half of eukaryotic Topo II and the ATPase fragment of MutL. Type II DNA topoisomerases catalyze the ATP-dependent transport of one DNA duplex through another, in the process generating transient double strand breaks via covalent attachments to both DNA strands at the 5' positions. Included in this group are proteins similar to human MLH1 and PMS2.  MLH1 forms a heterodimer with PMS2 which functions in meiosis and in DNA mismatch repair (MMR). Cells lacking either hMLH1 or hPMS2 have a strong mutator phenotype and display microsatellite instability (MSI). Mutation in hMLH1 accounts for a large fraction of Lynch syndrome (HNPCC) families.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>MutL_Trans_MutL: transducer domain, having a ribosomal S5 domain 2-like fold, found in proteins similar to Escherichia coli MutL.  EcMutL belongs to the DNA mismatch repair (MutL/MLH1/PMS2) family.  This transducer domain is homologous to the second domain of the DNA gyrase B subunit, which is known to be important in nucleotide hydrolysis and the transduction of structural signals from the ATP-binding site to the DNA breakage/reunion regions of the enzymes.  It has been suggested that during initiation of DNA mismatch repair in E. coli, the mismatch recognition protein MutS recruits MutL in the presence of ATP.  The MutS(ATP)-MutL ternary complex formed, then recruits the latent endonuclease MutH. Prokaryotic MutS and MutL are homodimers.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>MutL_Trans_MLH1: transducer domain, having a ribosomal S5 domain 2-like fold, found in proteins similar to yeast and human MLH1 (MutL homologue 1). This transducer domain is homologous to the second domain of the DNA gyrase B subunit, which is known to be important in nucleotide hydrolysis and the transduction of structural signals from ATP-binding site to the DNA breakage/reunion regions of the enzymes. MLH1 forms heterodimers with PMS2, PMS1 and MLH3. These three complexes have distinct functions in meiosis. hMLH1-hPMS2 also participates in the repair of all DNA mismatch repair (MMR) substrates. Roles for hMLH1-hPMS1 or hMLH1-hMLH3 in MMR have not been established. Cells lacking hMLH1 have a strong mutator phenotype and display microsatellite instability (MSI). Mutation in hMLH1 causes predisposition to HNPCC, Muir-Torre syndrome and Turcot syndrome (HNPCC variant). Mutation in hMLH1 accounts for a large fraction of HNPCC families.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>Histidine kinase-, DNA gyrase B-, and HSP90-like ATPase. This family represents, additionally, the structurally related ATPase domains of histidine kinase, DNA gyrase B and HSP90.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2585421</td></tr>
<tr><td id=al>end</td><td id=al>2587077</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA mismatch repair endonuclease MutL</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
