<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00096</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00096 ribonuclease III [Phormidium_A]</h3>
    <div>MTAVLGLNFKNTDLLNQALIHRSYASESPLPIEHNERLEFLGDAIINFISGSFLYQVYPKFSEEKLTKLRSKLVDEPHLASLARQLQLNDQIKLGQGERQANGSEKDSILSDAFEAVIGAYFLDAGIEAVREFVEPLLSQAIAEMAEAKDLEEQLSNLITDAKSRLQEWVQKQHGSPPNYRLVEKRGEPHIPTFVVQVLIEGKVWGTGTGSRKQKAEKQAAQEALKQLKAAGLMPQ</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>79</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>94</td>
            <td id=cg>95</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>93</td>
            <td id=cg>94</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>93</td>
            <td id=cg>94</td>
            <td id=al>ribonuclease III</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>92</td>
            <td id=cb>60</td>
            <td id=al>ribonuclease III</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>92</td>
            <td id=cg>94</td>
            <td id=al>ribonuclease III</td>
            <td id=co>CCY15215</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>45</td>
            <td id=cg>95</td>
            <td id=cg>98</td>
            <td id=al>PRK00102 ribonuclease III; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>TIGR02191 ribonuclease III, bacterial. This family consists of bacterial examples of ribonuclease III. This enzyme cleaves double-stranded rRNA. It is involved in processing ribosomal RNA precursors. It is found even in minimal genones such as Mycoplasma genitalium and Buchnera aphidicola, and in some cases has been shown to be an essential gene. These bacterial proteins contain a double-stranded RNA binding motif (pfam00035) and a ribonuclease III domain (pfam00636). Eukaryotic homologs tend to be much longer proteins with additional domains, localized to the nucleus, and not included in this family. [Transcription, RNA processing]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>94</td>
            <td id=cg>95</td>
            <td id=al>COG0571 dsRNA-specific ribonuclease [Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=al>cd00593 RIBOc. Ribonuclease III C terminal domain. This group consists of eukaryotic, bacterial and archeal ribonuclease III (RNAse III) proteins. RNAse III is a double stranded RNA-specific endonuclease. Prokaryotic RNAse III is important in post-transcriptional control of mRNA stability and translational efficiency. It is involved in the processing of ribosomal RNA precursors. Prokaryotic RNAse III also plays a role in the maturation of tRNA precursors and in the processing of phage and plasmid transcripts. Eukaryotic RNase III's participate (through direct cleavage) in rRNA processing, in processing of small nucleolar RNAs (snoRNAs) and snRNA's (components of the spliceosome). In eukaryotes RNase III or RNaseIII like enzymes such as Dicer are involved in RNAi (RNA interference) and miRNA (micro-RNA) gene silencing.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=al>smart00535 Ribonuclease III family. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=co>52</td>
            <td id=cg>97</td>
            <td id=al>pfam14622 Ribonuclease-III-like. Members of this family are involved in rDNA transcription and rRNA processing. They probably also cleave a stem-loop structure at the 3' end of U2 snRNA to ensure formation of the correct U2 3' end; they are involved in polyadenylation-independent transcription termination. Some members may be mitochondrial ribosomal protein subunit L15, others may be 60S ribosomal protein L3.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>96</td>
            <td id=co>55</td>
            <td id=al>PRK12372 ribonuclease III; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>97</td>
            <td id=co>49</td>
            <td id=al>PRK14718 ribonuclease III; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>94</td>
            <td id=cg>95</td>
            <td id=al>PRK12371 ribonuclease III; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=al>cd10845 double-stranded RNA binding motif of ribonuclease III (RNase III) and similar proteins. RNase III (EC 3.1.26.3; also known as ribonuclease 3) digests double-stranded RNA formed within single-strand substrates, but not RNA-DNA hybrids. It is involved in the processing of rRNA precursors, viral transcripts, some mRNAs, and at least 1 tRNA (metY, a minor form of tRNA-init-Met). It cleaves the 30S primary rRNA transcript to yield the immediate precursors to the 16S and 23S rRNAs. The cleavage can occur in assembled 30S, 50S, and even 70S subunits and is influenced by the presence of ribosomal proteins. The RNase III family also includes the mitochondrion-specific ribosomal protein mL44 subfamily, which is composed of mitochondrial 54S ribosomal protein L3 (MRPL3) and mitochondrial 39S ribosomal protein L44 (MRPL44). Members of this family contain an RNase III domain and a C-terminal double-stranded RNA binding motif (DSRM). DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=al>pfam00636 Ribonuclease III domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>smart00358 Double-stranded RNA binding motif. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>pfam00035 Double-stranded RNA binding motif. Sequences gathered for seed by HMM_iterative_training Putative motif shared by proteins that bind to dsRNA. At least some DSRM proteins seem to bind to specific RNA targets. Exemplified by Staufen, which is involved in localization of at least five different mRNAs in the early Drosophila embryo. Also by interferon-induced protein kinase in humans, which is part of the cellular response to dsRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=al>cd00048 double-stranded RNA binding motif (DSRM) superfamily. DSRM (also known as dsRBM) is a 65-70 amino acid domain that adopts an alpha-beta-beta-beta-alpha fold. It is not sequence specific, but highly specific for double-stranded RNAs (dsRNAs) of various origin and structure. The DSRM domains are found in a variety of proteins including dsRNA dependent protein kinase PKR, RNA helicases, Drosophila Staufen protein, E. coli RNase III, RNase H1, and dsRNA dependent adenosine deaminases. They are involved in numerous cellular mechanisms ranging from localization and transport of messenger RNAs, through maturation and degradation of RNAs, to viral response and signal transduction. Some members harbor tandem DSRMs that act in small RNA biogenesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>31</td>
            <td id=cg>98</td>
            <td id=al>cd19877 double-stranded RNA binding motif of metazoan ribonuclease III (RNase III) and similar proteins. RNase III (EC 3.1.26.3; also known as Drosha, or ribonuclease 3) is a double-stranded RNA (dsRNA)-specific endoribonuclease that is involved in the initial step of microRNA (miRNA) biogenesis. It is a component of the microprocessor complex that is required to process primary miRNA transcripts (pri-miRNAs) to release precursor miRNA (pre-miRNA) in the nucleus. Within the microprocessor complex, RNase III cleaves the 3' and 5' strands of a stem-loop in pri-miRNAs (processing center 11 bp from the dsRNA-ssRNA junction) to release hairpin-shaped pre-miRNAs that are subsequently cut by the cytoplasmic DICER to generate mature miRNAs. It is also involved in pre-rRNA processing. Metazoan RNase III is a larger protein than bacterial RNase III. It contains two RNase III domains in the C-terminal half of the protein followed by a double-stranded RNA binding motif (DSRM). DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>27</td>
            <td id=cg>95</td>
            <td id=al>cd19903 first double-stranded RNA binding motif of eukaryotic translation initiation factor 2-alpha kinase 2 (EIF2AK2) and similar proteins. EIF2AK2 (EC 2.7.11.1/EC 2.7.10.2; also known as interferon-induced, double-stranded RNA-activated protein kinase, eIF-2A protein kinase 2, interferon-inducible RNA-dependent protein kinase, P1/eIF-2A protein kinase, protein kinase RNA-activated (PKR), protein kinase R, tyrosine-protein kinase EIF2AK2, or p68 kinase) acts as an IFN-induced dsRNA-dependent serine/threonine-protein kinase which plays a key role in the innate immune response to viral infection and is also involved in the regulation of signal transduction, apoptosis, cell proliferation and differentiation. EIF2AK2 proteins contain two to three double-stranded RNA binding motifs (DSRMs). This model describes the first motif. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>28</td>
            <td id=cg>95</td>
            <td id=al>cd19862 first double-stranded RNA binding motif of protein activator of the interferon-induced protein kinase (PRKRA) and similar proteins. This family includes protein activator of the interferon-induced protein kinase (PRKRA) and the RISC-loading complex subunit TARBP2. PRKRA (also known as interferon-inducible double-stranded RNA-dependent protein kinase activator A, PKR-associated protein X (RAX), PKR-associating protein X, protein kinase, interferon-inducible double-stranded RNA-dependent activator, PACT, or HSD14) is a cellular activator for double-stranded RNA-dependent protein kinase during stress signaling. TARBP2 (also called TAR RNA-binding protein 2, or trans-activation-responsive RNA-binding protein (TRBP)), participates in the formation of the RNA-induced silencing complex (RISC). It is part of the RISC-loading complex (RLC), together with dicer1 and eif2c2/ago2, and is required to process precursor miRNAs. This family also includes Drosophila melanogaster Loquacious and similar proteins. Loquacious (Loqs) is a double-stranded RNA-binding domain (dsRBD) protein, a homolog of human TAR RNA binding protein (TRBP) that is a protein first identified as binding the HIV trans-activator RNA (TAR). Loqs interacts with Dicer1 (dmDcr1) to facilitate miRNA processing. PRKRA family proteins contain three double-stranded RNA binding motifs (DSRMs). This model describes the first motif. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>cd19875 double-stranded RNA binding motif of eukaryotic translation initiation factor 2-alpha kinase 2 (EIF2AK2) and similar proteins. The family includes EIF2AK2 and adenosine deaminase domain-containing proteins, ADAD1 and ADAD2. EIF2AK2 (EC 2.7.11.1/EC 2.7.10.2; also known as interferon-induced, double-stranded RNA-activated protein kinase, eIF-2A protein kinase 2, interferon-inducible RNA-dependent protein kinase, P1/eIF-2A protein kinase, protein kinase RNA-activated (PKR), protein kinase R, tyrosine-protein kinase EIF2AK2, or p68 kinase) acts as an IFN-induced dsRNA-dependent serine/threonine-protein kinase which plays a key role in the innate immune response to viral infection and is also involved in the regulation of signal transduction, apoptosis, cell proliferation and differentiation. ADAD1 (also called testis nuclear RNA-binding protein (TENR)) and ADAD2 (also called testis nuclear RNA-binding protein-like (TENRL)) are phylogenetically related to a family of adenosine deaminases involved in RNA editing. ADAD1 plays an essential function in spermatid morphogenesis. It may be involved in testis-specific nuclear post-transcriptional processes such as heterogeneous nuclear RNA (hnRNA) packaging, alternative splicing, or nuclear/cytoplasmic transport of mRNAs. ADAD2 is a double-stranded RNA binding protein with unclear biological function. Members of this group contains varying numbers of double-stranded RNA binding motifs (DSRMs). DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>28</td>
            <td id=cg>95</td>
            <td id=al>cd19902 double-stranded RNA binding motif of double-stranded RNA-specific adenosine deaminase (DRADA) and similar proteins. DRADA (EC 3.5.4.37; also known as 136 kDa double-stranded RNA-binding protein (p136), interferon-inducible protein 4 (IFI-4), K88DSRBP, ADAR1, G1P1, or ADAR) catalyzes the hydrolytic deamination of adenosine to inosine in double-stranded RNA (dsRNA), referred to as A-to-I RNA editing. DRADA family members contain at least one double-stranded RNA binding motifs (DSRM); vertebrate proteins contain three. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>27</td>
            <td id=cg>94</td>
            <td id=al>cd19904 second double-stranded RNA binding motif of eukaryotic translation initiation factor 2-alpha kinase 2 (EIF2AK2) and similar proteins. EIF2AK2 (EC 2.7.11.1/EC 2.7.10.2; also known as interferon-induced, double-stranded RNA-activated protein kinase, eIF-2A protein kinase 2, interferon-inducible RNA-dependent protein kinase, P1/eIF-2A protein kinase, protein kinase RNA-activated (PKR), protein kinase R, tyrosine-protein kinase EIF2AK2, or p68 kinase) acts as an IFN-induced dsRNA-dependent serine/threonine-protein kinase which plays a key role in the innate immune response to viral infection and is also involved in the regulation of signal transduction, apoptosis, cell proliferation and differentiation. EIF2AK2 proteins contain two to three double-stranded RNA binding motifs (DSRMs). This model describes the second motif. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>30</td>
            <td id=cg>97</td>
            <td id=al>cd19870 double-stranded RNA binding motif of protein SON and similar proteins. Protein SON (also known as Bax antagonist selected in saccharomyces 1 (BASS1), negative regulatory element-binding protein (NRE-binding protein), or protein DBP-5, or SON3) is an RNA-binding protein which acts as an mRNA splicing cofactor by promoting efficient splicing of transcripts that possess weak splice sites. It specifically promotes splicing of many cell-cycle and DNA-repair transcripts that possess weak splice sites, such as TUBG1, KATNB1, TUBGCP2, AURKB, PCNT, AKT1, RAD23A, and FANCG. Members of this group contain a double-stranded RNA binding motif (DSRM) at the C-terminus. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>cd19878 double-stranded RNA binding motif of Arabidopsis thaliana double-stranded RNA-binding proteins (AtDRBs)and similar proteins. This family includes a group of Arabidopsis thaliana double-stranded RNA-binding proteins (AtDRB1-5). They bind double-stranded RNA (dsRNA) and may be involved in RNA-mediated silencing. Members of this family contain two to three double-stranded RNA binding motifs (DSRMs). DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>19</td>
            <td id=cb>74</td>
            <td id=al>cd19865 first double-stranded RNA binding motif of STRBP, ILF3, RED1, RED2 and similar proteins. This family includes spermatid perinuclear RNA-binding protein (STRBP) and interleukin enhancer-binding factor 3 (ILF3), as well as two RNA-editing deaminases, RED1 and RED2. STRBP is a double-stranded DNA and RNA binding protein that is involved in spermatogenesis and sperm function. It plays a role in regulation of cell growth. ILF3 (also known as double-stranded RNA-binding protein 76 (DRBP76), M-phase phosphoprotein 4 (MPP4), nuclear factor associated with dsRNA (NFAR), nuclear factor of activated T-cells 90 kDa (NF-AT-90), or translational control protein 80 (TCP80)) is an RNA-binding protein that plays an essential role in the biogenesis of circular RNAs (circRNAs) which are produced by back-splicing circularization of pre-mRNAs. RED1 (EC 3.5.4.37; also called double-stranded RNA-specific editase 1, RNA-editing enzyme 1, dsRNA adenosine deaminase, ADARB1, ADAR2, or DRADA2) catalyzes the hydrolytic deamination of adenosine to inosine in double-stranded RNA (dsRNA), referred to as A-to-I RNA editing. RED2 (also called double-stranded RNA-specific editase B2, RNA-dependent adenosine deaminase 3, RNA-editing enzyme 2, dsRNA adenosine deaminase B2, ADAR3, or ADARB2) prevents the binding of other ADAR enzymes to targets in vitro, and decreases the efficiency of  these enzymes. It is capable of binding to dsRNA, but also to ssRNA. RED2 lacks editing activity for currently known substrate RNAs. Members of this group contain two double-stranded RNA binding motifs (DSRMs). This model describes the first motif. DSRM is not sequence specific, but highly specific for dsRNAs of various origin and structure.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>79108</td></tr>
<tr><td id=al>end</td><td id=al>79819</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>ribonuclease III</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>237</td></tr>
</table>

    </div>
</body>
</html>
