<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04788</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04788 signal peptide peptidase SppA [Phormidium_A]</h3>
    <div>MVWFFKRFRSKIARVEVSGAIASKTRQEVLKALETVAERKYPALLLRIDSPGGTVTDSHEIYQALRRLRDKHKVKVVASFGNISASGGVYIGMGAEKIVSNPGTITGSIGVILRGNNLERLLDKIGVSFKTIKSGPYKDILAFDRELSEGEQQILQSLIDNTYDQFVETVATARQLPVDTVREFADGRIFNGQQALDLKLVDALGTEEDARRLAASLAGLDPERAELDEIETAKPFWAKLLRSSSSVSRLSATGEWLEFELTTNGMPLWMYRP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>protease IV</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=co>Microcoleus</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=co>Microcoleus</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA</td>
            <td id=co>Microcoleus</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>signal peptide peptidase SppA, 36K type. The related but duplicated, double-length protein SppA (protease IV) of E. coli was shown experimentally to degrade signal peptides as are released by protein processing and secretion. This protein shows stronger homology to the C-terminal region of SppA than to the N-terminal domain or to the related putative protease SuhB. The member of this family from Bacillus subtilis was shown to have properties consistent with a role in degrading signal peptides after cleavage from precursor proteins, although it was not demonstrated conclusively. [Protein fate, Degradation of proteins, peptides, and glycopeptides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Signal peptide peptidase A (SppA), a serine protease, has catalytic Ser-Lys dyad. Signal peptide peptidase A (SppA; Peptidase S49; Protease IV): SppA is found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. This subfamily contains members with either a single domain (sometimes referred to as 36K type), such as sohB peptidase, protein C and archaeal signal peptide peptidase, or an amino-terminal domain in addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members (sometimes referred to as 67K type), similar to E. coli and Arabidopsis thaliana SppA peptidases. Site-directed mutagenesis and sequence analysis have shown these SppAs to be serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. Interestingly, the single membrane spanning E. coli SppA carries out catalysis using a Ser-Lys dyad with the serine located in the conserved carboxy-terminal protease domain and the lysine in the non-conserved amino-terminal domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cb>70</td>
            <td id=al>Periplasmic serine protease, ClpP class [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Peptidase family S49. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>signal peptide peptidase SppA, 67K type. This model represents the signal peptide peptidase A (SppA, protease IV) as found in E. coli, Treponema pallidum, Mycobacterium leprae, and several other species, in which it has a molecular mass around 67 kDa and a duplication such that the N-terminal half shares extensive homology with the C-terminal half. This enzyme was shown in E. coli to form homotetramers. E. coli SohB, which is most closely homologous to the C-terminal duplication of SppA, is predicted to perform a similar function of small peptide degradation, but in the periplasm. Many prokaryotes have a single SppA/SohB homolog that may perform the function of either or both. [Protein fate, Degradation of proteins, peptides, and glycopeptides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>39</td>
            <td id=al>signal peptide peptidase SppA, 67K type. This model represents the signal peptide peptidase A (SppA, protease IV) as found in E. coli, Treponema pallidum, Mycobacterium leprae, and several other species, in which it has a molecular mass around 67 kDa and a duplication such that the N-terminal half shares extensive homology with the C-terminal half. This enzyme was shown in E. coli to form homotetramers. E. coli SohB, which is most closely homologous to the C-terminal duplication of SppA, is predicted to perform a similar function of small peptide degradation, but in the periplasm. Many prokaryotes have a single SppA/SohB homolog that may perform the function of either or both. [Protein fate, Degradation of proteins, peptides, and glycopeptides]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>Signal peptide peptidase A (SppA) 36K type, a serine protease, has catalytic Ser-Lys dyad. Signal peptide peptidase A (SppA; Peptidase S49; Protease IV) 36K type: SppA is found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Members in this subfamily are all bacterial and include sohB peptidase and protein C. These are sometimes referred to as 36K type since they contain only one domain, unlike E. coli SppA that also contains an amino-terminal domain. Site-directed mutagenesis and sequence analysis have shown these SppAs to be serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>signal peptide peptidase SppA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Signal peptide peptidase A (SppA), a serine protease, has catalytic Ser-Lys dyad. Signal peptide peptidase A (SppA; Peptidase S49; Protease IV): SppAs in this subfamily are found in all three domains of life and are involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Site-directed mutagenesis and sequence analysis have shown these bacterial, archaeal and thylakoid SppAs to be serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad (both residues absolutely conserved within bacteria, chloroplast and mitochondrial signal peptidase family members) and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. In addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members, the E. coli SppA contains an amino-terminal domain, similar to Arabidopsis thaliana SppA1 peptidase. Others, including sohB peptidase, protein C and archaeal signal peptide peptidase, do not contain the amino-terminal domain. Interestingly, the single membrane spanning E. coli SppA carries out catalysis using a Ser-Lys dyad with the serine located in the conserved carboxy-terminal protease domain and the lysine in the non-conserved amino-terminal domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>Signal peptide peptidase A (SppA) 67K type, a serine protease, has catalytic Ser-Lys dyad. Signal peptide peptidase A (SppA; Peptidase S49; Protease IV) 67K type: SppA is found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Members in this subfamily contain an amino-terminal domain in addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members (sometimes referred to as 67K type), similar to E. coli and Arabidopsis thaliana SppA peptidases. Unlike the eukaryotic functional homologs that are proposed to be aspartic proteases, site-directed mutagenesis and sequence analysis have shown that members in this subfamily, mostly bacterial, are serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad (both residues absolutely conserved within bacteria, chloroplast and mitochondrial signal peptidase family members) and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. Interestingly, the single membrane spanning E. coli SppA carries out catalysis using a Ser-Lys dyad with the serine located in the conserved carboxy-terminal protease domain and the lysine in the non-conserved amino-terminal domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Signal peptide peptidase A. Signal peptide peptidase A (SppA; Peptidase S49; Protease IV): SppA is an intramembrane enzyme found in all three domains of life and is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Unlike the eukaryotic functional homologs that are proposed to be aspartic proteases, site-directed mutagenesis and sequence analysis have shown these bacterial, archaeal and thylakoid SppAs to be ClpP-like serine proteases. The predicted active site serine for members in this family occurs in a transmembrane domain, cleaving peptide bonds in the plane of the lipid bilayer. Mutagenesis studies also suggest that the catalytic center comprises a Ser-Lys dyad (both residues absolutely conserved within bacteria, chloroplast and mitochondrial signal peptidase family members) and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. In addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members, the E. coli SppA contains an amino-terminal domain (sometimes referred to as 67K type). Others, including sohB peptidase, protein C, protein 1510-N and archaeal signal peptide peptidase, do not contain the amino-terminal domain (sometimes referred to as 36K type). Interestingly, the single membrane spanning E. coli SppA carries out catalysis using a Ser-Lys dyad with the serine located in the conserved carboxy-terminal protease domain and the lysine in the non-conserved amino-terminal domain. This family also contains homologs that either have been found experimentally to be without peptidase activity, or lack amino acid residues that are believed to be essential for the catalytic activity of peptidases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>putative inner membrane peptidase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Caseinolytic protease (ClpP) is an ATP-dependent protease. Clp protease (caseinolytic protease; ClpP; endopeptidase Clp; Peptidase S14; ATP-dependent protease, ClpAP)-like enzymes are highly conserved serine proteases and belong to the ClpP/Crotonase superfamily. Included in this family are Clp proteases that are involved in a number of cellular processes such as degradation of misfolded proteins, regulation of short-lived proteins and housekeeping removal of dysfunctional proteins. They are also implicated in the control of cell growth, targeting DNA-binding protein from starved cells. The functional Clp protease is comprised of two components: a proteolytic component and one of several regulatory ATPase components, both of which are required for effective levels of protease activity in the presence of ATP. Active site consists of the triad Ser, His and Asp, preferring hydrophobic or non-polar residues at P1 or P1' positions. The protease exists as a tetradecamer made up of two heptameric rings stacked back-to-back such that the catalytic triad of each subunit is located at the interface between three monomers, thus making oligomerization essential for function. Another family included in this class of enzymes is the signal peptide peptidase A (SppA; S49) which is involved in the cleavage of signal peptides after their removal from the precursor proteins by signal peptidases. Mutagenesis studies suggest that the catalytic center of SppA comprises a Ser-Lys dyad and not the usual Ser-His-Asp catalytic triad found in the majority of serine proteases. In addition to the carboxyl-terminal protease domain that is conserved in all the S49 family members, the E. coli SppA contains an amino-terminal domain. Others, including sohB peptidase, protein C, protein 1510-N and archaeal signal peptide peptidase, do not contain the amino-terminal domain. The third family included in this hierarchy is nodulation formation efficiency D (NfeD) which is a membrane-bound Clp-class protease and only found in bacteria and archaea. Majority of the NfeD genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named stomatin operon partner protein (STOPP). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 from Pyrococcus horikoshii has been shown to possess serine protease activity having a Ser-Lys catalytic dyad.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>Nodulation formation efficiency D (NfeD) is a membrane-bound ClpP-class protease. Nodulation formation efficiency D (NfeD; stomatin operon partner protein, STOPP; DUF107) is a member of membrane-anchored ClpP-class proteases. Currently, more than 300 NfeD homologs have been identified - all of which are bacterial or archaeal in origin. Majority of these genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named STOPP (stomatin operon partner protein). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 (1510-N or PH1510-N) from Pyrococcus horikoshii has been shown to possess serine protease activity and has a Ser-Lys catalytic dyad, preferentially cleaving hydrophobic substrates. Difference in oligomeric form and catalytic residues between 1510-N (forming a dimer) and ClpP (forming a tetradecamer) shows a possible functional difference: 1510-N is likely to have a regulatory function while ClpP is involved in protein quality control.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>Nodulation formation efficiency D (NfeD) is a membrane-bound ClpP-class protease. Nodulation formation efficiency D (NfeD; stomatin operon partner protein, STOPP; DUF107) is a member of membrane-anchored ClpP-class proteases. Currently, more than 300 NfeD homologs have been identified - all of which are bacterial or archaeal in origin. Majority of these genomes have been shown to possess operons containing a homologous NfeD/stomatin gene pair, causing NfeD to be previously named STOPP (stomatin operon partner protein). NfeD homologs can be divided into two groups: long and short forms. Long-form homologs have a putative ClpP-class serine protease domain while the short form homologs do not. Downstream from the ClpP-class domain is the so-called NfeD or DUF107 domain. N-terminal region of the NfeD homolog PH1510 (1510-N or PH1510-N) from Pyrococcus horikoshii has been shown to possess serine protease activity and has a Ser-Lys catalytic dyad, preferentially cleaving hydrophobic substrates. Difference in oligomeric form and catalytic residues between 1510-N (forming a dimer) and ClpP (forming a tetradecamer) shows a possible functional difference: 1510-N is likely to have a regulatory function while ClpP is involved in protein quality control.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cr>25</td>
            <td id=al>Membrane-bound serine protease (ClpP class)  [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3893483</td></tr>
<tr><td id=al>end</td><td id=al>3894305</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[secondary-metabolites]</td></tr>
<tr><td id=al>descr</td><td id=al>signal peptide peptidase SppA</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al>(region 6) NRPS-like cds((PP-binding or NAD_binding_4) and (AMP-binding or A-OX))  </td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
