<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00610</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00610 HAD family hydrolase [Phormidium_A]</h3>
    <div>MNLSCGPKTFQNIRAVIFDKDGTLLDTQDFLRRLGIDRARQLDAQIPGVGECLMMAFGIEGDRLDPTYLLAVGSRQENLIAAAAYVAETGRSWAESLQVAQTAFDESDRLLKQEDTSPLFVGSLEVLYSLAEAGFKLGILSADTNERVNQFVQTCQLEEIIQVKMGVDDGPSKPDPTLFQQACAALGVRPEEAIMVGDSPLDIEMAQRAGAAGTIGICWHSQDHRHLSQADVTIEQLDQLQVVP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>phosphoglycolate phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>HAD family hydrolase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>HAD family hydrolase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>TPA: HAD family hydrolase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>HAD family hydrolase</td>
            <td id=co>PCC7113</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphoglycolate phosphatase, HAD superfamily [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>uncharacterized family of the haloacid dehalogenase-like (HAD) hydrolase superfamily, similar to Parvibaculum lavamentivorans  HAD-superfamily hydrolase, subfamily IA, variant 1. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pyrophosphatase similar to Bacillus subtilis PpaX. This family includes Bacillus subtilis PpaX which hydrolyzes pyrophosphate formed during serine-46-phosphorylated HPr (P-Ser-HPr) dephosphorylation by the bifunctional enzyme HPr kinase/phosphorylase. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>haloacid dehalogenase-like hydrolase. This family is structurally different from the alpha/beta hydrolase family (pfam00561). This family includes L-2-haloacid dehalogenase, epoxide hydrolases and phosphatases. The structure of the family consists of two domains. One is an inserted four helix bundle, which is the least well conserved region of the alignment, between residues 16 and 96 of Pseudomonas sp. (S)-2-haloacid dehalogenase 1. The rest of the fold is composed of the core alpha/beta domain. Those members with the characteristic DxD triad at the N-terminus are probably phosphatidylglycerolphosphate (PGP) phosphatases involved in cardiolipin biosynthesis in the mitochondria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Haloacid dehalogenase-like hydrolase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pyrophosphatase PpaX; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Beta-phosphoglucomutase or related phosphatase, HAD superfamily  [Carbohydrate transport and metabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>haloacid dehalogenase superfamily, subfamily IA, variant 1 with third motif having Dx(3-4)D or Dx(3-4)E. This model represents part of one structural subfamily of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called "capping domain", or the absence of such a domain. Subfamily I consists of sequences in which the capping domain is found in between the first and second catalytic motifs. Subfamily II consists of sequences in which the capping domain is found between the second and third motifs. Subfamily III sequences have no capping domain in either of these positions.The Subfamily IA and IB capping domains are predicted by PSI-PRED to consist of an alpha helical bundle. Subfamily I encompasses such a wide region of sequence space (the sequences are highly divergent) that representing it with a single model is impossible, resulting in an overly broad description which allows in many unrelated sequences. Subfamily IA and IB are separated based on an aparrent phylogenetic bifurcation. Subfamily IA is still too broad to model, but cannot be further subdivided into large chunks based on phylogenetic trees. Of the three motifs defining the HAD superfamily, the third has three variant forms: (1) hhhhsDxxx(x)(D/E), (2) hhhhssxxx(x)D and (3) hhhhDDxxx(x)s where _s_ refers to a small amino acid and _h_ to a hydrophobic one. All three of these variants are found in subfamily IA. Individual models were made based on seeds exhibiting only one of the variants each. Variant 1 (this model) is found in the enzymes phosphoglycolate phosphatase (TIGR01449) and enolase-phosphatase. These three variant models (see also TIGR01493 and TIGR01509) were created withthe knowledge that there will be overlap among them - this is by design and serves the purpose of eliminating the overlap with models of more distantly relatedHAD subfamilies caused by an overly broad single model. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>FMN phosphatase YigB, HAD superfamily [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-acetylmuramic acid 6-phosphate phosphatase MupP. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>Haloacid dehalogenase-like hydrolases. The haloacid dehalogenase-like (HAD) superfamily includes L-2-haloacid dehalogenase, epoxide hydrolase, phosphoserine phosphatase, phosphomannomutase, phosphoglycolate phosphatase, P-type ATPase, and many others. This superfamily includes a variety of enzymes that catalyze the cleavage of substrate C-Cl, P-C, and P-OP bonds via nucleophilic substitution pathways. All of which use a nucleophilic aspartate in their phosphoryl transfer reaction. They catalyze nucleophilic substitution reactions at phosphorus or carbon centers, using a conserved Asp carboxylate in covalent catalysis. All members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. Members of this superfamily are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>haloacid dehalogenase superfamily, subfamily IA, variant 3 with third motif having DD or ED. This model represents part of one structural subfamily of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The superfamily is defined by the presence of three short catalytic motifs. The subfamilies are defined based on the location and the observed or predicted fold of a so-called "capping domain", or the absence of such a domain. Subfamily I consists of sequences in which the capping domain is found in between the first and second catalytic motifs. Subfamily II consists of sequences in which the capping domain is found between the second and third motifs. Subfamily III sequences have no capping domain in either of these positions. The Subfamily IA and IB capping domains are predicted by PSI-PRED to consist of an alpha helical bundle. Subfamily I encompasses such a wide region of sequence space (the sequences are highly divergent) that representing it with a single model is impossible, resulting in an overly broad description which allows in many unrelated sequences. Subfamily IA and IB are separated based on an aparrent phylogenetic bifurcation. Subfamily IA is still too broad to model, but cannot be further subdivided into large chunks based on phylogenetic trees. Of the three motifs defining the HAD superfamily, the third has three variant forms: (1) hhhhsDxxx(x)D, (2) hhhhssxxx(x)D and (3) hhhhDDxxx(x)s where _s_ refers to a small amino acid and _h_ to a hydrophobic one. All three of these variants are found in subfamily IA. Individual models were made based on seeds exhibiting only one of the variants each. Variant 3 (this model) is found in the enzymes beta-phosphoglucomutase (TIGR01990) and deoxyglucose-6-phosphatase, while many other enzymes of subfamily IA exhibit this variant as well as variant 1 (TIGR01549). These three variant models were created with the knowledge that there will be overlap among them - this is by design and serves the purpose of eliminating the overlap with models of more distantly related HAD subfamilies caused by an overly broad single model. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Escherichia coli Gph phosphoglycolate phosphatase and related proteins; belongs to the haloacid dehalogenase-like superfamily. Phosphoglycolate phosphatase (PGP; EC 3.1.3.18) catalyzes the conversion of 2-phosphoglycolate into glycolate and phosphate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>uncharacterized family of the haloacid dehalogenase-like superfamily, similar to uncharacterized Drosophila melanogaster rhythmically expressed gene 2 protein and human haloacid dehalogenase-like hydrolase domain-containing protein 3. The haloacid dehalogenase-like (HAD) hydrolases are a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. Members are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-terminal lipase phosphatase domain of human soluble epoxide hydrolase, Escherichia coli YihX/HAD4 alpha-D-glucose 1-phosphate phosphatase, and related domains, may be inactive. This family includes the N-terminal phosphatase domain of human soluble epoxide hydrolase (sEH). sEH is a bifunctional enzyme with two distinct enzyme activities, the C-terminal domain has epoxide hydrolysis activity and the N-terminal domain (Ntermphos), which belongs to this family, has lipid phosphatase activity. The latter prefers mono-phosphate esters, and lysophosphatidic acids (LPAs) are the best natural substrates found to date.  In addition this family includes Gallus gallus sEH and Xenopus sEH which appears to lack phosphatase activity, and Escherichia coli YihX/HAD4 which selectively hydrolyzes alpha-Glucose-1-P, phosphatase, has significant phosphatase activity against pyridoxal phosphate, and has low beta phosphoglucomutase activity. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>human N-acetylneuraminate-9-phosphate phosphatase, Escherichia coli house-cleaning phosphatase YjjG, and related phosphatases. N-acetylneuraminate-9- phosphatase (Neu5Ac-9-Pase; E.C. 3.1.3.29) catalyzes the dephosphorylation of N-acylneuraminate 9-phosphate during the synthesis of N-acetylneuraminate; Escherichia coli nucleotide phosphatase YjjG has a broad pyrimidine nucleotide activity spectrum and functions as an in vivo house-cleaning phosphatase for noncanonical pyrimidine nucleotides. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>haloacid dehalogenase-like superfamily phosphoglycolate phosphatase, similar to Rhodobacter sphaeroides CbbZ. Phosphoglycolate phosphatase catalyzes the dephosphorylation of phosphoglycolate; its activity requires divalent cations, especially Mg++.  This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>uncharacterized subfamily of beta-phosphoglucomutase-like family, similar to uncharacterized Bacillus subtilis YhcW. This subfamily includes the uncharacterized Bacillus subtilis YhcW. It belongs to the beta-phosphoglucomutase-like family whose other members include Lactococcus lactis beta-PGM, a mutase which catalyzes the interconversion of beta-D-glucose 1-phosphate (G1P) and D-glucose 6-phosphate (G6P), Saccharomyces cerevisiae phosphatases GPP1 and GPP2 that dephosphorylate DL-glycerol-3-phosphate and DOG1 and DOG2 that dephosphorylate 2-deoxyglucose-6-phosphate, and Escherichia coli 6-phosphogluconate phosphatase YieH. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>L-2-haloacid dehalogenase. L-2-Haloacid dehalogenase catalyzes the hydrolytic dehalogenation of L-2-haloacids to produce the corresponding D-2-hydroxyacids with an inversion of the C2-configuration. 2-haloacid dehalogenases are of interest for their potential to degrade recalcitrant halogenated environmental pollutants and their use in the synthesis of industrial chemicals. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>HAD-hyrolase-like. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>2-phosphoglycolate phosphatase, prokaryotic. PGP is an essential enzyme in the glycolate salvage pathway in higher organisms (photorespiration in plants). Phosphoglycolate results from the oxidase activity of RubisCO in the Calvin cycle when concentrations of carbon dioxide are low relative to oxygen. In Ralstonia (Alcaligenes) eutropha and Rhodobacter sphaeroides, the PGP gene (CbbZ) is located on an operon along with other Calvin cycle enzymes including RubisCO. The only other pertinent experimental evidence concerns the gene from E. coli. The in vitro activity of the Ralstonia and Escherichia enzymes was determined with crude cell extracts of strains containing PGP on expression plasmids and compared to controls. In E. coli, however, there does not appear to be a functional Calvin cycle (RubisCO is absent), although the E. coli PGP gene (gph) is on the same operon (dam) with ribulose-5-phosphate-3-epimerase (rpe), a gene in the pentose-phosphate pathway (along with other, unrelated genes). The E. coli enzyme is not expressed under normal laboratory conditions; the pathway to which it belongs has not been determined. In fact, the possibility exists, although unlikely, that the E. coli enzyme and others within this equivalog have as their physiological substrate another, closely related molecule. The other seed chosen for this model, from Xylella fastidiosa has no experimental evidence, but is a plant pathogen and thus may obtain phosphoglycolate from its host. This model has been restricted to encompass only proteobacteria as no related PGP has been verified outside of this clade. Sequences from Aquifex aeolicus and Treponema pallidum fall between the trusted and noise cutoffs. Just below the noise cutoff is a gene which is part of the operon for the biosynthesis of the blue pigment, indigoidine, from Erwinia (Pectobacterium) chrysanthemi, a plant pathogen. It does not seem likely, considering the proposed biosynthetic mechanism, that the dephosphorylation of phosphoglycolate or a closely related compound is required. Possibly, this gene is fortuitously located in this operon, or has an indirect relationship to the necessity for the biosynthesis of this compound. Sequences from 11 species have been annotated as PGP or putative PGP but fall below the noise cutoff. None of these have experimental validation. This enzyme is a member of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolase enzymes (pfam00702). [Energy metabolism, Sugars]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>beta-phosphoglucomutase-like family of the haloacid dehalogenase-like (HAD) hydrolase superfamily. This family represents the beta-phosphoglucomutase-like family of the haloacid dehalogenase-like (HAD) hydrolase superfamily. Family members include Lactococcus lactis beta-PGM, a mutase which catalyzes the interconversion of beta-D-glucose 1-phosphate (G1P) and D-glucose 6-phosphate (G6P), Saccharomyces cerevisiae phosphatases GPP1 and GPP2 that dephosphorylate DL-glycerol-3-phosphate and DOG1 and DOG2 that dephosphorylate 2-deoxyglucose-6-phosphate, and Escherichia coli 6-phosphogluconate phosphatase YieH. It belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>Rhodobacter capsulatus Cbbz phosphoglycolate phosphatase and related proteins; ; belongs to the haloacid dehalogenase-like superfamily. Phosphoglycolate phosphatase (PGPase; EC 3.1.3.18) catalyzes the conversion of 2-phosphoglycolate into glycolate and phosphate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>510999</td></tr>
<tr><td id=al>end</td><td id=al>511734</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[Calvin-Benson-Bassham|phosphoglycolate phosphatase]</td></tr>
<tr><td id=al>descr</td><td id=al>HAD family hydrolase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
