<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04389</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04389 HAD family hydrolase [Phormidium_A]</h3>
    <div>MGVDEPLVLATDLDGTFLGGSVEQRSRFYQYLLQHRQRLVLIFVTGRDLELVRQLYREPEVPQPDYIIADVGTTVTCGQSFTPIQGVQDWIAQTWDNSTERVRSLLEGEPGLELQPITPQYRVSYYYTPNKLQSDTLTRITTAGFDYVLSADRFLDVMPKGIAKGATLLKTLESLNLSPEQTVVAGDTLNDRSLLETGLNGIVVGNAEPKLKDVARSLPNAYISPEPGAWGIWDGLKHHGRAF</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>94</td>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=al>HAD family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>HAD superfamily hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>alpha,alpha-trehalose-phosphate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=al>alpha,alpha-trehalose-phosphate synthase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>97</td>
            <td id=cg>97</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>TPA: HAD-IIB family hydrolase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=co>Microcoleus</td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>HAD-IIB family hydrolase</td>
            <td id=co>Lusitaniella</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pfam05116 Sucrose-6F-phosphate phosphohydrolase. This family consists of Sucrose-6F-phosphate phosphohydrolase proteins found in plants and cyanobacteria. Sucrose-6(F)-phosphate phosphohydrolase catalyzes the final step in the pathway of sucrose biosynthesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd02605 sucrose-phosphatase, similar to Synechocystis sp PCC 6803 SPP. Sucrose-phosphatase (SPP; EC 3.1.3.24) catalyzes the dephosphorylation of sucrose-6(F)-phosphate (Suc6P)-the final step in the pathway of sucrose biosynthesis in plants and cyanobacteria. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=al>TIGR02471 sucrose-phosphate synthase, sucrose phosphatase-like domain, bacterial. Sucrose phosphate synthase (SPS) and sucrose phosphate phosphatase (SPP) are the last two enzymes of sucrose biosynthesis. In cyanobacteria and plants, the C-terminal region of most or all versions of SPS has a domain homologous to the known SPP. This domain may serve a binding or regulatory rather than catalytic function. Sequences in this family are bacterial C-terminal regions found in all but two of the putative bacterial sucrose phosphate synthases described by TIGR02472.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>89</td>
            <td id=cg>87</td>
            <td id=al>TIGR01485 sucrose-6F-phosphate phosphohydrolase. This model describes the sucrose phosphate phosphohydrolase from plants and cyanobacteria (SPP). SPP is a member of the Class IIB subfamily (TIGR01484) of the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. SPP catalyzes the final step in the biosynthesis of sucrose, a critically important molecule for plants. Sucrose phosphate synthase (SPS), the prior step in the biosynthesis of sucrose, contains a domain which exhibits considerable similarity to SPP albeit without conservation of the catalytic residues. The catalytic machinery of the synthase resides in another domain. It seems likely that the phosphatase-like domain is involved in substrate binding, possibly binding both substrates in a "product-like" orientation prior to ligation by the synthase catalytic domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>PLN02382 probable sucrose-phosphatase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG0561 Hydroxymethylpyrimidine pyrophosphatase and other HAD family phosphatases [Coenzyme transport and metabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>99</td>
            <td id=cg>95</td>
            <td id=al>pfam08282 haloacid dehalogenase-like hydrolase. This family contains haloacid dehalogenase-like hydrolase enzymes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>29</td>
            <td id=cr>28</td>
            <td id=al>cd07516 phosphatase, similar to Escherichia coli Cof and Thermotoga maritima TM0651; belongs to the haloacid dehalogenase-like superfamily. Escherichia coli Cof is involved in the hydrolysis of HMP-PP (4-amino-2-methyl-5-hydroxymethylpyrimidine pyrophosphate, an intermediate in thiamin biosynthesis), Cof also has phosphatase activity against the coenzymes pyridoxal phosphate (PLP) and FMN. Thermotoga maritima TM0651 acts as a phosphatase with a phosphorylated carbohydrate molecule as a possible substrate. Escherichia coli YbhA is also a member of this family and catalyzes the dephosphorylation of PLP, YbhA can also hydrolyze erythrose-4-phosphate and fructose-1,6-bis-phosphate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>cd07518 Escherichia coli YbiV sugar phosphatase/phosphotransferase and related proteins; belongs to the haloacid dehalogenase-like superfamily. Escherichia coli YbiV can act as both a sugar phosphatase and as a phosphotransferase. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>35</td>
            <td id=co>40</td>
            <td id=al>cd07517 phosphatase, similar to Bacteroides thetaiotaomicron VPI-5482 BT4131 hexose phosphate phosphatase; belongs to the haloacid dehalogenase-like superfamily. Bacteroides thetaiotaomicron VPI-5482 BT4131 is a phosphatase with preference for hexose phosphates. In addition this family includes uncharacterized Bacillus subtilis YkrA, a putative phosphatase and uncharacterized Streptococcus pyogenes MGAS10394 a putative bifunctional phosphatase/peptidyl-prolyl cis-trans isomerase. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>TIGR01484 HAD-superfamily hydrolase, subfamily IIB. This subfamily falls within the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The Class II subfamilies are characterized by a domain that is located between the second and third conserved catalytic motifs of the superfamily domain. The IIB subfamily is distinguished from the IIA subfamily (TIGR01460) by homology and the predicted secondary structure of this domain by PSI-PRED. The IIB subfamily's Class II domain has the following predicted structure: Helix-Sheet-Sheet-(Helix or Sheet)-Helix-Sheet-(variable)-Helix-Sheet-Sheet. The IIB subfamily consists of Trehalose-6-phosphatase (TIGR00685), plant and cyanobacterial Sucrose-phosphatase and a closely related group of bacterial and archaeal sequences, eukaryotic phosphomannomutase (pfam03332), a large subfamily ("Cof-like hydrolases", TIGR00099) containing many closely related bacterial sequences, a hypothetical equivalog containing the E. coli YedP protein, as well as two small clusters containing OMNI|TC0379 and OMNI|SA2196 whose relationship to the other groups is unclear. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>31</td>
            <td id=co>56</td>
            <td id=al>cd07514 phosphatase, similar to Thermoplasma acidophilum TA0175 phosphoglycolate phosphatase (PCPase), and Pyrococcus horikoshii PH1421, a magnesium-dependent phosphatase; belongs to the haloacid dehalogenase-like superfamily. Thermoplasma acidophilum TA0175 phosphoglycolate phosphatase (PGPase) catalyzes the magnesium-dependent dephosphorylation of phosphoglycolate. This family also includes Pyrococcus horikoshii OT3 PH1421, a magnesium-dependent phosphatase. This family belongs to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>33</td>
            <td id=cr>35</td>
            <td id=al>PRK01158 phosphoglycolate phosphatase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3555704</td></tr>
<tr><td id=al>end</td><td id=al>3556436</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>HAD family hydrolase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>244</td></tr>
</table>

    </div>
</body>
</html>
