<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02520</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02520 peptide chain release factor N(5)-glutamine methyltransferase [Phormidium_A]</h3>
    <div>MEQMSQPPRCQRRHPAQGARVEVTVAQVRAWRNQARSGALAEDLDPAEVDWLLYEGARLERLKFRLGQPGDRLSLSRSLEDLTQLWQKRLGDRTPVQYLLGSVPWRNFNLLVSPAVLIPRPETEAMIDIVLETKGADSAGIWVDLGTGSGAIALGLAQALCHAKIHAVDCSAEALKLARSNAERHGLGDRIQFHQGSWFDPLAAQPGQLAGMLSNPPYIPSSEIPHLQPEVSHHEPHLALDGGDDGLTALRHLVQTAPQYLQPGGLWLVEIMSGQSPAVCQLLQDNGNYQAPQVIHDFAGHDRFVLTQTNQR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>release factor glutamine methyltransferase HemK</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=co>Spirulina_A</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cr>FACHB-8</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td>
            <td id=cr>O-77</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N5-glutamine S-adenosyl-L-methionine-dependent methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Methylase of polypeptide chain release factors [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>protein-(glutamine-N5) methyltransferase, release factor-specific. Members of this protein family are HemK (PrmC), a protein once thought to be involved in heme biosynthesis but now recognized to be a protein-glutamine methyltransferase that modifies the peptide chain release factors. All members of the seed alignment are encoded next to the release factor 1 gene (prfA) and confirmed by phylogenetic analysis. SIMBAL analysis (manuscript in prep.) shows the motif [LIV]PRx[DE]TE (in Escherichia coli, IPRPDTE) confers specificity for the release factors rather than for ribosomal protein L3. [Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>HemK family putative methylases. The gene hemK from E. coli was found to contribute to heme biosynthesis and originally suggested to be protoporphyrinogen oxidase. Functional analysis of the nearest homolog in Saccharomyces cerevisiae, YNL063w, finds it is not protoporphyrinogen oxidase and sequence analysis suggests that HemK homologs have S-adenosyl-methionine-dependent methyltransferase activity (Medline 99237242). Homologs are found, usually in a single copy, in nearly all completed genomes, but varying somewhat in apparent domain architecture. Both E. coli and H. influenzae have two members rather than one. The members from the Mycoplasmas have an additional C-terminal domain. [Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=co>49</td>
            <td id=al>bifunctional N5-glutamine S-adenosyl-L-methionine-dependent methyltransferase/tRNA (m7G46) methyltransferase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>50S ribosomal protein L3 N(5)-glutamine methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>unknown domain/N5-glutamine S-adenosyl-L-methionine-dependent methyltransferase fusion protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>Methyltransferase small domain. This domain is found in ribosomal RNA small subunit methyltransferase C as well as other methyltransferases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>tRNA1(Val) A37 N6-methylase TrmN6 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>putative methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>HemK-related putative methylase. The gene hemK from E. coli was found to contribute to heme biosynthesis and originally suggested to be protoporphyrinogen oxidase. Functional analysis of the nearest homolog in Saccharomyces cerevisiae, YNL063w, finds it is not protoporphyrinogen oxidase and sequence analysis suggests that HemK homologs have S-adenosyl-methionine-dependent methyltransferase activity (Medline 99237242). Homologs are found, usually in a single copy, in nearly all completed genomes, but varying somewhat in apparent domain architecture. This model represents an archaeal and eukaryotic protein family that lacks an N-terminal domain found in HemK and its eubacterial homologs. It is found in a single copy in the first six completed archaeal and eukaryotic genomes. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>putative methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>Methyltransferase domain. This family appears to be a methyltransferase domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>S-adenosylmethionine-dependent methyltransferases (SAM or AdoMet-MTase), class I;  AdoMet-MTases are enzymes that use S-adenosyl-L-methionine (SAM or AdoMet) as a substrate for methyltransfer, creating the product S-adenosyl-L-homocysteine (AdoHcy). There are at least five structurally distinct families of AdoMet-MTases, class I being the largest and most diverse. Within this class enzymes can be classified by different substrate specificities (small molecules, lipids, nucleic acids, etc.) and different target atoms for methylation (nitrogen, oxygen, carbon, sulfur, etc.).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>39</td>
            <td id=al>Methyltransferase domain. This family appears to have methyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>16S rRNA G1207 methylase RsmC [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2112609</td></tr>
<tr><td id=al>end</td><td id=al>2113548</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>peptide chain release factor N(5)-glutamine methyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
