<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03586</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03586 DNA phosphorothioation system restriction enzyme [Phormidium_A]</h3>
    <div>MPLDLASVLPSGRPRLPSDLTLRPYQEQAIANWFAASGRGTLKMATGSGKTITALAIASRLHQAIGLQVLLIICPYRHLVTQWDRECRRFDLDPILAFESIHHWQGPLIRSLTAVRSQSLPFLTIITTNATLMRDRFQWQLKYLPAKTLIIGDEVHNLGAPRLETSLPRHIGLRLGLSATPERHFDDEGTDAVLGYFGNVLQPELSLGDAIKAGALVRYQYYPLFVPLTSWESEQYAKLTHRIGWALGDDSRDPDELTALLVKRSRLIGTASRKLDILRWMMRSRLNTSHTLFYCSDGCDSETNTPQIEAVVRLLGRELGFRINTYTAETSINEREQLRQQFEAGELQGLVAIRCLDEGVDIPAIRTAAILASSSNPRQFVQRRGRILRPHPSKDHATLFDTIVLPPDLDRETWEVERNLLKKELKRFLEFAELADNAEEAQQQLRGLQAKYGIGDD</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA or RNA helicases of superfamily II</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=co>Oxynema</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>TPA: DNA phosphorothioation system restriction enzyme</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>DNA phosphorothioation system restriction enzyme</td>
            <td id=co>LEGE-06147</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA phosphorothioation system restriction enzyme. The DNA phosphorothioate modification system dnd (DNA instability during electrophoresis) recently has been shown to provide a modification essential to a restriction system. This protein family was detected by Partial Phylogenetic Profiling as linked to dnd, and its members usually are clustered with the dndABCDE genes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Superfamily II DNA or RNA helicase [Transcription, Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DEXH-box helicase domain of DEAD-like helicase restriction enzyme family proteins. This family is composed of helicase restriction enzymes and similar proteins such as TFIIH basal transcription factor complex helicase XPB subunit. These proteins are part of the DEAD-like helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Type III restriction enzyme, res subunit. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>DEAD-like helicases superfamily. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>C-terminal helicase domain of XPB-like helicases. TFIIH basal transcription factor complex helicase XPB (xeroderma pigmentosum type B) subunit (also known as DNA excision repair protein ERCC-3 or TFIIH 89 kDa subunit) is the ATP-dependent 3'-5' DNA helicase component of the core-TFIIH basal transcription factor, involved in nucleotide excision repair (NER) of DNA and, when complexed to CAK, in RNA transcription by RNA polymerase II. XPB is a DEAD-like helicase belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>helicase superfamily c-terminal domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Helicase conserved C-terminal domain. The Prosite family is restricted to DEAD/H helicases, whereas this domain family is found in a wide variety of helicases and helicase related proteins. It may be that this is not an autonomously folding unit, but an integral part of the helicase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DEXH-box helicase domain of type III restriction enzyme res subunit. Members of this cd includes both type I and type III restriction enzymes. Both are hetero-oligomeric proteins. Type I REs are encoded by three closely linked genes: a specificity subunit (HsdS or S) for recognizing a DNA sequence, a methylation subunit (HsdM or M) for methylating the recognized target bases, and a restriction subunit (HsdR or R) for the translocation and random cleavage of non-methylated DNA. They show diverse catalytic activities, including methyltransferase (MTase), ATP hydrolase (ATPase), DNA translocation and restriction activities. These enzymes cut at a site that differs, and is a random distance (at least 1000 bp) away, from their recognition site. Cleavage at these random sites follows a process of DNA translocation, which shows that these enzymes are also molecular motors. The recognition site is asymmetrical and is composed of two specific portions: one containing 3-4 nucleotides, and another containing 4-5 nucleotides, separated by a non-specific spacer of about 6-8 nucleotides. Type III enzymes are composed of two subunits, Res and Mod. The Mod subunit recognizes the DNA sequence specific for the system and is a modification methyltransferase; as such, it is functionally equivalent to the M and S subunits of type I restriction endonucleases. Res is required for restriction, although it has no enzymatic activity on its own. Type III enzymes recognize short 5-6 bp-long asymmetric DNA sequences and cleave 25-27 bp downstream to leave short, single-stranded 5' protrusions. They require the presence of two inversely oriented unmethylated recognition sites for restriction to occur. These enzymes methylate only one strand of the DNA, at the N-6 position of adenosyl residues, so newly replicated DNA will have only one strand methylated, which is sufficient to protect against restriction. Both type I and type III REs are members of the DEAD-like helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>C-terminal helicase domain of EcoAI HsdR-like restriction enzyme family helicases. This family is composed of helicase restriction enzymes, including the HsdR subunit of restriction-modification enzymes such as Escherichia coli type I restriction enzyme EcoAI R protein (R.EcoAI). The EcoAI enzyme recognizes 5'-GAGN(7)GTCA-3'. The HsdR or R subunit is required for both nuclease and ATPase activities, but not for modification. These proteins are DEAD-like helicases belonging to superfamily (SF)2, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Similar to SF1 helicases, SF2 helicases do not form toroidal structures like SF3-6 helicases. Their helicase core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>C-terminal helicase domain of superfamily 2 DEAD/H-box helicases. Superfamily (SF)2 helicases include DEAD-box helicases, UvrB, RecG, Ski2, Sucrose Non-Fermenting (SNF) family helicases, and dicer proteins, among others. Similar to SF1 helicases, they do not form toroidal structures like SF3-6 helicases. SF2 helicases are a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. Their helicase core is surrounded by C- and N-terminal domains with specific functions such as nucleases, RNA or DNA binding domains, or domains engaged in protein-protein interactions. The core consists of two similar protein domains that resemble the fold of the recombination protein RecA. This model describes the C-terminal domain, also called HelicC.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-terminal DEAD/H-box helicase domain of superfamily 2 helicases. The DEAD/H-like superfamily 2 helicases comprise a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This N-terminal domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DEXH-box helicase domain of TFIIH XPB subunit and similar proteins. TFIIH basal transcription factor complex helicase XPB subunit (also known as DNA excision repair protein ERCC-3 or TFIIH 89 kDa subunit) is the ATP-dependent 3'-5' DNA helicase component of the core-TFIIH basal transcription factor, involved in nucleotide excision repair (NER) of DNA and, when complexed to CAK, in RNA transcription by RNA polymerase II. XPB is a member of the DEAD-like helicase superfamily, a diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>DEXH/Q-box helicase domain of DEAD-like helicase Snf family proteins. Sucrose Non-Fermenting (SNF) proteins DEAD-like helicases superfamily. A diverse family of proteins involved in ATP-dependent RNA or DNA unwinding. This domain contains the ATP-binding region.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3037581</td></tr>
<tr><td id=al>end</td><td id=al>3038955</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA phosphorothioation system restriction enzyme</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
