<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02783</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02783 elongation factor Tu [Phormidium_A]</h3>
    <div>MARAKFERNKPHVNIGTIGHVDHGKTTLTAAITATLAVTGQAKARKYADIDAAPEEKERGITINTAHVEYETTQRHYAHVDCPGHADYVKNMITGAAQMDGGILVVSAADGPMPQTREHILLARQVGVPSLVVFLNKQDQVDDEELLELVELEVRELLSEYDFPGDDIPIVTGSALKAIEALEANPTVAAGEDEWVDKILKLMEEVDAYVPTPERDVDKDFLMAIEDVFSITGRGTVATGRIERGIVKVGDTVELVGIKDTRSTTVTGVEMFQKTLDQGMAGDNVGVLLRGVQKGDIERGMVLAKPGTINPHTKFEAEVYILNKNEGGRHTPFFPGYRPQFYVRTTDVTGTISDFTADDGSEAEMVMPGDRIKMTVELINPVAIEQGMRFAIREGGRTVGAGAVAKILQ</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tuf</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cr>Acaryochloris</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>elongation factor Tu</td>
            <td id=cr>Romeria</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>elongation factor Tu</td>
            <td id=co>Oxynema</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>elongation factor Tu</td>
            <td id=cr>JAAUTN01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>83</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>CHL00071 elongation factor Tu</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>PRK00049 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>PRK12735 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>PRK12736 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>TIGR00485 translation elongation factor TU. This model models orthologs of translation elongation factor EF-Tu in bacteria, mitochondria, and chloroplasts, one of several GTP-binding translation factors found by the more general pfam model GTP_EFTU. The eukaryotic conterpart, eukaryotic translation elongation factor 1 (eEF-1 alpha), is excluded from this model. EF-Tu is one of the most abundant proteins in bacteria, as well as one of the most highly conserved, and in a number of species the gene is duplicated with identical function. When bound to GTP, EF-Tu can form a complex with any (correctly) aminoacylated tRNA except those for initiation and for selenocysteine, in which case EF-Tu is replaced by other factors. Transfer RNA is carried to the ribosome in these complexes for protein translation. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>COG0050 Translation elongation factor EF-Tu, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>99</td>
            <td id=cg>85</td>
            <td id=al>PLN03126 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>PLN03127 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=al>cd01884 Elongation Factor Tu (EF-Tu) GTP-binding proteins. EF-Tu subfamily. This subfamily includes orthologs of translation elongation factor EF-Tu in bacteria, mitochondria, and chloroplasts. It is one of several GTP-binding translation factors found in the larger family of GTP-binding elongation factors. The eukaryotic counterpart, eukaryotic translation elongation factor 1 (eEF-1 alpha), is excluded from this family. EF-Tu is one of the most abundant proteins in bacteria, as well as, one of the most highly conserved, and in a number of species the gene is duplicated with identical function. When bound to GTP, EF-Tu can form a complex with any (correctly) aminoacylated tRNA except those for initiation and for selenocysteine, in which case EF-Tu is replaced by other factors. Transfer RNA is carried to the ribosome in these complexes for protein translation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG5256 Translation elongation factor EF-1alpha (GTPase)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK12317 elongation factor 1-alpha; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>pfam00009 Elongation factor Tu GTP binding domain. This domain contains a P-loop motif, also found in several other families such as pfam00071, pfam00025 and pfam00063. Elongation factor Tu consists of three structural domains, this plus two C-terminal beta barrel domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=al>cd03707 Domain III of Elongation Factor (EF) Tu. EF-Tu consists of three structural domains, designated I, II, and III. Domain III adopts a beta barrel structure. Domain III is involved in binding to both charged tRNA and to elongation factor Ts (EF-Ts). EF-Ts is the guanine-nucleotide-exchange factor for EF-Tu. EF-Tu and EF-G participate in the elongation phase during protein biosynthesis on the ribosome. Their functional cycles depend on GTP binding and its hydrolysis. The EF-Tu complexed with GTP and aminoacyl-tRNA delivers tRNA to the ribosome, whereas EF-G stimulates translocation, a process in which tRNA and mRNA movements occur in the ribosome. Crystallographic studies revealed structural similarities ("molecular mimicry") between tertiary structures of EF-G and the EF-Tu-aminoacyl-tRNA ternary complex. Domains III, IV, and V of EF-G mimic the tRNA structure in the EF-Tu ternary complex; domains III, IV and V can be related to the acceptor stem, anticodon helix and T stem of tRNA respectively.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>cd03697 Domain II of elongation factor Tu. Elongation factors Tu (EF-Tu) are three-domain GTPases with an essential function in the elongation phase of mRNA translation. The GTPase center of EF-Tu is in the N-terminal domain (domain I), also known as the catalytic or G-domain. The G-domain is composed of about 200 amino acid residues, arranged into a predominantly parallel six-stranded beta-sheet core surrounded by seven alpha helices. Non-catalytic domains II and III are beta-barrels of seven and six, respectively, antiparallel beta-strands that share an extended interface. Both non-catalytic domains are composed of about 100 amino acid residues. EF-Tu proteins exist in two principal conformations: a compact one, EF-Tu*GTP, with tight interfaces between all three domains and a high affinity for aminoacyl-tRNA; and an open one, EF-Tu*GDP, with essentially no G-domain-domain II interactions and a low affinity for aminoacyl-tRNA. EF-Tu has approximately a 100-fold higher affinity for GDP than for GTP.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>92</td>
            <td id=cg>84</td>
            <td id=al>COG3276 Selenocysteine-specific translation elongation factor [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=al>pfam03143 Elongation factor Tu C-terminal domain. Elongation factor Tu consists of three structural domains, this is the third domain. This domain adopts a beta barrel structure. This the third domain is involved in binding to both charged tRNA and binding to EF-Ts pfam00889.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PTZ00141 elongation factor 1- alpha; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>COG5258 GTPase  [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>95</td>
            <td id=cb>67</td>
            <td id=al>TIGR00475 selenocysteine-specific elongation factor SelB. In prokaryotes, the incorporation of selenocysteine as the 21st amino acid, encoded by TGA, requires several elements: SelC is the tRNA itself, SelD acts as a donor of reduced selenium, SelA modifies a serine residue on SelC into selenocysteine, and SelB is a selenocysteine-specific translation elongation factor. 3-prime or 5-prime non-coding elements of mRNA have been found as probable structures for directing selenocysteine incorporation. This model describes the elongation factor SelB, a close homolog rf EF-Tu. It may function by replacing EF-Tu. A C-terminal domain not found in EF-Tu is in all SelB sequences in the seed alignment except that from Methanococcus jannaschii. This model does not find an equivalent protein for eukaryotes. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=co>41</td>
            <td id=cg>93</td>
            <td id=al>cd00881 GTP translation factor family primarily contains translation initiation, elongation and release factors. The GTP translation factor family consists primarily of translation initiation, elongation, and release factors, which play specific roles in protein translation. In addition, the family includes Snu114p, a component of the U5 small nuclear riboprotein particle which is a component of the spliceosome and is involved in excision of introns, TetM, a tetracycline resistance gene that protects the ribosome from tetracycline binding, and the unusual subfamily CysN/ATPS, which has an unrelated function (ATP sulfurylase) acquired through lateral transfer of the EF1-alpha gene and development of a new function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TIGR03680 translation initiation factor 2 subunit gamma. This model represents the archaeal translation initiation factor 2 subunit gamma and is found in all known archaea. eIF-2 functions in the early steps of protein synthesis by forming a ternary complex with GTP and initiator tRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>PLN00043 elongation factor 1-alpha; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG5257 Translation initiation factor 2, gamma subunit (eIF-2gamma; GTPase)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>39</td>
            <td id=cb>73</td>
            <td id=al>cd01883 Elongation Factor 1-alpha (EF1-alpha) protein family. EF1 is responsible for the GTP-dependent binding of aminoacyl-tRNAs to the ribosomes. EF1 is composed of four subunits: the alpha chain which binds GTP and aminoacyl-tRNAs, the gamma chain that probably plays a role in anchoring the complex to other cellular components and the beta and delta (or beta') chains. This subfamily is the alpha subunit, and represents the counterpart of bacterial EF-Tu for the archaea (aEF1-alpha) and eukaryotes (eEF1-alpha). eEF1-alpha interacts with the actin of the eukaryotic cytoskeleton and may thereby play a role in cellular transformation and apoptosis. EF-Tu can have no such role in bacteria. In humans, the isoform eEF1A2 is overexpressed in 2/3 of breast cancers and has been identified as a putative oncogene. This subfamily also includes Hbs1, a G protein known to be important for efficient growth and protein synthesis under conditions of limiting translation initiation in yeast, and to associate with Dom34. It has been speculated that yeast Hbs1 and Dom34 proteins may function as part of a complex with a role in gene expression.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK04000 translation initiation factor IF-2 subunit gamma; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>70</td>
            <td id=co>47</td>
            <td id=al>PRK10512 selenocysteinyl-tRNA-specific translation factor; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cr>31</td>
            <td id=cb>77</td>
            <td id=al>cd04171 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec, and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains bacterial SelBs, as well as, one from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>82</td>
            <td id=cb>78</td>
            <td id=al>COG2895 Sulfate adenylyltransferase subunit 1, EFTu-like GTPase family  [Inorganic ion transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=al>cd03706 Domain III of mitochondrial EF-TU (mtEF-TU). mtEF-TU is highly conserved and is 55-60% identical to bacterial EF-TU. The overall structure is similar to that observed in the Escherichia coli and Thermus aquaticus EF-TU. However, compared with that observed in prokaryotic EF-TU, the nucleotide-binding domain (domain I) of mtEF-TU is in a different orientation relative to the rest of the structure. Furthermore, domain III is followed by a short 11-amino acid extension that forms one helical turn. This extension seems to be specific to the mitochondrial factors and has not been observed in any of the prokaryotic factors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>70</td>
            <td id=co>48</td>
            <td id=al>TIGR01394 GTP-binding protein TypA/BipA. This bacterial (and Arabidopsis) protein, termed TypA or BipA, a GTP-binding protein, is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways, but the precise function is unknown. [Regulatory functions, Other, Cellular processes, Adaptations to atypical conditions, Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>74</td>
            <td id=co>50</td>
            <td id=al>COG1217 Predicted membrane GTPase involved in stress response [Signal transduction mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>34</td>
            <td id=cb>67</td>
            <td id=al>cd04166 CysN, together with protein CysD, forms the ATP sulfurylase (ATPS) complex. CysN_ATPS subfamily. CysN, together with protein CysD, form the ATP sulfurylase (ATPS) complex in some bacteria and lower eukaryotes. ATPS catalyzes the production of ATP sulfurylase (APS) and pyrophosphate (PPi) from ATP and sulfate. CysD, which catalyzes ATP hydrolysis, is a member of the ATP pyrophosphatase (ATP PPase) family. CysN hydrolysis of GTP is required for CysD hydrolysis of ATP; however, CysN hydrolysis of GTP is not dependent on CysD hydrolysis of ATP. CysN is an example of lateral gene transfer followed by acquisition of new function. In many organisms, an ATPS exists which is not GTP-dependent and shares no sequence or structural similarity to CysN.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>38</td>
            <td id=cg>80</td>
            <td id=al>cd01888 Gamma subunit of initiation factor 2 (eIF2 gamma). eIF2 is a heterotrimeric translation initiation factor that consists of alpha, beta, and gamma subunits. The GTP-bound gamma subunit also binds initiator methionyl-tRNA and delivers it to the 40S ribosomal subunit. Following hydrolysis of GTP to GDP, eIF2:GDP is released from the ribosome. The gamma subunit has no intrinsic GTPase activity, but is stimulated by the GTPase activating protein (GAP) eIF5, and GDP/GTP exchange is stimulated by the guanine nucleotide exchange factor (GEF) eIF2B. eIF2B is a heteropentamer, and the epsilon chain binds eIF2. Both eIF5 and eIF2B-epsilon are known to bind strongly to eIF2-beta, but have also been shown to bind directly to eIF2-gamma. It is possible that eIF2-beta serves simply as a high-affinity docking site for eIF5 and eIF2B-epsilon, or that eIF2-beta serves a regulatory role. eIF2-gamma is found only in eukaryotes and archaea. It is closely related to SelB, the selenocysteine-specific elongation factor from eubacteria. The translational factor components of the ternary complex, IF2 in eubacteria and eIF2 in eukaryotes are not the same protein (despite their unfortunately similar names). Both factors are GTPases; however, eubacterial IF-2 is a single polypeptide, while eIF2 is heterotrimeric. eIF2-gamma is a member of the same family as eubacterial IF2, but the two proteins are only distantly related. This family includes translation initiation, elongation, and release factors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>36</td>
            <td id=cb>78</td>
            <td id=al>cd01889 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains eukaryotic SelBs and some from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=al>cd03696 Domain II of elongation factor SelB. This subfamily represents the domain of elongation factor SelB that is homologous to domain II of EF-Tu. SelB may function by replacing EF-Tu. In prokaryotes, the incorporation of selenocysteine as the 21st amino acid, encoded by TGA, requires several elements: SelC is the tRNA itself, SelD acts as a donor of reduced selenium, SelA modifies a serine residue on SelC into selenocysteine, and SelB is a selenocysteine-specific translation elongation factor. 3' or 5' non-coding elements of mRNA have been found as probable structures for directing selenocysteine incorporation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>70</td>
            <td id=cb>62</td>
            <td id=al>PTZ00327 eukaryotic translation initiation factor 2 gamma subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cb>64</td>
            <td id=co>41</td>
            <td id=al>PRK05506 bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>21</td>
            <td id=cg>96</td>
            <td id=al>cd03693 Domain II of elongation factor 1-alpha. This family represents domain II of elongation factor 1-alpha (EF-1A) that is found in archaea and all eukaryotic lineages. EF-1A is very abundant in the cytosol, where it is involved in the GTP-dependent binding of aminoacyl-tRNAs to the A site of the ribosomes in the second step of translation from mRNAs to proteins. Both domain II of EF-1A and domain IV of IF2/eIF5B have been implicated in recognition of the 3'-ends of tRNA. More than 61% of eukaryotic elongation factor 1A (eEF-1A) in cells is estimated to be associated with actin cytoskeleton. The binding of eEF-1A to actin is a noncanonical function that may link two distinct cellular processes, cytoskeleton organization and gene expression.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>pfam03144 Elongation factor Tu domain 2. Elongation factor Tu consists of three structural domains, this is the second domain. This domain adopts a beta barrel structure. This the second domain is involved in binding to charged tRNA. This domain is also found in other proteins such as elongation factor G and translation initiation factor IF-2. This domain is structurally related to pfam03143, and in fact has weak sequence matches to this domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=al>cd01513 Domain III of Elongation factor (EF) Tu (EF-TU) and related proteins. Elongation factor (EF) EF-Tu participates in the elongation phase during protein biosynthesis on the ribosome. Its functional cycles depend on GTP binding and its hydrolysis. The EF-Tu complexed with GTP and aminoacyl-tRNA delivers tRNA to the ribosome, whereas EF-G stimulates translocation, a process in which tRNA and mRNA movements occur in the ribosome. Experimental findings indicate an essential contribution of domain III to activation of GTP hydrolysis. This domain III, which is distinct from the domain III in EFG and related elongation factors, is found in several eukaryotic translation factors, like peptide chain release factors RF3, elongation factor 1, selenocysteine (Sec)-specific elongation factor, and in GT-1 family of GTPase (GTPBP1).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>62</td>
            <td id=co>43</td>
            <td id=al>TIGR00487 translation initiation factor IF-2. This model discriminates eubacterial (and mitochondrial) translation initiation factor 2 (IF-2), encoded by the infB gene in bacteria, from similar proteins in the Archaea and Eukaryotes. In the bacteria and in organelles, the initiator tRNA is charged with N-formyl-Met instead of Met. This translation factor acts in delivering the initator tRNA to the ribosome. It is one of a number of GTP-binding translation factors recognized by the pfam model GTP_EFTU. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cb>66</td>
            <td id=co>57</td>
            <td id=al>PRK05124 sulfate adenylyltransferase subunit 1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cb>62</td>
            <td id=co>42</td>
            <td id=al>PRK10218 translational GTPase TypA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=al>cd01342 Domain II of Elongation factor Tu (EF-Tu)-like proteins. Elongation factor Tu consists of three structural domains. Domain II adopts a beta barrel structure and is involved in binding to charged tRNA. Domain II is found in other proteins such as elongation factor G and translation initiation factor IF-2. This group also includes the C2 subdomain of domain IV of IF-2 that has the same fold as domain II of (EF-Tu). Like IF-2 from certain prokaryotes such as Thermus thermophilus, mitochondrial IF-2 lacks domain II, which is thought  to be involved in binding of E. coli IF-2 to 30S subunits.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>34</td>
            <td id=cb>72</td>
            <td id=al>cd01891 Tyrosine phosphorylated protein A (TypA)/BipA family belongs to ribosome-binding GTPases. BipA is a protein belonging to the ribosome-binding family of GTPases and is widely distributed in bacteria and plants. BipA was originally described as a protein that is induced in Salmonella typhimurium after exposure to bactericidal/permeability-inducing protein (a cationic antimicrobial protein produced by neutrophils), and has since been identified in E. coli as well. The properties thus far described for BipA are related to its role in the process of pathogenesis by enteropathogenic E. coli. It appears to be involved in the regulation of several processes important for infection, including rearrangements of the cytoskeleton of the host, bacterial resistance to host defense peptides, flagellum-mediated cell motility, and expression of K5 capsular genes. It has been proposed that BipA may utilize a novel mechanism to regulate the expression of target genes. In addition, BipA from enteropathogenic E. coli has been shown to be phosphorylated on a tyrosine residue, while BipA from Salmonella and from E. coli K12 strains is not phosphorylated under the conditions assayed. The phosphorylation apparently modifies the rate of nucleotide hydrolysis, with the phosphorylated form showing greatly increased GTPase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>31</td>
            <td id=cr>18</td>
            <td id=al>COG0480 Translation elongation factor EF-G, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>32</td>
            <td id=cr>18</td>
            <td id=al>TIGR00490 translation elongation factor aEF-2. This model represents archaeal elongation factor 2, a protein more similar to eukaryotic EF-2 than to bacterial EF-G, both in sequence similarity and in sharing with eukaryotes the property of having a diphthamide (modified His) residue at a conserved position. The diphthamide can be ADP-ribosylated by diphtheria toxin in the presence of NAD. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>33</td>
            <td id=cb>63</td>
            <td id=al>cd01885 Elongation Factor 2 (EF2) in archaea and eukarya. Translocation requires hydrolysis of a molecule of GTP and is mediated by EF-G in bacteria and by eEF2 in eukaryotes. The eukaryotic elongation factor eEF2 is a GTPase involved in the translocation of the peptidyl-tRNA from the A site to the P site on the ribosome. The 95-kDa protein is highly conserved, with 60% amino acid sequence identity between the human and yeast proteins. Two major mechanisms are known to regulate protein elongation and both involve eEF2. First, eEF2 can be modulated by reversible phosphorylation. Increased levels of phosphorylated eEF2 reduce elongation rates presumably because phosphorylated eEF2 fails to bind the ribosomes. Treatment of mammalian cells with agents that raise the cytoplasmic Ca2+ and cAMP levels reduce elongation rates by activating the kinase responsible for phosphorylating eEF2. In contrast, treatment of cells with insulin increases elongation rates by promoting eEF2 dephosphorylation. Second, the protein can be post-translationally modified by ADP-ribosylation. Various bacterial toxins perform this reaction after modification of a specific histidine residue to diphthamide, but there is evidence for endogenous ADP ribosylase activity. Similar to the bacterial toxins, it is presumed that modification by the endogenous enzyme also inhibits eEF2 activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>61</td>
            <td id=co>49</td>
            <td id=al>COG0532 Translation initiation factor IF-2, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>33</td>
            <td id=cr>19</td>
            <td id=al>PRK07560 elongation factor EF-2; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>22</td>
            <td id=cr>13</td>
            <td id=al>PRK12740 elongation factor G-like protein EF-G2. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>32</td>
            <td id=co>55</td>
            <td id=al>cd04168 Tet(M)-like family includes Tet(M), Tet(O), Tet(W), and OtrA, containing tetracycline resistant proteins. Tet(M), Tet(O), Tet(W), and OtrA are tetracycline resistance genes found in Gram-positive and Gram-negative bacteria. Tetracyclines inhibit protein synthesis by preventing aminoacyl-tRNA from binding to the ribosomal acceptor site. This subfamily contains tetracycline resistance proteins that function through ribosomal protection and are typically found on mobile genetic elements, such as transposons or plasmids, and are often conjugative. Ribosomal protection proteins are homologous to the elongation factors EF-Tu and EF-G. EF-G and Tet(M) compete for binding on the ribosomes. Tet(M) has a higher affinity than EF-G, suggesting these two proteins may have overlapping binding sites and that Tet(M) must be released before EF-G can bind. Tet(M) and Tet(O) have been shown to have ribosome-dependent GTPase activity. These proteins are part of the GTP translation factor family, which includes EF-G, EF-Tu, EF2, LepA, and SelB.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>22</td>
            <td id=cr>33</td>
            <td id=al>cd04170 Elongation factor G (EF-G) family. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains only bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=al>cd03694 Domain II of the GTPBP family of GTP binding proteins. This group includes proteins similar to GTPBP1 and GTPBP2. GTPBP1 is structurally related to elongation factor 1 alpha, a key component of the protein biosynthesis machinery. Immunohistochemical analyses on mouse tissues revealed that GTPBP1 is expressed in some neurons and smooth muscle cells of various organs as well as macrophages. Immunofluorescence analyses revealed that GTPBP1 is localized exclusively in cytoplasm and shows a diffuse granular network forming a gradient from the nucleus to the periphery of the cells in smooth muscle cell lines and macrophages. No significant difference was observed in the immune response to protein antigen between mutant mice and wild-type mice, suggesting normal function of antigen-presenting cells of the mutant mice. The absence of an eminent phenotype in GTPBP1-deficient mice may be due to functional compensation by GTPBP2, which is similar to GTPBP1 in structure and tissue distribution.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>30</td>
            <td id=cb>73</td>
            <td id=al>cd01887 Initiation Factor 2 (IF2)/ eukaryotic Initiation Factor 5B (eIF5B) family. IF2/eIF5B contribute to ribosomal subunit joining and function as GTPases that are maximally activated by the presence of both ribosomal subunits. As seen in other GTPases, IF2/IF5B undergoes conformational changes between its GTP- and GDP-bound states. Eukaryotic IF2/eIF5Bs possess three characteristic segments, including a divergent N-terminal region followed by conserved central and C-terminal segments. This core region is conserved among all known eukaryotic and archaeal IF2/eIF5Bs and eubacterial IF2s.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>33</td>
            <td id=cr>19</td>
            <td id=al>PRK13351 elongation factor G-like protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cb>61</td>
            <td id=cr>33</td>
            <td id=al>CHL00189 translation initiation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cr>17</td>
            <td id=cg>85</td>
            <td id=al>cd16267 Domain II of Hbs1-like proteins. S. cerevisiae Hbs1 is closely related to the eukaryotic class II release factor (eRF3). Hbs1, together with Dom34 (pelota), plays an important role in termination and recycling, but in contrast to eRF3/eRF1, Hbs1, together with Dom34 (pelota), functions on mRNA-bound ribosomes in a codon-independent manner and promotes subunit splitting on completely empty ribosomes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>18</td>
            <td id=cg>96</td>
            <td id=al>cd16265 Proteins related to domain II of EF-Tu and related translation factors. Elongation factor Tu consists of three structural domains; this family represents single domain proteins that are related to the second domain of EF-Tu. Domain II of EF-Tu adopts a beta barrel structure and is involved in binding to charged tRNA. Domain II is also found in other proteins such as elongation factor G and translation initiation factor IF-2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>17</td>
            <td id=cg>85</td>
            <td id=al>cd04089 Domain II of the eukaryotic class II release factor. In eukaryotes, translation termination is mediated by two interacting release factors, eRF1 and eRF3, which act as class I and II factors, respectively. eRF1 functions as an omnipotent release factor, decoding all three stop codons and triggering the release of the nascent peptide catalyzed by the ribosome. eRF3 is a GTPase, which enhances termination efficiency by stimulating eRF1 activity in a GTP-dependent manner. Sequence comparison of class II release factors with elongation factors shows that eRF3 is more similar to eEF-1alpha whereas prokaryote RF3 is more similar to EF-G, implying that their precise function may differ. Only eukaryote RF3s are found in this group. Saccharomyces cerevisiae eRF3 (Sup35p) is a translation termination factor which is divided into three regions N, M and a C-terminal eEF1a-like region essential for translation termination. Sup35NM  is a non-pathogenic prion-like protein with the property of aggregating into polymer-like fibrils.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2322885</td></tr>
<tr><td id=al>end</td><td id=al>2324115</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>elongation factor Tu</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>410</td></tr>
</table>

    </div>
</body>
</html>
