<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04170</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04170 cysteine desulfurase NifS [Phormidium_A]</h3>
    <div>MSVIYLDNNATTCVAPEVYEAMSPYFTEFYGNPSSMHRFGGQVARGLRQSREAIAALIHAEPNEIIFTSCGTEGDNAAIRAALDSQPDKRHIITTAVEHPAILKPLQRLEKQGYTVTYLGVDGKGQLKLEELRRAIRDDTALISVMAANNETGTIFPIERIGHLAKERGILFHVDAVQAAGKIPLDVSSGLIDFLVLSGHKFHGPKGIGVLYVRRGVRFRPLLAGGHQERGRRGGTENVPGIIGLGKAAELTLNSFDEDHTRVSQLRDRLEQGILSTISDTQRNGHPGLRLPNTANIGFKYVEGEAILLSLDREGICASSGSACTSGSLQPSHVLRAMGLPYTLLHGSIRFSLSRYTTGAEIEQVLAVLQPMIERLRDLSPFSRDNEATHPWLQEQTQRVGIA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=co>Coleofasciculus</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=co>Roseofilum</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=co>Hydrococcus</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>cysteine desulfurase NifS</td>
            <td id=co>Hydrococcus</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase NifS. Members of this protein family are NifS, one of several related families of cysteine desulfurase involved in iron-sulfur (FeS) cluster biosynthesis. NifS is part of the NIF system, usually associated with other nif genes involved in nitrogenase expression and nitrogen fixation. The protein family is given a fairly broad interpretation here. It includes a clade nearly always found in extended nitrogen fixation genomic regions, plus a second clade more closely related to the first than to IscS and also part of NifS-like/NifU-like systems. This model does not extend to a more distantly clade found in the epsilon proteobacteria such as Helicobacter pylori, also named NifS in the literature, built instead in TIGR03403.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Cysteine sulfinate desulfinase/cysteine desulfurase or related enzyme [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>IscS subfamily cysteine desulfurase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>cysteine desulfurase, NifS family, epsilon proteobacteria type. Members of this family are the NifS-like cysteine desulfurase of the epsilon division of the Proteobacteria, similar to the NifS protein of nitrogen-fixing bacteria. Like NifS, and unlike IscS, this protein is found as part of a system of just two proteins, a cysteine desulfurase and a scaffold, for iron-sulfur cluster biosynthesis. This protein is called NifS by Olsen, et al. (), so we use this designation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase DndA. This model describes DndA, a protein related to IscS and part of a larger family of cysteine desulfurases. It is encoded, typically, divergently from a conserved, sparsely distributed operon for sulfur modification of DNA. This modification system is designated dnd, after the phenotype of DNA degradation during electrophoresis. The system is sporadically distributed in bacteria, much like some restriction enzyme operons. DndB is described as a putative ATPase. [DNA metabolism, Restriction/modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Aminotransferase class-V. This domain is found in amino transferases, and other enzymes including cysteine desulphurase EC:4.4.1.-.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>IscS subfamily cysteine desulfurase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Selenocysteine lyase/Cysteine desulfurase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Cysteine desulfurase (SufS)-like. This family belongs to the pyridoxal phosphate (PLP)-dependent aspartate aminotransferase superfamily (fold I). The major groups in this CD correspond to cysteine desulfurase (SufS) and selenocysteine lyase. SufS catalyzes the removal of elemental sulfur and selenium atoms from L-cysteine, L-cystine, L-selenocysteine, and L-selenocystine to produce L-alanine; and selenocysteine lyase catalyzes the decomposition of L-selenocysteine.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Bifunctional selenocysteine lyase/cysteine desulfurase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase SufS. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cysteine desulfurase CsdA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>Kynureninase  [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Aspartate aminotransferase (AAT) superfamily (fold type I) of pyridoxal phosphate (PLP)-dependent enzymes. PLP combines with an alpha-amino acid to form a compound called a Schiff base or aldimine intermediate, which depending on the reaction, is the substrate in four kinds of reactions (1) transamination (movement of amino groups), (2) racemization (redistribution of enantiomers), (3) decarboxylation (removing COOH groups), and (4) various side-chain reactions depending on the enzyme involved. Pyridoxal phosphate (PLP) dependent enzymes were previously classified into alpha, beta and gamma classes, based on the chemical characteristics (carbon atom involved) of the reaction they catalyzed. The availability of several structures allowed a comprehensive analysis of  the evolutionary classification of PLP dependent enzymes, and it was found that the functional classification did not always agree with the evolutionary history of these enzymes. Structure and sequence analysis has revealed that the PLP dependent enzymes can be classified into four major groups of different evolutionary origin: aspartate aminotransferase superfamily (fold type I), tryptophan synthase beta superfamily (fold type II), alanine racemase superfamily (fold type III), and D-amino acid superfamily (fold type IV) and Glycogen phophorylase family (fold type V).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=co>43</td>
            <td id=al>Glutamate or tyrosine decarboxylase or a related PLP-dependent protein [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>Archaeal aspartate aminotransferase or a related aminotransferase, includes purine catabolism protein PucG [Amino acid transport and metabolism, Nucleotide transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>DOPA decarboxylase family. This family belongs to pyridoxal phosphate (PLP)-dependent aspartate aminotransferase superfamily (fold I). The major groups in this CD correspond to DOPA/tyrosine decarboxylase (DDC), histidine decarboxylase (HDC), and glutamate decarboxylase (GDC). DDC is active as a dimer and catalyzes the decarboxylation of tyrosine. GDC catalyzes the decarboxylation of glutamate and HDC catalyzes the decarboxylation of histidine.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>tyrosine decarboxylase MfnA. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3403473</td></tr>
<tr><td id=al>end</td><td id=al>3404685</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>cysteine desulfurase NifS</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
