<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02759</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02759 gamma carbonic anhydrase family protein [Phormidium_A]</h3>
    <div>MTSSPNLTPHGSLPTYWPPVDCSQASFVAANATVLGDVILEAGASVWYGAVIRGDVVSIRVGASSNVQDGAILHGDPGEPLVLEELVTVGHRAVIHSAYVERGCLVGIGATLLNGVRVGAGSIIGAGALVTKDVPPRSLVLGVPGKVVRQVSEEQALDLMDHARRYEQLAQVHAGTGSDLGFHGPTSEP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>putative carbonic anhydrase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=co>JAAUTY01</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>gamma carbonic anhydrase family protein</td>
            <td id=co>Microcoleus</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TPA: gamma carbonic anhydrase family protein</td>
            <td id=co>DVEG01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Gamma carbonic anhydrase-like: This family is composed of gamma carbonic anhydrase (CA), Ferripyochelin Binding Protein (FBP), E. coli paaY protein, and similar proteins. CAs are zinc-containing enzymes that catalyze the reversible hydration of carbon dioxide in a two-step mechanism, involving the nucleophilic attack of a zinc-bound hydroxide ion on carbon dioxide, followed by the regeneration of the active site by ionization of the zinc-bound water molecule and removal of a proton from the active site. They are ubiquitous enzymes involved in fundamental processes like photosynthesis, respiration, pH homeostasis and ion transport. There are three evolutionary distinct groups - alpha, beta and gamma carbonic anhydrases - which show no significant sequence identity or structural similarity. Gamma CAs are trimeric enzymes with left-handed parallel beta helix (LbH) structural domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Carbonic anhydrase or acetyltransferase, isoleucine patch superfamily [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Ferripyochelin Binding Protein (FBP): FBP is an outer membrane protein which plays a role in iron acquisition. It binds iron when it is complexed with pyochelin. It adopts the left-handed parallel beta-helix (LbH) structure, and contains imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. Acyltransferase activity has not been observed in this group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>carbonate dehydratase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>paaY-like: This group is composed by uncharacterized proteins with similarity to the protein product of the E. coli paaY gene, which is part of the paa gene cluster responsible for phenylacetic acid degradation. Proteins in this group are expected to adopt the left-handed parallel beta-helix (LbH) structure. They contain imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Similarity to gamma carbonic anhydrase and Ferripyochelin Binding Protein (FBP) may suggest metal binding capacity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>uncharacterized protein</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>carnitine operon protein CaiE; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>Gamma carbonic anhydrases (CA): Carbonic anhydrases are zinc-containing enzymes that catalyze the reversible hydration of carbon dioxide in a two-step mechanism, involving the nucleophilic attack of a zinc-bound hydroxide ion on carbon dioxide, followed by the regeneration of the active site by ionization of the zinc-bound water molecule and removal of a proton from the active site. They are ubiquitous enzymes involved in fundamental processes like photosynthesis, respiration, pH homeostasis and ion transport. There are three distinct groups of  carbonic anhydrases - alpha, beta and gamma - which show no significant sequence identity or structural similarity. Gamma CAs are homotrimeric enzymes, with each subunit containing a left-handed parallel beta helix (LbH) structural domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>Putative Acyltransferase (AT), Left-handed parallel beta-Helix (LbH) domain; This group is composed of mostly uncharacterized proteins containing an N-terminal helical subdomain followed by a LbH domain. The alignment contains 6 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. A few members are identified as NeuD, a sialic acid (Sia) O-acetyltransferase that is required for Sia synthesis and surface polysaccharide sialylation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>WcxM-like, Left-handed parallel beta-Helix (LbH) N-terminal domain: This group is composed of Xanthomonas campestris WcxM and proteins with similarity to the WcxM N-terminal domain. WcxM is thought to be bifunctional, catalyzing both the isomerization and transacetylation reactions of keto-hexoses. It contains an N-terminal LbH domain responsible for the transacetylation function and a C-terminal isomerase domain. The LbH domain contains imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), typical of enzymes with acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>sugar O-acyltransferase, sialic acid O-acetyltransferase NeuD family. This family of proteins includes the characterized NeuD sialic acid O-acetyltransferase enzymes from E. coli and Streptococcus agalactiae (group B strep). These two are quite closely related to one another, so extension of this annotation to other members of the family in unsupported without additional independent evidence. The neuD gene is often observed in close proximity to the neuABC genes for the biosynthesis of CMP-N-acetylneuraminic acid (CMP-sialic acid), and NeuD sequences from these organisms were used to construct the seed for this model. Nevertheless, there are numerous instances of sequences identified by this model which are observed in a different genomic context (although almost universally in exopolysaccharide biosynthesis-related loci), as well as in genomes for which the biosynthesis of sialic acid (SA) is undemonstrated. Even in the cases where the association with SA biosynthesis is strong, it is unclear in the literature whether the biological substrate is SA iteself, CMP-SA, or a polymer containing SA. Similarly, it is unclear to what extent the enzyme has a preference for acetylation at the 7, 8 or 9 positions. In the absence of evidence of association with SA, members of this family may be involved with the acetylation of differring sugar substrates, or possibly the delivery of alternative acyl groups. The closest related sequences to this family (and those used to root the phylogenetic tree constructed to create this model) are believed to be succinyltransferases involved in lysine biosynthesis. These proteins contain repeats of the bacterial transferase hexapeptide (pfam00132), although often these do not register above the trusted cutoff.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Xenobiotic acyltransferase (XAT): The XAT class of hexapeptide acyltransferases is composed of a large number of microbial enzymes that catalyze the CoA-dependent acetylation of a variety of hydroxyl-bearing acceptors such as chloramphenicol and streptogramin, among others. Members of this class of enzymes include Enterococcus faecium streptogramin A acetyltransferase and Pseudomonas aeruginosa chloramphenicol acetyltransferase. They contain repeated copies of a six-residue hexapeptide repeat sequence motif (X-[STAV]-X-[LIV]-[GAED]-X) and adopt a left-handed parallel beta helix (LbH) structure. The active enzyme is a trimer with CoA and substrate binding sites at the interface of two separate LbH subunits. XATs are implicated in inactivating xenobiotics leading to xenobiotic resistance in patients.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>Maltose O-acyltransferase (MAT)-like: This family is composed of maltose O-acetyltransferase, galactoside O-acetyltransferase (GAT), xenobiotic acyltransferase (XAT) and similar proteins. MAT and GAT catalyze the CoA-dependent acetylation of the 6-hydroxyl group of their respective sugar substrates. MAT acetylates maltose and glucose exclusively while GAT specifically acetylates galactopyranosides. XAT catalyzes the CoA-dependent acetylation of a variety of hydroxyl-bearing acceptors such as chloramphenicol and streptogramin, among others. XATs are implicated in inactivating xenobiotics leading to xenobiotic resistance in patients. Members of this family contain a a left-handed parallel beta-helix (LbH) domain with at least 5 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). They are trimeric in their active form.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cr>28</td>
            <td id=al>Maltose O-acetyltransferase (MAT) and Galactoside O-acetyltransferase (GAT): MAT and GAT catalyze the CoA-dependent acetylation of the 6-hydroxyl group of their respective sugar substrates. MAT acetylates maltose and glucose exclusively at the C6 position of the nonreducing end glucosyl moiety. GAT specifically acetylates galactopyranosides. Furthermore, MAT shows higher affinity toward artificial substrates containing an alkyl or hydrophobic chain as well as a glucosyl unit. Active MAT and GAT are homotrimers, with each subunit consisting of an N-terminal alpha-helical region and a C-terminal left-handed parallel alpha-helix (LbH) subdomain with 6 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>Acetyltransferase (isoleucine patch superfamily) [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cr>34</td>
            <td id=al>serine acetyltransferase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>Serine acetyltransferase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>Serine acetyltransferase (SAT): SAT catalyzes the CoA-dependent acetylation of the side chain hydroxyl group of L-serine to form O-acetylserine, as the first step of a two-step biosynthetic pathway in bacteria and plants leading to the formation of L-cysteine. This reaction represents a key metabolic point of regulation for the cysteine biosynthetic pathway due to its feedback inhibition by cysteine. The enzyme is a 175 kDa homohexamer, composed of a dimer of homotrimers. Each subunit contains an N-terminal alpha helical region and a C-terminal left-handed beta-helix (LbH) subdomain with 5 turns, each containing a hexapeptide repeat motif characteristic of the acyltransferase superfamily of enzymes. The trimer interface mainly involves the C-terminal LbH subdomain while the dimer (of trimers) interface is mediated by the N-terminal alpha helical subdomain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cr>22</td>
            <td id=al>serine O-acetyltransferase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>wcaF-like: This group is composed of the protein product of the E. coli wcaF gene and similar proteins. WcaF is part of the gene cluster responsible for the biosynthesis of the extracellular polysaccharide colanic acid. The wcaF protein is predicted to contain a left-handed parallel beta-helix (LbH) domain encoded by imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. Many are trimeric in their active forms.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cr>19</td>
            <td id=al>galactoside O-acetyltransferase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cr>23</td>
            <td id=al>serine acetyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>12</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Dynactin 5 (or subunit p25); Dynactin is a major component of the activator complex that stimulates dynein-mediated vesicle transport. Dynactin is a heterocomplex of at least eight subunits, including a 150,000-MW protein called Glued, the actin-capping protein Arp1, and dynamatin. In vitro binding experiments show that dynactin enhances dynein-dependent motility, possibly through interaction with microtubules and vesicles. Subunit p25 is part of the pointed-end subcomplex in dynactin that also includes p26, p27, and Arp11. This subcomplex interacts with membranous cargoes. p25 and p27 contain imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), indicating a left-handed parallel beta helix (LbH) structural domain. Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2302591</td></tr>
<tr><td id=al>end</td><td id=al>2303161</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>gamma carbonic anhydrase family protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
