<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00145</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00145 uroporphyrinogen-III C-methyltransferase [Phormidium_A]</h3>
    <div>MSLGNVYLVGAGVGDLKYLTQRGRQVLAQAEVLIYDALADNRLLELVPDNCLCLDMGKRGGRPSPPQGEIDRLLVAYCGQGRRVVRLKAGDPFIFGRSASELRALRAANCPVEVVPGISSALAAPLLAGIPLTDTQLSSGFAVFTGHDLAALDWLALARLPTLVFLMAGRNLGAICKALIDHGKLPNTPVAMIRACGRPEQRVWTGTLETIDRQTQGEIRSPTVIVMGEVVSLRDVFGETAMVASEMGLQGKTILVTRSTGQSSDFCDRLQHHGAIPLSTPALEIVPPSSWQELDGAIAQLEQVDWLILTSSNGVEFFLDRLLTLGQDLRSLAHLKIAVVGKKTARVLKQRGLSPDFIPPDFIADSLVERFPESVEGKTVLFPRVETGGRDVLVRQFREAGAKVLEAAAYESRCPDTLDPQALAALRQGTVDVVTFASSKTVRNFAQLIETAGDISLAGVAIASIGPQTSQSCHQAWGRCDIEAEEFTIEGLEQAILDWAIGQPSHP</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>uroporphyrinogen III methyltransferase / synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=co>Gloeocapsopsis</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=co>Chroogloeocystis</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase</td>
            <td id=cb>LEGE-11467</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uroporphyrin-III C-methyltransferase (also known as S-Adenosyl-L-methionine:uroporphyrinogen III methyltransferase, SUMT). SUMT is an enzyme of the cobalamin and siroheme biosynthetic pathway. It catalyzes the first of three steps leading to the formation of siroheme from uroporphyrinogen III; it transfers two methyl groups from S-adenosyl-L-methionine to the C-2 and C-7 atoms of uroporphyrinogen III to yield precorrin-2 via the intermediate formation of precorrin-1. Precorrin-2 is also a precursor for the biosynthesis of vitamin B12, coenzyme F430, siroheme and heme d1. This family includes proteins in which the SUMT domain is fused to other functional domains, such as to a uroporphyrinogen-III synthase domain to form bifunctional uroporphyrinogen-III methylase/uroporphyrinogen-III synthase, or to a dual function dehydrogenase-chelatase domain, as in the case of the multifunctional S-adenosyl-L-methionine (SAM)-dependent bismethyltransferase/dehydrogenase/ferrochelatase CysG, which catalyzes all three steps that transform uroporphyrinogen III into siroheme.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Uroporphyrinogen-III methylase (siroheme synthase) [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>uroporphyrinogen-III C-methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>uroporphyrin-III C-methyltransferase. This model represents enzymes, or enzyme domains, with uroporphyrin-III C-methyltransferase activity. This enzyme catalyzes the first step committed to the biosynthesis of either siroheme or cobalamin (vitamin B12) rather than protoheme (heme). Cobalamin contains cobalt while siroheme contains iron. Siroheme is a cofactor for nitrite and sulfite reductases and therefore plays a role in cysteine biosynthesis; many members of this family are CysG, siroheme synthase, with an additional N-terminal domain and with additional oxidation and iron insertion activities. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>uroporphyrin-III C-methyltransferase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uroporphyrinogen-III synthase (HemD) catalyzes the asymmetrical cyclization of tetrapyrrole (linear) to uroporphyrinogen-III, the fourth step in the biosynthesis of heme. This ubiquitous enzyme is present in eukaryotes, bacteria and archaea. Mutations in the human uroporphyrinogen-III synthase gene cause congenital erythropoietic porphyria, a recessive inborn error of metabolism also known as Gunther disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>Uroporphyrinogen-III synthase (HemD) catalyzes the asymmetrical cyclization of tetrapyrrole (linear) to uroporphyrinogen-III, the fourth step in the biosynthesis of heme. This ubiquitous enzyme is present in eukaryotes, bacteria and archaea. Mutations in the human uroporphyrinogen-III synthase gene cause congenital erythropoietic porphyria, a recessive inborn error of metabolism also known as Gunther disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Uroporphyrinogen-III synthase HemD. This family consists of uroporphyrinogen-III synthase HemD EC:4.2.1.75 also known as Hydroxymethylbilane hydrolyase (cyclizing) from eukaryotes, bacteria and archaea. This enzyme catalyzes the reaction: Hydroxymethylbilane <=> uroporphyrinogen-III + H(2)O. Some members of this family are multi-functional proteins possessing other enzyme activities related to porphyrin biosynthesis, such as HemD with pfam00590, however the aligned region corresponds with the uroporphyrinogen-III synthase EC:4.2.1.75 activity only. Uroporphyrinogen-III synthase is the fourth enzyme in the heme pathway. Mutant forms of the Uroporphyrinogen-III synthase gene cause congenital erythropoietic porphyria in humans a recessive inborn error of metabolism also known as Gunther disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>Uroporphyrinogen-III synthase HemD. This family consists of uroporphyrinogen-III synthase HemD EC:4.2.1.75 also known as Hydroxymethylbilane hydrolyase (cyclizing) from eukaryotes, bacteria and archaea. This enzyme catalyzes the reaction: Hydroxymethylbilane <=> uroporphyrinogen-III + H(2)O. Some members of this family are multi-functional proteins possessing other enzyme activities related to porphyrin biosynthesis, such as HemD with pfam00590, however the aligned region corresponds with the uroporphyrinogen-III synthase EC:4.2.1.75 activity only. Uroporphyrinogen-III synthase is the fourth enzyme in the heme pathway. Mutant forms of the Uroporphyrinogen-III synthase gene cause congenital erythropoietic porphyria in humans a recessive inborn error of metabolism also known as Gunther disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uroporphyrinogen-III synthase [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>50</td>
            <td id=al>Uroporphyrinogen-III synthase [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>uroporphyrin-III C-methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>uroporphyrinogen-III synthase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Tetrapyrrole (Corrin/Porphyrin) Methylases. This family uses S-AdoMet in the methylation of diverse substrates. This family includes a related group of bacterial proteins of unknown function. This family includes the methylase Dipthine synthase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>siroheme synthase CysG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Precorrin-4 methylase  [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>precorrin-4 C11-methyltransferase. This model represents precorrin-4 C11-methyltransferase, one of two methyltransferases commonly referred to as precorrin-3 methylase (the other is precorrin-3B C17-methyltransferase, EC 2.1.1.131). This enzyme participates in the pathway toward the biosynthesis of cobalamin and related products. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Precorrin-4 C11-methyltransferase (CbiF/CobM). Precorrin-4 C11-methyltransferase participates in the pathway toward the biosynthesis of cobalamin (vitamin B12). There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. In the aerobic pathway, CobM catalyzes the methylation of precorrin-4 at C-11 to yield precorrin-5. In the anaerobic pathway, CibF catalyzes the methylation of cobalt-precorrin-4 to cobalt-precorrin-5. Both CibF and CobM, which are homologous, are included in this model. There are about 30 enzymes involved in vitamin B12 synthetic pathway. The enzymes involved in the aerobic pathway are prefixed Cob and those of the anaerobic pathway Cbi. Most of the enzymes are shared in both pathways and several of these enzymes are pathway-specific.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S-AdoMet-dependent tetrapyrrole methylases. This superfamily uses S-AdoMet (S-adenosyl-L-methionine or SAM) in the methylation of diverse substrates. Most members catalyze various methylation steps in cobalamin (vitamin B12) biosynthesis. There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. The enzymes involved in the aerobic pathway are prefixed Cob and those of the anaerobic pathway Cbi. Most of the enzymes are shared by both pathways and a few enzymes are pathway-specific. Diphthine synthase and ribosomal RNA small subunit methyltransferase I (RsmI) are two superfamily members that are not involved in cobalamin biosynthesis. Diphthine synthase participates in the posttranslational modification of a specific histidine residue in elongation factor 2 (EF-2) of eukaryotes and archaea to diphthamide. RsmI catalyzes the 2-O-methylation of the ribose of cytidine 1402 (C1402) in 16S rRNA. Other superfamily members not involved in cobalamin biosynthesis include the N-terminal tetrapyrrole methylase domain of Bacillus subtilis YabN whose specific function is unknown, and Omphalotus olearius omphalotin methyltransferase which catalyzes the automethylation of its own C-terminus; this C terminus is subsequently released and macrocyclized to give Omphalotin A, a potent nematicide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>uncharacterized family of the tetrapyrrole methylase superfamily. Members of this superfamily use S-AdoMet (S-adenosyl-L-methionine or SAM) in the methylation of diverse substrates. Most members catalyze various methylation steps in cobalamin (vitamin B12) biosynthesis. There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. The enzymes involved in the aerobic pathway are prefixed Cob and those of the anaerobic pathway Cbi. Most of the enzymes are shared by both pathways and a few enzymes are pathway-specific. Diphthine synthase and Ribosomal RNA small subunit methyltransferase I (RsmI) are two superfamily members that are not involved in cobalamin biosynthesis. Diphthine synthase participates in the posttranslational modification of a specific histidine residue in elongation factor 2 (EF-2) of eukaryotes and archaea to diphthamide. RsmI catalyzes the 2-O-methylation of the ribose of cytidine 1402 (C1402) in 16S rRNA. Other superfamily members not involved in cobalamin biosynthesis include the N-terminal tetrapyrrole methylase domain of Bacillus subtilis YabN whose specific function is unknown, and Omphalotus olearius omphalotin methyltransferase which catalyzes the automethylation of its own C-terminus; this C terminus is subsequently released and macrocyclized to give Omphalotin A, a potent nematicide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>bifunctional uroporphyrinogen-III synthetase/uroporphyrin-III C-methyltransferase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Precorrin-6Y methyltransferase (also named CbiE). CbiE (precorrin-6Y methyltransferase, also known as cobalt-precorrin-7 C(5)-methyltransferase, also known as cobalt-precorrin-6Y C(5)-methyltransferase) catalyzes the methylation of C-5 in cobalt-precorrin-7 to form cobalt-precorrin-8. It participates in the pathway toward the biosynthesis of cobalamin (vitamin B12). There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. CbiE functions in the anaerobic pathway, it is a subunit of precorrin-6Y C5,15-methyltransferase, a bifunctional enzyme: cobalt-precorrin-7 C(5)-methyltransferase (CbiE)/cobalt-precorrin-6B C(15)-methyltransferase (decarboxylating) (CbiT), that catalyzes two methylations (at C-5 and C-15) in precorrin-6Y, as well as the decarboxylation of the acetate side chain located in ring C, in order to generate precorrin-8X. CbiE and CbiT can be found fused (CbiET, also called CobL), or on separate protein chains (CbiE and CbiT). In the aerobic pathway, a single enzyme called CobL catalyzes the methylations at C-5 and C-15, and the decarboxylation of the C-12 acetate side chain of precorrin-6B.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Precorrin-6B methylase 1  [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Precorrin-2 C20-methyltransferase, also named CobI or CbiL. Precorrin-2 C20-methyltransferase (also known as S-adenosyl-L-methionine--precorrin-2 methyltransferase) participates in the pathway toward the biosynthesis of cobalamin (vitamin B12). There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. Precorrin-2 C20-methyltransferase catalyzes methylation at the C-20 position of a cyclic tetrapyrrole ring of precorrin-2 using S-adenosylmethionine as a methyl group source to produce precorrin-3A. In the anaerobic pathway, cobalt is inserted into precorrin-2 by CbiK to generate cobalt-precorrin-2, which is the substrate for CbiL, a C20 methyltransferase. In Clostridium difficile, CbiK and CbiL are fused into a bifunctional enzyme. In the aerobic pathway, the precorrin-2 C20-methyltransferase is named CobI. This family includes CbiL and CobI precorrin-2 C20-methyltransferases, both as stand-alone enzymes and when CbiL forms part of a bifunctional enzyme.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Precorrin-3B C(17)-methyltransferase (also named CobJ or CbiH). Precorrin-3B C(17)-methyltransferase participates in the pathway toward the biosynthesis of cobalamin (vitamin B12). There are two distinct cobalamin biosynthetic pathways. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. This model includes CobJ of the aerobic pathway and CbiH of the anaerobic pathway, both as stand-alone enzymes and when CobJ or CbiH form part of bifunctional enzymes, such as in Mycobacterium tuberculosis CobIJ  where CobJ fuses with a precorrin-2 C(20)-methyltransferase domain, or Bacillus megaterium CbiH60, where CbiH is fused to a nitrite and sulfite reductase-like domain. In the aerobic pathway, once CobG has generated precorrin-3b, CobJ catalyzes the methylation of precorrin-3b at C-17 to form precorrin-4 (the extruded methylated C-20 fragment is left attached as an acyl group at C-1). In the corresponding anaerobic  pathway, CbiH carries out this ring contraction, using cobalt-precorrin-3b as a substrate to generate a tetramethylated delta-lactone.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>cobalt-precorrin-4 methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cobalt-precorrin-7 (C(5))-methyltransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Precorrin-3B methylase  [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>precorrin-2 C(20)-methyltransferase. This model represents precorrin-2 C(20)-methyltransferase, one of several closely related S-adenosylmethionine-dependent methyltransferases involved in cobalamin (vitamin B12) biosynthesis. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>precorrin-3B C17-methyltransferase. This model represents precorrin-3B C17-methyltransferase, one of two methyltransferases commonly referred to as precorrin-3 methylase (the other is precorrin-4 C11-methyltransferase, EC 2.1.1.133). This enzyme participates in the pathway toward the biosynthesis of cobalamin and related products. Members of this family may appear as fusion proteins with other enzymes of cobalamin biosynthesis. [Biosynthesis of cofactors, prosthetic groups, and carriers, Heme, porphyrin, and cobalamin]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Precorrin-2 methylase  [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>precorrin-3B C17-methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>bifunctional uroporphyrinogen-III synthetase/response regulator domain protein; Validated</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>118345</td></tr>
<tr><td id=al>end</td><td id=al>119869</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>uroporphyrinogen-III C-methyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
