<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05444</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05444 putative transcriptional regulator [Phormidium_A]</h3>
    <div>MLKVGDVADKLEINSQTLYFYERIGLIPSPQRSSGGYRLYSDRDVERLKFIRHLKGLGLSLQDIREILELQEDNLLSCDRVYQHLKLKEQEIDYKIEQLQALKSQLLPLIETCKVRLNSSETQTCSVLNHNQPHPADTPSLTPLLSNPMDTYKIEVLGTGCKKCHQLEANAQEAVRNLGLEAEISHITDMMEITKRGVMKTPAIVIGDRVMSQGKVPSAAEIETLING</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>putative transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MTH895/ArsE family thioredoxin-like protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>MTH895/ArsE family thioredoxin-like protein</td>
            <td id=co>Coleofasciculus</td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MTH895/ArsE family thioredoxin-like protein</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>thioredoxin family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>transcriptional regulator</td>
            <td id=co>Lyngbya</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>heavy metal-responsive transcriptional regulator</td>
            <td id=cr>Synechococcus_A</td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>heavy metal-responsive transcriptional regulator</td>
            <td id=co>Nodularia</td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>heavy metal-responsive transcriptional regulator</td>
            <td id=cr>Pseudanabaena</td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>heavy metal-responsive transcriptional regulator</td>
            <td id=co>Rivularia</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of Heavy Metal Resistance transcription regulators. Helix-turn-helix (HTH) heavy metal resistance transcription regulators (HMRTR): MerR1 (mercury), CueR (copper),  CadR (cadmium),  PbrR (lead), ZntR (zinc), and other related proteins. These transcription regulators mediate responses to heavy metal stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the CadR- and PbrR-like transcription regulators. Helix-turn-helix (HTH) CadR- and PbrR-like transcription regulators. CadR and PbrR regulate expression of the cadmium and lead resistance operons, respectively. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines which comprise a putative metal binding site. Some members in this group have a histidine-rich C-terminal extension. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the CadR and PbrR transcription regulators. Helix-turn-helix (HTH) CadR and PbrR transcription regulators including Pseudomonas aeruginosa CadR and Ralstonia metallidurans PbrR that regulate expression of the cadmium and lead resistance operons, respectively. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines which form a putative metal binding site. Some members in this group have a histidine-rich C-terminal extension. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of CueR-like transcription regulators. Helix-turn-helix (HTH) transcription regulators CueR and ActP, copper efflux regulators. In Bacillus subtilis, copper induced CueR regulates the copZA operon, preventing copper toxicity. In Rhizobium leguminosarum, ActP controls copper homeostasis; it detects cytoplasmic copper stress and activates transcription in response to increasing copper concentrations. These proteins are comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain winged HTH motifs that mediate DNA binding, while the C-terminal domains have two conserved cysteines that define a monovalent copper ion binding site. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of MerR-like transcription regulators. Helix-turn-helix (HTH) MerR-like transcription regulator, N-terminal domain. The MerR family transcription regulators have been shown to mediate responses to stress including exposure to heavy metals, drugs, or oxygen radicals in eubacterial and some archaeal species. They regulate transcription of multidrug/metal ion transporter genes and oxidative stress regulons by reconfiguring the spacer between the -35 and -10 promoter elements.  A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>helix_turn_helix, mercury resistance. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the MerR1 transcription regulator. Helix-turn-helix (HTH) transcription regulator MerR1. MerR1 transcription regulators, such as Tn21 MerR and Tn501 MerR, mediate response to mercury exposure in eubacteria. These proteins are comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain winged HTH motifs that mediate DNA binding, while the C-terminal domains have three conserved cysteines that define a mercury binding site. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Thioredoxin domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>DNA-binding transcriptional regulator, MerR family [Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the MerR-like transcription regulators YyaN and YraB. Putative helix-turn-helix (HTH) MerR-like transcription regulators of Bacillus subtilis, YyaN and YraB, and related proteins; N-terminal domain. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Cu(I)-responsive transcriptional regulator. This model represents the copper-, silver- and gold- (I) responsive transcriptional activator of the gamma proteobacterial copper efflux system. This protein is a member of the MerR family of transcriptional activators (pfam00376) and contains a distinctive pattern of cysteine residues in its metal binding loop, Cys-X7-Cys. This family also lacks a conserved cysteine at the N-terminal end of the dimerization helix which is required for the binding of divalent metals such as zinc; here it is replaced by a serine residue. [Regulatory functions, DNA interactions]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Zn(2+)-responsive transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>Hg(II)-responsive transcriptional regulator. This model represents the mercury (II) responsive transcriptional activator of the mer organomercurial resistance operon. This protein is a member of the MerR family of transcriptional activators (pfam00376) and contains a distinctive pattern of cysteine residues in its metal binding loop, Cys-X(8)-Cys-Pro, as well as a conserved and critical cysteine at the N-terminal end of the dimerization helix. [Cellular processes, Detoxification, Regulatory functions, DNA interactions]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the transcription regulators TipAL, Mta, and SkgA. Helix-turn-helix (HTH) TipAL, Mta, and SkgA transcription regulators, and related proteins, N-terminal domain. TipAL regulates resistance to and activation by numerous cyclic thiopeptide antibiotics, such as thiostrepton. Mta is a global transcriptional regulator; the N-terminal DNA-binding domain of Mta interacts directly with the promoters of mta, bmr, blt, and ydfK, and induces transcription of these multidrug-efflux transport genes. SkgA has been shown to control stationary-phase expression of catalase-peroxidase in Caulobacter crescentus. These proteins are comprised of distinct domains that harbor an  N-terminal active (DNA-binding) site and a regulatory (effector-binding) site. The conserved N-terminal domain of these transcription regulators contains winged HTH motifs that mediate DNA binding. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Unique to this family, is a TipAL-like, lineage specific Bacilli subgroup, which has five conserved cysteines in the C-terminus of the protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative Heavy Metal Resistance transcription regulators. Putative helix-turn-helix (HTH) heavy metal resistance transcription regulators (HMRTR), unknown subgroup. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to heavy metal stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules, such as, metal ions, drugs, and organic substrates. This subgroup lacks one of the conserved, metal-binding cysteines seen in the MerR1 group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily. Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 3). Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative Cfa-like transcription regulators. Putative helix-turn-helix (HTH) MerR-like transcription regulator; conserved, Cfa-like, unknown proteins (~172 a.a.). The N-terminal domain of these proteins appears to be related to the HTH domain of Cfa, a cyclopropane fatty acid synthase. These Cfa-like proteins have a unique C-terminal domain with conserved histidines (motif HXXFX7HXXF). Based on sequence similarity of the N-terminal domains, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MerR HTH family regulatory protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>small redox-active disulfide protein 2. This small protein is found in three archaeal species so far (Methanococcus jannaschii, Archeoglobus fulgidus, and Methanobacterium thermoautotrophicum) as well as in Anabaena PCC7120. It is homologous to thioredoxins, glutaredoxins, and protein disulfide isomerases, and shares with them a redox-active disulfide. The redox active disulfide region CXXC motif resembles neither thioredoxin nor glutaredoxin. A closely related protein found in the same three Archaea, described by redox_disulf_1, has a glutaredoxin-like CP[YH]C sequence; it has been characterized in functional assays as redox-active but unlikely to be a thioredoxin or glutaredoxin. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the transcription regulators NolA and AlbR. Helix-turn-helix (HTH) transcription regulators NolA and AlbR, N-terminal domain. In Bradyrhizobium (Arachis) sp. NC92, NolA is required for efficient nodulation of host plants. In Xanthomonas albilineans, AlbR regulates the expression of the pathotoxin, albicidin. These proteins are putatively comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains are often unrelated and bind specific coactivator molecules. They share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the HspR transcription regulator. Helix-turn-helix (HTH) transcription regulator HspR, N-terminal domain. Heat shock protein regulators (HspR) have been shown to regulate expression of specific regulons in response to high temperature or high osmolarity in Streptomyces and Helicobacter, respectively. These proteins share the N-terminal DNA binding domain  with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.  A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of MerR2-like transcription regulators. Helix-turn-helix (HTH) transcription regulator MerR2 and related proteins. MerR2 in Bacillus cereus RC607 regulates resistance to organomercurials. The MerR family transcription regulators have been shown to mediate responses to stress including exposure to heavy metals, drugs, or oxygen radicals in eubacterial and some archaeal species. They regulate transcription by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>HTH-type transcriptional regulator CueR. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the YfmP transcription regulator. Helix-turn-helix (HTH) transcription regulator, YfmP, and related proteins; N-terminal domain. YfmP regulates the multidrug efflux protein, YfmO, and indirectly regulates the expression of the Bacillus subtilis copZA operon encoding a metallochaperone, CopZ, and a CPx-type ATPase efflux protein, CopA. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the BltR transcription regulator. Helix-turn-helix (HTH) multidrug-efflux transporter transcription regulator, BltR (BmrR-like transporter) of Bacillus subtilis, and related proteins; N-terminal domain. Blt, like Bmr, is a membrane protein which causes the efflux of a variety of toxic substances and antibiotics. These regulators are comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the C-terminal domains are often unrelated and bind specific coactivator molecules. They share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of transcription regulators from the MerR superfamily. Helix-turn-helix (HTH) transcription regulator MerR superfamily, N-terminal domain. The MerR family transcription regulators have been shown to mediate responses to stress including exposure to heavy metals, drugs, or oxygen radicals in eubacterial and some archaeal species. They regulate transcription of multidrug/metal ion transporter genes and oxidative stress regulons by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the regulatory protein GnyR. Putative helix-turn-helix (HTH) regulatory protein, GnyR, and other related proteins. GnyR belongs to the gnyRDBHAL cluster, which is involved in acyclic isoprenoid degradation in Pseudomonas aeruginosa. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>MerR, DNA binding. Members of this family of DNA-binding domains are predominantly found in the prokaryotic transcriptional regulator MerR. They adopt a structure consisting of a core of three alpha helices, with an architecture that is similar to that of the 'winged helix' fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of BmrR-like transcription regulators. Helix-turn-helix (HTH) BmrR-like transcription regulators (TipAL, Mta, SkgA, BmrR, and BltR), N-terminal domain. These proteins have been shown to regulate expression of specific regulons in response to various toxic substances, antibiotics, or oxygen radicals in Bacillus subtilis, Streptomyces, and Caulobacter crescentus. They are comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain  HTH motifs that mediate DNA binding, while the C-terminal domains are often unrelated and bind specific coactivator molecules. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily. Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 6) with at least two conserved cysteines present in the C-terminal portion of the protein. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the BmrR transcription regulator. Helix-turn-helix (HTH) multidrug-efflux transporter transcription regulator, BmrR and YdfL of Bacillus subtilis, and related proteins; N-terminal domain. Bmr is a membrane protein which causes the efflux of a variety of toxic substances and antibiotics. BmrR is comprised of two distinct domains that harbor a regulatory (effector-binding) site and an active (DNA-binding) site. The conserved N-terminal domain contains a winged HTH motif  that mediates DNA binding, while the C-terminal domain binds coactivating, toxic compounds. BmrR shares the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the SoxR transcription regulator. Helix-turn-helix (HTH) transcriptional regulator SoxR. The global regulator, SoxR, up-regulates gene expression of another transcription activator, SoxS, which directly stimulates the oxidative stress regulon genes in E. coli. The soxRS response renders the bacterial cell resistant to superoxide-generating agents, macrophage-generated nitric oxide, organic solvents, and antibiotics. The SoxR proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the unusually long spacer between the -35 and -10 promoter elements. They also harbor a regulatory C-terminal domain containing an iron-sulfur center.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>mercuric resistance operon transcriptional regulator MerR. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of Cfa-like transcription regulators. Putative helix-turn-helix (HTH) MerR-like transcription regulators; the HTH domain of Cfa, a cyclopropane fatty acid synthase, and other related methyltransferases, as well as, the N-terminal domain of a conserved, uncharacterized ~172 a.a. protein. Based on sequence similarity of the N-terminal domain, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily. Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 4). Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>MerR family regulatory protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the Cfa transcription regulator. Putative helix-turn-helix (HTH) MerR-like transcription regulator; the N-terminal domain of Cfa, a cyclopropane fatty acid synthase and other related methyltransferases. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily. Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 7) with a conserved cysteine present in the C-terminal portion of the protein. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of HspR-like transcription regulators. Helix-turn-helix (HTH) transcription regulator HspR and related proteins, N-terminal domain. Heat shock protein regulators (HspR) have been shown to regulate expression of specific regulons in response to high temperature or high osmolarity in Streptomyces and Helicobacter, respectively. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>First Helix-Turn-Helix DNA binding domain of the regulatory protein TioE. Putative helix-turn-helix (HTH) regulatory protein, TioE, and related proteins. TioE is part of the thiocoraline gene cluster, which is involved in the biosynthesis of the antitumor thiocoraline from the marine actinomycete, Micromonospora. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Proteins in this family are unique within the MerR superfamily in that they are composed of just two adjacent MerR-like N-terminal domains; this CD contains the N-terminal or first repeat (rpt1) of these tandem MerR-like domain proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of the transcription regulators MlrA and CarA. Helix-turn-helix (HTH) transcription regulator MlrA (merR-like regulator A), N-terminal domain. The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium.  Its close homolog, CarA from Myxococcus xanthus, is involved in activation of the carotenoid biosynthesis genes by light. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules. Many MlrA- and CarA-like proteins in this group appear to lack the long dimerization helix seen in the N-terminal domains of typical MerR-like proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative MlrA-like transcription regulators. Putative helix-turn-helix (HTH) MlrA-like transcription regulators (subgroup 1). The MlrA protein, also known as YehV, has been shown to control cell-cell aggregation by co-regulating the expression of curli and extracellular matrix production in Escherichia coli and Salmonella typhimurium. These proteins belong to the MerR superfamily of transcription regulators that promote expression of several stress regulon genes by reconfiguring the spacer between the -35 and -10 promoter elements. Their conserved N-terminal domains contain predicted HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules. Many MlrA-like proteins in this group appear to lack the long dimerization helix seen in the N-terminal domains of typical MerR-like proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative transcription regulators from the MerR superfamily. Putative helix-turn-helix (HTH) MerR-like transcription regulators (subgroup 5), N-terminal domain. Based on sequence similarity, these proteins are predicted to function as transcription regulators that mediate responses to stress in eubacteria. They belong to the MerR superfamily of transcription regulators that promote transcription of various stress regulons by reconfiguring the operator sequence located between the -35 and -10 promoter elements. A typical MerR regulator is comprised of two distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their N-terminal domains are homologous and contain a DNA-binding winged HTH motif, while the C-terminal domains are often dissimilar and bind specific coactivator molecules such as metal ions, drugs, and organic substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Second Helix-Turn-Helix DNA binding domain of the regulatory protein TioE. Putative helix-turn-helix (HTH) regulatory protein, TioE, and related proteins. TioE is part of the thiocoraline gene cluster, which is involved in the biosynthesis of the antitumor thiocoraline from the marine actinomycete, Micromonospora. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. Proteins in this family are unique within the MerR superfamily in that they are composed of just two adjacent MerR-like N-terminal domains; this CD mainly contains the C-terminal or second repeat (rpt2) of these tandem MerR-like domain proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>Helix-Turn-Helix DNA binding domain of putative HspR-like transcription regulators. Putative helix-turn-helix (HTH) transcription regulator HspR-like proteins. Unlike the characterized HspR, these proteins have a C-terminal domain with putative metal binding cysteines (MBC). Heat shock protein regulators (HspR) have been shown to regulate expression of specific regulons in response to high temperature or high osmolarity in Streptomyces and Helicobacter, respectively. These proteins share the N-terminal DNA binding domain with other transcription regulators of the MerR superfamily that promote transcription by reconfiguring the spacer between the -35 and -10 promoter elements. A typical MerR regulator is comprised of distinct domains that harbor the regulatory (effector-binding) site and the active (DNA-binding) site. Their conserved N-terminal domains contain predicted winged HTH motifs that mediate DNA binding, while the dissimilar C-terminal domains bind specific coactivator molecules.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4429013</td></tr>
<tr><td id=al>end</td><td id=al>4429700</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>putative transcriptional regulator</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
