<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03107</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03107 polyribonucleotide nucleotidyltransferase [Phormidium_A]</h3>
    <div>MKAMEKSVSFDGRDIRLKVGTLAPQAGGSVVIELGDTSVLVTATRSAGREGIDFLPLLVDYEERLYAGGKIPGGFLRREGRPPEKVTLTCRLIDRPMRPLFPQWLRDDIQIVATTLSMDELVPPDVLAVTGASIATMLAKIPFKGPMAAVRVGLVGDDFILNPSFEEIRLGDLDLVVAGTHDGVVMVEAGANQLPEQDIIEAIDFGYEGALELIRHQEALMKDLGVELVEEEPPVVDDKLEIYIREAAEEPVKKVLSNFEHTKSSRDEALDEVKGEIAEKIEQLPEDDPLRLAVTEDAKSLGKVFKGVTKQLMRAQIVEDKVRVDGRKLDEVRPISCEVGLVPRRVHGSALFNRGLTQVLSLATLGSPGDAQELDDLHPQEQKRYLHHYNFPPYSVGETRPMRSPGRREIGHGALAERALVPVLPPKESFPYVIRVVSEVLSSNGSTSMGSVCGSTLSLMHAGVPLSKPVSGAAMGLIKEGDEVRILTDIQGIEDFLGDMDFKVAGTDTGITALQMDMKITGLPVKIVADAVKQALPARLHILNKMIEAISEPNPELSPFAPRLYSLKIDPDLIGLVIGPGGKTIKGITEETGCKIDIDNDGTVTISGLDGEKAMRARAIIQGMTRKIQTGEVFLGRVTRIIPIGAFVEIAPGKEGMVHISQVADYRIGKVEDELSVGDEVVVKVRDIDNKNRINLTRLGIDPDDEAAAREAGIIG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase</td>
            <td id=cb>RDYZ01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polynucleotide phosphorylase/polyadenylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polyribonucleotide nucleotidyltransferase. Members of this protein family are polyribonucleotide nucleotidyltransferase, also called polynucleotide phosphorylase. Some members have been shown also to have additional functions as guanosine pentaphosphate synthetase and as poly(A) polymerase (see model TIGR02696 for an exception clade, within this family). [Transcription, Degradation of RNA]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Polyribonucleotide nucleotidyltransferase (polynucleotide phosphorylase) [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>guanosine pentaphosphate synthetase I/polynucleotide phosphorylase. Sohlberg, et al. present characterization of two proteins from Streptomyces coelicolor. The protein in this family was shown to have poly(A) polymerase activity and may be responsible for polyadenylating RNA in this species. Reference 2 showed that a nearly identical plasmid-encoded protein from Streptomyces antibioticus is a bifunctional enzyme that acts also as a guanosine pentaphosphate synthetase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>polyribonucleotide nucleotidyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Polyribonucleotide nucleotidyltransferase, repeat 2. Polyribonucleotide nucleotidyltransferase (PNPase) is a member of the RNase_PH family, named after the bacterial Ribonuclease PH, a 3'-5' exoribonuclease. Structurally, all members of this family form hexameric rings. In the case of PNPase the complex is a trimer, since each monomer contains two tandem copies of the domain. PNPase is involved in mRNA degradation in a 3'-5' direction and in quality control of ribosomal RNA precursors, with the second repeat containing the active site. PNPase is part of the RNA degradosome complex and binds to the scaffolding domain of the endoribonuclease RNase E.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Polyribonucleotide nucleotidyltransferase, repeat 1. Polyribonucleotide nucleotidyltransferase (PNPase) is a member of the RNase_PH family, named after the bacterial Ribonuclease PH, a 3'-5' exoribonuclease. Structurally, all members of this family form hexameric rings. In the case of PNPase the complex is a trimer, since each monomer contains two tandem copies of the domain. PNPase is involved in mRNA degradation in a 3'-5' direction and in quality control of ribosomal RNA precursors. It is part of the RNA degradosome complex and binds to the scaffolding domain of the endoribonuclease RNase E.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>3' exoribonuclease family, domain 1. This family includes 3'-5' exoribonucleases. Ribonuclease PH contains a single copy of this domain, and removes nucleotide residues following the -CCA terminus of tRNA. Polyribonucleotide nucleotidyltransferase (PNPase) contains two tandem copies of the domain. PNPase is involved in mRNA degradation in a 3'-5' direction. The exosome is a 3'-5' exoribonuclease complex that is required for 3' processing of the 5.8S rRNA. Three of its five protein components contain a copy of this domain. A hypothetical protein from S. pombe appears to belong to an uncharacterized subfamily. This subfamily is found in both eukaryotes and archaebacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>3' exoribonuclease family, domain 1. This family includes 3'-5' exoribonucleases. Ribonuclease PH contains a single copy of this domain, and removes nucleotide residues following the -CCA terminus of tRNA. Polyribonucleotide nucleotidyltransferase (PNPase) contains two tandem copies of the domain. PNPase is involved in mRNA degradation in a 3'-5' direction. The exosome is a 3'-5' exoribonuclease complex that is required for 3' processing of the 5.8S rRNA. Three of its five protein components contain a copy of this domain. A hypothetical protein from S. pombe appears to belong to an uncharacterized subfamily. This subfamily is found in both eukaryotes and archaebacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S1_PNPase: Polynucleotide phosphorylase (PNPase), ), S1-like RNA-binding domain. PNPase  is a polyribonucleotide nucleotidyl transferase that degrades mRNA. It is a trimeric multidomain protein. The C-terminus contains the S1 domain which binds ssRNA. This family is classified based on the S1 domain. PNPase nonspecifically removes the 3' nucleotides from mRNA, but is stalled by double-stranded RNA structures such as a stem-loop. Evidence shows that a minimum of 7-10 unpaired nucleotides at the 3' end, is required for PNPase degradation. It is suggested that PNPase also dephosphorylates the RNA 5' end. This additional activity may regulate the 5'-dependent activity of RNaseE in vivo.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>type I K homology (KH) RNA-binding domain found in polyribonucleotide nucleotidyltransferase (PNPase) and similar proteins. PNPase, also called polynucleotide phosphorylase, is a polyribonucleotide nucleotidyl transferase that degrades mRNA in prokaryotes and plant chloroplasts. It catalyzes the phosphorolysis of single-stranded polyribonucleotides processively in the 3'- to 5'-direction.  It is also involved, along with RNase II, in tRNA processing. The C-terminal region of PNPase contains domains homologous to those in other RNA binding proteins: a KH domain and an S1 domain. The model corresponds to the KH domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>exosome complex exonuclease Rrp41; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNase PH-like 3'-5' exoribonucleases. RNase PH-like 3'-5' exoribonucleases are enzymes that catalyze the 3' to 5' processing and decay of RNA substrates. Evolutionarily related members can be fond in prokaryotes, archaea, and eukaryotes. Bacterial ribonuclease PH contains a single copy of this domain, and removes nucleotide residues following the -CCA terminus of tRNA. Polyribonucleotide nucleotidyltransferase (PNPase) contains two tandem copies of the domain and is involved in mRNA degradation in a 3'-5' direction. Archaeal exosomes contain two individually encoded RNase PH-like 3'-5' exoribonucleases and are required for 3' processing of the 5.8S rRNA. The eukaryotic exosome core is composed of six individually encoded RNase PH-like subunits, but it is not a phosphorolytic enzyme per se; it directly associates with Rrp44 and Rrp6, which are hydrolytic exoribonucleases related to bacterial RNase II/R and RNase D. All members of the RNase PH-like family form ring structures by oligomerization of six domains or subunits, except for a total of 3 subunits with tandem repeats in the case of PNPase, with a central channel through which the RNA substrate must pass to gain access to the phosphorolytic active sites.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNase PH-like 3'-5' exoribonucleases. RNase PH-like 3'-5' exoribonucleases are enzymes that catalyze the 3' to 5' processing and decay of RNA substrates. Evolutionarily related members can be fond in prokaryotes, archaea, and eukaryotes. Bacterial ribonuclease PH contains a single copy of this domain, and removes nucleotide residues following the -CCA terminus of tRNA. Polyribonucleotide nucleotidyltransferase (PNPase) contains two tandem copies of the domain and is involved in mRNA degradation in a 3'-5' direction. Archaeal exosomes contain two individually encoded RNase PH-like 3'-5' exoribonucleases and are required for 3' processing of the 5.8S rRNA. The eukaryotic exosome core is composed of six individually encoded RNase PH-like subunits, but it is not a phosphorolytic enzyme per se; it directly associates with Rrp44 and Rrp6, which are hydrolytic exoribonucleases related to bacterial RNase II/R and RNase D. All members of the RNase PH-like family form ring structures by oligomerization of six domains or subunits, except for a total of 3 subunits with tandem repeats in the case of PNPase, with a central channel through which the RNA substrate must pass to gain access to the phosphorolytic active sites.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>S1_RPS1_repeat_hs4: Ribosomal protein S1 (RPS1) domain. RPS1 is a component of the small ribosomal subunit thought to be involved in the recognition and binding of mRNA's during translation initiation. The bacterial RPS1 domain architecture consists of 4-6 tandem S1 domains. In some bacteria, the tandem S1 array is located C-terminal to a 4-hydroxy-3-methylbut-2-enyl diphosphate reductase (HMBPP reductase) domain. While RPS1 is found primarily in bacteria, proteins with tandem RPS1-like domains have been identified in plants and humans, however these lack the N-terminal HMBPP reductase domain. This CD includes S1 repeat 4 (hs4) of the H. sapiens RPS1 homolog. Autoantibodies to double-stranded DNA from patients with systemic lupus erythematosus cross-react with the human RPS1 homolog.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cr>10</td>
            <td id=al>bifunctional 4-hydroxy-3-methylbut-2-enyl diphosphate reductase/30S ribosomal protein S1. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>12</td>
            <td id=al>bifunctional 4-hydroxy-3-methylbut-2-enyl diphosphate reductase/30S ribosomal protein S1. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>Predicted RNA-binding protein, contains ribosomal protein S1 (RPS1) domain  [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>23</td>
            <td id=al>30S ribosomal protein S1; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>30S ribosomal protein S1; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RRP41 subunit of archaeal exosome. The RRP41 subunit of the archaeal exosome is a member of the RNase_PH family, named after the bacterial Ribonuclease PH, a 3'-5' exoribonuclease. Structurally all members of this family form hexameric rings (trimers of dimers). In archaea, the ring is formed by three Rrp41:Rrp42 dimers. The central chamber within the ring contains three phosphorolytic active sites located in an Rrp41 pocket at the interface between Rrp42 and Rrp41. The ring is capped by three copies of Rrp4 and/or Csl4 which contain putative RNA interaction domains. The archaeal exosome degrades single-stranded RNA (ssRNA) in the 3'-5' direction, but also can catalyze the reverse reaction of adding nucleoside diphosphates to the 3'-end of RNA which has been shown to lead to the formation of poly-A-rich tails on RNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Ribosomal protein S1-like RNA-binding domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>general stress protein 13; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>30S ribosomal protein S1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>30S ribosomal protein S1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=co>50</td>
            <td id=al>RNA-binding protein S1. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>3' exoribonuclease family, domain 2. This family includes 3'-5' exoribonucleases. Ribonuclease PH contains a single copy of this domain, and removes nucleotide residues following the -CCA terminus of tRNA. Polyribonucleotide nucleotidyltransferase (PNPase) contains two tandem copies of the domain. PNPase is involved in mRNA degradation in a 3'-5' direction. The exosome is a 3'-5' exoribonuclease complex that is required for 3' processing of the 5.8S rRNA. Three of its five protein components contain a copy of this domain. A hypothetical protein from S. pombe appears to belong to an uncharacterized subfamily. This subfamily is found in both eukaryotes and archaebacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>S1 RNA binding domain. The S1 domain occurs in a wide range of RNA associated proteins. It is structurally similar to cold shock protein which binds nucleic acids. The S1 domain has an OB-fold structure.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>Ribosomal protein S1 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>16</td>
            <td id=al>Ribosomal protein S1 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>Ribosomal protein S1 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>30S ribosomal protein S1; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>S1_like: Ribosomal protein S1-like RNA-binding domain. Found in a wide variety of RNA-associated proteins. Originally identified in S1 ribosomal protein. This superfamily also contains the Cold Shock Domain (CSD), which is a homolog of the S1 domain. Both domains are members of the Oligonucleotide/oligosaccharide Binding (OB) fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>S1_Tex: The C-terminal S1 domain of a transcription accessory factor called Tex, which has been characterized in Bordetella pertussis and Pseudomonas aeruginosa. The tex gene is essential in Bortella pertusis and is named for its role in toxin expression. Tex has two functional domains, an N-terminal domain homologous to the Escherichia coli maltose repression protein, which is a poorly defined transcriptional factor, and a C-terminal S1 RNA-binding domain. Tex is found in prokaryotes, eukaryotes, and archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>S1_DHX8_helicase: The  N-terminal S1 domain of human ATP-dependent RNA helicase DHX8, a DEAH (Asp-Glu-Ala-His) box polypeptide.  The DEAH-box RNA helicases are thought to play key roles in pre-mRNA splicing and DHX8 facilitates nuclear export of spliced mRNA by releasing the RNA from the spliceosome. DHX8 is also known as HRH1 (human RNA helicase 1) in Homo sapiens and PRP22 in Saccharomyces cerevisiae.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>RNA-binding protein S1. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Ribonuclease PH [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Ribonuclease PH [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>RRP41 subunit of eukaryotic exosome. The RRP41 subunit of eukaryotic exosome is a member of the RNase_PH family, named after the bacterial Ribonuclease PH, a 3'-5' exoribonuclease. Structurally all members of this family form hexameric rings (trimers of Rrp41-Rrp45, Rrp46-Rrp43, and Mtr3-Rrp42 dimers). The eukaryotic exosome core is composed of six individually encoded RNase PH-like subunits and three additional proteins (Rrp4, Csl4 and Rrp40) that form a stable cap and contain RNA-binding domains. The RNase PH-like subunits are no longer phosphorolytic enzymes, the exosome directly associates with Rrp44 and Rrp6, hydrolytic exoribonucleases related to bacterial RNase II/R and RNase D. The exosome plays an important role in RNA turnover. It plays a crucial role in the maturation of stable RNA species such as rRNA, snRNA and snoRNA, quality control of mRNA, and the degradation of RNA processing by-products and non-coding transcripts.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Polyribonucleotide nucleotidyltransferase, RNA binding domain. This family contains the RNA binding domain of Polyribonucleotide nucleotidyltransferase (PNPase) PNPase is involved in mRNA degradation in a 3'-5' direction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>KH domain. KH motifs bind RNA in vitro. Autoantibodies to Nova, a KH domain protein, cause paraneoplastic opsoclonus ataxia.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>type I K homology (KH) RNA-binding domain found in DEAD box protein 43 (DDX43), DEAD box protein 53 (DDX53) and similar proteins. DDX43 (also called cancer/testis antigen 13, or DEAD box protein HAGE, or helical antigen) displays tumor-specific expression. Diseases associated with DDX43 include rheumatoid lung disease. DDX53 (also called cancer-associated gene protein, or cancer/testis antigen 26, or DEAD box protein CAGE) shows high expression level in various tumors and is involved in anti-cancer drug resistance. Both DDX46 and DDX53 are members of the DEAD-box helicases, a diverse family of proteins involved in ATP-dependent RNA unwinding, needed in a variety of cellular processes including splicing, ribosome biogenesis and RNA degradation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>K homology RNA-binding domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>second type I K homology (KH) RNA-binding domain found in a family of fragile X mental retardation protein (FMR1) and fragile X related (FXR) proteins. The FMR1/FXR family includes FMR1 (also known as FMRP) and its two homologues, fragile X related 1 (FXR1) and 2 (FXR2). They are involved in translational regulation, particularly in neuronal cells and play an important role in the regulation of glutamate-mediated neuronal activity and plasticity. Each of these three proteins can form heteromers with the others, and each can also form homomers. Lack of expression of FMR1 results in mental retardation and macroorchidism. FXR1 and FXR2 may play important roles in the function of FMR1 and in the pathogenesis of the Fragile X Mental Retardation Syndrome. Members of this family contain three K-homology (KH) RNA-binding domains. The model corresponds to the second one.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>first type I K homology (KH) RNA-binding domain found in Pyrococcus horikoshii Dim2p and similar proteins. The family includes a group of conserved KH domain-containing protein mainly from archaea, such as Dim2p homologues from Pyrococcus horikoshii and Aeropyrum pernix. Dim2p acts as a preribosomal RNA processing factor that has been identified as an essential protein for the maturation of 40S ribosomal subunit in Saccharomyces cerevisiae. It is required for the cleavage at processing site A2 to generate the pre-20S rRNA and for the dimethylation of the 18S rRNA by 18S rRNA dimethyltransferase, Dim1p. Dim2p contains two K-homology (KH) RNA-binding domains. The model corresponds to the first one.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>30S ribosomal protein S1; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>30S ribosomal protein S1; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>K homology (KH) RNA-binding domain, type I. KH binds single-stranded RNA or DNA. It is found in a wide variety of proteins including ribosomal proteins, transcription factors and post-transcriptional modifiers of mRNA. There are two different KH domains that belong to different protein folds, but they share a single KH motif. The KH motif is folded into a beta alpha alpha beta unit. In addition to the core, type II KH domains (e.g. ribosomal protein S3) include an N-terminal extension and type I KH domains (e.g. hnRNP K) contain a C-terminal extension. Some KH-I superfamily members contain a divergent KH domain that lacks the RNA-binding GXXG motif. Some others have a mutated GXXG motif which may or may not have nucleic acid binding ability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>S1_RPS1_repeat_ec3: Ribosomal protein S1 (RPS1) domain. RPS1 is a component of the small ribosomal subunit thought to be involved in the recognition and binding of mRNA's during translation initiation. The bacterial RPS1 domain architecture consists of 4-6 tandem S1 domains. In some bacteria, the tandem S1 array is located C-terminal to a 4-hydroxy-3-methylbut-2-enyl diphosphate reductase (HMBPP reductase) domain. While RPS1 is found primarily in bacteria, proteins with tandem RPS1-like domains have been identified in plants and humans, however these lack the N-terminal HMBPP reductase domain. This CD includes S1 repeat 3 (ec3) of the Escherichia coli RPS1. Autoantibodies to double-stranded DNA from patients with systemic lupus erythematosus cross-react with the human RPS1 homolog.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cr>13</td>
            <td id=al>ribosomal protein S1. This model describes ribosomal protein S1, RpsA. This protein is found in most bacterial genomes in a single copy, but is not present in the Mycoplasmas. It is heterogeneous with respect to the number of repeats of the S1 RNA binding domain described by pfam00575: six repeats in E. coli and most other bacteria, four in Bacillus subtilis and some other species. rpsA is an essential gene in E. coli but not in B. subtilis. It is associated with the cytidylate kinase gene cmk in many species, and fused to it in Treponema pallidum. RpsA is proposed (Medline:97323001) to assist in mRNA degradation. This model provides trusted hits to most long form (6 repeat) examples of RpsA. Among homologs with only four repeats are some to which other (perhaps secondary) functions have been assigned. [Protein synthesis, Ribosomal proteins: synthesis and modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cr>13</td>
            <td id=al>ribosomal protein S1. This model describes ribosomal protein S1, RpsA. This protein is found in most bacterial genomes in a single copy, but is not present in the Mycoplasmas. It is heterogeneous with respect to the number of repeats of the S1 RNA binding domain described by pfam00575: six repeats in E. coli and most other bacteria, four in Bacillus subtilis and some other species. rpsA is an essential gene in E. coli but not in B. subtilis. It is associated with the cytidylate kinase gene cmk in many species, and fused to it in Treponema pallidum. RpsA is proposed (Medline:97323001) to assist in mRNA degradation. This model provides trusted hits to most long form (6 repeat) examples of RpsA. Among homologs with only four repeats are some to which other (perhaps secondary) functions have been assigned. [Protein synthesis, Ribosomal proteins: synthesis and modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>49</td>
            <td id=al>S1 RNA-binding domain-containing protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>S1_RNase_E: RNase E and RNase G, S1-like RNA-binding domain. RNase E is an essential endoribonuclease in the processing and degradation of RNA. In addition to its role in mRNA degradation, RNase E has also been implicated in the processing of rRNA, and the maturation of tRNA, 10Sa RNA and the M1 precursor of RNase P. RNase E associates with PNPase (3' to 5' exonuclease), Rhl B (DEAD-box RNA helicase) and enolase (glycolytic enzyme)  to form the RNA degradosome. RNase E tends to cut mRNA within single-stranded regions that are rich in A/U nucleotides. The N-terminal region of RNase E contains the catalytic site. Within the conserved N-terminal domain of RNAse E and RNase G, there is an S1-like subdomain, which is an ancient single-stranded RNA-binding domain. S1 domain is an RNA-binding module originally identified in the ribosomal protein S1. The S1 domain is required for RNA cleavage by RNase E. RNase G is paralogous to RNase E with an N-terminal catalytic domain that is highly homologous to that of RNase E. RNase G not only shares sequence similarity with RNase E, but also functionally overlaps with RNase E. In Escherichia coli, RNase G is involved in the maturation of the 5' end of the 16S rRNA. RNase G plays a secondary role in mRNA decay.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>putative RNA-processing protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>S1_pNO40: pNO40 , S1-like RNA-binding domain. pNO40 is a nucleolar protein of unknown function with an N-terminal S1 RNA binding domain, a CCHC type zinc finger, and clusters of basic amino acids representing a potential nucleolar targeting signal.  pNO40 was identified through a yeast two-hybrid interaction screen of a human kidney cDNA library using the pinin (pnn) protein as bait. pNO40 is thought to play a role in ribosome maturation and/or biogenesis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>S1_IF2_alpha: The alpha subunit of translation Initiation Factor 2, S1-like RNA-binding domain. S1-like RNA-binding domains are found in a wide variety of RNA-associated proteins. Eukaryotic and archaeal Initiation Factor 2 (e- and aIF2, respectively) are heterotrimeric proteins with three subunits (alpha, beta, and gamma). IF2 plays a crucial role in the process of translation initiation. The IF2 gamma subunit contains a GTP-binding site. The IF2 beta and gamma subunits together are thought to be responsible for binding methionyl-initiator tRNA. The ternary complex consisting of IF2, GTP, and the methionyl-initiator tRNA binds to the small subunit of the ribosome, as part of a pre-initiation complex that scans the mRNA to find the AUG start codon. The IF2-bound GTP is hydrolyzed to GDP when the methionyl-initiator tRNA binds the AUG start codon, at which time the IF2 is released with its bound GDP. The large ribosomal subunit then joins with the small subunit to complete the initiation complex, which is competent to begin translation. The IF2a subunit is a major site of control of the translation initiation process, via phosphorylation of a specific serine residue. This alpha subunit is well conserved in eukaryotes and archaea but is not present in bacteria. IF2 is a cold-shock-inducible protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cr>23</td>
            <td id=al>Translation initiation factor 2, alpha subunit (eIF-2alpha)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>10</td>
            <td id=al>Transcriptional accessory protein Tex/SPT6 [Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>S1_Rrp5_repeat_sc12: Rrp5 is a trans-acting factor important for biogenesis of both the 40S and 60S eukaryotic ribosomal subunits. Rrp5 has two distinct regions, an N-terminal region containing tandemly repeated S1 RNA-binding domains (12 S1 repeats in Saccharomyces cerevisiae Rrp5 and 14 S1 repeats in Homo sapiens Rrp5) and a C-terminal region containing tetratricopeptide repeat (TPR) motifs thought to be involved in protein-protein interactions.  Mutational studies have shown that each region represents a specific functional domain. Deletions within the S1-containing region inhibit pre-rRNA processing at either site A3 or A2, whereas deletions within the TPR region confer an inability to support cleavage of A0-A2. This CD includes S. cerevisiae S1 repeat 12 (sc12). Rrp5 is found in eukaryotes but not in prokaryotes or archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>translation initiation factor IF-2 subunit alpha; Validated</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2606045</td></tr>
<tr><td id=al>end</td><td id=al>2608196</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>polyribonucleotide nucleotidyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
