<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05623</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05623 16S rRNA (cytidine(1402)-2'-O)-methyltransferase [Phormidium_A]</h3>
    <div>MAGLYLVGTPIGNLQDMTLRAIQTLKEVDLIAAEDTRHTGKLLQHFQVDTPLISYHHHNRHERLGELLSRLQGGTAIALVTDAGMPGISDPGYELVQACAAAGIQVTPIPGVSACVTAVSVAGIEGDRFCFEGFLPIKGKERQERLEQLQNEPRLLVFYEAPHRLRQTLEDLQTVLGEERPVTLGRELTKRFEEIWRGTLAGAIAHYQNHDPKGEYTLVIGGCPSPSRQFSEAALKSELQALMEQGHSRSQASRLLAQSTPLSRRDIYQIALTLTD</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase RsmI</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>97</td>
            <td id=cg>92</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cr>JAAUSF01</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>98</td>
            <td id=cg>96</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cr>JAAUTN01</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>98</td>
            <td id=cg>96</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cr>FACHB-1407</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>97</td>
            <td id=cg>94</td>
            <td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td>
            <td id=cr>Elainella</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>54</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>COG0313 16S rRNA C1402 (ribose-2'-O) methylase RsmI [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=al>cd11648 Ribosomal RNA small subunit methyltransferase I (RsmI), also known as rRNA (cytidine-2'-O-)-methyltransferase. RsmI is an S-AdoMet (S-adenosyl-L-methionine or SAM)-dependent methyltransferase responsible for the 2'-O-methylation of cytidine 1402 (C1402) at the P site of bacterial 16S rRNA. Another S-AdoMet-dependent methyltransferase, RsmH (not included in this family), is responsible for N4-methylation at C1402. These methylation reactions may occur at a late step during 30S assembly in the cell. The dimethyl modification is believed to be conserved in bacteria, may play a role in fine-tuning the shape and functions of the P-site to increase the translation fidelity, and has been shown for Staphylococcus aureus, to contribute to virulence in host animals by conferring resistance to oxidative stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=al>cd19917 tetrapyrrole methylase family protein similar to ribosomal RNA small subunit methyltransferase I (RsmI). RsmI, also known as rRNA (cytidine-2'-O-)-methyltransferase, is an S-AdoMet (S-adenosyl-L-methionine or SAM)-dependent methyltransferase responsible for the 2'-O-methylation of cytidine 1402 (C1402) at the P site of bacterial 16S rRNA. Another S-AdoMet-dependent methyltransferase, RsmH (not included in this family), is responsible for N4-methylation at C1402. These methylation reactions may occur at a late step during 30S assembly in the cell. The dimethyl modification is believed to be conserved in bacteria, may play a role in fine-tuning the shape and functions of the P-site to increase the translation fidelity, and has been shown for Staphylococcus aureus, to contribute to virulence in host animals by conferring resistance to oxidative stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>98</td>
            <td id=cg>94</td>
            <td id=al>PRK14994 SAM-dependent 16S ribosomal RNA C1402 ribose 2'-O-methyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cb>77</td>
            <td id=cg>99</td>
            <td id=al>cd19918 uncharacterized subfamily of the tetrapyrrole methylase family similar to Ribosomal RNA small subunit methyltransferase I (RsmI). RsmI, also known as rRNA (cytidine-2'-O-)-methyltransferase, is an S-AdoMet (S-adenosyl-L-methionine or SAM)-dependent methyltransferase responsible for the 2'-O-methylation of cytidine 1402 (C1402) at the P site of bacterial 16S rRNA. Another S-AdoMet-dependent methyltransferase, RsmH (not included in this family), is responsible for N4-methylation at C1402. These methylation reactions may occur at a late step during 30S assembly in the cell. The dimethyl modification is believed to be conserved in bacteria, may play a role in fine-tuning the shape and functions of the P-site to increase the translation fidelity, and has been shown for Staphylococcus aureus, to contribute to virulence in host animals by conferring resistance to oxidative stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>TIGR00096 16S rRNA (cytidine(1402)-2'-O)-methyltransferase. This protein, previously known as YraL, is RsmI, one of a pair of genes involved in a unique dimethyl modification of a cytidine in 16S rRNA. See pfam00590 (tetrapyrrole methylase), which demonstrates homology between this family and other members, including several methylases for the tetrapyrrole class of compound, as well as the enzyme diphthine synthase. [Protein synthesis, tRNA and rRNA base modification]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cb>74</td>
            <td id=cg>98</td>
            <td id=al>pfam00590 Tetrapyrrole (Corrin/Porphyrin) Methylases. This family uses S-AdoMet in the methylation of diverse substrates. This family includes a related group of bacterial proteins of unknown function. This family includes the methylase Dipthine synthase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cb>75</td>
            <td id=cg>91</td>
            <td id=al>cd11649 uncharacterized subfamily of the tetrapyrrole methylase family similar to Ribosomal RNA small subunit methyltransferase I (RsmI). RsmI, also known as rRNA (cytidine-2'-O-)-methyltransferase, is an S-AdoMet (S-adenosyl-L-methionine or SAM)-dependent methyltransferase responsible for the 2'-O-methylation of cytidine 1402 (C1402) at the P site of bacterial 16S rRNA. Another S-AdoMet-dependent methyltransferase, RsmH (not included in this family), is responsible for N4-methylation at C1402. These methylation reactions may occur at a late step during 30S assembly in the cell. The dimethyl modification is believed to be conserved in bacteria, may play a role in fine-tuning the shape and functions of the P-site to increase the translation fidelity, and has been shown for Staphylococcus aureus, to contribute to virulence in host animals by conferring resistance to oxidative stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cb>79</td>
            <td id=cg>100</td>
            <td id=al>cd09815 S-AdoMet-dependent tetrapyrrole methylases. This superfamily uses S-AdoMet (S-adenosyl-L-methionine or SAM) in the methylation of diverse substrates. Most members catalyze various methylation steps in cobalamin (vitamin B12) biosynthesis. There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. The enzymes involved in the aerobic pathway are prefixed Cob and those of the anaerobic pathway Cbi. Most of the enzymes are shared by both pathways and a few enzymes are pathway-specific. Diphthine synthase and ribosomal RNA small subunit methyltransferase I (RsmI) are two superfamily members that are not involved in cobalamin biosynthesis. Diphthine synthase participates in the posttranslational modification of a specific histidine residue in elongation factor 2 (EF-2) of eukaryotes and archaea to diphthamide. RsmI catalyzes the 2-O-methylation of the ribose of cytidine 1402 (C1402) in 16S rRNA. Other superfamily members not involved in cobalamin biosynthesis include the N-terminal tetrapyrrole methylase domain of Bacillus subtilis YabN whose specific function is unknown, and Omphalotus olearius omphalotin methyltransferase which catalyzes the automethylation of its own C-terminus; this C terminus is subsequently released and macrocyclized to give Omphalotin A, a potent nematicide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>84</td>
            <td id=cg>95</td>
            <td id=al>COG0007 Uroporphyrinogen-III methylase (siroheme synthase) [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>cd11724 uncharacterized family of the tetrapyrrole methylase superfamily. Members of this superfamily use S-AdoMet (S-adenosyl-L-methionine or SAM) in the methylation of diverse substrates. Most members catalyze various methylation steps in cobalamin (vitamin B12) biosynthesis. There are two distinct cobalamin biosynthetic pathways in bacteria. The aerobic pathway requires oxygen, and cobalt is inserted late in the pathway; the anaerobic pathway does not require oxygen, and cobalt insertion is the first committed step towards cobalamin synthesis. The enzymes involved in the aerobic pathway are prefixed Cob and those of the anaerobic pathway Cbi. Most of the enzymes are shared by both pathways and a few enzymes are pathway-specific. Diphthine synthase and Ribosomal RNA small subunit methyltransferase I (RsmI) are two superfamily members that are not involved in cobalamin biosynthesis. Diphthine synthase participates in the posttranslational modification of a specific histidine residue in elongation factor 2 (EF-2) of eukaryotes and archaea to diphthamide. RsmI catalyzes the 2-O-methylation of the ribose of cytidine 1402 (C1402) in 16S rRNA. Other superfamily members not involved in cobalamin biosynthesis include the N-terminal tetrapyrrole methylase domain of Bacillus subtilis YabN whose specific function is unknown, and Omphalotus olearius omphalotin methyltransferase which catalyzes the automethylation of its own C-terminus; this C terminus is subsequently released and macrocyclized to give Omphalotin A, a potent nematicide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=al>cd11642 Uroporphyrin-III C-methyltransferase (also known as S-Adenosyl-L-methionine:uroporphyrinogen III methyltransferase, SUMT). SUMT is an enzyme of the cobalamin and siroheme biosynthetic pathway. It catalyzes the first of three steps leading to the formation of siroheme from uroporphyrinogen III; it transfers two methyl groups from S-adenosyl-L-methionine to the C-2 and C-7 atoms of uroporphyrinogen III to yield precorrin-2 via the intermediate formation of precorrin-1. Precorrin-2 is also a precursor for the biosynthesis of vitamin B12, coenzyme F430, siroheme and heme d1. This family includes proteins in which the SUMT domain is fused to other functional domains, such as to a uroporphyrinogen-III synthase domain to form bifunctional uroporphyrinogen-III methylase/uroporphyrinogen-III synthase, or to a dual function dehydrogenase-chelatase domain, as in the case of the multifunctional S-adenosyl-L-methionine (SAM)-dependent bismethyltransferase/dehydrogenase/ferrochelatase CysG, which catalyzes all three steps that transform uroporphyrinogen III into siroheme.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4592698</td></tr>
<tr><td id=al>end</td><td id=al>4593529</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>16S rRNA (cytidine(1402)-2'-O)-methyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>277</td></tr>
</table>

    </div>
</body>
</html>
