<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00187</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00187 GTPase HflX [Phormidium_A]</h3>
    <div>MAAISTDLGKPVSTYLNRRGQVIRVGVGSPRQTQIPPLELPRYGASRLSGIRCIATQLKRDQPGTATLTAMAIQRLDALICLTLTGSGFQRRGGGETGYVHRTYLAHLVPNPSEASWTVSEEMTLEAIAQQDFLDLVEALEGEFEREFVGQSVDSDQDRVLLVGLRTQNTSDVEFEEHLAEVVRLVDTAGGVVLQTIQQGRSRPHPQTVIGSGKVDELALAVQTLGATLVVFDRDLSPAQVRNLEKRLGVRVVDRTEVILDIFAQRAQSRAGKLQVELAQLEYSLPRLRGQGQKMSRLGGGIGTRGPGETQLETERRAIAQRISRLQREVTNLQAHRARMRQQRQAQDVPSIALVGYTNAGKSTLLNVLANSEIYTADQLFATLDPTTRRLSIQEDVTHVVHQLVLTDTVGFIHELPPALVDAFRATLEEVTEADALLHVVDLSHAAWQNQIHSVMGILSEMPITPGPILLVFNKIDSVDGDTLELAKEEYPQATFISATAGFGLATLRQRLLQLVDYTKH</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>GTPase HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>GTPase HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>GTP-binding protein HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>GTPase HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>GTPase HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>GTPase HflX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>GTPase HflX</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>GTPase HflX</td>
            <td id=co>FACHB-40</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>GTPase HflX</td>
            <td id=cr>JAAUTJ01</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>GTPase HflX</td>
            <td id=cr>FACHB-46</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein HflX. This protein family is one of a number of homologous small, well-conserved GTP-binding proteins with pleiotropic effects. Bacterial members are designated HflX, following the naming convention in Escherichia coli where HflX is encoded immediately downstream of the RNA chaperone Hfq, and immediately upstream of HflKC, a membrane-associated protease pair with an important housekeeping function. Over large numbers of other bacterial genomes, the pairing with hfq is more significant than with hflK and hlfC. The gene from Homo sapiens in this family has been named PGPL (pseudoautosomal GTP-binding protein-like). [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>50S ribosomal subunit-associated GTPase HflX [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>HflX GTPase family. HflX subfamily. A distinct conserved domain with a glycine-rich segment N-terminal of the GTPase domain characterizes the HflX subfamily. The E. coli HflX has been implicated in the control of the lambda cII repressor proteolysis, but the actual biological functions of these GTPases remain unclear. HflX is widespread, but not universally represented in all three superkingdoms.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>GTPase HflX; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding GTPase N-terminal. This is the N-terminal region of GTP-binding HflX-like proteins. The full-length members bind and interact with the 50S ribosome and are GTPases, hydrolysing GTP/GDP/ATP/ADP. This N-terminal region is necessary for stability of the whole protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>50S ribosome-binding GTPase. The full-length GTPase protein is required for the complete activity of the protein of interacting with the 50S ribosome and binding of both adenine and guanine nucleotides, with a preference for guanine nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding GTPase Middle Region. This family locates between the N-terminal domain and MMR_HSR1 50S ribosome-binding GTPase of GTP-binding HflX-like proteins. The full-length members bind and interact with the 50S ribosome and are GTPases, hydrolysing GTP/GDP/ATP/ADP. This region is unknown for its function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>E. coli Ras-like protein (Era)-like GTPase. The Era (E. coli Ras-like protein)-like family includes several distinct subfamilies (TrmE/ThdF, FeoB, YihA (EngB), Era, and EngA/YfgK) that generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. TrmE is ubiquitous in bacteria and is a widespread mitochondrial protein in eukaryotes, but is absent from archaea. The yeast member of TrmE family, MSS1, is involved in mitochondrial translation; bacterial members are often present in translation-related operons. FeoB represents an unusual adaptation of GTPases for high-affinity iron (II) transport. YihA (EngB) family of GTPases is typified by the E. coli YihA, which is an essential protein involved in cell division control. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain. EngA and its orthologs are composed of two GTPase domains and, since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>E. coli Ras-like protein (Era) is a multifunctional GTPase. Era (E. coli Ras-like protein) is a multifunctional GTPase found in all bacteria except some eubacteria. It binds to the 16S ribosomal RNA (rRNA) of the 30S subunit and appears to play a role in the assembly of the 30S subunit, possibly by chaperoning the 16S rRNA. It also contacts several assembly elements of the 30S subunit. Era couples cell growth with cytokinesis and plays a role in cell division and energy metabolism. Homologs have also been found in eukaryotes. Era contains two domains: the N-terminal GTPase domain and a C-terminal domain KH domain that is critical for RNA binding. Both domains are important for Era function. Era is functionally able to compensate for deletion of RbfA, a cold-shock adaptation protein that is required for efficient processing of the 16S rRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Rat sarcoma (Ras)-like superfamily of small guanosine triphosphatases (GTPases). Ras-like GTPase superfamily. The Ras-like superfamily of small GTPases consists of several families with an extremely high degree of structural and functional similarity. The Ras superfamily is divided into at least four families in eukaryotes: the Ras, Rho, Rab, and Sar1/Arf families. This superfamily also includes proteins like the GTP translation factors, Era-like GTPases, and G-alpha chain of the heterotrimeric G proteins. Members of the Ras superfamily regulate a wide variety of cellular functions: the Ras family regulates gene expression, the Rho family regulates cytoskeletal reorganization and gene expression, the Rab and Sar1/Arf families regulate vesicle trafficking, and the Ran family regulates nucleocytoplasmic transport and microtubule organization. The GTP translation factor family regulates initiation, elongation, termination, and release in translation, and the Era-like GTPase family regulates cell division, sporulation, and DNA replication. Members of the Ras superfamily are identified by the GTP binding site, which is made up of five characteristic sequence motifs, and the switch I and switch II regions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>GTPase Era, involved in 16S rRNA processing [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>GTPase Era; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>tRNA U34 5-carboxymethylaminomethyl modifying GTPase MnmE/TrmE [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Obg-like family of GTPases consist of five subfamilies: Obg, DRG, YyaF/YchF, Ygr210, and NOG1. The Obg-like subfamily consists of five well-delimited, ancient subfamilies, namely Obg, DRG, YyaF/YchF, Ygr210, and NOG1. Four of these groups (Obg, DRG, YyaF/YchF, and Ygr210) are characterized by a distinct glycine-rich motif immediately following the Walker B motif (G3 box). Obg/CgtA is an essential gene that is involved in the initiation of sporulation and DNA replication in the bacteria Caulobacter and Bacillus, but its exact molecular role is unknown. Furthermore, several OBG family members possess a C-terminal RNA-binding domain, the TGS domain, which is also present in threonyl-tRNA synthetase and in bacterial guanosine polyphosphatase SpoT. Nog1 is a nucleolar protein that might function in ribosome assembly. The DRG and Nog1 subfamilies are ubiquitous in archaea and eukaryotes, the Ygr210 subfamily is present in archaea and fungi, and the Obg and YyaF/YchF subfamilies are ubiquitous in bacteria and eukaryotes. The Obg/Nog1 and DRG subfamilies appear to form one major branch of the Obg family and the Ygr210 and YchF subfamilies form another branch. No GEFs, GAPs, or GDIs for Obg have been identified.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>EngA1 GTPase contains the first domain of EngA. This EngA1 subfamily CD represents the first GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>YfjP GTPase. The Era (E. coli Ras-like protein)-like YfjP subfamily includes several uncharacterized bacterial GTPases that are similar to Era. They generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>152929</td></tr>
<tr><td id=al>end</td><td id=al>154495</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>GTPase HflX</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
