<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02049</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02049 SDR family oxidoreductase [Phormidium_A]</h3>
    <div>MNIAILGCGYVGVAAAQHWQAAGHHVTATTTSPERVKELEAIASRVEVIQGNNPDDLKRLLQGQDTLVMSVASKGRAPYKSVYLETAQTLVSVLPEAPELRQIIYTGSGSVYGNWGGGWVNETTPPQPTMENYEILVETEQTLLNAAHDDLRVCILRLGGIYGPGREIARIYRNWAGQTRPGDGNRPSNWIHLDDIVGALAFAQDKRLEGIYNLVDDDHQPTRAVVEAVCQTYQLPPVTWDGSEPATVRYSGKLSNQKLKDAGYQFIHPQRTI</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>nucleoside-diphosphate-sugar epimerase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>Desertifilum</td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>Lusitaniella</td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>Hormoscilla</td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>SDR family oxidoreductase</td>
            <td id=co>Hormoscilla</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>atypical (a) SDRs, subgroup 4. Atypical SDRs in this subgroup are poorly defined, one member is identified as a putative NAD-dependent epimerase/dehydratase. Atypical SDRs are distinct from classical SDRs. Members of this subgroup have a glycine-rich NAD(P)-binding motif that is related to, but is different from, the archetypical SDRs, GXGXXG. This subgroup also lacks most of the characteristic active site residues of the SDRs; however, the upstream Ser is present at the usual place, and some potential catalytic residues are present in place of the usual YXXXK active site motif. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Nucleoside-diphosphate-sugar epimerase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>atypical (a) SDRs, subgroup 1. Atypical SDRs in this subgroup are poorly defined and have been identified putatively as isoflavones reductase, sugar dehydratase, mRNA binding protein etc. Atypical SDRs are distinct from classical SDRs. Members of this subgroup retain the canonical active site triad (though not the upstream Asn found in most SDRs) but have an unusual putative glycine-rich NAD(P)-binding motif, GGXXXXG, in the usual location. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>atypical (a) SDRs, subgroup 8. This subgroup contains atypical SDRs of unknown function. Proteins in this subgroup have a glycine-rich NAD(P)-binding motif consensus that resembles that of the extended SDRs, (GXXGXXG or GGXGXXG), but lacks the characteristic active site residues of the SDRs. A Cys often replaces the usual Lys of the YXXXK active site motif, while the upstream Ser is generally present and Arg replaces the usual Asn. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>extended (e) SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>TIGR01777 family protein. This model represents a clade of proteins of unknown function including the E. coli yfcH protein. [Hypothetical proteins, Conserved]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Extended (e) and atypical (a) SDRs. Extended or atypical short-chain dehydrogenases/reductases (SDRs, aka tyrosine-dependent oxidoreductases) are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>biliverdin IX beta reductase (BVR-B, aka flavin reductase)-like proteins; atypical (a) SDRs. Human BVR-B catalyzes pyridine nucleotide-dependent production of bilirubin-IX beta during fetal development; in the adult BVR-B has flavin and ferric reductase activities. Human BVR-B catalyzes the reduction of FMN, FAD, and riboflavin. Recognition of flavin occurs mostly by hydrophobic interactions, accounting for the broad substrate specificity. Atypical SDRs are distinct from classical SDRs. BVR-B does not share the key catalytic triad, or conserved tyrosine typical of SDRs. The glycine-rich NADP-binding motif of BVR-B is GXXGXXG, which is similar but not identical to the pattern seen in extended SDRs. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>UDP-glucose 4-epimerase (G4E), subgroup 5, extended (e) SDRs. This subgroup partially conserves the characteristic active site tetrad and NAD-binding motif of the extended SDRs, and has been identified as possible UDP-glucose 4-epimerase (aka UDP-galactose 4-epimerase), a homodimeric member of the extended SDR family. UDP-glucose 4-epimerase catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>3beta-hydroxysteroid dehydrogenases (3b-HSD)-like, extended (e) SDRs. Extended SDR family domains belonging to this subgroup have the characteristic active site tetrad and a fairly well-conserved NAD(P)-binding motif. 3b-HSD catalyzes the NAD-dependent conversion of various steroids, such as pregnenolone to progesterone, or androstenediol to testosterone. This subgroup includes an unusual bifunctional 3b-HSD/C-4 decarboxylase from Arabidopsis thaliana, and Saccharomyces cerevisiae ERG26, a 3b-HSD/C-4 decarboxylase, involved in the synthesis of ergosterol, the major sterol of yeast. It also includes human 3 beta-HSD/HSD3B1 and C(27) 3beta-HSD/ [3beta-hydroxy-delta(5)-C(27)-steroid oxidoreductase; HSD3B7].  C(27) 3beta-HSD/HSD3B7 is a membrane-bound enzyme of the endoplasmic reticulum, that catalyzes the isomerization and oxidation of 7alpha-hydroxylated sterol intermediates, an early step in bile acid biosynthesis. Mutations in the human NSDHL (NAD(P)H steroid dehydrogenase-like protein) cause CHILD syndrome (congenital hemidysplasia with ichthyosiform nevus and limb defects), an X-linked dominant, male-lethal trait. Mutations in the human gene encoding C(27) 3beta-HSD underlie a rare autosomal recessive form of neonatal cholestasis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid sythase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>atypical (a) SDRs, subgroup 3. These atypical SDR family members of unknown function have a glycine-rich NAD(P)-binding motif consensus that is very similar to the extended SDRs, GXXGXXG.  Generally, this group has poor conservation of the active site tetrad, However, individual sequences do contain matches to the YXXXK active site motif, and generally Tyr or Asn in place of the upstream Ser found in most SDRs. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Atypical SDRs include biliverdin IX beta reductase (BVR-B,aka flavin reductase), NMRa (a negative transcriptional regulator of various fungi), progesterone 5-beta-reductase like proteins, phenylcoumaran benzylic ether and pinoresinol-lariciresinol reductases, phenylpropene synthases, eugenol synthase, triphenylmethane reductase, isoflavone reductases, and others. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. In addition to the Rossmann fold core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids, and typically have a TGXXGXXG cofactor binding motif. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>UDP-glucose 4 epimerase (G4E), subgroup 3, extended (e) SDRs. Members of this bacterial subgroup are identified as possible sugar epimerases, such as UDP-glucose 4 epimerase. However, while the NAD(P)-binding motif is fairly well conserved, not all members retain the canonical active site tetrad of the extended SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>UDP-glucose 4 epimerase, subgroup 4, extended (e) SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup is comprised of bacterial proteins, and includes the Staphylococcus aureus capsular polysaccharide Cap5N, which may have a role in the synthesis of UDP-N-acetyl-d-fucosamine. This subgroup has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NAD dependent epimerase/dehydratase family. This family of proteins utilize NAD as a cofactor. The proteins in this family use nucleotide-sugar substrates for a variety of chemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>NAD(P)H-binding. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>UDP-glucose 4 epimerase, subgroup 2, extended (e) SDRs. UDP-glucose 4 epimerase (aka UDP-galactose-4-epimerase), is a homodimeric extended SDR. It catalyzes the NAD-dependent conversion of UDP-galactose to UDP-glucose, the final step in Leloir galactose synthesis. This subgroup is comprised of archaeal and bacterial proteins, and has the characteristic active site tetrad and NAD-binding motif of the extended SDRs. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>52</td>
            <td id=al>uncharacterized subgroup of aldehyde reductase and flavonoid reductase related proteins, extended (e) SDRs. This subgroup contains proteins of unknown function related to aldehyde reductase and flavonoid reductase of the extended SDR-type. Aldehyde reductase I (aka carbonyl reductase) is an NADP-binding SDR; it has an NADP-binding motif consensus that is slightly different from the canonical SDR form and lacks the Asn of the extended SDR active site tetrad. Aldehyde reductase I catalyzes the NADP-dependent  reduction of ethyl 4-chloro-3-oxobutanoate to ethyl (R)-4-chloro-3-hydroxybutanoate. The related flavonoid reductases act in the NADP-dependent reduction of  flavonoids, ketone-containing plant secondary metabolites. Extended SDRs are distinct from classical SDRs. In addition to the Rossmann fold (alpha/beta folding pattern with a central beta-sheet) core region typical of all SDRs, extended SDRs have a less conserved C-terminal extension of approximately 100 amino acids. Extended SDRs are a diverse collection of proteins, and include isomerases, epimerases, oxidoreductases, and lyases; they typically have a TGXXGXXG cofactor binding motif. SDRs are a functionally diverse family of oxidoreductases that have a single domain with a structurally conserved Rossmann fold, an NAD(P)(H)-binding region, and a structurally diverse C-terminal region. Sequence identity between different SDR enzymes is typically in the 15-30% range; they catalyze a wide range of activities including the metabolism of steroids, cofactors, carbohydrates, lipids, aromatic compounds, and amino acids, and act in redox sensing. Classical SDRs have an TGXXX[AG]XG cofactor binding motif and a YXXXK active site motif, with the Tyr residue of the active site motif serving as a critical catalytic residue (Tyr-151, human 15-hydroxyprostaglandin dehydrogenase numbering). In addition to the Tyr and Lys, there is often an upstream Ser and/or an Asn, contributing to the active site; while substrate binding is in the C-terminal region, which determines specificity. The standard reaction mechanism is a 4-pro-S hydride transfer and proton relay involving the conserved Tyr and Lys, a water molecule stabilized by Asn, and nicotinamide. Atypical SDRs generally lack the catalytic residues characteristic of the SDRs, and their glycine-rich NAD(P)-binding motif is often different from the forms normally seen in classical or extended SDRs. Complex (multidomain) SDRs such as ketoreductase domains of fatty acid synthase have a GGXGXXG NAD(P)-binding motif and an altered active site motif (YXXXN). Fungal type ketoacyl reductases have a TGXXXGX(1-2)G NAD(P)-binding motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>Putative NADH-flavin reductase  [General function prediction only]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1734814</td></tr>
<tr><td id=al>end</td><td id=al>1735636</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>SDR family oxidoreductase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
