<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01129</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01129 methionine--tRNA ligase [Phormidium_A]</h3>
    <div>MSPNPIQHQTFALTTPLYYVNALPHIGSAYTTVAADVVARFQRLLGHSVSFVTGTDEHGQKIQRTAQEKERDPQAYCDEVVACFEDLWQKLNIQYDSFSRTTASRHEAIVREFFQRVWDNGDIYLGQQKGWYCVACEEFKDEGELLEDHYCPLHPNRQVEWRDEENYFFRLSKYQPALEDLYKNQPDFIQPTMRRNEVLSFVQGGLQDFSISRVNMDWGFPVPENPEHTLYVWFDALLGYITGLLQPEDEPTLERVLERGYPFQLHLIGKDILRFHAVYWPAMLLSAGMPLPERVFGHGFLTKDGHKISKTLGNTVDPVDLVERFGSDAFRYYFMKEIEFGRDGDYNETRFINTINADLANDLGNLLNRTLGMARKYCGGSVPNLDVAGISPENPLKALGDSLTPQVRSAYESLAFHQACGASLSLVRAGNKYIDDRAPWSLYKQGEQAAVEEVLYSVLESVRLVAYLLSPVVPHLSTAIYGQLGYSTDFDQPQSVRETAPFEHHGTWGALPPGQSLNKPEPVFRRIEVPSA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>methionyl-tRNA synthetase MetG</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=co>Mastigocladopsis</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=co>Nostoc</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>methionine--tRNA ligase</td>
            <td id=co>Nostoc</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>methionyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Methionyl-tRNA synthetase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>methionine-tRNA ligase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionine--tRNA ligase. The methionyl-tRNA synthetase (metG) is a class I amino acyl-tRNA ligase. This model appears to recognize the methionyl-tRNA synthetase of every species, including eukaryotic cytosolic and mitochondrial forms. The UPGMA difference tree calculated after search and alignment according to this model shows an unusual deep split between two families of MetG. One family contains forms from the Archaea, yeast cytosol, spirochetes, and E. coli, among others. The other family includes forms from yeast mitochondrion, Synechocystis sp., Bacillus subtilis, the Mycoplasmas, Aquifex aeolicus, and Helicobacter pylori. The E. coli enzyme is homodimeric, although monomeric forms can be prepared that are fully active. Activity of this enzyme in bacteria includes aminoacylation of fMet-tRNA with Met; subsequent formylation of the Met to fMet is catalyzed by a separate enzyme. Note that the protein from Aquifex aeolicus is split into an alpha (large) and beta (small) subunit; this model does not include the C-terminal region corresponding to the beta chain. [Protein synthesis, tRNA aminoacylation]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>catalytic core domain of methioninyl-tRNA synthetases. Methionine tRNA synthetase (MetRS) catalytic core domain. This class I enzyme aminoacylates the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. MetRS, which consists of the core domain and an anti-codon binding domain, functions as a monomer. However, in some species the anti-codon binding domain is followed by an EMAP domain. In this case, MetRS functions as a homodimer. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding.  As a result of a deletion event, MetRS has a significantly shorter core domain insertion than IleRS, ValRS, and LeuR.  Consequently, the MetRS insertion lacks the editing function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>tRNA synthetases class I (M). This family includes methionyl tRNA synthetases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>methionyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>catalytic core domain of isoleucyl, leucyl, valyl and methioninyl tRNA synthetases. Catalytic core domain of isoleucyl, leucyl, valyl and methioninyl tRNA synthetases. These class I enzymes are all monomers. However, in some species, MetRS functions as a homodimer, as a result of an additional C-terminal domain. These enzymes aminoacylate the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding.  Enzymes in this subfamily share an insertion in the core domain, which is subject to both deletions and rearrangements. This editing region hydrolyzes mischarged cognate tRNAs and thus prevents the incorporation of chemically similar amino acids. MetRS has a significantly shorter insertion, which lacks the editing function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>probable methionyl-tRNA synthetase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Anticodon-binding domain of methionyl tRNA synthetases. This domain is found in methionyl tRNA synthetases (MetRS), which belong to the class Ia aminoacyl tRNA synthetases. It lies C-terminal to the catalytic core domain, and recognizes and specifically binds to the tRNA anticodon (CAU). MetRS catalyzes the transfer of methionine to the 3'-end of its tRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>catalytic core domain of valyl-tRNA synthetases. Valine amino-acyl tRNA synthetase (ValRS) catalytic core domain. This enzyme is a monomer which aminoacylates the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding.  ValRS has an insertion in the core domain, which is subject to both deletions and rearrangements. This editing region hydrolyzes mischarged cognate tRNAs and thus prevents the incorporation of chemically similar amino acids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>catalytic core domain of isoleucyl-tRNA synthetases. Isoleucine amino-acyl tRNA synthetases (IleRS) catalytic core domain . This class I enzyme is a monomer which aminoacylates the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding.  IleRS has an insertion in the core domain, which is subject to both deletions and rearrangements. This editing region hydrolyzes mischarged cognate tRNAs and thus prevents the incorporation of chemically similar amino acids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>catalytic core domain of leucyl-tRNA synthetases. Leucyl tRNA synthetase (LeuRS) catalytic core domain. This class I enzyme is a monomer which aminoacylates the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding. In Aquifex aeolicus, the gene encoding LeuRS is split in two, just before the KMSKS motif. Consequently, LeuRS is a heterodimer, which likely superimposes with the LeuRS monomer found in most other organisms. LeuRS has an insertion in the core domain, which is subject to both deletions and rearrangements and thus differs between prokaryotic LeuRS and archaeal/eukaryotic LeuRS. This editing region hydrolyzes mischarged cognate tRNAs and thus prevents the incorporation of chemically similar amino acids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>Valyl-tRNA synthetase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>Isoleucyl-tRNA synthetase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>valyl-tRNA synthetase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>52</td>
            <td id=al>catalytic core domain of class I amino acyl-tRNA synthetase. Class I amino acyl-tRNA synthetase (aaRS) catalytic core domain. These enzymes are mostly monomers which aminoacylate the 2'-OH of the nucleotide at the 3' of the appropriate tRNA. The core domain is based on the Rossman fold and is responsible for the ATP-dependent formation of the enzyme bound aminoacyl-adenylate. It contains the characteristic class I HIGH and KMSKS motifs, which are involved in ATP binding.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>24</td>
            <td id=al>tRNA synthetases class I (I, L, M and V). Other tRNA synthetase sub-families are too dissimilar to be included.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>38</td>
            <td id=al>tRNA synthetases class I (I, L, M and V). Other tRNA synthetase sub-families are too dissimilar to be included.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>valyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>valyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>27</td>
            <td id=al>leucyl-tRNA synthetase; Reviewed</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>936096</td></tr>
<tr><td id=al>end</td><td id=al>937695</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>methionine--tRNA ligase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
