<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02900</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02900 glucose-1-phosphate thymidylyltransferase [Phormidium_A]</h3>
    <div>MKALILSGGKGTRLRPLTYTGAKQLVPVANKPILWYGIESIVSAGITEIGIIISPETGEEVQKLTGDGDRFGANITYILQDEPAGLAHAVKIAQPFLGDSPFVMYLGDNLIQNDLTPFLNHFREKQLDALILLRPVENPSAFGVAKVDDSGRVLELVEKPAVPPSNLALVGIYYFSAEVHQAIARIKPSERGELEITDAISNLLDHKNPVEAQILEGWWLDTGKKDDLLSANQIVLDTQLQTAIFGELDADSQVVGRVQIGQGTQLINCSVRGPVIIGDNCHLENCFIGPYSSIADGVILKDVDLEHSVILQETRIEGIQQRIVDSLIGRRAQLGTGRPRPKALRFMVGDDCQIELA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase RmlA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>TPA: glucose-1-phosphate thymidylyltransferase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cr>JSC-12</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylylransferase, long form</td>
            <td id=cr>JSC-12</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase</td>
            <td id=cr>Stenomitos</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate thymidylylransferase, long form. The family of known and putative glucose-1-phosphate thymidyltransferase (also called dTDP-glucose synthase) shows a deep split into a short form (see TIGR01207) and a long form described by this model. The homotetrameric short form is found in numerous bacterial species that incorporate dTDP-L-rhamnose, which it helps synthesize, into the cell wall. It is subject to feedback inhibition. This form, in contrast, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced. Alternate name: dTDP-D-glucose synthase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>dTDP-glucose pyrophosphorylase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>G1P_TT_long represents the long form of glucose-1-phosphate thymidylyltransferase. This family is the long form of Glucose-1-phosphate thymidylyltransferase.  Glucose-1-phosphate thymidylyltransferase catalyses the formation of dTDP-glucose, from dTTP and glucose 1-phosphate. It is the first enzyme in the biosynthesis of dTDP-L-rhamnose, a cell wall constituent and a feedback inhibitor of the enzyme.There are two forms of   Glucose-1-phosphate thymidylyltransferase in bacteria and archeae; short form and long form.  The long form, which has an extra 50 amino acids c-terminal, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced.The long from enzymes also have a left-handed parallel helix domain at the c-terminus, whereas, th eshort form enzymes do not have this domain. The homotetrameric, feedback inhibited short form is found in numerous bacterial species that produce dTDP-L-rhamnose.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase. The MJ_1101 protein from Methanococcus jannaschii has been characterized as the GlmU enzyme catalyzing the final two steps of UDP-GlcNAc biosynthesis. Many of the genes identified by this model are in proximity to the GlmS and GlmM genes and are also presumed to be GlmU. However, some archaeal genomes contain multiple closely-related homologs from this family and it is not clear what the substrate specificity is for each of them.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>G1P_TT_short is the short form of glucose-1-phosphate thymidylyltransferase. This family is the short form of glucose-1-phosphate thymidylyltransferase.  Glucose-1-phosphate thymidylyltransferase catalyses the formation of dTDP-glucose, from dTTP and glucose 1-phosphate. It is the first enzyme in the biosynthesis of dTDP-L-rhamnose, a cell wall constituent and a feedback inhibitor of the enzyme.There are two forms of   Glucose-1-phosphate thymidylyltransferase in bacteria and archeae; short form and long form. The homotetrameric, feedback inhibited short form is found in numerous bacterial species that produce dTDP-L-rhamnose. The long form, which has an extra 50 amino acids c-terminal, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_transferases catalyze the transfer of nucleotides onto phosphosugars. Nucleotidyltransferases transfer nucleotides onto phosphosugars.  The enzyme family includes Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase. The products are activated sugars that are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Nucleotidyl transferase. This family includes a wide range of enzymes which transfer nucleotides onto phosphosugars.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NDP-sugar pyrophosphorylase, includes eIF-2Bgamma, eIF-2Bepsilon, and LPS biosynthesis proteins [Translation, ribosomal structure and biogenesis, Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Prokaryotic UGPase catalyses the synthesis of UDP-glucose. Prokaryotic UDP-Glucose Pyrophosphorylase (UGPase) catalyzes a reversible production of UDP-Glucose  and pyrophosphate (PPi) from glucose-1-phosphate and UTP.  UDP-glucose plays pivotal roles in galactose utilization, in glycogen synthesis, and in the synthesis of the carbohydrate moieties of glycolipids , glycoproteins , and proteoglycans. UGPase is found in both prokaryotes and eukaryotes, although prokaryotic and eukaryotic forms of UGPase catalyze the same reaction, they share low sequence similarity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase RfbA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>UTP-glucose-1-phosphate uridylyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-terminal domain of the M1P-guanylyltransferase B-isoform like proteins. GDP-mannose pyrophosphorylase  (GTP: alpha-d-mannose-1-phosphate guanyltransferase) catalyzes the formation of GDP-d-mannose from GTP and alpha-d-mannose-1-Phosphate. It contains an N-terminal catalytic domain and a C-terminal Lefthanded-beta-Helix fold domain. GDP-d-mannose is the activated form of mannose for formation of cell wall lipoarabinomannan and various mannose-containing glycolipids and polysaccharides. The function of GDP-mannose pyrophosphorylase is essential for cell wall integrity, morphogenesis and viability. Repression of GDP-mannose pyrophosphorylase in yeast leads to phenotypes, such as cell lysis, defective cell wall, and failure of polarized growth and cell separation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_trnasferase_like_2 is a member of the nucleotidyl transferase family. This is a subfamily of nucleotidyl transferases. Nucleotidyl transferases transfer nucleotides onto phosphosugars. The activated sugars are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides. Other subfamilies of nucleotidyl transferases include Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>WcbM_like is a subfamily of nucleotidyl transferases. WcbM protein of Burkholderia mallei is involved in the biosynthesis, export or translocation of capsule. It is a subfamily of nucleotidyl transferases that transfer nucleotides onto phosphosugars.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>UTP--glucose-1-phosphate uridylyltransferase GalF. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>UTP--glucose-1-phosphate uridylyltransferase GalU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_transferase_like_1 is a member of the nucleotidyl transferase family. This is a subfamily of nucleotidyl transferases. Nucleotidyl transferases transfer nucleotides onto phosphosugars. The activated sugars are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides. Other subfamilies of nucleotidyl transferases include Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>Bifunctional protein GlmU, N-acetylglucosamine-1-phosphate-uridyltransferase/glucosamine-1-phosphate-acetyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>N-terminal domain of bacterial GlmU. The N-terminal domain of N-Acetylglucosamine-1-phosphate uridyltransferase (GlmU). GlmU is an essential bacterial enzyme with both an acetyltransferase and an uridyltransferase activity which have been mapped to the C-terminal and N-terminal domains, respectively. This family represents the N-terminal uridyltransferase. GlmU performs the last two steps in the synthesis of UDP-N-acetylglucosamine (UDP-GlcNAc), which is an essential precursor in both the peptidoglycan and the lipopolysaccharide metabolic pathways in Gram-positive and Gram-negative bacteria, respectively.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>ADP-glucose pyrophosphorylase [Carbohydrate transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphocholine cytidylyltransferases catalyze the synthesis of CDP-choline. This family contains proteins similar to prokaryotic phosphocholine (P-cho) cytidylyltransferases. Phosphocholine (PC) cytidylyltransferases catalyze the transfer of a cytidine monophosphate from CTP to phosphocholine to form CDP-choline. PC is the most abundant phospholipid in eukaryotic membranes and it is also important in prokaryotic membranes. For pathogenic prokaryotes, the cell surface PC facilitates the interaction with host surface and induces attachment and invasion. In addition cell wall PC serves as scaffold for a group of choline-binding proteins that are secreted from the cells. Phosphocholine (PC) cytidylyltransferase is a key enzyme in the prokaryotic choline metabolism pathway. It has been hypothesized to consist of a choline transport system, a choline kinase, CTP:phosphocholine cytidylyltransferase, and a choline phosphotransferase that transfers P-Cho from CDP-Cho to either lipoteichoic acid or lipopolysaccharide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>The N-terminal domain of gamma subunit of the eIF-2B is a subfamily of glycosyltransferase 2. N-terminal domain of gamma subunit of the eukaryotic translation initiation factor 2B (eIF-2B): eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit gamma shares sequence similarity with epsilon subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>N-terminal domain of M1P_guanylyl_A_ like proteins are likely to be a isoform of GDP-mannose pyrophosphorylase. N-terminal domain of the M1P-guanylyltransferase A-isoform like proteins:  The proteins of this family are likely to be a isoform of GDP-mannose pyrophosphorylase. Their sequences are highly conserved with mannose-1-phosphate guanyltransferase, but  generally about 40-60 bases longer.  GDP-mannose pyrophosphorylase (GTP: alpha-d-mannose-1-phosphate guanyltransferase) catalyzes the formation of GDP-d-mannose from GTP and alpha-d-mannose-1-Phosphate. It contains an N-terminal catalytic domain that resembles a dinucleotide-binding Rossmann fold and a C-terminal LbH fold domain. GDP-d-mannose is the activated form of mannose for formation of cell wall lipoarabinomannan and various mannose-containing glycolipids and polysaccharides. The function of GDP-mannose pyrophosphorylase is essential for cell wall integrity, morphogenesis and viability.  Repression of GDP-mannose pyrophosphorylase in yeast leads to phenotypes including cell lysis, defective cell wall, and failure of polarized growth and cell separation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>GT2_BcbE_like is likely involved in the biosynthesis of the polysaccharide capsule. GT2_BcbE_like:  The bcbE gene is one of the genes in the capsule biosynthetic locus of Pasteurella multocida. Its deducted product is likely involved in the biosynthesis of the polysaccharide capsule, which is found on surface of a wide range of bacteria. It is a subfamily of Glycosyltransferase Family GT2, which includes diverse families of glycosyltransferases with a common GT-A type structural fold, which has two tightly associated beta/alpha/beta domains that tend to form a continuous central sheet of at least eight beta-strands. These are enzymes that catalyze the transfer of sugar moieties from activated donor molecules to specific acceptor molecules, forming glycosidic bonds.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>The N-terminal of eIF-2B_gamma_like is predicted to have glycosyltransferase activity. N-terminal domain of eEIF-2B epsilon and gamma, subunits of eukaryotic translation initiators, is a subfamily of glycosyltranferase 2 and is predicted to have glycosyltranferase activity. eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit epsilon shares sequence similarity with gamma subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>Putative glucose-1-phosphate thymidylyltransferase, C-terminal Left-handed parallel beta-Helix (LbH) domain: Proteins in this family show simlarity to glucose-1-phosphate adenylyltransferases in that they contain N-terminal catalytic domains that resemble a dinucleotide-binding Rossmann fold and C-terminal LbH fold domains. Members in this family are predicted to be glucose-1-phosphate thymidylyltransferases, which are involved in the dTDP-L-rhamnose biosynthetic pathway. Glucose-1-phosphate thymidylyltransferase catalyzes the synthesis of deoxy-thymidine di-phosphate (dTDP)-L-rhamnose, an important component of the cell wall of many microorganisms. The C-terminal LbH domain contains multiple turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cr>22</td>
            <td id=al>CTP:phosphocholine cytidylyltransferase involved in choline phosphorylation for cell surface LPS epi... [Cell wall/membrane/envelope biogenesis, Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2425522</td></tr>
<tr><td id=al>end</td><td id=al>2426596</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>glucose-1-phosphate thymidylyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
