<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04173</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04173 homocitrate synthase [Phormidium_A]</h3>
    <div>MNDCQNAFGHDSLQPRLLDTTLRDGEQMAGVALTPEEKLAIARLLDAIGVPEIEVGVAAMGGDEALAITSLVNSHLNAELLGWNRARISDLAASFNCGLQRVHISLPISDIQIAAKFGGDRQRVWQQLHESLSFARDRSTFISVGAEDASRADPEFLLAVAKTAQSLGAQRFRFCDTVGILDPFQMIEQVQRLVQSLDIPVEVHTHDDLGMATANALAGLRAGARIVDVTVNGIGERAGNAALEEVVVAMRRLQGWDFGIDSRQLLKLSRNVQQWIQGGRLPPWKAIVGENVFTHESGIHADGILKNPQTYEPFPPHWLGTQHQIAIGKHSGRRAIAACLEHHQPHLTPTNRDHQQVLLDAVRVRASQLKRGLAVDEVLALAASSPPS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>homocitrate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>homocitrate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>homocitrate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>homocitrate synthase NifV</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>homocitrate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>homocitrate synthase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>homocitrate synthase</td>
            <td id=cr>Neosynechococcus</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TPA: homocitrate synthase</td>
            <td id=cr>DVEB01</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>homocitrate synthase</td>
            <td id=co>Aulosira</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>homocitrate synthase</td>
            <td id=cr>FACHB-1407</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>trans-homoaconitate synthase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>homocitrate synthase NifV. This family consists of the NifV clade of homocitrate synthases, most of which are found in operons for nitrogen fixation. Members are closely homologous to enzymes that include 2-isopropylmalate synthase, (R)-citramalate synthase, and homocitrate synthases associated with other processes. The homocitrate made by this enzyme becomes a part of the iron-molybdenum cofactor of nitrogenase. [Biosynthesis of cofactors, prosthetic groups, and carriers, Other, Central intermediary metabolism, Nitrogen fixation]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Streptomyces rubellomurinus FrbC and related proteins, catalytic TIM barrel domain. FrbC (NifV) of Streptomyces rubellomurinus catalyzes the condensation of acetyl-CoA and alpha-ketoglutarate to form homocitrate and CoA, a reaction similar to one catalyzed by homocitrate synthase.  The gene encoding FrbC is one of several genes required for the biosynthesis of FR900098, a potent antimalarial antibiotic.  This protein is also required for assembly of the nitrogenase MoFe complex but its exact role is unknown.   This family also includes the NifV proteins of Heliobacterium chlorum and Gluconacetobacter diazotrophicus, which appear to be orthologous to FrbC.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>Isopropylmalate/homocitrate/citramalate synthases [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cb>67</td>
            <td id=al>(R)-citramalate synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>isopropylmalate/citramalate/homocitrate synthases. Methanogenic archaea contain three closely related homologs of the 2-isopropylmalate synthases (LeuA) represented by TIGR00973. Two of these in Methanococcus janaschii (MJ1392 - CimA; MJ0503 - AksA) have been characterized as catalyzing alternative reactions leaving the third (MJ1195) as the presumptive LeuA enzyme. CimA is citramalate (2-methylmalate) synthase which condenses acetyl-CoA with pyruvate. This enzyme is believed to be involved in the biosynthesis of isoleucine in methanogens and possibly other species lacking threonine dehydratase. AksA is a homocitrate synthase which also produces (homo)2-citrate and (homo)3-citrate in the biosynthesis of Coenzyme B which is restricted solely to methanogenic archaea. Methanogens, then should and aparrently do contain all three of these enzymes. Unfortunately, phylogenetic trees do not resolve into three unambiguous clades, making assignment of function to particular genes problematic. Other archaea which lack a threonine dehydratase (mainly Euryarchaeota) should contain both a CimA and a LeuA gene. This is true of, for example, archaeoglobus fulgidis, but not for the Pyrococci which have none in this clade, but one in TIGR00973 and one in TIGRT00977 which may fulfill these roles. Other species which have only one hit to this model and lack threonine dehydratase are very likely LeuA enzymes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>2-isopropylmalate synthase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>2-isopropylmalate synthase (IPMS), N-terminal catalytic TIM barrel domain. 2-isopropylmalate synthase (IPMS) catalyzes an aldol-type condensation of acetyl-CoA and 2-oxoisovalerate yielding 2-isopropylmalate and CoA, the first committed step in leucine biosynthesis.  This family includes the Arabidopsis thaliana IPMS1 and IPMS2 proteins, the Glycine max GmN56 protein, and the Brassica insularis BatIMS protein.  This family also includes a group of archeal IPMS-like proteins represented by the Methanocaldococcus jannaschii AksA protein.  AksA catalyzes the condensation of alpha-ketoglutarate and acetyl-CoA to form trans-homoaconitate, one of 13 steps in the conversion of alpha-ketoglutarate and acetylCoA to alpha-ketosuberate, a precursor to coenzyme B and biotin.  AksA also catalyzes the condensation of alpha-ketoadipate or alpha-ketopimelate with acetylCoA to form, respectively, the (R)-homocitrate homologs (R)-2-hydroxy-1,2,5-pentanetricarboxylic acid and (R)-2-hydroxy-1,2,6- hexanetricarboxylic acid.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>HMGL-like. This family contains a diverse set of enzymes. These include various aldolases and a region of pyruvate carboxylase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DRE-TIM metallolyase superfamily. The DRE-TIM metallolyase superfamily includes 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>methylthioalkylmalate synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>homocitrate synthase. This model includes the yeast LYS21 gene which carries out the first step of the alpha-aminoadipate (AAA) lysine biosynthesis pathway. A related pathway is found in Thermus thermophilus. This enzyme is closely related to 2-isopropylmalate synthase (LeuA) and citramalate synthase (CimA), both of which are present in the euryarchaeota. Some archaea have a separate homocitrate synthase (AksA) which also synthesizes longer homocitrate analogs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>2-isopropylmalate synthase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>putative alpha-isopropylmalate/homocitrate synthase family transferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>Saccharomyces cerevisiae homocitrate synthase and related proteins, catalytic TIM barrel domain. Homocitrate synthase (HCS) catalyzes the condensation of acetyl-CoA and alpha-ketoglutarate to form homocitrate, the first step in the lysine biosynthesis pathway.  This family includes the Yarrowia lipolytica LYS1 protein as well as the Saccharomyces cerevisiae LYS20 and LYS21 proteins.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Leptospira interrogans citramalate synthase (CMS) and related proteins, N-terminal catalytic TIM barrel domain. Citramalate synthase (CMS) catalyzes the conversion of pyruvate and acetyl-CoA to (R)-citramalate in the first dedicated step of the citramalate pathway.  Citramalate is only found in Leptospira interrogans and a few other microorganisms.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Desulfobacterium autotrophicum LeuA3 and related proteins, N-terminal catalytic TIM barrel domain. Desulfobacterium autotrophicum LeuA3 is sequence-similar to alpha-isopropylmalate synthase (LeuA) but its exact function is unknown.  Members of this family have an N-terminal TIM barrel domain that belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>4-hydroxy-2-oxovalerate aldolase, N-terminal catalytic TIM barrel domain. 4-hydroxy 2-ketovalerate aldolase  (Also known as 4-hydroxy-2-ketovalerate aldolase and 4-hydroxy-2-oxopentanoate aldolase (HOA)) converts 4-hydroxy-2-oxopentanoate to acetaldehyde and pyruvate, the penultimate step in the meta-cleavage pathway for the degradation of phenols, cresols and catechol.  This family includes the Escherichia coli MhpE aldolase, the Pseudomonas DmpG aldolase, and the Burkholderia xenovorans BphI pyruvate aldolase.  In Pseudomonas, the DmpG aldolase tightly associates with a dehydrogenase (DmpF ) and is inactive without it.  HOA has a canonical TIM-barrel fold with a C-terminal extension that forms a funnel leading to the active site.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Clostridium kluyveri Re-citrate synthase and related proteins, catalytic TIM barrel domain. Re-citrate synthase (Re-CS) is a Clostridium kluyveri enzyme that converts acetyl-CoA and oxaloacetate to citrate.  In most organisms, this reaction is catalyzed by Si-citrate synthase which is Si-face stereospecific with respect to C-2 of oxaloacetate, and phylogenetically unrelated to Re-citrate synthase.  Re-citrate synthase is also found in a few other strictly anaerobic organisms.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>4-hyroxy-2-oxovalerate/4-hydroxy-2-oxopentanoic acid aldolase,; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>3-hydroxy-3-methylglutaryl-CoA lyase, catalytic TIM barrel domain. 3-hydroxy-3-methylglutaryl-CoA lyase (HMGL) catalyzes the cleavage of HMG-CoA to acetyl-CoA and acetoacetate, one of the terminal steps in ketone body generation and leucine degradation, and is a key enzyme in the pathway that supplies metabolic fuel to extrahepatic tissues.  Mutations in HMGL cause a human autosomal recessive disorder called primary metabolic aciduria that affects ketogenesis and leucine catabolism and can be fatal due to an inability to tolerate hypoglycemia.  HMGL has a TIM barrel domain with a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  The cleavage of HMG-CoA requires the presence of a divalent cation like Mg2+ or Mn2+, and the reaction is thought to involve general acid/base catalysis.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>4-hydroxy-2-oxovalerate aldolase-like, N-terminal catalytic TIM barrel domain. This family of bacterial enzymes is sequence-similar to 4-hydroxy-2-oxovalerate aldolase (HOA) but its exact function is unknown.  This family includes the Bacteroides vulgatus Bvu_2661 protein and belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>oxaloacetate decarboxylase subunit alpha. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>Pyruvate/oxaloacetate carboxyltransferase  [Energy production and conversion]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>Pyruvate carboxylase and Transcarboxylase 5S, carboxyltransferase domain. This family includes the carboxyltransferase domains of pyruvate carboxylase (PC) and the transcarboxylase (TC) 5S subunit.  Transcarboxylase 5S is a cobalt-dependent metalloenzyme subunit of the biotin-dependent transcarboxylase multienzyme complex. Transcarboxylase 5S transfers carbon dioxide from the 1.3S biotin to pyruvate in the second of two carboxylation reactions catalyzed by TC. The first reaction involves the transfer of carbon dioxide from methylmalonyl-CoA to the 1.3S biotin, and is catalyzed by the 12S subunit.  These two steps allow a carboxylate group to be transferred from oxaloacetate to propionyl-CoA to yield pyruvate and methylmalonyl-CoA.  The catalytic domain of transcarboxylase 5S has a canonical TIM-barrel fold with a large C-terminal extension that forms a funnel leading to the active site.  Transcarboxylase 5S forms a homodimer and there are six dimers per complex.  In addition to the catalytic domain, transcarboxylase 5S has several other domains including a carbamoyl-phosphate synthase domain, a biotin carboxylase domain, a carboxyltransferase domain, and an ATP-grasp domain.  Pyruvate carboxylase, like TC, is a biotin-dependent enzyme that catalyzes the carboxylation of pyruvate to produce oxaloacetate.  In mammals, PC has critical roles in gluconeogenesis, lipogenesis, glyceroneogenesis, and insulin secretion.  Inherited PC deficiencies are linked to serious diseases in humans such as lactic acidemia, hypoglycemia, psychomotor retardation, and death.  PC is a single-chain enzyme and is active only in its homotetrameric form.  PC has three domains, an N-terminal biotin carboxylase domain, a carboxyltransferase domain (this alignment model), and a C-terminal biotin-carboxyl carrier protein domain.  This family belongs to the DRE-TIM metallolyase superfamily.  DRE-TIM metallolyases include 2-isopropylmalate synthase (IPMS), alpha-isopropylmalate synthase (LeuA), 3-hydroxy-3-methylglutaryl-CoA lyase, homocitrate synthase, citramalate synthase, 4-hydroxy-2-oxovalerate aldolase, re-citrate synthase, transcarboxylase 5S, pyruvate carboxylase, AksA, and FrbC.  These members all share a conserved  triose-phosphate isomerase (TIM) barrel domain consisting of a core beta(8)-alpha(8) motif with the eight parallel beta strands forming an enclosed barrel surrounded by eight alpha helices.  The domain has a catalytic center containing a divalent cation-binding site formed by a cluster of invariant residues that cap the core of the barrel.  In addition, the catalytic site includes three invariant residues - an aspartate (D), an arginine (R), and a glutamate (E) - which is the basis for the domain name "DRE-TIM".</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>oxaloacetate decarboxylase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>19</td>
            <td id=al>pyruvate carboxylase subunit B; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>pyruvate carboxylase subunit B. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>hydroxymethylglutaryl-CoA lyase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cr>34</td>
            <td id=al>hydroxymethylglutaryl-CoA lyase</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3405794</td></tr>
<tr><td id=al>end</td><td id=al>3406961</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>homocitrate synthase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
