<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03725</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03725 prephenate dehydratase [Phormidium_A]</h3>
    <div>MAYCNWLEETSGRKGKLVPHPSISQTLEALLQGTCDVAVIPIENSIQGSVTVSLDSFWEKEQLQVQQALVLPIHHALISRGAGLEEITTVYSHPQALAQCQHWLETQLSDVTLHSTNSTAEALQYLQDPQVGAIASERAAQLYDLPILAHGINDHRDNCTRFWVVRRVQEGMALPQAPGPAYLSLAFSLPQNSPGALLNPLQVLANYGINMTRIESRPTKRLLGEYLFFVDVDAPEEEGVLSEAIAALGKYTEILKIFGSYTVLTVGNSPLEG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>prephenate dehydratase PheA2</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>TPA: prephenate dehydratase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>prephenate dehydratase</td>
            <td id=cr>JAAUTN01</td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>prephenate dehydratase</td>
            <td id=co>LEGE-07092</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>prephenate dehydratase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>Prephenate dehydratase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Catalytic domain of prephenate dehydratase and similar proteins, subgroup 1; the type 2 periplasmic binding protein fold. Prephenate dehydratase (PDT, EC:4.2.1.51) converts prephenate to phenylpyruvate through dehydration and decarboxylation reactions. PDT plays a key role in the biosynthesis of L-Phe in organisms that utilize the shikimate pathway. PDT is allosterically regulated by L-Phe and other amino acids. The catalytic PDT domain consists of two similar subdomains with a cleft in between, which hosts the highly conserved active site. In gram-postive bacteria and archaea, PDT is a monofunctional enzyme, consisting of a catalytic domain (PDT domain) and a regulatory domain (ACT) (aspartokinase, chorismate mustase domain). In gram-negative bacteria, PDT exists as fusion protein with chorismate mutase (CM), forming a bifunctional enzyme, P-protein (PheA). The CM in the P-protein catalyzes the pericycle isomerization of chorismate to prephenate that serves as a substrate for PDT. The CM and PDT are essentail enzymes for the biosynthesis of aromatic amino acids in microorganisms but are not found in humans. Thus, both CM and PDT can potentially serve as drug targets against microbial pathogens. The PDT domain has the same structural fold as the type 2 periplasmic binding proteins (PBP2), many of which are involved in chemotaxis and uptake of nutrients and other small molecules from the extracellular space as a primary receptor. The PBP2 proteins are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Prephenate dehydratase. This protein is involved in Phenylalanine biosynthesis. This protein catalyzes the decarboxylation of prephenate to phenylpyruvate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Catalytic domain of prephenate dehydratase from Staphylococcus aureus and similar proteins, subgroup 4; the type 2 periplasmic binding protein fold. Prephenate dehydratase (PDT, EC:4.2.1.51) converts prephenate to phenylpyruvate through dehydration and decarboxylation reactions. PDT plays a key role in the biosynthesis of L-Phe in organisms that utilize the shikimate pathway. PDT is allosterically regulated by L-Phe and other amino acids. The catalytic PDT domain consists of two similar subdomains with a cleft in between, which hosts the highly conserved active site. In gram-postive bacteria and archaea, PDT is a monofunctional enzyme, consisting of a catalytic domain (PDT domain) and a regulatory domain (ACT) (aspartokinase, chorismate mustase domain). In gram-negative bacteria, PDT exists as fusion protein with chorismate mutase (CM), forming a bifunctional enzyme, P-protein (PheA). The CM in the P-protein catalyzes the pericycle isomerization of chorismate to prephenate that serves as a substrate for PDT. The CM and PDT are essentail enzymes for the biosynthesis of aromatic amino acids in microorganisms but are not found in humans. Thus, both CM and PDT can potentially serve as drug targets against microbial pathogens. The PDT domain has the same structural fold as the type 2 periplasmic binding proteins (PBP2), many of which are involved in chemotaxis and uptake of nutrients and other small molecules from the extracellular space as a primary receptor. The PBP2 proteins are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Catalytic domain of prephenate dehydratase and similar proteins; the type 2 periplasmic binding protein fold. Prephenate dehydratase (PDT, EC:4.2.1.51) converts prephenate to phenylpyruvate through dehydration and decarboxylation reactions. PDT plays a key role in the biosynthesis of L-Phe in organisms that utilize the shikimate pathway. PDT is allosterically regulated by L-Phe and other amino acids. The catalytic PDT domain consists of two similar subdomains with a cleft in between, which hosts the highly conserved active site. In gram-postive bacteria and archaea, PDT is a monofunctional enzyme, consisting of a catalytic domain (PDT domain) and a regulatory domain (ACT) (aspartokinase, chorismate mustase domain). In gram-negative bacteria, PDT exists as fusion protein with chorismate mutase (CM), forming a bifunctional enzyme, P-protein (PheA). The CM in the P-protein catalyzes the pericycle isomerization of chorismate to prephenate that serves as a substrate for PDT. The CM and PDT are essentail enzymes for the biosynthesis of aromatic amino acids in microorganisms but are not found in humans. Thus, both CM and PDT can potentially serve as drug targets against microbial pathogens. The PDT domain has the same structural fold as the type 2 periplasmic binding proteins (PBP2), many of which are involved in chemotaxis and uptake of nutrients and other small molecules from the extracellular space as a primary receptor. The PBP2 proteins are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>Catalytic domain of prephenate dehydratase from Chlorobium tepidum and similar proteins, subgroup 2; the type 2 periplasmic binding protein fold. Prephenate dehydratase (PDT, EC:4.2.1.51) converts prephenate to phenylpyruvate through dehydration and decarboxylation reactions. PDT plays a key role in the biosynthesis of L-Phe in organisms that utilize the shikimate pathway. PDT is allosterically regulated by L-Phe and other amino acids. The catalytic PDT domain consists of two similar subdomains with a cleft in between, which hosts the highly conserved active site. In gram-postive bacteria and archaea, PDT is a monofunctional enzyme, consisting of a catalytic domain (PDT domain) and a regulatory domain (ACT) (aspartokinase, chorismate mustase domain). In gram-negative bacteria, PDT exists as fusion protein with chorismate mutase (CM), forming a bifunctional enzyme, P-protein (PheA). The CM in the P-protein catalyzes the pericycle isomerization of chorismate to prephenate that serves as a substrate for PDT. The CM and PDT are essentail enzymes for the biosynthesis of aromatic amino acids in microorganisms but are not found in humans. Thus, both CM and PDT can potentially serve as drug targets against microbial pathogens. The PDT domain has the same structural fold as the type 2 periplasmic binding proteins (PBP2), many of which are involved in chemotaxis and uptake of nutrients and other small molecules from the extracellular space as a primary receptor. The PBP2 proteins are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>arogenate dehydratase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>prephenate dehydratase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>Catalytic domain of prephenate dehydratase from Arthrobacter aurescens and similar proteins, subgroup 3; the type 2 periplasmic binding protein fold. Prephenate dehydratase (PDT, EC:4.2.1.51) converts prephenate to phenylpyruvate through dehydration and decarboxylation reactions. PDT plays a key role in the biosynthesis of L-Phe in organisms that utilize the shikimate pathway. PDT is allosterically regulated by L-Phe and other amino acids. The catalytic PDT domain consists of two similar subdomains with a cleft in between, which hosts the highly conserved active site. In gram-postive bacteria and archaea, PDT is a monofunctional enzyme, consisting of a catalytic domain (PDT domain) and a regulatory domain (ACT) (aspartokinase, chorismate mustase domain). In gram-negative bacteria, PDT exists as fusion protein with chorismate mutase (CM), forming a bifunctional enzyme, P-protein (PheA). The CM in the P-protein catalyzes the pericycle isomerization of chorismate to prephenate that serves as a substrate for PDT. The CM and PDT are essentail enzymes for the biosynthesis of aromatic amino acids in microorganisms but are not found in humans. Thus, both CM and PDT can potentially serve as drug targets against microbial pathogens. The PDT domain has the same structural fold as the type 2 periplasmic binding proteins (PBP2), many of which are involved in chemotaxis and uptake of nutrients and other small molecules from the extracellular space as a primary receptor. The PBP2 proteins are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>bifunctional chorismate mutase/prephenate dehydratase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>C-terminal ACT domain of the bifunctional chorismate mutase-prephenate dehydratase (CM-PDT) enzyme and the prephenate dehydratase (PDT) enzyme. The C-terminal ACT domain of the bifunctional chorismate mutase-prephenate dehydratase (CM-PDT) enzyme and the prephenate dehydratase (PDT) enzyme, found in plants, fungi, bacteria, and archaea. The P-protein of E. coli (CM-PDT, PheA) catalyzes the conversion of chorismate to prephenate and then the decarboxylation and dehydration to form phenylpyruvate. These are the first two steps in the biosynthesis of L-Phe and L-Tyr via the shikimate pathway in microorganisms and plants. The E. coli P-protein (CM-PDT) has three domains with an N-terminal domain with chorismate mutase activity, a middle domain with prephenate dehydratase activity, and an ACT regulatory C-terminal domain. The prephenate dehydratase enzyme has a PDT and ACT domain. The ACT domain is essential to bring about the negative allosteric regulation by L-Phe binding. L-Phe binds with positive cooperativity; with this binding, there is a shift in the protein to less active tetrameric and higher oligomeric forms from a more active dimeric form. Members of this CD belong to the superfamily of ACT regulatory domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ACT domain of the nonheme iron-dependent, aromatic amino acid hydroxylases (AAAH). ACT domain of the nonheme iron-dependent, aromatic amino acid hydroxylases (AAAH): Phenylalanine hydroxylases (PAH), tyrosine hydroxylases (TH) and tryptophan hydroxylases (TPH), both peripheral (TPH1) and neuronal (TPH2) enzymes. This family of enzymes shares a common catalytic mechanism, in which dioxygen is used by an active site containing a single, reduced iron atom to hydroxylate an unactivated aromatic substrate, concomitant with a two-electron oxidation of tetrahydropterin (BH4) cofactor to its quinonoid dihydropterin form. Eukaryotic AAAHs have an N-terminal  ACT (regulatory) domain, a middle catalytic domain and a C-terminal domain which is responsible for the oligomeric state of the enzyme forming a domain-swapped tetrameric coiled-coil. The PAH, TH, and TPH enzymes contain highly conserved catalytic domains but distinct N-terminal ACT domains and differ in their mechanisms of regulation. One commonality is that all three eukaryotic enzymes appear to be regulated, in part, by the phosphorylation of serine residues N-terminal of the ACT domain. Also included in this CD are the C-terminal ACT domains of the bifunctional chorismate mutase-prephenate dehydratase (CM-PDT) enzyme and the prephenate dehydratase (PDT) enzyme found in plants, fungi, bacteria, and archaea. The P-protein of Escherichia coli (CM-PDT) catalyzes the conversion of chorismate to prephenate and then the decarboxylation and dehydration to form phenylpyruvate. These are the first two steps in the biosynthesis of L-Phe and L-Tyr via the shikimate pathway in microorganisms and plants. The E. coli P-protein (CM-PDT) has three domains with an N-terminal domain with chorismate mutase activity, a middle domain with prephenate dehydratase activity, and an ACT regulatory C-terminal domain. The prephenate dehydratase enzyme has a PDT and ACT domain. The ACT domain is essential to bring about the negative allosteric regulation by L-Phe binding. L-Phe binds with positive cooperativity; with this binding, there is a shift in the protein to less active tetrameric and higher oligomeric forms from a more active dimeric form. Members of this CD belong to the superfamily of ACT regulatory domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ACT domain of the nonheme iron-dependent, aromatic amino acid hydroxylases (AAAH). ACT domain of the nonheme iron-dependent, aromatic amino acid hydroxylases (AAAH): Phenylalanine hydroxylases (PAH), tyrosine hydroxylases (TH) and tryptophan hydroxylases (TPH), both peripheral (TPH1) and neuronal (TPH2) enzymes. This family of enzymes shares a common catalytic mechanism, in which dioxygen is used by an active site containing a single, reduced iron atom to hydroxylate an unactivated aromatic substrate, concomitant with a two-electron oxidation of tetrahydropterin (BH4) cofactor to its quinonoid dihydropterin form. PAH catalyzes the hydroxylation of L-Phe to L-Tyr, the first step in the catabolic degradation of L-Phe; TH catalyses the hydroxylation of L-Tyr to 3,4-dihydroxyphenylalanine, the rate limiting step in the biosynthesis of catecholamines; and TPH catalyses the hydroxylation of L-Trp to 5-hydroxytryptophan, the rate limiting step in the biosynthesis of 5-hydroxytryptamine (serotonin) and the first reaction in the synthesis of melatonin. Eukaryotic AAAHs have an N-terminal  ACT (regulatory) domain, a middle catalytic domain and a C-terminal domain which is responsible for the oligomeric state of the enzyme forming a domain-swapped tetrameric coiled-coil. The PAH, TH, and TPH enzymes contain highly conserved catalytic domains but distinct N-terminal ACT domains (this CD) and differ in their mechanisms of regulation. One commonality is that all three eukaryotic enzymes are regulated in part by the phosphorylation of serine residues N-terminal of the ACT domain. Members of this CD belong to the superfamily of ACT regulatory domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>ACT domain of the nonheme iron-dependent aromatic amino acid hydroxylase, phenylalanine hydroxylases (PAH). ACT domain of the nonheme iron-dependent aromatic amino acid hydroxylase, phenylalanine hydroxylases (PAH). PAH catalyzes the hydroxylation of L-Phe to L-Tyr, the first step in the catabolic degradation of L-Phe. In PAH, an autoregulatory sequence, N-terminal of the ACT domain, extends across the catalytic domain active site and regulates the enzyme by intrasteric regulation. It appears that the activation by L-Phe induces a conformational change that converts the enzyme to a high-affinity and high-activity state. Modulation of activity is achieved through inhibition by BH4 and activation by phosphorylation of serine residues of the autoregulatory region. The molecular basis for the cooperative activation process is not fully understood yet. Members of this CD belong to the superfamily of ACT regulatory domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cr>17</td>
            <td id=al>phenylalanine-4-hydroxylase, tetrameric form. This model describes the larger, tetrameric form of phenylalanine-4-hydroxylase, as found in metazoans. The enzyme irreversibly converts phenylalanine to tryosine and is known to be the rate-limiting step in phenylalanine catabolism in some systems. It is closely related to metazoan tyrosine 3-monooxygenase and tryptophan 5-monoxygenase, and more distantly to monomeric phenylalanine-4-hydroxylases of some Gram-negative bacteria. The member of this family from Drosophila has been described as having both phenylalanine-4-hydroxylase and tryptophan 5-monoxygenase activity (. However, a Drosophila member of the tryptophan 5-monoxygenase clade has subsequently been discovered.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>Type 2 periplasmic binding fold superfamily. This evolutionary model and hierarchy represent the ligand-binding domains found in solute binding proteins that serve as initial receptors in the transport, signal transduction and channel gating.  The PBP2 proteins share the same architecture as periplasmic binding proteins type 1 (PBP1), but have a different topology.  They are typically comprised of two globular subdomains connected by a flexible hinge and bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. The origin of PBP module can be traced across the distant phyla, including eukaryotes, archebacteria, and prokaryotes.  The majority of PBP2 proteins are involved in the uptake of a variety of soluble substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.  Besides transport proteins, the family includes ionotropic glutamate receptors and unorthodox sensor proteins involved in signal transduction. The substrate binding domain of the LysR transcriptional regulators and the oligopeptide-like transport systems also contain the type 2 periplasmic binding fold and thus they are significantly homologous to that of the PBP2; however, these two families are grouped into a separate hierarchy of the PBP2 superfamily due to the large number of protein sequences.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ACT domains are commonly involved in specifically binding an amino acid or other small ligand leading to regulation of the enzyme. Members of this CD belong to the superfamily of ACT regulatory domains. Pairs of ACT domains are commonly involved in specifically binding an amino acid or other small ligand leading to regulation of the enzyme. The ACT domain has been detected in a number of diverse proteins; some of these proteins are involved in amino acid and purine biosynthesis, phenylalanine hydroxylation, regulation of bacterial metabolism and transcription, and many remain to be characterized. ACT domain-containing enzymes involved in amino acid and purine synthesis are in many cases allosteric enzymes with complex regulation enforced by the binding of ligands. The ACT domain is commonly involved in the binding of a small regulatory molecule, such as the amino acids L-Ser and L-Phe in the case of D-3-phosphoglycerate dehydrogenase and the bifunctional chorismate mutase-prephenate dehydratase enzyme (P-protein), respectively. Aspartokinases typically consist of two C-terminal ACT domains in a tandem repeat, but  the second ACT domain is inserted within the first, resulting in, what is normally the terminal beta strand of ACT2, formed from a region N-terminal of ACT1. ACT domain repeats have been shown to have nonequivalent ligand-binding sites with complex regulatory patterns such as those seen in the bifunctional enzyme, aspartokinase-homoserine dehydrogenase (ThrA). In other enzymes, such as phenylalanine hydroxylases, the ACT domain appears to function as a flexible small module providing allosteric regulation via transmission of conformational changes, these conformational changes are not necessarily initiated by regulatory ligand binding at the ACT domain itself. ACT domains are present either singularly, N- or C-terminal, or in pairs present C-terminal or between two catalytic domains. Unique to cyanobacteria are four ACT domains C-terminal to an aspartokinase domain. A few proteins are composed almost entirely of ACT domain repeats as seen in the four ACT domain protein, the ACR protein, found in higher plants; and the two ACT domain protein, the glycine cleavage system transcriptional repressor (GcvR) protein, found in some bacteria. Also seen are single ACT domain proteins similar to the Streptococcus pneumoniae ACT domain protein (uncharacterized pdb structure 1ZPV) found in both bacteria and archaea. Purportedly, the ACT domain is an evolutionarily mobile ligand binding regulatory module that has been fused to different enzymes at various times.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3142819</td></tr>
<tr><td id=al>end</td><td id=al>3143641</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>prephenate dehydratase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
