<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02275</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02275 pyruvate formate lyase-activating protein [Phormidium_A]</h3>
    <div>MPISQSPISGRIHSFESCGTVDGPGIRFVVFTQGCPLRCLYCHNPDCQDSSGGRDVTVTEILEEVVKYRSYMEFSQGGVTVTGGEPLMQPEFVGEIFRQCQALGIHTALDTSGYIPVEVAKPVLENTDLVLLDIKSYQPETYRAVTCVSVEPTLKFARYLEAINKPTWIRFVLVPELTNQVENMEGLAQFLSGFRNVERLEILPFHKMGEYKWEALGYDYQLKDTPEPTDEQIQVAREIFSQYRLPVVA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase activating enzyme PflA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate-lyase-activating protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=co>Limnospira</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=co>Limnospira</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>98</td>
            <td id=cg>97</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=co>Halothece</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>97</td>
            <td id=cg>95</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=co>SIO1D9</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>96</td>
            <td id=cg>96</td>
            <td id=al>pyruvate formate lyase-activating protein</td>
            <td id=co>Hyella</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>57</td>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=al>TIGR02493 pyruvate formate-lyase 1-activating enzyme. An iron-sulfur protein with a radical-SAM domain (pfam04055). A single glycine residue in EC 2.3.1.54, formate C-acetyltransferase (formate-pyruvate lyase), is oxidized to the corresponding radical by transfer of H from its CH2 to AdoMet with concomitant cleavage of the latter. The reaction requires Fe2+. The first stage is reduction of the AdoMet to give methionine and the 5'-deoxyadenosin-5-yl radical, which then abstracts a hydrogen radical from the glycine residue. [Energy metabolism, Anaerobic, Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>96</td>
            <td id=cg>97</td>
            <td id=al>PRK11145 pyruvate formate lyase 1-activating protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>97</td>
            <td id=cg>93</td>
            <td id=al>COG1180 Pyruvate-formate lyase-activating enzyme [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>TIGR02494 glycyl-radical enzyme activating protein. This subset of the radical-SAM family (pfam04055) includes a number of probable activating proteins acting on different enzymes all requiring an amino-acid-centered radical. The closest relatives to this family are the pyruvate-formate lyase activating enzyme (PflA, 1.97.1.4, TIGR02493) and the anaerobic ribonucleotide reductase activating enzyme (TIGR02491). Included within this subfamily are activators of hydroxyphenyl acetate decarboxylase (HdpA), benzylsuccinate synthase (BssD), gycerol dehydratase (DhaB2) as well as enzymes annotated in E. coli as activators of different isozymes of pyruvate-formate lyase (PFLC and PFLE) however, these appear to lack characterization and may activate enzymes with distinctive functions. Most of the sequence-level variability between these forms is concentrated within an N-terminal domain which follows a conserved group of three cysteines and contains a variable pattern of 0 to 8 additional cysteines.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>TIGR04041 glycine radical enzyme activase, YjjW family. Members of this family are radical SAM enzymes, designated YjjW in E. coli, that are paired with and appear to activate a glycyl radical enzyme of unknown function, designated YjjI. This activase and its target are found in Clostridial species as well as E. coli and cousins. Members of this family may be misannotated as pyruvate formate lyase activating enzyme. [Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>NF033719 indoleacetate decarboxylase activase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>NF033717 4-hydroxyphenylacetate decarboxylase activase. 4-hydroxyphenylacetate decarboxylase activase is a radical SAM enzyme, found in anaerobic bacteria where 4-hydroxyphenylacetate decarboxylase occurs and required to prepare the glycyl radical active site of the enzyme.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cb>74</td>
            <td id=cg>87</td>
            <td id=al>PRK10076 pyruvate formate lyase II activase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>TIGR04003 [benzylsuccinate synthase]-activating enzyme. Members of this radical SAM protein family are [benzylsuccinate synthase]-activating enzyme, a glycyl radical active site-creating enzyme related to [pyruvate formate-lyase]-activating enzyme and additional uncharacterized homologs activating additional glycyl radical-containing enzymes. [Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cb>62</td>
            <td id=cg>81</td>
            <td id=al>TIGR02495 anaerobic ribonucleoside-triphosphate reductase activating protein. This enzyme is a member of the radical-SAM family (pfam04055). It is often gene clustered with the class III (anaerobic) ribonucleotide triphosphate reductase (NrdD, TIGR02487) and presumably fulfills the identical function as NrdG, which utilizes S-adenosyl methionine, an iron-sulfur cluster and a reductant (dihydroflavodoxin) to produce a glycine-centered radical in NrdD. [Purines, pyrimidines, nucleosides, and nucleotides, 2'-Deoxyribonucleotide metabolism, Protein fate, Protein modification and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>48</td>
            <td id=cg>88</td>
            <td id=al>pfam13353 4Fe-4S single cluster domain. This family includes proteins containing domains which bind to iron-sulfur clusters. Members include bacterial ferredoxins, various dehydrogenases, and various reductases. The structure of the domain is an alpha-antiparallel beta sandwich.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>52</td>
            <td id=cb>61</td>
            <td id=al>COG0602 Organic radical activating enzyme [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=al>pfam04055 Radical SAM superfamily. Radical SAM proteins catalyze diverse reactions, including unusual methylations, isomerisation, sulphur insertion, ring formation, anaerobic oxidation and protein radical formation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cb>75</td>
            <td id=cg>92</td>
            <td id=al>cd01335 Radical SAM superfamily. Enzymes of this family generate radicals by combining a 4Fe-4S cluster and S-adenosylmethionine (SAM) in close proximity. They are characterized by a conserved CxxxCxxC motif, which coordinates the conserved iron-sulfur cluster. Mechanistically, they share the transfer of a single electron from the iron-sulfur cluster to SAM, which leads to its reductive cleavage to methionine and a 5'-deoxyadenosyl radical, which, in turn, abstracts a hydrogen from the appropriately positioned carbon atom. Depending on the enzyme, SAM is consumed during this process or it is restored and reused. Radical SAM enzymes catalyze steps in metabolism, DNA repair, the biosynthesis of vitamins and coenzymes, and the biosynthesis of many antibiotics. Examples are biotin synthase (BioB), lipoyl synthase (LipA), pyruvate formate-lyase (PFL), coproporphyrinogen oxidase (HemN), lysine 2,3-aminomutase (LAM), anaerobic ribonucleotide reductase (ARR), and  MoaA, an enzyme of the biosynthesis of molybdopterin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>82</td>
            <td id=cb>63</td>
            <td id=al>COG2896 Molybdenum cofactor biosynthesis enzyme MoaA [Coenzyme transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>87</td>
            <td id=cb>62</td>
            <td id=al>COG0535 Radical SAM superfamily enzyme, MoaA/NifB/PqqE/SkfB family [General function prediction only]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1928825</td></tr>
<tr><td id=al>end</td><td id=al>1929575</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>pyruvate formate lyase-activating protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>250</td></tr>
</table>

    </div>
</body>
</html>
