<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00311</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00311 DNA topoisomerase (ATP-hydrolyzing) subunit B [Phormidium_A]</h3>
    <div>MTSSYSADQIQVLEGLEAVRKRPGMYIGSTGPRGLHHLVYEVVDNAIDEALAGYCTHIEIDIHADGSVSVTDNGRGIPTDVHSKTGRSALETVMTVLHAGGKFGGGGYKVSGGLHGVGVSVVNALSEWVEVTVWREKKVYLQRYERGKPASELTSKPSSEKQRTGTSVRFLPDRDIFVTTGIEFDFNAIGARVRELAYLNAGVRITFTDYRLDLLGKEEPRVETYFYEGGIREYIQYMNRDKEPLHEEIIYVEGERDGVVVEAALQWSVNAYSDSLLGFANNIRTVDGGTHLEGLKSVLTRTMNVTARKRNKIKENQPNLGGENIREGLTGIISVKVPDPEFEGQTKTKLGNPEVRGIVDSLVGEVLTEYLEFRPHVADAILEKAIQAFNAAEAARRARELVRRKSVLESSTLPGKLADCSSKDPSESEIFIVEGDSAGGSAKQGRDRRFQAILPLRGKILNIEKTDDAKIYKNNEVQSLIAALGLGIKGEDFEVSQLRYHRIILMTDADVDGAHIRTLLLTFFYRYQRDMVEQGFVYIACPPLYKVERGRSHHYCYSDRELQRHLNELPSNANYNIQRFKGLGEMMPTQLWDTTMNPESRLLKRVEIEDAVEADRIFTVLMGDKVAPRREFIETEGPKLNLTDLDI</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA gyrase subunit GyrB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=co>PCC7113</td>
            </tr><tr>
            <td id=cg>82</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td>
            <td id=co>Desertifilum</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK05644 DNA gyrase subunit B; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TIGR01059 DNA gyrase, B subunit. This model describes the common type II DNA topoisomerase (DNA gyrase). Two apparently independently arising families, one in the Proteobacteria and one in Gram-positive lineages, are both designated toposisomerase IV. Proteins scoring above the noise cutoff for this model and below the trusted cutoff for topoisomerase IV models probably should be designated GyrB. [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>COG0187 DNA gyrase/topoisomerase IV, subunit B [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK14939 DNA gyrase subunit B; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK05559 DNA topoisomerase IV subunit B; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=al>smart00433 TopoisomeraseII. Eukaryotic DNA topoisomerase II, GyrB, ParE</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TIGR01058 DNA topoisomerase IV, B subunit, Gram-positive. Operationally, topoisomerase IV is a type II topoisomerase required for the decatenation step of chromosome segregation. Not every bacterium has both a topo II and a topo IV. The topo IV families of the Gram-positive bacteria and the Gram-negative bacteria appear not to represent a single clade among the type II topoisomerases, and are represented by separate models for this reason. [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>PTZ00109 DNA gyrase subunit b; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>TIGR01055 DNA topoisomerase IV, B subunit, proteobacterial. Operationally, topoisomerase IV is a type II topoisomerase required for the decatenation of chromosome segregation. Not every bacterium has both a topo II and a topo IV. The topo IV families of the Gram-positive bacteria and the Gram-negative bacteria appear not to represent a single clade among the type II topoisomerases, and are represented by separate models for this reason. This protein is active as an alpha(2)beta(2) heterotetramer. [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>cd16928 Histidine kinase-like ATPase domain of the B subunit of DNA gyrase. This family includes histidine kinase-like ATPase domain of the B subunit of DNA gyrase. Bacterial DNA gyrase is a type II topoisomerase (type II as it transiently cleaves both strands of DNA) which catalyzes the introduction of negative supercoils into DNA, possibly by a mechanism in which one segment of the double-stranded DNA substrate is passed through a transient break in a second segment. It consists of GyrA and GyrB subunits in an A2B2 stoichiometry; GyrA subunits catalyze strand-breakage and reunion reactions, and GyrB subunits hydrolyze ATP. DNA gyrase is found in bacteria, plants and archaea, but as it is absent in humans it is a possible drug target for the treatment of bacterial and parasite infections.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>cd03366 TOPRIM_TopoIIA_GyrB: topoisomerase-primase (TOPRIM) nucleotidyl transferase/hydrolase domain of the type found in proteins of the type IIA family of DNA topoisomerases similar to the Escherichia coli GyrB subunit. TopoIIA enzymes cut both strands of the duplex DNA to remove (relax) both positive and negative supercoils in DNA.  These enzymes covalently attach to the 5' ends of the cut DNA, separate the free ends of the cleaved strands, pass another region of the duplex through this gap, then rejoin the ends. These proteins also catenate/ decatenate duplex rings.  DNA gyrase is more effective at relaxing supercoils than decatentating DNA.  DNA gyrase in addition inserts negative supercoils in the presence of ATP.  The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). The conserved glutamate may act as a general base in strand joining and as a general acid in strand cleavage by topisomerases.  The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cr>24</td>
            <td id=cg>93</td>
            <td id=al>cd00822 TopoIIA_Trans_DNA_gyrase: Transducer domain, having a ribosomal S5 domain 2-like fold, of the type found in proteins of the type IIA family of DNA topoisomerases similar to the B subunits of E. coli DNA gyrase and E. coli Topoisomerase IV which are  heterodimers composed of two subunits.  The type IIA enzymes are the predominant form of topoisomerase and are found in some bacteriophages, viruses and archaea, and in all bacteria and eukaryotes.  All type IIA topoisomerases are related to each other at amino acid sequence level, though their oligomeric organization sometimes differs.  TopoIIA enzymes cut both strands of the duplex DNA to remove (relax) both positive and negative supercoils in DNA.  These enzymes covalently attach to the 5' ends of the cut DNA, separate the free ends of the cleaved strands, pass another region of the duplex through this gap, then rejoin the ends. TopoIIA enzymes also catenate/ decatenate duplex rings. E.coli DNA gyrase is a heterodimer composed of two subunits. E. coli DNA gyrase B subunit is known to be important in nucleotide hydrolysis and the transduction of structural signals from ATP-binding site to the DNA breakage/reunion regions of the enzymes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>cd01030 TOPRIM_TopoIIA_like: topoisomerase-primase (TOPRIM) nucleotidyl transferase/hydrolase domain of the type found in proteins of the type IIA family of DNA topoisomerases similar to Saccharomyces cerevisiae Topoisomerase II. TopoIIA enzymes cut both strands of the duplex DNA to remove (relax) both positive and negative supercoils in DNA.  These enzymes covalently attach to the 5' ends of the cut DNA, separate the free ends of the cleaved strands, pass another region of the duplex through this gap, then rejoin the ends. These proteins also catenate/ decatenate duplex rings.  The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). The conserved glutamate may act as a general base in strand joining and as a general acid in strand cleavage by topisomerases.  The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cr>24</td>
            <td id=cg>93</td>
            <td id=al>pfam00204 DNA gyrase B. This family represents the second domain of DNA gyrase B which has a ribosomal S5 domain 2-like fold. This family is structurally related to PF01119.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PHA02569 DNA topoisomerase II large subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cr>9</td>
            <td id=cg>100</td>
            <td id=al>pfam00986 DNA gyrase B subunit, carboxyl terminus. The amino terminus of eukaryotic and prokaryotic DNA topoisomerase II are similar, but they have a different carboxyl terminus. The amino-terminal portion of the DNA gyrase B protein is thought to catalyze the ATP-dependent super-coiling of DNA. See pfam00204. The carboxyl-terminal end supports the complexation with the DNA gyrase A protein and the ATP-independent relaxation. This family also contains Topoisomerase IV. This is a bacterial enzyme that is closely related to DNA gyrase,.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>99</td>
            <td id=co>46</td>
            <td id=al>PTZ00108 DNA topoisomerase 2-like protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>PLN03128 DNA topoisomerase 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>15</td>
            <td id=cg>81</td>
            <td id=al>cd03365 TOPRIM_TopoIIA: topoisomerase-primase (TOPRIM) nucleotidyl transferase/hydrolase domain of the type found in proteins of the type IIA family of DNA topoisomerases similar to Saccharomyces cerevisiae Topoisomerase II. TopoIIA enzymes cut both strands of the duplex DNA to remove (relax) both positive and negative supercoils in DNA.  These enzymes covalently attach to the 5' ends of the cut DNA, separate the free ends of the cleaved strands, pass another region of the duplex through this gap, then rejoin the ends. These proteins also catenate/ decatenate duplex rings.  The TOPRIM domain has two conserved motifs, one of which centers at a conserved glutamate and the other one at two conserved aspartates (DxD). This glutamate and two aspartates, cluster together to form a highly acid surface patch. The conserved glutamate may act as a general base in strand joining and as a general acid in strand cleavage by topisomerases.  The DXD motif may co-ordinate Mg2+, a cofactor required for full catalytic function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>96</td>
            <td id=co>42</td>
            <td id=al>PLN03237 DNA topoisomerase 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=al>cd00329 MutL_Trans: transducer domain, having a ribosomal S5 domain 2-like fold, conserved in the C-terminal domain of type II DNA topoisomerases (Topo II) and DNA mismatch repair (MutL/MLH1/PMS2) proteins. This transducer domain is homologous to the second domain of the DNA gyrase B subunit, which is known to be important in nucleotide hydrolysis and the transduction of structural signals from ATP-binding site to the DNA breakage/reunion regions of the enzymes. The GyrB dimerizes in response to ATP binding, and is homologous to the N-terminal half of eukaryotic Topo II and the ATPase fragment of MutL. Type II DNA topoisomerases catalyze the ATP-dependent transport of one DNA duplex through another, in the process generating transient double strand breaks via covalent attachments to both DNA strands at the 5' positions. Included in this group are proteins similar to human MLH1 and PMS2.  MLH1 forms a heterodimer with PMS2 which functions in meiosis and in DNA mismatch repair (MMR). Cells lacking either hMLH1 or hPMS2 have a strong mutator phenotype and display microsatellite instability (MSI). Mutation in hMLH1 accounts for a large fraction of Lynch syndrome (HNPCC) families.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>14</td>
            <td id=cg>100</td>
            <td id=al>pfam01751 Toprim domain. This is a conserved region from DNA primase. This corresponds to the Toprim domain common to DnaG primases, topoisomerases, OLD family nucleases and RecR proteins. Both DnaG motifs IV and V are present in the alignment, the DxD (V) motif may be involved in Mg2+ binding and mutations to the conserved glutamate (IV) completely abolish DnaG type primase activity. DNA primase EC:2.7.7.6 is a nucleotidyltransferase it synthesizes the oligoribonucleotide primers required for DNA replication on the lagging strand of the replication fork; it can also prime the leading stand and has been implicated in cell division. This family also includes the atypical archaeal A subunit from type II DNA topoisomerases. Type II DNA topoisomerases catalyze the relaxation of DNA supercoiling by causing transient double strand breaks.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cr>16</td>
            <td id=cg>98</td>
            <td id=al>pfam02518 Histidine kinase-, DNA gyrase B-, and HSP90-like ATPase. This family represents the structurally related ATPase domains of histidine kinase, DNA gyrase B and HSP90.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>smart00387 Histidine kinase-like ATPases. Histidine kinase-, DNA gyrase B-, phytochrome-like ATPases.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>248390</td></tr>
<tr><td id=al>end</td><td id=al>250334</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA topoisomerase (ATP-hydrolyzing) subunit B</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>648</td></tr>
</table>

    </div>
</body>
</html>
