<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00408</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00408 AAA family ATPase [Microcystis]</h3>
    <div>MTSLRHSLGFYLPSIAGGVLVGGIAIASGGSGLQSGIAAIATGVSGSTSAALVAQRSRHERQLFESERNQLQKNLDRQQKALGLNIEVESLERKLGLLGEEIQKQEETLEQRQQKIHQEIQEREEILEELQRRISTVKAEMPNLEDLENLRAELASYNTQLTERQAELTTIKQEIEEGEARRRNLDEITVELNYKQNEVKDFRDKINMLNQQTVELEIFRSTYDALVQESENLKEQKQQLLAEVPRLEQERDRVLAEIQKIESRAQESARLSREVTQLEARLRGKQGQRRTLEHQIEQLDAERAGLQEQIGRYQQEIQMRNGQLRELSQQIETREQQQERLGSQVRELQAQIEPLERLRGEVNQLDADLRVKRGQQEDLERRIEQLNYERAKLDDEVARRQQKIEAHDNRLKALTVELQTLQQQQEQLGSQVRELQSQVEPLERLRDEVNQLDADLRNKKSLNTNLERQIEQLQSEQWRQQAAQETESQILKDLQAQIQRAKGELKDFESSAKEALQSLESPIEIGAKQKGSFANEAQFLNAFTAYLEQKGLFFPERILRAFHTSLKVQDISALVILAGISGTGKSELPQAYAEFIGAPLVMLPVQPRWDSPQDLQGFYNYIERKYKPTELMRYLYQHSHNPQLQGRMVLVLLDEMNLARVEYYFSDFLSKLESRRNKSTYLELEVGSLKLTEQQRRVLIPKEFLFVGTMNEDETTQALSDKVLDRANVLTFGRPEELKLRGERPQTAEVSQQYLTWDQFDSWIEQPTEESELAKEVKSYVDRANGIMEKLGRPFAHRVYQAIAKYVINYPQADQNESVRKQAIADQFGQKLLPKLRGVMVEDSLVREALEQMKVLISEFGDEPLTQAFEQACQGQYGQFQWKGMVYPQQ</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Microcystis</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cb>64</td>
            <td id=al>AAA family ATPase</td>
            <td id=cr>Pseudanabaena</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>AAA domain-containing protein</td>
            <td id=co>Geminocystis</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Microcystis</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>PCC-10605</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=co>43</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Cyanobacterium</td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Crinalium</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Limnospira</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Aulosira</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>AAA family ATPase</td>
            <td id=co>Aulosira</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>31</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>31</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>27</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>21</td>
            <td id=al>Chromosome segregation ATPase [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cr>28</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cr>28</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>chromosome segregation protein SMC, primarily archaeal type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. It is found in a single copy and is homodimeric in prokaryotes, but six paralogs (excluded from this family) are found in eukarotes, where SMC proteins are heterodimeric. This family represents the SMC protein of archaea and a few bacteria (Aquifex, Synechocystis, etc); the SMC of other bacteria is described by TIGR02168. The N- and C-terminal domains of this protein are well conserved, but the central hinge region is skewed in composition and highly divergent. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>27</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>27</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cr>32</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>chromosome segregation protein SMC, common bacterial type. SMC (structural maintenance of chromosomes) proteins bind DNA and act in organizing and segregating chromosomes for partition. SMC proteins are found in bacteria, archaea, and eukaryotes. This family represents the SMC protein of most bacteria. The smc gene is often associated with scpB (TIGR00281) and scpA genes, where scp stands for segregation and condensation protein. SMC was shown (in Caulobacter crescentus) to be induced early in S phase but present and bound to DNA throughout the cell cycle. [Cellular processes, Cell division, DNA metabolism, Chromosome-associated proteins]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>5-methylcytosine-specific restriction endonuclease McrBC, GTP-binding regulatory subunit McrB [Defense mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>helix-rich Mycoplasma protein. Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>helix-rich Mycoplasma protein. Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>helix-rich Mycoplasma protein. Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>helix-rich Mycoplasma protein. Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>helix-rich Mycoplasma protein. Members of this family occur strictly within a subset of Mycoplasma species. Members average 750 amino acids in length, including signal peptide. Sequences are predicted (Jpred 3) to be almost entirely alpha-helical. These sequences show strong periodicity (consistent with long alpha helical structures) and low complexity rich in D,E,N,Q, and K. Genes encoding these proteins are often found in tandem. The function is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>DNA repair exonuclease SbcCD ATPase subunit [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>DNA repair exonuclease SbcCD ATPase subunit [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>DNA repair exonuclease SbcCD ATPase subunit [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>DNA repair exonuclease SbcCD ATPase subunit [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cr>38</td>
            <td id=al>Coiled-coil domain-containing protein 158. CCDC158 is a family of proteins found in eukaryotes. The function is not known.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>Coiled-coil domain-containing protein 158. CCDC158 is a family of proteins found in eukaryotes. The function is not known.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>Coiled-coil domain-containing protein 158. CCDC158 is a family of proteins found in eukaryotes. The function is not known.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>DNA double-strand break repair ATPase Rad50. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>DNA double-strand break repair ATPase Rad50. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>40</td>
            <td id=al>DNA double-strand break repair ATPase Rad50. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>DNA double-strand break repair ATPase Rad50. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>DNA double-strand break repair ATPase Rad50. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>exonuclease SbcC. All proteins in this family for which functions are known are part of an exonuclease complex with sbcD homologs. This complex is involved in the initiation of recombination to regulate the levels of palindromic sequences in DNA. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>15</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>exonuclease SbcC. All proteins in this family for which functions are known are part of an exonuclease complex with sbcD homologs. This complex is involved in the initiation of recombination to regulate the levels of palindromic sequences in DNA. This family is based on the phylogenomic analysis of JA Eisen (1999, Ph.D. Thesis, Stanford University). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>DNA double-strand break repair Rad50 ATPase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>DNA double-strand break repair Rad50 ATPase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>DNA double-strand break repair Rad50 ATPase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>Chromosome condensin MukBEF, ATPase and DNA-binding subunit MukB [Escherichia coli str. K-12 substr. MG1655 [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>Uncharacterized conserved protein, contains DUF3084 domain [Function unknown]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>Uncharacterized conserved protein, contains DUF3084 domain [Function unknown]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>40</td>
            <td id=al>Myosin tail. The myosin molecule is a multi-subunit complex made up of two heavy chains and four light chains it is a fundamental contractile protein found in all eukaryote cell types. This family consists of the coiled-coil myosin heavy chain tail region. The coiled-coil is composed of the tail from two molecules of myosin. These can then assemble into the macromolecular thick filament. The coiled-coil region provides the structural backbone the thick filament.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=co>41</td>
            <td id=al>Family of unknown function (DUF5401). This is a family of unknown function found in Chromadorea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>putative mechanosensitive channel protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=co>47</td>
            <td id=al>pneumococcal surface protein A. The pneumococcal surface protein proteins, found in Streptococcus pneumoniae, are repetitive, with patterns of localized high sequence identity across pairs of proteins given different specific names that recombination may be presumed. This protein, PspA, has an N-terminal region that lacks a cross-wall-targeting YSIRK type extended signal peptide, in contrast to the closely related choline-binding protein CbpA which has a similar C-terminus but a YSIRK-containing region at the N-terminus.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>Protein of unknown function (DUF3584). This protein is found in bacteria and eukaryotes. Proteins in this family are typically between 943 to 1234 amino acids in length. This family contains a P-loop motif suggesting it is a nucleotide binding protein. It may be involved in replication.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=cb>64</td>
            <td id=al>Septal ring factor EnvC, activator of murein hydrolases AmiA and AmiB [Cell cycle control, cell division, chromosome partitioning]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>330549</td></tr>
<tr><td id=al>end</td><td id=al>333222</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>AAA family ATPase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Microcystaceae; Microcystis; Microcystis</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>1</td></tr>
<tr><td id=al>tmh_topology</td><td id=al>i,7-29</td></tr>
</table>

    </div>
</body>
</html>
