<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01605</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01605 polysaccharide deacetylase family protein [Phormidium_A]</h3>
    <div>MKLIKVSPYLMRRRISLQPYPHLVAQIAIVILILLMGANRLDSLRPVTYAPKLLPLEVAPVEQTVSQHTDISLARLERLQETTQRQIERDRLPPDIYHSVPEPYRGQIVRSLPLAPGDRAIALTFDDGPSPYTAQILDILRDHDILASFFVLGRVLSTYPELLQRIVAEGHILGNHTWSHPYVVKSDALAYQEIEQTAELIYQYTRVRTQLFRPPGGYLNNALTPYAAAQNYAITMWSVDSSDYFLPRDGIIERVLNGIHPGAIVLFHDGGGPRHHTVAALPVIIKELRERGYEFVTVPELMERSQAAAAQASN</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>putative xylanase/chitin deacetylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=co>Rivularia</td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=co>Rivularia</td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>polysaccharide deacetylase family protein</td>
            <td id=co>Rivularia</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Catalytic NodB homology domain of uncharacterized bacterial glycosyl transferase, group 2-like family proteins. This family includes many uncharacterized bacterial proteins containing an N-terminal GH18 (glycosyl hydrolase, family 18) domain, a middle NodB-like homology domain, and a C-terminal GT2-like (glycosyl transferase group 2) domain. Although their biological function is unknown, members in this family contain a middle NodB homology domain that is similar to the catalytic domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Like SpPgdA, this family is a member of the carbohydrate esterase 4 (CE4) superfamily. The presence of three domains suggests that members of this family may be multifunctional.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Catalytic NodB homology domain of rhizobial NodB-like proteins. This family belongs to the large and functionally diverse carbohydrate esterase 4 (CE4) superfamily, whose members show strong sequence similarity with some variability due to their distinct carbohydrate substrates. It includes many rhizobial NodB chitooligosaccharide N-deacetylase (EC 3.5.1.-)-like proteins, mainly from bacteria and eukaryotes, such as chitin deacetylases (EC 3.5.1.41), bacterial peptidoglycan N-acetylglucosamine deacetylases (EC 3.5.1.-), and acetylxylan esterases (EC 3.1.1.72), which catalyze the N- or O-deacetylation of substrates such as acetylated chitin, peptidoglycan, and acetylated xylan. All members of this family contain a catalytic NodB homology domain with the same overall topology and a deformed (beta/alpha)8 barrel fold with 6- or 7 strands. Their catalytic activity is dependent on the presence of a divalent cation, preferably cobalt or zinc, and they employ a conserved His-His-Asp zinc-binding triad closely associated with the conserved catalytic base (aspartic acid) and acid (histidine) to carry out acid/base catalysis. Several family members show diversity both in metal ion specificities and in the residues that coordinate the metal.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Catalytic NodB homology domain of Clostridium thermocellum acetylxylan esterase and its bacterial homologs. This family is represented by Clostridium thermocellum acetylxylan esterase (CtAXE, EC 3.1.1.72), a member of the carbohydrate esterase 4 (CE4) superfamily. CtAXE deacetylates O-acetylated xylan, a key component of plant cell walls. It shows no detectable activity on generic esterase substrates including para-nitrophenyl acetate. It is specific for sugar-based substrates and will precipitate acetylxylan, as a consequence of deacetylation. CtAXE is a monomeric protein containing a catalytic NodB homology domain with the same overall topology and a deformed (beta/alpha)8 barrel fold as other CE4 esterases. However, due to differences in the topography of the substrate-binding groove, the chemistry of the active center, and metal ion coordination, CtAXE has different metal ion preference and lacks activity on N-acetyl substrates. It is significantly activated by Co2+. Moreover, CtAXE displays distinctly different ligand coordination to the metal ion, utilizing an aspartate, a histidine, and four water molecules, as opposed to the conserved His-His-Asp zinc-binding triad of other CE4 esterases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>polysaccharide deacetylase family sporulation protein PdaB. This model describes the YbaN protein family, also called PdaB and SpoVIE, of Gram-positive bacteria. Although ybaN null mutants have only a mild sporulation defect, ybaN/ytrI double mutants show drastically reducted sporulation efficiencies. This synthetic defect suggests the role of this sigmaE-controlled gene in sporulation had been masked by functional redundancy. Members of this family are homologous to a characterized polysaccharide deacetylase; the exact function this protein family is unknown. [Cellular processes, Sporulation and germination]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Catalytic NodB homology domain of uncharacterized bacterial polysaccharide deacetylases. This family includes many uncharacterized bacterial polysaccharide deacetylases. Although their biological function still remains unknown, members in this family show high sequence homology to the catalytic NodB homology domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), which is an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Like SpPgdA, this family is a member of the carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Catalytic NodB homology domain of Streptococcus mutans polysaccharide deacetylase PgdA, Bacillus subtilis YheN, and similar proteins. This family is represented by a putative polysaccharide deacetylase PgdA from the oral pathogen Streptococcus mutans (SmPgdA) and Bacillus subtilis YheN (BsYheN), which are members of the carbohydrate esterase 4 (CE4) superfamily. SmPgdA is an extracellular metal-dependent polysaccharide deacetylase with a typical CE4 fold, with metal bound to a His-His-Asp triad. It possesses de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. SmPgdA plays a role in tuning cell surface properties and in interactions with (salivary) agglutinin, an essential component of the innate immune system, most likely through deacetylation of an as-yet-unidentified polysaccharide. SmPgdA shows significant homology to the catalytic domains of peptidoglycan deacetylases from Streptococcus pneumoniae (SpPgdA) and Listeria monocytogenes (LmPgdA), both of which are involved in the bacterial defense mechanism against human mucosal lysozyme. The Bacillus subtilis genome contains six polysaccharide deacetylase gene homologs: pdaA, pdaB (previously known as ybaN), yheN, yjeA, yxkH and ylxY. The biological function of BsYheN is still unknown. This family also includes many uncharacterized polysaccharide deacetylases mainly found in bacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Putative catalytic NodB homology domain of uncharacterized BH1302 protein from Bacillus halodurans and its bacterial homologs. This family is represented by a putative polysaccharide deacetylase BH1302 from Bacillus halodurans. Although its biological function is unknown, BH1302 shows high sequence homology to the catalytic NodB homology domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), which is an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Both BH1302 and SpPgdA belong to the carbohydrate esterase 4 (CE4) superfamily. This family also includes many uncharacterized bacterial polysaccharide deacetylases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Putative catalytic NodB homology domain of uncharacterized protein YlxY from Bacillus subtilis and its bacterial homologs. The Bacillus subtilis genome contains six polysaccharide deacetylase gene homologs: pdaA, pdaB (previously known as ybaN), yheN, yjeA, yxkH and ylxY. This family is represented by Bacillus subtilis putative polysaccharide deacetylase BsYlxY, encoded by the ylxY gene, which is a member of the carbohydrate esterase 4 (CE4) superfamily. Although its biological function still remains unknown, BsYlxY shows high sequence homology to the catalytic domain of Bacillus subtilis pdaB gene encoding a putative polysaccharide deacetylase (BsPdaB), which is essential for the maintenance of spores after the late stage of sporulation and is highly conserved in spore-forming bacteria. However, disruption of the ylxY gene in B. subtilis did not cause any sporulation defect. Moreover, the Asp residue in the classical His-His-Asp zinc-binding motif of CE4 esterases is mutated to a Val residue in this family. Other catalytically relevant residues of CE4 esterases are also not conserved, which suggest that members of this family may be inactive.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Catalytic NodB homology domain of Streptococcus pneumoniae peptidoglycan deacetylase PgdA, Bacillus subtilis BsYjeA protein, and their bacterial homologs. This family is represented by Streptococcus pneumoniae peptidoglycan GlcNAc deacetylase (SpPgdA), a member of the carbohydrate esterase 4 (CE4) superfamily. SpPgdA protects gram-positive bacterial cell wall from host lysozymes by deacetylating peptidoglycan N-acetylglucosamine (GlcNAc) residues. It consists of three separate domains: N-terminal, middle and C-terminal (catalytic) domains. The catalytic NodB homology domain is similar to the deformed (beta/alpha)8 barrel fold adopted by other CE4 esterases, which harbors a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad closely associated with conserved catalytic base (aspartic acid) and acid (histidine) to carry out acid/base catalysis. The enzyme is able to accept GlcNAc3 as a substrate, with the N-acetyl of the middle sugar being removed by the enzyme. This family also includes Bacillus subtilis BsYjeA protein encoded by the yjeA gene, which is one of the six polysaccharide deacetylase gene homologs (pdaA, pdaB/ybaN, yheN, yjeA, yxkH and ylxY) in the Bacillus subtilis genome. Although homology comparison shows that the BsYjeA protein contains a polysaccharide deacetylase domain, and was predicted to be a membrane-bound xylanase or a membrane-bound chitooligosaccharide deacetylase, more recent research indicates BsYjeA might be a novel non-specific secretory endonuclease which creates random nicks progressively on the two strands of dsDNA, resulting in highly distinguishable intermediates/products very different in chemical and physical compositions over time. In addition, BsYjeA shares several enzymatic properties with the well-understood DNase I endonuclease. Both enzymes are active on ssDNA and dsDNA, both generate random nicks, and both require Mg2+ or Mn2+ for hydrolytic activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Peptidoglycan/xylan/chitin deacetylase, PgdA/CDA1 family [Carbohydrate transport and metabolism, Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Catalytic NodB homology domain of Bacillus subtilis polysaccharide deacetylase PdaA, and its bacterial homologs. The Bacillus subtilis genome contains six polysaccharide deacetylase gene homologs: pdaA, pdaB (previously known as ybaN), yheN, yjeA, yxkH and ylxY. This family is represented by Bacillus subtilis pdaA gene encoding polysaccharide deacetylase BsPdaA, which is a member of the carbohydrate esterase 4 (CE4) superfamily. BsPdaA deacetylates peptidoglycan N-acetylmuramic acid (MurNAc) residues to facilitate the formation of muramic delta-lactam, which is required for recognition of germination lytic enzymes. BsPdaA deficiency leads to the absence of muramic delta-lactam residues in the spore cortex. Like other CE4 esterases, BsPdaA consists of a single catalytic NodB homology domain that appears to adopt a deformed (beta/alpha)8 barrel fold with a putative substrate binding groove harboring the majority of the conserved residues. It utilizes a general acid/base catalytic mechanism involving a tetrahedral transition intermediate, where a water molecule functions as the nucleophile tightly associated to the zinc cofactor.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Catalytic NodB homology domain of uncharacterized chitin deacetylases and hypothetical proteins. This family includes some uncharacterized chitin deacetylases and hypothetical proteins, mainly from eukaryotes. Although their biological function is unknown, members in this family show high sequence homology to the catalytic NodB homology domain of Colletotrichum lindemuthianum chitin deacetylase (endo-chitin de-N-acetylase, ClCDA, EC 3.5.1.41), which catalyzes the hydrolysis of N-acetamido groups of N-acetyl-D-glucosamine residues in chitin, converting it to chitosan in fungal cell walls. Like ClCDA, this family is a member the carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Catalytic NodB homology domain of Colletotrichum lindemuthianum chitin deacetylase and similar proteins. This family is represented by the chitin deacetylase (endo-chitin de-N-acetylase, ClCDA, EC 3.5.1.41) from Colletotrichum lindemuthianum (also known as Glomerella lindemuthiana), which is a member of the carbohydrate esterase 4 (CE4) superfamily. ClCDA catalyzes the hydrolysis of N-acetamido groups of N-acetyl-D-glucosamine residues in chitin, converting it to chitosan in fungal cell walls. It consists of a single catalytic domain similar to the deformed (alpha/beta)8 barrel fold adopted by other CE4 esterases, which encompasses a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad closely associated with the conserved catalytic base (aspartic acid) and acid (histidine), to carry out acid/base catalysis. It possesses a highly conserved substrate-binding groove, with subtle alterations that influence substrate specificity and subsite affinity. Unlike its bacterial homologs, ClCDA contains two intramolecular disulfide bonds that may add stability to this secreted protein. The family also includes many uncharacterized deacetylases and hypothetical proteins mainly from eukaryotes, which show high sequence similarity to ClCDA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>delta-lactam-biosynthetic de-N-acetylase. Muramic delta-lactam is an unusual constituent of peptidoglycan, found only in bacterial spores in the peptidoglycan wall, or spore cortex. The proteins in this family are PdaA (yfjS), a member of a larger family of polysaccharide deacetylases, and are specificially involved in delta-lactam biosynthesis. PdaA acts immediately after CwlD, an N-acetylmuramoyl-L-alanine amidase and performs a de-N-acetylation. PdaA may also perform the following transpeptidation for lactam ring formation, as heterologous expression in E. coli of CwlD and PdaA together is sufficient for delta-lactam production. [Cell envelope, Biosynthesis and degradation of murein sacculus and peptidoglycan, Cellular processes, Sporulation and germination]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Putative catalytic NodB homology domain of Bacillus subtilis putative polysaccharide deacetylase PdaB, and its bacterial homologs. The Bacillus subtilis genome contains six polysaccharide deacetylase gene homologs: pdaA, pdaB (previously known as ybaN), yheN, yjeA, yxkH and ylxY. This family is represented by the putative polysaccharide deacetylase PdaB encoded by the pdaB gene on sporulation of Bacillus subtilis. Although its biochemical properties remain to be determined, the PdaB (YbaN) protein is essential for maintaining spores after the late stage of sporulation and is highly conserved in spore-forming bacteria. The glycans of the spore cortex may be candidate PdaB substrates. Based on sequence similarity, the family members are classified as carbohydrate esterase 4 (CE4) superfamily members. However, the classical His-His-Asp zinc-binding motif of CE4 esterases is missing in this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Putative catalytic domain of rhizobial NodB chitooligosaccharide N-deacetylase and its bacterial homologs. This family corresponds to rhizobial NodB chitooligosaccharide N-deacetylase (EC 3.5.1.-), encoded by nodB gene from the nodulation (nod) gene cluster that is responsible for the biosynthesis of bacterial nodulation signals, termed Nod factors. NodB is involved in de-N-acetylating the nonreducing N-acetylglucosamine residue of chitooligosaccharides to allow for the attachment of the fatty acyl group by the acyltransferase NodA. The monosaccharide N-acetylglucosamine cannot be deacetylated by NodB. NodB is composed of a 6-stranded barrel catalytic domain with detectable sequence similarity to the 7-stranded barrel homology domain of polysaccharide deacetylase (DCA)-like proteins in the larger carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Polysaccharide deacetylase. This domain is found in polysaccharide deacetylase. This family of polysaccharide deacetylases includes NodB (nodulation protein B from Rhizobium) which is a chitooligosaccharide deacetylase. It also includes chitin deacetylase from yeast, and endoxylanases which hydrolyzes glucosidic bonds in xylan.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Putative catalytic NodB homology domain of uncharacterized BH0857 protein from Bacillus halodurans and its bacterial homologs. This family is represented by a putative polysaccharide deacetylase BH0857 from Bacillus halodurans. Although its biological function still remains unknown, BH0857 shows high sequence homology to the catalytic NodB homology domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), which is an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Both BH0857 and SpPgdA belong to the carbohydrate esterase 4 (CE4) superfamily. This family also includes many uncharacterized bacterial polysaccharide deacetylases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Catalytic NodB homology domain of Streptomyces lividans acetylxylan esterase and its bacterial homologs. This family is represented by Streptomyces lividans acetylxylan esterase (SlAXE, EC 3.1.1.72), a member of the carbohydrate esterase 4 (CE4) superfamily. SlAXE deacetylates O-acetylated xylan, a key component of plant cell walls. It shows no detectable activity on generic esterase substrates including para-nitrophenyl acetate. It is specific for sugar-based substrates and will precipitate acetylxylan as a result of deacetylation. SlAXE also functions as a chitin and chitooligosaccharide de-N-acetylase with equal efficiency to its activity on xylan. SlAXE forms a dimer. Each monomer contains a catalytic NodB homology domain with the same overall topology and a deformed (beta/alpha)8 barrel fold as other CE4 esterases, which encompasses a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad closely associated with the conserved catalytic base (aspartic acid) and acid (histidine), to carry out acid/base catalysis. SlAXE possess a single metal center with a chemical preference for Co2+.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>Catalytic NodB homology domain of Mucor rouxii chitin deacetylase and similar proteins. This family is represented by the chitin deacetylase (MrCDA, EC 3.5.1.41) encoded from the fungus Mucor rouxii (also known as Amylomyces rouxii). MrCDA is an acidic glycoprotein with a very stringent specificity for beta1-4-linked N-acetylglucosamine homopolymers. It requires at least four residues (chitotetraose) for catalysis, and can achieve extensive deacetylation on chitin polymers. MrCDA shows high sequence similarity to Colletotrichum lindemuthianum chitin deacetylase (endo-chitin de-N-acetylase, ClCDA), which consists of a single catalytic domain similar to the deformed (beta/alpha)8 barrel fold adopted by the carbohydrate esterase 4 (CE4) superfamily, which encompasses a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad closely associated with the conserved catalytic base (aspartic acid) and acid (histidine) to carry out acid/base catalysis. The family also includes some uncharacterized eukaryotic and bacterial homologs of MrCDA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Catalytic NodB homology domain of uncharacterized bacterial polysaccharide deacetylases. This family includes many uncharacterized bacterial polysaccharide deacetylases. Although their biological function still remains unknown, members in this family show high sequence homology to the catalytic NodB homology domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), which is an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Like SpPgdA, this family is a member of the carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=co>45</td>
            <td id=al>Catalytic domain of bacterial PuuE allantoinases, Helicobacter pylori peptidoglycan deacetylase (HpPgdA), and similar proteins. This family is a member of the very large and functionally diverse carbohydrate esterase 4 (CE4) superfamily. It contains bacterial PuuE (purine utilization E) allantoinases, a peptidoglycan deacetylase from Helicobacter pylori (HpPgdA), Escherichia coli ArnD, and many uncharacterized homologs from all three kingdoms of life. PuuE allantoinase appears to be metal-independent and specifically catalyzes the hydrolysis of (S)-allantoin into allantoic acid. Different from PuuE allantoinase, HpPgdA has the ability to bind a metal ion at the active site and is responsible for a peptidoglycan modification that counteracts the host immune response. Both PuuE allantoinase and HpPgdA function as a homotetramer. The monomer is composed of a 7-stranded barrel with detectable sequence similarity to the 6-stranded barrel NodB homology domain of polysaccharide deacetylase (DCA)-like proteins in the CE4 superfamily, which removes N-linked or O-linked acetyl groups from cell wall polysaccharides. However, in contrast with the typical DCAs, PuuE allantoinase and HpPgdA might not exhibit a solvent-accessible polysaccharide binding groove and only recognize a small substrate molecule. ArnD catalyzes the deformylation of 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol to 4-amino-4-deoxy-L-arabinose-phosphoundecaprenol.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>Catalytic domain of Helicobacter pylori peptidoglycan deacetylase (HpPgdA) and similar proteins. This family is represented by a peptidoglycan deacetylase (HP0310, HpPgdA) from the gram-negative pathogen Helicobacter pylori. HpPgdA has the ability to bind a metal ion at the active site and is responsible for a peptidoglycan modification that counteracts the host immune response. It functions as a homotetramer. The monomer is composed of a 7-stranded barrel with detectable sequence similarity to the 6-stranded barrel NodB homology domain of polysaccharide deacetylase (DCA)-like proteins in the CE4 superfamily, which removes N-linked or O-linked acetyl groups from cell wall polysaccharides. In contrast to typical NodB-like DCAs, HpPgdA does not exhibit a solvent-accessible polysaccharide binding groove, suggesting that the enzyme binds a small molecule at the active site.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Putative catalytic NodB homology domain of uncharacterized Mll8295 protein encoded from Rhizobium loti and its bacterial homologs. This family is represented by a putative polysaccharide deacetylase Mll8295 encoded from Rhizobium loti. Although its biological function still remains unknown, Mll8295 shows high sequence homology to the catalytic domain of Streptococcus pneumoniae polysaccharide deacetylase PgdA (SpPgdA), which is an extracellular metal-dependent polysaccharide deacetylase with de-N-acetylase activity toward a hexamer of chitooligosaccharide N-acetylglucosamine, but not shorter chitooligosaccharides or a synthetic peptidoglycan tetrasaccharide. Both Mll8295 and SpPgdA belong to the carbohydrate esterase 4 (CE4) superfamily. This family also includes many uncharacterized bacterial polysaccharide deacetylases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>Putative catalytic domain of uncharacterized prokaryotic polysaccharide deacetylases similar to bacterial PuuE allantoinases and Helicobacter pylori peptidoglycan deacetylase (HpPgdA). This family contains many uncharacterized prokaryotic polysaccharide deacetylases (DCAs) that show high sequence similarity to the catalytic domain of bacterial PuuE allantoinases and Helicobacter pylori peptidoglycan deacetylase (HpPgdA). PuuE allantoinase appears to be metal-independent and specifically catalyzes the hydrolysis of (S)-allantoin into allantoic acid. Different from PuuE allantoinase, HpPgdA has the ability to bind a metal ion at the active site and is responsible for a peptidoglycan modification that counteracts the host immune response. Both PuuE allantoinase and HpPgdA function as homotetramers. The monomer is composed of a 7-stranded barrel with detectable sequence similarity to the 6-stranded barrel NodB homology domain of DCA-like proteins in the CE4 superfamily, which removes N-linked or O-linked acetyl groups from cell wall polysaccharides. In contrast to typical NodB-like DCAs, PuuE allantoinase and HpPgdA do not exhibit a solvent-accessible polysaccharide binding groove and might only bind a small molecule at the active site.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>Putative catalytic domain of uncharacterized bacterial proteins from the carbohydrate esterase 4 superfamily. This family corresponds to a group of uncharacterized bacterial proteins with high sequence similarity to the catalytic domain of the six-stranded barrel rhizobial NodB-like proteins, which remove N-linked or O-linked acetyl groups from cell wall polysaccharides and belong to the larger carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>58</td>
            <td id=al>Putative catalytic NodB homology domain of gellan lyase and similar proteins. This family is represented by the extracellular polysaccharide-degrading enzyme, gellan lyase (gellanase, EC 4.2.2.-), from Bacillus sp. The enzyme acts on gellan exolytically and releases a tetrasaccharide of glucuronyl-glucosyl-rhamnosyl-glucose with unsaturated glucuronic acid at the nonreducing terminus. The family also includes many uncharacterized prokaryotic polysaccharide deacetylases, which show high sequence similarity to Bacillus sp. gellan lyase. Although their biological functions remain unknown, all members of the family contain a conserved domain with a 6-stranded beta/alpha barrel, which is similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, belonging to the larger carbohydrate esterase 4 (CE4) superfamily.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>Catalytic NodB homology domain of the carbohydrate esterase 4 superfamily. The carbohydrate esterase 4 (CE4) superfamily mainly includes chitin deacetylases (EC 3.5.1.41), bacterial peptidoglycan N-acetylglucosamine deacetylases (EC 3.5.1.-), and acetylxylan esterases (EC 3.1.1.72), which catalyze the N- or O-deacetylation of substrates such as acetylated chitin, peptidoglycan, and acetylated xylan, respectively. Members in this superfamily contain a NodB homology domain that adopts a deformed (beta/alpha)8 barrel fold, which encompasses a mononuclear metalloenzyme employing a conserved His-His-Asp zinc-binding triad, closely associated with the conserved catalytic base (aspartic acid) and acid (histidine) to carry out acid/base catalysis. The NodB homology domain of CE4 superfamily is remotely related to the 7-stranded beta/alpha barrel catalytic domain of the superfamily consisting of family 38 glycoside hydrolases (GH38), family 57 heat stable retaining glycoside hydrolases (GH57), lactam utilization protein LamB/YcsF family proteins, and YdjC-family proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Putative catalytic NodB homology domain of PgaB, IcaB, and similar proteins which consist of a deformed (beta/alpha)8 barrel fold with 5- or 6-strands. This family belongs to the large and functionally diverse carbohydrate esterase 4 (CE4) superfamily, whose members show strong sequence similarity with some variability due to their distinct carbohydrate substrates. It includes bacterial poly-beta-1,6-N-acetyl-D-glucosamine N-deacetylase PgaB, hemin storage system HmsF protein in gram-negative species, intercellular adhesion proteins IcaB, and many uncharacterized prokaryotic polysaccharide deacetylases. It also includes a putative polysaccharide deacetylase YxkH encoded by the Bacillus subtilis yxkH gene, which is one of six polysaccharide deacetylase gene homologs present in the Bacillus subtilis genome. Sequence comparison shows all family members contain a conserved domain similar to the catalytic NodB homology domain of rhizobial NodB-like proteins, which consists of a deformed (beta/alpha)8 barrel fold with 6 or 7 strands. However, in this family, most proteins have 5 strands and some have 6 strands. Moreover, long insertions are found in many family members, whose function remains unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>Putative catalytic domain of uncharacterized bacterial polysaccharide deacetylases similar to bacterial PuuE allantoinases and Helicobacter pylori peptidoglycan deacetylase (HpPgdA). This family contains many uncharacterized bacterial polysaccharide deacetylases (DCAs) that show high sequence similarity to the catalytic domain of bacterial PuuE allantoinases and Helicobacter pylori peptidoglycan deacetylase (HpPgdA). PuuE allantoinase appears to be metal-independent and specifically catalyzes the hydrolysis of (S)-allantoin into allantoic acid. Different from PuuE allantoinase, HpPgdA has the ability to bind a metal ion at the active site and is responsible for a peptidoglycan modification that counteracts the host immune response. Both PuuE allantoinase and HpPgdA function as homotetramers. The monomer is composed of a 7-stranded barrel with detectable sequence similarity to the 6-stranded barrel NodB homology domain of DCA-like proteins in the CE4 superfamily, which removes N-linked or O-linked acetyl groups from cell wall polysaccharides. In contrast to typical NodB-like DCAs, PuuE allantoinase and HpPgdA do not exhibit a solvent-accessible polysaccharide binding groove and might only bind a small molecule at the active site.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1361157</td></tr>
<tr><td id=al>end</td><td id=al>1362102</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>polysaccharide deacetylase family protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>1</td></tr>
<tr><td id=al>tmh_topology</td><td id=al>i,20-39</td></tr>
</table>

    </div>
</body>
</html>
