<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05760</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05760 nucleoside triphosphate pyrophosphohydrolase [Phormidium_A]</h3>
    <div>MSDRPSNPSQALTALEQLIAVVAQLRDPDGGCPWDLEQTPESLIPYIIEEAYESVDAIRRGDSEHISEELGDLLLQVVLQAQIAQEAGDFSLAEVAQGITDKLIRRHPHVFADTEANTSEEVRQNWQAIKAAEKGDDETKLSSKLARYARTLPPLRASQKISKKAAEVGFEWEKVEDVWTKFDEELGEFKEAIAQESRQRQEEELGDLLFVLVNLARWYDLDPEAALQGTHDRFLARFTQVEAACDRPLSDYSVEELEALWQIAKQRLNS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>98</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>XTP/dITP diphosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>89</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>98</td>
            <td id=cg>98</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>74</td>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>95</td>
            <td id=cg>95</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=co>Chroococcidiopsis</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>96</td>
            <td id=cg>95</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=co>Chroococcidiopsis</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>98</td>
            <td id=cg>95</td>
            <td id=al>nucleoside triphosphate pyrophosphohydrolase</td>
            <td id=co>Kamptonema</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>53</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>PRK09562 nucleoside triphosphate pyrophosphohydrolase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>TIGR00444 MazG family protein. This family of prokaryotic proteins has no known function. It includes the uncharacterized protein MazG in E. coli. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>93</td>
            <td id=co>51</td>
            <td id=al>COG3956 Uncharacterized conserved protein YabN, contains tetrapyrrole methylase and MazG-like pyrophosphatase domain [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=al>cd11528 Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) N-terminal tandem-domain of MazG proteins from Escherichia coli and bacterial homologs. MazG is a NTP-PPase that hydrolyzes all canonical NTPs into their corresponding nucleoside monophosphates and pyrophosphate. The prototype of this family is MazG proteins from Escherichia coli (EcMazG) that represents the most abundant form consisting two sequence-related domains in tandem, this family corresponding to the N-terminal MazG-like domain. EcMazG functions as a regulator of cellular response to starvation by lowering the cellular concentration of guanosine 3',5'-bispyrophosphate (ppGpp). EcMazG exists as a dimer; each monomer contains two tandem MazG-like domains with similarly folded globular structures. However, only the C-terminal domain has well-ordered active site and exhibits an NTPase activity responsible for the regulation of bacterial cell survival under nutritional stress. Divalent ions, such as Mg2+ or Mn2+, are required for activity; however, this domain does not exhibit an NTPase activity despite containing structural features such as the EEXX(E/D) motif and key basic catalytic residues responsible for nucleotide pyrophosphohydrolysis activity. It is suggested that the N-terminal domain of EcMazG might have a house-cleaning function by hydrolyzing noncanonical NTPs whose incorporation into the nascent DNA leads to increased mutagenesis and DNA damage.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>90</td>
            <td id=cg>88</td>
            <td id=al>PRK12334 nucleoside triphosphate pyrophosphohydrolase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=al>PRK12333 nucleoside triphosphate pyrophosphohydrolase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=al>cd11529 Nucleoside Triphosphate Pyrophosphohydrolase (EC 3.6.1.8) C-terminal tandem-domain of MazG proteins from Escherichia coli and bacterial homologs'. MazG is a NTP-PPase that hydrolyzes all canonical NTPs into their corresponding nucleoside monophosphates and pyrophosphate. The prototype of this family is MazG proteins from Escherichia coli (EcMazG) that represents the most abundant form consisting two sequence-related domains in tandem, this family corresponding to the C-terminal MazG-like domain. EcMazG functions as a regulator of cellular response to starvation by lowering the cellular concentration of guanosine 3',5'-bispyrophosphate (ppGpp). EcMazG exists as a dimer. Each monomer contains two tandem MazG-like domains with similarly folded globular structures. However, only the C-terminal domain has well-ordered active sites and exhibits an NTPase activity responsible for the regulation of bacterial cell survival under nutritional stress. Divalent ions, such as Mg2+ or Mn2+, are required for activity, along with structural features such as EEXX(E/D) motifs and key basic catalytic residues. It has been shown that the C-terminus NTPase activity is responsible for regulation of bacterial cell survival under nutritional stress.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=al>pfam03819 MazG nucleotide pyrophosphohydrolase domain. This domain is about 100 amino acid residues in length. It is found in the MazG protein from E. coli. It contains four conserved negatively charged residues that probably form an active site or metal binding site. This domain is found in isolation in some proteins as well as associated with pfam00590. This domain is clearly related to pfam01503 another pyrophosphohydrolase involved in histidine biosynthesis. This family may be structurally related to the NUDIX domain pfam00293 (Bateman A pers. obs.).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cr>37</td>
            <td id=cg>99</td>
            <td id=al>COG1694 NTP pyrophosphatase, house-cleaning of non-canonical NTPs [Defense mechanisms]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4714905</td></tr>
<tr><td id=al>end</td><td id=al>4715718</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>nucleoside triphosphate pyrophosphohydrolase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>271</td></tr>
</table>

    </div>
</body>
</html>
