<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03192</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03192 RNA-binding protein [Phormidium_A]</h3>
    <div>MSVYVGNLSYEVTSEDLSDVFSEYGTIKRIHVPTDRETGRMRGFAFVEMEQESEEAAAIEALDGAEWMDRVIRVNPAKPRTDKPRYNRR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding proteins (RRM domain)</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>RNA-binding protein</td>
            <td id=co>Gomphosphaeria</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>RNA-binding protein</td>
            <td id=co>Okeania</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cb>79</td>
            <td id=al>RNA-binding protein</td>
            <td id=cr>Myxacorys</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>RNA-binding protein</td>
            <td id=co>Okeania</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in Nicotiana sylvestris chloroplastic 33 kDa ribonucleoprotein (NsCP33) and similar proteins. The family includes NsCP33, Arabidopsis thaliana chloroplastic 31 kDa ribonucleoprotein (CP31A) and mitochondrial glycine-rich RNA-binding protein 2 (AtGR-RBP2). NsCP33 may be involved in splicing and/or processing of chloroplast RNA's. AtCP31A, also called RNA-binding protein 1/2/3 (AtRBP33), or RNA-binding protein CP31A, or RNA-binding protein RNP-T, or RNA-binding protein cp31, is required for specific RNA editing events in chloroplasts and stabilizes specific chloroplast mRNAs, as well as for normal chloroplast development under cold stress conditions by stabilizing transcripts of numerous mRNAs under these conditions. CP31A may modulate telomere replication through RNA binding domains. AtGR-RBP2, also called AtRBG2, or glycine-rich protein 2 (AtGRP2), or mitochondrial RNA-binding protein 1a (At-mRBP1a), plays a role in RNA transcription or processing during stress. It binds RNAs and DNAs sequence with a preference to single-stranded nucleic acids. AtGR-RBP2 displays strong affinity to poly(U) sequence. It exerts cold and freezing tolerance, probably by exhibiting an RNA chaperone activity during the cold and freezing adaptation process. Some members in this family contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the second RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cr>29</td>
            <td id=al>RNA recognition motif (RRM) domain [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif (RRM) found in Helicobacter pylori HP0827 protein and similar proteins. This subfamily corresponds to the RRM of H. pylori HP0827, a putative ssDNA-binding protein 12rnp2 precursor, containing one RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). The ssDNA binding may be important in activation of HP0827. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>RNA recognition motif (RRM) found in cleavage stimulation factor subunit 2 (CSTF2), yeast ortholog mRNA 3'-end-processing protein RNA15 and similar proteins. This subfamily corresponds to the RRM domain of CSTF2, its tau variant and eukaryotic homologs. CSTF2, also termed cleavage stimulation factor 64 kDa subunit (CstF64), is the vertebrate conterpart of yeast mRNA 3'-end-processing protein RNA15. It is expressed in all somatic tissues and is one of three cleavage stimulatory factor (CstF) subunits required for polyadenylation. CstF64 contains an N-terminal RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), a CstF77-binding domain, a repeated MEARA helical region and a conserved C-terminal domain reported to bind the transcription factor PC-4. During polyadenylation, CstF interacts with the pre-mRNA through the RRM of CstF64 at U- or GU-rich sequences within 10 to 30 nucleotides downstream of the cleavage site. CSTF2T, also termed tauCstF64, is a paralog of the X-linked cleavage stimulation factor CstF64 protein that supports polyadenylation in most somatic cells. It is expressed during meiosis and subsequent haploid differentiation in a more limited set of tissues and cell types, largely in meiotic and postmeiotic male germ cells, and to a lesser extent in brain. The loss of CSTF2T will cause male infertility, as it is necessary for spermatogenesis and fertilization. Moreover, CSTF2T is required for expression of genes involved in morphological differentiation of spermatids, as well as for genes having products that function during interaction of motile spermatozoa with eggs. It promotes germ cell-specific patterns of polyadenylation by using its RRM to bind to different sequence elements downstream of polyadenylation sites than does CstF64. The family also includes yeast ortholog mRNA 3'-end-processing protein RNA15 and similar proteins. RNA15 is a core subunit of cleavage factor IA (CFIA), an essential transcriptional 3'-end processing factor from Saccharomyces cerevisiae. RNA recognition by CFIA is mediated by an N-terminal RRM, which is contained in the RNA15 subunit of the complex. The RRM of RNA15 has a strong preference for GU-rich RNAs, mediated by a binding pocket that is entirely conserved in both yeast and vertebrate RNA15 orthologs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in yeast protein gar2 and similar proteins. This subfamily corresponds to the RRM2 of yeast protein gar2, a novel nucleolar protein required for 18S rRNA and 40S ribosomal subunit accumulation. It shares similar domain architecture with nucleolin from vertebrates and NSR1 from Saccharomyces cerevisiae. The highly phosphorylated N-terminal domain of gar2 is made up of highly acidic regions separated from each other by basic sequences, and contains multiple phosphorylation sites. The central domain of gar2 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The C-terminal RGG (or GAR) domain of gar2 is rich in glycine, arginine and phenylalanine residues. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif (RRM) found in cyclophilin-33 (Cyp33) and similar proteins. This subfamily corresponds to the RRM of Cyp33, also termed peptidyl-prolyl cis-trans isomerase E (PPIase E), or cyclophilin E, or rotamase E. Cyp33 is a nuclear RNA-binding cyclophilin with an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a C-terminal PPIase domain. Cyp33 possesses RNA-binding activity and preferentially binds to polyribonucleotide polyA and polyU, but hardly to polyG and polyC. It binds specifically to mRNA, which can stimulate its PPIase activity. Moreover, Cyp33 interacts with the third plant homeodomain (PHD3) zinc finger cassette of the mixed lineage leukemia (MLL) proto-oncoprotein and a poly-A RNA sequence through its RRM domain. It further mediates downregulation of the expression of MLL target genes HOXC8, HOXA9, CDKN1B, and C-MYC, in a proline isomerase-dependent manner. Cyp33 also possesses a PPIase activity that catalyzes cis-trans isomerization of the peptide bond preceding a proline, which has been implicated in the stimulation of folding and conformational changes in folded and unfolded proteins. The PPIase activity can be inhibited by the immunosuppressive drug cyclosporin A. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in chloroplastic plastid-specific 30S ribosomal protein 2 (PSRP-2) and similar proteins. PSRP-2, also called chloroplastic 30S ribosomal protein 2, or chloroplastic small ribosomal subunit protein cS22, is a component of the chloroplast ribosome (chloro-ribosome), a dedicated translation machinery responsible for the synthesis of chloroplast genome-encoded proteins, including proteins of the transcription and translation machinery and components of the photosynthetic apparatus. It binds single strand DNA (ssDNA) and RNA in vitro. It exhibits RNA chaperone activity and regulates negatively resistance responses to abiotic stresses during seed germination (e.g. salt, dehydration, and low temperature) and seedling growth (e.g. salt). The family also includes Nicotiana sylvestris chloroplastic 33 kDa ribonucleoprotein (NsCP33) and Arabidopsis thaliana chloroplastic 31 kDa ribonucleoprotein (AtCP31A). NsCP33 may be involved in splicing and/or processing of chloroplast RNA's. AtCP31A, also called RNA-binding protein 1/2/3 (AtRBP33), or RNA-binding protein CP31A, or RNA-binding protein RNP-T, or RNA-binding protein cp31, is required for specific RNA editing events in chloroplasts and stabilizes specific chloroplast mRNAs, as well as for normal chloroplast development under cold stress conditions by stabilizing transcripts of numerous mRNAs under these conditions. CP31A may modulate telomere replication through RNA binding domains. Members in this family contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain). The RRM motif is probably diagnostic of an RNA binding protein. RRMs are found in a variety of RNA binding proteins, including various hnRNP proteins, proteins implicated in regulation of alternative splicing, and protein components of snRNPs. The motif also appears in a few single stranded DNA binding proteins. The RRM structure consists of four strands and two helices arranged in an alpha/beta sandwich, with a third helix present during RNA binding in some cases The C-terminal beta strand (4th strand) and final helix are hard to align and have been omitted in the SEED alignment The LA proteins have an N terminal rrm which is included in the seed. There is a second region towards the C-terminus that has some features characteristic of a rrm but does not appear to have the important structural core of a rrm. The LA proteins are one of the main autoantigens in Systemic lupus erythematosus (SLE), an autoimmune disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>glycine-rich RNA-binding protein 4; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>RNA recognition motif (RRM) superfamily. RRM, also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), is a highly abundant domain in eukaryotes found in proteins involved in post-transcriptional gene expression processes including mRNA and rRNA processing, RNA export, and RNA stability. This domain is 90 amino acids in length and consists of a four-stranded beta-sheet packed against two alpha-helices. RRM usually interacts with ssRNA, but is also known to interact with ssDNA as well as proteins. RRM binds a variable number of nucleotides, ranging from two to eight. The active site includes three aromatic side-chains located within the conserved RNP1 and RNP2 motifs of the domain. The RRM domain is found in a variety heterogeneous nuclear ribonucleoproteins (hnRNPs), proteins implicated in regulation of alternative splicing, and protein components of small nuclear ribonucleoproteins (snRNPs).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif (RRM) found in eukaryotic translation initiation factor 3 subunit G (eIF-3G) and similar proteins. This subfamily corresponds to the RRM of eIF-3G and similar proteins. eIF-3G, also termed eIF-3 subunit 4, or eIF-3-delta, or eIF3-p42, or eIF3-p44, is the RNA-binding subunit of eIF3, a large multisubunit complex that plays a central role in the initiation of translation by binding to the 40 S ribosomal subunit and promoting the binding of methionyl-tRNAi and mRNA. eIF-3G binds 18 S rRNA and beta-globin mRNA, and therefore appears to be a nonspecific RNA-binding protein. eIF-3G is one of the cytosolic targets and interacts with mature apoptosis-inducing factor (AIF). eIF-3G contains one RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). This family also includes yeast eIF3-p33, a homolog of vertebrate eIF-3G, plays an important role in the initiation phase of protein synthesis in yeast. It binds both, mRNA and rRNA, fragments due to an RRM near its C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>RNA recognition motif (RRM) found in ist3 family. This subfamily corresponds to the RRM of the ist3 family that includes fungal U2 small nuclear ribonucleoprotein (snRNP) component increased sodium tolerance protein 3 (ist3), X-linked 2 RNA-binding motif proteins (RBMX2) found in Metazoa and plants, and similar proteins. Gene IST3 encoding ist3, also termed U2 snRNP protein SNU17 (Snu17p), is a novel yeast Saccharomyces cerevisiae protein required for the first catalytic step of splicing and for progression of spliceosome assembly. It binds specifically to the U2 snRNP and is an intrinsic component of prespliceosomes and spliceosomes. Yeast ist3 contains an atypical RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). In the yeast pre-mRNA retention and splicing complex, the atypical RRM of ist3 functions as a scaffold that organizes the other two constituents, Bud13p (bud site selection 13) and Pml1p (pre-mRNA leakage 1). Fission yeast Schizosaccharomyces pombe gene cwf29 encoding ist3, also termed cell cycle control protein cwf29, is an RNA-binding protein complexed with cdc5 protein 29. It also contains one RRM. The biological function of RBMX2 remains unclear. It shows high sequence similarity to yeast ist3 protein and harbors one RRM as well. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif 3 (RRM3) found in RNA-binding protein 19 (RBM19) and RNA recognition motif 2 found in multiple RNA-binding domain-containing protein 1 (MRD1). This subfamily corresponds to the RRM3 of RBM19 and RRM2 of MRD1. RBM19, also termed RNA-binding domain-1 (RBD-1), is a nucleolar protein conserved in eukaryotes involved in ribosome biogenesis by processing rRNA and is essential for preimplantation development. It has a unique domain organization containing 6 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). MRD1 is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). It is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. MRD1 contains 5 conserved RRMs, which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in Arabidopsis thaliana phragmoplastin interacting protein 1 (PHIP1) and similar proteins. The CD corresponds to the RRM2 of PHIP1. A. thaliana PHIP1 and its homologs represent a novel class of plant-specific RNA-binding proteins that may play a unique role in the polarized mRNA transport to the vicinity of the cell plate. The family members consist of multiple functional domains, including a lysine-rich domain (KRD domain) that contains three nuclear localization motifs (KKKR/NK), two RNA recognition motifs (RRMs), and three CCHC-type zinc fingers. PHIP1 is a peripheral membrane protein and is localized at the cell plate during cytokinesis in plants. In addition to phragmoplastin, PHIP1 interacts with two Arabidopsis small GTP-binding proteins, Rop1 and Ran2. However, PHIP1 interacted only with the GTP-bound form of Rop1 but not the GDP-bound form. It also binds specifically to Ran2 mRNA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>RNA recognition motif (RRM) found in cold inducible RNA binding protein (CIRBP), RNA binding motif protein 3 (RBM3) and similar proteins. This subfamily corresponds to the RRM domain of two structurally related heterogenous nuclear ribonucleoproteins, CIRBP (also termed CIRP or A18 hnRNP) and RBM3 (also termed RNPL), both of which belong to a highly conserved cold shock proteins family. The cold shock proteins can be induced after exposure to a moderate cold-shock and other cellular stresses such as UV radiation and hypoxia. CIRBP and RBM3 may function in posttranscriptional regulation of gene expression by binding to different transcripts, thus allowing the cell to response rapidly to environmental signals. However, the kinetics and degree of cold induction are different between CIRBP and RBM3. Tissue distribution of their expression is different. CIRBP and RBM3 may be differentially regulated under physiological and stress conditions and may play distinct roles in cold responses of cells. CIRBP, also termed glycine-rich RNA-binding protein CIRP, is localized in the nucleus and mediates the cold-induced suppression of cell cycle progression. CIRBP also binds DNA and possibly serves as a chaperone that assists in the folding/unfolding, assembly/disassembly and transport of various proteins. RBM3 may enhance global protein synthesis and the formation of active polysomes while reducing the levels of ribonucleoprotein complexes containing microRNAs. RBM3 may also serve to prevent the loss of muscle mass by its ability to decrease cell death. Furthermore, RBM3 may be essential for cell proliferation and mitosis. Both, CIRBP and RBM3, contain an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), that is involved in RNA binding, and C-terminal glycine-rich domain (RGG motif) that probably enhances RNA-binding via protein-protein and/or protein-RNA interactions. Like CIRBP, RBM3 can also bind to both RNA and DNA via its RRM domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM1 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in yeast multiple RNA-binding domain-containing protein 1 (MRD1) and similar proteins. This subgroup corresponds to the RRM2 of MRD1 which is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well-conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). It is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. MRD1 contains 5 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>RNA recognition motif 3 (RRM3) found in RNA-binding protein 19 (RBM19) and similar proteins. This subgroup corresponds to the RRM3 of RBM19, also termed RNA-binding domain-1 (RBD-1), which is a nucleolar protein conserved in eukaryotes. It is involved in ribosome biogenesis by processing rRNA. In addition, it is essential for preimplantation development. RBM19 has a unique domain organization containing 6 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>RNA recognition motif (RRM) found in cleavage stimulation factor subunit 2 (CSTF2), cleavage stimulation factor subunit 2 tau variant (CSTF2T) and similar proteins. This subgroup corresponds to the RRM domain of CSTF2, its tau variant and eukaryotic homologs. CSTF2, also termed cleavage stimulation factor 64 kDa subunit (CstF64), is the vertebrate conterpart of yeast mRNA 3'-end-processing protein RNA15. It is expressed in all somatic tissues and is one of three cleavage stimulatory factor (CstF) subunits required for polyadenylation. CstF64 contains an N-terminal RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), a CstF77-binding domain, a repeated MEARA helical region and a conserved C-terminal domain reported to bind the transcription factor PC-4. During polyadenylation, CstF interacts with the pre-mRNA through the RRM of CstF64 at U- or GU-rich sequences within 10 to 30 nucleotides downstream of the cleavage site. CSTF2T, also termed tauCstF64, is a paralog of the X-linked cleavage stimulation factor CstF64 protein that supports polyadenylation in most somatic cells. It is expressed during meiosis and subsequent haploid differentiation in a more limited set of tissues and cell types, largely in meiotic and postmeiotic male germ cells, and to a lesser extent in brain. The loss of CSTF2T will cause male infertility, as it is necessary for spermatogenesis and fertilization. Moreover, CSTF2T is required for expression of genes involved in morphological differentiation of spermatids, as well as for genes having products that function during interaction of motile spermatozoa with eggs. It promotes germ cell-specific patterns of polyadenylation by using its RRM to bind to different sequence elements downstream of polyadenylation sites than does CstF64. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in vertebrate RNA-binding protein RBM23, RBM39 and similar proteins. This subfamily corresponds to the RRM2 of RBM39 (also termed HCC1), a nuclear autoantigen that contains an N-terminal arginine/serine rich (RS) motif and three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). An octapeptide sequence called the RS-ERK motif is repeated six times in the RS region of RBM39. Although the cellular function of RBM23 remains unclear, it shows high sequence homology to RBM39 and contains two RRMs. It may possibly function as a pre-mRNA splicing factor. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in nucleolin-like proteins mainly from plants. This subfamily corresponds to the RRM1 of a group of plant nucleolin-like proteins, including nucleolin 1 (also termed protein nucleolin like 1) and nucleolin 2 (also termed protein nucleolin like 2, or protein parallel like 1). They play roles in the regulation of ribosome synthesis and in the growth and development of plants. Like yeast nucleolin, nucleolin-like proteins possess two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains).  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in yeast nucleolar protein 13 (Nop13p) and similar proteins. This subfamily corresponds to the RRM2 of Nop13p encoded by YNL175c from Saccharomyces cerevisiae. It shares high sequence similarity with nucleolar protein 12 (Nop12p). Both Nop12p and Nop13p are not essential for growth. However, unlike Nop12p that is localized to the nucleolus, Nop13p localizes primarily to the nucleolus but is also present in the nucleoplasm to a lesser extent. Nop13p contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>RNA recognition motif in type II polyadenylate-binding proteins. This subfamily corresponds to the RRM of type II polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 2 (PABP-2 or PABPN1), embryonic polyadenylate-binding protein 2 (ePABP-2 or PABPN1L) and similar proteins. PABPs are highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in the regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. ePABP-2 is predominantly located in the cytoplasm and PABP-2 is located in the nucleus. In contrast to the type I PABPs containing four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), the type II PABPs contains a single highly-conserved RRM. This subfamily also includes Saccharomyces cerevisiae RBP29 (SGN1, YIR001C) gene encoding cytoplasmic mRNA-binding protein Rbp29 that binds preferentially to poly(A). Although not essential for cell viability, Rbp29 plays a role in modulating the expression of cytoplasmic mRNA. Like other type II PABPs, Rbp29 contains one RRM only. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif (RRM) found in Saccharomyces cerevisiae nucleolar protein 6 (Nop6) and similar proteins. This subfamily corresponds to the RRM of Nop6, also known as Ydl213c, a component of 90S pre-ribosomal particles in yeast S. cerevisiae. It is enriched in the nucleolus and is required for 40S ribosomal subunit biogenesis. Nop6 is a non-essential putative RNA-binding protein with two N-terminal putative nuclear localisation sequences (NLS-1 and NLS-2) and an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). It binds to the pre-rRNA early during transcription and plays an essential role in pre-rRNA processing. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>RNA recognition motif (RRM) found in serine/arginine-rich splicing factor SRSF2, SRSF8 and similar proteins. This subfamily corresponds to the RRM of SRSF2 and SRSF8. SRSF2, also termed protein PR264, or splicing component, 35 kDa (splicing factor SC35 or SC-35), is a prototypical SR protein that plays important roles in the alternative splicing of pre-mRNA. It is also involved in transcription elongation by directly or indirectly mediating the recruitment of elongation factors to the C-terminal domain of polymerase II. SRSF2 is exclusively localized in the nucleus and is restricted to nuclear processes. It contains a single N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), followed by a C-terminal RS domain rich in serine-arginine dipeptides. The RRM is responsible for the specific recognition of 5'-SSNG-3' (S=C/G) RNA. In the regulation of alternative splicing events, it specifically binds to cis-regulatory elements on the pre-mRNA. The RS domain modulates SRSF2 activity through phosphorylation, directly contacts RNA, and promotes protein-protein interactions with the spliceosome. SRSF8, also termed SRP46 or SFRS2B, is a novel mammalian SR splicing factor encoded by a PR264/SC35 functional retropseudogene. SRSF8 is localized in the nucleus and does not display the same activity as PR264/SC35. It functions as an essential splicing factor in complementing a HeLa cell S100 extract deficient in SR proteins. Like SRSF2, SRSF8 contains a single N-terminal RRM and a C-terminal RS domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein hnRNP A and hnRNP D subfamilies and similar proteins. This subfamily corresponds to the RRM1 in the hnRNP A subfamily which includes hnRNP A0, hnRNP A1, hnRNP A2/B1, hnRNP A3 and similar proteins. hnRNP A0 is a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. hnRNP A1 is an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A2/B1 is an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). hnRNP A3 is also a RNA trafficking response element-binding protein that participates in the trafficking of A2RE-containing RNA. The hnRNP A subfamily is characterized by two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. The hnRNP D subfamily includes hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. hnRNP D0 is a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP A/B is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. hnRNP A/B has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus, plays an important role in apoB mRNA editing. hnRNP DL (or hnRNP D-like) is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. All members in this subfamily contain two putative RRMs and a glycine- and tyrosine-rich C-terminus. The family also contains DAZAP1 (Deleted in azoospermia-associated protein 1), RNA-binding protein Musashi homolog Musashi-1, Musashi-2 and similar proteins. They all harbor two RRMs. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>RNA recognition motif (RRM) found in transformer-2 protein homolog TRA2-alpha, TRA2-beta and similar proteins. This subfamily corresponds to the RRM of two mammalian homologs of Drosophila transformer-2 (Tra2), TRA2-alpha, TRA2-beta (also termed SFRS10), and similar proteins found in eukaryotes. TRA2-alpha is a 40-kDa serine/arginine-rich (SR) protein that specifically binds to gonadotropin-releasing hormone (GnRH) exonic splicing enhancer on exon 4 (ESE4) and is necessary for enhanced GnRH pre-mRNA splicing. It strongly stimulates GnRH intron A excision in a dose-dependent manner. In addition, TRA2-alpha can interact with either 9G8 or SRp30c, which may also be crucial for ESE-dependent GnRH pre-mRNA splicing. TRA2-beta is a serine/arginine-rich (SR) protein that controls the pre-mRNA alternative splicing of the calcitonin/calcitonin gene-related peptide (CGRP), the survival motor neuron 1 (SMN1) protein and the tau protein. Both, TRA2-alpha and TRA2-beta, contains a well conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), flanked by the N- and C-terminal arginine/serine (RS)-rich regions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A subfamily. This subfamily corresponds to the RRM2 of hnRNP A0, hnRNP A1, hnRNP A2/B1, hnRNP A3 and similar proteins. hnRNP A0 is a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. It has been identified as the substrate for MAPKAP-K2 and may be involved in the lipopolysaccharide (LPS)-induced post-transcriptional regulation of tumor necrosis factor-alpha (TNF-alpha), cyclooxygenase 2 (COX-2) and macrophage inflammatory protein 2 (MIP-2). hnRNP A1 is an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A2/B1 is an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). Many mRNAs, such as myelin basic protein (MBP), myelin-associated oligodendrocytic basic protein (MOBP), carboxyanhydrase II (CAII), microtubule-associated protein tau, and amyloid precursor protein (APP) are trafficked by hnRNP A2/B1. hnRNP A3 is also a RNA trafficking response element-binding protein that participates in the trafficking of A2RE-containing RNA. The hnRNP A subfamily is characterized by two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>RNA recognition motif (RRM) found in SRA stem-loop-interacting RNA-binding protein (SLIRP) and similar proteins. This subfamily corresponds to the RRM of SLIRP, a widely expressed small steroid receptor RNA activator (SRA) binding protein, which binds to STR7, a functional substructure of SRA. SLIRP is localized predominantly to the mitochondria and plays a key role in modulating several nuclear receptor (NR) pathways. It functions as a co-repressor to repress SRA-mediated nuclear receptor coactivation. It modulates SHARP- and SKIP-mediated co-regulation of NR activity. SLIRP contains an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), which is required for SLIRP's corepression activities. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>RNA recognition motif (RRM) found in RNA-binding protein with serine-rich domain 1 (RNPS1) and similar proteins. This subfamily corresponds to the RRM of RNPS1 and its eukaryotic homologs. RNPS1, also termed RNA-binding protein prevalent during the S phase, or SR-related protein LDC2, was originally characterized as a general pre-mRNA splicing activator, which activates both constitutive and alternative splicing of pre-mRNA in vitro.It has been identified as a protein component of the splicing-dependent mRNP complex, or exon-exon junction complex (EJC), and is directly involved in mRNA surveillance. Furthermore, RNPS1 is a splicing regulator whose activator function is controlled in part by CK2 (casein kinase II) protein kinase phosphorylation. It can also function as a squamous-cell carcinoma antigen recognized by T cells-3 (SART3)-binding protein, and is involved in the regulation of mRNA splicing. RNPS1 contains an N-terminal serine-rich (S) domain, a central RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and the C-terminal arginine/serine/proline-rich (RS/P) domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in nucleolin-like proteins mainly from plants. This subfamily corresponds to the RRM2 of a group of plant nucleolin-like proteins, including nucleolin 1 (also termed protein nucleolin like 1) and nucleolin 2 (also termed protein nucleolin like 2, or protein parallel like 1). They play roles in the regulation of ribosome synthesis and in the growth and development of plants. Like yeast nucleolin, nucleolin-like proteins possess two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains).  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A0 (hnRNP A0) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A0, a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. It has been identified as the substrate for MAPKAP-K2 and may be involved in the lipopolysaccharide (LPS)-induced post-transcriptional regulation of tumor necrosis factor-alpha (TNF-alpha), cyclooxygenase 2 (COX-2) and macrophage inflammatory protein 2 (MIP-2). hnRNP A0 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in yeast nucleolar protein 4 (Nop4p) and similar proteins. This subgroup corresponds to the RRM1 of Nop4p (also known as Nop77p), encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>RNA recognition motif (RRM) found in RNA-binding protein RBM8A, RBM8B nd similar proteins. This subfamily corresponds to the RRM of RBM8, also termed binder of OVCA1-1 (BOV-1), or RNA-binding protein Y14, which is one of the components of the exon-exon junction complex (EJC). It has two isoforms, RBM8A and RBM8B, both of which are identical except that RBM8B is 16 amino acids shorter at its N-terminus. RBM8, together with other EJC components (such as Magoh, Aly/REF, RNPS1, Srm160, and Upf3), plays critical roles in postsplicing processing, including nuclear export and cytoplasmic localization of the mRNA, and the nonsense-mediated mRNA decay (NMD) surveillance process. RBM8 binds to mRNA 20-24 nucleotides upstream of a spliced exon-exon junction. It is also involved in spliced mRNA nuclear export, and the process of nonsense-mediated decay of mRNAs with premature stop codons. RBM8 forms a specific heterodimer complex with the EJC protein Magoh which then associates with Aly/REF, RNPS1, DEK, and SRm160 on the spliced mRNA, and inhibits ATP turnover by eIF4AIII, thereby trapping the EJC core onto RNA. RBM8 contains an N-terminal putative bipartite nuclear localization signal, one RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), in the central region, and a C-terminal serine-arginine rich region (SR domain) and glycine-arginine rich region (RG domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in RNA-binding protein 34 (RBM34) and similar proteins. This subfamily corresponds to the RRM2 of RBM34, a putative RNA-binding protein containing two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). Although the function of RBM34 remains unclear currently, its RRM domains may participate in mRNA processing. RBM34 may act as an mRNA processing-related protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>RNA recognition motif (RRM) found in heterogeneous nuclear ribonucleoprotein G (hnRNP G), Y chromosome RNA recognition motif 1 (hRBMY), testis-specific heterogeneous nuclear ribonucleoprotein G-T (hnRNP G-T) and similar proteins. This subfamily corresponds to the RRM domain of hnRNP G, also termed glycoprotein p43 or RBMX, an RNA-binding motif protein located on the X chromosome. It is expressed ubiquitously and has been implicated in the splicing control of several pre-mRNAs. Moreover, hnRNP G may function as a regulator of transcription for SREBP-1c and GnRH1. Research has shown that hnRNP G may also act as a tumor-suppressor since it upregulates the Txnip gene and promotes the fidelity of DNA end-joining activity. In addition, hnRNP G appears to play a critical role in proper neural development of zebrafish and frog embryos. The family also includes several paralogs of hnRNP G, such as hRBMY and hnRNP G-T (also termed RNA-binding motif protein, X-linked-like-2). Both, hRBMY and hnRNP G-T, are exclusively expressed in testis and critical for male fertility. Like hnRNP G, hRBMY and hnRNP G-T interact with factors implicated in the regulation of pre-mRNA splicing, such as hTra2-beta1 and T-STAR. Although members in this family share a high conserved N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), they appear to recognize different RNA targets. For instance, hRBMY interacts specifically with a stem-loop structure in which the loop is formed by the sequence CA/UCAA. In contrast, hnRNP G associates with single stranded RNA sequences containing a CCA/C motif. In addition to the RRM, hnRNP G contains a nascent transcripts targeting domain (NTD) in the middle region and a novel auxiliary RNA-binding domain (RBD) in its C-terminal region. The C-terminal RBD exhibits distinct RNA binding specificity, and would play a critical role in the regulation of alternative splicing by hnRNP G. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in splicing factor 3B subunit 4 (SF3B4) and similar proteins. This subfamily corresponds to the RRM1 of SF3B4, also termed pre-mRNA-splicing factor SF3b 49 kDa (SF3b50), or spliceosome-associated protein 49 (SAP 49). SF3B4 a component of the multiprotein complex splicing factor 3b (SF3B), an integral part of the U2 small nuclear ribonucleoprotein (snRNP) and the U11/U12 di-snRNP. SF3B is essential for the accurate excision of introns from pre-messenger RNA, and is involved in the recognition of the pre-mRNA's branch site within the major and minor spliceosomes. SF3B4 functions to tether U2 snRNP with pre-mRNA at the branch site during spliceosome assembly. It is an evolutionarily highly conserved protein with orthologs across diverse species. SF3B4 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). It binds directly to pre-mRNA and also interacts directly and highly specifically with another SF3B subunit called SAP 145. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in yeast nuclear and cytoplasmic polyadenylated RNA-binding protein PUB1 and similar proteins. This subgroup corresponds to the RRM2 of yeast protein PUB1, also termed ARS consensus-binding protein ACBP-60, or poly uridylate-binding protein, or poly(U)-binding protein. PUB1 has been identified as both, a heterogeneous nuclear RNA-binding protein (hnRNP) and a cytoplasmic mRNA-binding protein (mRNP), which may be stably bound to a translationally inactive subpopulation of mRNAs within the cytoplasm. It is distributed in both, the nucleus and the cytoplasm, and binds to poly(A)+ RNA (mRNA or pre-mRNA). Although it is one of the major cellular proteins cross-linked by UV light to polyadenylated RNAs in vivo, PUB1 is nonessential for cell growth in yeast. PUB1 also binds to T-rich single stranded DNA (ssDNA). However, there is no strong evidence implicating PUB1 in the mechanism of DNA replication. PUB1 contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a GAR motif (glycine and arginine rich stretch) that is located between RRM2 and RRM3. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM2 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. This subfamily corresponds to the RRM2 of hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. hnRNP D0, a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP A/B is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. It has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus plays an important role in apoB mRNA editing. hnRNP DL (or hnRNP D-like) is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. All memembers in this family contain two putative RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glycine- and tyrosine-rich C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>RNA recognition motif 3 in fungal pre-messenger RNA splicing protein 24 (Prp24) and similar proteins. This subfamily corresponds to the RRM3 of Prp24, also termed U4/U6 snRNA-associated-splicing factor PRP24 (U4/U6 snRNP), an RNA-binding protein with four well conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). It facilitates U6 RNA base-pairing with U4 RNA during spliceosome assembly. Prp24 specifically binds free U6 RNA primarily with RRMs 1 and 2 and facilitates pairing of U6 RNA bases with U4 RNA bases. Additionally, it may also be involved in dissociation of the U4/U6 complex during spliceosome activation. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>RNA recognition motif (RRM) found in eukaryotic translation initiation factor 4B (eIF-4B) and similar proteins. This subfamily corresponds to the RRM of eIF-4B, a multi-domain RNA-binding protein that has been primarily implicated in promoting the binding of 40S ribosomal subunits to mRNA during translation initiation. It contains two RNA-binding domains; the N-terminal well-conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), binds the 18S rRNA of the 40S ribosomal subunit and the C-terminal basic domain (BD), including two arginine-rich motifs (ARMs), binds mRNA during initiation, and is primarily responsible for the stimulation of the helicase activity of eIF-4A. eIF-4B also contains a DRYG domain (a region rich in Asp, Arg, Tyr, and Gly amino acids) in the middle, which is responsible for both, self-association of eIF-4B and  binding to the p170 subunit of eIF3. Additional research indicates that eIF-4B can interact with the poly(A) binding protein (PABP) in mammalian cells, which can stimulate both, the eIF-4B-mediated activation of the helicase activity of eIF-4A and binding of poly(A) by PABP. eIF-4B has also been shown to interact specifically with the internal ribosome entry sites (IRES) of several picornaviruses which facilitate cap-independent translation initiation. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in yeast protein gar2 and similar proteins. This subfamily corresponds to the RRM1 of yeast protein gar2, a novel nucleolar protein required for 18S rRNA and 40S ribosomal subunit accumulation. It shares similar domain architecture with nucleolin from vertebrates and NSR1 from Saccharomyces cerevisiae. The highly phosphorylated N-terminal domain of gar2 is made up of highly acidic regions separated from each other by basic sequences, and contains multiple phosphorylation sites. The central domain of gar2 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The C-terminal RGG (or GAR) domain of gar2 is rich in glycine, arginine and phenylalanine residues. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>RNA recognition motif (RRM) found in U1 small nuclear ribonucleoprotein 70 kDa (U1-70K) and similar proteins. This subfamily corresponds to the RRM of U1-70K, also termed snRNP70, a key component of the U1 snRNP complex, which is one of the key factors facilitating the splicing of pre-mRNA via interaction at the 5' splice site, and is involved in regulation of polyadenylation of some viral and cellular genes, enhancing or inhibiting efficient poly(A) site usage. U1-70K plays an essential role in targeting the U1 snRNP to the 5' splice site through protein-protein interactions with regulatory RNA-binding splicing factors, such as the RS protein ASF/SF2. Moreover, U1-70K protein can specifically bind to stem-loop I of the U1 small nuclear RNA (U1 snRNA) contained in the U1 snRNP complex. It also mediates the binding of U1C, another U1-specific protein, to the U1 snRNP complex. U1-70K contains a conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), followed by an adjacent glycine-rich region at the N-terminal half, and two serine/arginine-rich (SR) domains at the C-terminal half. The RRM is responsible for the binding of stem-loop I of U1 snRNA molecule. Additionally, the most prominent immunodominant region that can be recognized by auto-antibodies from autoimmune patients may be located within the RRM. The SR domains are involved in protein-protein interaction with SR proteins that mediate 5' splice site recognition. For instance, the first SR domain is necessary and sufficient for ASF/SF2 Binding. The family also includes Drosophila U1-70K that is an essential splicing factor required for viability in flies, but its SR domain is dispensable. The yeast U1-70k doesn't contain easily recognizable SR domains and shows low sequence similarity in the RRM region with other U1-70k proteins and therefore not included in this family. The RRM domain is dispensable for yeast U1-70K function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in squamous cell carcinoma antigen recognized by T-cells 3 (SART3) and similar proteins. This subfamily corresponds to the RRM2 of SART3, also termed Tat-interacting protein of 110 kDa (Tip110), is an RNA-binding protein expressed in the nucleus of the majority of proliferating cells, including normal cells and malignant cells, but not in normal tissues except for the testes and fetal liver. It is involved in the regulation of mRNA splicing probably via its complex formation with RNA-binding protein with a serine-rich domain (RNPS1), a pre-mRNA-splicing factor. SART3 has also been identified as a nuclear Tat-interacting protein that regulates Tat transactivation activity through direct interaction and functions as an important cellular factor for HIV-1 gene expression and viral replication. In addition, SART3 is required for U6 snRNP targeting to Cajal bodies. It binds specifically and directly to the U6 snRNA, interacts transiently with the U6 and U4/U6 snRNPs, and promotes the reassembly of U4/U6 snRNPs after splicing in vitro. SART3 contains an N-terminal half-a-tetratricopeptide repeat (HAT)-rich domain, a nuclearlocalization signal (NLS) domain, and two C-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif (RRM) found in Zinc finger CCHC-type and RNA-binding motif-containing protein 1 (ZCRB1) and similar proteins. This subfamily corresponds to the RRM of ZCRB1, also termed MADP-1, or U11/U12 small nuclear ribonucleoprotein 31 kDa protein (U11/U12 snRNP 31 or U11/U12-31K), a novel multi-functional nuclear factor, which may be involved in morphine dependence, cold/heat stress, and hepatocarcinoma. It is located in the nucleoplasm, but outside the nucleolus. ZCRB1 is one of the components of U11/U12 snRNPs that bind to U12-type pre-mRNAs and form a di-snRNP complex, simultaneously recognizing the 5' splice site and branchpoint sequence. ZCRB1 is characterized by an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a CCHC-type Zinc finger motif. In addition, it contains core nucleocapsid motifs, and Lys- and Glu-rich domains.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 3 (RRM3) found found in type I polyadenylate-binding proteins. This subfamily corresponds to the RRM3 of type I poly(A)-binding proteins (PABPs), highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in the regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. The family represents type I polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 1 (PABP-1 or PABPC1), polyadenylate-binding protein 3 (PABP-3 or PABPC3), polyadenylate-binding protein 4 (PABP-4 or APP-1 or iPABP), polyadenylate-binding protein 5 (PABP-5 or PABPC5), polyadenylate-binding protein 1-like (PABP-1-like or PABPC1L), polyadenylate-binding protein 1-like 2 (PABPC1L2 or RBM32), polyadenylate-binding protein 4-like (PABP-4-like or PABPC4L), yeast polyadenylate-binding protein, cytoplasmic and nuclear (PABP or ACBP-67), and similar proteins. PABP-1 is an ubiquitously expressed multifunctional protein that may play a role in 3' end formation of mRNA, translation initiation, mRNA stabilization, protection of poly(A) from nuclease activity, mRNA deadenylation, inhibition of mRNA decapping, and mRNP maturation. Although PABP-1 is thought to be a cytoplasmic protein, it is also found in the nucleus. PABP-1 may be involved in nucleocytoplasmic trafficking and utilization of mRNP particles. PABP-1 contains four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), a less well conserved linker region, and a proline-rich C-terminal conserved domain (CTD). PABP-3 is a testis-specific poly(A)-binding protein specifically expressed in round spermatids. It is mainly found in mammalian and may play an important role in the testis-specific regulation of mRNA homeostasis. PABP-3 shows significant sequence similarity to PABP-1. However, it binds to poly(A) with a lower affinity than PABP-1. PABP-1 possesses an A-rich sequence in its 5'-UTR and allows binding of PABP and blockage of translation of its own mRNA. In contrast, PABP-3 lacks the A-rich sequence in its 5'-UTR. PABP-4 is an inducible poly(A)-binding protein (iPABP) that is primarily localized to the cytoplasm. It shows significant sequence similarity to PABP-1 as well. The RNA binding properties of PABP-1 and PABP-4 appear to be identical. PABP-5 is encoded by PABPC5 gene within the X-specific subinterval, and expressed in fetal brain and in a range of adult tissues in mammalian, such as ovary and testis. It may play an important role in germ cell development. Moreover, unlike other PABPs, PABP-5 contains only four RRMs, but lacks both the linker region and the CTD. PABP-1-like and PABP-1-like 2 are the orthologs of PABP-1. PABP-4-like is the ortholog of PABP-5. Their cellular functions remain unclear. The family also includes the yeast PABP, a conserved poly(A) binding protein containing poly(A) tails that can be attached to the 3'-ends of mRNAs. The yeast PABP and its homologs may play important roles in the initiation of translation and in mRNA decay. Like vertebrate PABP-1, the yeast PABP contains four RRMs, a linker region, and a proline-rich CTD as well. The first two RRMs are mainly responsible for specific binding to poly(A). The proline-rich region may be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>RNA recognition motif (RRM) found in eukaryotic RNA-binding protein RBM24, RBM38 and similar proteins. This subfamily corresponds to the RRM of RBM24 and RBM38 from vertebrate, SUPpressor family member SUP-12 from Caenorhabditis elegans and similar proteins. Both, RBM24 and RBM38, are preferentially expressed in cardiac and skeletal muscle tissues. They regulate myogenic differentiation by controlling the cell cycle in a p21-dependent or -independent manner. RBM24, also termed RNA-binding region-containing protein 6, interacts with the 3'-untranslated region (UTR) of myogenin mRNA and regulates its stability in C2C12 cells. RBM38, also termed CLL-associated antigen KW-5, or HSRNASEB, or RNA-binding region-containing protein 1(RNPC1), or ssDNA-binding protein SEB4, is a direct target of the p53 family. It is required for maintaining the stability of the basal and stress-induced p21 mRNA by binding to their 3'-UTRs. It also binds the AU-/U-rich elements in p63 3'-UTR and regulates p63 mRNA stability and activity. SUP-12 is a novel tissue-specific splicing factor that controls muscle-specific splicing of the ADF/cofilin pre-mRNA in C. elegans. All family members contain a conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>splicing factor, CC1-like family. This model represents a subfamily of RNA splicing factors including the Pad-1 protein (N. crassa), CAPER (M. musculus) and CC1.3 (H.sapiens). These proteins are characterized by an N-terminal arginine-rich, low complexity domain followed by three (or in the case of 4 H. sapiens paralogs, two) RNA recognition domains (rrm: pfam00706). These splicing factors are closely related to the U2AF splicing factor family (TIGR01642). A homologous gene from Plasmodium falciparum was identified in the course of the analysis of that genome at TIGR and was included in the seed.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein D-like (hnRNP DL) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP DL (or hnRNP D-like), also termed AU-rich element RNA-binding factor, or JKT41-binding protein (protein laAUF1 or JKTBP), is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. hnRNP DL binds single-stranded DNA (ssDNA) or double-stranded DNA (dsDNA) in a non-sequencespecific manner, and interacts with poly(G) and poly(A) tenaciously. It contains two putative two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glycine- and tyrosine-rich C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>RNA recognition motif 3 (RRM3) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM3 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cr>38</td>
            <td id=al>RNA recognition motif (RRM) found in Saccharomyces cerevisiae U1 small nuclear ribonucleoprotein SNP1 and similar proteins. SNP1, also called U1 snRNP protein SNP1, or U1 small nuclear ribonucleoprotein 70 kDa homolog, or U1 70K, or U1 snRNP 70 kDa homolog, interacts with mRNA and is involved in nuclear mRNA splicing. It is a component of the spliceosome, where it is associated with snRNP U1 by binding stem loop I of U1 snRNA. Members in this family contain an N-terminal U1snRNP70 domain and an RNA recognition motif (RRM), also called RBD (RNA binding domain) or RNP (ribonucleoprotein domain).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in yeast nuclear polyadenylated RNA-binding protein 4 (Hrp1p or Nab4p) and similar proteins. This subfamily corresponds to the RRM1 of Hrp1p and similar proteins. Hrp1p or Nab4p, also termed cleavage factor IB (CFIB), is a sequence-specific trans-acting factor that is essential for mRNA 3'-end formation in yeast Saccharomyces cerevisiae. It can be UV cross-linked to RNA and specifically recognizes the (UA)6 RNA element required for both, the cleavage and poly(A) addition steps. Moreover, Hrp1p can shuttle between the nucleus and the cytoplasm, and play an additional role in the export of mRNAs to the cytoplasm. Hrp1p also interacts with Rna15p and Rna14p, two components of CF1A. In addition, Hrp1p functions as a factor directly involved in modulating the activity of the nonsense-mediated mRNA decay (NMD) pathway; it binds specifically to a downstream sequence element (DSE)-containing RNA and interacts with Upf1p, a component of the surveillance complex, further triggering the NMD pathway. Hrp1p contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an arginine-glycine-rich region harboring repeats of the sequence RGGF/Y. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in Saccharomyces cerevisiae protein HRB1, G-strand-binding protein 2 (GBP2) and similar proteins. The family includes Saccharomyces cerevisiae protein HRB1 (also called protein TOM34) and GBP2, both of which are SR-like mRNA-binding proteins which shuttle from the nucleus to the cytoplasm when bound to the mature mRNA molecules. They act as quality control factors for spliced mRNAs. GBP2, also called RAP1 localization factor 6, is a single-strand telomeric DNA-binding protein that binds single-stranded telomeric sequences of the type (TG[1-3])n in vitro. It also binds to RNA. GBP2 influences the localization of RAP1 in the nuclei and plays a role in modulating telomere length. Members in this family contain three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>RNA recognition motif 3 (RRM3) found in yeast nucleolar protein 4 (Nop4p) and similar proteins. This subgroup corresponds to the RRM3 of Nop4p (also known as Nop77p), encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>RNA recognition motif (RRM) found in nuclear cap-binding protein subunit 2 (CBP20) and similar proteins. This subfamily corresponds to the RRM of CBP20, also termed nuclear cap-binding protein subunit 2 (NCBP2), or cell proliferation-inducing gene 55 protein, or NCBP-interacting protein 1 (NIP1). CBP20 is the small subunit of the nuclear cap binding complex (CBC), which is a conserved eukaryotic heterodimeric protein complex binding to 5'-capped polymerase II transcripts and plays a central role in the maturation of pre-mRNA and uracil-rich small nuclear RNA (U snRNA). CBP20 is most likely responsible for the binding of capped RNA. It contains an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and interacts with the second and third domains of CBP80, the large subunit of CBC. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in RNA-binding protein Musashi homologs Musashi-1, Musashi-2 and similar proteins. This subfamily corresponds to the RRM2.in Musashi-1 (also termed Msi1), a neural RNA-binding protein putatively expressed in central nervous system (CNS) stem cells and neural progenitor cells, and associated with asymmetric divisions in neural progenitor cells. It is evolutionarily conserved from invertebrates to vertebrates. Musashi-1 is a homolog of Drosophila Musashi and Xenopus laevis nervous system-specific RNP protein-1 (Nrp-1). It has been implicated in the maintenance of the stem-cell state, differentiation, and tumorigenesis. It translationally regulates the expression of a mammalian numb gene by binding to the 3'-untranslated region of mRNA of Numb, encoding a membrane-associated inhibitor of Notch signaling, and further influences neural development. Moreover, Musashi-1 represses translation by interacting with the poly(A)-binding protein and competes for binding of the eukaryotic initiation factor-4G (eIF-4G). Musashi-2 (also termed Msi2) has been identified as a regulator of the hematopoietic stem cell (HSC) compartment and of leukemic stem cells after transplantation of cells with loss and gain of function of the gene. It influences proliferation and differentiation of HSCs and myeloid progenitors, and further modulates normal hematopoiesis and promotes aggressive myeloid leukemia. Both, Musashi-1 and Musashi-2, contain two conserved N-terminal tandem RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), along with other domains of unknown function. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=co>58</td>
            <td id=al>RNA recognition motif (RRM) found in pre-mRNA branch site protein p14 (SF3B14) and similar proteins. This subfamily corresponds to the RRM of SF3B14 (also termed p14), a 14 kDa protein subunit of SF3B which is a multiprotein complex that is an integral part of the U2 small nuclear ribonucleoprotein (snRNP) and the U11/U12 di-snRNP. SF3B is essential for the accurate excision of introns from pre-messenger RNA and has been involved in the recognition of the pre-mRNA's branch site within the major and minor spliceosomes. SF3B14 associates directly with another SF3B subunit called SF3B155. It is also present in both U2- and U12-dependent spliceosomes and may contribute to branch site positioning in both the major and minor spliceosome. Moreover, SF3B14 interacts directly with the pre-mRNA branch adenosine early in spliceosome assembly and within the fully assembled spliceosome. SF3B14 contains one well conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in tRNA selenocysteine-associated protein 1 (SECp43) and similar proteins. This subfamily corresponds to the RRM2 in tRNA selenocysteine-associated protein 1 (SECp43), yeast negative growth regulatory protein NGR1 (RBP1), yeast protein NAM8, and similar proteins. SECp43 is an RNA-binding protein associated specifically with eukaryotic selenocysteine tRNA [tRNA(Sec)]. It may play an adaptor role in the mechanism of selenocysteine insertion. SECp43 is located primarily in the nucleus and contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal polar/acidic region. Yeast proteins, NGR1 and NAM8, show high sequence similarity with SECp43. NGR1 is a putative glucose-repressible protein that binds both RNA and single-stranded DNA (ssDNA). It may function in regulating cell growth in early log phase, possibly through its participation in RNA metabolism. NGR1 contains three RRMs, two of which are followed by a glutamine-rich stretch that may be involved in transcriptional activity. In addition, NGR1 has an asparagine-rich region near the C-terminus which also harbors a methionine-rich region. NAM8 is a putative RNA-binding protein that acts as a suppressor of mitochondrial splicing deficiencies when overexpressed in yeast. It may be a non-essential component of the mitochondrial splicing machinery. NAM8 also contains three RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>RNA recognition motif (RRM) found in serine/arginine-rich splicing factor SRSF10, SRSF12 and similar proteins. This subfamily corresponds to the RRM of SRSF10 and SRSF12. SRSF10, also termed 40 kDa SR-repressor protein (SRrp40), or FUS-interacting serine-arginine-rich protein 1 (FUSIP1), or splicing factor SRp38, or splicing factor, arginine/serine-rich 13A (SFRS13A), or TLS-associated protein with Ser-Arg repeats (TASR). It is a serine-arginine (SR) protein that acts as a potent and general splicing repressor when dephosphorylated. It mediates global inhibition of splicing both in M phase of the cell cycle and in response to heat shock. SRSF10 emerges as a modulator of cholesterol homeostasis through the regulation of low-density lipoprotein receptor (LDLR) splicing efficiency. It also regulates cardiac-specific alternative splicing of triadin pre-mRNA and is required for proper Ca2+ handling during embryonic heart development. In contrast, the phosphorylated SRSF10 functions as a sequence-specific splicing activator in the presence of a nuclear cofactor. It activates distal alternative 5' splice site of adenovirus E1A pre-mRNA in vivo. Moreover, SRSF10 strengthens pre-mRNA recognition by U1 and U2 snRNPs. SRSF10 localizes to the nuclear speckles and can shuttle between nucleus and cytoplasm. SRSF12, also termed 35 kDa SR repressor protein (SRrp35), or splicing factor, arginine/serine-rich 13B (SFRS13B), or splicing factor, arginine/serine-rich 19 (SFRS19), is a serine/arginine (SR) protein-like alternative splicing regulator that antagonizes authentic SR proteins in the modulation of alternative 5' splice site choice. For instance, it activates distal alternative 5' splice site of the adenovirus E1A pre-mRNA in vivo. Both, SRSF10 and SRSF12, contain a single N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), followed by a C-terminal RS domain rich in serine-arginine dipeptides. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>polyadenylate binding protein, human types 1, 2, 3, 4 family. These eukaryotic proteins recognize the poly-A of mRNA and consists of four tandem RNA recognition domains at the N-terminus (rrm: pfam00076) followed by a PABP-specific domain (pfam00658) at the C-terminus. The protein is involved in the transport of mRNA's from the nucleus to the cytoplasm. There are four paralogs in Homo sapiens which are expressed in testis, platelets, broadly expressed and of unknown tissue range.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cr>8</td>
            <td id=al>polyadenylate binding protein, human types 1, 2, 3, 4 family. These eukaryotic proteins recognize the poly-A of mRNA and consists of four tandem RNA recognition domains at the N-terminus (rrm: pfam00076) followed by a PABP-specific domain (pfam00658) at the C-terminus. The protein is involved in the transport of mRNA's from the nucleus to the cytoplasm. There are four paralogs in Homo sapiens which are expressed in testis, platelets, broadly expressed and of unknown tissue range.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in granule-associated RNA binding proteins p40-TIA-1 and TIAR. This subfamily corresponds to the RRM2 of nucleolysin TIA-1 isoform p40 (p40-TIA-1 or TIA-1) and nucleolysin TIA-1-related protein (TIAR), both of which are granule-associated RNA binding proteins involved in inducing apoptosis in cytotoxic lymphocyte (CTL) target cells. TIA-1 and TIAR share high sequence similarity. They are expressed in a wide variety of cell types. TIA-1 can be phosphorylated by a serine/threonine kinase that is activated during Fas-mediated apoptosis. TIAR is mainly localized in the nucleus of hematopoietic and nonhematopoietic cells. It is translocated from the nucleus to the cytoplasm in response to exogenous triggers of apoptosis. Both, TIA-1 and TIAR, bind specifically to poly(A) but not to poly(C) homopolymers. They are composed of three N-terminal highly homologous RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glutamine-rich C-terminal auxiliary domain containing a lysosome-targeting motif. TIA-1 and TIAR interact with RNAs containing short stretches of uridylates and their RRM2 can mediate the specific binding to uridylate-rich RNAs. The C-terminal auxiliary domain may be responsible for interacting with other proteins. In addition, TIA-1 and TIAR share a potential serine protease-cleavage site (Phe-Val-Arg) localized at the junction between their RNA binding domains and their C-terminal auxiliary domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in yeast nuclear polyadenylated RNA-binding protein 4 (Hrp1p or Nab4p) and similar proteins. This subfamily corresponds to the RRM1 of Hrp1p and similar proteins. Hrp1p or Nab4p, also termed cleavage factor IB (CFIB), is a sequence-specific trans-acting factor that is essential for mRNA 3'-end formation in yeast Saccharomyces cerevisiae. It can be UV cross-linked to RNA and specifically recognizes the (UA)6 RNA element required for both, the cleavage and poly(A) addition, steps. Moreover, Hrp1p can shuttle between the nucleus and the cytoplasm, and play an additional role in the export of mRNAs to the cytoplasm. Hrp1p also interacts with Rna15p and Rna14p, two components of CF1A. In addition, Hrp1p functions as a factor directly involved in modulating the activity of the nonsense-mediated mRNA decay (NMD) pathway. It binds specifically to a downstream sequence element (DSE)-containing RNA and interacts with Upf1p, a component of the surveillance complex, further triggering the NMD pathway. Hrp1p contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an arginine-glycine-rich region harboring repeats of the sequence RGGF/Y. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif in type II polyadenylate-binding protein 2 (PABP-2) and similar proteins. This subgroup corresponds to the RRM of PABP-2, also termed poly(A)-binding protein 2, or nuclear poly(A)-binding protein 1 (PABPN1), or poly(A)-binding protein II (PABII), which is a ubiquitously expressed type II nuclear poly(A)-binding protein that directs the elongation of mRNA poly(A) tails during pre-mRNA processing. Although PABP-2 binds poly(A) with high affinity and specificity as type I poly(A)-binding proteins, it contains only one highly conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), which is responsible for the poly(A) binding. In addition, PABP-2 possesses an acidic N-terminal domain that is essential for the stimulation of PAP, and an arginine-rich C-terminal domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>RNA recognition motif (RRM) found in RNA-binding protein 7 (RBM7) and similar proteins. This subfamily corresponds to the RRM of RBM7, RBM11 and their eukaryotic homologous. RBM7 is an ubiquitously expressed pre-mRNA splicing factor that enhances messenger RNA (mRNA) splicing in a cell-specific manner or in a certain developmental process, such as spermatogenesis. It interacts with splicing factors SAP145 (the spliceosomal splicing factor 3b subunit 2) and SRp20, and may play a more specific role in meiosis entry and progression. Together with additional testis-specific RNA-binding proteins, RBM7 may regulate the splicing of specific pre-mRNA species that are important in the meiotic cell cycle. RBM11 is a novel tissue-specific splicing regulator that is selectively expressed in brain, cerebellum and testis, and to a lower extent in kidney. It is localized in the nucleoplasm and enriched in SRSF2-containing splicing speckles. It may play a role in the modulation of alternative splicing during neuron and germ cell differentiation. Both, RBM7 and RBM11, contain an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a region lacking known homology at the C-terminus. The RRM is responsible for RNA binding, whereas the C-terminal region permits nuclear localization and homodimerization. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in La-related protein 7 (LARP7) and similar proteins. This subfamily corresponds to the RRM1 of LARP7, also termed La ribonucleoprotein domain family member 7, or P-TEFb-interaction protein for 7SK stability (PIP7S), an oligopyrimidine-binding protein that binds to the highly conserved 3'-terminal U-rich stretch (3' -UUU-OH) of 7SK RNA. LARP7 is a stable component of the 7SK small nuclear ribonucleoprotein (7SK snRNP). It intimately associates with all the nuclear 7SK and is required for 7SK stability. LARP7 also acts as a negative transcriptional regulator of cellular and viral polymerase II genes, acting by means of the 7SK snRNP system. It plays an essential role in the inhibition of positive transcription elongation factor b (P-TEFb)-dependent transcription, which has been linked to the global control of cell growth and tumorigenesis. LARP7 contains a La motif (LAM) and an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), at the N-terminal region, which mediates binding to the U-rich 3' terminus of 7SK RNA. LARP7 also carries another putative RRM domain at its C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A2/B1 (hnRNP A2/B1) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A2/B1, an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). Many mRNAs, such as myelin basic protein (MBP), myelin-associated oligodendrocytic basic protein (MOBP), carboxyanhydrase II (CAII), microtubule-associated protein tau, and amyloid precursor protein (APP) are trafficked by hnRNP A2/B1. hnRNP A2/B1 also functions as a splicing factor that regulates alternative splicing of the tumor suppressors, such as BIN1, WWOX, the antiapoptotic proteins c-FLIP and caspase-9B, the insulin receptor (IR), and the RON proto-oncogene among others. Overexpression of hnRNP A2/B1 has been described in many cancers. It functions as a nuclear matrix protein involving in RNA synthesis and the regulation of cellular migration through alternatively splicing pre-mRNA. It may play a role in tumor cell differentiation. hnRNP A2/B1 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 6 (RRM6) found in RNA-binding protein 19 (RBM19 or RBD-1) and RNA recognition motif 5 (RRM5) found in multiple RNA-binding domain-containing protein 1 (MRD1). This subfamily corresponds to the RRM6 of RBM19 and RRM5 of MRD1. RBM19, also termed RNA-binding domain-1 (RBD-1), is a nucleolar protein conserved in eukaryotes. It is involved in ribosome biogenesis by processing rRNA and is essential for preimplantation development. It has a unique domain organization containing 6 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). MRD1 is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well-conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). It is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. MRD1 contains 5 conserved RRMs, which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>RNA recognition motif 4 (RRM4) found in type I polyadenylate-binding proteins. This subfamily corresponds to the RRM4 of type I poly(A)-binding proteins (PABPs), highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in theThe CD corresponds to the RRM. regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. The family represents type I polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 1 (PABP-1 or PABPC1), polyadenylate-binding protein 3 (PABP-3 or PABPC3), polyadenylate-binding protein 4 (PABP-4 or APP-1 or iPABP), polyadenylate-binding protein 5 (PABP-5 or PABPC5), polyadenylate-binding protein 1-like (PABP-1-like or PABPC1L), polyadenylate-binding protein 1-like 2 (PABPC1L2 or RBM32), polyadenylate-binding protein 4-like (PABP-4-like or PABPC4L), yeast polyadenylate-binding protein, cytoplasmic and nuclear (PABP or ACBP-67), and similar proteins. PABP-1 is an ubiquitously expressed multifunctional protein that may play a role in 3' end formation of mRNA, translation initiation, mRNA stabilization, protection of poly(A) from nuclease activity, mRNA deadenylation, inhibition of mRNA decapping, and mRNP maturation. Although PABP-1 is thought to be a cytoplasmic protein, it is also found in the nucleus. PABP-1 may be involved in nucleocytoplasmic trafficking and utilization of mRNP particles. PABP-1 contains four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), a less well conserved linker region, and a proline-rich C-terminal conserved domain (CTD). PABP-3 is a testis-specific poly(A)-binding protein specifically expressed in round spermatids. It is mainly found in mammalian and may play an important role in the testis-specific regulation of mRNA homeostasis. PABP-3 shows significant sequence similarity to PABP-1. However, it binds to poly(A) with a lower affinity than PABP-1. Moreover, PABP-1 possesses an A-rich sequence in its 5'-UTR and allows binding of PABP and blockage of translation of its own mRNA. In contrast, PABP-3 lacks the A-rich sequence in its 5'-UTR. PABP-4 is an inducible poly(A)-binding protein (iPABP) that is primarily localized to the cytoplasm. It shows significant sequence similarity to PABP-1 as well. The RNA binding properties of PABP-1 and PABP-4 appear to be identical. PABP-5 is encoded by PABPC5 gene within the X-specific subinterval, and expressed in fetal brain and in a range of adult tissues in mammalian, such as ovary and testis. It may play an important role in germ cell development. Moreover, unlike other PABPs, PABP-5 contains only four RRMs, but lacks both the linker region and the CTD. PABP-1-like and PABP-1-like 2 are the orthologs of PABP-1. PABP-4-like is the ortholog of PABP-5. Their cellular functions remain unclear. The family also includes the yeast PABP, a conserved poly(A) binding protein containing poly(A) tails that can be attached to the 3'-ends of mRNAs. The yeast PABP and its homologs may play important roles in the initiation of translation and in mRNA decay. Like vertebrate PABP-1, the yeast PABP contains four RRMs, a linker region, and a proline-rich CTD as well. The first two RRMs are mainly responsible for specific binding to poly(A). The proline-rich region may be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>RNA recognition motif (RRM) found in eukaryotic translation initiation factor 4H (eIF-4H) and similar proteins. This subfamily corresponds to the RRM of eIF-4H, also termed Williams-Beuren syndrome chromosomal region 1 protein, which, together with elf-4B/eIF-4G, serves as the accessory protein of RNA helicase eIF-4A. eIF-4H contains a well conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). It stimulates protein synthesis by enhancing the helicase activity of eIF-4A in the initiation step of mRNA translation. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in (U)-binding-splicing factor PUF60 and similar proteins. This subfamily corresponds to the RRM1 of PUF60, also termed FUSE-binding protein-interacting repressor (FBP-interacting repressor or FIR), or Ro-binding protein 1 (RoBP1), or Siah-binding protein 1 (Siah-BP1). PUF60 is an essential splicing factor that functions as a poly-U RNA-binding protein required to reconstitute splicing in depleted nuclear extracts. Its function is enhanced through interaction with U2 auxiliary factor U2AF65. PUF60 also controls human c-myc gene expression by binding and inhibiting the transcription factor far upstream sequence element (FUSE)-binding-protein (FBP), an activator of c-myc promoters. PUF60 contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal U2AF (U2 auxiliary factor) homology motifs (UHM) that harbors another RRM and binds to tryptophan-containing linear peptide motifs (UHM ligand motifs, ULMs) in several nuclear proteins. Research indicates that PUF60 binds FUSE as a dimer, and only the first two RRM domains participate in the single-stranded DNA recognition. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cb>62</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. This subfamily corresponds to the RRM1 in hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. hnRNP D0 is a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP A/B is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. hnRNP A/B has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus plays an important role in apoB mRNA editing. hnRNP DL (or hnRNP D-like) is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. All members in this family contain two putative RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glycine- and tyrosine-rich C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in Arabidopsis thaliana phragmoplastin interacting protein 1 (PHIP1) and similar proteins. This subfamily corresponds to the RRM1 of PHIP1. A. thaliana PHIP1 and its homologs represent a novel class of plant-specific RNA-binding proteins that may play a unique role in the polarized mRNA transport to the vicinity of the cell plate. The family members consist of multiple functional domains, including a lysine-rich domain (KRD domain) that contains three nuclear localization motifs (KKKR/NK), two RNA recognition motifs (RRMs), and three CCHC-type zinc fingers. PHIP1 is a peripheral membrane protein and is localized at the cell plate during cytokinesis in plants. In addition to phragmoplastin, PHIP1 interacts with two Arabidopsis small GTP-binding proteins, Rop1 and Ran2. However, PHIP1 interacted only with the GTP-bound form of Rop1 but not the GDP-bound form. It also binds specifically to Ran2 mRNA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=co>56</td>
            <td id=al>RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein A/B (hnRNP A/B) and similar proteins. This subgroup corresponds to the RRM1 of hnRNP A/B, also termed APOBEC1-binding protein 1 (ABBP-1), which is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. hnRNP A/B has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus plays an important role in apoB mRNA editing. hnRNP A/B contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long C-terminal glycine-rich domain that contains a potential ATP/GTP binding loop. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>RNA recognition motif 2 (RRM2) found in nucleolysin TIA-1 isoform p40 (p40-TIA-1) and similar proteins. This subgroup corresponds to the RRM2 of p40-TIA-1, the 40-kDa isoform of T-cell-restricted intracellular antigen-1 (TIA-1), and a cytotoxic granule-associated RNA-binding protein mainly found in the granules of cytotoxic lymphocytes. TIA-1 can be phosphorylated by a serine/threonine kinase that is activated during Fas-mediated apoptosis, and function as the granule component responsible for inducing apoptosis in cytolytic lymphocyte (CTL) targets. It is composed of three N-terminal highly homologous RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glutamine-rich C-terminal auxiliary domain containing a lysosome-targeting motif. TIA-1 interacts with RNAs containing short stretches of uridylates and its RRM2 can mediate the specific binding to uridylate-rich RNAs. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2692483</td></tr>
<tr><td id=al>end</td><td id=al>2692753</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>RNA-binding protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
