<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03817</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03817 DNA repair protein RadA [Phormidium_A]</h3>
    <div>MAKPRTIYVCQECGAEFPQYFGRCTNCGAWNSLVETEPIEALTTTRPASAVAWNRNLGVSNGEKPEGQPRASVRLSEIADASVARMPSGYAELDRVLGGGIVPGSLVLIGGEPGIGKSTLLLQTASGLAQESRVLYVSAEESGQQIKLRAQRLLDGHPHVETSEAAEGEYVEEHQRESRLRSPQASRSQSGGRVSESNSPQEGEAGLGEGSLAVEPNFFLLPETDLEAILRELESLKPDFAVIDSIQTIYFSSLTSAPGSVAQVRECTSALMQVAKRENITLFVVGHVTKDGSIAGPRVLEHLVDTVLFFEGDRFASHRILRSMKNRFGATHEIGVFEMIANGLREVSNPSELFMGSRDETVSGTCTVVACEGTRPILVELQALVSPASFGSPRRSATGIDSNRLQQILAVLEKRVGVPLSKLDTYLASVGGIRVEEPAADLGVAIAVVSGFRDRLVDPHTVLIGEVGLGGQVRQVSQLELRLREAAKLGFKRAIIPKGQNPPELGLTLIPVARVLDAILEAIPSASTASPMALEAGETPSGLSLPSLDAVDDDEDWDDDLGSAGGPETDLF</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DNA repair and recombination protein Sms</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>91</td>
            <td id=cg>97</td>
            <td id=al>TPA: DNA repair protein RadA</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>92</td>
            <td id=cg>98</td>
            <td id=al>DNA repair protein RadA</td>
            <td id=co>FACHB-68</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>51</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>PRK11823 DNA repair protein RadA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=al>COG1066 Predicted ATP-dependent serine protease [Posttranslational modification, protein turnover, chaperones]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=al>TIGR00416 DNA repair protein RadA. The gene protuct codes for a probable ATP-dependent protease involved in both DNA repair and degradation of proteins, peptides, glycopeptides. Also known as sms. Residues 11-28 of the SEED alignment contain a putative Zn binding domain. Residues 110-117 of the seed contain a putative ATP binding site both documented in Haemophilus (SP:P45266) and in Listeria monocytogenes (SP:Q48761) . for E.coli see ( J. BACTERIOL. 178:5045-5048(1996)). [DNA metabolism, DNA replication, recombination, and repair]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=al>cd01121 bacterial RadA DNA repair protein. Sms or bacterial RadA is a DNA repair protein that plays a role in recombination and recombinational repair of DNA damaged by UV radiation, X-rays, and chemical agent and is responsible for the stabilization or processing of branched DNA molecules.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=al>cd01124 Circadian Clock Protein KaiC. KaiC is a circadian clock protein, most studied in cyanobacteria.  KaiC, an autokinase, autophosphatase, and ATPase, is part of the core oscillator, composed of three proteins: KaiA, KaiB, and KaiC. The circadian oscillation is regulated via KaiC phosphorylation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>COG0467 RecA-superfamily ATPase, KaiC/GvpD/RAD55 family [Signal transduction mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>37</td>
            <td id=co>42</td>
            <td id=al>PRK09302 circadian clock protein KaiC; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=al>pfam13481 AAA domain. This AAA domain is found in a wide variety of presumed DNA repair proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=al>pfam06745 KaiC. This family is in the P-loop NTPase superfamily and is found in archaea, bacteria and eukaryotes. More than one copy is sometimes found in each protein. This family includes KaiC, which is one of the Kai proteins among which direct protein-protein association may be a critical process in the generation of circadian rhythms in cyanobacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cr>4</td>
            <td id=cg>100</td>
            <td id=al>pfam18073 Rubredoxin metal binding domain. This is the C-terminal rubredoxin metal binding domain found in Interest in lipopolysaccharide (LPS) assembly protein B (LapB). Rubredoxin proteins form small non-heme iron binding sites that use four cysteine residues to coordinate a single metal ion in a tetrahedral environment. Rubredoxins are most commonly found in bacterial systems, but have also been found in eukaryotes. The key features of these rubredoxin-like domains are the extended loops or 'knuckles' and the tetracysteine mode of iron binding. Structural analysis of LapB from Escherichia coli show that the rubredoxin metal binding domain is intimately bound to the TPR motifs and that this association to the TPR motifs is essential to LPS regulation and growth in vivo. Other family members include RadA proteins which play a role in DNA damage repair. In E. coli, a protein known as RadA (or Sms) participates in the recombinational repair of radiation-damaged DNA in a process that uses an undamaged DNA strand in one DNA duplex to fill a DNA strand gap in a homologous sister DNA duplex. RadA carries a zinc finger at the N-terminal domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>30</td>
            <td id=cb>67</td>
            <td id=al>cd00984 C-terminal domain of DnaB helicase. DnaB helicase C-terminal domain. The hexameric helicase DnaB unwinds the DNA duplex at the  chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerization of the N-terminal domain has been observed and may occur during the enzymatic cycle. This C-terminal domain contains an ATP-binding site and is therefore probably the site of ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=al>cd19488 N-terminal domain of KaiC family protein; uncharacterized subfamily. KaiC is a circadian clock protein, most studied in cyanobacteria.  KaiC, an autokinase, autophosphatase, and ATPase, is part of the core oscillator, composed of three proteins: KaiA, KaiB, and KaiC. The circadian oscillation is regulated via KaiC phosphorylation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>34</td>
            <td id=cg>91</td>
            <td id=al>cd19484 C-terminal domain of Circadian Clock Protein KaiC. KaiC is a circadian clock protein, most studied in cyanobacteria.  KaiC, an autokinase, autophosphatase, and ATPase, is part of the core oscillator, composed of three proteins: KaiA, KaiB, and KaiC. The circadian oscillation is regulated via KaiC phosphorylation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cr>30</td>
            <td id=co>40</td>
            <td id=al>COG0305 Replicative DNA helicase [Replication, recombination and repair]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cr>9</td>
            <td id=cr>23</td>
            <td id=al>COG2874 Archaellum biogenesis protein FlaH, an ATPase [Cell motility]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>12</td>
            <td id=cr>27</td>
            <td id=al>pfam03796 DnaB-like helicase C terminal domain. The hexameric helicase DnaB unwinds the DNA duplex at the Escherichia coli chromosome replication fork. Although the mechanism by which DnaB both couples ATP hydrolysis to translocation along DNA and denatures the duplex is unknown, a change in the quaternary structure of the protein involving dimerization of the N-terminal domain has been observed and may occur during the enzymatic cycle. This C-terminal domain contains an ATP-binding site and is therefore probably the site of ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cr>9</td>
            <td id=cr>24</td>
            <td id=al>PRK06067 flagellar accessory protein FlaH; Validated</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3219209</td></tr>
<tr><td id=al>end</td><td id=al>3220928</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DNA repair protein RadA</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>573</td></tr>
</table>

    </div>
</body>
</html>
