<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00778</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00778 LysR family transcriptional regulator [Phormidium_A]</h3>
    <div>MSDLPFTLDQLRILKAIASEGSFKRAADSLYVSQPAVSLQVQNLERQLDVPLFDRGGRRAQLTEAGYLLLSYGEKVLTLCQETCRAIEDLQNLQGGTLIVGASQTTGTYLLPRTIGAFRQRYPDVAVQLHVHSTRRTSWSVANGQVDLAIVGGEVPTDLQDSLEIIPYAEDELALILPVSHPLSKLPSVHKEELYNLQFIALDSQSTIRKVIDRVLARCDIDTRRLTLEMELNSIEAIKNAVQSGLGAAFVSVSAIEKELEMGVLHRATIEEVVVKRTLNVIYNPNRYRSKAAEAFITEILPLFSNQELDPKLLEPITRVKANSKAKDPPAAETPEMPVTSVAESLADSSDSGS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>91</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=cr>PCC-7407</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LysR family transcriptional regulator</td>
            <td id=co>Phormidium</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LysR transcriptional regulator; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>C-terminal substrate binding domain of LysR-type transcriptional regulator CysL, which activates the transcription of the cysJI operon encoding sulfite reductase, contains the type 2 periplasmic binding fold. CysL, also known as YwfK, is a regular of sulfur metabolism in Bacillus subtilis. Sulfur is required for the synthesis of proteins and essential cofactors in all living organism. Sulfur can be assimilated either from inorganic sources (sulfate and thiosulfate), or from organic sources (sulfate esters, sulfamates, and sulfonates). CysL activates the transcription of the cysJI operon encoding sulfite reductase, which reduces sulfite to sulfide. Both cysL mutant and cysJI mutant are unable to grow using sulfate or sulfite as the sulfur source. Like other LysR-type regulators, CysL also negatively regulates its own transcription. In Escherichia coli, three LysR-type activators are involved in the regulation of sulfur metabolism: CysB, Cbl and MetR.  The topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA-binding transcriptional regulator, LysR family [Transcription]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The substrate binding domain of LysR-type transcriptional regulators (LTTRs), a member of the type 2 periplasmic binding fold protein superfamily. This model and hierarchy represent the the substrate-binding domain of the LysR-type transcriptional regulators that form the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, oxidative stress responses, nodule formation of nitrogen-fixing bacteria, synthesis of virulence factors, toxin production, attachment and secretion, to name a few. The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.  Besides transport proteins, the PBP2 superfamily includes the substrate-binding domains from ionotropic glutamate receptors, LysR-like transcriptional regulators, and unorthodox sensor proteins involved in signal transduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LysR substrate binding domain. The structure of this domain is known and is similar to the periplasmic binding proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>putative DNA-binding transcriptional regulator; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>DNA-binding transcriptional regulator CynR; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>transcriptional regulator CysB-like protein; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>The C-terminal substrate binding of LysR-type transcriptional regulator (CbbR) of RubisCO operon, which is involved in the carbon dioxide fixation, contains the type 2 periplasmic binding fold. CbbR, a LysR-type transcriptional regulator, is required to activate expression of RubisCO, one of two unique enzymes in the Calvin-Benson-Bassham (CBB) cycle pathway. All plants, cyanobacteria, and many autotrophic bacteria use the CBB cycle to fix carbon dioxide. Thus, this cycle plays an essential role in assimilating CO2 into organic carbon on earth. The key CBB cycle enzyme is ribulose 1,5-bisphosphate carboxylase/oxygenase (RubisCO), which catalyzes the actual CO2 fixation reaction. The CO2 concentration affects the expression of RubisCO genes.  It has also shown that NADPH enhances the DNA-binding ability of the CbbR. RubisCO is composed of eight large (CbbL) and eight small subunits (CbbS).  The topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>CysB family HTH-type transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TThe C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>transcriptional regulator CysB-like protein; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The substrate binding domain of LysR-type transcriptional regulator GltC, which activates gltA expression of glutamate synthase operon, contains type 2 periplasmic binding fold. GltC, a member of the LysR family of bacterial transcriptional factors, activates the expression of gltA gene of glutamate synthase operon and is essential for cell growth in the absence of glutamate. Glutamate synthase is a heterodimeric protein that encoded by gltA and gltB, whose expression is subject to nutritional regulation. GltC also negatively auto-regulates its own expression. This substrate-binding domain has strong homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>DNA-binding transcriptional regulator OxyR; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate-domain of LysR-type transcriptional regulators involved in the catabolism of opines and that of related regulators, contains the type 2 periplasmic binding fold. This CD includes the C-terminal substrate-domain of LysR-type transcriptional regulators, OccR and NocR, involved in the catabolism of opines and that of LysR for lysine biosynthesis which clustered together in phylogenetic trees. Opines, such as octopine and nopaline, are low molecular weight compounds found in plant crown gall tumors that are produced by the parasitic bacterium Agrobacterium. There are at least 30 different opines identified so far. Opines are utilized by tumor-colonizing bacteria as a source of carbon, nitrogen, and energy. NocR and OccR belong to the family of LysR-type transcriptional regulators that positively regulates the catabolism of nopaline and octopine, respectively. Both nopaline and octopalin are arginine derivatives. In Agrobacterium tumefaciens, NocR regulates expression of the divergently transcribed nocB and nocR genes of the nopaline catabolism (noc) region.  OccR protein activates the occQ operon of the Ti plasmid in response to octopine. This operon encodes proteins required for the uptake and catabolism of octopine. The occ operon also encodes the TraR protein, which is a quorum-sensing transcriptional regulator of the Ti plasmid tra regulon.  LysR is the transcriptional activator of lysA gene encoding diaminopimelate decarboxylase, an enzyme that catalyses the decarboxylation of diaminopimelate to produce lysine. This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>The C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>HTH-type transcriptional regulator Cbl. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Bacterial regulatory helix-turn-helix protein, lysR family. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of LysR-like transcriptional regulator CidR, contains the type 2 periplasmic binding fold. This CD includes the substrate binding domain of CidR which positively up-regulates the expression of cidABC operon in the presence of acetic acid produced by the metabolism of excess glucose. The CidR affects the control of murein hydrolase activity by enhancing cidABC expression in the presence of acetic acid. Thus, up-regulation of cidABC expression results in increased murein hydrolase activity. This substrate binding domain has significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>HTH-type transcriptional regulator CysB. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>DNA-binding transcriptional regulator HcaR; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate-binding domain of the LysR-type transcriptional regulator OxyR, a member of the type 2 periplasmic binding fold protein superfamily. OxyR senses hydrogen peroxide and is activated through the formation of an intramolecular disulfide bond. The OxyR activation induces the transcription of genes necessary for the bacterial defense against oxidative stress. The OxyR of LysR-type transcriptional regulator family is composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The C-terminal domain also contains the redox-active cysteines that mediate the redox-dependent conformational switch. Thus, the interaction between the OxyR-tetramer and DNA is notably different between the oxidized and reduced forms. The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-teminal substrate binding domain of LysR-like nitrogen assimilation control (NAC) protein, contains the type 2 periplasmic binding fold. The NAC is a LysR-type transcription regulator that activates expression of operons such as hut (histidine utilization) and ure (urea utilization), allowing use of non-preferred (poor) nitrogen sources, and represses expression of operons, such as glutamate dehydrogenase (gdh), allowing assimilation of the preferred nitrogen source.  The expression of the nac gene is fully dependent on the nitrogen regulatory system (NTR) and the sigma54-containing RNA polymerase (sigma54-RNAP). In response to nitrogen starvation, NTR system activates the expression of nac, and NAC activates the expression of hut, ure, and put (proline utilization). NAC is not involved in the transcription of Sigma70-RNAP operons such as glnA, which directly respond by the NTR system, but activates the transcription of sigma70-RNAP dependent operons such as hut. Hence, NAC allows the coupling of sigma70-RNAP dependent operons to the sigma54-RNAP dependent NTR system.  This substrate-binding domain has significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>DNA-binding transcriptional repressor LrhA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>nitrogen assimilation transcriptional regulator; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type transcriptional regulators involved in the catabolism of aromatic compounds and that of other related regulators, contains type 2 periplasmic binding fold. This CD includes the C-terminal substrate binding domain of LTTRs involved in degradation of aromatic compounds, such as CbnR, BenM, CatM, ClcR and TfdR, as well as that of other transcriptional regulators clustered together in phylogenetic trees, including XapR, HcaR, MprR, IlvR, BudR, AlsR, LysR, and OccR. The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.  Besides transport proteins, the PBP2 superfamily includes the substrate-binding domains from ionotropic glutamate receptors, LysR-like transcriptional regulators, and unorthodox sensor proteins involved in signal transduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>The C-terminal substrate domain of LysR-type transcriptional regulators CysB-like contains type 2 periplasmic binding fold. CysB is a transcriptional activator of genes involved in sulfate and thiosulfate transport, sulfate reduction, and cysteine synthesis. In Escherichia coli, the regulation of transcription in response to sulfur source is attributed to two transcriptional regulators, CysB and Cbl. CysB, in association with Cbl, downregulates the expression of ssuEADCB operon which is required for the utilization of sulfur from aliphatic sulfonates, in the presence of cysteine. Also, Cbl and CysB together directly function as transcriptional activators of tauABCD genes, which are required for utilization of taurine as sulfur source for growth. Like many other members of the LTTR family, CysB is composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.  Besides transport proteins, the PBP2 superfamily includes the substrate-binding domains from ionotropic glutamate receptors, LysR-like transcriptional regulators, and unorthodox sensor proteins involved in signal transduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>LysR family transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of galactose-binding protein regulator contains the type 2 periplasmic binding fold. Galactose-binding protein regulator (GbpR), a member of the LysR family of bacterial transcriptional regulators, regulates the expression of chromosomal virulence gene chvE.   The chvE gene is involved in the uptake of specific sugars, in chemotaxis to these sugars, and in the VirA-VirG two-component signal transduction system. In the presence of an inducing sugar such as L-arabinose, D-fucose, or D-galactose, GbpR activates chvE expression, while in the absence of an inducing sugar, GbpR represses expression. The topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=co>51</td>
            <td id=al>DNA-binding transcriptional activator GcvA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>HTH-type transcriptional activator IlvY. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>The C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>LysR family transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>DNA-binding transcriptional regulator LysR; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type transcriptional regulators, YofA and SoxR, contains the type 2 periplasmic binding fold. YofA is a LysR-like transcriptional regulator of cell growth in Bacillus subtillis. YofA controls cell viability and the formation of constrictions during cell division. YofaA positively regulates expression of the cell division gene ftsW, and thus is essential for cell viability during stationary-phase growth of Bacillus substilis. YofA shows significant homology to SoxR from Arthrobacter sp. TE1826. SoxR is a negative regulator for the sarcosine oxidase gene soxA. Sarcosine oxidase catalyzes the oxidative demethylation of sarcosine, which is involved in the metabolism of creatine and choline. The topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate binding domain of LysR, transcriptional regulator for lysine biosynthesis, contains the type 2 periplasmic binding fold. LysR, the transcriptional activator of lysA encoding diaminopimelate decarboxylase, catalyses the decarboxylation of diaminopimelate to produce lysine. The LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>The C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>DNA-binding transcriptional regulator DsdC. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>58</td>
            <td id=al>HTH-type transcriptional regulator MetR. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>HTH-type transcriptional activator AllS. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>The C-terminal substrate domain of LysR-type transcriptional regulator CysB contains type 2 periplasmic binding fold. CysB is a transcriptional activator of genes involved in sulfate and thiosulfate transport, sulfate reduction, and cysteine synthesis. In Escherichia coli, the regulation of transcription in response to sulfur source is attributed to two transcriptional regulators, CysB and Cbl. CysB, in association with Cbl, downregulates the expression of ssuEADCB operon which is required for the utilization of sulfur from aliphatic sulfonates, in the presence of cysteine. Also, Cbl and CysB together directly function as transcriptional activators of tauABCD genes, which are required for utilization of taurine as sulfur source for growth. Like many other members of the LTTR family, CysB is composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type transcriptional regulator Cbl, which is required for expression of sulfate starvation-inducible (ssi) genes, contains the type 2 periplasmic binding fold. Cbl is a member of the LysR transcriptional regulators that comprise the largest family of prokaryotic transcription factor. Cbl shows high sequence similarity to CysB, the LysR-type transcriptional activator of genes involved in sulfate and thiosulfate transport, sulfate reduction, and cysteine synthesis. In Escherichia coli, the function of Cbl is required for expression of sulfate starvation-inducible (ssi) genes, coupled with the biosynthesis of cysteine from the organic sulfur sources (sulfonates). The ssi genes include the ssuEADCB and tauABCD operons encoding uptake systems for organosulfur compounds, aliphatic sulfonates, and taurine. The genes in these operons encode an ABC-type transport system required for uptake of aliphatic sulfonates and a desulfonation enzyme. Both Cbl and CysB require expression of the tau and ssu genes.  Like many other members of the LTTR family, the Cbl is composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>putative DNA-binding transcriptional regulator; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cr>31</td>
            <td id=al>ArgP/LysG family DNA-binding transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>The C-terminal substrate binding domain of an uncharacterized LysR-type transcriptional regulator, contains the type 2 periplasmic binding fold. LysR-transcriptional regulators comprise the largest family of prokaryotic transcription factor. Homologs of some of LTTRs with similar domain organizations are also found in the archaea and eukaryotic organisms. The LTTRs are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of nitrogen-fixing bacteria, and synthesis of virulence factors, to a name a few.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type transcriptional regulator HcaR in involved in 3-phenylpropionic acid catabolism, contains the type2 periplasmic binding fold. HcaR, a member of the LysR family of transcriptional regulators, controls the expression of the hcA1, A2, B, C, and D operon, encoding for the 3-phenylpropionate dioxygenase complex and 3-phenylpropionate-2',3'-dihydrodiol dehydrogenase, that oxidizes 3-phenylpropionate to 3-(2,3-dihydroxyphenyl) propionate.  Dioxygenases play an important role in protecting the cell against the toxic effects of dioxygen. The expression of hcaR is negatively auto-regulated, as for other members of the LysR family, and is strongly repressed in the presence of glucose. This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type trnascriptional regulator AlsR, which regulates acetoin formation under stationary phase growth conditions; contains the type 2 periplasmic binding fold. AlsR is responsible for activating the expression of the acetoin operon (alsSD) in response to inducing signals such as glucose and acetate.  Like many other LysR family proteins, AlsR is transcribed divergently from the alsSD operon. The alsS gene encodes acetolactate synthase, an enzyme involved in the production of acetoin in cells of stationary-phase. AlsS catalyzes the conversion of two pyruvate molecules to acetolactate and carbon dioxide. Acetolactate is then converted to acetoin at low pH by acetolactate decarboxylase which encoded by the alsD gene. Acetoin is an important physiological metabolite excreted by many microorganisms grown on glucose or other fermentable carbon sources. This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>transcriptional regulator TdcA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>The C-terminal substrate binding domain of LysR-type transcrptional regulator BudR, which is responsible for activation of the expression of the butanediol operon genes; contains the type 2 periplasmic binding fold. This CD represents the substrate binding domain of BudR regulator, which is responsible for induction of the butanediol formation pathway under fermentative growth conditions. Three enzymes are involved in the production of 1 mol of 2,3 butanediol from the condensation of 2 mol of pyruvate with acetolactate and acetoin as intermediates: acetolactate synthetase, acetolactate decarboxylase, and acetoin reductase. In Klebsiella terrigena, BudR regulates the expression of the budABC operon genes, encoding these three enzymes of the butanediol pathway. In many bacterial species, the use of this pathway can prevent intracellular acidification by diverting metabolism from acid production to the formation of neutral compounds (acetoin and butanediol). This substrate-binding domain has significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate-binding domain of the LysR-type transcriptional regulator CynR, contains the type 2 periplasmic binding fold. CynR is a LysR-like transcriptional regulator of the cyn operon, which encodes genes that allow cyanate to be used as a sole source of nitrogen. The operon includes three genes in the following order: cynT (cyanate permease), cynS (cyanase), and cynX (a protein of unknown function).  CynR negatively regulates its own expression independently of cyanate. CynR binds to DNA and induces bending of DNA in the presence or absence of cyanate, but the amount of bending is decreased by cyanate. The CynR of LysR-type transcriptional regulator family is composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins (PBP2). The PBP2 are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>42</td>
            <td id=al>LysR family transcriptional regulator. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate-binding domain of putative LysR-type transcriptional regulator PAO1-like, a member of the type 2 periplasmic binding fold protein superfamily. This family includes the C-terminal substrate domain of a putative LysR-type transcriptional regulator from the plant pathogen Pseudomonas aeruginosa PAO1and its closely related homologs. The LysR-type transcriptional regulators (LTTRs) are composed of two functional domains joined by a linker helix involved in oligomerization: an N-terminal HTH (helix-turn-helix) domain, which is responsible for the DNA-binding specificity, and a C-terminal substrate-binding domain, which is structurally homologous to the type 2 periplasmic binding proteins. As also observed in the periplasmic binding proteins, the C-terminal domain of the bacterial transcriptional repressor undergoes a conformational change upon substrate binding which in turn changes the DNA binding affinity of the repressor.  The genes controlled by the LTTRs have diverse functional roles including amino acid biosynthesis, CO2 fixation, antibiotic resistance, degradation of aromatic compounds, nodule formation of N2 fixing bacteria, and synthesis of virulence factors, to a name a few. The structural topology of this substrate-binding domain is most similar to that of the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.  Besides transport proteins, the PBP2 superfamily includes the substrate-binding domains from ionotropic glutamate receptors, LysR-like transcriptional regulators, and unorthodox sensor proteins involved in signal transduction.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cr>21</td>
            <td id=al>HTH-type transcriptional regulator HdfR. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cr>28</td>
            <td id=al>HTH-type transcriptional regulator ArgP. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The C-terminal substrate-domain of LysR-type transcriptional regulator, OccR, involved in the catabolism of octopine, contains the type 2 periplasmic binding fold. This CD includes the C-terminal substrate-domain of LysR-type transcriptional regulator OccR, which is involved in the catabolism of octopine. Opines are low molecular weight compounds found in plant crown gall tumors produced by the parasitic bacterium Agrobacterium. There are at least 30 different opines identified so far. Opines are utilized by tumor-colonizing bacteria as a source of carbon, nitrogen, and energy. In Agrobacterium tumefaciens,  OccR protein activates the occQ operon of the Ti plasmid in response to octopine. This operon encodes proteins required for the uptake and catabolism of octopine, an arginine derivative. The occ operon also encodes the TraR protein, which is a quorum-sensing transcriptional regulator of the Ti plasmid tra regulon.  This substrate-binding domain shows significant homology to the type 2 periplasmic binding proteins (PBP2), which are responsible for the uptake of a variety of substrates such as phosphate, sulfate, polysaccharides, lysine/arginine/ornithine, and histidine. The PBP2 bind their ligand in the cleft between these domains in a manner resembling a Venus flytrap. After binding their specific ligand with high affinity, they can interact with a cognate membrane transport complex comprised of two integral membrane domains and two cytoplasmically located ATPase domains. This interaction triggers the ligand translocation across the cytoplasmic membrane energized by ATP hydrolysis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cr>39</td>
            <td id=al>HTH-type transcriptional activator AaeR. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>648048</td></tr>
<tr><td id=al>end</td><td id=al>649113</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>LysR family transcriptional regulator</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
