<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00759</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00759 mannose-1-phosphate guanyltransferase [Phormidium_A]</h3>
    <div>MRAVLMAGGSGTRLRPLTCDLPKPMVPILNRPIAEHILNLLKRHNITEVIATVHYLPDVMRDYFQDGKDFGVQMTYAVEEEQPLGTAGCVKNVEDLLDDTFVVISGDAITDFDLQEAIAFHREKGSKATLVLKHVPNPIEFGVVITDEQQRIVRFLEKPSASEIFSDTVNTGTYILEPEVLQYLPSNQECDFSKDLFPTLLDKGEPLYGYIAEGYWCDVGHLEAYRESQYDALNRKVAITYPYEEREPGVWVGQNSFIDDSVKIKPPVLIGENCRIGPRTNIEPGTVIGDNVTIGADADLKRPIIWNGAIIGDEVHLRACTICRGTRVDRRAHILEGAVVGPLSSVGEEAQISPGVRIWPSKKVESGAILNINLIWGQTARRNLFGQRGVSGLANIDITPEFAVKLGAAYGSSLKQGEQVVVSRDQRSISRMVSRSIIAGIMSVGVNVQNLEATAIPLSRSFARASRNGMTSRPTIAGGIHVRIHPDRSDSILIEFFDANGINITKSKEKKIEGGYFKEDIRRAQIAEIGDMTYPSQVLEMYNTGFEKLLNVEAIRHSNSKIVIDYAYAVSGAILPMLLAKFGSDAVVLNASLNQLALANEEREGLLTQLGQVVEALRANFGAQVSANGEQFILVDESGLPIRGEQLTAFMVNMILTAHPRGTVVVPVHTSSAIEQIARRHDGQVIRTKANPTALMEACHTNPNVVLGGSGEMGFIFPQLHPGFDAMFSIAKLIEMETLQGRSLGQIRAELPRVYHRCQTIRCPWTAKGALMRHMVETHSADRIELTDGVKVCEPQSDNWVLILPDAGEPVVHIFANGDDRDWVDEMLLDYRGRIQKFVEQQQGGQNLPLDEI</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bifunctional mannose-1-phosphate guanylyltransferase / phosphomannomutase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanylyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanylyltransferase</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mannose-1-phosphate guanyltransferase</td>
            <td id=co>Oxynema</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-mannose-1-phosphate guanyltransferase (MPG1 transferase), also known as GDP-mannose pyrophosphorylase, is a bifunctional enzyme with both phosphomannose isomerase (PMI) activity and GDP-mannose phosphorylase (GMP) activity.  The protein contains an N-terminal NTP transferase domain, an L-beta-H domain, and a C-terminal PGM-like domain that belongs to the alpha-D-phosphohexomutase superfamily.  This subfamily is limited to bacteria and archaea. The alpha-D-phosphohexomutases include several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. Members of this group appear to lack conserved residues necessary for metal binding and catalytic activity. Other members of this superfamily include the phosphoglucomutases (PGM1 and PGM2), phosphoglucosamine mutase (PNGM), phosphoacetylglucosamine mutase (PAGM), the bacterial phosphomannomutase ManB, the bacterial phosphoglucosamine mutase GlmM, and the bifunctional phosphomannomutase/phosphoglucomutase (PMM/PGM). Each of these enzymes has four domains with a centrally located active site formed by four loops, one from each domain. All four domains are included in this alignment model.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NDP-sugar pyrophosphorylase, includes eIF-2Bgamma, eIF-2Bepsilon, and LPS biosynthesis proteins [Translation, ribosomal structure and biogenesis, Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_transferases catalyze the transfer of nucleotides onto phosphosugars. Nucleotidyltransferases transfer nucleotides onto phosphosugars.  The enzyme family includes Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase. The products are activated sugars that are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_trnasferase_like_2 is a member of the nucleotidyl transferase family. This is a subfamily of nucleotidyl transferases. Nucleotidyl transferases transfer nucleotides onto phosphosugars. The activated sugars are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides. Other subfamilies of nucleotidyl transferases include Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase. The MJ_1101 protein from Methanococcus jannaschii has been characterized as the GlmU enzyme catalyzing the final two steps of UDP-GlcNAc biosynthesis. Many of the genes identified by this model are in proximity to the GlmS and GlmM genes and are also presumed to be GlmU. However, some archaeal genomes contain multiple closely-related homologs from this family and it is not clear what the substrate specificity is for each of them.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Nucleotidyl transferase. This family includes a wide range of enzymes which transfer nucleotides onto phosphosugars.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>N-terminal domain of the M1P-guanylyltransferase B-isoform like proteins. GDP-mannose pyrophosphorylase  (GTP: alpha-d-mannose-1-phosphate guanyltransferase) catalyzes the formation of GDP-d-mannose from GTP and alpha-d-mannose-1-Phosphate. It contains an N-terminal catalytic domain and a C-terminal Lefthanded-beta-Helix fold domain. GDP-d-mannose is the activated form of mannose for formation of cell wall lipoarabinomannan and various mannose-containing glycolipids and polysaccharides. The function of GDP-mannose pyrophosphorylase is essential for cell wall integrity, morphogenesis and viability. Repression of GDP-mannose pyrophosphorylase in yeast leads to phenotypes, such as cell lysis, defective cell wall, and failure of polarized growth and cell separation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>This archaeal PGM-like (phosphoglucomutase-like) protein of unknown function belongs to the alpha-D-phosphohexomutase superfamily which includes several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. The alpha-D-phosphohexomutases include several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. Members of this superfamily include the phosphoglucomutases (PGM1 and PGM2), phosphoglucosamine mutase (PNGM), phosphoacetylglucosamine mutase (PAGM), the bacterial phosphomannomutase ManB, the bacterial phosphoglucosamine mutase GlmM, and the bifunctional phosphomannomutase/phosphoglucomutase (PMM/PGM). Each of these enzymes has four domains with a centrally located active site formed by four loops, one from each domain. All four domains are included in this alignment model.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>WcbM_like is a subfamily of nucleotidyl transferases. WcbM protein of Burkholderia mallei is involved in the biosynthesis, export or translocation of capsule. It is a subfamily of nucleotidyl transferases that transfer nucleotides onto phosphosugars.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>NTP_transferase_like_1 is a member of the nucleotidyl transferase family. This is a subfamily of nucleotidyl transferases. Nucleotidyl transferases transfer nucleotides onto phosphosugars. The activated sugars are precursors for synthesis of lipopolysaccharide, glycolipids and polysaccharides. Other subfamilies of nucleotidyl transferases include Alpha-D-Glucose-1-Phosphate Cytidylyltransferase, Mannose-1-phosphate guanyltransferase, and Glucose-1-phosphate thymidylyltransferase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>G1P_TT_long represents the long form of glucose-1-phosphate thymidylyltransferase. This family is the long form of Glucose-1-phosphate thymidylyltransferase.  Glucose-1-phosphate thymidylyltransferase catalyses the formation of dTDP-glucose, from dTTP and glucose 1-phosphate. It is the first enzyme in the biosynthesis of dTDP-L-rhamnose, a cell wall constituent and a feedback inhibitor of the enzyme.There are two forms of   Glucose-1-phosphate thymidylyltransferase in bacteria and archeae; short form and long form.  The long form, which has an extra 50 amino acids c-terminal, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced.The long from enzymes also have a left-handed parallel helix domain at the c-terminus, whereas, th eshort form enzymes do not have this domain. The homotetrameric, feedback inhibited short form is found in numerous bacterial species that produce dTDP-L-rhamnose.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>glucose-1-phosphate thymidylylransferase, long form. The family of known and putative glucose-1-phosphate thymidyltransferase (also called dTDP-glucose synthase) shows a deep split into a short form (see TIGR01207) and a long form described by this model. The homotetrameric short form is found in numerous bacterial species that incorporate dTDP-L-rhamnose, which it helps synthesize, into the cell wall. It is subject to feedback inhibition. This form, in contrast, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced. Alternate name: dTDP-D-glucose synthase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>dTDP-glucose pyrophosphorylase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>This PGM-like (phosphoglucomutase-like) domain is located C-terminal to a mannose-1-phosphate guanyltransferase domain in a protein of unknown function that is found in both prokaryotes and eukaryotes. This domain belongs to the alpha-D-phosphohexomutase superfamily which includes several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. Members of this superfamily include the phosphoglucomutases (PGM1 and PGM2), phosphoglucosamine mutase (PNGM), phosphoacetylglucosamine mutase (PAGM), the bacterial phosphomannomutase ManB, the bacterial phosphoglucosamine mutase GlmM, and the bifunctional phosphomannomutase/phosphoglucomutase (PMM/PGM). Each of these enzymes has four domains with a centrally located active site formed by four loops, one from each domain. All four domains are included in this alignment model.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>G1P_cytidylyltransferase catalyzes the production of CDP-D-Glucose. Alpha-D-Glucose-1-phosphate Cytidylyltransferase catalyzes the production of CDP-D-Glucose from alpha-D-Glucose-1-phosphate and MgCTP as substrate. CDP-D-Glucose is the precursor  for synthesizing four of the five naturally occurring 3,6-dideoxy sugars-abequose (3,6-dideoxy-D-Xylo-hexose), ascarylose (3,6-dideoxy-L-arabino-hexose), paratose (3,6-dideoxy-D-ribohexose), and tyvelose (3,6-dideoxy-D-arabino-hexose. Deoxysugars are ubiquitous in nature where they function in a variety of biological processes, including cell adhesion, immune response, determination of ABO blood groups, fertilization, antibiotic function, and microbial pathogenicity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ADP-glucose pyrophosphorylase [Carbohydrate transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>N-terminal domain of M1P_guanylyl_A_ like proteins are likely to be a isoform of GDP-mannose pyrophosphorylase. N-terminal domain of the M1P-guanylyltransferase A-isoform like proteins:  The proteins of this family are likely to be a isoform of GDP-mannose pyrophosphorylase. Their sequences are highly conserved with mannose-1-phosphate guanyltransferase, but  generally about 40-60 bases longer.  GDP-mannose pyrophosphorylase (GTP: alpha-d-mannose-1-phosphate guanyltransferase) catalyzes the formation of GDP-d-mannose from GTP and alpha-d-mannose-1-Phosphate. It contains an N-terminal catalytic domain that resembles a dinucleotide-binding Rossmann fold and a C-terminal LbH fold domain. GDP-d-mannose is the activated form of mannose for formation of cell wall lipoarabinomannan and various mannose-containing glycolipids and polysaccharides. The function of GDP-mannose pyrophosphorylase is essential for cell wall integrity, morphogenesis and viability.  Repression of GDP-mannose pyrophosphorylase in yeast leads to phenotypes including cell lysis, defective cell wall, and failure of polarized growth and cell separation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The phosphomannomutase/phosphoglucomutase (PMM/PGM) bifunctional enzyme catalyzes the reversible conversion of 1-phospho to 6-phospho-sugars (e.g. between mannose-1-phosphate and mannose-6-phosphate or glucose-1-phosphate and glucose-6-phosphate) via a bisphosphorylated sugar intermediate. The reaction involves two phosphoryl transfers, with an intervening 180 degree reorientation of the reaction intermediate during catalysis. Reorientation of the intermediate occurs without dissociation from the active site of the enzyme and is thus, a simple example of processivity, as defined by multiple rounds of catalysis without release of substrate. Glucose-6-phosphate and glucose-1-phosphate are known to be utilized for energy metabolism and cell surface construction, respectively. PMM/PGM belongs to the alpha-D-phosphohexomutase superfamily which includes several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. Other members of this superfamily include phosphoglucosamine mutase (PNGM), phosphoacetylglucosamine mutase (PAGM), the bacterial phosphomannomutase ManB, the bacterial phosphoglucosamine mutase GlmM, and the phosphoglucomutases (PGM1 and PGM2). Each of these enzymes has four domains with a centrally located active site formed by four loops, one from each domain. All four domains are included in this alignment model.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>phosphoglucosamine mutase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>Bifunctional protein GlmU, N-acetylglucosamine-1-phosphate-uridyltransferase/glucosamine-1-phosphate-acetyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>G1P_TT_short is the short form of glucose-1-phosphate thymidylyltransferase. This family is the short form of glucose-1-phosphate thymidylyltransferase.  Glucose-1-phosphate thymidylyltransferase catalyses the formation of dTDP-glucose, from dTTP and glucose 1-phosphate. It is the first enzyme in the biosynthesis of dTDP-L-rhamnose, a cell wall constituent and a feedback inhibitor of the enzyme.There are two forms of   Glucose-1-phosphate thymidylyltransferase in bacteria and archeae; short form and long form. The homotetrameric, feedback inhibited short form is found in numerous bacterial species that produce dTDP-L-rhamnose. The long form, which has an extra 50 amino acids c-terminal, is found in many species for which it serves as a sugar-activating enzyme for antibiotic biosynthesis and or other, unknown pathways, and in which dTDP-L-rhamnose is not necessarily produced.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The N-terminal of eIF-2B_gamma_like is predicted to have glycosyltransferase activity. N-terminal domain of eEIF-2B epsilon and gamma, subunits of eukaryotic translation initiators, is a subfamily of glycosyltranferase 2 and is predicted to have glycosyltranferase activity. eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit epsilon shares sequence similarity with gamma subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>N-terminal domain of bacterial GlmU. The N-terminal domain of N-Acetylglucosamine-1-phosphate uridyltransferase (GlmU). GlmU is an essential bacterial enzyme with both an acetyltransferase and an uridyltransferase activity which have been mapped to the C-terminal and N-terminal domains, respectively. This family represents the N-terminal uridyltransferase. GlmU performs the last two steps in the synthesis of UDP-N-acetylglucosamine (UDP-GlcNAc), which is an essential precursor in both the peptidoglycan and the lipopolysaccharide metabolic pathways in Gram-positive and Gram-negative bacteria, respectively.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphoglucomutase/phosphomannomutase, alpha/beta/alpha domain I. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cb>73</td>
            <td id=al>The N-terminal domain of gamma subunit of the eIF-2B is a subfamily of glycosyltransferase 2. N-terminal domain of gamma subunit of the eukaryotic translation initiation factor 2B (eIF-2B): eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit gamma shares sequence similarity with epsilon subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>glucose-1-phosphate adenylyltransferase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>glucose-1-phosphate adenylyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The N-terminal domain of epsilon subunit of the eIF-2B is a subfamily of glycosyltransferase 2. N-terminal domain of epsilon subunit of the eukaryotic translation initiation factor 2B (eIF-2B): eIF-2B is a guanine nucleotide-exchange factor which mediates the exchange of GDP (bound to initiation factor eIF2) for GTP, generating active eIF2.GTP complex. EIF2B is a complex multimeric protein consisting of five subunits named alpha, beta, gamma, delta and epsilon. Subunit epsilon shares sequence similarity with gamma subunit, and with a family of bifunctional nucleotide-binding enzymes such as ADP-glucose pyrophosphorylase, suggesting that epsilon subunit may play roles in nucleotide binding activity. In yeast, eIF2B gamma enhances the activity of eIF2B-epsilon leading to the idea that these subunits form the catalytic subcomplex.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>Putative glucose-1-phosphate thymidylyltransferase, C-terminal Left-handed parallel beta-Helix (LbH) domain: Proteins in this family show simlarity to glucose-1-phosphate adenylyltransferases in that they contain N-terminal catalytic domains that resemble a dinucleotide-binding Rossmann fold and C-terminal LbH fold domains. Members in this family are predicted to be glucose-1-phosphate thymidylyltransferases, which are involved in the dTDP-L-rhamnose biosynthetic pathway. Glucose-1-phosphate thymidylyltransferase catalyzes the synthesis of deoxy-thymidine di-phosphate (dTDP)-L-rhamnose, an important component of the cell wall of many microorganisms. The C-terminal LbH domain contains multiple turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>This PGM-like (phosphoglucomutase-like) protein of unknown function belongs to the alpha-D-phosphohexomutase superfamily and is found in both archaea and bacteria. The alpha-D-phosphohexomutases include several related enzymes that catalyze a reversible intramolecular phosphoryl transfer on their sugar substrates. Other members of this superfamily include phosphoglucosamine mutase (PNGM), phosphoacetylglucosamine mutase (PAGM), the bacterial phosphomannomutase ManB, the bacterial phosphoglucosamine mutase GlmM, and the bifunctional phosphomannomutase/phosphoglucomutase (PMM/PGM). Each of these enzymes has four structural domains (subdomains) with a centrally located active site formed by four loops, one from each subdomain. All four subdomains are included in this alignment model.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>ADP-glucose pyrophosphorylase is involved in the biosynthesis of glycogen or starch. ADP-glucose pyrophosphorylase (glucose-1-phosphate adenylyltransferase) catalyzes a very important step in the biosynthesis of alpha 1,4-glucans (glycogen or starch) in bacteria and plants: synthesis of the activated glucosyl donor, ADP-glucose, from glucose-1-phosphate and ATP.  ADP-glucose pyrophosphorylase is a tetrameric allosterically regulated enzyme. While a homotetramer in bacteria, in plant chloroplasts and amyloplasts, it is a heterotetramer of two different, yet evolutionary related, subunits.  There are a number of conserved regions in the sequence of bacterial and plant ADP-glucose pyrophosphorylase subunits. It is a subfamily of a very diverse glycosy transferase family 2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Prokaryotic UGPase catalyses the synthesis of UDP-glucose. Prokaryotic UDP-Glucose Pyrophosphorylase (UGPase) catalyzes a reversible production of UDP-Glucose  and pyrophosphate (PPi) from glucose-1-phosphate and UTP.  UDP-glucose plays pivotal roles in galactose utilization, in glycogen synthesis, and in the synthesis of the carbohydrate moieties of glycolipids , glycoproteins , and proteoglycans. UGPase is found in both prokaryotes and eukaryotes, although prokaryotic and eukaryotic forms of UGPase catalyze the same reaction, they share low sequence similarity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>phosphoglucosamine mutase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>glucose-1-phosphate thymidylyltransferase RfbA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>phosphoglucosamine mutase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>Phosphocholine cytidylyltransferases catalyze the synthesis of CDP-choline. This family contains proteins similar to prokaryotic phosphocholine (P-cho) cytidylyltransferases. Phosphocholine (PC) cytidylyltransferases catalyze the transfer of a cytidine monophosphate from CTP to phosphocholine to form CDP-choline. PC is the most abundant phospholipid in eukaryotic membranes and it is also important in prokaryotic membranes. For pathogenic prokaryotes, the cell surface PC facilitates the interaction with host surface and induces attachment and invasion. In addition cell wall PC serves as scaffold for a group of choline-binding proteins that are secreted from the cells. Phosphocholine (PC) cytidylyltransferase is a key enzyme in the prokaryotic choline metabolism pathway. It has been hypothesized to consist of a choline transport system, a choline kinase, CTP:phosphocholine cytidylyltransferase, and a choline phosphotransferase that transfers P-Cho from CDP-Cho to either lipoteichoic acid or lipopolysaccharide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=co>52</td>
            <td id=al>UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>26</td>
            <td id=al>UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>phosphoglucosamine mutase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>eIF-2B gamma subunit, C-terminal Left-handed parallel beta-Helix (LbH) domain: eIF-2B is a eukaryotic translation initiator, a guanine nucleotide exchange factor (GEF) composed of five different subunits (alpha, beta, gamma, delta and epsilon). eIF2B is important for regenerating GTP-bound eIF2 during the initiation process. This event is obligatory for eIF2 to bind initiator methionyl-tRNA, forming the ternary initiation complex. The eIF-2B gamma subunit contains an N-terminal domain that resembles a dinucleotide-binding Rossmann fold and a C-terminal LbH domain with 4 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). The epsilon and gamma subunits form the catalytic subcomplex of eIF-2B, which binds eIF2 and catalyzes guanine nucleotide exchange.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cr>36</td>
            <td id=al>UDP-3-O-acyl-glucosamine N-acyltransferase (LpxD): The enzyme catalyzes the transfer of 3-hydroxymyristic acid or 3-hydroxy-arachidic acid, depending on the organism, from the acyl carrier protein (ACP) to UDP-3-O-acyl-glucosamine to produce UDP-2,3-diacyl-GlcNAc. This constitutes the third step in the lipid A biosynthetic pathway in Gram-negative bacteria. LpxD is a homotrimer, with each subunit consisting of a novel combination of an N-terminal uridine-binding domain, a core lipid-binding left-handed parallel beta helix (LbH) domain, and a C-terminal alpha-helical extension. The LbH domain contains 9 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>31</td>
            <td id=al>UDP-3-O-acyl-glucosamine N-acyltransferase (LpxD): The enzyme catalyzes the transfer of 3-hydroxymyristic acid or 3-hydroxy-arachidic acid, depending on the organism, from the acyl carrier protein (ACP) to UDP-3-O-acyl-glucosamine to produce UDP-2,3-diacyl-GlcNAc. This constitutes the third step in the lipid A biosynthetic pathway in Gram-negative bacteria. LpxD is a homotrimer, with each subunit consisting of a novel combination of an N-terminal uridine-binding domain, a core lipid-binding left-handed parallel beta helix (LbH) domain, and a C-terminal alpha-helical extension. The LbH domain contains 9 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=co>52</td>
            <td id=al>UDP-3-O-acyl-glucosamine N-acyltransferase (LpxD): The enzyme catalyzes the transfer of 3-hydroxymyristic acid or 3-hydroxy-arachidic acid, depending on the organism, from the acyl carrier protein (ACP) to UDP-3-O-acyl-glucosamine to produce UDP-2,3-diacyl-GlcNAc. This constitutes the third step in the lipid A biosynthetic pathway in Gram-negative bacteria. LpxD is a homotrimer, with each subunit consisting of a novel combination of an N-terminal uridine-binding domain, a core lipid-binding left-handed parallel beta helix (LbH) domain, and a C-terminal alpha-helical extension. The LbH domain contains 9 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>bifunctional UDP-N-acetylglucosamine diphosphorylase/glucosamine-1-phosphate N-acetyltransferase GlmU. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cr>24</td>
            <td id=al>UDP-3-O-[3-hydroxymyristoyl] glucosamine N-acyltransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>UTP-glucose-1-phosphate uridylyltransferase [Cell wall/membrane/envelope biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>N-acetyl-glucosamine-1-phosphate uridyltransferase (GlmU), C-terminal left-handed beta-helix (LbH) acetyltransferase domain: GlmU is also known as UDP-N-acetylglucosamine pyrophosphorylase. It is a bifunctional bacterial enzyme that catalyzes two consecutive steps in the formation of UDP-N-acetylglucosamine (UDP-GlcNAc), an important precursor in bacterial cell wall formation. The two enzymatic activities, uridyltransferase and acetyltransferase, are carried out by two independent domains. The C-terminal LbH domain possesses the acetyltransferase activity. It catalyzes the CoA-dependent acetylation of GlcN-1-phosphate to GlcNAc-1-phosphate. The LbH domain contains 10 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X. The acetyltransferase active site is located at the interface between two subunits of the active LbH trimer.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>N-acetyl-glucosamine-1-phosphate uridyltransferase (GlmU), C-terminal left-handed beta-helix (LbH) acetyltransferase domain: GlmU is also known as UDP-N-acetylglucosamine pyrophosphorylase. It is a bifunctional bacterial enzyme that catalyzes two consecutive steps in the formation of UDP-N-acetylglucosamine (UDP-GlcNAc), an important precursor in bacterial cell wall formation. The two enzymatic activities, uridyltransferase and acetyltransferase, are carried out by two independent domains. The C-terminal LbH domain possesses the acetyltransferase activity. It catalyzes the CoA-dependent acetylation of GlcN-1-phosphate to GlcNAc-1-phosphate. The LbH domain contains 10 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X. The acetyltransferase active site is located at the interface between two subunits of the active LbH trimer.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>Left-handed parallel beta-Helix (LbH) domain of a group of proteins with similarity to glucose-1-phosphate adenylyltransferase: Included in this family are glucose-1-phosphate adenylyltransferase, mannose-1-phosphate guanylyltransferase, and the eukaryotic translation initiation factor eIF-2B subunits, epsilon and gamma. Most members of this family contains an N-terminal catalytic domain that resembles a dinucleotide-binding Rossmann fold, followed by a LbH fold domain with at least 4 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). eIF-2B epsilon contains an additional domain of unknown function at the C-terminus. Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>Left-handed parallel beta-Helix (LbH) domain of a group of proteins with similarity to glucose-1-phosphate adenylyltransferase: Included in this family are glucose-1-phosphate adenylyltransferase, mannose-1-phosphate guanylyltransferase, and the eukaryotic translation initiation factor eIF-2B subunits, epsilon and gamma. Most members of this family contains an N-terminal catalytic domain that resembles a dinucleotide-binding Rossmann fold, followed by a LbH fold domain with at least 4 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). eIF-2B epsilon contains an additional domain of unknown function at the C-terminus. Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>CTP:phosphocholine cytidylyltransferase involved in choline phosphorylation for cell surface LPS epi... [Cell wall/membrane/envelope biogenesis, Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>WcxM-like, Left-handed parallel beta-Helix (LbH) N-terminal domain: This group is composed of Xanthomonas campestris WcxM and proteins with similarity to the WcxM N-terminal domain. WcxM is thought to be bifunctional, catalyzing both the isomerization and transacetylation reactions of keto-hexoses. It contains an N-terminal LbH domain responsible for the transacetylation function and a C-terminal isomerase domain. The LbH domain contains imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), typical of enzymes with acyltransferase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphoglucomutase/phosphomannomutase, alpha/beta/alpha domain III. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Gamma carbonic anhydrase-like: This family is composed of gamma carbonic anhydrase (CA), Ferripyochelin Binding Protein (FBP), E. coli paaY protein, and similar proteins. CAs are zinc-containing enzymes that catalyze the reversible hydration of carbon dioxide in a two-step mechanism, involving the nucleophilic attack of a zinc-bound hydroxide ion on carbon dioxide, followed by the regeneration of the active site by ionization of the zinc-bound water molecule and removal of a proton from the active site. They are ubiquitous enzymes involved in fundamental processes like photosynthesis, respiration, pH homeostasis and ion transport. There are three evolutionary distinct groups - alpha, beta and gamma carbonic anhydrases - which show no significant sequence identity or structural similarity. Gamma CAs are trimeric enzymes with left-handed parallel beta helix (LbH) structural domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>eIF-2B epsilon subunit, central Left-handed parallel beta-Helix (LbH) domain: eIF-2B is a eukaryotic translation initiator, a guanine nucleotide exchange factor (GEF) composed of five different subunits (alpha, beta, gamma, delta and epsilon). eIF2B is important for regenerating GTP-bound eIF2 during the initiation process. This event is obligatory for eIF2 to bind initiator methionyl-tRNA, forming the ternary initiation complex. The eIF-2B epsilon subunit contains an N-terminal domain that resembles a dinucleotide-binding Rossmann fold, a central LbH domain containing 4 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X), and a C-terminal domain of unknown function that is present in eIF-4 gamma, eIF-5, and eIF-2B epsilon. The epsilon and gamma subunits form the catalytic subcomplex of eIF-2B, which binds eIF2 and catalyzes guanine nucleotide exchange.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>Putative Acyltransferase (AT), Left-handed parallel beta-Helix (LbH) domain; This group is composed of mostly uncharacterized proteins containing an N-terminal helical subdomain followed by a LbH domain. The alignment contains 6 turns, each containing three imperfect tandem repeats of a hexapeptide repeat motif (X-[STAV]-X-[LIV]-[GAED]-X). Proteins containing hexapeptide repeats are often enzymes showing acyltransferase activity. A few members are identified as NeuD, a sialic acid (Sia) O-acetyltransferase that is required for Sia synthesis and surface polysaccharide sialylation.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>621747</td></tr>
<tr><td id=al>end</td><td id=al>624309</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[secondary-metabolites]</td></tr>
<tr><td id=al>descr</td><td id=al>mannose-1-phosphate guanyltransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al>(region 2) terpene (Terpene_synth or Terpene_synth_C or phytoene_synt or Lycopene_cycl or terpene_cyclase or NapT7 or fung_ggpps or fung_ggpps2 or trichodiene_synth or TRI5) biosynthetic-additional (smcogs) SMCOG1064:glucose-1-phosphate adenylyl/thymidylyltransferase (Score: 225; E-value: 1.9e-68) biosynthetic-additional</td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
