<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04799</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04799 amino acid adenylation domain-containing protein, partial [Phormidium_A]</h3>
    <div>MNILPDPTQPLQSTWSGSVLNKFNEQVARVPNNVALVDIHKKWSYNELDHSSDQLAQQLQNKGLGYQDIIAIYASRSASLVWALLSILKIGASFTILDSAYPGSRLRDCCRLSGLKGLIHLSSSGDLPQELEQFLSEKEIPYISLPRIPSPVTQQKPEAHFSQPAQLIDADDRAYIAFTSGSTGHPKGILGSHRPLSHFIDWHCRQFGLSEQDRFALLSGLGHDPLLRDIFTPLWLGATLVIPEQQTIETSGKLAQWMQEQDITIAHLTPAMGMLLAEGTRSSASTTALVPSLRYLFWGGDVLTELDIRRSYELSPNVTNVNFYGATETPQAMSYYVVPKTSDFSSHKVPIGQGIDGVQILIVRENGVLAKLGESGEIWIRTPYLSMGYLQDDVLTQTKFVSNPFRDDPSENGGDLDLIYKTGDLGRYKSDGTVEIIGRLDNQVKLRGFRIELSEIESVLAQYPHIQKAVVLLQENGLKKIIAYCLPQKDKPLETAHIREFLIERLPYYMIPQAFVAIAELPLTPNGKIDRQALTQLDISPQIVTSYEAPQDELEEQLVRIWQNHLEQEKIGVQDNFFDLGGHSLLAIALCREVEEVLNRSIPISVLFQNPTIRELAERLRNEPGDLTGSSIITIQANGTQLPIFGVHVLGKGLRYYRPLVKHLGDHQPVYGLAFNLLDDSKGSSKGIKELADIYIKDMKTLQPQGPYHLLGVSIGGRVAFEMAYQLKKQGEEVALLGLLDTTARTGVKHLPASSRVSGHWGKFKAEGLGYVTRKLNSRFGRIKYRVKAIRAQLYQRLGRQVPDDLKSIAGLETNIKIKKQHFPQPYHGKVTLFRAKDRKAEVGTEIDPDYGWRDLAVGGLDIYDIPGDHLLMLQEPNVQVLAAKIKEIIQPSPEMSPGDSPM</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>73</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>amino acid adenylation domain-containing protein, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>amino acid adenylation domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>putative non-ribosomal peptide synthetase module</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>amino acid adenylation domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>48</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>amino acid adenylation domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>non-ribosomal peptide synthetase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>non-ribosomal peptide synthetase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=co>53</td>
            <td id=al>amino acid adenylation (plasmid)</td>
            <td id=co>Cylindrospermum</td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=co>46</td>
            <td id=al>amino acid adenylation domain-containing protein</td>
            <td id=co>Brasilonema</td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>non-ribosomal peptide synthetase</td>
            <td id=co>FACHB-68</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=co>52</td>
            <td id=al>L-aminoadipate-semialdehyde dehydrogenase. Members of this protein family are L-aminoadipate-semialdehyde dehydrogenase (EC 1.2.1.31), product of the LYS2 gene. It is also called alpha-aminoadipate reductase. In fungi, lysine is synthesized via aminoadipate. Currently, all members of this family are fungal.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation domain of nonribosomal peptide synthetases (NRPS). The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>68</td>
            <td id=al>enterobactin non-ribosomal peptide synthetase EntF. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>18</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>16</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation domain of nonribosomal peptide synthetases (NRPS), including Bacillus subtilis termination module Surfactin (SrfA-C). The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and, in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions. This family includes the adenylation domain of the Bacillus subtilis termination module (Surfactin domain, SrfA-C) which recognizes a specific amino acid building block, which is then activated and transferred to the terminal thiol of the 4'-phosphopantetheine (Ppan) arm of the downstream peptidyl carrier protein (PCP) domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Alpha-aminoadipate reductase. This family contains L-2-aminoadipate reductase, also known as alpha-aminoadipate reductase (EC 1.2.1.95) or alpha-AR or L-aminoadipate-semialdehyde dehydrogenase (EC 1.2.1.31), which catalyzes the activation of alpha-aminoadipate by ATP-dependent adenylation and the reduction of activated alpha-aminoadipate by NADPH. The activated alpha-aminoadipate is bound to the phosphopantheinyl group of the enzyme itself before it is reduced to (S)-2-amino-6-oxohexanoate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>similar to adenylation domain of virginiamycin S synthetase. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes virginiamycin S synthetase (VisG) in Streptomyces virginiae; VisG is involved in virginiamycin S (VS) biosynthesis as the provider of an L-pheGly molecule, a highly specific substrate for the last condensation step by VisF. This family also includes linear gramicidin synthetase B (LgrB) in Brevibacillus brevis. Substrate specificity analysis using residues of the substrate-binding pockets of all 16 adenylation domains has shown good agreement of the substrate amino acids predicted with the sequence of linear gramicidin. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation domain of nonribosomal peptide synthetases (NRPS), including salinosporamide A polyketide synthase. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions. This family includes the myxovirescin (TA) antibiotic biosynthetic gene in Myxococcus xanthus; TA production plays a role in predation. It also includes the salinosporamide A polyketide synthase which is involved in the biosynthesis of salinosporamide A, a marine microbial metabolite whose chlorine atom is crucial for potent proteasome inhibition and anticancer activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>12</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cr>11</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>12</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cr>11</td>
            <td id=al>peptide synthase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Peptide Synthetase. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>bacitracin synthetase and related proteins. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes bacitracin synthetases 1, 2, and 3 (BA1, also known as ATP-dependent cysteine adenylase or cysteine activase, BA2, also known as ATP-dependent lysine adenylase or lysine activase, and BA3, also known as ATP-dependent isoleucine adenylase or isoleucine activase) in Bacilli. Bacitracin is a mixture of related cyclic peptides used as a polypeptide antibiotic. This family also includes gramicidin synthetase 1 involved in synthesis of the cyclic peptide antibiotic gramicidin S via activation of phenylalanine. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>amino acid adenylation domain. This model represents a domain responsible for the specific recognition of amino acids and activation as adenylyl amino acids. The reaction catalyzed is aa + ATP -> aa-AMP + PPi. These domains are usually found as components of multi-domain non-ribosomal peptide synthetases and are usually called "A-domains" in that context. A-domains are almost invariably followed by "T-domains" (thiolation domains, pfam00550) to which the amino acid adenylate is transferred as a thiol-ester to a bound pantetheine cofactor with the release of AMP (these are also called peptide carrier proteins, or PCPs. When the A-domain does not represent the first module (corresponding to the first amino acid in the product molecule) it is usually preceded by a "C-domain" (condensation domain, pfam00668) which catalyzes the ligation of two amino acid thiol-esters from neighboring modules. This domain is a subset of the AMP-binding domain found in Pfam (pfam00501) which also hits substrate--CoA ligases and luciferases. Sequences scoring in between trusted and noise for this model may be ambiguous as to whether they activate amino acids or other molecules lacking an alpha amino group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>similar to adenylation domain of anabaenopeptin synthetase (ApnA). This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes Planktothrix agardhii anabaenopeptin synthetase (ApnA A1), which is capable of activating two chemically distinct amino acids (Arg and Tyr). Structural studies show that the architecture of the active site forces Arg to adopt a Tyr-like conformation, thus explaining the bispecificity. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>similar to adenylation domain of cytotrienin synthetase CytC1. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes Streptomyces sp. cytotrienin synthetase (CytC1), a relatively promiscuous adenylation enzyme that installs the aminoacyl moieties on the phosphopantetheinyl arm of the holo carrier protein CytC2. Also included are Streptomyces sp Thr1, involved in the biosynthesis of 4-chlorothreonine, Pseudomonas aeruginosa pyoverdine synthetase D (PvdD), involved in the biosynthesis of the siderophore pyoverdine and Pseudomonas syringae syringopeptin synthetase, where syringpeptin is a necrosis-inducing phytotoxin that functions as a virulence determinant in the plant-pathogen interaction. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation (A) domain of siderophore-synthesizing nonribosomal peptide synthetases (NRPS). The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. This family of siderophore-synthesizing NRPS includes the third adenylation domain of SidN from the endophytic fungus Neotyphodium lolii, ferrichrome siderophore synthetase, HC-toxin synthetase, and enniatin synthase. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>D-alanine:D-alanyl carrier protein ligase (DltA) and similar proteins. This family includes D-alanyl carrier protein ligase DltA and aliphatic beta-amino acid adenylation enzymes IdnL1 and CmiS6. DltA incorporates D-ala in techoic acids in gram-positive bacteria via a two-step process, starting with adenylation of D-alanine that transfers D-alanine to the D-alanyl carrier protein. IdnL1, a short-chain aliphatic beta-amino acid adenylation enzyme, recognizes 3-aminobutanoic acid, and is involved in the synthesis of the macrolactam antibiotic incednine. CmiS6 is a medium-chain beta-amino acid adenylation enzyme that recognizes 3-aminononanoic acid, and is involved in the synthesis of cremimycin, also a macrolactam antibiotic. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>non-ribosomal peptide synthetase. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes pyoverdine biosynthesis protein PvdJ involved in the synthesis of pyoverdine, which consists of a chromophore group attached to a variable peptide chain and comprises around 6-12 amino acids that are specific for each Pseudomonas species, and for which the peptide might be first synthesized before the chromophore assembly. Also included is ornibactin biosynthesis protein OrbI; ornibactin is a tetrapeptide siderophore with an l-ornithine-d-hydroxyaspartate-l-serine-l-ornithine backbone. The adenylation domain at the N-terminal of OrbI possibly initiates the ornibactin with the binding of N5-hydroxyornithine. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation domain of nonribosomal peptide synthetases (NRPS), including Streptoalloteichus tallysomycin biosynthesis genes. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions. This family includes the TLM biosynthetic gene cluster from Streptoalloteichus that consists of nine NRPS genes; the N-terminal module of TlmVI (NRPS-5) and the starter module of BlmVI (NRPS-5) are comprised of the acyl CoA ligase (AL) and acyl carrier protein (ACP)-like domains, which are thought to be involved in the biosynthesis of the beta-aminoalaninamide moiety.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>peptide synthase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>peptide synthase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cr>14</td>
            <td id=al>peptide synthase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cr>7</td>
            <td id=al>peptide synthase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>adenylation (A) domain of linear gramicidin synthetase (LgrA) and similar proteins. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes linear gramicidin synthetase (LgrA) in Brevibacillus brevis. LgrA has a formylation domain fused to the N-terminal end that formylates its substrate for linear gramicidin synthesis to proceed. This formyl group is essential for the clinically important antibacterial activity of gramicidin by enabling head-to-head gramicidin dimers to make a beta-helical pore in gram-positive bacterial membranes, allowing free passage of monovalent cations, destroying the ion gradient and killing bacteria. This family also includes bacitracin synthetase 1 (known as ATP-dependent cysteine adenylase or BA1); it activates cysteine, incorporates two D-amino acids, releases and cyclizes the mature bacitracin, an antibiotic that is a mixture of related cyclic peptides that disrupt gram positive bacteria by interfering with cell wall and peptidoglycan synthesis. Also included is surfactin synthetase which activates and polymerizes the amino acids Leu, Glu, Asp, and Val to form the antibiotic surfactin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>The adenylation domain of nonribosomal peptide synthetases (NRPS), including Saframycin A gene cluster from Streptomyces lavendulae. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions. This family includes the saframycin A gene cluster from Streptomyces lavendulae which implicates the NRPS system for assembling the unusual tetrapeptidyl skeleton in an iterative manner. It also includes saframycin Mx1 produced by Myxococcus xanthus NRPS.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>similar to adenylation domain of chondramide synthase cmdD. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes phosphinothricin tripeptide (PTT, phosphinothricylalanylalanine) synthetase, where PTT is a natural-product antibiotic and potent herbicide that is produced by Streptomyces hygroscopicus. This adenylation domain has been confirmed to directly activate beta-tyrosine, and fluorinated chondramides are produced through precursor-directed biosynthesis. Also included in this family is chondramide synthase D (also known as ATP-dependent phenylalanine adenylase or phenylalanine activase or tyrosine activase). Chondramides A-D are depsipeptide antitumor and antifungal antibiotics produced by C. crocatus, are a class of mixed peptide/polyketide depsipeptides comprised of three amino acids (alanine, N-methyltryptophan, plus the unusual amino acid beta-tyrosine or alpha-methoxy-beta-tyrosine) and a polyketide chain ([E]-7-hydroxy-2,4,6-trimethyloct-4-enoic acid).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>similar to adenylation domain of plipastatin synthase (PpsD). This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) includes bacitracin synthetase 1 (BacA) in Bacillus licheniformis, tyrocidine synthetase in Brevibacillus brevis, plipastatin synthase (PpsD, an important antifungal protein) in Bacillus subtilis and mannopeptimycin peptide synthetase (MppB) in Streptomyces hygroscopicus. Plipastatin has strong fungitoxic activity and is involved in inhibition of phospholipase A2 and biofilm formation. Bacitracin, a mixture of related cyclic peptides, is used as a polypeptide antibiotic while function of tyrocidine is thought to be regulation of sporulation. MppB is involved in biosynthetic pathway of mannopeptimycin, a novel class of mannosylated lipoglycopeptides. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>Non-ribosomal peptide synthetase component F [Secondary metabolites biosynthesis, transport and catabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>gramicidin S synthase 2, also known as ATP-dependent proline adenylase. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) contains gramicidin S synthase 2 (also known as ATP-dependent proline adenylase or proline activase or ProA). ProA is a multifunctional enzyme involved in synthesis of the cyclic peptide antibiotic gramicidin S and able to activate and polymerize the amino acids proline, valine, ornithine and leucine. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>nonribosomal peptide synthase GliP-like. This family includes the adenylation (A) domain of nonribosomal peptide synthases (NRPS) gliotoxin biosynthesis protein P (GliP), thioclapurine biosynthesis protein P (tcpP) and Sirodesmin biosynthesis protein P (SirP). In the filamentous fungus Aspergillus fumigatus, NRPS GliP is involved in the biosynthesis of gliotoxin, which is initiated by the condensation of serine and phenylalanine. Studies show that GliP is not required for invasive aspergillosis, suggesting that the principal targets of gliotoxin are neutrophils or other phagocytes. SirP is a phytotoxin produced by the fungus Leptosphaeria maculans, which causes blackleg disease of canola (Brassica napus). In the fungus Claviceps purpurea, NRPS tcpP catalyzes condensation of tyrosine and glycine, part of biosynthesis of an unusual class of epipolythiodioxopiperazines (ETPs) that lacks the reactive thiol group for toxicity. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino) acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester bond to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>AMP-binding enzyme. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=co>44</td>
            <td id=al>myxochelin non-ribosomal peptide synthetase MxcG. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Acyl-CoA synthetase (AMP-forming)/AMP-acid ligase II [Lipid transport and metabolism, Secondary metabolites biosynthesis, transport and catabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-(5-amino-5-carboxypentanoyl)-L-cysteinyl-D-valine synthase. This family contains ACV synthetase (ACVS, EC 6.3.2.26; also known as N-(5-amino-5-carboxypentanoyl)-L-cysteinyl-D-valine synthase or delta-(L-alpha-aminoadipyl)-L-cysteinyl-D-valine synthetase) is involved in medically important antibiotic biosynthesis. ACV synthetase is active in an early step in the penicillin G biosynthesis pathway which involves the formation of the tripeptide 6-(L-alpha-aminoadipyl)-L-cysteinyl-D-valine (ACV); each of the constituent amino acids of the tripeptide ACV are activated as aminoacyl-adenylates with peptide bonds formed through the participation of amino acid thioester intermediates. ACV is then cyclized by the action of isopenicillin N synthase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>D-alanine--poly(phosphoribitol) ligase subunit DltA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Adenylate forming domain, Class I, also known as the ANL superfamily. This family is known as the ANL (acyl-CoA synthetases, the NRPS adenylation domains, and the Luciferase enzymes) superfamily. It includes acyl- and aryl-CoA ligases, as well as the adenylation domain of nonribosomal peptide synthetases and firefly luciferases.The adenylate-forming enzymes catalyze an ATP-dependent two-step reaction to first activate a carboxylate substrate as an adenylate and then transfer the carboxylate to the pantetheine group of either coenzyme A or an acyl-carrier protein. The active site of the domain is located at the interface of a large N-terminal subdomain and a smaller C-terminal subdomain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Acyl-coenzyme A synthetase/AMP-(fatty) acid ligase [Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>fatty acyl-CoA synthetase, including FadD13. This family contains fatty acyl-CoA synthetases, including Mycobacterium tuberculosis acid-induced operon MymA encoding the fatty acyl-CoA synthetase FadD13 which is essential for virulence and intracellular growth of the pathogen. The fatty acyl-CoA synthetase activates lipids before entering into the metabolic pathways and is also involved in transmembrane lipid transport. However, unlike soluble fatty acyl-CoA synthetases, but like the mammalian integral-membrane very-long-chain acyl-CoA synthetases, FadD13 accepts lipid substrates up to the maximum length of C26, and this is facilitated by an extensive hydrophobic tunnel from the active site to a positively charged patch. Also included is feruloyl-CoA synthetase (Fcs) in Rhodococcus strains where it is involved in biotechnological vanillin production from eugenol and ferulic acid via a non-beta-oxidative pathway.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase family member 4. This family of the adenylation (A) domain of nonribosomal peptide synthases (NRPS) contains acyl-CoA synthethase family member 4, also known as 2-aminoadipic 6-semialdehyde dehydrogenase or aminoadipate-semialdehyde dehydrogenase, most of which are uncharacterized. Acyl-CoA synthetase catalyzes the initial reaction in fatty acid metabolism, by forming a thioester with CoA. NRPSs are large multifunctional enzymes which synthesize many therapeutically useful peptides in bacteria and fungi via a template-directed, nucleic acid independent nonribosomal mechanism. These natural products include antibiotics, immunosuppressants, plant and animal toxins, and enzyme inhibitors. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Prokaryotic long-chain fatty acid CoA synthetases similar to Escherichia coli FadD. This subfamily of the AMP-forming adenylation family contains Escherichia coli FadD and similar prokaryotic fatty acid CoA synthetases. FadD was characterized as a long-chain fatty acid CoA synthetase. The gene fadD is regulated by the fatty acid regulatory protein FadR. Fatty acid CoA synthetase catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, followed by the formation of a fatty acyl-CoA. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Subfamily of fatty acid CoA ligase (FACL) similar to Fum10p of Gibberella moniliformis. FACL catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, followed by the formation of a fatty acyl-CoA. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions. Fum10p is a fatty acid CoA ligase involved in the synthesis of fumonisin, a polyketide mycotoxin, in Gibberella moniliformis.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Firefly luciferase of light emitting insects and 4-Coumarate-CoA Ligase (4CL). This family contains insect firefly luciferases that share significant sequence similarity to plant 4-coumarate:coenzyme A ligases, despite their functional diversity. Luciferase catalyzes the production of light in the presence of MgATP, molecular oxygen, and luciferin. In the first step, luciferin is activated by acylation of its carboxylate group with ATP, resulting in an enzyme-bound luciferyl adenylate. In the second step, luciferyl adenylate reacts with molecular oxygen, producing an enzyme-bound excited state product (Luc=O*) and releasing AMP. This excited-state product then decays to the ground state (Luc=O), emitting a quantum of visible light.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Cyclohexanecarboxylate-CoA ligase (also called cyclohex-1-ene-1-carboxylate:CoA ligase). Cyclohexanecarboxylate-CoA ligase activates the aliphatic ring compound, cyclohexanecarboxylate, for degradation. It catalyzes the synthesis of cyclohexanecarboxylate-CoA thioesters in a two-step reaction involving the formation of cyclohexanecarboxylate-AMP anhydride, followed by the nucleophilic substitution of AMP by CoA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Non-ribosomal peptide synthetase component E (peptide arylation enzyme) [Secondary metabolites biosynthesis, transport and catabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of medium-chain acyl-CoA synthetase (MACS). MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. MACS enzymes are localized to mitochondria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>medium-chain fatty-acid--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions. Members of this family include DitJ from Pseudomonas and similar proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>O-succinylbenzoic acid-CoA ligase. This family contains O-succinylbenzoyl-CoA (OSB-CoA) synthetase (also known as O-succinylbenzoic acid CoA ligase) that belongs to the ANL superfamily and catalyzes the ligation of CoA to o-succinylbenzoate (OSB). It includes MenE in the bacterial menaquinone biosynthesis pathway which is a promising target for the development of novel antibacterial agents. MenE catalyzes CoA ligation via an acyl-adenylate intermediate; tight-binding inhibitors of MenE based on stable acyl-sulfonyladenosine analogs of this intermediate provide a pathway toward the development of optimized MenE inhibitors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>2,3-dihydroxybenzoate-AMP ligase. 2,3-dihydroxybenzoate-AMP ligase activates 2,3-dihydroxybenzoate (DHB) by ligation of AMP from ATP with the release of pyrophosphate. However, it can also catalyze the ATP-PPi exchange for 2,3-DHB analogs, such as salicyclic acid (o-hydrobenzoate), as well as 2,4-DHB and 2,5-DHB, but with less efficiency. Proteins in this family are the stand-alone adenylation components of non-ribosomal peptide synthases (NRPSs) involved in the biosynthesis of siderophores, which are low molecular weight iron-chelating compounds synthesized by many bacteria to aid in the acquisition of this vital trace elements. In Escherichia coli, the 2,3-dihydroxybenzoate-AMP ligase is called EntE, the adenylation component of the enterobactin NRPS system.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Benzoate CoA ligase (BCL) and 4-Hydroxybenzoate-Coenzyme A Ligase (4-HBA-CoA ligase). Benzoate CoA ligase and 4-hydroxybenzoate-coenzyme A ligase catalyze the first activating step for benzoate and 4-hydroxybenzoate catabolic pathways, respectively. Although these two enzymes share very high sequence homology, they have their own substrate preference. The reaction proceeds via a two-step process; the first ATP-dependent step forms the substrate-AMP intermediate, while the second step forms the acyl-CoA ester, releasing the AMP. Aromatic compounds represent the second most abundant class of organic carbon compounds after carbohydrates. Some bacteria can use benzoic acid or benzenoid compounds as the sole source of carbon and energy through degradation. Benzoate CoA ligase and 4-hydroxybenzoate-Coenzyme A ligase are key enzymes of this process.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Malonyl-CoA synthetase (MCS). MCS catalyzes the formation of malonyl-CoA in a two-step reaction consisting of the adenylation of malonate with ATP, followed by malonyl transfer from malonyl-AMP to CoA. Malonic acid and its derivatives are the building blocks of polyketides and malonyl-CoA serves as the substrate of polyketide synthases. Malonyl-CoA synthetase has broad substrate tolerance and can activate a variety of malonyl acid derivatives. MCS may play an important role in biosynthesis of polyketides, the important secondary metabolites with therapeutic and agrochemical utility.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>fatty acid-CoA ligase VraA. This family of acyl-CoA ligases includes Bacillus subtilis YhfT, as well as long-chain fatty acid-CoA ligase VraA, all of which are as yet to be characterized. These proteins belong to the adenylate-forming enzymes which catalyze an ATP-dependent two-step reaction to first activate a carboxylate substrate as an adenylate and then transfer the carboxylate to the pantetheine group of either coenzyme A or an acyl-carrier protein. The active site of the domain is located at the interface of a large N-terminal subdomain and a smaller C-terminal subdomain</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Benzoate CoA ligase (BCL) and similar adenylate forming enzymes. This family contains benzoate CoA ligase (BCL) and related ligases that catalyze the acylation of benzoate derivatives, 2-aminobenzoate and 4-hydroxybenzoate. Aromatic compounds represent the second most abundant class of organic carbon compounds after carbohydrates. Xenobiotic aromatic compounds are also a major class of man-made pollutants. Some bacteria use benzoate as the sole source of carbon and energy through benzoate degradation. Benzoate degradation starts with its activation to benzoyl-CoA by benzoate CoA ligase. The reaction catalyzed by benzoate CoA ligase proceeds via a two-step process; the first ATP-dependent step forms an acyl-AMP intermediate, and the second step forms the acyl-CoA ester with release of the AMP.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>fatty acid--CoA ligase family protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of Acetyl-CoA synthetase like family (ACS). This family is most similar to acetyl-CoA synthetase. Acetyl-CoA synthetase (ACS) catalyzes the formation of acetyl-CoA from acetate, CoA, and ATP. Synthesis of acetyl-CoA is carried out in a two-step reaction. In the first step, the enzyme catalyzes the synthesis of acetyl-AMP intermediate from acetate and ATP. In the second step, acetyl-AMP reacts with CoA to produce acetyl-CoA. This enzyme is only present in bacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Thioesterase domain of type I polyketide synthase or non-ribosomal peptide synthetase [Secondary metabolites biosynthesis, transport and catabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>o-succinylbenzoate--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Medium-chain acyl-CoA synthetase (MACS or ACSM). MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. The acyl-CoA is a key intermediate in many important biosynthetic and catabolic processes.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>adenylate forming domain, fatty acid CoA ligase (FadD10). This family contains long chain fatty acid CoA ligases, including FadD10 which is involved in the synthesis of a virulence-related lipopeptide. FadD10 is a fatty acyl-AMP ligase (FAAL) that transfers fatty acids to an acyl carrier protein. Structures of FadD10 in apo- and complexed form with dodecanoyl-AMP, show a novel open conformation, facilitated by its unique inter-domain and intermolecular interactions, which is critical for the enzyme to carry out the acyl transfer onto the acyl carrier protein (Rv0100) rather than coenzyme A.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>O-succinylbenzoate-CoA ligase (also known as O-succinylbenzoate-CoA synthase, OSB-CoA synthetase, or MenE). O-succinylbenzoic acid-CoA synthase catalyzes the coenzyme A (CoA)- and ATP-dependent conversion of o-succinylbenzoic acid to o-succinylbenzoyl-CoA. The reaction is the fourth step of the biosynthesis pathway of menaquinone (vitamin K2). In certain bacteria, menaquinone is used during fumarate reduction in anaerobic respiration. In cyanobacteria, the product of the menaquinone pathway is phylloquinone (2-methyl-3-phytyl-1,4-naphthoquinone), a molecule used exclusively as an electron transfer cofactor in Photosystem 1. In green sulfur bacteria and heliobacteria, menaquinones are used as loosely bound secondary electron acceptors in the photosynthetic reaction center.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>acetate-CoA ligase. This family includes acyl- and aryl-CoA ligases, as well as the adenylation domain of nonribosomal peptide synthetases and firefly luciferases. The adenylate-forming enzymes catalyze an ATP-dependent two-step reaction to first activate a carboxylate substrate as an adenylate and then transfer the carboxylate to the pantetheine group of either coenzyme A or an acyl-carrier protein. The active site of the domain is located at the interface of a large N-terminal subdomain and a smaller C-terminal subdomain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cyclohexanecarboxylate-CoA ligase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>Fatty acyl-CoA synthetases similar to LC-FACS from Thermus thermophiles. This family includes fatty acyl-CoA synthetases that can activate medium-chain to long-chain fatty acids. They catalyze the ATP-dependent acylation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. The fatty acyl-CoA synthetases are responsible for fatty acid degradation as well as physiological regulation of cellular functions via the production of fatty acyl-CoA esters. The fatty acyl-CoA synthetase from Thermus thermophiles in this family catalyzes the long-chain fatty acid, myristoyl acid, while another member in this family, the AlkK protein identified from Pseudomonas oleovorans, targets medium chain fatty acids. This family also includes uncharacterized FACS proteins.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>crotonobetaine/carnitine-CoA ligase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>insect luciferase, similar to plant 4-coumarate: CoA ligases. This family contains insect firefly luciferases that share significant sequence similarity to plant 4-coumarate:coenzyme A ligases, despite their functional diversity. Luciferase catalyzes the production of light in the presence of MgATP, molecular oxygen, and luciferin. In the first step, luciferin is activated by acylation of its carboxylate group with ATP, resulting in an enzyme-bound luciferyl adenylate. In the second step, luciferyl adenylate reacts with molecular oxygen, producing an enzyme-bound excited state product (Luc=O*) and releasing AMP. This excited-state product then decays to the ground state (Luc=O), emitting a quantum of visible light.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>acyl-CoA synthetase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>4-Chlorobenzoate-CoA ligase (CBAL). CBAL catalyzes the conversion of 4-chlorobenzoate (4-CB) to 4-chlorobenzoyl-coenzyme A (4-CB-CoA) by the two-step adenylation and thioester-forming reactions. 4-Chlorobenzoate (4-CBA) is an environmental pollutant derived from microbial breakdown of aromatic pollutants, such as polychlorinated biphenyls (PCBs), DDT, and certain herbicides. The 4-CBA degrading pathway converts 4-CBA to the metabolite 4-hydroxybezoate (4-HBA), allowing some soil-dwelling microbes to utilize 4-CBA as an alternate carbon source. This pathway consists of three chemical steps catalyzed by 4-CBA-CoA ligase, 4-CBA-CoA dehalogenase, and 4HBA-CoA thioesterase in sequential reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>The adenylation domain (A domain) of a family of nonribosomal peptide synthetases (NRPSs) synthesizing toxins and antitumor agents. The adenylation (A) domain of NRPS recognizes a specific amino acid or hydroxy acid and activates it as an (amino)-acyl adenylate by hydrolysis of ATP. The activated acyl moiety then forms a thioester to the enzyme-bound cofactor phosphopantetheine of a peptidyl carrier protein domain. This family includes NRPSs that synthesize toxins and antitumor agents; for example, TubE for Tubulysine, CrpA for cryptophycin, TdiA for terrequinone A, KtzG for kutzneride, and Vlm1/Vlm2 for Valinomycin. Nonribosomal peptide synthetases are large multifunctional enzymes which synthesize many therapeutically useful peptides. NRPS has a distinct modular structure in which each module is responsible for the recognition, activation, and, in some cases, modification of a single amino acid residue of the final peptide product. The modules can be subdivided into domains that catalyze specific biochemical reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of medium-chain acyl-CoA synthetase (MACS). MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. MACS enzymes are localized to mitochondria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>2-aminobenzoate-CoA ligase (ABCL). ABCL catalyzes the initial step in the 2-aminobenzoate aerobic degradation pathway by activating 2-aminobenzoate to 2-aminobenzoyl-CoA. The reaction is carried out via a two-step process; the first step is ATP-dependent and forms a 2-aminobenzoyl-AMP intermediate, and the second step forms the 2-aminobenzoyl-CoA ester and releases the AMP. 2-Aminobenzoyl-CoA is further converted to 2-amino-5-oxo-cyclohex-1-ene-1-carbonyl-CoA catalyzed by 2-aminobenzoyl-CoA monooxygenase/reductase. ABCL has been purified from cells aerobically grown with 2-aminobenzoate as sole carbon, energy, and nitrogen source, and has been characterized as a monomer.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>o-succinylbenzoate--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>78</td>
            <td id=al>C-terminal domain of the acyl-acyl carrier protein synthetase (also called 2-acylglycerophosphoethanolamine acyltransferase, Aas). Acyl-acyl carrier protein synthase (Aas) is a membrane protein responsible for a minor pathway of incorporating exogenous fatty acids into membrane phospholipids. Its in vitro activity is characterized by the ligation of free fatty acids between 8 and 18 carbons in length to the acyl carrier protein sulfydryl group (ACP-SH) in the presence of ATP and Mg2+. However, its in vivo function is as a 2-acylglycerophosphoethanolamine (2-acyl-GPE) acyltransferase. The reaction occurs in two steps: the acyl chain is first esterified to acyl carrier protein (ACP) via a thioester bond, followed by a second step where the acyl chain is transferred to a 2-acyllysophospholipid, thus completing the transacylation reaction. This model represents the C-terminal domain of the enzyme, which belongs to the class I adenylate-forming enzyme family, including acyl-CoA synthetases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>4-Coumarate-CoA Ligase (4CL). 4-Coumarate:coenzyme A ligase is a key enzyme in the phenylpropanoid metabolic pathway for monolignol and flavonoid biosynthesis. It catalyzes the synthesis of hydroxycinnamate-CoA thioesters in a two-step reaction, involving the formation of hydroxycinnamate-AMP anhydride and the nucleophilic substitution of AMP by CoA. The phenylpropanoid pathway is one of the most important secondary metabolism pathways in plants and hydroxycinnamate-CoA thioesters are the precursors of lignin and other important phenylpropanoids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>o-succinylbenzoate-CoA ligase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Long-chain fatty acid CoA synthetases and Bubblegum-like very long-chain fatty acid CoA synthetases. This family includes long-chain fatty acid (C12-C20) CoA synthetases and Bubblegum-like very long-chain (>C20) fatty acid CoA synthetases. FACS catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, and the formation of a fatty acyl-CoA. Eukaryotes generally have multiple isoforms of LC-FACS genes with multiple splice variants. For example, nine genes are found in Arabidopsis and six genes are expressed in mammalian cells. Drosophila melanogaster mutant bubblegum (BGM) have elevated levels of very-long-chain fatty acids (VLCFA) caused by a defective gene later named bubblegum. The human homolog (hsBG) of bubblegum has been characterized as a very long chain fatty acid CoA synthetase that functions specifically in the brain; hsBG may play a central role in brain VLCFA metabolism and myelinogenesis. Free fatty acids must be "activated" to their CoA thioesters before participating in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>O-succinylbenzoic acid--CoA ligase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acetyl-CoA synthetase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Putative long-chain fatty acid CoA ligase. The members of this family are putative long-chain fatty acyl-CoA synthetases, which catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. Fatty acyl-CoA synthetases are responsible for fatty acid degradation as well as physiological regulation of cellular functions via the production of fatty acyl-CoA esters.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase FadD3 and similar proteins. This family contains long chain fatty acid CoA ligases, including FadD3 which is an acyl-CoA synthetase that initiates catabolism of cholesterol rings C and D in actinobacteria. The cholesterol catabolic pathway occurs in most mycolic acid-containing actinobacteria, such as Rhodococcus jostii RHA1, and is critical for Mycobacterium tuberculosis (Mtb) during infection. FadD3 catalyzes the ATP-dependent CoA thioesterification of 3a-alpha-H-4alpha(3'-propanoate)-7a-beta-methylhexahydro-1,5-indanedione (HIP) to yield HIP-CoA. Hydroxylated analogs of HIP, 5alpha-OH HIP and 1beta-OH HIP, can also be used.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>Long-chain acyl-CoA synthetase (AMP-forming)  [Lipid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>35</td>
            <td id=al>2-acyl-glycerophospho-ethanolamine acyltransferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>acyl-CoA synthetase (AMP-forming)/AMP-acid ligase II. This family contains fatty acid CoA ligases, including acyl-CoA synthetase (AMP-forming)/AMP-acid ligase II, most of which are yet to be characterized, but may be similar to Carnitine-CoA ligase (CaiC) which catalyzes the transfer of CoA to carnitine. Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of bifunctional fatty acid transporter/very-long-chain acyl-CoA synthetase in fungi. Fatty acid transport protein (FATP) transports long-chain or very-long-chain fatty acids across the plasma membrane. FATPs also have fatty acid CoA synthetase activity, thus playing dual roles as fatty acid transporters and its activation enzymes. FATPs are the key players in the trafficking of exogenous fatty acids into the cell and in intracellular fatty acid homeostasis. Members of this family are fungal FATPs, including FAT1 from Cochliobolus heterostrophus.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>putative crotonobetaine/carnitine-CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>4-coumarate--CoA ligase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>p-hydroxybenzoic acid--AMP ligase FadD22. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Medium-chain acyl-CoA synthetase (MACS) of AAE_MA like. MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This family of MACS enzymes is found in archaea and bacteria. It is represented by the acyl-adenylating enzyme from Methanosarcina acetivorans (AAE_MA). AAE_MA is most active with propionate, butyrate, and the branched analogs: 2-methyl-propionate, butyrate, and pentanoate. The specific activity is weaker for smaller or larger acids.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>(2,3-dihydroxybenzoyl)adenylate synthase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>4-coumarate--CoA ligase-like</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). The members of this family are bacterial long-chain fatty acid CoA synthetase, most of which are as yet uncharacterized. LC-FACS catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, and the formation of a fatty acyl-CoA. Free fatty acids must be "activated" to their CoA thioesters before participating in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>63</td>
            <td id=al>Propionyl-CoA synthetase (PrpE). EC 6.2.1.17: propanoate:CoA ligase (AMP-forming) or propionate#CoA ligase (PrpE) catalyzes the first step of the 2-methylcitric acid cycle for propionate catabolism. It activates propionate to propionyl-CoA in a two-step reaction, which proceeds through a propionyl-AMP intermediate and requires ATP and Mg2+. In Salmonella enterica, the PrpE protein is required for growth of Salmonella enterica on propionate and can substitute for the acetyl-CoA synthetase (Acs) enzyme during growth on acetate. PrpE can also activate acetate, 3HP, and butyrate to their corresponding CoA-thioesters, although with less efficiency.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>fatty acid--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Thioesterase domain. Peptide synthetases are involved in the non-ribosomal synthesis of peptide antibiotics. Next to the operons encoding these enzymes, in almost all cases, are genes that encode proteins that have similarity to the type II fatty acid thioesterases of vertebrates. There are also modules within the peptide synthetases that also share this similarity. With respect to antibiotic production, thioesterases are required for the addition of the last amino acid to the peptide antibiotic, thereby forming a cyclic antibiotic. Thioesterases (non-integrated) have molecular masses of 25-29 kDa.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Fatty acyl-AMP ligase (FAAL). FAAL belongs to the class I adenylate forming enzyme family and is homologous to fatty acyl-coenzyme A (CoA) ligases (FACLs). However, FAALs produce only the acyl adenylate and are unable to perform the thioester-forming reaction, while FACLs perform a two-step catalytic reaction; AMP ligation followed by CoA ligation using ATP and CoA as cofactors. FAALs have insertion motifs between the N-terminal and C-terminal subdomains that distinguish them from the FACLs. This insertion motif precludes the binding of CoA, thus preventing CoA ligation. It has been suggested that the acyl adenylates serve as substrates for multifunctional polyketide synthases to permit synthesis of complex lipids such as phthiocerol dimycocerosate, sulfolipids, mycolic acids, and mycobactin.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>Bacterial Bile acid CoA ligases and similar proteins. Bile acid-Coenzyme A ligase catalyzes the formation of bile acid-CoA conjugates in a two-step reaction: the formation of a bile acid-AMP molecule as an intermediate, followed by the formation of a bile acid-CoA. This ligase requires a bile acid with a free carboxyl group, ATP, Mg2+, and CoA for synthesis of the final bile acid-CoA conjugate. The bile acid-CoA ligation is believed to be the initial step in the bile acid 7alpha-dehydroxylation pathway in the intestinal bacterium Eubacterium sp.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>o-succinylbenzoate--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Long-chain fatty acid CoA synthetase. This family includes long-chain fatty acid (C12-C20) CoA synthetases, including an Arabidopsis gene At4g14070 that plays a role in activation and elongation of exogenous fatty acids. FACS catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, and the formation of a fatty acyl-CoA. Eukaryotes generally have multiple isoforms of LC-FACS genes with multiple splice variants. For example, nine genes are found in Arabidopsis and six genes are expressed in mammalian cells. Free fatty acids must be "activated" to their CoA thioesters before participating in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cr>27</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>Eukaryotic Medium-chain acyl-CoA synthetase (MACS or ACSM). MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. The acyl-CoA is a key intermediate in many important biosynthetic and catabolic processes. MACS enzymes are localized to mitochondria. Two murine MACS family proteins are found in liver and kidney. In rodents, a MACS member is detected particularly in the olfactory epithelium and is called O-MACS. O-MACS demonstrates substrate preference for the fatty acid lengths of C6-C12.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>long chain fatty acid--[acyl-carrier-protein] ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Phosphopantetheine attachment site. A 4'-phosphopantetheine prosthetic group is attached through a serine. This prosthetic group acts as a a 'swinging arm' for the attachment of activated fatty acid and amino-acid groups. This domain forms a four helix bundle. This family includes members not included in Prosite. The inclusion of these members is supported by sequence analysis and functional evidence. The related domain of the anguibactin system regulator AngR has the attachment serine replaced by an alanine.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>acyl-CoA synthetase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>33</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Phosphopantetheine attachment site. Phosphopantetheine (or pantetheine 4' phosphate) is the prosthetic group of acyl carrier proteins (ACP) in some multienzyme complexes where it serves as a 'swinging arm' for the attachment of activated fatty acid and amino-acid groups.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>69</td>
            <td id=al>long-chain fatty acid--CoA ligase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cr>38</td>
            <td id=al>long-chain-fatty-acid--CoA ligase FadD. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>Bacterial long-chain fatty acid CoA synthetase (LC-FACS), including Marinobacter hydrocarbonoclasticus isoprenoid Coenzyme A synthetase. The members of this family are bacterial long-chain fatty acid CoA synthetase. Marinobacter hydrocarbonoclasticus isoprenoid Coenzyme A synthetase in this family is involved in the synthesis of isoprenoid wax ester storage compounds when grown on phytol as the sole carbon source. LC-FACS catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, and the formation of a fatty acyl-CoA. Free fatty acids must be "activated" to their CoA thioesters before participating in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>long-chain-fatty-acid--CoA ligase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>O-succinylbenzoic acid--CoA ligase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>4-coumarate--CoA ligase-like 1</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cr>38</td>
            <td id=al>Long-chain-fatty-acid--CoA ligase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of medium-chain acyl-CoA synthetase (MACS). MACS catalyzes the two-step activation of medium chain fatty acids (containing 4-12 carbons). The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. MACS enzymes are localized to mitochondria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cr>37</td>
            <td id=al>Eukaryotic long-chain fatty acid CoA synthetase (LC-FACS), including fungal proteins. The members of this family are eukaryotic fatty acid CoA synthetases (EC 6.2.1.3) that activate fatty acids with chain lengths of 12 to 20 and includes fungal proteins. They act on a wide range of long-chain saturated and unsaturated fatty acids, but the enzymes from different tissues show some variation in specificity. LC-FACS catalyzes the formation of fatty acyl-CoA in a two-step reaction: the formation of a fatty acyl-AMP molecule as an intermediate, and the formation of a fatty acyl-CoA. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions. Organisms tend to have multiple isoforms of LC-FACS genes with multiple splice variants. For example, nine genes are found in Arabidopsis and six genes are expressed in mammalian cells. In Schizosaccharomyces pombe, lcf1 gene encodes a new fatty acyl-CoA synthetase that preferentially recognizes myristic acid as a substrate.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Uncharacterized subfamily of fatty acid CoA ligase (FACL). Fatty acyl-CoA ligases catalyze the ATP-dependent activation of fatty acids in a two-step reaction. The carboxylate substrate first reacts with ATP to form an acyl-adenylate intermediate, which then reacts with CoA to produce an acyl-CoA ester. This is a required step before free fatty acids can participate in most catabolic and anabolic reactions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>AMP-binding enzyme C-terminal domain. This is a small domain that is found C terminal to pfam00501. It has a central beta sheet core that is flanked by alpha helices.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3904699</td></tr>
<tr><td id=al>end</td><td id=al>3907411</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al>[secondary-metabolites]</td></tr>
<tr><td id=al>descr</td><td id=al>amino acid adenylation domain-containing protein, partial</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al>(region 6) NRPS-like cds((PP-binding or NAD_binding_4) and (AMP-binding or A-OX)) biosynthetic-additional (smcogs) SMCOG1002:AMP-dependent synthetase and ligase (Score: 276.9; E-value: 4.4e-84) biosynthetic</td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
