<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03457</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03457 RNA-binding protein [Phormidium_A]</h3>
    <div>MSIRLYVGNLPKDVERDELQDFFQDAGELLSVKVITDRKTGKCRGFGFVTAKNDEQADQIVEAFNGKEFKETALKVEKAMPRKKGKEKSDSAPAAKSSPSGSRSSRSSSSSRSSSSDSAQPDPRWADELSKLKQMLASQTTNS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding proteins (RRM domain)</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>hypothetical protein</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>59</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cr>JACYMB01</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TPA: RNA-binding protein</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>RNA-binding protein</td>
            <td id=cr>RECH01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>50</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd21608 RNA recognition motif 2 (RRM2) found in Nicotiana sylvestris chloroplastic 33 kDa ribonucleoprotein (NsCP33) and similar proteins. The family includes NsCP33, Arabidopsis thaliana chloroplastic 31 kDa ribonucleoprotein (CP31A) and mitochondrial glycine-rich RNA-binding protein 2 (AtGR-RBP2). NsCP33 may be involved in splicing and/or processing of chloroplast RNA's. AtCP31A, also called RNA-binding protein 1/2/3 (AtRBP33), or RNA-binding protein CP31A, or RNA-binding protein RNP-T, or RNA-binding protein cp31, is required for specific RNA editing events in chloroplasts and stabilizes specific chloroplast mRNAs, as well as for normal chloroplast development under cold stress conditions by stabilizing transcripts of numerous mRNAs under these conditions. CP31A may modulate telomere replication through RNA binding domains. AtGR-RBP2, also called AtRBG2, or glycine-rich protein 2 (AtGRP2), or mitochondrial RNA-binding protein 1a (At-mRBP1a), plays a role in RNA transcription or processing during stress. It binds RNAs and DNAs sequence with a preference to single-stranded nucleic acids. AtGR-RBP2 displays strong affinity to poly(U) sequence. It exerts cold and freezing tolerance, probably by exhibiting an RNA chaperone activity during the cold and freezing adaptation process. Some members in this family contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the second RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>smart00360 RNA recognition motif. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=al>cd21609 RNA recognition motif 1 (RRM1) found in chloroplastic plastid-specific 30S ribosomal protein 2 (PSRP-2) and similar proteins. PSRP-2, also called chloroplastic 30S ribosomal protein 2, or chloroplastic small ribosomal subunit protein cS22, is a component of the chloroplast ribosome (chloro-ribosome), a dedicated translation machinery responsible for the synthesis of chloroplast genome-encoded proteins, including proteins of the transcription and translation machinery and components of the photosynthetic apparatus. It binds single strand DNA (ssDNA) and RNA in vitro. It exhibits RNA chaperone activity and regulates negatively resistance responses to abiotic stresses during seed germination (e.g. salt, dehydration, and low temperature) and seedling growth (e.g. salt). The family also includes Nicotiana sylvestris chloroplastic 33 kDa ribonucleoprotein (NsCP33) and Arabidopsis thaliana chloroplastic 31 kDa ribonucleoprotein (AtCP31A). NsCP33 may be involved in splicing and/or processing of chloroplast RNA's. AtCP31A, also called RNA-binding protein 1/2/3 (AtRBP33), or RNA-binding protein CP31A, or RNA-binding protein RNP-T, or RNA-binding protein cp31, is required for specific RNA editing events in chloroplasts and stabilizes specific chloroplast mRNAs, as well as for normal chloroplast development under cold stress conditions by stabilizing transcripts of numerous mRNAs under these conditions. CP31A may modulate telomere replication through RNA binding domains. Members in this family contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>cd00590 RNA recognition motif (RRM) superfamily. RRM, also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), is a highly abundant domain in eukaryotes found in proteins involved in post-transcriptional gene expression processes including mRNA and rRNA processing, RNA export, and RNA stability. This domain is 90 amino acids in length and consists of a four-stranded beta-sheet packed against two alpha-helices. RRM usually interacts with ssRNA, but is also known to interact with ssDNA as well as proteins. RRM binds a variable number of nucleotides, ranging from two to eight. The active site includes three aromatic side-chains located within the conserved RNP1 and RNP2 motifs of the domain. The RRM domain is found in a variety heterogeneous nuclear ribonucleoproteins (hnRNPs), proteins implicated in regulation of alternative splicing, and protein components of small nuclear ribonucleoproteins (snRNPs).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cb>63</td>
            <td id=cr>30</td>
            <td id=al>COG0724 RNA recognition motif (RRM) domain [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=al>pfam00076 RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain). The RRM motif is probably diagnostic of an RNA binding protein. RRMs are found in a variety of RNA binding proteins, including various hnRNP proteins, proteins implicated in regulation of alternative splicing, and protein components of snRNPs. The motif also appears in a few single stranded DNA binding proteins. The RRM structure consists of four strands and two helices arranged in an alpha/beta sandwich, with a third helix present during RNA binding in some cases The C-terminal beta strand (4th strand) and final helix are hard to align and have been omitted in the SEED alignment The LA proteins have an N terminal rrm which is included in the seed. There is a second region towards the C-terminus that has some features characteristic of a rrm but does not appear to have the important structural core of a rrm. The LA proteins are one of the main autoantigens in Systemic lupus erythematosus (SLE), an autoimmune disease.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>52</td>
            <td id=cg>97</td>
            <td id=al>cd12398 RNA recognition motif (RRM) found in cleavage stimulation factor subunit 2 (CSTF2), yeast ortholog mRNA 3'-end-processing protein RNA15 and similar proteins. This subfamily corresponds to the RRM domain of CSTF2, its tau variant and eukaryotic homologs. CSTF2, also termed cleavage stimulation factor 64 kDa subunit (CstF64), is the vertebrate conterpart of yeast mRNA 3'-end-processing protein RNA15. It is expressed in all somatic tissues and is one of three cleavage stimulatory factor (CstF) subunits required for polyadenylation. CstF64 contains an N-terminal RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), a CstF77-binding domain, a repeated MEARA helical region and a conserved C-terminal domain reported to bind the transcription factor PC-4. During polyadenylation, CstF interacts with the pre-mRNA through the RRM of CstF64 at U- or GU-rich sequences within 10 to 30 nucleotides downstream of the cleavage site. CSTF2T, also termed tauCstF64, is a paralog of the X-linked cleavage stimulation factor CstF64 protein that supports polyadenylation in most somatic cells. It is expressed during meiosis and subsequent haploid differentiation in a more limited set of tissues and cell types, largely in meiotic and postmeiotic male germ cells, and to a lesser extent in brain. The loss of CSTF2T will cause male infertility, as it is necessary for spermatogenesis and fertilization. Moreover, CSTF2T is required for expression of genes involved in morphological differentiation of spermatids, as well as for genes having products that function during interaction of motile spermatozoa with eggs. It promotes germ cell-specific patterns of polyadenylation by using its RRM to bind to different sequence elements downstream of polyadenylation sites than does CstF64. The family also includes yeast ortholog mRNA 3'-end-processing protein RNA15 and similar proteins. RNA15 is a core subunit of cleavage factor IA (CFIA), an essential transcriptional 3'-end processing factor from Saccharomyces cerevisiae. RNA recognition by CFIA is mediated by an N-terminal RRM, which is contained in the RNA15 subunit of the complex. The RRM of RNA15 has a strong preference for GU-rich RNAs, mediated by a binding pocket that is entirely conserved in both yeast and vertebrate RNA15 orthologs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>cd12362 RNA recognition motif 3 (RRM3) found in CELF/Bruno-like family of RNA binding proteins CELF1, CELF2, CELF3, CELF4, CELF5, CELF6 and similar proteins. This subgroup corresponds to the RRM3 of the CUGBP1 and ETR-3-like factors (CELF) or BRUNOL (Bruno-like) proteins, a family of structurally related RNA-binding proteins involved in the regulation of pre-mRNA splicing in the nucleus and in the control of mRNA translation and deadenylation in the cytoplasm. The family contains six members: CELF-1 (also termed BRUNOL-2, or CUG-BP1, or NAPOR, or EDEN-BP), CELF-2 (also termed BRUNOL-3, or ETR-3, or CUG-BP2, or NAPOR-2), CELF-3 (also termed BRUNOL-1, or TNRC4, or ETR-1, or CAGH4, or ER DA4), CELF-4 (also termed BRUNOL-4), CELF-5 (also termed BRUNOL-5), CELF-6 (also termed BRUNOL-6). They all contain three highly conserved RNA recognition motifs (RRMs), also known as RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains): two consecutive RRMs (RRM1 and RRM2) situated in the N-terminal region followed by a linker region and the third RRM (RRM3) close to the C-terminus of the protein. The low sequence conservation of the linker region is highly suggestive of a large variety in the co-factors that associate with the various CELF family members. Based on both sequence similarity and function, the CELF family can be divided into two subfamilies, the first containing CELFs 1 and 2, and the second containing CELFs 3, 4, 5, and 6. The different CELF proteins may act through different sites on at least some substrates. Furthermore, CELF proteins may interact with each other in varying combinations to influence alternative splicing in different contexts. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=al>cd12408 RNA recognition motif (RRM) found in eukaryotic translation initiation factor 3 subunit G (eIF-3G) and similar proteins. This subfamily corresponds to the RRM of eIF-3G and similar proteins. eIF-3G, also termed eIF-3 subunit 4, or eIF-3-delta, or eIF3-p42, or eIF3-p44, is the RNA-binding subunit of eIF3, a large multisubunit complex that plays a central role in the initiation of translation by binding to the 40 S ribosomal subunit and promoting the binding of methionyl-tRNAi and mRNA. eIF-3G binds 18 S rRNA and beta-globin mRNA, and therefore appears to be a nonspecific RNA-binding protein. eIF-3G is one of the cytosolic targets and interacts with mature apoptosis-inducing factor (AIF). eIF-3G contains one RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). This family also includes yeast eIF3-p33, a homolog of vertebrate eIF-3G, plays an important role in the initiation phase of protein synthesis in yeast. It binds both, mRNA and rRNA, fragments due to an RRM near its C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=co>52</td>
            <td id=cg>95</td>
            <td id=al>cd12380 RNA recognition motif 3 (RRM3) found found in type I polyadenylate-binding proteins. This subfamily corresponds to the RRM3 of type I poly(A)-binding proteins (PABPs), highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in the regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. The family represents type I polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 1 (PABP-1 or PABPC1), polyadenylate-binding protein 3 (PABP-3 or PABPC3), polyadenylate-binding protein 4 (PABP-4 or APP-1 or iPABP), polyadenylate-binding protein 5 (PABP-5 or PABPC5), polyadenylate-binding protein 1-like (PABP-1-like or PABPC1L), polyadenylate-binding protein 1-like 2 (PABPC1L2 or RBM32), polyadenylate-binding protein 4-like (PABP-4-like or PABPC4L), yeast polyadenylate-binding protein, cytoplasmic and nuclear (PABP or ACBP-67), and similar proteins. PABP-1 is an ubiquitously expressed multifunctional protein that may play a role in 3' end formation of mRNA, translation initiation, mRNA stabilization, protection of poly(A) from nuclease activity, mRNA deadenylation, inhibition of mRNA decapping, and mRNP maturation. Although PABP-1 is thought to be a cytoplasmic protein, it is also found in the nucleus. PABP-1 may be involved in nucleocytoplasmic trafficking and utilization of mRNP particles. PABP-1 contains four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), a less well conserved linker region, and a proline-rich C-terminal conserved domain (CTD). PABP-3 is a testis-specific poly(A)-binding protein specifically expressed in round spermatids. It is mainly found in mammalian and may play an important role in the testis-specific regulation of mRNA homeostasis. PABP-3 shows significant sequence similarity to PABP-1. However, it binds to poly(A) with a lower affinity than PABP-1. PABP-1 possesses an A-rich sequence in its 5'-UTR and allows binding of PABP and blockage of translation of its own mRNA. In contrast, PABP-3 lacks the A-rich sequence in its 5'-UTR. PABP-4 is an inducible poly(A)-binding protein (iPABP) that is primarily localized to the cytoplasm. It shows significant sequence similarity to PABP-1 as well. The RNA binding properties of PABP-1 and PABP-4 appear to be identical. PABP-5 is encoded by PABPC5 gene within the X-specific subinterval, and expressed in fetal brain and in a range of adult tissues in mammalian, such as ovary and testis. It may play an important role in germ cell development. Moreover, unlike other PABPs, PABP-5 contains only four RRMs, but lacks both the linker region and the CTD. PABP-1-like and PABP-1-like 2 are the orthologs of PABP-1. PABP-4-like is the ortholog of PABP-5. Their cellular functions remain unclear. The family also includes the yeast PABP, a conserved poly(A) binding protein containing poly(A) tails that can be attached to the 3'-ends of mRNAs. The yeast PABP and its homologs may play important roles in the initiation of translation and in mRNA decay. Like vertebrate PABP-1, the yeast PABP contains four RRMs, a linker region, and a proline-rich CTD as well. The first two RRMs are mainly responsible for specific binding to poly(A). The proline-rich region may be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>50</td>
            <td id=cg>98</td>
            <td id=al>cd12395 RNA recognition motif 2 (RRM2) found in RNA-binding protein 34 (RBM34) and similar proteins. This subfamily corresponds to the RRM2 of RBM34, a putative RNA-binding protein containing two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). Although the function of RBM34 remains unclear currently, its RRM domains may participate in mRNA processing. RBM34 may act as an mRNA processing-related protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>54</td>
            <td id=cg>92</td>
            <td id=al>cd12671 RNA recognition motif (RRM) found in cleavage stimulation factor subunit 2 (CSTF2), cleavage stimulation factor subunit 2 tau variant (CSTF2T) and similar proteins. This subgroup corresponds to the RRM domain of CSTF2, its tau variant and eukaryotic homologs. CSTF2, also termed cleavage stimulation factor 64 kDa subunit (CstF64), is the vertebrate conterpart of yeast mRNA 3'-end-processing protein RNA15. It is expressed in all somatic tissues and is one of three cleavage stimulatory factor (CstF) subunits required for polyadenylation. CstF64 contains an N-terminal RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), a CstF77-binding domain, a repeated MEARA helical region and a conserved C-terminal domain reported to bind the transcription factor PC-4. During polyadenylation, CstF interacts with the pre-mRNA through the RRM of CstF64 at U- or GU-rich sequences within 10 to 30 nucleotides downstream of the cleavage site. CSTF2T, also termed tauCstF64, is a paralog of the X-linked cleavage stimulation factor CstF64 protein that supports polyadenylation in most somatic cells. It is expressed during meiosis and subsequent haploid differentiation in a more limited set of tissues and cell types, largely in meiotic and postmeiotic male germ cells, and to a lesser extent in brain. The loss of CSTF2T will cause male infertility, as it is necessary for spermatogenesis and fertilization. Moreover, CSTF2T is required for expression of genes involved in morphological differentiation of spermatids, as well as for genes having products that function during interaction of motile spermatozoa with eggs. It promotes germ cell-specific patterns of polyadenylation by using its RRM to bind to different sequence elements downstream of polyadenylation sites than does CstF64. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12399 RNA recognition motif (RRM) found in Helicobacter pylori HP0827 protein and similar proteins. This subfamily corresponds to the RRM of H. pylori HP0827, a putative ssDNA-binding protein 12rnp2 precursor, containing one RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). The ssDNA binding may be important in activation of HP0827. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=co>54</td>
            <td id=cg>98</td>
            <td id=al>cd12413 RNA recognition motif 1 (RRM1) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM1 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=co>40</td>
            <td id=cb>79</td>
            <td id=al>cd12328 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A subfamily. This subfamily corresponds to the RRM2 of hnRNP A0, hnRNP A1, hnRNP A2/B1, hnRNP A3 and similar proteins. hnRNP A0 is a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. It has been identified as the substrate for MAPKAP-K2 and may be involved in the lipopolysaccharide (LPS)-induced post-transcriptional regulation of tumor necrosis factor-alpha (TNF-alpha), cyclooxygenase 2 (COX-2) and macrophage inflammatory protein 2 (MIP-2). hnRNP A1 is an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A2/B1 is an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). Many mRNAs, such as myelin basic protein (MBP), myelin-associated oligodendrocytic basic protein (MOBP), carboxyanhydrase II (CAII), microtubule-associated protein tau, and amyloid precursor protein (APP) are trafficked by hnRNP A2/B1. hnRNP A3 is also a RNA trafficking response element-binding protein that participates in the trafficking of A2RE-containing RNA. The hnRNP A subfamily is characterized by two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cr>36</td>
            <td id=cb>70</td>
            <td id=al>cd12400 RNA recognition motif (RRM) found in Saccharomyces cerevisiae nucleolar protein 6 (Nop6) and similar proteins. This subfamily corresponds to the RRM of Nop6, also known as Ydl213c, a component of 90S pre-ribosomal particles in yeast S. cerevisiae. It is enriched in the nucleolus and is required for 40S ribosomal subunit biogenesis. Nop6 is a non-essential putative RNA-binding protein with two N-terminal putative nuclear localisation sequences (NLS-1 and NLS-2) and an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). It binds to the pre-rRNA early during transcription and plays an essential role in pre-rRNA processing. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>cd12448 RNA recognition motif 2 (RRM2) found in yeast protein gar2 and similar proteins. This subfamily corresponds to the RRM2 of yeast protein gar2, a novel nucleolar protein required for 18S rRNA and 40S ribosomal subunit accumulation. It shares similar domain architecture with nucleolin from vertebrates and NSR1 from Saccharomyces cerevisiae. The highly phosphorylated N-terminal domain of gar2 is made up of highly acidic regions separated from each other by basic sequences, and contains multiple phosphorylation sites. The central domain of gar2 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The C-terminal RGG (or GAR) domain of gar2 is rich in glycine, arginine and phenylalanine residues. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cr>37</td>
            <td id=cb>71</td>
            <td id=al>cd12384 RNA recognition motif (RRM) found in eukaryotic RNA-binding protein RBM24, RBM38 and similar proteins. This subfamily corresponds to the RRM of RBM24 and RBM38 from vertebrate, SUPpressor family member SUP-12 from Caenorhabditis elegans and similar proteins. Both, RBM24 and RBM38, are preferentially expressed in cardiac and skeletal muscle tissues. They regulate myogenic differentiation by controlling the cell cycle in a p21-dependent or -independent manner. RBM24, also termed RNA-binding region-containing protein 6, interacts with the 3'-untranslated region (UTR) of myogenin mRNA and regulates its stability in C2C12 cells. RBM38, also termed CLL-associated antigen KW-5, or HSRNASEB, or RNA-binding region-containing protein 1(RNPC1), or ssDNA-binding protein SEB4, is a direct target of the p53 family. It is required for maintaining the stability of the basal and stress-induced p21 mRNA by binding to their 3'-UTRs. It also binds the AU-/U-rich elements in p63 3'-UTR and regulates p63 mRNA stability and activity. SUP-12 is a novel tissue-specific splicing factor that controls muscle-specific splicing of the ADF/cofilin pre-mRNA in C. elegans. All family members contain a conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>44</td>
            <td id=cg>84</td>
            <td id=al>cd12447 RNA recognition motif 1 (RRM1) found in yeast protein gar2 and similar proteins. This subfamily corresponds to the RRM1 of yeast protein gar2, a novel nucleolar protein required for 18S rRNA and 40S ribosomal subunit accumulation. It shares similar domain architecture with nucleolin from vertebrates and NSR1 from Saccharomyces cerevisiae. The highly phosphorylated N-terminal domain of gar2 is made up of highly acidic regions separated from each other by basic sequences, and contains multiple phosphorylation sites. The central domain of gar2 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The C-terminal RGG (or GAR) domain of gar2 is rich in glycine, arginine and phenylalanine residues. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>50</td>
            <td id=cg>92</td>
            <td id=al>cd12284 RNA recognition motif 2 (RRM2) found in vertebrate RNA-binding protein RBM23, RBM39 and similar proteins. This subfamily corresponds to the RRM2 of RBM39 (also termed HCC1), a nuclear autoantigen that contains an N-terminal arginine/serine rich (RS) motif and three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). An octapeptide sequence called the RS-ERK motif is repeated six times in the RS region of RBM39. Although the cellular function of RBM23 remains unclear, it shows high sequence homology to RBM39 and contains two RRMs. It may possibly function as a pre-mRNA splicing factor. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=al>cd12330 RNA recognition motif 2 (RRM2) found in yeast nuclear polyadenylated RNA-binding protein 4 (Hrp1p or Nab4p) and similar proteins. This subfamily corresponds to the RRM1 of Hrp1p and similar proteins. Hrp1p or Nab4p, also termed cleavage factor IB (CFIB), is a sequence-specific trans-acting factor that is essential for mRNA 3'-end formation in yeast Saccharomyces cerevisiae. It can be UV cross-linked to RNA and specifically recognizes the (UA)6 RNA element required for both, the cleavage and poly(A) addition steps. Moreover, Hrp1p can shuttle between the nucleus and the cytoplasm, and play an additional role in the export of mRNAs to the cytoplasm. Hrp1p also interacts with Rna15p and Rna14p, two components of CF1A. In addition, Hrp1p functions as a factor directly involved in modulating the activity of the nonsense-mediated mRNA decay (NMD) pathway; it binds specifically to a downstream sequence element (DSE)-containing RNA and interacts with Upf1p, a component of the surveillance complex, further triggering the NMD pathway. Hrp1p contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an arginine-glycine-rich region harboring repeats of the sequence RGGF/Y. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=al>cd12450 RNA recognition motif 1 (RRM1) found in nucleolin-like proteins mainly from plants. This subfamily corresponds to the RRM1 of a group of plant nucleolin-like proteins, including nucleolin 1 (also termed protein nucleolin like 1) and nucleolin 2 (also termed protein nucleolin like 2, or protein parallel like 1). They play roles in the regulation of ribosome synthesis and in the growth and development of plants. Like yeast nucleolin, nucleolin-like proteins possess two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains).  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>56</td>
            <td id=co>56</td>
            <td id=al>PLN03134 glycine-rich RNA-binding protein 4; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>56</td>
            <td id=cr>14</td>
            <td id=al>TIGR01628 polyadenylate binding protein, human types 1, 2, 3, 4 family. These eukaryotic proteins recognize the poly-A of mRNA and consists of four tandem RNA recognition domains at the N-terminus (rrm: pfam00076) followed by a PABP-specific domain (pfam00658) at the C-terminus. The protein is involved in the transport of mRNA's from the nucleus to the cytoplasm. There are four paralogs in Homo sapiens which are expressed in testis, platelets, broadly expressed and of unknown tissue range.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>54</td>
            <td id=cr>14</td>
            <td id=al>TIGR01628 polyadenylate binding protein, human types 1, 2, 3, 4 family. These eukaryotic proteins recognize the poly-A of mRNA and consists of four tandem RNA recognition domains at the N-terminus (rrm: pfam00076) followed by a PABP-specific domain (pfam00658) at the C-terminus. The protein is involved in the transport of mRNA's from the nucleus to the cytoplasm. There are four paralogs in Homo sapiens which are expressed in testis, platelets, broadly expressed and of unknown tissue range.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=co>54</td>
            <td id=cr>13</td>
            <td id=al>TIGR01628 polyadenylate binding protein, human types 1, 2, 3, 4 family. These eukaryotic proteins recognize the poly-A of mRNA and consists of four tandem RNA recognition domains at the N-terminus (rrm: pfam00076) followed by a PABP-specific domain (pfam00658) at the C-terminus. The protein is involved in the transport of mRNA's from the nucleus to the cytoplasm. There are four paralogs in Homo sapiens which are expressed in testis, platelets, broadly expressed and of unknown tissue range.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>53</td>
            <td id=cg>96</td>
            <td id=al>cd12382 RNA recognition motif (RRM) found in heterogeneous nuclear ribonucleoprotein G (hnRNP G), Y chromosome RNA recognition motif 1 (hRBMY), testis-specific heterogeneous nuclear ribonucleoprotein G-T (hnRNP G-T) and similar proteins. This subfamily corresponds to the RRM domain of hnRNP G, also termed glycoprotein p43 or RBMX, an RNA-binding motif protein located on the X chromosome. It is expressed ubiquitously and has been implicated in the splicing control of several pre-mRNAs. Moreover, hnRNP G may function as a regulator of transcription for SREBP-1c and GnRH1. Research has shown that hnRNP G may also act as a tumor-suppressor since it upregulates the Txnip gene and promotes the fidelity of DNA end-joining activity. In addition, hnRNP G appears to play a critical role in proper neural development of zebrafish and frog embryos. The family also includes several paralogs of hnRNP G, such as hRBMY and hnRNP G-T (also termed RNA-binding motif protein, X-linked-like-2). Both, hRBMY and hnRNP G-T, are exclusively expressed in testis and critical for male fertility. Like hnRNP G, hRBMY and hnRNP G-T interact with factors implicated in the regulation of pre-mRNA splicing, such as hTra2-beta1 and T-STAR. Although members in this family share a high conserved N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), they appear to recognize different RNA targets. For instance, hRBMY interacts specifically with a stem-loop structure in which the loop is formed by the sequence CA/UCAA. In contrast, hnRNP G associates with single stranded RNA sequences containing a CCA/C motif. In addition to the RRM, hnRNP G contains a nascent transcripts targeting domain (NTD) in the middle region and a novel auxiliary RNA-binding domain (RBD) in its C-terminal region. The C-terminal RBD exhibits distinct RNA binding specificity, and would play a critical role in the regulation of alternative splicing by hnRNP G. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>52</td>
            <td id=cg>93</td>
            <td id=al>cd12412 RNA recognition motif (RRM) found in AZoospermia (DAZ) autosomal homologs, DAZL (DAZ-like) and BOULE. This subfamily corresponds to the RRM domain of two Deleted in AZoospermia (DAZ) autosomal homologs, DAZL (DAZ-like) and BOULE. BOULE is the founder member of the family and DAZL arose from BOULE in an ancestor of vertebrates. The DAZ gene subsequently originated from a duplication transposition of the DAZL gene. Invertebrates contain a single DAZ homolog, BOULE, while vertebrates, other than catarrhine primates, possess both BOULE and DAZL genes. The catarrhine primates possess BOULE, DAZL, and DAZ genes. The family members encode closely related RNA-binding proteins that are required for fertility in numerous organisms. These proteins contain an RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a varying number of copies of a DAZ motif, believed to mediate protein-protein interactions. DAZL and BOULE contain a single copy of the DAZ motif, while DAZ proteins can contain 8-24 copies of this repeat. Although their specific biochemical functions remain to be investigated, DAZL proteins may interact with poly(A)-binding proteins (PABPs), and act as translational activators of specific mRNAs during gametogenesis.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>52</td>
            <td id=cg>98</td>
            <td id=al>cd12375 RNA recognition motif 1 (RRM1) found in the Hu proteins family, Drosophila sex-lethal (SXL), and similar proteins. This subfamily corresponds to the RRM1 of Hu proteins and SXL. The Hu proteins family represents a group of RNA-binding proteins involved in diverse biological processes. Since the Hu proteins share high homology with the Drosophila embryonic lethal abnormal vision (ELAV) protein, the Hu family is sometimes referred to as the ELAV family. Drosophila ELAV is exclusively expressed in neurons and is required for the correct differentiation and survival of neurons in flies. The neuronal members of the Hu family include Hu-antigen B (HuB or ELAV-2 or Hel-N1), Hu-antigen C (HuC or ELAV-3 or PLE21), and Hu-antigen D (HuD or ELAV-4), which play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. Hu-antigen R (HuR or ELAV-1 or HuA) is ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. Hu proteins perform their cytoplasmic and nuclear molecular functions by coordinately regulating functionally related mRNAs. In the cytoplasm, Hu proteins recognize and bind to AU-rich RNA elements (AREs) in the 3' untranslated regions (UTRs) of certain target mRNAs, such as GAP-43, vascular epithelial growth factor (VEGF), the glucose transporter GLUT1, eotaxin and c-fos, and stabilize those ARE-containing mRNAs. They also bind and regulate the translation of some target mRNAs, such as neurofilament M, GLUT1, and p27. In the nucleus, Hu proteins function as regulators of polyadenylation and alternative splicing. Each Hu protein contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. This family also includes the sex-lethal protein (SXL) from Drosophila melanogaster. SXL governs sexual differentiation and X chromosome dosage compensation in flies. It induces female-specific alternative splicing of the transformer (tra) pre-mRNA by binding to the tra uridine-rich polypyrimidine tract at the non-sex-specific 3' splice site during the sex-determination process. SXL binds to its own pre-mRNA and promotes female-specific alternative splicing. It contains an N-terminal Gly/Asn-rich domain that may be responsible for the protein-protein interaction, and tandem RRMs that show high preference to bind single-stranded, uridine-rich target RNA transcripts. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>52</td>
            <td id=cg>95</td>
            <td id=al>cd12449 RNA recognition motif (RRM) found in cold inducible RNA binding protein (CIRBP), RNA binding motif protein 3 (RBM3) and similar proteins. This subfamily corresponds to the RRM domain of two structurally related heterogenous nuclear ribonucleoproteins, CIRBP (also termed CIRP or A18 hnRNP) and RBM3 (also termed RNPL), both of which belong to a highly conserved cold shock proteins family. The cold shock proteins can be induced after exposure to a moderate cold-shock and other cellular stresses such as UV radiation and hypoxia. CIRBP and RBM3 may function in posttranscriptional regulation of gene expression by binding to different transcripts, thus allowing the cell to response rapidly to environmental signals. However, the kinetics and degree of cold induction are different between CIRBP and RBM3. Tissue distribution of their expression is different. CIRBP and RBM3 may be differentially regulated under physiological and stress conditions and may play distinct roles in cold responses of cells. CIRBP, also termed glycine-rich RNA-binding protein CIRP, is localized in the nucleus and mediates the cold-induced suppression of cell cycle progression. CIRBP also binds DNA and possibly serves as a chaperone that assists in the folding/unfolding, assembly/disassembly and transport of various proteins. RBM3 may enhance global protein synthesis and the formation of active polysomes while reducing the levels of ribonucleoprotein complexes containing microRNAs. RBM3 may also serve to prevent the loss of muscle mass by its ability to decrease cell death. Furthermore, RBM3 may be essential for cell proliferation and mitosis. Both, CIRBP and RBM3, contain an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), that is involved in RNA binding, and C-terminal glycine-rich domain (RGG motif) that probably enhances RNA-binding via protein-protein and/or protein-RNA interactions. Like CIRBP, RBM3 can also bind to both RNA and DNA via its RRM domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>52</td>
            <td id=cg>98</td>
            <td id=al>cd12397 RNA recognition motif 2 (RRM2) found in yeast nucleolar protein 13 (Nop13p) and similar proteins. This subfamily corresponds to the RRM2 of Nop13p encoded by YNL175c from Saccharomyces cerevisiae. It shares high sequence similarity with nucleolar protein 12 (Nop12p). Both Nop12p and Nop13p are not essential for growth. However, unlike Nop12p that is localized to the nucleolus, Nop13p localizes primarily to the nucleolus but is also present in the nucleoplasm to a lesser extent. Nop13p contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=co>40</td>
            <td id=cb>72</td>
            <td id=al>cd12581 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A2/B1 (hnRNP A2/B1) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A2/B1, an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). Many mRNAs, such as myelin basic protein (MBP), myelin-associated oligodendrocytic basic protein (MOBP), carboxyanhydrase II (CAII), microtubule-associated protein tau, and amyloid precursor protein (APP) are trafficked by hnRNP A2/B1. hnRNP A2/B1 also functions as a splicing factor that regulates alternative splicing of the tumor suppressors, such as BIN1, WWOX, the antiapoptotic proteins c-FLIP and caspase-9B, the insulin receptor (IR), and the RON proto-oncogene among others. Overexpression of hnRNP A2/B1 has been described in many cancers. It functions as a nuclear matrix protein involving in RNA synthesis and the regulation of cellular migration through alternatively splicing pre-mRNA. It may play a role in tumor cell differentiation. hnRNP A2/B1 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cb>64</td>
            <td id=cr>18</td>
            <td id=al>TIGR01622 splicing factor, CC1-like family. This model represents a subfamily of RNA splicing factors including the Pad-1 protein (N. crassa), CAPER (M. musculus) and CC1.3 (H.sapiens). These proteins are characterized by an N-terminal arginine-rich, low complexity domain followed by three (or in the case of 4 H. sapiens paralogs, two) RNA recognition domains (rrm: pfam00706). These splicing factors are closely related to the U2AF splicing factor family (TIGR01642). A homologous gene from Plasmodium falciparum was identified in the course of the analysis of that genome at TIGR and was included in the seed.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12565 RNA recognition motif 1 (RRM1) found in yeast multiple RNA-binding domain-containing protein 1 (MRD1) and similar proteins. This subgroup corresponds to the RRM1 of MRD1 which is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well-conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). MRD1 is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. It contains 5 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>55</td>
            <td id=cg>98</td>
            <td id=al>cd12402 RNA recognition motif (RRM) found in eukaryotic translation initiation factor 4B (eIF-4B) and similar proteins. This subfamily corresponds to the RRM of eIF-4B, a multi-domain RNA-binding protein that has been primarily implicated in promoting the binding of 40S ribosomal subunits to mRNA during translation initiation. It contains two RNA-binding domains; the N-terminal well-conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), binds the 18S rRNA of the 40S ribosomal subunit and the C-terminal basic domain (BD), including two arginine-rich motifs (ARMs), binds mRNA during initiation, and is primarily responsible for the stimulation of the helicase activity of eIF-4A. eIF-4B also contains a DRYG domain (a region rich in Asp, Arg, Tyr, and Gly amino acids) in the middle, which is responsible for both, self-association of eIF-4B and  binding to the p170 subunit of eIF3. Additional research indicates that eIF-4B can interact with the poly(A) binding protein (PABP) in mammalian cells, which can stimulate both, the eIF-4B-mediated activation of the helicase activity of eIF-4A and binding of poly(A) by PABP. eIF-4B has also been shown to interact specifically with the internal ribosome entry sites (IRES) of several picornaviruses which facilitate cap-independent translation initiation. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>54</td>
            <td id=cg>97</td>
            <td id=al>cd12674 RNA recognition motif 1 (RRM1) found in yeast nucleolar protein 4 (Nop4p) and similar proteins. This subgroup corresponds to the RRM1 of Nop4p (also known as Nop77p), encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>43</td>
            <td id=cg>88</td>
            <td id=al>cd12243 RNA recognition motif 1 (RRM1) found in the c-myc gene single-strand binding proteins (MSSP) family. This subfamily corresponds to the RRM1 of c-myc gene single-strand binding proteins (MSSP) family, including single-stranded DNA-binding protein MSSP-1 (also termed RBMS1 or SCR2) and MSSP-2 (also termed RBMS2 or SCR3). All MSSP family members contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), both of which are responsible for the specific DNA binding activity. Both, MSSP-1 and -2, have been identified as protein factors binding to a putative DNA replication origin/transcriptional enhancer sequence present upstream from the human c-myc gene in both single- and double-stranded forms. Thus, they have been implied in regulating DNA replication, transcription, apoptosis induction, and cell-cycle movement, via the interaction with c-MYC, the product of protooncogene c-myc. Moreover, the family includes a new member termed RNA-binding motif, single-stranded-interacting protein 3 (RBMS3), which is not a transcriptional regulator. RBMS3 binds with high affinity to A/U-rich stretches of RNA, and to A/T-rich DNA sequences, and functions as a regulator of cytoplasmic activity. In addition, a putative meiosis-specific RNA-binding protein termed sporulation-specific protein 5 (SPO5, or meiotic RNA-binding protein 1, or meiotically up-regulated gene 12 protein), encoded by Schizosaccharomyces pombe Spo5/Mug12 gene, is also included in this family. SPO5 is a novel meiosis I regulator that may function in the vicinity of the Mei2 dot. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>53</td>
            <td id=cg>93</td>
            <td id=al>cd12769 RNA recognition motif 1 (RRM1) found in vertebrate Hu-antigen R (HuR). This subgroup corresponds to the RRM1 of HuR, also termed ELAV-like protein 1 (ELAV-1), a ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. HuR has an anti-apoptotic function during early cell stress response; it binds to mRNAs and enhances the expression of several anti-apoptotic proteins, such as p21waf1, p53, and prothymosin alpha. Meanwhile, HuR also has pro-apoptotic function by promoting apoptosis when cell death is unavoidable. Furthermore, HuR may be important in muscle differentiation, adipogenesis, suppression of inflammatory response and modulation of gene expression in response to chronic ethanol exposure and amino acid starvation. Like other Hu proteins, HuR contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=al>cd12579 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A0 (hnRNP A0) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A0, a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. It has been identified as the substrate for MAPKAP-K2 and may be involved in the lipopolysaccharide (LPS)-induced post-transcriptional regulation of tumor necrosis factor-alpha (TNF-alpha), cyclooxygenase 2 (COX-2) and macrophage inflammatory protein 2 (MIP-2). hnRNP A0 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>46</td>
            <td id=cg>88</td>
            <td id=al>cd12414 RNA recognition motif 2 (RRM2) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM2 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>50</td>
            <td id=cg>94</td>
            <td id=al>cd12377 RNA recognition motif 3 (RRM3) found in the Hu proteins family. This subfamily corresponds to the RRM3 of the Hu proteins family which represent a group of RNA-binding proteins involved in diverse biological processes. Since the Hu proteins share high homology with the Drosophila embryonic lethal abnormal vision (ELAV) protein, the Hu family is sometimes referred to as the ELAV family. Drosophila ELAV is exclusively expressed in neurons and is required for the correct differentiation and survival of neurons in flies. The neuronal members of the Hu family include Hu-antigen B (HuB or ELAV-2 or Hel-N1), Hu-antigen C (HuC or ELAV-3 or PLE21), and Hu-antigen D (HuD or ELAV-4), which play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. Hu-antigen R (HuR or ELAV-1 or HuA) is the ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. Hu proteins perform their cytoplasmic and nuclear molecular functions by coordinately regulating functionally related mRNAs. In the cytoplasm, Hu proteins recognize and bind to AU-rich RNA elements (AREs) in the 3' untranslated regions (UTRs) of certain target mRNAs, such as GAP-43, vascular epithelial growth factor (VEGF), the glucose transporter GLUT1, eotaxin and c-fos, and stabilize those ARE-containing mRNAs. They also bind and regulate the translation of some target mRNAs, such as neurofilament M, GLUT1, and p27. In the nucleus, Hu proteins function as regulators of polyadenylation and alternative splicing. Each Hu protein contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>40</td>
            <td id=cg>81</td>
            <td id=al>cd12325 RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein hnRNP A and hnRNP D subfamilies and similar proteins. This subfamily corresponds to the RRM1 in the hnRNP A subfamily which includes hnRNP A0, hnRNP A1, hnRNP A2/B1, hnRNP A3 and similar proteins. hnRNP A0 is a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. hnRNP A1 is an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A2/B1 is an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). hnRNP A3 is also a RNA trafficking response element-binding protein that participates in the trafficking of A2RE-containing RNA. The hnRNP A subfamily is characterized by two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. The hnRNP D subfamily includes hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. hnRNP D0 is a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP A/B is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. hnRNP A/B has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus, plays an important role in apoB mRNA editing. hnRNP DL (or hnRNP D-like) is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. All members in this subfamily contain two putative RRMs and a glycine- and tyrosine-rich C-terminus. The family also contains DAZAP1 (Deleted in azoospermia-associated protein 1), RNA-binding protein Musashi homolog Musashi-1, Musashi-2 and similar proteins. They all harbor two RRMs. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=al>cd12409 RNA recognition motif 1 (RRM1) found in yeast regulator of rDNA transcription protein 5 (RRT5) and similar proteins. This subfamily corresponds to the RRM1 of the lineage specific family containing a group of uncharacterized yeast regulators of rDNA transcription protein 5 (RRT5), which may play roles in the modulation of rDNA transcription. RRT5 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>53</td>
            <td id=cr>21</td>
            <td id=al>TIGR01661 ELAV/HuD family splicing factor. This model describes the ELAV/HuD subfamily of splicing factors found in metazoa. HuD stands for the human paraneoplastic encephalomyelitis antigen D of which there are 4 variants in human. ELAV stnds for the Drosophila Embryonic lethal abnormal visual protein. ELAV-like splicing factors are also known in human as HuB (ELAV-like protein 2), HuC (ELAV-like protein 3, Paraneoplastic cerebellar degeneration-associated antigen) and HuR (ELAV-like protein 1). These genes are most closely related to the sex-lethal subfamily of splicing factors found in Dipteran insects (TIGR01659). These proteins contain 3 RNA-recognition motifs (rrm: pfam00076).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>50</td>
            <td id=cr>20</td>
            <td id=al>TIGR01661 ELAV/HuD family splicing factor. This model describes the ELAV/HuD subfamily of splicing factors found in metazoa. HuD stands for the human paraneoplastic encephalomyelitis antigen D of which there are 4 variants in human. ELAV stnds for the Drosophila Embryonic lethal abnormal visual protein. ELAV-like splicing factors are also known in human as HuB (ELAV-like protein 2), HuC (ELAV-like protein 3, Paraneoplastic cerebellar degeneration-associated antigen) and HuR (ELAV-like protein 1). These genes are most closely related to the sex-lethal subfamily of splicing factors found in Dipteran insects (TIGR01659). These proteins contain 3 RNA-recognition motifs (rrm: pfam00076).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>52</td>
            <td id=cg>96</td>
            <td id=al>cd21610 RNA recognition motif 2 (RRM2) found in chloroplastic plastid-specific 30S ribosomal protein 2 (PSRP-2) and similar proteins. PSRP-2, also called chloroplastic 30S ribosomal protein 2, or chloroplastic small ribosomal subunit protein cS22, is a component of the chloroplast ribosome (chloro-ribosome), a dedicated translation machinery responsible for the synthesis of chloroplast genome-encoded proteins, including proteins of the transcription and translation machinery and components of the photosynthetic apparatus. It binds single strand DNA (ssDNA) and RNA in vitro. It exhibits RNA chaperone activity and regulates negatively resistance responses to abiotic stresses during seed germination (e.g. salt, dehydration, and low temperature) and seedling growth (e.g. salt). PSRP-2 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the second RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>52</td>
            <td id=cg>93</td>
            <td id=al>cd12619 RNA recognition motif 2 (RRM2) found in yeast nuclear and cytoplasmic polyadenylated RNA-binding protein PUB1 and similar proteins. This subgroup corresponds to the RRM2 of yeast protein PUB1, also termed ARS consensus-binding protein ACBP-60, or poly uridylate-binding protein, or poly(U)-binding protein. PUB1 has been identified as both, a heterogeneous nuclear RNA-binding protein (hnRNP) and a cytoplasmic mRNA-binding protein (mRNP), which may be stably bound to a translationally inactive subpopulation of mRNAs within the cytoplasm. It is distributed in both, the nucleus and the cytoplasm, and binds to poly(A)+ RNA (mRNA or pre-mRNA). Although it is one of the major cellular proteins cross-linked by UV light to polyadenylated RNAs in vivo, PUB1 is nonessential for cell growth in yeast. PUB1 also binds to T-rich single stranded DNA (ssDNA). However, there is no strong evidence implicating PUB1 in the mechanism of DNA replication. PUB1 contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a GAR motif (glycine and arginine rich stretch) that is located between RRM2 and RRM3. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=al>cd12226 RNA recognition motif (RRM) found in nucleolar protein 8 (NOL8) and similar proteins. This model corresponds to the RRM of NOL8 (also termed Nop132) encoded by a novel NOL8 gene that is up-regulated in the majority of diffuse-type, but not intestinal-type, gastric cancers. Thus, NOL8 may be a good molecular target for treatment of diffuse-type gastric cancer. Also, NOL8 is a phosphorylated protein that contains an N-terminal RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), suggesting NOL8 is likely to function as a novel RNA-binding protein. It may be involved in regulation of gene expression at the post-transcriptional level or in ribosome biogenesis in cancer cells.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=al>cd12358 RNA recognition motif 1 (RRM1) found in the VICKZ family proteins. Thid subfamily corresponds to the RRM1 of IGF2BPs (or IMPs) found in the VICKZ family that have been implicated in the post-transcriptional regulation of several different RNAs and in subcytoplasmic localization of mRNAs during embryogenesis. IGF2BPs are composed of two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and four hnRNP K homology (KH) domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=co>43</td>
            <td id=cb>79</td>
            <td id=al>cd12381 RNA recognition motif 4 (RRM4) found in type I polyadenylate-binding proteins. This subfamily corresponds to the RRM4 of type I poly(A)-binding proteins (PABPs), highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in theThe CD corresponds to the RRM. regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. The family represents type I polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 1 (PABP-1 or PABPC1), polyadenylate-binding protein 3 (PABP-3 or PABPC3), polyadenylate-binding protein 4 (PABP-4 or APP-1 or iPABP), polyadenylate-binding protein 5 (PABP-5 or PABPC5), polyadenylate-binding protein 1-like (PABP-1-like or PABPC1L), polyadenylate-binding protein 1-like 2 (PABPC1L2 or RBM32), polyadenylate-binding protein 4-like (PABP-4-like or PABPC4L), yeast polyadenylate-binding protein, cytoplasmic and nuclear (PABP or ACBP-67), and similar proteins. PABP-1 is an ubiquitously expressed multifunctional protein that may play a role in 3' end formation of mRNA, translation initiation, mRNA stabilization, protection of poly(A) from nuclease activity, mRNA deadenylation, inhibition of mRNA decapping, and mRNP maturation. Although PABP-1 is thought to be a cytoplasmic protein, it is also found in the nucleus. PABP-1 may be involved in nucleocytoplasmic trafficking and utilization of mRNP particles. PABP-1 contains four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), a less well conserved linker region, and a proline-rich C-terminal conserved domain (CTD). PABP-3 is a testis-specific poly(A)-binding protein specifically expressed in round spermatids. It is mainly found in mammalian and may play an important role in the testis-specific regulation of mRNA homeostasis. PABP-3 shows significant sequence similarity to PABP-1. However, it binds to poly(A) with a lower affinity than PABP-1. Moreover, PABP-1 possesses an A-rich sequence in its 5'-UTR and allows binding of PABP and blockage of translation of its own mRNA. In contrast, PABP-3 lacks the A-rich sequence in its 5'-UTR. PABP-4 is an inducible poly(A)-binding protein (iPABP) that is primarily localized to the cytoplasm. It shows significant sequence similarity to PABP-1 as well. The RNA binding properties of PABP-1 and PABP-4 appear to be identical. PABP-5 is encoded by PABPC5 gene within the X-specific subinterval, and expressed in fetal brain and in a range of adult tissues in mammalian, such as ovary and testis. It may play an important role in germ cell development. Moreover, unlike other PABPs, PABP-5 contains only four RRMs, but lacks both the linker region and the CTD. PABP-1-like and PABP-1-like 2 are the orthologs of PABP-1. PABP-4-like is the ortholog of PABP-5. Their cellular functions remain unclear. The family also includes the yeast PABP, a conserved poly(A) binding protein containing poly(A) tails that can be attached to the 3'-ends of mRNAs. The yeast PABP and its homologs may play important roles in the initiation of translation and in mRNA decay. Like vertebrate PABP-1, the yeast PABP contains four RRMs, a linker region, and a proline-rich CTD as well. The first two RRMs are mainly responsible for specific binding to poly(A). The proline-rich region may be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>47</td>
            <td id=cg>84</td>
            <td id=al>cd12672 RNA recognition motif (RRM) found in vertebrate deleted in azoospermia-like (DAZL) proteins. This subgroup corresponds to the RRM of DAZL, also termed SPGY-like-autosomal, encoded by the autosomal homolog of DAZ gene, DAZL. It is ancestral to the deleted in azoospermia (DAZ) protein. DAZL is germ-cell-specific RNA-binding protein that contains a RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a DAZ motif, a protein-protein interaction domain. Although their specific biochemical functions remain to be investigated, DAZL proteins may interact with poly(A)-binding proteins (PABPs), and act as translational activators of specific mRNAs during gametogenesis. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>56</td>
            <td id=cg>98</td>
            <td id=al>cd12280 RNA recognition motif (RRM) found in the FET family of RNA-binding proteins. This subfamily corresponds to the RRM of FET (previously TET) (FUS/TLS, EWS, TAF15) family of RNA-binding proteins. This ubiquitously expressed family of similarly structured proteins predominantly localizing to the nuclear, includes FUS (also known as TLS or Pigpen or hnRNP P2), EWS (also known as EWSR1), TAF15 (also known as hTAFII68 or TAF2N or RPB56), and Drosophila Cabeza (also known as SARFH). The corresponding coding genes of these proteins are involved in deleterious genomic rearrangements with transcription factor genes in a variety of human sarcomas and acute leukemias. All FET proteins interact with each other and are therefore likely to be part of the very same protein complexes, which suggests a general bridging role for FET proteins coupling RNA transcription, processing, transport, and DNA repair. The FET proteins contain multiple copies of a degenerate hexapeptide repeat motif at the N-terminus. The C-terminal region consists of a conserved nuclear import and retention signal (C-NLS), a putative zinc-finger domain, and a conserved RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), which is flanked by 3 arginine-glycine-glycine (RGG) boxes. FUS and EWS might have similar sequence specificity; both bind preferentially to GGUG-containing RNAs. FUS has also been shown to bind strongly to human telomeric RNA and to small low-copy-number RNAs tethered to the promoter of cyclin D1. To date, nothing is known about the RNA binding specificity of TAF15. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>52</td>
            <td id=cg>91</td>
            <td id=al>cd12383 RNA recognition motif (RRM) found in RNA-binding protein 42 (RBM42) and similar proteins. This subfamily corresponds to the RRM of RBM42 which has been identified as a heterogeneous nuclear ribonucleoprotein K (hnRNP K)-binding protein. It also directly binds the 3' untranslated region of p21 mRNA that is one of the target mRNAs for hnRNP K. Both, hnRNP K and RBM42, are components of stress granules (SGs). Under nonstress conditions, RBM42 predominantly localizes within the nucleus and co-localizes with hnRNP K. Under stress conditions, hnRNP K and RBM42 form cytoplasmic foci where the SG marker TIAR localizes, and may play a role in the maintenance of cellular ATP level by protecting their target mRNAs. RBM42 contains an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>53</td>
            <td id=cg>90</td>
            <td id=al>cd12772 RNA recognition motif 1 (RRM1) found in vertebrate Hu-antigen C (HuC). This subgroup corresponds to the RRM1 of HuC, also termed ELAV-like protein 3 (ELAV-3), or paraneoplastic cerebellar degeneration-associated antigen, or paraneoplastic limbic encephalitis antigen 21 (PLE21), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. Like other Hu proteins, HuC contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). The AU-rich element binding of HuC can be inhibited by flavonoids. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>52</td>
            <td id=cg>97</td>
            <td id=al>cd12650 RNA recognition motif 1 (RRM1) found in the Hu proteins family. This subfamily corresponds to the RRM1 of the Hu proteins family which represents a group of RNA-binding proteins involved in diverse biological processes. Since the Hu proteins share high homology with the Drosophila embryonic lethal abnormal vision (ELAV) protein, the Hu family is sometimes referred to as the ELAV family. Drosophila ELAV is exclusively expressed in neurons and is required for the correct differentiation and survival of neurons in flies. The neuronal members of the Hu family include Hu-antigen B (HuB or ELAV-2 or Hel-N1), Hu-antigen C (HuC or ELAV-3 or PLE21), and Hu-antigen D (HuD or ELAV-4), which play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. Hu-antigen R (HuR or ELAV-1 or HuA) is the ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. HuR has an anti-apoptotic function during early cell stress response. It binds to mRNAs and enhances the expression of several anti-apoptotic proteins, such as p21waf1, p53, and prothymosin alpha. HuR also has pro-apoptotic function by promoting apoptosis when cell death is unavoidable. Furthermore, HuR may be important in muscle differentiation, adipogenesis, suppression of inflammatory response and modulation of gene expression in response to chronic ethanol exposure and amino acid starvation. Hu proteins perform their cytoplasmic and nuclear molecular functions by coordinately regulating functionally related mRNAs. In the cytoplasm, Hu proteins recognize and bind to AU-rich RNA elements (AREs) in the 3' untranslated regions (UTRs) of certain target mRNAs, such as GAP-43, vascular epithelial growth factor (VEGF), the glucose transporter GLUT1, eotaxin and c-fos, and stabilize those ARE-containing mRNAs. They also bind and regulate the translation of some target mRNAs, such as neurofilament M, GLUT1, and p27. In the nucleus, Hu proteins function as regulators of polyadenylation and alternative splicing. Each Hu protein contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>44</td>
            <td id=cg>80</td>
            <td id=al>cd12633 RNA recognition motif 1 (RRM1) found in plant flowering time control protein FCA and similar proteins. This subgroup corresponds to the RRM1 of FCA, a gene controlling flowering time in Arabidopsis, encoding a flowering time control protein that functions in the posttranscriptional regulation of transcripts involved in the flowering process. FCA contains two RNA recognition motifs (RRMs), also known as RBDs (RNA binding domains) or RNP (ribonucleoprotein domains), and a WW protein interaction domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cr>36</td>
            <td id=cb>71</td>
            <td id=al>cd12321 RNA recognition motif 1 (RRM1) found in TAR DNA-binding protein 43 (TDP-43) and similar proteins. This subfamily corresponds to the RRM1 of TDP-43 (also termed TARDBP), a ubiquitously expressed pathogenic protein whose normal function and abnormal aggregation are directly linked to the genetic disease cystic fibrosis, and two neurodegenerative disorders: frontotemporal lobar degeneration (FTLD) and amyotrophic lateral sclerosis (ALS). TDP-43 binds both DNA and RNA, and has been implicated in transcriptional repression, pre-mRNA splicing and translational regulation. TDP-43 is a dimeric protein with two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal glycine-rich domain. The RRMs are responsible for DNA and RNA binding; they bind to TAR DNA and RNA sequences with UG-repeats. The glycine-rich domain can interact with the hnRNP family proteins to form the hnRNP-rich complex involved in splicing inhibition. It is also essential for the cystic fibrosis transmembrane conductance regulator (CFTR) exon 9-skipping activity. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>43</td>
            <td id=cg>84</td>
            <td id=al>cd12353 RNA recognition motif 2 (RRM2) found in granule-associated RNA binding proteins p40-TIA-1 and TIAR. This subfamily corresponds to the RRM2 of nucleolysin TIA-1 isoform p40 (p40-TIA-1 or TIA-1) and nucleolysin TIA-1-related protein (TIAR), both of which are granule-associated RNA binding proteins involved in inducing apoptosis in cytotoxic lymphocyte (CTL) target cells. TIA-1 and TIAR share high sequence similarity. They are expressed in a wide variety of cell types. TIA-1 can be phosphorylated by a serine/threonine kinase that is activated during Fas-mediated apoptosis. TIAR is mainly localized in the nucleus of hematopoietic and nonhematopoietic cells. It is translocated from the nucleus to the cytoplasm in response to exogenous triggers of apoptosis. Both, TIA-1 and TIAR, bind specifically to poly(A) but not to poly(C) homopolymers. They are composed of three N-terminal highly homologous RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glutamine-rich C-terminal auxiliary domain containing a lysosome-targeting motif. TIA-1 and TIAR interact with RNAs containing short stretches of uridylates and their RRM2 can mediate the specific binding to uridylate-rich RNAs. The C-terminal auxiliary domain may be responsible for interacting with other proteins. In addition, TIA-1 and TIAR share a potential serine protease-cleavage site (Phe-Val-Arg) localized at the junction between their RNA binding domains and their C-terminal auxiliary domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>43</td>
            <td id=co>52</td>
            <td id=al>cd21615 RNA recognition motif (RRM) found in Saccharomyces cerevisiae U1 small nuclear ribonucleoprotein SNP1 and similar proteins. SNP1, also called U1 snRNP protein SNP1, or U1 small nuclear ribonucleoprotein 70 kDa homolog, or U1 70K, or U1 snRNP 70 kDa homolog, interacts with mRNA and is involved in nuclear mRNA splicing. It is a component of the spliceosome, where it is associated with snRNP U1 by binding stem loop I of U1 snRNA. Members in this family contain an N-terminal U1snRNP70 domain and an RNA recognition motif (RRM), also called RBD (RNA binding domain) or RNP (ribonucleoprotein domain).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>50</td>
            <td id=cg>84</td>
            <td id=al>cd12653 RNA recognition motif 3 (RRM3) found in vertebrate Hu-antigen R (HuR). This subgroup corresponds to the RRM3 of HuR, also termed ELAV-like protein 1 (ELAV-1), the ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. HuR has an anti-apoptotic function during early cell stress response. It binds to mRNAs and enhances the expression of several anti-apoptotic proteins, such as p21waf1, p53, and prothymosin alpha. HuR also has pro-apoptotic function by promoting apoptosis when cell death is unavoidable. Furthermore, HuR may be important in muscle differentiation, adipogenesis, suppression of inflammatory response and modulation of gene expression in response to chronic ethanol exposure and amino acid starvation. Like other Hu proteins, HuR contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12373 RNA recognition motif (RRM) found in serine/arginine-rich splicing factor 3 (SRSF3) and similar proteins. This subfamily corresponds to the RRM of two serine/arginine (SR) proteins, serine/arginine-rich splicing factor 3 (SRSF3) and serine/arginine-rich splicing factor 7 (SRSF7). SRSF3, also termed pre-mRNA-splicing factor SRp20, modulates alternative splicing by interacting with RNA cis-elements in a concentration- and cell differentiation-dependent manner. It is also involved in termination of transcription, alternative RNA polyadenylation, RNA export, and protein translation. SRSF3 is critical for cell proliferation, and tumor induction and maintenance. It can shuttle between the nucleus and cytoplasm. SRSF7, also termed splicing factor 9G8, plays a crucial role in both constitutive splicing and alternative splicing of many pre-mRNAs. Its localization and functions are tightly regulated by phosphorylation. SRSF7 is predominantly present in the nuclear and can shuttle between nucleus and cytoplasm. It cooperates with the export protein, Tap/NXF1, helps mRNA export to the cytoplasm, and enhances the expression of unspliced mRNA. Moreover, SRSF7 inhibits tau E10 inclusion through directly interacting with the proximal downstream intron of E10, a clustering region for frontotemporal dementia with Parkinsonism (FTDP) mutations. Both SRSF3 and SRSF7 contain a single N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a C-terminal RS domain rich in serine-arginine dipeptides. The RRM domain is involved in RNA binding, and the RS domain has been implicated in protein shuttling and protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>52</td>
            <td id=cg>98</td>
            <td id=al>cd12371 RNA recognition motif 2 (RRM2) found in (U)-binding-splicing factor PUF60 and similar proteins. This subfamily corresponds to the RRM2 of PUF60, also termed FUSE-binding protein-interacting repressor (FBP-interacting repressor or FIR), or Ro-binding protein 1 (RoBP1), or Siah-binding protein 1 (Siah-BP1). PUF60 is an essential splicing factor that functions as a poly-U RNA-binding protein required to reconstitute splicing in depleted nuclear extracts. Its function is enhanced through interaction with U2 auxiliary factor U2AF65. PUF60 also controls human c-myc gene expression by binding and inhibiting the transcription factor far upstream sequence element (FUSE)-binding-protein (FBP), an activator of c-myc promoters. PUF60 contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal U2AF (U2 auxiliary factor) homology motifs (UHM) that harbors another RRM and binds to tryptophan-containing linear peptide motifs (UHM ligand motifs, ULMs) in several nuclear proteins. Research indicates that PUF60 binds FUSE as a dimer, and only the first two RRM domains participate in the single-stranded DNA recognition. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>50</td>
            <td id=cg>83</td>
            <td id=al>cd12654 RNA recognition motif 3 (RRM3) found in vertebrate Hu-antigen B (HuB). This subgroup corresponds to the RRM3 of HuB, also termed ELAV-like protein 2 (ELAV-2), or ELAV-like neuronal protein 1, or nervous system-specific RNA-binding protein Hel-N1 (Hel-N1), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. It is up-regulated during neuronal differentiation of embryonic carcinoma P19 cells. Like other Hu proteins, HuB contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>43</td>
            <td id=cb>78</td>
            <td id=al>cd12376 RNA recognition motif 2 (RRM2) found in the Hu proteins family, Drosophila sex-lethal (SXL), and similar proteins. This subfamily corresponds to the RRM2 of Hu proteins and SXL. The Hu proteins family represents a group of RNA-binding proteins involved in diverse biological processes. Since the Hu proteins share high homology with the Drosophila embryonic lethal abnormal vision (ELAV) protein, the Hu family is sometimes referred to as the ELAV family. Drosophila ELAV is exclusively expressed in neurons and is required for the correct differentiation and survival of neurons in flies. The neuronal members of the Hu family include Hu-antigen B (HuB or ELAV-2 or Hel-N1), Hu-antigen C (HuC or ELAV-3 or PLE21), and Hu-antigen D (HuD or ELAV-4), which play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. Hu-antigen R (HuR or ELAV-1 or HuA) is the ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. Hu proteins perform their cytoplasmic and nuclear molecular functions by coordinately regulating functionally related mRNAs. In the cytoplasm, Hu proteins recognize and bind to AU-rich RNA elements (AREs) in the 3' untranslated regions (UTRs) of certain target mRNAs, such as GAP-43, vascular epithelial growth factor (VEGF), the glucose transporter GLUT1, eotaxin and c-fos, and stabilize those ARE-containing mRNAs. They also bind and regulate the translation of some target mRNAs, such as neurofilament M, GLUT1, and p27. In the nucleus, Hu proteins function as regulators of polyadenylation and alternative splicing. Each Hu protein contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. Also included in this subfamily is the sex-lethal protein (SXL) from Drosophila melanogaster. SXL governs sexual differentiation and X chromosome dosage compensation in flies. It induces female-specific alternative splicing of the transformer (tra) pre-mRNA by binding to the tra uridine-rich polypyrimidine tract at the non-sex-specific 3' splice site during the sex-determination process. SXL binds also to its own pre-mRNA and promotes female-specific alternative splicing. SXL contains an N-terminal Gly/Asn-rich domain that may be responsible for the protein-protein interaction, and tandem RRMs that show high preference to bind single-stranded, uridine-rich target RNA transcripts. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>44</td>
            <td id=cg>80</td>
            <td id=al>cd12363 RNA recognition motif (RRM) found in transformer-2 protein homolog TRA2-alpha, TRA2-beta and similar proteins. This subfamily corresponds to the RRM of two mammalian homologs of Drosophila transformer-2 (Tra2), TRA2-alpha, TRA2-beta (also termed SFRS10), and similar proteins found in eukaryotes. TRA2-alpha is a 40-kDa serine/arginine-rich (SR) protein that specifically binds to gonadotropin-releasing hormone (GnRH) exonic splicing enhancer on exon 4 (ESE4) and is necessary for enhanced GnRH pre-mRNA splicing. It strongly stimulates GnRH intron A excision in a dose-dependent manner. In addition, TRA2-alpha can interact with either 9G8 or SRp30c, which may also be crucial for ESE-dependent GnRH pre-mRNA splicing. TRA2-beta is a serine/arginine-rich (SR) protein that controls the pre-mRNA alternative splicing of the calcitonin/calcitonin gene-related peptide (CGRP), the survival motor neuron 1 (SMN1) protein and the tau protein. Both, TRA2-alpha and TRA2-beta, contains a well conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), flanked by the N- and C-terminal arginine/serine (RS)-rich regions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>46</td>
            <td id=cg>90</td>
            <td id=al>cd12316 RNA recognition motif 3 (RRM3) found in RNA-binding protein 19 (RBM19) and RNA recognition motif 2 found in multiple RNA-binding domain-containing protein 1 (MRD1). This subfamily corresponds to the RRM3 of RBM19 and RRM2 of MRD1. RBM19, also termed RNA-binding domain-1 (RBD-1), is a nucleolar protein conserved in eukaryotes involved in ribosome biogenesis by processing rRNA and is essential for preimplantation development. It has a unique domain organization containing 6 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). MRD1 is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). It is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. MRD1 contains 5 conserved RRMs, which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>52</td>
            <td id=cg>98</td>
            <td id=al>cd12231 RNA recognition motif 2 (RRM2) found in U2 large nuclear ribonucleoprotein auxiliary factor U2AF 65 kDa subunit (U2AF65) and similar proteins. This subfamily corresponds to the RRM2 of U2AF65 and dU2AF50. U2AF65, also termed U2AF2, is the large subunit of U2 small nuclear ribonucleoprotein (snRNP) auxiliary factor (U2AF), which has been implicated in the recruitment of U2 snRNP to pre-mRNAs and is a highly conserved heterodimer composed of large and small subunits. U2AF65 specifically recognizes the intron polypyrimidine tract upstream of the 3' splice site and promotes binding of U2 snRNP to the pre-mRNA branchpoint. U2AF65 also plays an important role in the nuclear export of mRNA. It facilitates the formation of a messenger ribonucleoprotein export complex, containing both the NXF1 receptor and the RNA substrate. Moreover, U2AF65 interacts directly and specifically with expanded CAG RNA, and serves as an adaptor to link expanded CAG RNA to NXF1 for RNA export. U2AF65 contains an N-terminal RS domain rich in arginine and serine, followed by a proline-rich segment and three C-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The N-terminal RS domain stabilizes the interaction of U2 snRNP with the branch point (BP) by contacting the branch region, and further promotes base pair interactions between U2 snRNA and the BP. The proline-rich segment mediates protein-protein interactions with the RRM domain of the small U2AF subunit (U2AF35 or U2AF1). The RRM1 and RRM2 are sufficient for specific RNA binding, while RRM3 is responsible for protein-protein interactions. The family also includes Splicing factor U2AF 50 kDa subunit (dU2AF50), the Drosophila ortholog of U2AF65. dU2AF50 functions as an essential pre-mRNA splicing factor in flies. It associates with intronless mRNAs and plays a significant and unexpected role in the nuclear export of a large number of intronless mRNAs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=co>46</td>
            <td id=cg>84</td>
            <td id=al>cd12566 RNA recognition motif 2 (RRM2) found in yeast multiple RNA-binding domain-containing protein 1 (MRD1) and similar proteins. This subgroup corresponds to the RRM2 of MRD1 which is encoded by a novel yeast gene MRD1 (multiple RNA-binding domain). It is well-conserved in yeast and its homologs exist in all eukaryotes. MRD1 is present in the nucleolus and the nucleoplasm. It interacts with the 35 S precursor rRNA (pre-rRNA) and U3 small nucleolar RNAs (snoRNAs). It is essential for the initial processing at the A0-A2 cleavage sites in the 35 S pre-rRNA. MRD1 contains 5 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), which may play an important structural role in organizing specific rRNA processing events. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>52</td>
            <td id=cr>14</td>
            <td id=al>TIGR01642 U2 snRNP auxilliary factor, large subunit, splicing factor. These splicing factors consist of an N-terminal arginine-rich low complexity domain followed by three tandem RNA recognition motifs (pfam00076). The well-characterized members of this family are auxilliary components of the U2 small nuclear ribonuclearprotein splicing factor (U2AF). These proteins are closely related to the CC1-like subfamily of splicing factors (TIGR01622). Members of this subfamily are found in plants, metazoa and fungi.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>53</td>
            <td id=cg>92</td>
            <td id=al>cd12771 RNA recognition motif 1 (RRM1) found in vertebrate Hu-antigen B (HuB). This subgroup corresponds to the RRM1 of HuB, also termed ELAV-like protein 2 (ELAV-2), or ELAV-like neuronal protein 1, or nervous system-specific RNA-binding protein Hel-N1 (Hel-N1), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads and is up-regulated during neuronal differentiation of embryonic carcinoma P19 cells. Like other Hu proteins, HuB contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>43</td>
            <td id=cb>76</td>
            <td id=al>cd12651 RNA recognition motif 2 (RRM2) found in Drosophila sex-lethal (SXL) and similar proteins. This subfamily corresponds to the RRM2 of the sex-lethal protein (SXL) which governs sexual differentiation and X chromosome dosage compensation in Drosophila melanogaster. It induces female-specific alternative splicing of the transformer (tra) pre-mRNA by binding to the tra uridine-rich polypyrimidine tract at the non-sex-specific 3' splice site during the sex-determination process. SXL binds also to its own pre-mRNA and promotes female-specific alternative splicing. SXL contains an N-terminal Gly/Asn-rich domain that may be responsible for the protein-protein interaction, and tandem RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), that show high preference to bind single-stranded, uridine-rich target RNA transcripts. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>49</td>
            <td id=cg>88</td>
            <td id=al>cd21601 RNA recognition motif 1 (RRM1) found in Saccharomyces cerevisiae protein PES4, protein MIP6 and similar proteins. The family includes PES4 (also called DNA polymerase epsilon suppressor 4) and MIP6 (also called MEX67-interacting protein 6), both of which are predicted RNA binding proteins that may act as regulators of late translation, protection, and mRNA localization. MIP6 acts as a novel factor for nuclear mRNA export, binds to both poly(A)+ RNA and nuclear pores. It interacts with MEX67. Members in this family contain four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>40</td>
            <td id=cb>75</td>
            <td id=al>cd12580 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A1 (hnRNP A1) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A1, also termed helix-destabilizing protein, or single-strand RNA-binding protein, or hnRNP core protein A1, an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A1 has been characterized as a splicing silencer, often acting in opposition to an activating hnRNP H. It silences exons when bound to exonic elements in the alternatively spliced transcripts of c-src, HIV, GRIN1, and beta-tropomyosin. hnRNP A1 can shuttle between the nucleus and the cytoplasm. Thus, it may be involved in transport of cellular RNAs, including the packaging of pre-mRNA into hnRNP particles and transport of poly A+ mRNA from the nucleus to the cytoplasm. The cytoplasmic hnRNP A1 has high affinity with AU-rich elements, whereas the nuclear hnRNP A1 has high affinity with a polypyrimidine stretch bordered by AG at the 3' ends of introns. hnRNP A1 is also involved in the replication of an RNA virus, such as mouse hepatitis virus (MHV), through an interaction with the transcription-regulatory region of viral RNA. Moreover, hnRNP A1, together with the scaffold protein septin 6, serves as host proteins to form a complex with NS5b and viral RNA, and further play important roles in the replication of Hepatitis C virus (HCV). hnRNP A1 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. The RRMs of hnRNP A1 play an important role in silencing the exon and the glycine-rich domain is responsible for protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>45</td>
            <td id=cg>85</td>
            <td id=al>cd21605 RNA recognition motif 1 (RRM1) found in Saccharomyces cerevisiae protein HRB1, G-strand-binding protein 2 (GBP2) and similar proteins. The family includes Saccharomyces cerevisiae protein HRB1 (also called protein TOM34) and GBP2, both of which are SR-like mRNA-binding proteins which shuttle from the nucleus to the cytoplasm when bound to the mature mRNA molecules. They act as quality control factors for spliced mRNAs. GBP2, also called RAP1 localization factor 6, is a single-strand telomeric DNA-binding protein that binds single-stranded telomeric sequences of the type (TG[1-3])n in vitro. It also binds to RNA. GBP2 influences the localization of RAP1 in the nuclei and plays a role in modulating telomere length. Members in this family contain three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>43</td>
            <td id=cg>81</td>
            <td id=al>cd12379 RNA recognition motif 2 (RRM2) found found in type I polyadenylate-binding proteins. This subfamily corresponds to the RRM2 of type I poly(A)-binding proteins (PABPs), highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in the regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. The family represents type I polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 1 (PABP-1 or PABPC1), polyadenylate-binding protein 3 (PABP-3 or PABPC3), polyadenylate-binding protein 4 (PABP-4 or APP-1 or iPABP), polyadenylate-binding protein 5 (PABP-5 or PABPC5), polyadenylate-binding protein 1-like (PABP-1-like or PABPC1L), polyadenylate-binding protein 1-like 2 (PABPC1L2 or RBM32), polyadenylate-binding protein 4-like (PABP-4-like or PABPC4L), yeast polyadenylate-binding protein, cytoplasmic and nuclear (PABP or ACBP-67), and similar proteins. PABP-1 is a ubiquitously expressed multifunctional protein that may play a role in 3' end formation of mRNA, translation initiation, mRNA stabilization, protection of poly(A) from nuclease activity, mRNA deadenylation, inhibition of mRNA decapping, and mRNP maturation. Although PABP-1 is thought to be a cytoplasmic protein, it is also found in the nucleus. PABP-1 may be involved in nucleocytoplasmic trafficking and utilization of mRNP particles. PABP-1 contains four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), a less well conserved linker region, and a proline-rich C-terminal conserved domain (CTD). PABP-3 is a testis-specific poly(A)-binding protein specifically expressed in round spermatids. It is mainly found in mammalian and may play an important role in the testis-specific regulation of mRNA homeostasis. PABP-3 shows significant sequence similarity to PABP-1. However, it binds to poly(A) with a lower affinity than PABP-1. Moreover, PABP-1 possesses an A-rich sequence in its 5'-UTR and allows binding of PABP and blockage of translation of its own mRNA. In contrast, PABP-3 lacks the A-rich sequence in its 5'-UTR. PABP-4 is an inducible poly(A)-binding protein (iPABP) that is primarily localized to the cytoplasm. It shows significant sequence similarity to PABP-1 as well. The RNA binding properties of PABP-1 and PABP-4 appear to be identical. PABP-5 is encoded by PABPC5 gene within the X-specific subinterval, and expressed in fetal brain and in a range of adult tissues in mammalian, such as ovary and testis. It may play an important role in germ cell development. Unlike other PABPs, PABP-5 contains only four RRMs, but lacks both the linker region and the CTD. PABP-1-like and PABP-1-like 2 are the orthologs of PABP-1. PABP-4-like is the ortholog of PABP-5. Their cellular functions remain unclear. The family also includes the yeast PABP, a conserved poly(A) binding protein containing poly(A) tails that can be attached to the 3'-ends of mRNAs. The yeast PABP and its homologs may play important roles in the initiation of translation and in mRNA decay. Like vertebrate PABP-1, the yeast PABP contains four RRMs, a linker region, and a proline-rich CTD as well. The first two RRMs are mainly responsible for specific binding to poly(A). The proline-rich region may be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=co>50</td>
            <td id=cg>84</td>
            <td id=al>cd12655 RNA recognition motif 3 (RRM3) found in vertebrate Hu-antigen C (HuC). This subgroup corresponds to the RRM3 of HuC, also termed ELAV-like protein 3 (ELAV-3), or paraneoplastic cerebellar degeneration-associated antigen, or paraneoplastic limbic encephalitis antigen 21 (PLE21), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. Like other Hu proteins, HuC contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). The AU-rich element binding of HuC can be inhibited by flavonoids. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12418 RNA recognition motif (RRM) found in the Aly/REF family. This subfamily corresponds to the RRM of  Aly/REF family which includes THO complex subunit 4 (THOC4, also termed Aly/REF), S6K1 Aly/REF-like target (SKAR, also termed PDIP3 or PDIP46) and similar proteins. THOC4 is an mRNA transporter protein with a well conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). It is involved in RNA transportation from the nucleus, and was initially identified as a transcription coactivator of LEF-1 and AML-1 for the TCRalpha enhancer function. In addition, THOC4 specifically binds to rhesus (RH) promoter in erythroid, and might be a novel transcription cofactor for erythroid-specific genes. SKAR shows high sequence homology with THOC4 and possesses one RRM as well. SKAR is widely expressed and localizes to the nucleus. It may be a critical player in the function of S6K1 in cell and organism growth control by binding the activated, hyperphosphorylated form of S6K1 but not S6K2. Furthermore, SKAR functions as a protein partner of the p50 subunit of DNA polymerase delta. In addition, SKAR may have particular importance in pancreatic beta cell size determination and insulin secretion. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>50</td>
            <td id=cg>98</td>
            <td id=al>cd12307 RNA recognition motif in nucleolar protein interacting with the FHA domain of pKI-67 (NIFK) and similar proteins. This subgroup corresponds to the RRM of NIFK and Nop15p. NIFK, also termed MKI67 FHA domain-interacting nucleolar phosphoprotein, or nucleolar phosphoprotein Nopp34, is a putative RNA-binding protein interacting with the forkhead associated (FHA) domain of pKi-67 antigen in a mitosis-specific and phosphorylation-dependent manner. It is nucleolar in interphase but associates with condensed mitotic chromosomes. This family also includes Saccharomyces cerevisiae YNL110C gene encoding ribosome biogenesis protein 15 (Nop15p), also termed nucleolar protein 15. Both, NIFK and Nop15p, contain an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>53</td>
            <td id=cg>95</td>
            <td id=al>cd12770 RNA recognition motif 1 (RRM1) found in vertebrate Hu-antigen D (HuD). This subgroup corresponds to the RRM1 of HuD, also termed ELAV-like protein 4 (ELAV-4), or paraneoplastic encephalomyelitis antigen HuD, one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuD has been implicated in various aspects of neuronal function, such as the commitment and differentiation of neuronal precursors as well as synaptic remodeling in mature neurons. HuD also functions as an important regulator of mRNA expression in neurons by interacting with AU-rich RNA element (ARE) and stabilizing multiple transcripts. Moreover, HuD regulates the nuclear processing/stability of N-myc pre-mRNA in neuroblastoma cells, as well as the neurite elongation and morphological differentiation. HuD specifically binds poly(A) RNA. Like other Hu proteins, HuD contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12608 RNA recognition motif 1 (RRM1) found in vertebrate RRM-containing coactivator activator/modulator (CoAA). This subgroup corresponds to the RRM1 of CoAA, also termed RNA-binding protein 14 (RBM14), or paraspeckle protein 2 (PSP2), or synaptotagmin-interacting protein (SYT-interacting protein), a heterogeneous nuclear ribonucleoprotein (hnRNP)-like protein identified as a nuclear receptor coactivator. It mediates transcriptional coactivation and RNA splicing effects in a promoter-preferential manner and is enhanced by thyroid hormone receptor-binding protein (TRBP). CoAA contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a TRBP-interacting domain. It stimulates transcription through its interactions with coactivators, such as TRBP and CREB-binding protein CBP/p300, via the TRBP-interacting domain and interaction with an RNA-containing complex, such as DNA-dependent protein kinase-poly(ADP-ribose) polymerase complexes, via the RRMs. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>50</td>
            <td id=cg>98</td>
            <td id=al>cd12417 RNA recognition motif (RRM) found in the scaffold attachment factor (SAFB) family. This subfamily corresponds to the RRM domain of the SAFB family, including scaffold attachment factor B1 (SAFB1), scaffold attachment factor B2 (SAFB2), SAFB-like transcriptional modulator (SLTM), and similar proteins, which are ubiquitously expressed. SAFB1, SAFB2 and SLTM have been implicated in many diverse cellular processes including cell growth and transformation, stress response, and apoptosis. They share high sequence similarities and all contain a scaffold attachment factor-box (SAF-box, also known as SAP domain) DNA-binding motif, an RNA recognition motif (RRM), also known as RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a region rich in glutamine and arginine residues. SAFB1 is a nuclear protein with a distribution similar to that of SLTM, but unlike that of SAFB2, which is also found in the cytoplasm. To a large extent, SAFB1 and SLTM might share similar functions, such as the inhibition of an oestrogen reporter gene. The additional cytoplasmic localization of SAFB2 implies that it could play additional roles in the cytoplasmic compartment which are distinct from the nuclear functions shared with SAFB1 and SLTM. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=al>cd12582 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein A3 (hnRNP A3) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP A3, a novel RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE) independently of hnRNP A2 and participates in the trafficking of A2RE-containing RNA. hnRNP A3 can shuttle between the nucleus and the cytoplasm. It contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>45</td>
            <td id=cb>77</td>
            <td id=al>cd12473 RNA recognition motif 2 (RRM2) found in vertebrate single-stranded DNA-binding protein MSSP-1. This subgroup corresponds to the RRM2 of MSSP-1, also termed RNA-binding motif, single-stranded-interacting protein 1 (RBMS1), or suppressor of CDC2 with RNA-binding motif 2 (SCR2). MSSP-1 is a double- and single-stranded DNA binding protein that belongs to the c-myc single-strand binding proteins (MSSP) family. It specifically recognizes the sequence CT(A/T)(A/T)T, and stimulates DNA replication in the system using SV40 DNA. MSSP-1 is identical with Scr2, a human protein which complements the defect of cdc2 kinase in Schizosaccharomyces pombe. MSSP-1 has been implied in regulating DNA replication, transcription, apoptosis induction, and cell-cycle movement, via the interaction with c-MYC, the product of protooncogene c-myc. MSSP-1 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), both of which are responsible for the specific DNA binding activity as well as induction of apoptosis. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>44</td>
            <td id=cb>78</td>
            <td id=al>cd12344 RNA recognition motif 1 (RRM1) found in tRNA selenocysteine-associated protein 1 (SECp43) and similar proteins. This subfamily corresponds to the RRM1 in tRNA selenocysteine-associated protein 1 (SECp43), yeast negative growth regulatory protein NGR1 (RBP1), yeast protein NAM8, and similar proteins. SECp43 is an RNA-binding protein associated specifically with eukaryotic selenocysteine tRNA [tRNA(Sec)]. It may play an adaptor role in the mechanism of selenocysteine insertion. SECp43 is located primarily in the nucleus and contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal polar/acidic region. Yeast proteins, NGR1 and NAM8, show high sequence similarity with SECp43. NGR1 is a putative glucose-repressible protein that binds both RNA and single-stranded DNA (ssDNA). It may function in regulating cell growth in early log phase, possibly through its participation in RNA metabolism. NGR1 contains three RRMs, two of which are followed by a glutamine-rich stretch that may be involved in transcriptional activity. In addition, NGR1 has an asparagine-rich region near the C-terminus which also harbors a methionine-rich region. NAM8 is a putative RNA-binding protein that acts as a suppressor of mitochondrial splicing deficiencies when overexpressed in yeast. It may be a non-essential component of the mitochondrial splicing machinery. NAM8 also contains three RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=co>50</td>
            <td id=cg>83</td>
            <td id=al>cd12656 RNA recognition motif 3 (RRM3) found in vertebrate Hu-antigen D (HuD). This subgroup corresponds to the RRM3 of HuD, also termed ELAV-like protein 4 (ELAV-4), or paraneoplastic encephalomyelitis antigen HuD, one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuD has been implicated in various aspects of neuronal function, such as the commitment and differentiation of neuronal precursors as well as synaptic remodeling in mature neurons. HuD also functions as an important regulator of mRNA expression in neurons by interacting with AU-rich RNA element (ARE) and stabilizing multiple transcripts. Moreover, HuD regulates the nuclear processing/stability of N-myc pre-mRNA in neuroblastoma cells. And it also regulates the neurite elongation and morphological differentiation. HuD specifically bound poly(A) RNA. Like other Hu proteins, HuD contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>50</td>
            <td id=cg>98</td>
            <td id=al>cd12283 RNA recognition motif 1 (RRM1) found in vertebrate RNA-binding protein 39 (RBM39) and similar proteins. This subfamily corresponds to the RRM1 of RNA-binding protein 39 (RBM39), RNA-binding protein 23 (RBM23) and similar proteins. RBM39 (also termed HCC1) is a nuclear autoantigen that contains an N-terminal arginine/serine rich (RS) motif and three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). An octapeptide sequence called the RS-ERK motif is repeated six times in the RS region of RBM39. Although the cellular function of RBM23 remains unclear, it shows high sequence homology to RBM39 and contains two RRMs. It may possibly function as a pre-mRNA splicing factor. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>50</td>
            <td id=cg>90</td>
            <td id=al>cd12392 RNA recognition motif 2 (RRM2) found in squamous cell carcinoma antigen recognized by T-cells 3 (SART3) and similar proteins. This subfamily corresponds to the RRM2 of SART3, also termed Tat-interacting protein of 110 kDa (Tip110), is an RNA-binding protein expressed in the nucleus of the majority of proliferating cells, including normal cells and malignant cells, but not in normal tissues except for the testes and fetal liver. It is involved in the regulation of mRNA splicing probably via its complex formation with RNA-binding protein with a serine-rich domain (RNPS1), a pre-mRNA-splicing factor. SART3 has also been identified as a nuclear Tat-interacting protein that regulates Tat transactivation activity through direct interaction and functions as an important cellular factor for HIV-1 gene expression and viral replication. In addition, SART3 is required for U6 snRNP targeting to Cajal bodies. It binds specifically and directly to the U6 snRNA, interacts transiently with the U6 and U4/U6 snRNPs, and promotes the reassembly of U4/U6 snRNPs after splicing in vitro. SART3 contains an N-terminal half-a-tetratricopeptide repeat (HAT)-rich domain, a nuclearlocalization signal (NLS) domain, and two C-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>50</td>
            <td id=cg>100</td>
            <td id=al>cd12272 RNA recognition motif 2 (RRM2) found in Arabidopsis thaliana phragmoplastin interacting protein 1 (PHIP1) and similar proteins. The CD corresponds to the RRM2 of PHIP1. A. thaliana PHIP1 and its homologs represent a novel class of plant-specific RNA-binding proteins that may play a unique role in the polarized mRNA transport to the vicinity of the cell plate. The family members consist of multiple functional domains, including a lysine-rich domain (KRD domain) that contains three nuclear localization motifs (KKKR/NK), two RNA recognition motifs (RRMs), and three CCHC-type zinc fingers. PHIP1 is a peripheral membrane protein and is localized at the cell plate during cytokinesis in plants. In addition to phragmoplastin, PHIP1 interacts with two Arabidopsis small GTP-binding proteins, Rop1 and Ran2. However, PHIP1 interacted only with the GTP-bound form of Rop1 but not the GDP-bound form. It also binds specifically to Ran2 mRNA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>44</td>
            <td id=cg>83</td>
            <td id=al>cd12361 RNA recognition motif 1 (RRM1) and 2 (RRM2) found in CELF/Bruno-like family of RNA binding proteins and plant flowering time control protein FCA. This subfamily corresponds to the RRM1 and RRM2 domains of the CUGBP1 and ETR-3-like factors (CELF) as well as plant flowering time control protein FCA. CELF, also termed BRUNOL (Bruno-like) proteins, is a family of structurally related RNA-binding proteins involved in regulation of pre-mRNA splicing in the nucleus, and control of mRNA translation and deadenylation in the cytoplasm. The family contains six members: CELF-1 (also known as BRUNOL-2, CUG-BP1, NAPOR, EDEN-BP), CELF-2 (also known as BRUNOL-3, ETR-3, CUG-BP2, NAPOR-2), CELF-3 (also known as BRUNOL-1, TNRC4, ETR-1, CAGH4, ER DA4), CELF-4 (BRUNOL-4), CELF-5 (BRUNOL-5) and CELF-6 (BRUNOL-6). They all contain three highly conserved RNA recognition motifs (RRMs), also known as RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains): two consecutive RRMs (RRM1 and RRM2) situated in the N-terminal region followed by a linker region and the third RRM (RRM3) close to the C-terminus of the protein. The low sequence conservation of the linker region is highly suggestive of a large variety in the co-factors that associate with the various CELF family members. Based on both, sequence similarity and function, the CELF family can be divided into two subfamilies, the first containing CELFs 1 and 2, and the second containing CELFs 3, 4, 5, and 6. The different CELF proteins may act through different sites on at least some substrates. Furthermore, CELF proteins may interact with each other in varying combinations to influence alternative splicing in different contexts. This subfamily also includes plant flowering time control protein FCA that functions in the posttranscriptional regulation of transcripts involved in the flowering process. FCA contains two RRMs, and a WW protein interaction domain.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>45</td>
            <td id=cb>76</td>
            <td id=al>cd12474 RNA recognition motif 2 (RRM2) found in vertebrate single-stranded DNA-binding protein MSSP-2. This subgroup corresponds to the RRM2 of MSSP-2, also termed RNA-binding motif, single-stranded-interacting protein 2 (RBMS2), or suppressor of CDC2 with RNA-binding motif 3 (SCR3). MSSP-2 is a double- and single-stranded DNA binding protein that belongs to the c-myc single-strand binding proteins (MSSP) family. It specifically recognizes the sequence T(C/A)TT, and stimulates DNA replication in the system using SV40 DNA. MSSP-2 is identical with Scr3, a human protein which complements the defect of cdc2 kinase in Schizosaccharomyces pombe. MSSP-2 has been implied in regulating DNA replication, transcription, apoptosis induction, and cell-cycle movement, via the interaction with C-MYC, the product of protooncogene c-myc. MSSP-2 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), both of which are responsible for the specific DNA binding activity as well as induction of apoptosis. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12337 RNA recognition motif 1 (RRM1) found in serine/arginine-rich splicing factor 4 (SRSF4) and similar proteins. This subfamily corresponds to the RRM1 in three serine/arginine (SR) proteins: serine/arginine-rich splicing factor 4 (SRSF4 or SRp75 or SFRS4), serine/arginine-rich splicing factor 5 (SRSF5 or SRp40 or SFRS5 or HRS), serine/arginine-rich splicing factor 6 (SRSF6 or SRp55). SRSF4 plays an important role in both, constitutive  and alternative, splicing of many pre-mRNAs. It can shuttle between the nucleus and cytoplasm. SRSF5 regulates both alternative splicing and basal splicing. It is the only SR protein efficiently selected from nuclear extracts (NE) by the splicing enhancer (ESE) and essential for enhancer activation. SRSF6 preferentially interacts with a number of purine-rich splicing enhancers (ESEs) to activate splicing of the ESE-containing exon. It is the only protein from HeLa nuclear extract or purified SR proteins that specifically binds B element RNA after UV irradiation. SRSF6 may also recognize different types of RNA sites. Members in this family contain two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a C-terminal RS domains rich in serine-arginine dipeptides. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=co>51</td>
            <td id=cg>100</td>
            <td id=al>cd12343 RNA recognition motif 1 (RRM1) and 2 (RRM2) found in RRM-containing coactivator activator/modulator (CoAA) and similar proteins. This subfamily corresponds to the RRM in CoAA (also known as RBM14 or PSP2) and RNA-binding protein 4 (RBM4). CoAA is a heterogeneous nuclear ribonucleoprotein (hnRNP)-like protein identified as a nuclear receptor coactivator. It mediates transcriptional coactivation and RNA splicing effects in a promoter-preferential manner, and is enhanced by thyroid hormone receptor-binding protein (TRBP). CoAA contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a TRBP-interacting domain. RBM4 is a ubiquitously expressed splicing factor with two isoforms, RBM4A (also known as Lark homolog) and RBM4B (also known as RBM30), which are very similar in structure and sequence. RBM4 may also function as a translational regulator of stress-associated mRNAs as well as play a role in micro-RNA-mediated gene regulation. RBM4 contains two N-terminal RRMs, a CCHC-type zinc finger, and three alanine-rich regions within their C-terminal regions. This family also includes Drosophila RNA-binding protein lark (Dlark), a homolog of human RBM4. It plays an important role in embryonic development and in the circadian regulation of adult eclosion. Dlark shares high sequence similarity with RBM4 at the N-terminal region. However, Dlark has three proline-rich segments instead of three alanine-rich segments within the C-terminal region. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>43</td>
            <td id=cb>76</td>
            <td id=al>cd12229 RNA recognition motif (RRM) found in ras GTPase-activating protein-binding protein G3BP1, G3BP2 and similar proteins. This subfamily corresponds to the RRM domain in the G3BP family of RNA-binding and SH3 domain-binding proteins. G3BP acts at the level of RNA metabolism in response to cell signaling, possibly as RNA transcript stabilizing factors or an RNase. Members include G3BP1, G3BP2 and similar proteins. These proteins associate directly with the SH3 domain of GTPase-activating protein (GAP), which functions as an inhibitor of Ras. They all contain an N-terminal nuclear transfer factor 2 (NTF2)-like domain, an acidic domain, a domain containing PXXP motif(s), an RNA recognition motif (RRM), and an Arg-Gly-rich region (RGG-rich region, or arginine methylation motif).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>40</td>
            <td id=cb>78</td>
            <td id=al>cd12329 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. This subfamily corresponds to the RRM2 of hnRNP D0, hnRNP A/B, hnRNP DL and similar proteins. hnRNP D0, a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP A/B is an RNA unwinding protein with a high affinity for G- followed by U-rich regions. It has also been identified as an APOBEC1-binding protein that interacts with apolipoprotein B (apoB) mRNA transcripts around the editing site and thus plays an important role in apoB mRNA editing. hnRNP DL (or hnRNP D-like) is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. All memembers in this family contain two putative RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glycine- and tyrosine-rich C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>46</td>
            <td id=cg>81</td>
            <td id=al>cd12244 RNA recognition motif 2 (RRM2) found in the c-myc gene single-strand binding proteins (MSSP) family. This subfamily corresponds to the RRM2 of c-myc gene single-strand binding proteins (MSSP) family, including single-stranded DNA-binding protein MSSP-1 (also termed RBMS1 or SCR2) and MSSP-2 (also termed RBMS2 or SCR3). All MSSP family members contain two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), both of which are responsible for the specific DNA binding activity. Both, MSSP-1 and -2, have been identified as protein factors binding to a putative DNA replication origin/transcriptional enhancer sequence present upstream from the human c-myc gene in both single- and double-stranded forms. Thus they have been implied in regulating DNA replication, transcription, apoptosis induction, and cell-cycle movement, via the interaction with C-MYC, the product of protooncogene c-myc. Moreover, they family includes a new member termed RNA-binding motif, single-stranded-interacting protein 3 (RBMS3), which is not a transcriptional regulator. RBMS3 binds with high affinity to A/U-rich stretches of RNA, and to A/T-rich DNA sequences, and functions as a regulator of cytoplasmic activity. In addition, a putative meiosis-specific RNA-binding protein termed sporulation-specific protein 5 (SPO5, or meiotic RNA-binding protein 1, or meiotically up-regulated gene 12 protein), encoded by Schizosaccharomyces pombe Spo5/Mug12 gene, is also included in this family. SPO5 is a novel meiosis I regulator that may function in the vicinity of the Mei2 dot. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>49</td>
            <td id=cg>97</td>
            <td id=al>cd12306 RNA recognition motif in type II polyadenylate-binding proteins. This subfamily corresponds to the RRM of type II polyadenylate-binding proteins (PABPs), including polyadenylate-binding protein 2 (PABP-2 or PABPN1), embryonic polyadenylate-binding protein 2 (ePABP-2 or PABPN1L) and similar proteins. PABPs are highly conserved proteins that bind to the poly(A) tail present at the 3' ends of most eukaryotic mRNAs. They have been implicated in the regulation of poly(A) tail length during the polyadenylation reaction, translation initiation, mRNA stabilization by influencing the rate of deadenylation and inhibition of mRNA decapping. ePABP-2 is predominantly located in the cytoplasm and PABP-2 is located in the nucleus. In contrast to the type I PABPs containing four copies of RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), the type II PABPs contains a single highly-conserved RRM. This subfamily also includes Saccharomyces cerevisiae RBP29 (SGN1, YIR001C) gene encoding cytoplasmic mRNA-binding protein Rbp29 that binds preferentially to poly(A). Although not essential for cell viability, Rbp29 plays a role in modulating the expression of cytoplasmic mRNA. Like other type II PABPs, Rbp29 contains one RRM only. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>50</td>
            <td id=cg>96</td>
            <td id=al>cd12347 RNA recognition motif (RRM) found in cyclophilin-33 (Cyp33) and similar proteins. This subfamily corresponds to the RRM of Cyp33, also termed peptidyl-prolyl cis-trans isomerase E (PPIase E), or cyclophilin E, or rotamase E. Cyp33 is a nuclear RNA-binding cyclophilin with an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a C-terminal PPIase domain. Cyp33 possesses RNA-binding activity and preferentially binds to polyribonucleotide polyA and polyU, but hardly to polyG and polyC. It binds specifically to mRNA, which can stimulate its PPIase activity. Moreover, Cyp33 interacts with the third plant homeodomain (PHD3) zinc finger cassette of the mixed lineage leukemia (MLL) proto-oncoprotein and a poly-A RNA sequence through its RRM domain. It further mediates downregulation of the expression of MLL target genes HOXC8, HOXA9, CDKN1B, and C-MYC, in a proline isomerase-dependent manner. Cyp33 also possesses a PPIase activity that catalyzes cis-trans isomerization of the peptide bond preceding a proline, which has been implicated in the stimulation of folding and conformational changes in folded and unfolded proteins. The PPIase activity can be inhibited by the immunosuppressive drug cyclosporin A. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>51</td>
            <td id=cg>97</td>
            <td id=al>cd12564 RNA recognition motif 1 (RRM1) found in RNA-binding protein 19 (RBM19) and similar proteins. This subgroup corresponds to the RRM1 of RBM19, also termed RNA-binding domain-1 (RBD-1), a nucleolar protein conserved in eukaryotes. It is involved in ribosome biogenesis by processing rRNA. In addition, it is essential for preimplantation development. RBM19 has a unique domain organization containing 6 conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>50</td>
            <td id=cg>96</td>
            <td id=al>cd21606 RNA recognition motif 2 (RRM2) found in Saccharomyces cerevisiae protein HRB1, G-strand-binding protein 2 (GBP2) and similar proteins. The family includes Saccharomyces cerevisiae protein HRB1 (also called protein TOM34) and GBP2, both of which are SR-like mRNA-binding proteins which shuttle from the nucleus to the cytoplasm when bound to the mature mRNA molecules. They act as quality control factors for spliced mRNAs. GBP2, also called RAP1 localization factor 6, is a single-strand telomeric DNA-binding protein that binds single-stranded telomeric sequences of the type (TG[1-3])n in vitro. It also binds to RNA. GBP2 influences the localization of RAP1 in the nuclei and plays a role in modulating telomere length. Members in this family contain three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the second RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>45</td>
            <td id=cb>75</td>
            <td id=al>cd12475 RNA recognition motif 2 (RRM2) found in vertebrate RNA-binding motif, single-stranded-interacting protein 3 (RBMS3). This subgroup corresponds to the RRM2 of RBMS3, a new member of the c-myc gene single-strand binding proteins (MSSP) family of DNA regulators. Unlike other MSSP proteins, RBMS3 is not a transcriptional regulator. It binds with high affinity to A/U-rich stretches of RNA, and to A/T-rich DNA sequences, and functions as a regulator of cytoplasmic activity. RBMS3 contain two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and its C-terminal region is acidic and enriched in prolines, glutamines and threonines. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>52</td>
            <td id=cg>89</td>
            <td id=al>cd12223 RNA recognition motif (RRM) found in U2-associated protein SR140 and similar proteins. This subgroup corresponds to the RRM of SR140 (also termed U2 snRNP-associated SURP motif-containing protein orU2SURP, or 140 kDa Ser/Arg-rich domain protein) which is a putative splicing factor mainly found in higher eukaryotes. Although it is initially identified as one of the 17S U2 snRNP-associated proteins, the molecular and physiological function of SR140 remains unclear. SR140 contains an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), a SWAP/SURP domain that is found in a number of pre-mRNA splicing factors in the middle region, and a C-terminal arginine/serine-rich domain (RS domain).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>37</td>
            <td id=cb>75</td>
            <td id=al>cd12391 RNA recognition motif 1 (RRM1) found in squamous cell carcinoma antigen recognized by T-cells 3 (SART3) and similar proteins. This subfamily corresponds to the RRM1 of SART3, also termed Tat-interacting protein of 110 kDa (Tip110), an RNA-binding protein expressed in the nucleus of the majority of proliferating cells, including normal cells and malignant cells, but not in normal tissues except for the testes and fetal liver. It is involved in the regulation of mRNA splicing probably via its complex formation with RNA-binding protein with a serine-rich domain (RNPS1), a pre-mRNA-splicing factor. SART3 has also been identified as a nuclear Tat-interacting protein that regulates Tat transactivation activity through direct interaction and functions as an important cellular factor for HIV-1 gene expression and viral replication. In addition, SART3 is required for U6 snRNP targeting to Cajal bodies. It binds specifically and directly to the U6 snRNA, interacts transiently with the U6 and U4/U6 snRNPs, and promotes the reassembly of U4/U6 snRNPs after splicing in vitro. SART3 contains an N-terminal half-a-tetratricopeptide repeat (HAT)-rich domain, a nuclearlocalization signal (NLS) domain, and two C-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>48</td>
            <td id=cg>88</td>
            <td id=al>cd12451 RNA recognition motif 2 (RRM2) found in nucleolin-like proteins mainly from plants. This subfamily corresponds to the RRM2 of a group of plant nucleolin-like proteins, including nucleolin 1 (also termed protein nucleolin like 1) and nucleolin 2 (also termed protein nucleolin like 2, or protein parallel like 1). They play roles in the regulation of ribosome synthesis and in the growth and development of plants. Like yeast nucleolin, nucleolin-like proteins possess two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains).  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>56</td>
            <td id=cg>97</td>
            <td id=al>cd21620 RNA recognition motif 1 (RRM1) found in Schizosaccharomyces pombe meiotically up-regulated gene 28 protein (Mug28) and similar proteins. Mug28 is a meiosis-specific protein that regulates spore wall formation. Members in this family contain three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the first RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>50</td>
            <td id=cg>97</td>
            <td id=al>cd12336 RNA recognition motif (RRM) found in RNA-binding protein 7 (RBM7) and similar proteins. This subfamily corresponds to the RRM of RBM7, RBM11 and their eukaryotic homologous. RBM7 is an ubiquitously expressed pre-mRNA splicing factor that enhances messenger RNA (mRNA) splicing in a cell-specific manner or in a certain developmental process, such as spermatogenesis. It interacts with splicing factors SAP145 (the spliceosomal splicing factor 3b subunit 2) and SRp20, and may play a more specific role in meiosis entry and progression. Together with additional testis-specific RNA-binding proteins, RBM7 may regulate the splicing of specific pre-mRNA species that are important in the meiotic cell cycle. RBM11 is a novel tissue-specific splicing regulator that is selectively expressed in brain, cerebellum and testis, and to a lower extent in kidney. It is localized in the nucleoplasm and enriched in SRSF2-containing splicing speckles. It may play a role in the modulation of alternative splicing during neuron and germ cell differentiation. Both, RBM7 and RBM11, contain an N-terminal RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and a region lacking known homology at the C-terminus. The RRM is responsible for RNA binding, whereas the C-terminal region permits nuclear localization and homodimerization. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>43</td>
            <td id=cg>80</td>
            <td id=al>cd12552 RNA recognition motif in yeast ribosome biogenesis protein 15 (Nop15p) and similar proteins. This subgroup corresponds to the RRM of Nop15p, also termed nucleolar protein 15, which is encoded by YNL110C from Saccharomyces cerevisiae, and localizes to the nucleoplasm and nucleolus. Nop15p has been identified as a component of a pre-60S particle. It interacts with RNA components of the early pre-60S particles. Furthermore, Nop15p binds directly to a pre-rRNA transcript in vitro and is required for pre-rRNA processing. It functions as a ribosome synthesis factor required for the 5' to 3' exonuclease digestion that generates the 5' end of the major, short form of the 5.8S rRNA as well as for processing of 27SB to 7S pre-rRNA. Nop15p also play a specific role in cell cycle progression. Nop15p contains an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>54</td>
            <td id=cg>97</td>
            <td id=al>cd12345 RNA recognition motif 2 (RRM2) found in tRNA selenocysteine-associated protein 1 (SECp43) and similar proteins. This subfamily corresponds to the RRM2 in tRNA selenocysteine-associated protein 1 (SECp43), yeast negative growth regulatory protein NGR1 (RBP1), yeast protein NAM8, and similar proteins. SECp43 is an RNA-binding protein associated specifically with eukaryotic selenocysteine tRNA [tRNA(Sec)]. It may play an adaptor role in the mechanism of selenocysteine insertion. SECp43 is located primarily in the nucleus and contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal polar/acidic region. Yeast proteins, NGR1 and NAM8, show high sequence similarity with SECp43. NGR1 is a putative glucose-repressible protein that binds both RNA and single-stranded DNA (ssDNA). It may function in regulating cell growth in early log phase, possibly through its participation in RNA metabolism. NGR1 contains three RRMs, two of which are followed by a glutamine-rich stretch that may be involved in transcriptional activity. In addition, NGR1 has an asparagine-rich region near the C-terminus which also harbors a methionine-rich region. NAM8 is a putative RNA-binding protein that acts as a suppressor of mitochondrial splicing deficiencies when overexpressed in yeast. It may be a non-essential component of the mitochondrial splicing machinery. NAM8 also contains three RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>43</td>
            <td id=cb>74</td>
            <td id=al>cd12464 RNA recognition motif (RRM) found in ras GTPase-activating protein-binding protein 2 (G3BP2) and similar proteins. This subgroup corresponds to the RRM of G3BP2, also termed GAP SH3 domain-binding protein 2, a cytoplasmic protein that interacts with both IkappaBalpha and IkappaBalpha/NF-kappaB complexes, indicating that G3BP2 may play a role in the control of nucleocytoplasmic distribution of IkappaBalpha and cytoplasmic anchoring of the IkappaBalpha/NF-kappaB complex. G3BP2 contains an N-terminal nuclear transfer factor 2 (NTF2)-like domain, an acidic domain, a domain containing five PXXP motifs, an RNA recognition motif (RRM domain), and an Arg-Gly-rich region (RGG-rich region, or arginine methylation motif). It binds to the SH3 domain of RasGAP, a multi-functional protein controlling Ras activity, through its N-terminal NTF2-like domain. The acidic domain is sufficient for the interaction of G3BP2 with the IkappaBalpha cytoplasmic retention sequence. Furthermore, G3BP2 might influence stability or translational efficiency of particular mRNAs by binding to RNA-containing structures within the cytoplasm through its RNA-binding domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=al>cd12614 RNA recognition motif 1 (RRM1) found in yeast nuclear and cytoplasmic polyadenylated RNA-binding protein PUB1 and similar proteins. This subgroup corresponds to the RRM1 of yeast protein PUB1, also termed ARS consensus-binding protein ACBP-60, or poly uridylate-binding protein, or poly(U)-binding protein. PUB1 has been identified as both, a heterogeneous nuclear RNA-binding protein (hnRNP) and a cytoplasmic mRNA-binding protein (mRNP), which may be stably bound to a translationally inactive subpopulation of mRNAs within the cytoplasm. It is distributed in both, the nucleus and the cytoplasm, and binds to poly(A)+ RNA (mRNA or pre-mRNA). Although it is one of the major cellular proteins cross-linked by UV light to polyadenylated RNAs in vivo, PUB1 is nonessential for cell growth in yeast. PUB1 also binds to T-rich single stranded DNA (ssDNA); however, there is no strong evidence implicating PUB1 in the mechanism of DNA replication. PUB1 contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a GAR motif (glycine and arginine rich stretch) that is located between RRM2 and RRM3. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>48</td>
            <td id=cg>83</td>
            <td id=al>cd12775 RNA recognition motif 2 (RRM2) found in vertebrate Hu-antigen B (HuB). This subgroup corresponds to the RRM2 of HuB, also termed ELAV-like protein 2 (ELAV-2), or ELAV-like neuronal protein 1, or nervous system-specific RNA-binding protein Hel-N1 (Hel-N1), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. It is up-regulated during neuronal differentiation of embryonic carcinoma P19 cells. Like other Hu proteins, HuB contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=al>cd12577 RNA recognition motif 1 (RRM1) found in yeast nuclear polyadenylated RNA-binding protein 4 (Hrp1p or Nab4p) and similar proteins. This subfamily corresponds to the RRM1 of Hrp1p and similar proteins. Hrp1p or Nab4p, also termed cleavage factor IB (CFIB), is a sequence-specific trans-acting factor that is essential for mRNA 3'-end formation in yeast Saccharomyces cerevisiae. It can be UV cross-linked to RNA and specifically recognizes the (UA)6 RNA element required for both, the cleavage and poly(A) addition, steps. Moreover, Hrp1p can shuttle between the nucleus and the cytoplasm, and play an additional role in the export of mRNAs to the cytoplasm. Hrp1p also interacts with Rna15p and Rna14p, two components of CF1A. In addition, Hrp1p functions as a factor directly involved in modulating the activity of the nonsense-mediated mRNA decay (NMD) pathway. It binds specifically to a downstream sequence element (DSE)-containing RNA and interacts with Upf1p, a component of the surveillance complex, further triggering the NMD pathway. Hrp1p contains two central RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an arginine-glycine-rich region harboring repeats of the sequence RGGF/Y. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>56</td>
            <td id=cg>90</td>
            <td id=al>cd12390 RNA recognition motif 3 (RRM3) found in ribonucleoprotein PTB-binding raver-1, raver-2 and similar proteins. This subfamily corresponds to the RRM3 of raver-1 and raver-2. Raver-1 is a ubiquitously expressed heterogeneous nuclear ribonucleoprotein (hnRNP) that serves as a co-repressor of the nucleoplasmic splicing repressor polypyrimidine tract-binding protein (PTB)-directed splicing of select mRNAs. It shuttles between the cytoplasm and the nucleus and can accumulate in the perinucleolar compartment, a dynamic nuclear substructure that harbors PTB. Raver-1 also modulates focal adhesion assembly by binding to the cytoskeletal proteins, including alpha-actinin, vinculin, and metavinculin (an alternatively spliced isoform of vinculin) at adhesion complexes, particularly in differentiated muscle tissue. Raver-2 is a novel member of the heterogeneous nuclear ribonucleoprotein (hnRNP) family. It shows high sequence homology to raver-1. Raver-2 exerts a spatio-temporal expression pattern during embryogenesis and is mainly limited to differentiated neurons and glia cells. Although it displays nucleo-cytoplasmic shuttling in heterokaryons, raver2 localizes to the nucleus in glia cells and neurons. Raver-2 can interact with PTB and may participate in PTB-mediated RNA-processing. However, there is no evidence indicating that raver-2 can bind to cytoplasmic proteins. Both, raver-1 and raver-2, contain three N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), two putative nuclear localization signals (NLS) at the N- and C-termini, a central leucine-rich region, and a C-terminal region harboring two [SG][IL]LGxxP motifs. They binds to RNA through the RRMs. In addition, the two [SG][IL]LGxxP motifs serve as the PTB-binding motifs in raver1. However, raver-2 interacts with PTB through the SLLGEPP motif only. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>54</td>
            <td id=cg>98</td>
            <td id=al>cd12355 RNA recognition motif (RRM) found in eukaryotic RNA-binding protein 18 and similar proteins. This subfamily corresponds to the RRM of RBM18, a putative RNA-binding protein containing a well-conserved RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain). The biological role of RBM18 remains unclear. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>44</td>
            <td id=cb>76</td>
            <td id=al>cd12774 RNA recognition motif 2 (RRM2) found in vertebrate Hu-antigen D (HuD). This subgroup corresponds to the RRM2 of HuD, also termed ELAV-like protein 4 (ELAV-4), or paraneoplastic encephalomyelitis antigen HuD, one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. HuD has been implicated in various aspects of neuronal function, such as the commitment and differentiation of neuronal precursors as well as synaptic remodeling in mature neurons. HuD also functions as an important regulator of mRNA expression in neurons by interacting with AU-rich RNA element (ARE) and stabilizing multiple transcripts. Moreover, HuD regulates the nuclear processing/stability of N-myc pre-mRNA in neuroblastoma cells and also regulates the neurite elongation and morphological differentiation. HuD specifically binds poly(A) RNA. Like other Hu proteins, HuD contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>53</td>
            <td id=cg>95</td>
            <td id=al>cd12649 RNA recognition motif 1 (RRM1) found in Drosophila sex-lethal (SXL) and similar proteins. This subfamily corresponds to the RRM1 of SXL which governs sexual differentiation and X chromosome dosage compensation in Drosophila melanogaster. It induces female-specific alternative splicing of the transformer (tra) pre-mRNA by binding to the tra uridine-rich polypyrimidine tract at the non-sex-specific 3' splice site during the sex-determination process. SXL binds also to its own pre-mRNA and promotes female-specific alternative splicing. SXL contains an N-terminal Gly/Asn-rich domain that may be responsible for the protein-protein interaction, and tandem RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), that show high preference to bind single-stranded, uridine-rich target RNA transcripts. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=co>45</td>
            <td id=cg>83</td>
            <td id=al>cd12482 RNA recognition motif 1 (RRM1) found in vertebrate heterogeneous nuclear ribonucleoprotein R (hnRNP R). This subgroup corresponds to the RRM1 of hnRNP R, which is a ubiquitously expressed nuclear RNA-binding protein that specifically binds mRNAs with a preference for poly(U) stretches. Upon binding of RNA, hnRNP R forms oligomers, most probably dimers. hnRNP R has been implicated in mRNA processing and mRNA transport, and also acts as a regulator to modify binding to ribosomes and RNA translation. It is predominantly located in axons of motor neurons and to a much lower degree in sensory axons. In axons of motor neurons, it also functions as a cytosolic protein and interacts with wild type of survival motor neuron (SMN) proteins directly, further providing a molecular link between SMN and the spliceosome. Moreover, hnRNP R plays an important role in neural differentiation and development, and in retinal development and light-elicited cellular activities. hnRNP R contains an acidic auxiliary N-terminal region, followed by two well defined and one degenerated RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal RGG motif; it binds RNA through its RRM domains. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=co>44</td>
            <td id=cb>77</td>
            <td id=al>cd12335 RNA recognition motif 2 (RRM2) found in splicing factor 3B subunit 4 (SF3B4) and similar proteins. This subfamily corresponds to the RRM2 of SF3B4, also termed pre-mRNA-splicing factor SF3b 49 kDa (SF3b50), or spliceosome-associated protein 49 (SAP 49). SF3B4 is a component of the multiprotein complex splicing factor 3b (SF3B), an integral part of the U2 small nuclear ribonucleoprotein (snRNP) and the U11/U12 di-snRNP. SF3B is essential for the accurate excision of introns from pre-messenger RNA, and is involved in the recognition of the pre-mRNA's branch site within the major and minor spliceosomes. SF3B4 functions to tether U2 snRNP with pre-mRNA at the branch site during spliceosome assembly. It is an evolutionarily highly conserved protein with orthologs across diverse species. SF3B4 contains two closely adjacent N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). It binds directly to pre-mRNA and also interacts directly and highly specifically with another SF3B subunit called SAP 145. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>45</td>
            <td id=cb>78</td>
            <td id=al>cd12483 RNA recognition motif 1 (RRM1) found in vertebrate heterogeneous nuclear ribonucleoprotein Q (hnRNP Q).  This subgroup corresponds to the RRM1 of hnRNP Q, also termed glycine- and tyrosine-rich RNA-binding protein (GRY-RBP), or NS1-associated protein 1 (NASP1), or synaptotagmin-binding, cytoplasmic RNA-interacting protein (SYNCRIP). It is a ubiquitously expressed nuclear RNA-binding protein identified as a component of the spliceosome complex, as well as a component of the apobec-1 editosome. As an alternatively spliced version of NSAP, it acts as an interaction partner of a multifunctional protein required for viral replication, and is implicated in the regulation of specific mRNA transport. hnRNP Q has also been identified as SYNCRIP, a dual functional protein participating in both viral RNA replication and translation. As a synaptotagmin-binding protein, hnRNP Q plays a putative role in organelle-based mRNA transport along the cytoskeleton. Moreover, hnRNP Q has been found in protein complexes involved in translationally coupled mRNA turnover and mRNA splicing. It functions as a wild-type survival motor neuron (SMN)-binding protein that may participate in pre-mRNA splicing and modulate mRNA transport along microtubuli. hnRNP Q contains an acidic auxiliary N-terminal region, followed by two well-defined and one degenerated RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal RGG motif; hnRNP Q binds RNA through its RRM domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>43</td>
            <td id=cg>84</td>
            <td id=al>cd12365 RNA recognition motif (RRM) found in RNA-binding protein with serine-rich domain 1 (RNPS1) and similar proteins. This subfamily corresponds to the RRM of RNPS1 and its eukaryotic homologs. RNPS1, also termed RNA-binding protein prevalent during the S phase, or SR-related protein LDC2, was originally characterized as a general pre-mRNA splicing activator, which activates both constitutive and alternative splicing of pre-mRNA in vitro.It has been identified as a protein component of the splicing-dependent mRNP complex, or exon-exon junction complex (EJC), and is directly involved in mRNA surveillance. Furthermore, RNPS1 is a splicing regulator whose activator function is controlled in part by CK2 (casein kinase II) protein kinase phosphorylation. It can also function as a squamous-cell carcinoma antigen recognized by T cells-3 (SART3)-binding protein, and is involved in the regulation of mRNA splicing. RNPS1 contains an N-terminal serine-rich (S) domain, a central RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), and the C-terminal arginine/serine/proline-rich (RS/P) domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>50</td>
            <td id=cg>92</td>
            <td id=al>cd12639 RNA recognition motif 2 (RRM2) found in CUGBP Elav-like family member CELF-3, CELF-4, CELF-5, CELF-6 and similar proteins. This subgroup corresponds to the RRM3 of CELF-3, CELF-4, CELF-5, and CELF-6, all of which belong to the CUGBP1 and ETR-3-like factors (CELF) or BRUNOL (Bruno-like) family of RNA-binding proteins that display dual nuclear and cytoplasmic localizations and have been implicated in the regulation of pre-mRNA splicing and in the control of mRNA translation and deadenylation. CELF-3, expressed in brain and testis only, is also known as bruno-like protein 1 (BRUNOL-1), or CAG repeat protein 4, or CUG-BP- and ETR-3-like factor 3, or embryonic lethal abnormal vision (ELAV)-type RNA-binding protein 1 (ETR-1), or expanded repeat domain protein CAG/CTG 4, or trinucleotide repeat-containing gene 4 protein (TNRC4). It plays an important role in the pathogenesis of tauopathies. CELF-3 contains three highly conserved RNA recognition motifs (RRMs), also known as RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains): two consecutive RRMs (RRM1 and RRM2) situated in the N-terminal region followed by a linker region and the third RRM (RRM3) close to the C-terminus of the protein.The effect of CELF-3 on tau splicing is mediated mainly by the RNA-binding activity of RRM2. The divergent linker region might mediate the interaction of CELF-3 with other proteins regulating its activity or involved in target recognition. CELF-4, highly expressed throughout the brain and in glandular tissues, moderately expressed in heart, skeletal muscle, and liver, is also known as bruno-like protein 4 (BRUNOL-4), or CUG-BP- and ETR-3-like factor 4. Like CELF-3, CELF-4 also contains three highly conserved RRMs. The splicing activation or repression activity of CELF-4 on some specific substrates is mediated by its RRM1/RRM2. Both, RRM1 and RRM2 of CELF-4, can activate cardiac troponin T (cTNT) exon 5 inclusion. CELF-5, expressed in brain, is also known as bruno-like protein 5 (BRUNOL-5), or CUG-BP- and ETR-3-like factor 5. Although its biological role remains unclear, CELF-5 shares same domain architecture with CELF-3. CELF-6, strongly expressed in kidney, brain, and testis, is also known as bruno-like protein 6 (BRUNOL-6), or CUG-BP- and ETR-3-like factor 6. It activates exon inclusion of a cardiac troponin T minigene in transient transfection assays in an muscle-specific splicing enhancer (MSE)-dependent manner and can activate inclusion via multiple copies of a single element, MSE2. CELF-6 also promotes skipping of exon 11 of insulin receptor, a known target of CELF activity that is expressed in kidney. In addition to three highly conserved RRMs, CELF-6 also possesses numerous potential phosphorylation sites, a potential nuclear localization signal (NLS) at the C terminus, and an alanine-rich region within the divergent linker region. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>40</td>
            <td id=cb>77</td>
            <td id=al>cd12758 RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein D-like (hnRNP D-like or hnRNP DL) and similar proteins. This subgroup corresponds to the RRM1 of hnRNP DL (or hnRNP D-like), also termed AU-rich element RNA-binding factor, or JKT41-binding protein (protein laAUF1 or JKTBP), which is a dual functional protein that possesses DNA- and RNA-binding properties. It has been implicated in mRNA biogenesis at the transcriptional and post-transcriptional levels. hnRNP DL binds single-stranded DNA (ssDNA) or double-stranded DNA (dsDNA) in a non-sequencespecific manner, and interacts with poly(G) and poly(A) tenaciously. It contains two putative two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glycine- and tyrosine-rich C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>44</td>
            <td id=cb>79</td>
            <td id=al>cd12776 RNA recognition motif 2 (RRM2) found in vertebrate Hu-antigen C (HuC). This subgroup corresponds to the RRM2 of HuC, also termed ELAV-like protein 3 (ELAV-3), or paraneoplastic cerebellar degeneration-associated antigen, or paraneoplastic limbic encephalitis antigen 21 (PLE21), one of the neuronal members of the Hu family. The neuronal Hu proteins play important roles in neuronal differentiation, plasticity and memory. Like other Hu proteins, HuC contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an AU-rich RNA element (ARE). The AU-rich element binding of HuC can be inhibited by flavonoids. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>43</td>
            <td id=cb>79</td>
            <td id=al>cd12652 RNA recognition motif 2 (RRM2) found in the Hu proteins family. This subfamily corresponds to the RRM2 of Hu proteins family which represents a group of RNA-binding proteins involved in diverse biological processes. Since the Hu proteins share high homology with the Drosophila embryonic lethal abnormal vision (ELAV) protein, the Hu family is sometimes referred to as the ELAV family. Drosophila ELAV is exclusively expressed in neurons and is required for the correct differentiation and survival of neurons in flies. The neuronal members of the Hu family include Hu-antigen B (HuB or ELAV-2 or Hel-N1), Hu-antigen C (HuC or ELAV-3 or PLE21), and Hu-antigen D (HuD or ELAV-4), which play important roles in neuronal differentiation, plasticity and memory. HuB is also expressed in gonads. Hu-antigen R (HuR or ELAV-1 or HuA) is the ubiquitously expressed Hu family member. It has a variety of biological functions mostly related to the regulation of cellular response to DNA damage and other types of stress. Moreover, HuR has an anti-apoptotic function during early cell stress response. It binds to mRNAs and enhances the expression of several anti-apoptotic proteins, such as p21waf1, p53, and prothymosin alpha. HuR also has pro-apoptotic function by promoting apoptosis when cell death is unavoidable. Furthermore, HuR may be important in muscle differentiation, adipogenesis, suppression of inflammatory response and modulation of gene expression in response to chronic ethanol exposure and amino acid starvation. Hu proteins perform their cytoplasmic and nuclear molecular functions by coordinately regulating functionally related mRNAs. In the cytoplasm, Hu proteins recognize and bind to AU-rich RNA elements (AREs) in the 3' untranslated regions (UTRs) of certain target mRNAs, such as GAP-43, vascular epithelial growth factor (VEGF), the glucose transporter GLUT1, eotaxin and c-fos, and stabilize those ARE-containing mRNAs. They also bind and regulate the translation of some target mRNAs, such as neurofilament M, GLUT1, and p27. In the nucleus, Hu proteins function as regulators of polyadenylation and alternative splicing. Each Hu protein contains three RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). RRM1 and RRM2 may cooperate in binding to an ARE. RRM3 may help to maintain the stability of the RNA-protein complex, and might also bind to poly(A) tails or be involved in protein-protein interactions. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=co>48</td>
            <td id=cg>98</td>
            <td id=al>cd12354 RNA recognition motif 2 (RRM2) found in granule-associated RNA binding proteins (p40-TIA-1 and TIAR), and yeast nuclear and cytoplasmic polyadenylated RNA-binding protein PUB1. This subfamily corresponds to the RRM3 of TIA-1, TIAR, and PUB1. Nucleolysin TIA-1 isoform p40 (p40-TIA-1 or TIA-1) and nucleolysin TIA-1-related protein (TIAR) are granule-associated RNA binding proteins involved in inducing apoptosis in cytotoxic lymphocyte (CTL) target cells. They share high sequence similarity and are expressed in a wide variety of cell types. TIA-1 can be phosphorylated by a serine/threonine kinase that is activated during Fas-mediated apoptosis.TIAR is mainly localized in the nucleus of hematopoietic and nonhematopoietic cells. It is translocated from the nucleus to the cytoplasm in response to exogenous triggers of apoptosis. Both TIA-1 and TIAR bind specifically to poly(A) but not to poly(C) homopolymers. They are composed of three N-terminal highly homologous RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a glutamine-rich C-terminal auxiliary domain containing a lysosome-targeting motif. TIA-1 and TIAR interact with RNAs containing short stretches of uridylates and their RRM2 can mediate the specific binding to uridylate-rich RNAs. The C-terminal auxiliary domain may be responsible for interacting with other proteins. In addition, TIA-1 and TIAR share a potential serine protease-cleavage site (Phe-Val-Arg) localized at the junction between their RNA binding domains and their C-terminal auxiliary domains. This subfamily also includes a yeast nuclear and cytoplasmic polyadenylated RNA-binding protein PUB1, termed ARS consensus-binding protein ACBP-60, or poly uridylate-binding protein, or poly(U)-binding protein, which has been identified as both a heterogeneous nuclear RNA-binding protein (hnRNP) and a cytoplasmic mRNA-binding protein (mRNP). It may be stably bound to a translationally inactive subpopulation of mRNAs within the cytoplasm. PUB1 is distributed in both, the nucleus and the cytoplasm, and binds to poly(A)+ RNA (mRNA or pre-mRNA). Although it is one of the major cellular proteins cross-linked by UV light to polyadenylated RNAs in vivo, PUB1 is nonessential for cell growth in yeast. PUB1 also binds to T-rich single stranded DNA (ssDNA); however, there is no strong evidence implicating PUB1 in the mechanism of DNA replication. PUB1 contains three RRMs, and a GAR motif (glycine and arginine rich stretch) that is located between RRM2 and RRM3. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=co>44</td>
            <td id=cb>77</td>
            <td id=al>cd12415 RNA recognition motif 3 (RRM3) found in RNA-binding protein 28 (RBM28) and similar proteins. This subfamily corresponds to the RRM3 of RBM28 and Nop4p. RBM28 is a specific nucleolar component of the spliceosomal small nuclear ribonucleoproteins (snRNPs), possibly coordinating their transition through the nucleolus. It specifically associates with U1, U2, U4, U5, and U6 small nuclear RNAs (snRNAs), and may play a role in the maturation of both small nuclear and ribosomal RNAs. RBM28 has four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and an extremely acidic region between RRM2 and RRM3. The family also includes nucleolar protein 4 (Nop4p or Nop77p) encoded by YPL043W from Saccharomyces cerevisiae. It is an essential nucleolar protein involved in processing and maturation of 27S pre-rRNA and biogenesis of 60S ribosomal subunits. Nop4p also contains four RRMs.  </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>40</td>
            <td id=cb>78</td>
            <td id=al>cd12583 RNA recognition motif 2 (RRM2) found in heterogeneous nuclear ribonucleoprotein D0 (hnRNP D0) and similar proteins. This subgroup corresponds to the RRM2 of hnRNP D0, also termed AU-rich element RNA-binding protein 1, a UUAG-specific nuclear RNA binding protein that may be involved in pre-mRNA splicing and telomere elongation. hnRNP D0 contains two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), in the middle and an RGG box rich in glycine and arginine residues in the C-terminal part. Each of RRMs can bind solely to the UUAG sequence specifically. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>40</td>
            <td id=cb>79</td>
            <td id=al>cd12323 RNA recognition motif 2 (RRM2) found in RNA-binding protein Musashi homologs Musashi-1, Musashi-2 and similar proteins. This subfamily corresponds to the RRM2.in Musashi-1 (also termed Msi1), a neural RNA-binding protein putatively expressed in central nervous system (CNS) stem cells and neural progenitor cells, and associated with asymmetric divisions in neural progenitor cells. It is evolutionarily conserved from invertebrates to vertebrates. Musashi-1 is a homolog of Drosophila Musashi and Xenopus laevis nervous system-specific RNP protein-1 (Nrp-1). It has been implicated in the maintenance of the stem-cell state, differentiation, and tumorigenesis. It translationally regulates the expression of a mammalian numb gene by binding to the 3'-untranslated region of mRNA of Numb, encoding a membrane-associated inhibitor of Notch signaling, and further influences neural development. Moreover, Musashi-1 represses translation by interacting with the poly(A)-binding protein and competes for binding of the eukaryotic initiation factor-4G (eIF-4G). Musashi-2 (also termed Msi2) has been identified as a regulator of the hematopoietic stem cell (HSC) compartment and of leukemic stem cells after transplantation of cells with loss and gain of function of the gene. It influences proliferation and differentiation of HSCs and myeloid progenitors, and further modulates normal hematopoiesis and promotes aggressive myeloid leukemia. Both, Musashi-1 and Musashi-2, contain two conserved N-terminal tandem RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), along with other domains of unknown function. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=al>cd12578 RNA recognition motif 1 (RRM1) found in heterogeneous nuclear ribonucleoprotein A subfamily. This subfamily corresponds to the RRM1 in hnRNP A0, hnRNP A1, hnRNP A2/B1, hnRNP A3 and similar proteins. hnRNP A0 is a low abundance hnRNP protein that has been implicated in mRNA stability in mammalian cells. It has been identified as the substrate for MAPKAP-K2 and may be involved in the lipopolysaccharide (LPS)-induced post-transcriptional regulation of tumor necrosis factor-alpha (TNF-alpha), cyclooxygenase 2 (COX-2) and macrophage inflammatory protein 2 (MIP-2). hnRNP A1 is an abundant eukaryotic nuclear RNA-binding protein that may modulate splice site selection in pre-mRNA splicing. hnRNP A2/B1 is an RNA trafficking response element-binding protein that interacts with the hnRNP A2 response element (A2RE). Many mRNAs, such as myelin basic protein (MBP), myelin-associated oligodendrocytic basic protein (MOBP), carboxyanhydrase II (CAII), microtubule-associated protein tau, and amyloid precursor protein (APP) are trafficked by hnRNP A2/B1. hnRNP A3 is also a RNA trafficking response element-binding protein that participates in the trafficking of A2RE-containing RNA. The hnRNP A subfamily is characterized by two RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), followed by a long glycine-rich region at the C-terminus. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=co>49</td>
            <td id=cg>97</td>
            <td id=al>cd21603 RNA recognition motif 3 (RRM3) found in Saccharomyces cerevisiae protein PES4, protein MIP6 and similar proteins. The family includes PES4 (also called DNA polymerase epsilon suppressor 4) and MIP6 (also called MEX67-interacting protein 6), both of which are predicted RNA binding proteins that may act as regulators of late translation, protection, and mRNA localization. MIP6 acts as a novel factor for nuclear mRNA export, binds to both poly(A)+ RNA and nuclear pores. It interacts with MEX67. Members in this family contain four RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains). The model corresponds to the third RRM motif.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>36</td>
            <td id=cb>71</td>
            <td id=al>cd12242 RNA recognition motif (RRM) found in SRA stem-loop-interacting RNA-binding protein (SLIRP) and similar proteins. This subfamily corresponds to the RRM of SLIRP, a widely expressed small steroid receptor RNA activator (SRA) binding protein, which binds to STR7, a functional substructure of SRA. SLIRP is localized predominantly to the mitochondria and plays a key role in modulating several nuclear receptor (NR) pathways. It functions as a co-repressor to repress SRA-mediated nuclear receptor coactivation. It modulates SHARP- and SKIP-mediated co-regulation of NR activity. SLIRP contains an RNA recognition motif (RRM), also termed RBD (RNA binding domain) or RNP (ribonucleoprotein domain), which is required for SLIRP's corepression activities. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=al>cd12463 RNA recognition motif (RRM) found in ras GTPase-activating protein-binding protein 1 (G3BP1) and similar proteins. This subgroup corresponds to the RRM of G3BP1, also termed ATP-dependent DNA helicase VIII (DH VIII), or GAP SH3 domain-binding protein 1, which has been identified as a phosphorylation-dependent endoribonuclease that interacts with the SH3 domain of RasGAP, a multi-functional protein controlling Ras activity. The acidic RasGAP binding domain of G3BP1 harbors an arsenite-regulated phosphorylation site and dominantly inhibits stress granule (SG) formation. G3BP1 also contains an N-terminal nuclear transfer factor 2 (NTF2)-like domain, an RNA recognition motif (RRM domain), and an Arg-Gly-rich region (RGG-rich region, or arginine methylation motif). The RRM domain and RGG-rich region are canonically associated with RNA binding. G3BP1 co-immunoprecipitates with mRNAs. It binds to and cleaves the 3'-untranslated region (3'-UTR) of the c-myc mRNA in a phosphorylation-dependent manner. Thus, G3BP1 may play a role in coupling extra-cellular stimuli to mRNA stability. It has been shown that G3BP1 is a novel Dishevelled-associated protein that is methylated upon Wnt3a stimulation and that arginine methylation of G3BP1 regulates both Ctnnb1 mRNA and canonical Wnt/beta-catenin signaling. Furthermore, G3BP1 can be associated with the 3'-UTR of beta-F1 mRNA in cytoplasmic RNA-granules, demonstrating that G3BP1 may specifically repress the translation of the transcript.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=al>cd12327 RNA recognition motif 2 (RRM2) found in Deleted in azoospermia-associated protein 1 (DAZAP1) and similar proteins. This subfamily corresponds to the RRM2 of DAZAP1 or DAZ-associated protein 1, also termed proline-rich RNA binding protein (Prrp), a multi-functional ubiquitous RNA-binding protein expressed most abundantly in the testis and essential for normal cell growth, development, and spermatogenesis. DAZAP1 is a shuttling protein whose acetylated is predominantly nuclear and the nonacetylated form is in cytoplasm. DAZAP1 also functions as a translational regulator that activates translation in an mRNA-specific manner. DAZAP1 was initially identified as a binding partner of Deleted in Azoospermia (DAZ). It also interacts with numerous hnRNPs, including hnRNP U, hnRNP U like-1, hnRNPA1, hnRNPA/B, and hnRNP D, suggesting DAZAP1 might associate and cooperate with hnRNP particles to regulate adenylate-uridylate-rich elements (AU-rich element or ARE)-containing mRNAs. DAZAP1 contains two N-terminal RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains), and a C-terminal proline-rich domain. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>53</td>
            <td id=cg>100</td>
            <td id=al>cd12251 RNA recognition motif 3 (RRM3) found in heterogeneous nuclear ribonucleoprotein R (hnRNP R) and similar proteins. This subfamily corresponds to the RRM3 in hnRNP R, hnRNP Q, and APOBEC-1 complementation factor (ACF). hnRNP R is a ubiquitously expressed nuclear RNA-binding protein that specifically bind mRNAs with a preference for poly(U) stretches and has been implicated in mRNA processing and mRNA transport, and also acts as a regulator to modify binding to ribosomes and RNA translation. hnRNP Q is also a ubiquitously expressed nuclear RNA-binding protein. It has been identified as a component of the spliceosome complex, as well as a component of the apobec-1 editosome, and has been implicated in the regulation of specific mRNA transport. ACF is an RNA-binding subunit of a core complex that interacts with apoB mRNA to facilitate C to U RNA editing. It may also act as an apoB mRNA recognition factor and chaperone and play a key role in cell growth and differentiation. This family also includes two functionally unknown RNA-binding proteins, RBM46 and RBM47. All members contain three conserved RNA recognition motifs (RRMs), also termed RBDs (RNA binding domains) or RNPs (ribonucleoprotein domains).</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2927668</td></tr>
<tr><td id=al>end</td><td id=al>2928100</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>RNA-binding protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>144</td></tr>
</table>

    </div>
</body>
</html>
