<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04684</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04684 biosynthetic arginine decarboxylase [Phormidium_A]</h3>
    <div>MVDSATSPMSDAVAHNGTGPQTPSRSAWSIEDSEELYRIRGWGDPYFGINSAGHITVSPQGDRGGSLDLYELVEALKKRNLELPLLLRFSDILEDRIERINACFAKAIARYNYPGVYRGVFPVKCNQQRHIIEDLVRFGEPYHFGLEAGSKPELLIALASLKTSGSLLICNGYKDRGYIETAMLATRLGQQPIIVLEQPEEVELVISVSQQLGINPVVGVRAKLSAKGIGRWGGSSGDRAKFGLTIPEIITAVDQLRQANLLDSLQLLHYHIGSQISCISAIKDGIREASQIYVELAQLGANMQYLDVGGGLGVDYDGSKTNFYASKNYNMQNYANDIVAEVKEACEEKQQAVPTLISESGRAIASHHSLLIFNVLGSSDAPSGIPEVGPDKEHLILRNLRETYDSITPQTYQEAYHDAVQFKDEATSIFSFGYLSLSERAKAEQLYWACCRKIYDIVRQQADVPTELEVLEKTLASIYYANLSIFRSAPDSWAIEQLFPILPIHRLNEEPSLRATLADLTCDSDGKIDRFIDRLKTKSILELHPLKPEEPYYLGMFLVGAYQEIMGNLHNLFGDTNAVHIHLTPNGYEIEHLVKGDTMTQVLEYVEYNGKDLLENIRKATESALQKGNISIEESQRLLQNYEQSLRSYTYLA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>arginine decarboxylase SpeA</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>78</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>TPA: biosynthetic arginine decarboxylase</td>
            <td id=cb>DVED01</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=co>FACHB-68</td>
            </tr><tr>
            <td id=cb>76</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>biosynthetic arginine decarboxylase</td>
            <td id=cr>JAAUTN01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>biosynthetic arginine decarboxylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Arginine decarboxylase (spermidine biosynthesis) [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>arginine decarboxylase, biosynthetic. Two alternative pathways can convert arginine to putrescine. One is decarboxylation by this enzyme followed by removal of the urea moeity by agmatinase. In the other, the ureohydrolase (arginase) acts first, followed by ornithine decarboxylase. This pathway leads to spermidine biosynthesis, hence the gene symbol speA. A distinct biodegradative form is also pyridoxal phosphate-dependent but is not similar in sequence. [Central intermediary metabolism, Polyamine biosynthesis]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>arginine decarboxylase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme Arginine Decarboxylase. This subfamily includes plants and biosynthetic prokaryotic arginine decarboxylases (ADC, EC 4.1.1.19). ADC is involved in the biosynthesis of putrescine, which is the precursor of aliphatic polyamines in many organisms. It catalyzes the decarboxylation of L-arginine to agmatine, which is then hydrolyzed to putrescine by agmatinase. ADC is homologous to eukaryotic ornithine decarboxylase (ODC) and diaminopimelate decarboxylase (DapDC), which are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. Homodimer formation and the presence of both PLP and Mg2+ cofactors may be required for catalytic activity. Prokaryotic ADCs (biodegradative), which are fold type I PLP-dependent enzymes, are not included in this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>18</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzymes, Ornithine and Diaminopimelate Decarboxylases, and Related Enzymes. This family includes eukaryotic ornithine decarboxylase (ODC, EC 4.1.1.17), diaminopimelate decarboxylase (DapDC, EC 4.1.1.20), plant and prokaryotic biosynthetic arginine decarboxylase (ADC, EC 4.1.1.19), carboxynorspermidine decarboxylase (CANSDC), and ODC-like enzymes from diverse bacterial species. These proteins are fold type III PLP-dependent enzymes that catalyze essential steps in the  biosynthesis of polyamine and lysine. ODC and ADC participate in alternative pathways of the biosynthesis of putrescine, which is the precursor of aliphatic polyamines in many organisms. ODC catalyzes the direct synthesis of putrescine from L-ornithine, while ADC converts L-arginine to agmatine, which is hydrolysed to putrescine by agmatinase in a pathway that exists only in plants and bacteria. DapDC converts meso-2,6-diaminoheptanedioate to L-lysine, which is the final step of lysine biosynthesis. CANSDC catalyzes the decarboxylation of carboxynorspermidine, which is the last step in the synthesis of norspermidine. The PLP-dependent decarboxylases in this family contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. They exist as homodimers with active sites that lie at the interface between the TIM barrel domain of one subunit and the beta-sandwich domain of the other subunit. Prokaryotic ornithine, lysine and biodegradative arginine decarboxylases are fold type I PLP-dependent enzymes and are not included in this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Pyridoxal-dependent decarboxylase, pyridoxal binding domain. These pyridoxal-dependent decarboxylases acting on ornithine, lysine, arginine and related substrates This domain has a TIM barrel fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Arginine decarboxylase helical bundle domain. This entry represents a helical bundle domain that is found between the two enzymatic domains of the arginine decarboxylases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Pyridoxal-dependent decarboxylase, C-terminal sheet domain. These pyridoxal-dependent decarboxylases act on ornithine, lysine, arginine and related substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>Diaminopimelate decarboxylase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme Diaminopimelate Decarboxylase. Diaminopimelate decarboxylase (DapDC, EC 4.1.1.20) participates in the last step of lysine biosynthesis. It converts meso-2,6-diaminoheptanedioate to L-lysine. It is a fold type III PLP-dependent enzyme that contains an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. DapDC exists as homodimers with active sites that lie at the interface between the TIM barrel domain of one subunit and the beta-sandwich domain of the other subunit. Homodimer formation and the presence of the PLP cofactor are required for catalytic activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cb>76</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme MccE. This subfamily is composed of uncharacterized proteins with similarity to Escherichia coli MccE, a hypothetical protein that is homologous to eukaryotic ornithine decarboxylase (ODC) and diaminopimelate decarboxylase (DapDC). ODC and DapDC are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. ODC participates in the formation of putrescine by catalyzing the decarboxylation of ornithine, the first step in polyamine biosynthesis. DapDC participates in the last step of lysine biosynthesis, the conversion of meso-2,6-diaminoheptanedioate to L-lysine. Most members of this subfamily share the same domain architecture as ODC and DapDC. A few members, including Escherichia coli MccE, contain an additional acetyltransferase domain at the C-terminus.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>75</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme Ornithine Decarboxylase. This subfamily is composed mainly of eukaryotic ornithine decarboxylases (ODC, EC 4.1.1.17) and ODC-like enzymes from prokaryotes represented by Vibrio vulnificus LysineOrnithine decarboxylase. These are fold type III PLP-dependent enzymes that differ from most bacterial ODCs which are fold type I PLP-dependent enzymes. ODC participates in the formation of putrescine by catalyzing the decarboxylation of ornithine, the first step in polyamine biosynthesis. Members of this subfamily contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. They exist as homodimers with active sites that lie at the interface between the TIM barrel domain of one subunit and the beta-sandwich domain of the other subunit. Homodimer formation and the presence of the PLP cofactor are required for catalytic activity. Also members of this subfamily are proteins with homology to ODC but do not possess any catalytic activity, the Antizyme inhibitor (AZI) and ODC-paralogue (ODC-p). AZI binds to the regulatory protein Antizyme with a higher affinity than ODC and prevents ODC degradation. ODC-p is a novel ODC-like protein, present only in mammals, that is specifically exressed in the brain and testes. ODC-p may function as a tissue-specific antizyme inhibitory protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Arginine decarboxylase C-terminal helical extension. This small three helical domain is found at the C-terminus of the arginine decarboxylase enzyme.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>77</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzymes, Uncharacterized Proteins with similarity to Ornithine and Diaminopimelate Decarboxylases. This subfamily contains uncharacterized proteins with similarity to ornithine decarboxylase (ODC) and diaminopimelate decarboxylase (DapDC). ODC and DapDC are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. They exist as homodimers with active sites that lie at the interface between the TIM barrel domain of one subunit and the beta-sandwich domain of the other subunit. ODC participates in the formation of putrescine by catalyzing the decarboxylation of ornithine, the first step in polyamine biosynthesis. DapDC participates in the last step of lysine biosynthesis, the conversion of meso-2,6-diaminoheptanedioate to L-lysine. Proteins in this subfamily may function as PLP-dependent decarboxylases. Homodimer formation and the presence of the PLP cofactor may be required for catalytic activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>59</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme Y4yA. This subfamily is composed of the hypothetical Rhizobium sp. protein Y4yA and similar uncharacterized bacterial proteins. These proteins are homologous to eukaryotic ornithine decarboxylase (ODC) and diaminopimelate decarboxylase (DapDC). ODC and DapDC are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases. ODC participates in the formation of putrescine by catalyzing the decarboxylation of ornithine, the first step in polyamine biosynthesis. DapDC participates in the last step of lysine biosynthesis, the conversion of meso-2,6-diaminoheptanedioate to L-lysine. Proteins in this subfamily may function as PLP-dependent decarboxylases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cb>64</td>
            <td id=al>Type III Pyridoxal 5-phosphate (PLP)-Dependent Enzyme Btrk Decarboxylase. This subfamily is composed of Bacillus circulans BtrK decarboxylase and similar proteins. These proteins are fold type III PLP-dependent enzymes that contain an N-terminal PLP-binding TIM-barrel domain and a C-terminal beta-sandwich domain, similar to bacterial alanine racemases, eukaryotic ornithine decarboxylases and diaminopimelate decarboxylases. BtrK is presumed to function as a PLP-dependent decarboxylase involved in the biosynthesis of the aminoglycoside antibiotic butirosin. Homodimer formation and the presence of the PLP cofactor may be required for catalytic activity.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3815255</td></tr>
<tr><td id=al>end</td><td id=al>3817217</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>biosynthetic arginine decarboxylase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
