<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00127</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00127 translation initiation factor IF-2 [Phormidium_A]</h3>
    <div>MNNGKVRLYDLSRELNVDNKQVLAFCDQLDIAYKSHSSTITEADAARIRQEAENAPAVSEPARPSGSGPNTHRTKQKILKIEHRSPNAKPAVQGSDPETTGDNRANSDATSSNLDSPELVDKPVRPKSPSPAKPQPANTPPVEATNPKKPVSSQAPSRDDSDSAPVNPASEAKTIEADSKAEPEQKQEEVPQLIGPPPRPQPRAKKHKGRDKPTVERPTIAKDKPSSAPRPSAPSKDKEADSPRPVSSSPQKPSRPAERDEQEPVSAGPPSRPESKRPVPKLKRPPELVRLNAKTEEVEPEAPENEPLAVEPEDDTDETEQLTLKRPKLPRKAKTKKRRTTEEDPVESLEAANKANKPKRRVRPIDDDDEDFEADLNTDSNDAMEVSLSIARPPKATKSKSESRPSTQQRRKPAPQQRGSGDSSGGSSSSSRNRRDRRERANKVERPELVEIRDNLTVRELADLLLVPETEIVKRLFMQGIAVNVTQTLDVPTATMVAEELDVLVETPEKEQEARKVTDMIDATDLDSLQRRPPVVTIMGHVDHGKTTLLDSIRKTKVAQGEAGGITQHTGAYHVDVEHEGVMHQIVFLDTPGHEAFTAMRARGARVTDIAILVVAADDGVQPQTIEAISHARAAGVPLVVAVNKMDKEQANPDRVKQELMEHGLVPEEWGGDTVMVPVSAITGDNLDTLLEMIILVSEVEELSANPDRPARGTVIEAHLDKARGSVATLLVQNGTLRVGDTLVAGSVFAKVRAMIDDRGNRVDEAKPSFAVEVLGMNDVPEAGDEFEVFTNEKEARSEADRRKDESRQSRLQQTMASRRVTLSSLSARASEGELKELNLVLKADVQGSLEAILGSLEQLPQDEVQLRVLYAAPGEIAETDIDLAAASDAVILGFNTTLASGARQAAEREGVDVRDYSVIYNLLDDIQGAMEGLLDPEMVEEELGQAQVRAIFPSGRGQVAGCYVLSGQLVRNAKVRILRNDKVVHENVLDSLRRMREDVRDVKSGYECGLGIDRFSDWQEGDIVYCYRMASKRRTLASR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2 InfB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=cb>RDYZ01</td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation initiation factor IF-2</td>
            <td id=co>Crinalium</td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>TPA: translation initiation factor IF-2</td>
            <td id=cb>DVED01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>46</td>
            <td id=cb>71</td>
            <td id=cg>99</td>
            <td id=al>PRK05306 translation initiation factor IF-2; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cr>18</td>
            <td id=cr>25</td>
            <td id=al>PRK05306 translation initiation factor IF-2; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=co>57</td>
            <td id=cg>100</td>
            <td id=al>TIGR00487 translation initiation factor IF-2. This model discriminates eubacterial (and mitochondrial) translation initiation factor 2 (IF-2), encoded by the infB gene in bacteria, from similar proteins in the Archaea and Eukaryotes. In the bacteria and in organelles, the initiator tRNA is charged with N-formyl-Met instead of Met. This translation factor acts in delivering the initator tRNA to the ribosome. It is one of a number of GTP-binding translation factors recognized by the pfam model GTP_EFTU. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cb>69</td>
            <td id=cg>98</td>
            <td id=al>CHL00189 translation initiation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>57</td>
            <td id=co>49</td>
            <td id=cg>100</td>
            <td id=al>COG0532 Translation initiation factor IF-2, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd01887 Initiation Factor 2 (IF2)/ eukaryotic Initiation Factor 5B (eIF5B) family. IF2/eIF5B contribute to ribosomal subunit joining and function as GTPases that are maximally activated by the presence of both ribosomal subunits. As seen in other GTPases, IF2/IF5B undergoes conformational changes between its GTP- and GDP-bound states. Eukaryotic IF2/eIF5Bs possess three characteristic segments, including a divergent N-terminal region followed by conserved central and C-terminal segments. This core region is conserved among all known eukaryotic and archaeal IF2/eIF5Bs and eubacterial IF2s.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=al>PRK04004 translation initiation factor IF-2; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=co>53</td>
            <td id=cg>93</td>
            <td id=al>TIGR00491 translation initiation factor aIF-2/yIF-2. This model describes archaeal and eukaryotic orthologs of bacterial IF-2. Like IF-2, it helps convey the initiator tRNA to the ribosome, although the initiator is N-formyl-Met in bacteria and Met here. This protein is not closely related to the subunits of eIF-2 of eukaryotes, which is also involved in the initiation of translation. The aIF-2 of Methanococcus jannaschii contains a large intein interrupting a region of very strongly conserved sequence very near the amino end; the alignment generated by this model does not correctly align the sequences from Methanococcus jannaschii and Pyrococcus horikoshii in this region. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>17</td>
            <td id=cg>96</td>
            <td id=al>pfam00009 Elongation factor Tu GTP binding domain. This domain contains a P-loop motif, also found in several other families such as pfam00071, pfam00025 and pfam00063. Elongation factor Tu consists of three structural domains, this plus two C-terminal beta barrel domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=co>55</td>
            <td id=co>54</td>
            <td id=al>PRK14845 translation initiation factor IF-2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>51</td>
            <td id=cr>9</td>
            <td id=cg>99</td>
            <td id=al>pfam11987 Translation-initiation factor 2. IF-2 is a translation initiator in each of the three main phylogenetic domains (Eukaryotes, Bacteria and Archaea). IF2 interacts with formylmethionine-tRNA, GTP, IF1, IF3 and both ribosomal subunits. Through these interactions, IF2 promotes the binding of the initiator tRNA to the A site in the smaller ribosomal subunit and catalyzes the hydrolysis of GTP following initiation-complex formation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cr>9</td>
            <td id=cg>100</td>
            <td id=al>cd03702 Domain II of bacterial and mitochondrial Initiation Factor 2. This family represents domain II of bacterial Initiation Factor 2 (IF2) and its eukaryotic mitochondrial homolog mtIF2. IF2, the largest initiation factor, is an essential GTP binding protein. In E. coli, three natural forms of IF2 exist in the cell, IF2alpha, IF2beta1, and IF2beta2.  Bacterial IF-2 is structurally and functionally related to eukaryotic mitochondrial mtIF-2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>17</td>
            <td id=cg>98</td>
            <td id=al>cd00881 GTP translation factor family primarily contains translation initiation, elongation and release factors. The GTP translation factor family consists primarily of translation initiation, elongation, and release factors, which play specific roles in protein translation. In addition, the family includes Snu114p, a component of the U5 small nuclear riboprotein particle which is a component of the spliceosome and is involved in excision of introns, TetM, a tetracycline resistance gene that protects the ribosome from tetracycline binding, and the unusual subfamily CysN/ATPS, which has an unrelated function (ATP sulfurylase) acquired through lateral transfer of the EF1-alpha gene and development of a new function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>50</td>
            <td id=cr>8</td>
            <td id=cg>100</td>
            <td id=al>cd03692 C2 subdomain of domain IV in mitochondrial translation initiation factor 2. This model represents the C2 subdomain of domain IV of mitochondrial translation initiation factor 2 (mtIF2) which adopts a beta-barrel fold displaying a high degree of structural similarity with domain II of the translation elongation factor EF-Tu. The C-terminal part of mtIF2 contains the entire fMet-tRNAfmet binding site of IF-2 and is resistant to proteolysis. This C-terminal portion consists of two domains, IF2 C1 and IF2 C2.  IF2 C2 has been shown to contain all molecular determinants necessary and sufficient for the recognition and binding of fMet-tRNAfMet. Like IF2 from certain prokaryotes such as Thermus thermophilus, mtIF2lacks domain II which is thought to be involved in binding of E.coli IF-2 to 30S subunits.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>TIGR00231 small GTP-binding protein domain. Proteins with a small GTP-binding domain recognized by this model include Ras, RhoA, Rab11, translation elongation factor G, translation initiation factor IF-2, tetratcycline resistance protein TetM, CDC42, Era, ADP-ribosylation factors, tdhF, and many others. In some proteins the domain occurs more than once.This model recognizes a large number of small GTP-binding proteins and related domains in larger proteins. Note that the alpha chains of heterotrimeric G proteins are larger proteins in which the NKXD motif is separated from the GxxxxGK[ST] motif (P-loop) by a long insert and are not easily detected by this model. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>14</td>
            <td id=cr>20</td>
            <td id=al>COG0480 Translation elongation factor EF-G, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd04171 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec, and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains bacterial SelBs, as well as, one from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>21</td>
            <td id=cr>38</td>
            <td id=al>TIGR00475 selenocysteine-specific elongation factor SelB. In prokaryotes, the incorporation of selenocysteine as the 21st amino acid, encoded by TGA, requires several elements: SelC is the tRNA itself, SelD acts as a donor of reduced selenium, SelA modifies a serine residue on SelC into selenocysteine, and SelB is a selenocysteine-specific translation elongation factor. 3-prime or 5-prime non-coding elements of mRNA have been found as probable structures for directing selenocysteine incorporation. This model describes the elongation factor SelB, a close homolog rf EF-Tu. It may function by replacing EF-Tu. A C-terminal domain not found in EF-Tu is in all SelB sequences in the seed alignment except that from Methanococcus jannaschii. This model does not find an equivalent protein for eukaryotes. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>14</td>
            <td id=cr>22</td>
            <td id=al>PRK12740 elongation factor G-like protein EF-G2. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cr>30</td>
            <td id=cb>73</td>
            <td id=al>COG5256 Translation elongation factor EF-1alpha (GTPase)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>24</td>
            <td id=co>56</td>
            <td id=al>COG3276 Selenocysteine-specific translation elongation factor [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>29</td>
            <td id=cb>72</td>
            <td id=al>PRK12317 elongation factor 1-alpha; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>20</td>
            <td id=cr>34</td>
            <td id=al>PRK10512 selenocysteinyl-tRNA-specific translation factor; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd00882 Rat sarcoma (Ras)-like superfamily of small guanosine triphosphatases (GTPases). Ras-like GTPase superfamily. The Ras-like superfamily of small GTPases consists of several families with an extremely high degree of structural and functional similarity. The Ras superfamily is divided into at least four families in eukaryotes: the Ras, Rho, Rab, and Sar1/Arf families. This superfamily also includes proteins like the GTP translation factors, Era-like GTPases, and G-alpha chain of the heterotrimeric G proteins. Members of the Ras superfamily regulate a wide variety of cellular functions: the Ras family regulates gene expression, the Rho family regulates cytoskeletal reorganization and gene expression, the Rab and Sar1/Arf families regulate vesicle trafficking, and the Ran family regulates nucleocytoplasmic transport and microtubule organization. The GTP translation factor family regulates initiation, elongation, termination, and release in translation, and the Era-like GTPase family regulates cell division, sporulation, and DNA replication. Members of the Ras superfamily are identified by the GTP binding site, which is made up of five characteristic sequence motifs, and the switch I and switch II regions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>24</td>
            <td id=co>42</td>
            <td id=al>COG0481 Translation elongation factor EF-4, membrane-bound GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>16</td>
            <td id=cg>97</td>
            <td id=al>cd01890 LepA also known as Elongation Factor 4 (EF4). LepA (also known as elongation factor 4, EF4) belongs to the GTPase family and exhibits significant homology to the translation factors EF-G and EF-Tu, indicating its possible involvement in translation and association with the ribosome. LepA is ubiquitous in bacteria and eukaryota (e.g. yeast GUF1p), but is missing from archaea. This pattern of phyletic distribution suggests that LepA evolved through a duplication of the EF-G gene in bacteria, followed by early transfer into the eukaryotic lineage, most likely from the promitochondrial endosymbiont. Yeast GUF1p is not essential and mutant cells did not reveal any marked phenotype.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>23</td>
            <td id=cb>62</td>
            <td id=al>COG0050 Translation elongation factor EF-Tu, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>26</td>
            <td id=co>46</td>
            <td id=al>TIGR01394 GTP-binding protein TypA/BipA. This bacterial (and Arabidopsis) protein, termed TypA or BipA, a GTP-binding protein, is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways, but the precise function is unknown. [Regulatory functions, Other, Cellular processes, Adaptations to atypical conditions, Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>24</td>
            <td id=cb>64</td>
            <td id=al>PRK12736 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>14</td>
            <td id=cr>21</td>
            <td id=al>PRK13351 elongation factor G-like protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>23</td>
            <td id=co>40</td>
            <td id=al>PRK05433 GTP-binding protein LepA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>16</td>
            <td id=cg>88</td>
            <td id=al>cd01889 SelB, the dedicated elongation factor for delivery of selenocysteinyl-tRNA to the ribosome. SelB is an elongation factor needed for the co-translational incorporation of selenocysteine. Selenocysteine is coded by a UGA stop codon in combination with a specific downstream mRNA hairpin. In bacteria, the C-terminal part of SelB recognizes this hairpin, while the N-terminal part binds GTP and tRNA in analogy with elongation factor Tu (EF-Tu). It specifically recognizes the selenocysteine charged tRNAsec, which has a UCA anticodon, in an EF-Tu like manner. This allows insertion of selenocysteine at in-frame UGA stop codons. In E. coli SelB binds GTP, selenocysteyl-tRNAsec and a stem-loop structure immediately downstream of the UGA codon (the SECIS sequence). The absence of active SelB prevents the participation of selenocysteyl-tRNAsec in translation. Archaeal and animal mechanisms of selenocysteine incorporation are more complex. Although the SECIS elements have different secondary structures and conserved elements between archaea and eukaryotes, they do share a common feature. Unlike in E. coli, these SECIS elements are located in the 3' UTRs. This group contains eukaryotic SelBs and some from archaea.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>18</td>
            <td id=cg>96</td>
            <td id=al>cd01891 Tyrosine phosphorylated protein A (TypA)/BipA family belongs to ribosome-binding GTPases. BipA is a protein belonging to the ribosome-binding family of GTPases and is widely distributed in bacteria and plants. BipA was originally described as a protein that is induced in Salmonella typhimurium after exposure to bactericidal/permeability-inducing protein (a cationic antimicrobial protein produced by neutrophils), and has since been identified in E. coli as well. The properties thus far described for BipA are related to its role in the process of pathogenesis by enteropathogenic E. coli. It appears to be involved in the regulation of several processes important for infection, including rearrangements of the cytoskeleton of the host, bacterial resistance to host defense peptides, flagellum-mediated cell motility, and expression of K5 capsular genes. It has been proposed that BipA may utilize a novel mechanism to regulate the expression of target genes. In addition, BipA from enteropathogenic E. coli has been shown to be phosphorylated on a tyrosine residue, while BipA from Salmonella and from E. coli K12 strains is not phosphorylated under the conditions assayed. The phosphorylation apparently modifies the rate of nucleotide hydrolysis, with the phosphorylated form showing greatly increased GTPase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>12</td>
            <td id=cb>67</td>
            <td id=al>cd01884 Elongation Factor Tu (EF-Tu) GTP-binding proteins. EF-Tu subfamily. This subfamily includes orthologs of translation elongation factor EF-Tu in bacteria, mitochondria, and chloroplasts. It is one of several GTP-binding translation factors found in the larger family of GTP-binding elongation factors. The eukaryotic counterpart, eukaryotic translation elongation factor 1 (eEF-1 alpha), is excluded from this family. EF-Tu is one of the most abundant proteins in bacteria, as well as, one of the most highly conserved, and in a number of species the gene is duplicated with identical function. When bound to GTP, EF-Tu can form a complex with any (correctly) aminoacylated tRNA except those for initiation and for selenocysteine, in which case EF-Tu is replaced by other factors. Transfer RNA is carried to the ribosome in these complexes for protein translation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>24</td>
            <td id=co>55</td>
            <td id=al>PLN03127 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>24</td>
            <td id=cb>63</td>
            <td id=al>PRK00049 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cr>28</td>
            <td id=cb>71</td>
            <td id=al>COG5257 Translation initiation factor 2, gamma subunit (eIF-2gamma; GTPase)  [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>19</td>
            <td id=cg>90</td>
            <td id=al>cd01883 Elongation Factor 1-alpha (EF1-alpha) protein family. EF1 is responsible for the GTP-dependent binding of aminoacyl-tRNAs to the ribosomes. EF1 is composed of four subunits: the alpha chain which binds GTP and aminoacyl-tRNAs, the gamma chain that probably plays a role in anchoring the complex to other cellular components and the beta and delta (or beta') chains. This subfamily is the alpha subunit, and represents the counterpart of bacterial EF-Tu for the archaea (aEF1-alpha) and eukaryotes (eEF1-alpha). eEF1-alpha interacts with the actin of the eukaryotic cytoskeleton and may thereby play a role in cellular transformation and apoptosis. EF-Tu can have no such role in bacteria. In humans, the isoform eEF1A2 is overexpressed in 2/3 of breast cancers and has been identified as a putative oncogene. This subfamily also includes Hbs1, a G protein known to be important for efficient growth and protein synthesis under conditions of limiting translation initiation in yeast, and to associate with Dom34. It has been speculated that yeast Hbs1 and Dom34 proteins may function as part of a complex with a role in gene expression.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>27</td>
            <td id=co>46</td>
            <td id=al>COG1217 Predicted membrane GTPase involved in stress response [Signal transduction mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>13</td>
            <td id=cb>60</td>
            <td id=al>cd04168 Tet(M)-like family includes Tet(M), Tet(O), Tet(W), and OtrA, containing tetracycline resistant proteins. Tet(M), Tet(O), Tet(W), and OtrA are tetracycline resistance genes found in Gram-positive and Gram-negative bacteria. Tetracyclines inhibit protein synthesis by preventing aminoacyl-tRNA from binding to the ribosomal acceptor site. This subfamily contains tetracycline resistance proteins that function through ribosomal protection and are typically found on mobile genetic elements, such as transposons or plasmids, and are often conjugative. Ribosomal protection proteins are homologous to the elongation factors EF-Tu and EF-G. EF-G and Tet(M) compete for binding on the ribosomes. Tet(M) has a higher affinity than EF-G, suggesting these two proteins may have overlapping binding sites and that Tet(M) must be released before EF-G can bind. Tet(M) and Tet(O) have been shown to have ribosome-dependent GTPase activity. These proteins are part of the GTP translation factor family, which includes EF-G, EF-Tu, EF2, LepA, and SelB.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>24</td>
            <td id=cb>63</td>
            <td id=al>CHL00071 elongation factor Tu</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>24</td>
            <td id=cb>63</td>
            <td id=al>PRK04000 translation initiation factor IF-2 subunit gamma; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>12</td>
            <td id=cb>61</td>
            <td id=al>cd01885 Elongation Factor 2 (EF2) in archaea and eukarya. Translocation requires hydrolysis of a molecule of GTP and is mediated by EF-G in bacteria and by eEF2 in eukaryotes. The eukaryotic elongation factor eEF2 is a GTPase involved in the translocation of the peptidyl-tRNA from the A site to the P site on the ribosome. The 95-kDa protein is highly conserved, with 60% amino acid sequence identity between the human and yeast proteins. Two major mechanisms are known to regulate protein elongation and both involve eEF2. First, eEF2 can be modulated by reversible phosphorylation. Increased levels of phosphorylated eEF2 reduce elongation rates presumably because phosphorylated eEF2 fails to bind the ribosomes. Treatment of mammalian cells with agents that raise the cytoplasmic Ca2+ and cAMP levels reduce elongation rates by activating the kinase responsible for phosphorylating eEF2. In contrast, treatment of cells with insulin increases elongation rates by promoting eEF2 dephosphorylation. Second, the protein can be post-translationally modified by ADP-ribosylation. Various bacterial toxins perform this reaction after modification of a specific histidine residue to diphthamide, but there is evidence for endogenous ADP ribosylase activity. Similar to the bacterial toxins, it is presumed that modification by the endogenous enzyme also inhibits eEF2 activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>13</td>
            <td id=co>52</td>
            <td id=al>cd01886 Elongation factor G (EF-G) family involved in both the elongation and ribosome recycling phases of protein synthesis. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains both eukaryotic and bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>23</td>
            <td id=cb>62</td>
            <td id=al>PRK12735 elongation factor Tu; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>18</td>
            <td id=cg>93</td>
            <td id=al>cd04166 CysN, together with protein CysD, forms the ATP sulfurylase (ATPS) complex. CysN_ATPS subfamily. CysN, together with protein CysD, form the ATP sulfurylase (ATPS) complex in some bacteria and lower eukaryotes. ATPS catalyzes the production of ATP sulfurylase (APS) and pyrophosphate (PPi) from ATP and sulfate. CysD, which catalyzes ATP hydrolysis, is a member of the ATP pyrophosphatase (ATP PPase) family. CysN hydrolysis of GTP is required for CysD hydrolysis of ATP; however, CysN hydrolysis of GTP is not dependent on CysD hydrolysis of ATP. CysN is an example of lateral gene transfer followed by acquisition of new function. In many organisms, an ATPS exists which is not GTP-dependent and shares no sequence or structural similarity to CysN.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>14</td>
            <td id=co>54</td>
            <td id=al>cd04170 Elongation factor G (EF-G) family. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains only bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>24</td>
            <td id=co>53</td>
            <td id=al>PLN03126 Elongation factor Tu; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>25</td>
            <td id=cb>61</td>
            <td id=al>COG2895 Sulfate adenylyltransferase subunit 1, EFTu-like GTPase family  [Inorganic ion transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd00880 E. coli Ras-like protein (Era)-like GTPase. The Era (E. coli Ras-like protein)-like family includes several distinct subfamilies (TrmE/ThdF, FeoB, YihA (EngB), Era, and EngA/YfgK) that generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. TrmE is ubiquitous in bacteria and is a widespread mitochondrial protein in eukaryotes, but is absent from archaea. The yeast member of TrmE family, MSS1, is involved in mitochondrial translation; bacterial members are often present in translation-related operons. FeoB represents an unusual adaptation of GTPases for high-affinity iron (II) transport. YihA (EngB) family of GTPases is typified by the E. coli YihA, which is an essential protein involved in cell division control. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain. EngA and its orthologs are composed of two GTPase domains and, since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>18</td>
            <td id=cg>97</td>
            <td id=al>cd01888 Gamma subunit of initiation factor 2 (eIF2 gamma). eIF2 is a heterotrimeric translation initiation factor that consists of alpha, beta, and gamma subunits. The GTP-bound gamma subunit also binds initiator methionyl-tRNA and delivers it to the 40S ribosomal subunit. Following hydrolysis of GTP to GDP, eIF2:GDP is released from the ribosome. The gamma subunit has no intrinsic GTPase activity, but is stimulated by the GTPase activating protein (GAP) eIF5, and GDP/GTP exchange is stimulated by the guanine nucleotide exchange factor (GEF) eIF2B. eIF2B is a heteropentamer, and the epsilon chain binds eIF2. Both eIF5 and eIF2B-epsilon are known to bind strongly to eIF2-beta, but have also been shown to bind directly to eIF2-gamma. It is possible that eIF2-beta serves simply as a high-affinity docking site for eIF5 and eIF2B-epsilon, or that eIF2-beta serves a regulatory role. eIF2-gamma is found only in eukaryotes and archaea. It is closely related to SelB, the selenocysteine-specific elongation factor from eubacteria. The translational factor components of the ternary complex, IF2 in eubacteria and eIF2 in eukaryotes are not the same protein (despite their unfortunately similar names). Both factors are GTPases; however, eubacterial IF-2 is a single polypeptide, while eIF2 is heterotrimeric. eIF2-gamma is a member of the same family as eubacterial IF2, but the two proteins are only distantly related. This family includes translation initiation, elongation, and release factors.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>17</td>
            <td id=cr>38</td>
            <td id=al>PRK05124 sulfate adenylyltransferase subunit 1; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>16</td>
            <td id=co>57</td>
            <td id=al>PRK00089 GTPase Era; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>12</td>
            <td id=cb>62</td>
            <td id=al>cd04167 Snu114p, a spliceosome protein, is a GTPase. Snu114p subfamily. Snu114p is one of several proteins that make up the U5 small nuclear ribonucleoprotein (snRNP) particle. U5 is a component of the spliceosome, which catalyzes the splicing of pre-mRNA to remove introns. Snu114p is homologous to EF-2, but typically contains an additional N-terminal domain not found in Ef-2. This protein is part of the GTP translation factor family and the Ras superfamily, characterized by five G-box motifs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>28</td>
            <td id=co>56</td>
            <td id=al>COG5258 GTPase  [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>19</td>
            <td id=cb>68</td>
            <td id=al>COG1159 GTPase Era, involved in 16S rRNA processing [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cr>29</td>
            <td id=co>41</td>
            <td id=al>PRK07560 elongation factor EF-2; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=co>41</td>
            <td id=co>46</td>
            <td id=al>PTZ00449 104 kDa microneme/rhoptry antigen; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>30</td>
            <td id=cr>34</td>
            <td id=al>PTZ00449 104 kDa microneme/rhoptry antigen; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cr>9</td>
            <td id=cg>98</td>
            <td id=al>cd03701 Domain II of prokaryotic Initiation Factor 2 and archaeal and eukaryotic Initiation Factor 5. This family represents domain II of prokaryotic Initiation Factor 2 (IF2) and its archaeal and eukaryotic homologue aeIF5B. IF2, the largest initiation factor, is an essential GTP binding protein. In E. coli, three natural forms of IF2 exist in the cell, IF2alpha, IF2beta1, and IF2beta2. Disruption of the eIF5B gene (FUN12) in yeast causes a severe slow-growth phenotype, associated with a defect in translation. eIF5B has a function analogous to prokaryotic IF2 in mediating the joining of the 60S ribosomal subunit. The eIF5B consists of three N-terminal domains (I, II, II) connected by a long helix to domain IV. Domain I is a G domain, domain II and IV are beta-barrels and domain III has a novel alpha-beta-alpha sandwich fold. The G domain and the beta-barrel domain II display a similar structure and arrangement to the homologous domains in EF1A, eEF1A and aeIF2gamma.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>14</td>
            <td id=cr>21</td>
            <td id=al>PRK12739 elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>33</td>
            <td id=cr>12</td>
            <td id=cr>16</td>
            <td id=al>PTZ00416 elongation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>22</td>
            <td id=cr>37</td>
            <td id=al>PRK05506 bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>16</td>
            <td id=cg>100</td>
            <td id=al>cd04163 E. coli Ras-like protein (Era) is a multifunctional GTPase. Era (E. coli Ras-like protein) is a multifunctional GTPase found in all bacteria except some eubacteria. It binds to the 16S ribosomal RNA (rRNA) of the 30S subunit and appears to play a role in the assembly of the 30S subunit, possibly by chaperoning the 16S rRNA. It also contacts several assembly elements of the 30S subunit. Era couples cell growth with cytokinesis and plays a role in cell division and energy metabolism. Homologs have also been found in eukaryotes. Era contains two domains: the N-terminal GTPase domain and a C-terminal domain KH domain that is critical for RNA binding. Both domains are important for Era function. Era is functionally able to compensate for deletion of RbfA, a cold-shock adaptation protein that is required for efficient processing of the 16S rRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cr>29</td>
            <td id=cr>9</td>
            <td id=al>PHA03247 large tegument protein UL36; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cr>24</td>
            <td id=cr>8</td>
            <td id=al>PHA03247 large tegument protein UL36; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>17</td>
            <td id=cr>29</td>
            <td id=cr>9</td>
            <td id=al>PHA03247 large tegument protein UL36; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>17</td>
            <td id=cg>100</td>
            <td id=al>cd01895 EngA2 GTPase contains the second domain of EngA. This EngA2 subfamily CD represents the second GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cr>4</td>
            <td id=cg>96</td>
            <td id=al>pfam04760 Translation initiation factor IF-2, N-terminal region. This conserved feature at the N-terminus of bacterial translation initiation factor IF2 has recently had its structure solved. It shows structural similarity to the tRNA anticodon Stem Contact Fold domains of the methionyl-tRNA and glutaminyl-tRNA synthetases, and a similar fold is also found in the B5 domain of the phenylalanine-tRNA synthetase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cr>13</td>
            <td id=cr>20</td>
            <td id=al>PRK00007 elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cr>13</td>
            <td id=cr>23</td>
            <td id=al>PRK10218 translational GTPase TypA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>14</td>
            <td id=co>56</td>
            <td id=al>cd04169 Release Factor 3 (RF3) protein involved in the terminal step of translocation in bacteria. Peptide chain release factor 3 (RF3) is a protein involved in the termination step of translation in bacteria. Termination occurs when class I release factors (RF1 or RF2) recognize the stop codon at the A-site of the ribosome and activate the release of the nascent polypeptide. The class II release factor RF3 then initiates the release of the class I RF from the ribosome. RF3 binds to the RF/ribosome complex in the inactive (GDP-bound) state. GDP/GTP exchange occurs, followed by the release of the class I RF. Subsequent hydrolysis of GTP to GDP triggers the release of RF3 from the ribosome. RF3 also enhances the efficiency of class I RFs at less preferred stop codons and at stop codons in weak contexts.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cr>28</td>
            <td id=cr>22</td>
            <td id=al>PHA03307 transcriptional regulator ICP4; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>17</td>
            <td id=co>41</td>
            <td id=al>COG1160 Predicted GTPases [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cr>18</td>
            <td id=co>43</td>
            <td id=al>COG1160 Predicted GTPases [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>26</td>
            <td id=cb>60</td>
            <td id=al>PTZ00327 eukaryotic translation initiation factor 2 gamma subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>13</td>
            <td id=cr>16</td>
            <td id=al>PLN00116 translation elongation factor EF-2 subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>13</td>
            <td id=cr>27</td>
            <td id=al>COG4108 Peptide chain release factor RF-3 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cr>8</td>
            <td id=co>55</td>
            <td id=al>cd01894 EngA1 GTPase contains the first domain of EngA. This EngA1 subfamily CD represents the first GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>19</td>
            <td id=cr>34</td>
            <td id=cb>63</td>
            <td id=al>NF033839 pneumococcal surface protein PspC, LPXTG-anchored form. The pneumococcal surface protein PspC, as described in Streptococcus pneumoniae, is a repetitive and highly variable protein, recognized by a conserved N-terminal domain and also by genomic location. This form, subgroup 2, is anchored covalently after cleavage by sortase at a C-terminal LPXTG site. The other form, subgroup 1, has variable numbers of a choline-binding repeat in the C-terminal region, and is also known as choline-binding protein A.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cr>20</td>
            <td id=co>49</td>
            <td id=al>PRK00093 GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cr>16</td>
            <td id=cg>91</td>
            <td id=al>COG2229 Signal recognition particle receptor subunit beta, a GTPase [Intracellular trafficking, secretion, and vesicular transport]. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>103673</td></tr>
<tr><td id=al>end</td><td id=al>106796</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>translation initiation factor IF-2</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>1041</td></tr>
</table>

    </div>
</body>
</html>
