<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.02782</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.02782 elongation factor G [Phormidium_A]</h3>
    <div>MVHKMGEVHDGTAVTDWMAQERERGITITAAAITTNWLDHQINIIDTPGHVDFTIEVERSMRVLDGVVAVFCSVGGVQPQSETVWRQADRYKVPRIAFVNKMDRTGANFFKVYNQMRDRLRANAVAIQVPIGSESNFHGVVDLVGMKAYIYTNDLGTDIQETDIPEEVSELAAEYREKLLEAVAETDDDLMEKYLEGEEISEEQVRATLRRGTIDGTIVPVLCGSAFKNKGVQRLLDCVVDYLPSPTEVPPIQGTLPDGEVATRPADDDVPLSALAFKVMADPYGRLTFVRVYSGILQKGSYVYNATKDQKERVSRLIVLKSDDRIEVDELRAGDLGAALGLKDTLTGDTICDEEEPIILESLYIPEPVISVAVEPKTKADMEKLSKALQSLSEEDPTFRVTTDAETNQTVIAGMGELHLEILVDRMLREFKVEAEVGAPQVAYRETIRKPSTAEGKFVRQSGGKGQYGHVVIELEPGETGTGFEFVSKIVGGSVPKEYIGPAENGMKEACESGILAGYPLIDVKATLVDGSYHDVDSNEMAFKIAGSMAMKNAVTKASPVLLEPMMKVEVEVPEDFMGNVIGDLNSRRGQIEGQETEDGLTKVTAKVPLASMFGYATDIRSKTQGRGIFSMEFSHYEEVPRSVAEEIIAKSKGNA</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>translation elongation factor G</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=co>Planktothrix</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=co>Limnospira</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=co>Limnospira</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=cr>JAAUTN01</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G</td>
            <td id=co>Planktothrix</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>elongation factor G; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>61</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>elongation factor G-like protein EF-G2. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>translation elongation factor EF-G. After peptide bond formation, this elongation factor of bacteria and organelles catalyzes the translocation of the tRNA-mRNA complex, with its attached nascent polypeptide chain, from the A-site to the P-site of the ribosome. Every completed bacterial genome has at least one copy, but some species have additional EF-G-like proteins. The closest homolog to canonical (e.g. E. coli) EF-G in the spirochetes clusters as if it is derived from mitochondrial forms, while a more distant second copy is also present. Synechocystis PCC6803 has a few proteins more closely related to EF-G than to any other characterized protein. Two of these resemble E. coli EF-G more closely than does the best match from the spirochetes; it may be that both function as authentic EF-G. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>Translation elongation factor EF-G, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>55</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>elongation factor G-like protein. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>Elongation factor G (EF-G) family involved in both the elongation and ribosome recycling phases of protein synthesis. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains both eukaryotic and bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>elongation factor EF-2; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>Elongation factor G (EF-G) family. Translocation is mediated by EF-G (also called translocase). The structure of EF-G closely resembles that of the complex between EF-Tu and tRNA. This is an example of molecular mimicry; a protein domain evolved so that it mimics the shape of a tRNA molecule. EF-G in the GTP form binds to the ribosome, primarily through the interaction of its EF-Tu-like domain with the 50S subunit. The binding of EF-G to the ribosome in this manner stimulates the GTPase activity of EF-G. On GTP hydrolysis, EF-G undergoes a conformational change that forces its arm deeper into the A site on the 30S subunit. To accommodate this domain, the peptidyl-tRNA in the A site moves to the P site, carrying the mRNA and the deacylated tRNA with it. The ribosome may be prepared for these rearrangements by the initial binding of EF-G as well. The dissociation of EF-G leaves the ribosome ready to accept the next aminoacyl-tRNA into the A site. This group contains only bacterial members.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>65</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>EFG_mtEFG1_IV: domains similar to domain IV of the bacterial translational elongation factor (EF) EF-G.  Included in this group is a domain of mitochondrial Elongation factor G1 (mtEFG1) proteins homologous to domain IV of EF-G. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. During the process of peptide synthesis and tRNA site changes, the ribosome is moved along the mRNA a distance equal to one codon with the addition of each amino acid. In bacteria this translocation step is catalyzed by EF-G_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Eukaryotic mtEFG1 proteins show significant homology to bacterial EF-Gs.  Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. There are two forms of mtEFG present in mammals (designated mtEFG1s and mtEFG2s) mtEFG2s are not present in this group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor G, domain IV. Translation elongation factors are responsible for two main processes during protein synthesis on the ribosome. EF1A (or EF-Tu) is responsible for the selection and binding of the cognate aminoacyl-tRNA to the A-site (acceptor site) of the ribosome. EF2 (or EF-G) is responsible for the translocation of the peptidyl-tRNA from the A-site to the P-site (peptidyl-tRNA site) of the ribosome, thereby freeing the A-site for the next aminoacyl-tRNA to bind. Elongation factors are responsible for achieving accuracy of translation and both EF1A and EF2 are remarkably conserved throughout evolution. Elongation factor EF2 (EF-G) is a G-protein. It brings about the translocation of peptidyl-tRNA and mRNA through a ratchet-like mechanism: the binding of GTP-EF2 to the ribosome causes a counter-clockwise rotation in the small ribosomal subunit; the hydrolysis of GTP to GDP by EF2 and the subsequent release of EF2 causes a clockwise rotation of the small subunit back to the starting position. This twisting action destabilises tRNA-ribosome interactions, freeing the tRNA to translocate along the ribosome upon GTP-hydrolysis by EF2. EF2 binding also affects the entry and exit channel openings for the mRNA, widening it when bound to enable the mRNA to translocate along the ribosome. EF2 has five domains. This entry represents domain IV found in EF2 (or EF-G) of both prokaryotes and eukaryotes. The EF2-GTP-ribosome complex undergoes extensive structural rearrangement for tRNA-mRNA movement to occur. Domain IV, which extends from the 'body' of the EF2 molecule much like a lever arm, appears to be essential for the structural transition to take place.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor G, domain IV. This domain is found in elongation factor G, elongation factor 2 and some tetracycline resistance proteins and adopts a ribosomal protein S5 domain 2-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>90</td>
            <td id=al>elongation factor 2; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Tet(M)-like family includes Tet(M), Tet(O), Tet(W), and OtrA, containing tetracycline resistant proteins. Tet(M), Tet(O), Tet(W), and OtrA are tetracycline resistance genes found in Gram-positive and Gram-negative bacteria. Tetracyclines inhibit protein synthesis by preventing aminoacyl-tRNA from binding to the ribosomal acceptor site. This subfamily contains tetracycline resistance proteins that function through ribosomal protection and are typically found on mobile genetic elements, such as transposons or plasmids, and are often conjugative. Ribosomal protection proteins are homologous to the elongation factors EF-Tu and EF-G. EF-G and Tet(M) compete for binding on the ribosomes. Tet(M) has a higher affinity than EF-G, suggesting these two proteins may have overlapping binding sites and that Tet(M) must be released before EF-G can bind. Tet(M) and Tet(O) have been shown to have ribosome-dependent GTPase activity. These proteins are part of the GTP translation factor family, which includes EF-G, EF-Tu, EF2, LepA, and SelB.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor G C-terminus. This domain includes the carboxyl terminal regions of Elongation factor G, elongation factor 2 and some tetracycline resistance proteins and adopt a ferredoxin-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor Tu GTP binding domain. This domain contains a P-loop motif, also found in several other families such as pfam00071, pfam00025 and pfam00063. Elongation factor Tu consists of three structural domains, this plus two C-terminal beta barrel domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>82</td>
            <td id=al>Peptide chain release factor RF-3 [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>translation elongation factor EF-2 subunit; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>EFG_mtEFG_C: domains similar to the C-terminal domain of the bacterial translational elongation factor (EF) EF-G.  Included in this group is the C-terminus of mitochondrial Elongation factor G1 (mtEFG1) and G2 (mtEFG2) proteins. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. During the process of peptide synthesis and tRNA site changes, the ribosome is moved along the mRNA a distance equal to one codon with the addition of each amino acid. In bacteria this translocation step is catalyzed by EF-G_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Eukaryotic mtEFG1 proteins show significant homology to bacterial EF-Gs.  Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. No clear phenotype has been found for mutants in the yeast homologue of mtEFG2, MEF2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Domain III of Elongation Factor G (EFG). This model represents domain III of bacterial Elongation factor G (EF-G), and mitochondrial Elongation factor G1 (mtEFG1) and G2 (mtEFG2), which play an important role during peptide synthesis and tRNA site changes. In bacteria, this translocation step is catalyzed by EF-G_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. mtEFG1 and mtEFG2 show significant homology to bacterial EF-Gs. Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects, and a tendency to lose mitochondrial DNA. No clear phenotype has been found for mutants of the yeast homolog of mtEFG2, MEF2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>63</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Domain II of bacterial elongation factor G and C-terminal domain of mitochondrial Elongation factors G1 and G2. This family represents the domain II of bacterial Elongation factor G (EF-G)and mitochondrial Elongation factors G1 (mtEFG1) and G2 (mtEFG2). During the process of peptide synthesis and tRNA site changes, the ribosome is moved along the mRNA a distance equal to one codon with the addition of each amino acid. In bacteria this translocation step is catalyzed by EF-G_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. mtEFG1 and mtEFG2 show significant homology to bacterial EF-Gs. Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. No clear phenotype has been found for mutants in the yeast homolog of mtEFG2, MEF2.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Predicted membrane GTPase involved in stress response [Signal transduction mechanisms]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>peptide chain release factor 3; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>64</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation Factor G, domain II. This domain is found in Elongation Factor G. It shares a similar structure with domain V (pfam00679).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>81</td>
            <td id=al>peptide chain release factor 3. This translation releasing factor, RF-3 (prfC) was originally described as stop codon-independent, in contrast to peptide chain release factor 1 (RF-1, prfA) and RF-2 (prfB). RF-1 and RF-2 are closely related to each other, while RF-3 is similar to elongation factors EF-Tu and EF-G; RF-1 is active at UAA and UAG and RF-2 is active at UAA and UGA. More recently, RF-3 was shown to be active primarily at UGA stop codons in E. coli. All bacteria and organelles have RF-1. The Mycoplasmas and organelles, which translate UGA as Trp rather than as a stop codon, lack RF-2. RF-3, in contrast, seems to be rare among bacteria and is found so far only in Escherichia coli and some other gamma subdivision Proteobacteria, in Synechocystis PCC6803, and in Staphylococcus aureus. [Protein synthesis, Translation factors]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>54</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor G C-terminus. This domain includes the carboxyl terminal regions of Elongation factor G, elongation factor 2 and some tetracycline resistance proteins and adopt a ferredoxin-like fold.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor G C-terminus. This domain includes the carboxyl terminal regions of elongation factors (EFs) bacterial EF-G, eukaryotic and archeal EF-2 and eukaryotic mitochondrial mtEFG1s and mtEFG2s. This group also includes proteins similar to the ribosomal protection proteins Tet(M) and Tet(O), BipA, LepA and, spliceosomal proteins: human 116kD U5 small nuclear ribonucleoprotein (snRNP) protein (U5-116 kD) and yeast counterpart Snu114p.  This domain adopts a ferredoxin-like fold consisting of an alpha-beta sandwich with anti-parallel beta-sheets, resembling the topology of domain III found in the elongation factors EF-G and eukaryotic EF-2, with which it forms the C-terminal block. The two domains however are not superimposable and domain III lacks some of the characteristics of this domain.  EF-2/EF-G in complex with GTP, promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site, the uncharged tRNA from the P site to the E-site and, the mRNA is shifted one codon relative to the ribosome. Tet(M) and Tet(O) mediate Tc resistance. Typical Tcs bind to the ribosome and inhibit the elongation phase of protein synthesis, by inhibiting the occupation of site A by aminoacyl-tRNA. Tet(M) and Tet(O) catalyze the release of tetracycline (Tc) from the ribosome in a GTP-dependent manner.  BipA is a highly conserved protein with global regulatory properties in Escherichia coli. Yeast Snu114p is essential for cell viability and for splicing in vivo. Experiments suggest that GTP binding and probably GTP hydrolysis is important for the function of the U5-116 kD/Snu114p. The function of LepA proteins is unknown.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>Release Factor 3 (RF3) protein involved in the terminal step of translocation in bacteria. Peptide chain release factor 3 (RF3) is a protein involved in the termination step of translation in bacteria. Termination occurs when class I release factors (RF1 or RF2) recognize the stop codon at the A-site of the ribosome and activate the release of the nascent polypeptide. The class II release factor RF3 then initiates the release of the class I RF from the ribosome. RF3 binds to the RF/ribosome complex in the inactive (GDP-bound) state. GDP/GTP exchange occurs, followed by the release of the class I RF. Subsequent hydrolysis of GTP to GDP triggers the release of RF3 from the ribosome. RF3 also enhances the efficiency of class I RFs at less preferred stop codons and at stop codons in weak contexts.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mtEFG1_C: C-terminus of mitochondrial Elongation factor G1 (mtEFG1)-like proteins found in eukaryotes.  Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species.  Eukaryotic EF-2 operates in the cytosolic protein synthesis machinery of eukaryotes, EF-Gs in protein synthesis in bacteria.  Eukaryotic mtEFG1 proteins show significant homology to bacterial EF-Gs.  Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. There are two forms of mtEFG present in mammals (designated mtEFG1s and mtEFG2s) mtEFG2s are not present in this group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation Factor G-like domain IV. This family includes the translational elongation factor termed EF-2 (for Archaea and Eukarya) and EF-G (for Bacteria), ribosomal protection proteins that mediate tetracycline resistance and, an evolutionarily conserved U5 snRNP-specific protein (U5-116kD). In complex with GTP, EF-G/EF-2  promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site of the small subunit of ribosome and the mRNA is shifted one codon relative to the ribosome. It has been shown that EF-G/EF-2_IV domain mimics the shape of anticodon arm of the tRNA in the structurally homologous ternary complex of Petra, EF-Tu (another transcriptional elongation factor) and GTP analog. The tip portion of this domain is found in a position that overlaps the anticodon arm of the A-site tRNA, implying that EF-G/EF-2 displaces the A-site tRNA to the P-site by physical interaction with the anticodon arm.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>GTP translation factor family primarily contains translation initiation, elongation and release factors. The GTP translation factor family consists primarily of translation initiation, elongation, and release factors, which play specific roles in protein translation. In addition, the family includes Snu114p, a component of the U5 small nuclear riboprotein particle which is a component of the spliceosome and is involved in excision of introns, TetM, a tetracycline resistance gene that protects the ribosome from tetracycline binding, and the unusual subfamily CysN/ATPS, which has an unrelated function (ATP sulfurylase) acquired through lateral transfer of the EF1-alpha gene and development of a new function.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>58</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Domain II of mitochondrial elongation factor G2-like proteins found in eukaryotes. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. Eukaryotic EF-2 operates in the cytosolic protein synthesis machinery of eukaryotes, EF-Gs in protein synthesis in bacteria.  Eukaryotic mtEFG1 proteins show significant homology to bacterial EF-Gs. No clear phenotype has been found for mutants in the yeast homolog of mtEFG2, MEF2. There are two forms of mtEFG present in mammals (designated mtEFG1s and mtEFG2s); mtEFG1s are not present in this group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=co>48</td>
            <td id=al>Elongation Factor 2 (EF2) in archaea and eukarya. Translocation requires hydrolysis of a molecule of GTP and is mediated by EF-G in bacteria and by eEF2 in eukaryotes. The eukaryotic elongation factor eEF2 is a GTPase involved in the translocation of the peptidyl-tRNA from the A site to the P site on the ribosome. The 95-kDa protein is highly conserved, with 60% amino acid sequence identity between the human and yeast proteins. Two major mechanisms are known to regulate protein elongation and both involve eEF2. First, eEF2 can be modulated by reversible phosphorylation. Increased levels of phosphorylated eEF2 reduce elongation rates presumably because phosphorylated eEF2 fails to bind the ribosomes. Treatment of mammalian cells with agents that raise the cytoplasmic Ca2+ and cAMP levels reduce elongation rates by activating the kinase responsible for phosphorylating eEF2. In contrast, treatment of cells with insulin increases elongation rates by promoting eEF2 dephosphorylation. Second, the protein can be post-translationally modified by ADP-ribosylation. Various bacterial toxins perform this reaction after modification of a specific histidine residue to diphthamide, but there is evidence for endogenous ADP ribosylase activity. Similar to the bacterial toxins, it is presumed that modification by the endogenous enzyme also inhibits eEF2 activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>Translation elongation factor EF-4, membrane-bound GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Domain II of mitochondrial elongation factor G1-like proteins found in eukaryotes. Eukaryotic cells harbor 2 protein synthesis systems: one localized in the cytoplasm, the other in the mitochondria. Most factors regulating mitochondrial protein synthesis are encoded by nuclear genes, translated in the cytoplasm, and then transported to the mitochondria. The eukaryotic system of elongation factor (EF) components is more complex than that in prokaryotes, with both cytoplasmic and mitochondrial elongation factors and multiple isoforms being expressed in certain species. Eukaryotic EF-2 operates in the cytosolic protein synthesis machinery of eukaryotes, EF-Gs in protein synthesis in bacteria. Eukaryotic mtEFG1 proteins show significant homology to bacterial EF-Gs. Mutants in yeast mtEFG1 have impaired mitochondrial protein synthesis, respiratory defects and a tendency to lose mitochondrial DNA. There are two forms of mtEFG present in mammals (designated mtEFG1s and mtEFG2s); mtEFG2s are not present in this group.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>47</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Domain III of Elongation factor G (EF-G) and related proteins. Bacterial Elongation factor G (EF-G) and related proteins play a role in translation and share a similar domain architecture. Elongation factor EFG participates in the elongation phase during protein biosynthesis on the ribosome by stimulating translocation. Its functional cycles depend on GTP binding and its hydrolysis. Domain III is involved in the activation of GTP hydrolysis. This domain III, which is different from domain III in EF-TU and related elongation factors, is found in several translation factors, like bacterial release factors RF3, elongation factor 4, elongation factor 2, GTP-binding protein BipA and tetracycline resistance protein Tet.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cb>74</td>
            <td id=al>translational GTPase TypA. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>Tyrosine phosphorylated protein A (TypA)/BipA family belongs to ribosome-binding GTPases. BipA is a protein belonging to the ribosome-binding family of GTPases and is widely distributed in bacteria and plants. BipA was originally described as a protein that is induced in Salmonella typhimurium after exposure to bactericidal/permeability-inducing protein (a cationic antimicrobial protein produced by neutrophils), and has since been identified in E. coli as well. The properties thus far described for BipA are related to its role in the process of pathogenesis by enteropathogenic E. coli. It appears to be involved in the regulation of several processes important for infection, including rearrangements of the cytoskeleton of the host, bacterial resistance to host defense peptides, flagellum-mediated cell motility, and expression of K5 capsular genes. It has been proposed that BipA may utilize a novel mechanism to regulate the expression of target genes. In addition, BipA from enteropathogenic E. coli has been shown to be phosphorylated on a tyrosine residue, while BipA from Salmonella and from E. coli K12 strains is not phosphorylated under the conditions assayed. The phosphorylation apparently modifies the rate of nucleotide hydrolysis, with the phosphorylated form showing greatly increased GTPase activity.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>eEF2_snRNP_like_C: this family represents a C-terminal domain of eukaryotic elongation factor 2 (eEF-2) and a homologous domain of the spliceosomal human 116kD U5 small nuclear ribonucleoprotein (snRNP) protein (U5-116 kD) and, its yeast counterpart Snu114p.  Yeast Snu114p is essential for cell viability and for splicing in vivo. U5-116 kD binds GTP.  Experiments suggest that GTP binding and probably GTP hydrolysis is important for the function of the U5-116 kD/Snu114p.   In complex with GTP, EF-2 promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site, the uncharged tRNA from the P site to the E-site and, the mRNA is shifted one codon relative to the ribosome.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Tet_C: C-terminus of ribosomal protection proteins Tet(M) and Tet(O). This domain has homology to the C terminal domains of the elongation factors EF-G and EF-2. Tet(M) and Tet(O) catalyze the release of tetracycline (Tc) from the ribosome in a GTP-dependent manner thereby mediating Tc resistance.  Tcs are broad-spectrum antibiotics.  Typical Tcs bind to the ribosome and inhibit the elongation phase of protein synthesis, by inhibiting the  occupation of site A by aminoacyl-tRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Domain III of Elongation Factor 2 (EF2). This model represents domain III of Elongation factor 2 (EF2) found in eukaryotes and archaea, and the spliceosomal human 116kD U5 small nuclear ribonucleoprotein (snRNP) protein (U5-116 kD) and its yeast counterpart Snu114p. During the process of peptide synthesis and tRNA site changes, the ribosome is moved along the mRNA a distance equal to one codon with the addition of each amino acid. This translocation step is catalyzed by EF-2_GTP, which is hydrolyzed to provide the required energy. Thus, this action releases the uncharged tRNA from the P site and transfers the newly formed peptidyl-tRNA from the A site to the P site. Yeast Snu114p is essential for cell viability and for splicing in vivo. U5-116 kD binds GTP. Experiments suggest that GTP binding and probably GTP hydrolysis are important for the function of the U5-116 kD/Snu114p.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein LepA; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>BipA_TypA_C: a C-terminal portion of BipA or TypA having homology to the C terminal domains of the elongation factors EF-G and EF-2. A member of the ribosome binding GTPase superfamily, BipA is widely distributed in bacteria and plants.  BipA is a highly conserved protein with global regulatory properties in Escherichia coli. BipA is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways. BipA functions as a translation factor that is required specifically for the expression of the transcriptional modulator Fis.  BipA binds to ribosomes at a site that coincides with that of EF-G and has a GTPase activity that is sensitive to high GDP:GTP ratios and, is stimulated  by 70S ribosomes programmed with mRNA and aminoacylated tRNAs. The growth rate-dependent induction of BipA allows the efficient expression of Fis, thereby modulating a range of downstream processes, including DNA metabolism and type III secretion.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>87</td>
            <td id=al>small GTP-binding protein domain. Proteins with a small GTP-binding domain recognized by this model include Ras, RhoA, Rab11, translation elongation factor G, translation initiation factor IF-2, tetratcycline resistance protein TetM, CDC42, Era, ADP-ribosylation factors, tdhF, and many others. In some proteins the domain occurs more than once.This model recognizes a large number of small GTP-binding proteins and related domains in larger proteins. Note that the alpha chains of heterotrimeric G proteins are larger proteins in which the NKXD motif is separated from the GxxxxGK[ST] motif (P-loop) by a long insert and are not easily detected by this model. [Unknown function, General]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>LepA also known as Elongation Factor 4 (EF4). LepA (also known as elongation factor 4, EF4) belongs to the GTPase family and exhibits significant homology to the translation factors EF-G and EF-Tu, indicating its possible involvement in translation and association with the ribosome. LepA is ubiquitous in bacteria and eukaryota (e.g. yeast GUF1p), but is missing from archaea. This pattern of phyletic distribution suggests that LepA evolved through a duplication of the EF-G gene in bacteria, followed by early transfer into the eukaryotic lineage, most likely from the promitochondrial endosymbiont. Yeast GUF1p is not essential and mutant cells did not reveal any marked phenotype.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>100</td>
            <td id=cr>30</td>
            <td id=al>Snu114p, a spliceosome protein, is a GTPase. Snu114p subfamily. Snu114p is one of several proteins that make up the U5 small nuclear ribonucleoprotein (snRNP) particle. U5 is a component of the spliceosome, which catalyzes the splicing of pre-mRNA to remove introns. Snu114p is homologous to EF-2, but typically contains an additional N-terminal domain not found in Ef-2. This protein is part of the GTP translation factor family and the Ras superfamily, characterized by five G-box motifs.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Elongation factor Tu domain 2. Elongation factor Tu consists of three structural domains, this is the second domain. This domain adopts a beta barrel structure. This the second domain is involved in binding to charged tRNA. This domain is also found in other proteins such as elongation factor G and translation initiation factor IF-2. This domain is structurally related to pfam03143, and in fact has weak sequence matches to this domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Domain II of Elongation factor Tu (EF-Tu)-like proteins. Elongation factor Tu consists of three structural domains. Domain II adopts a beta barrel structure and is involved in binding to charged tRNA. Domain II is found in other proteins such as elongation factor G and translation initiation factor IF-2. This group also includes the C2 subdomain of domain IV of IF-2 that has the same fold as domain II of (EF-Tu). Like IF-2 from certain prokaryotes such as Thermus thermophilus, mitochondrial IF-2 lacks domain II, which is thought  to be involved in binding of E. coli IF-2 to 30S subunits.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>86</td>
            <td id=al>Domain II of BipA. BipA (also called TypA) is a highly conserved protein with global regulatory properties in Escherichia coli.  BipA is phosphorylated on a tyrosine residue under some cellular conditions. Mutants show altered regulation of some pathways. BipA functions as a translation factor that is required specifically for the expression of the transcriptional modulator Fis. BipA binds to ribosomes at a site that coincides with that of EF-G and has a GTPase activity that is sensitive to high GDP:GTP ratios and is stimulated  by 70S ribosomes programmed with mRNA and aminoacylated tRNAs. The growth rate-dependent induction of BipA allows the efficient expression of Fis, thereby modulating a range of downstream processes, including DNA metabolism and type III secretion. The domain II of BipA shows similarity to the domain II of the elongation factors (EFs) EF-G and EF-Tu.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>Domain II of ribosomal protection proteins Tet(M) and Tet(O). This subfamily represents domain II of ribosomal protection proteins Tet(M) and Tet(O). This domain has homology to domain II of the elongation factors EF-G and EF-2. Tet(M) and Tet(O) catalyze the release of tetracycline (Tc) from the ribosome in a GTP-dependent manner thereby mediating Tc resistance. Tcs are broad-spectrum antibiotics.  Typical Tcs bind to the ribosome and inhibit the elongation phase of protein synthesis, by inhibiting the occupation of site A by aminoacyl-tRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>Domain III of Tetracycline resistance protein Tet. Tetracycline resistance proteins, including TetM and TetO, catalyze the release of tetracycline (Tc) from the ribosome in a GTP-dependent manner thereby mediating Tc resistance. Tcs are broad-spectrum antibiotics. Typical Tcs bind to the ribosome and inhibit the elongation phase of protein synthesis, by inhibiting the occupation of site A by aminoacyl-tRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>mtEF-G2 domain IV. This subfamily is a part the of mitochondrial transcriptional elongation factor, mtEF-G2. Mitochondrial translation is crucial for maintaining mitochondrial function and mutations in this system lead to a breakdown in the respiratory chain-oxidative phosphorylation system and to impaired maintenance of mitochondrial DNA. In complex with GTP, EF-G promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site of the small subunit of ribosome and the mRNA is shifted one codon relative to the ribosome.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cr>15</td>
            <td id=al>Translation initiation factor IF-2, a GTPase [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>eEF2_C_snRNP: This family includes a C-terminal portion of the spliceosomal human 116kD U5 small nuclear ribonucleoprotein (snRNP) protein (U5-116 kD) and, its yeast counterpart Snu114p.  This domain is homologous to the C-terminal domain of the eukaryotic translational elongation factor EF-2.  Yeast Snu114p is essential for cell viability and for splicing in vivo. U5-116 kD binds GTP.  Experiments suggest that GTP binding and probably GTP hydrolysis is important for the function of the U5-116 kD/Snu114p.   In complex with GTP, EF-2 promotes the translocation step of translation. During translocation the peptidyl-tRNA is moved from the A site to the P site, the uncharged tRNA from the P site to the E-site and, the mRNA is shifted one codon relative to the ribosome.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2320890</td></tr>
<tr><td id=al>end</td><td id=al>2322861</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>elongation factor G</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
