<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.03254</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.03254 inositol monophosphatase [Phormidium_A]</h3>
    <div>MNAVTADKLKFFLDIATEAAMQGGAVLKSYWGKLEGVEEKGRSGDLVTVADKESEAKVLEVLQRHSPDCGVLAEESGVLGNAESPFRWAIDPLDGTTNYAHQYPICSVSIGLLINGVPEVGVVYNPIREDLFQAATELGATYNRQPMRVSQTKTLDRSLLVTGFAYDRRERDDNNYAEFCYLTHLTQGVRRGGSASLDLADVACGRIDGYWERGLSIWDLAAGIVLVREAGGVVTAYDQSPLDLPSGRLLATNGHLQSALSDRLADVQPLGELPGIGRLSERSPV</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>94</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>myo-inositol-1(or 4)-monophosphatase SuhB</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>75</td>
            <td id=cg>97</td>
            <td id=cg>97</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>93</td>
            <td id=cg>96</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cr>Leptothoe</td>
            </tr><tr>
            <td id=cb>72</td>
            <td id=cg>90</td>
            <td id=cg>91</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cr>Thermosynechococcus</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>94</td>
            <td id=cg>97</td>
            <td id=al>inositol monophosphatase</td>
            <td id=co>SIO2C1</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>90</td>
            <td id=cg>91</td>
            <td id=al>inositol monophosphatase</td>
            <td id=cr>Thermosynechococcus</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>94</td>
            <td id=cg>98</td>
            <td id=al>inositol monophosphatase</td>
            <td id=co>PCC7113</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>51</td>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=al>cd01639 IMPase, inositol monophosphatase and related domains. A family of Mg++ dependent phosphatases, inhibited by lithium, many of which may act on inositol monophosphate substrate. They dephosphorylate inositol phosphate to generate inositol, which may be recycled into inositol lipids; in eukaryotes IMPase plays a vital role in intracellular signaling. IMPase is one of the proposed targets of Li+ therapy in manic-depressive illness. This family contains some bacterial members of the inositol monophosphatase family classified as SuhB-like. E. coli SuhB has been suggested to participate in posstranscriptional control of gene expression, and its inositol monophosphatase activity doesn't appear to be sufficient for its cellular function. It has been proposed, that SuhB plays a role in the biosynthesis of phosphatidylinositol in mycobacteria.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>90</td>
            <td id=cg>99</td>
            <td id=al>COG0483 Archaeal fructose-1,6-bisphosphatase or related enzyme of inositol monophosphatase family [Carbohydrate transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>44</td>
            <td id=cg>87</td>
            <td id=cg>92</td>
            <td id=al>pfam00459 Inositol monophosphatase family. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=al>cd01637 Inositol-monophosphatase-like domains. This family of phosphatases is dependent on bivalent metal ions such as Mg++, and many members are inhibited by Li+ (which is thought to displace a bivalent ion in the active site). Substrates include fructose-1,6-bisphosphate, inositol poly- and monophosphates, PAP and PAPS, sedoheptulose-1,7-bisphosphate and probably others.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cg>90</td>
            <td id=cb>71</td>
            <td id=al>PLN02737 inositol monophosphatase family protein</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=al>PLN02553 inositol-phosphate phosphatase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>91</td>
            <td id=cg>98</td>
            <td id=al>PRK10757 inositol-1-monophosphatase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cg>83</td>
            <td id=cg>98</td>
            <td id=al>cd01643 Bacterial family of Mg++ dependent phosphatases, related to inositol monophosphatases. These enzymes may dephosphorylate inositol monophosphate or similar substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>89</td>
            <td id=cg>100</td>
            <td id=al>cd01641 Predominantly bacterial family of Mg++ dependend phosphatases, related to inositol monophosphatases. These enzymes may dephosphorylate fructose-1,6-bisphosphate, inositol monophospate, 3'-phosphoadenosine-5'-phosphate,  or similar substrates.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>91</td>
            <td id=cg>99</td>
            <td id=al>PRK12676 bifunctional fructose-bisphosphatase/inositol-phosphate phosphatase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=al>cd01638 CysQ, a 3'-Phosphoadenosine-5'-phosphosulfate (PAPS) 3'-phosphatase, is a bacterial member of the inositol monophosphatase family. It has been proposed that CysQ helps control intracellular levels of PAPS, which is an intermediate in cysteine biosynthesis (a principal route of sulfur assimilation).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>85</td>
            <td id=cg>94</td>
            <td id=al>cd01515 Archaeal fructose-1,6-bisphosphatase and related enzymes of inositol monophosphatase family (FBPase class IV). These are Mg++ dependent phosphatases. Members in this family may have both fructose-1,6-bisphosphatase and inositol-monophosphatase activity. In hyperthermophilic archaea, inositol monophosphatase is thought to play a role in the biosynthesis of di-myo-inositol-1,1'-phosphate, an osmolyte unique to hyperthermophiles.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>88</td>
            <td id=cg>91</td>
            <td id=al>COG1218 3'-Phosphoadenosine 5'-phosphosulfate (PAPS) 3'-phosphatase [Inorganic ion transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>86</td>
            <td id=cg>99</td>
            <td id=al>TIGR01331 3'(2'),5'-bisphosphate nucleotidase, bacterial. Sulfate is incorporated into 3-phosphoadenylylsulfate, PAPS, for utilization in pathways such as methionine biosynthesis. Transfer of sulfate from PAPS to an acceptor leaves adenosine 3'-5'-bisphosphate, APS. This model describes a form found in bacteria of the enzyme 3'(2'),5'-bisphosphate nucleotidase, which removes the 3'-phosphate from APS to regenerate AMP and help drive the cycle. [Central intermediary metabolism, Sulfur metabolism]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>84</td>
            <td id=cg>88</td>
            <td id=al>cd01517 PAP-phosphatase_like domains. PAP-phosphatase is a member of the inositol monophosphatase family, and catalyses the hydrolysis of 3'-phosphoadenosine-5'-phosphate (PAP) to AMP. In Saccharomyces cerevisiae, HAL2 (MET22) is involved in methionine biosynthesis and provides increased salt tolerance when over-expressed. Bacterial members of this domain family may differ in their substrate specificity and dephosphorylate different targets, as the substrate binding site does not appear to be conserved in that sub-set.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>81</td>
            <td id=cg>100</td>
            <td id=al>cd01636 FIG, FBPase/IMPase/glpX-like domain. A superfamily of metal-dependent phosphatases with various substrates. Fructose-1,6-bisphospatase (both the major and the glpX-encoded variant) hydrolyze fructose-1,6,-bisphosphate to fructose-6-phosphate in gluconeogenesis. Inositol-monophosphatases and inositol polyphosphatases play vital roles in eukaryotic signalling, as they participate in metabolizing the messenger molecule Inositol-1,4,5-triphosphate. Many of these enzymes are inhibited by Li+.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>86</td>
            <td id=cg>83</td>
            <td id=al>PLN02911 inositol-phosphate phosphatase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>91</td>
            <td id=co>46</td>
            <td id=al>PRK14076 bifunctional NADP phosphatase/NAD kinase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cb>69</td>
            <td id=cg>81</td>
            <td id=al>cd01642 Putative fructose-1,6-bisphosphatase or related enzymes of inositol monophosphatase family. These are Mg++ dependent phosphatases. Members in this family may have fructose-1,6-bisphosphatase and/or inositol-monophosphatase activity. Fructose-1,6-bisphosphatase catalyzes the hydrolysis of fructose-1,6-biphosphate  into fructose-6-phosphate and is critical in gluconeogenesis pathway.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>84</td>
            <td id=cg>98</td>
            <td id=al>PRK10931 adenosine-3'(2'),5'-bisphosphate nucleotidase; Provisional</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>2742770</td></tr>
<tr><td id=al>end</td><td id=al>2743628</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>inositol monophosphatase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>286</td></tr>
</table>

    </div>
</body>
</html>
