<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.05956</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.05956 YqeG family HAD IIIA-type phosphatase [Phormidium_A]</h3>
    <div>MSWGKLLQPDLVLGGTVMDITPDLLVEHNLKGLVFDVDETLVPIGQMEASEELRQRIVELRQVATLWLVSNNVSETRIGNIAKSLELPYIFFAAKPFRRKLWRAVSAMDLPVAQVAMVGDRLFTDVLAGNRMGMFTVLVEPMVDPGESVRQSPVRSVEVWLSQVLGASLTANR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>hydrolase YqeG</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>80</td>
            <td id=cg>99</td>
            <td id=cg>97</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=co>Phormidium</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=co>Oscillatoria</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cr>Stenomitos</td>
            </tr><tr>
            <td id=cb>66</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>YqeG family HAD IIIA-type phosphatase</td>
            <td id=cr>O-77</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=co>43</td>
            <td id=cg>95</td>
            <td id=cg>95</td>
            <td id=al>COG2179 Predicted phosphohydrolase YqeG, HAD superfamily [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>95</td>
            <td id=cg>97</td>
            <td id=al>TIGR01668 HAD superfamily (subfamily IIIA) phosphatase, TIGR01668. This family of hypothetical proteins is a member of the IIIA subfamily of the haloacid dehalogenase (HAD) superfamily of hydrolases. All characterized members of this subfamily (TIGR01662) and most characterized members of the HAD superfamily are phosphatases. HAD superfamily phosphatases contain active site residues in several conserved catalytic motifs, all of which are found conserved here. This family consists of sequences from fungi, plants, cyanobacteria, gram-positive bacteria and Deinococcus. There is presently no characterization of any sequence in this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>41</td>
            <td id=cb>62</td>
            <td id=cg>100</td>
            <td id=al>cd16416 Uncharacterized family of the the haloacid dehalogenase-like superfamily, similar to the uncharacterized protein Bacillus subtilis YqeG. The haloacid dehalogenase-like (HAD) hydrolases are a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members include 2-L-haloalkanoic acid dehalogenase (C-Cl bond hydrolysis), azetidine hydrolase (C-N bond hydrolysis); phosphonoacetaldehyde hydrolase (C-P bond hydrolysis), phosphoserine phosphatase and phosphomannomutase (CO-P bond hydrolysis), P-type ATPases (PO-P bond hydrolysis) and many others. Members are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>77</td>
            <td id=cg>100</td>
            <td id=al>TIGR01662 HAD-superfamily hydrolase, subfamily IIIA. This subfamily falls within the Haloacid Dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. The Class III subfamilies are characterized by the lack of any domains located between either between the first and second conserved catalytic motifs (as in the Class I subfamilies, TIGR01493, TIGR01509, TIGR01488 and TIGR01494) or between the second and third conserved catalytic motifs (as in the Class II subfamilies, TIGR01460 and TIGR01484) of the superfamily domain. The IIIA subfamily contains five major clades: histidinol-phosphatase (TIGR01261) and histidinol-phosphatase-related protein (TIGR00213) which together form a subfamily (TIGR01656), DNA 3'-phosphatase (TIGR01663, TIGR01664), YqeG (TIGR01668) and YrbI (TIGR01670). In the case of histidinol phosphatase and PNK-3'-phosphatase, this model represents a domain of a bifunctional system. In the histidinol phosphatase HisB, a C-terminal domain is an imidazoleglycerol-phosphate dehydratase which catalyzes a related step in histidine biosynthesis. In PNK-3'-phosphatase, N- and C-terminal domains constitute the polynucleotide kinase and DNA-binding components of the enzyme. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cb>66</td>
            <td id=cg>100</td>
            <td id=al>cd01427 Haloacid dehalogenase-like hydrolases. The haloacid dehalogenase-like (HAD) superfamily includes L-2-haloacid dehalogenase, epoxide hydrolase, phosphoserine phosphatase, phosphomannomutase, phosphoglycolate phosphatase, P-type ATPase, and many others. This superfamily includes a variety of enzymes that catalyze the cleavage of substrate C-Cl, P-C, and P-OP bonds via nucleophilic substitution pathways. All of which use a nucleophilic aspartate in their phosphoryl transfer reaction. They catalyze nucleophilic substitution reactions at phosphorus or carbon centers, using a conserved Asp carboxylate in covalent catalysis. All members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. Members of this superfamily are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cr>27</td>
            <td id=cb>62</td>
            <td id=al>pfam13242 HAD-hyrolase-like. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>35</td>
            <td id=cr>37</td>
            <td id=cr>26</td>
            <td id=al>cd07530 UmpH/NagD family phosphatase, similar to Escherichia coli UmpH UMP phosphatase/NagD nucleotide phosphatase and Mycobacterium tuberculosis Rv1692 glycerol 3-phosphate phosphatase. Escherichia coli UmpH/NagD is a ribonucleoside tri-, di-, and monophosphatase with a preference for purines, it shows peak activity with UMP and functions in UMP-degradation. It is also an effective phosphatase with AMP, GMP and CMP. Mycobacterium tuberculosis phosphatase, Rv1692 is a glycerol 3-phosphate phosphatase. Rv1692 is the final enzyme involved in glycerophospholipid recycling/catabolism.  This subfamily belongs to the UmpH/NagD phosphatase family, and to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>4880139</td></tr>
<tr><td id=al>end</td><td id=al>4880661</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>YqeG family HAD IIIA-type phosphatase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>174</td></tr>
</table>

    </div>
</body>
</html>
