<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.01365</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.01365 DUF697 domain-containing protein [Phormidium_A]</h3>
    <div>MPNPRSSNFERELDEAIFSFADIQSELNYKRAKTALHDLVERLDLSERERTGLEAEIDGLETMLDKLESDVVQIAAFGMVGRGKSSLLNALVGQPVFETGPLHGVTRSQQSVDWTLTRSAVGEGIDDVLQVSLPTVGTARVELVDTPGLDEVHGQTRAELAHKIAKRADLILFVISGDITQVEYDALSEIREAGKPLLLVFNKIDQYPDADRQAIYDQVRNERVRQLLTSDRIVMAAASPLTPTMVEKSDGRKGVVMEAGKPQVQELKLKILEILQREGKSLVALNSMLYADEVNEQVVQRKMEIRDRQANRIIWNGVMAKALAIALNPVTVVDIVTAATVDVVLLLSLSRLYGIPMTQTGAIDLLQKIALSMGGISASELLANLGLSGLKSLLGVATPVTGGVAFGGYVSVAITQASVAGVSTYAIGQVAKTYLANGASWGPEGPKVVVQQILDSLDETSILSQIKEELQQKLVG</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DUF697 domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DUF697 domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>98</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>97</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>protein of unknown function containing DUF697 domain</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DUF697 domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>95</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>DUF697 domain-containing protein</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>83</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>GTP-binding protein</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>70</td>
            <td id=cg>98</td>
            <td id=cg>96</td>
            <td id=al>GTP-binding protein</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>GTP-binding protein</td>
            <td id=co>Hassallia</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>34</td>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=al>cd00880 E. coli Ras-like protein (Era)-like GTPase. The Era (E. coli Ras-like protein)-like family includes several distinct subfamilies (TrmE/ThdF, FeoB, YihA (EngB), Era, and EngA/YfgK) that generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. TrmE is ubiquitous in bacteria and is a widespread mitochondrial protein in eukaryotes, but is absent from archaea. The yeast member of TrmE family, MSS1, is involved in mitochondrial translation; bacterial members are often present in translation-related operons. FeoB represents an unusual adaptation of GTPases for high-affinity iron (II) transport. YihA (EngB) family of GTPases is typified by the E. coli YihA, which is an essential protein involved in cell division control. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain. EngA and its orthologs are composed of two GTPase domains and, since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=al>pfam05128 Domain of unknown function (DUF697). Family of bacterial hypothetical proteins that is sometimes associated with GTPase domains.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>31</td>
            <td id=cg>93</td>
            <td id=al>cd00882 Rat sarcoma (Ras)-like superfamily of small guanosine triphosphatases (GTPases). Ras-like GTPase superfamily. The Ras-like superfamily of small GTPases consists of several families with an extremely high degree of structural and functional similarity. The Ras superfamily is divided into at least four families in eukaryotes: the Ras, Rho, Rab, and Sar1/Arf families. This superfamily also includes proteins like the GTP translation factors, Era-like GTPases, and G-alpha chain of the heterotrimeric G proteins. Members of the Ras superfamily regulate a wide variety of cellular functions: the Ras family regulates gene expression, the Rho family regulates cytoskeletal reorganization and gene expression, the Rab and Sar1/Arf families regulate vesicle trafficking, and the Ran family regulates nucleocytoplasmic transport and microtubule organization. The GTP translation factor family regulates initiation, elongation, termination, and release in translation, and the Era-like GTPase family regulates cell division, sporulation, and DNA replication. Members of the Ras superfamily are identified by the GTP binding site, which is made up of five characteristic sequence motifs, and the switch I and switch II regions.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>32</td>
            <td id=cg>86</td>
            <td id=al>cd09912 Dynamin-like protein including dynamins, mitofusins, and guanylate-binding proteins. The dynamin family of large mechanochemical GTPases includes the classical dynamins and dynamin-like proteins (DLPs) that are found throughout the Eukarya. This family also includes bacterial DLPs. These proteins catalyze membrane fission during clathrin-mediated endocytosis. Dynamin consists of five domains; an N-terminal G domain that binds and hydrolyzes GTP, a middle domain (MD) involved in self-assembly and oligomerization, a pleckstrin homology (PH) domain responsible for interactions with the plasma membrane, GED, which is also involved in self-assembly, and a proline arginine rich domain (PRD) that interacts with SH3 domains on accessory proteins. To date, three vertebrate dynamin genes have been identified; dynamin 1, which is brain specific, mediates uptake of synaptic vesicles in presynaptic terminals; dynamin-2 is expressed ubiquitously and similarly participates in membrane fission; mutations in the MD, PH and GED domains of dynamin 2 have been linked to human diseases such as Charcot-Marie-Tooth peripheral neuropathy and rare forms of centronuclear myopathy. Dynamin 3 participates in megakaryocyte progenitor amplification, and is also involved in cytoplasmic enlargement and the formation of the demarcation membrane system. This family also includes mitofusins (MFN1 and MFN2 in mammals) that are involved in mitochondrial fusion. Dynamin oligomerizes into helical structures around the neck of budding vesicles in a GTP hydrolysis-dependent manner.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>34</td>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=al>pfam01926 50S ribosome-binding GTPase. The full-length GTPase protein is required for the complete activity of the protein of interacting with the 50S ribosome and binding of both adenine and guanine nucleotides, with a preference for guanine nucleotide.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=co>46</td>
            <td id=cg>100</td>
            <td id=al>COG1100 GTPase SAR1 family domain [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cr>30</td>
            <td id=cg>90</td>
            <td id=al>cd04164 trmE is a tRNA modification GTPase. TrmE (MnmE, ThdF, MSS1) is a 3-domain protein found in bacteria and eukaryotes. It controls modification of the uridine at the wobble position (U34) of tRNAs that read codons ending with A or G in the mixed codon family boxes. TrmE contains a GTPase domain that forms a canonical Ras-like fold. It functions a molecular switch GTPase, and apparently uses a conformational change associated with GTP hydrolysis to promote the tRNA modification reaction, in which the conserved cysteine in the C-terminal domain is thought to function as a catalytic residue. In bacteria that are able to survive in extremely low pH conditions, TrmE regulates glutamate-dependent acid resistance.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>28</td>
            <td id=cg>96</td>
            <td id=al>cd11383 YfjP GTPase. The Era (E. coli Ras-like protein)-like YfjP subfamily includes several uncharacterized bacterial GTPases that are similar to Era. They generally show sequence conservation in the region between the Walker A and B motifs (G1 and G3 box motifs), to the exclusion of other GTPases. Era is characterized by a distinct derivative of the KH domain (the pseudo-KH domain) which is located C-terminal to the GTPase domain.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>28</td>
            <td id=cr>30</td>
            <td id=al>COG1160 Predicted GTPases [General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>35</td>
            <td id=co>56</td>
            <td id=al>COG1159 GTPase Era, involved in 16S rRNA processing [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>29</td>
            <td id=cg>83</td>
            <td id=al>cd04163 E. coli Ras-like protein (Era) is a multifunctional GTPase. Era (E. coli Ras-like protein) is a multifunctional GTPase found in all bacteria except some eubacteria. It binds to the 16S ribosomal RNA (rRNA) of the 30S subunit and appears to play a role in the assembly of the 30S subunit, possibly by chaperoning the 16S rRNA. It also contacts several assembly elements of the 30S subunit. Era couples cell growth with cytokinesis and plays a role in cell division and energy metabolism. Homologs have also been found in eukaryotes. Era contains two domains: the N-terminal GTPase domain and a C-terminal domain KH domain that is critical for RNA binding. Both domains are important for Era function. Era is functionally able to compensate for deletion of RbfA, a cold-shock adaptation protein that is required for efficient processing of the 16S rRNA.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=co>45</td>
            <td id=co>48</td>
            <td id=al>PRK05291 tRNA uridine-5-carboxymethylaminomethyl(34) synthesis GTPase MnmE. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>37</td>
            <td id=co>40</td>
            <td id=al>PRK00093 GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cr>28</td>
            <td id=cr>31</td>
            <td id=al>PRK00093 GTP-binding protein Der; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cr>31</td>
            <td id=cg>94</td>
            <td id=al>cd01894 EngA1 GTPase contains the first domain of EngA. This EngA1 subfamily CD represents the first GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cr>37</td>
            <td id=cr>39</td>
            <td id=al>COG0486 tRNA U34 5-carboxymethylaminomethyl modifying GTPase MnmE/TrmE [Translation, ribosomal structure and biogenesis]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cr>31</td>
            <td id=cg>85</td>
            <td id=al>cd01895 EngA2 GTPase contains the second domain of EngA. This EngA2 subfamily CD represents the second GTPase domain of EngA and its orthologs, which are composed of two adjacent GTPase domains. Since the sequences of the two domains are more similar to each other than to other GTPases, it is likely that an ancient gene duplication, rather than a fusion of evolutionarily distinct GTPases, gave rise to this family. Although the exact function of these proteins has not been elucidated, studies have revealed that the E. coli EngA homolog, Der, and Neisseria gonorrhoeae EngA are essential for cell viability. A recent report suggests that E. coli Der functions in ribosome assembly and stability.</td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>1133457</td></tr>
<tr><td id=al>end</td><td id=al>1134888</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>DUF697 domain-containing protein</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>477</td></tr>
</table>

    </div>
</body>
</html>
