<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.00981</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.00981 LL-diaminopimelate aminotransferase [Phormidium_A]</h3>
    <div>MLHFAQRLTSLRSNVFADMDRAKAAARQAGREIIDLSLGSSDLPTADFILQEIERSLWDSQTHQYLLFHGTKAFREAAADWYQRRYQIEVNPETEVLPLVGSQEGTAHLPLALLNPGDYALLLDPGYPSHYGGVALAGGQIHRLPLRAEREFLPDFSEIPPQVCDRARMMVLNYPHNPTTASASISFFKEALTFCDKYNVALVHDFPYGDLVFDGHCPPSILQVDVERRRSIEFFTLSKSYNMGGFRVGYAIGNRELILALRQVKAAVDFNQYQGIFNGAIAALNSDSKAVQATVDCFRERRDAFVDAMKTIGWAVPTPQATMYVWAQLPEPWRGDSIGFCEALVQETGVAVSPGVGFGESGEGYVRFALVRSHQVLTTAVQRISVFLRESSATS</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>94</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>90</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>acetylornithine/N-succinyldiaminopimelate aminotransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>86</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>96</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LL-diaminopimelate aminotransferase, partial</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>71</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=cb>67</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=co>Stanieria</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cr>PCC-7407</td>
            </tr><tr>
            <td id=cb>69</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cr>JAAUSV01</td>
            </tr><tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>LL-diaminopimelate aminotransferase</td>
            <td id=cb>RDYZ01</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cb>68</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>LL-diaminopimelate aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>43</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>45</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>LL-diaminopimelate aminotransferase. This clade of the pfam00155 superfamily of aminotransferases includes several which are adjacent to elements of the lysine biosynthesis via diaminopimelate pathway (GenProp0125). Every member of this clade is from a genome which possesses most of the lysine biosynthesis pathway but lacks any of the known aminotransferases, succinylases, desuccinylases, acetylases or deacetylases typical of the acylated versions of this pathway nor do they have the direct, NADPH-dependent enzyme (ddh). Although there is no experimental characterization of any of the sequences in this clade, a direct pathway is known in plants and Chlamydia and the clade containing the Chlamydia gene is a neighboring one in the same pfam00155 superfamily so it seems quite reasonable that these enzymes catalyze the same transformation.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>40</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>transaminase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>LL-diaminopimelate aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=co>42</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>aspartate aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>Aspartate/methionine/tyrosine aminotransferase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>aminotransferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>LL-diaminopimelate aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>38</td>
            <td id=cg>100</td>
            <td id=cg>95</td>
            <td id=al>LL-diaminopimelate aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>32</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Aspartate aminotransferase family. This family belongs to pyridoxal phosphate (PLP)-dependent aspartate aminotransferase superfamily (fold I). Pyridoxal phosphate combines with an alpha-amino acid to form a compound called a Schiff base or aldimine intermediate, which depending on the reaction, is the substrate in four kinds of reactions (1) transamination (movement of amino groups), (2) racemization (redistribution of enantiomers), (3) decarboxylation (removing COOH groups), and (4) various side-chain reactions depending on the enzyme involved. Pyridoxal phosphate (PLP) dependent enzymes were previously classified into alpha, beta and gamma classes, based on the chemical characteristics (carbon atom involved) of the reaction they catalyzed. The availability of several structures allowed a comprehensive analysis of  the evolutionary classification of PLP dependent enzymes, and it was found that the functional classification did not always agree with the evolutionary history of these enzymes. The major groups in this CD corresponds to Aspartate aminotransferase a, b and c, Tyrosine, Alanine, Aromatic-amino-acid, Glutamine phenylpyruvate, 1-Aminocyclopropane-1-carboxylate synthase, Histidinol-phosphate, gene products of malY and cobC, Valine-pyruvate aminotransferase and Rhizopine catabolism regulatory protein.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>LL-diaminopimelate aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>succinyldiaminopimelate transaminase. The four sequences which make up the seed for this model are not closely related, although they are all members of the pfam00155 family of aminotransferases and are more closely related to each other than to anything else. Additionally, all of them are found in the vicinity of genes involved in the biosynthesis of lysine via the diaminopimelate pathway (GenProp0125), although this amount to a separation of 12 genes in the case of Sulfurihydrogenibium azorense Az-Fu1. None of these genomes contain another strong candidate for this role in the pathway. Note: the detailed information included in the EC:2.6.1.17 record includes the assertions that the enzyme uses the pyridoxal pyrophosphate cofactor, which is consistent with the pfam00155 family, and the assertion that the amino group donor is L-glutamate, which is undetermined for the sequences in this clade.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aspartate aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Aminotransferase class I and II. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>putative succinyldiaminopimelate transaminase DapC. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>aspartate aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>N-succinyldiaminopimelate aminotransferase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>succinyldiaminopimelate transaminase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>aminotransferase family protein; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>aminotransferase A; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>28</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>Histidinol-phosphate/aromatic aminotransferase or cobyric acid decarboxylase [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>alanine aminotransferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>succinyldiaminopimelate transaminase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>methionine aminotransferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>83</td>
            <td id=al>DNA-binding transcriptional regulator, MocR family, contains an aminotransferase domain [Transcription, Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>91</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>26</td>
            <td id=cg>100</td>
            <td id=cg>97</td>
            <td id=al>Bifunctional PLP-dependent enzyme with beta-cystathionase and maltose regulon repressor activities [Amino acid transport and metabolism, General function prediction only]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>aspartate transaminase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>84</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>88</td>
            <td id=al>transaminase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>22</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>tyrosine/nicotianamine family aminotransferase. This subfamily of pyridoxal phosphate-dependent enzymes includes known examples of both tyrosine aminotransferase from animals and nicotianamine aminotransferase from barley.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>tyrosine aminotransferase, eukaryotic. This model describes tyrosine aminotransferase as found in animals and Trypanosoma cruzi. It is the first enzyme of a pathway of tyrosine degradation via homogentisate. Several plant enzyme designated as probable tyrosine aminotransferases are very closely related to an experimentally demonstrated nicotianamine aminotransferase, an enzyme in a siderophore (iron uptake chelator) biosynthesis pathway. These plant sequences are excluded from the model seed and score between the trusted an noise cutoffs. [Energy metabolism, Amino acids and amines]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>93</td>
            <td id=al>aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>25</td>
            <td id=cg>100</td>
            <td id=cg>99</td>
            <td id=al>L,L-diaminopimelate aminotransferase; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>94</td>
            <td id=al>tyrosine aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>61</td>
            <td id=al>putative aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>100</td>
            <td id=cb>66</td>
            <td id=al>putative aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>80</td>
            <td id=al>1-aminocyclopropane-1-carboxylate synthase</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=co>57</td>
            <td id=al>bifunctional HTH-domain containing protein/aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>29</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=co>55</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>aminotransferase AlaT; Validated</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cg>92</td>
            <td id=al>threonine-phosphate decarboxylase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>21</td>
            <td id=cg>100</td>
            <td id=cb>60</td>
            <td id=al>beta-methylarginine biosynthesis bifunctional aminotransferase. Members of this family are the bifunctional aminotransferase that catalyzes the first and third steps in the three-step conversion of arginine to beta-methylarginine. It first converts arginine to 2-ketoarginine, then converts 3-methyl-2-ketoarginine to 3-methylarginine. All members of the seed alignment are encoded next to a 2-ketoarginine methyltransferase (EC 2.1.1.243).</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>100</td>
            <td id=cg>85</td>
            <td id=al>enduracididine biosynthesis enzyme MppQ. Members of this family are MppQ, one of three enzymes involved in synthesizing enduracididine, a non-proteinogenic amino acid used in non-ribosomal peptide synthases to make natural products such as enduracidin from Streptomyces fungicidicus ATCC 21013. MppQ is a PLP-dependent enzyme, predicted by homology to be an aminotransferase.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>36</td>
            <td id=cg>100</td>
            <td id=cr>28</td>
            <td id=al>histidinol-phosphate aminotransferase; Provisional</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=co>54</td>
            <td id=al>pyridoxal phosphate-dependent aminotransferase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>23</td>
            <td id=cg>100</td>
            <td id=cb>72</td>
            <td id=al>histidinol-phosphate transaminase. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>27</td>
            <td id=cg>100</td>
            <td id=cb>71</td>
            <td id=al>histidinol-phosphate transaminase. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>801693</td></tr>
<tr><td id=al>end</td><td id=al>802881</td></tr>
<tr><td id=al>strand</td><td id=al>-1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>LL-diaminopimelate aminotransferase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
</table>

    </div>
</body>
</html>
