<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>pha.pha.04332</title>
</head>
<body>
    <style>
        div {font-family: Calibri, sans-serif; word-wrap: break-word;}
        h3 {text-align: left; font-family: Calibri, sans-serif; margin: 10px 0px 0 px 0px;}
        #al {text-align: left;}
        #cg {color: green;}
        #cr {color: red;}
        #cb {color: blue;}
        #co {color: orange;}
        th {
          background-color: white; margin: 0px 10px 0 px 0px;
            }
        #al {
          text-align: left;
            }
        #f {
          font-family: Calibri, sans-serif;
          text-align: center;
          padding: 0px;
          margin: 0px;
           }
    </style>
    <h3>>pha.pha.04332 HAD-IB family phosphatase [Phormidium_A]</h3>
    <div>MSPIVFCDFDGTITLEDTFISTVERFAPQAASRVLPQIYNLSLSLQEGVHQMLEAIPSACYPEIIKLSTTYEIRPGFVELLDFLDERHVPLVVVSGGIQGVIEGVLGDLIQRVRGIHAVKLQTHGTYCSPYSDFEGEGEMVNKLAVIEHYGEPDWILIGDSVTDLKAALAAPRVFARDRLCRYLTERGVTYDSFETFTQVRDRLAQVWR</div>
    <div id=f>
    <h3>Top Blast Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cg>93</td>
            <td id=cg>99</td>
            <td id=cg>100</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>87</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>88</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>2-hydroxy-3-keto-5-methylthiopentenyl-1-phosphate phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>96</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cg>84</td>
            <td id=cg>100</td>
            <td id=cg>89</td>
            <td id=al>2-hydroxy-3-keto-5-methylthiopentenyl-1-phosphate phosphatase MtnX</td>
            <td id=cg>Phormidium_A</td>
            </tr><tr>
            <td id=cb>60</td>
            <td id=cg>97</td>
            <td id=cg>96</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=cb>Geitlerinema</td>
            </tr><tr>
            <td id=co>56</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=cb>LEGE-11467</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>99</td>
            <td id=cg>98</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=co>Fischerella</td>
            </tr><tr>
            <td id=co>52</td>
            <td id=cg>99</td>
            <td id=cg>98</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=co>Fischerella</td>
            </tr><tr>
            <td id=co>53</td>
            <td id=cg>99</td>
            <td id=cg>99</td>
            <td id=al>HAD-IB family phosphatase</td>
            <td id=co>Limnospira</td>
            </tr></tbody>
</table>

    <h3>Top Conserved Domain Database Hits</h3>
    <table><thead><tr> <th>percent id</th>
<th>query align</th>
<th>hit align</th>
<th id=al>description</th>
<th>taxon</th>
</tr><thead>
<tbody>
<tr>
            <td id=cr>39</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>cd07524 phosphatase, similar to Bacillus subtilis MtnX; belongs to the haloacid dehalogenase-like superfamily. Bacillus subtilis recycles two toxic byproducts of polyamine metabolism, methylthioadenosine and methylthioribose, into methionine by a salvage pathway. The sixth reaction in this pathway is catalyzed by B. subtilis MtnX: the dephosphorylation of 2- hydroxy-3-keto-5-methylthiopentenyl-1-phosphate (HKMTP- 1-P) into 1,2-dihydroxy-3-keto-5-methylthiopentene. The hydrolysis of HK-MTP-1-P is a two-step mechanism involving the formation of a transiently phosphorylated aspartyl intermediate. Members of this family belong to the haloacid dehalogenase-like (HAD) hydrolases, a large superfamily of diverse enzymes that catalyze carbon or phosphoryl group transfer reactions on a range of substrates, using an active site aspartate in nucleophilic catalysis. Members of this superfamily include 2-L-haloalkanoic acid dehalogenase, azetidine hydrolase, phosphonoacetaldehyde hydrolase, phosphoserine phosphatase, phosphomannomutase, P-type ATPases and many others. HAD hydrolases are found in all three kingdoms of life, and most genomes are predicted to contain multiple HAD-like proteins. Members possess a highly conserved alpha/beta core domain, and many also possess a small cap domain, the fold and function of which is variable. HAD hydrolases are sometimes referred to as belonging to the DDDD superfamily of phosphohydrolases.</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>37</td>
            <td id=cg>100</td>
            <td id=cg>100</td>
            <td id=al>PRK09552 2-hydroxy-3-keto-5-methylthiopentenyl-1-phosphate phosphatase; Reviewed</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>31</td>
            <td id=cg>100</td>
            <td id=cg>98</td>
            <td id=al>COG4359 2-hydroxy-3-keto-5-methylthiopentenyl-1-phosphate phosphatase (methionine salvage) [Amino acid transport and metabolism]. </td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>30</td>
            <td id=cg>92</td>
            <td id=cg>100</td>
            <td id=al>TIGR01489 2,3-diketo-5-methylthio-1-phosphopentane phosphatase. This phosphatase is a member of the IB subfamily (TIGR01488) of the haloacid dehalogenase (HAD) superfamily of aspartate-nucleophile hydrolases. With the exception of OMNI|NTL01BS01361 from B. subtilis and GP|15024582 from Clostridium acetabutylicum, the members of this group are all eukaryotic, spanning metazoa, plants and fungi. The B. subtilus gene (YkrX, renamed MtnX) is part of an operon for the conversion of methylthioribose (MTR) to methionine. It works with the enolase MtnW, a RuBisCO homolog. The combination of MtnW and MtnX achieves the same overall reaction as the enolase-phosphatase MtnC. The function of MtnX was shown by Ashida, et al. (2003) to be 2,3-diketo-5-methylthio-1-phosphopentane phosphatase, rather than 2,3-diketo-5-methylthio-1-phosphopentane phosphatase as proposed earlier. See the Genome Property for methionine salvage for more details. In eukaryotes, methionine salvage from methylthioadenosine also occurs. It seems reasonable that members of this family in eukaryotes fulfill a similar role as in Bacillus. A more specific, equivalog-level model is TIGR03333. Note that SP|P53981 from S. cerevisiae, a member of this family, is annotated as a "probable membrane protein" due to a predicted transmembrane helix. The region in question contains the second of the three conserved HAD superfamily catalytic motifs and thus, considering the fold of the HAD catalytic domain, is unlikely to be a transmembrane region in fact. [Central intermediary metabolism, Other]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>24</td>
            <td id=cg>85</td>
            <td id=cg>100</td>
            <td id=al>TIGR01488 Haloacid Dehalogenase superfamily, subfamily IB, phosphoserine phosphatase-like. This model represents a subfamily of the Haloacid Dehalogenase superfamily of aspartate-nucleophile hydrolases. Subfamily IA, B, C and D are distinguished from the rest of the superfamily by the presence of a variable domain between the first and second conserved catalytic motifs. In subfamilies IA and IB, this domain consists of an alpha-helical bundle. It was necessary to model these two subfamilies separately, breaking them at a an apparent phylogenetic bifurcation, so that the resulting model(s) are not so broadly defined that members of subfamily III (which lack the variable domain) are included. Subfamily IA includes the enzyme phosphoserine phosphatase (TIGR00338) as well as three hypothetical equivalogs. Many members of these hypothetical equivalogs have been annotated as PSPase-like or PSPase-family proteins. In particular, the hypothetical equivalog which appears to be most closely related to PSPase contains only Archaea (while TIGR00338 contains only eukaryotes and bacteria) of which some are annotated as PSPases. Although this is a reasonable conjecture, none of these sequences has sufficient evidence for this assignment. If such should be found, this model should be retired while the PSPase model should be broadened to include these sequences. [Unknown function, Enzymes of unknown specificity]</td>
            <td id=cr></td>
            </tr><tr>
            <td id=cr>20</td>
            <td id=cg>88</td>
            <td id=cg>99</td>
            <td id=al>pfam12710 haloacid dehalogenase-like hydrolase. </td>
            <td id=cr></td>
            </tr></tbody>
</table>

    <h3>Other attributes</h3>
    <table>
<tr><td id=al>start</td><td id=al>3500436</td></tr>
<tr><td id=al>end</td><td id=al>3501066</td></tr>
<tr><td id=al>strand</td><td id=al>1</td></tr>
<tr><td id=al>type</td><td id=al>CDS</td></tr>
<tr><td id=al>inference</td><td id=al>prodigal</td></tr>
<tr><td id=al>subsystems</td><td id=al></td></tr>
<tr><td id=al>descr</td><td id=al>HAD-IB family phosphatase</td></tr>
<tr><td id=al>taxon</td><td id=al>Bacteria; Cyanobacteria; Cyanobacteriia; Cyanobacteriales; Geitlerinemaceae; Phormidium_A; Phormidium_A</td></tr>
<tr><td id=al>notes</td><td id=al></td></tr>
<tr><td id=al>antismash</td><td id=al></td></tr>
<tr><td id=al>signal_peptide</td><td id=al></td></tr>
<tr><td id=al>tmh</td><td id=al>0</td></tr>
<tr><td id=al>tmh_topology</td><td id=al></td></tr>
<tr><td id=al>length</td><td id=al>210</td></tr>
</table>

    </div>
</body>
</html>
